## Installation only needs to happen once.
# install.packages(c("knitr", "tidyverse"))

library(knitr)
library(tidyverse)

unlink("Tables", recursive = TRUE)
dir.create("Tables", showWarnings = FALSE)

assignment = c("09 - Strings", "14 - Regular Expressions 1", "15 - Regular Expressions 2", "19 - Additional Practice", "19 - Additional Practice")
exercise = c("05 - Is CpG island", "06 - Find words that start with a vowel", "06 - Switch column order", "08 - Make inducible promoter - Part B", "10 - Make inducible promoter - Part D")
#core_prompt = c("The human genome consists of 3 billion A, C, G, and T nucleotides. Some parts of the genome are known as 'promoter' regions. These regions typically lie upstream of genes and regulate gene expression. Promoter regions typically contain 'CpG islands,' which have a relatively high number of CG pairs (a C nucleotide followed by a G nucleotide). If 10% or more of the DNA in a region consists of these CG pairs, that region may be classified as a 'CpG' island. Please write a function called 'isCpgIsland' that accepts a single argument: a string. This string will contain a DNA sequence. Your function should calculate the percentage of nucleotides in the DNA sequence that are CG pairs. For example, if the DNA sequence were ACGACGAATT, the percentage would be 40. Return a Boolean value that indicates whether the DNA sequence is a CpG island (*greater than* 10%).", "Please write a function called 'findVowelStartWords' that accepts two arguments. The first argument will be the name of a text file that contains part of a classic biology paper. The second argument will be the name of a text file that your function will need to create. Your function should read the specified input file and use the 're' module to identify any multi-letter words that start with an upper-case vowel. Your function should write each of these words to the output file on a separate line (not including any white space before or after the word). For simplicity, focus on words that are preceded and followed by a space character (don't worry about words at the beginning of a line or that are followed by a punctuation mark).", "Suppose you are running a service to match hospital patients to potential blood donors. You have created a tab-delimited text file that indicates a person's name in the first column and their blood type in the second column. When you encounter a new patient at the hospital who needs a blood transfusion, you want to be able to search for people who match that person's blood type. As you have begin building that text file, you realize that you made a mistake. You have listed each person's name first and then their blood type. To make it easier for you to search, you would like to have the blood type first and then their name. Please write a function called 'switchColumnOrder' that accepts two arguments: 1) the name of an input file and 2) the name of a new file that your function will create. Your function should read the specified input file, switch the columns, and write the output to the new file.", "Now that you can identify translation initiation sequence, the next step is to identify the location of a stop codon within the gene. A stop codon ('TAG', 'TAA', or 'TGA') indicates the location where translation should stop. Please write a Python script that accepts one argument via 'sys.argv': the name of a FASTA file. Your script should read the FASTA file, three nucleotides at a time, and identify the first in-frame occurrence of start codon. Then starting from that location, your code should read the DNA sequence, three nucleotides at a time, and identify the first in-frame occurrence of a stop codon. Your script should print the start location first and then the stop location (on a separate line). Important: Ignore any stop sequences that are not in frame.", "Now the goal will be to identify any location(s) upstream of the gene where a restriction enzyme can bind. But you also want to ensure that the restriction enzyme will not bind to a location within the gene itself. Please write a Python script that accepts one argument via 'sys.argv': the name of a FASTA file. As in the previous exercise, your code should use information in 'RestrictionEnzymes.txt' and 'Placeholders.txt' to determine a DNA pattern associated with each enzyme. Your script should read the FASTA file and extract the upstream sequence and the gene sequence. For each restriction enzyme, your code should identify any restriction enzyme that that matches the upstream sequence and does not match the gene sequence. Your script should print the name of any restriction enzyme that meets these criteria. It should also print the sequence from the upstream DNA that matches the restriction enzyme. If there is no matching sequence in the upstream DNA, print 'No match' for that enzyme.")
prompt_summary = c("Count the proportion of a DNA sequence that consists of 'CG' nucleotide pairs", "Find words in a biological text that begin with vowels", "Switch the first two columns in a tab-delieted text file", "Identify in-frame start and stop codons in an mRNA sequence", "Identify restriction-enzyme binding sites upstream of a gene in a DNA sequence")
#output_type = c("text", "text", "text", "text", "text")
#bio_oriented = c("Yes", "Yes", "No", "Yes", "Yes")
#num_tests = c(3, 1, 1, 5, 5)
#included_data_file = c("No", "Yes", "Yes", "Yes", "Yes")
skills_emphasized = c("Manipulating strings; performing mathematical calculations", "Writing regular expressions; reading files", "Writing regular expressions or using lists; reading files; writing files", "Manipulating strings; reading files; using complex iteration logic", "Reading files; writing regular expressions; using complex iteration logic; using lists; using dictionaries; using conditionals")
failure_description = c("Failed on one test representing an edge case (exactly 10% of the sequence were CG pairs)", "Trouble dealing with extra spaces or punctuation marks", "Runtime errors, logic errors dealing with + or - portion of blood types", "Failed to follow the instruction to look for the start codon in frame", "Runtime errors, various logic errors, failure to fully comprehend the prompt")

#tbl = tibble(`Assignment`=assignment, `Exercise`=exercise, `Prompt summary`=prompt_summary, `Output type`=output_type, `Biology oriented`=bio_oriented, `Used data file(s)`=included_data_file, `Skills emphasized`=skills_emphasized, `Failure summary`=failure_description)
tbl = tibble(`Assignment`=assignment, `Exercise`=exercise, `Prompt summary`=prompt_summary, `Skills emphasized`=skills_emphasized, `Failure summary`=failure_description)

tbl %>%
  kable(format="simple") %>%
  write("Tables/Failure_Summary.md")
