<meta http-equiv="Content-Type" content="text/html; charset=utf-8"><link type="text/css" rel="stylesheet" href="resources/sheet.css" >
<style type="text/css">.ritz .waffle a { color: inherit; }.ritz .waffle .s4{background-color:#f4cccc;text-align:left;color:#000000;font-family:'Courier New';font-size:11pt;vertical-align:top;white-space:normal;overflow:hidden;word-wrap:break-word;direction:ltr;padding:0px 3px 0px 3px;}.ritz .waffle .s0{background-color:#ffffff;text-align:left;font-weight:bold;color:#000000;font-family:'Courier New';font-size:11pt;vertical-align:top;white-space:normal;overflow:hidden;word-wrap:break-word;direction:ltr;padding:0px 3px 0px 3px;}.ritz .waffle .s1{background-color:#ffffff;text-align:left;color:#000000;font-family:'Courier New';font-size:11pt;vertical-align:top;white-space:normal;overflow:hidden;word-wrap:break-word;direction:ltr;padding:0px 3px 0px 3px;}.ritz .waffle .s3{background-color:#b6d7a8;text-align:left;color:#000000;font-family:'Courier New';font-size:11pt;vertical-align:top;white-space:normal;overflow:hidden;word-wrap:break-word;direction:ltr;padding:0px 3px 0px 3px;}.ritz .waffle .s5{background-color:#f4cccc;text-align:right;color:#000000;font-family:'Courier New';font-size:11pt;vertical-align:top;white-space:normal;overflow:hidden;word-wrap:break-word;direction:ltr;padding:0px 3px 0px 3px;}.ritz .waffle .s2{background-color:#ffffff;text-align:right;color:#000000;font-family:'Courier New';font-size:11pt;vertical-align:top;white-space:normal;overflow:hidden;word-wrap:break-word;direction:ltr;padding:0px 3px 0px 3px;}</style><div class="ritz grid-container" dir="ltr"><table class="waffle" cellspacing="0" cellpadding="0"><thead><tr><th class="row-header"></th><th id="2089702959C0" style="width:155px;" class="column-headers-background">A</th><th id="2089702959C1" style="width:160px;" class="column-headers-background">B</th><th class="freezebar-cell frozen-column-cell freezebar-vertical-handle"></th><th id="2089702959C2" style="width:723px;" class="column-headers-background">C</th><th id="2089702959C3" style="width:141px;" class="column-headers-background">D</th><th id="2089702959C4" style="width:141px;" class="column-headers-background">E</th><th id="2089702959C5" style="width:141px;" class="column-headers-background">F</th><th id="2089702959C6" style="width:79px;" class="column-headers-background">G</th><th id="2089702959C7" style="width:102px;" class="column-headers-background">H</th><th id="2089702959C8" style="width:101px;" class="column-headers-background">I</th><th id="2089702959C9" style="width:105px;" class="column-headers-background">J</th><th id="2089702959C10" style="width:694px;" class="column-headers-background">K</th><th id="2089702959C11" style="width:83px;" class="column-headers-background">L</th><th id="2089702959C12" style="width:123px;" class="column-headers-background">M</th><th id="2089702959C13" style="width:141px;" class="column-headers-background">N</th><th id="2089702959C14" style="width:530px;" class="column-headers-background">O</th><th id="2089702959C15" style="width:140px;" class="column-headers-background">P</th><th id="2089702959C16" style="width:122px;" class="column-headers-background">Q</th><th id="2089702959C17" style="width:122px;" class="column-headers-background">R</th><th id="2089702959C18" style="width:122px;" class="column-headers-background">S</th><th id="2089702959C19" style="width:122px;" class="column-headers-background">T</th><th id="2089702959C20" style="width:122px;" class="column-headers-background">U</th><th id="2089702959C21" style="width:122px;" class="column-headers-background">V</th><th id="2089702959C22" style="width:151px;" class="column-headers-background">W</th><th id="2089702959C23" style="width:201px;" class="column-headers-background">X</th><th id="2089702959C24" style="width:151px;" class="column-headers-background">Y</th><th id="2089702959C25" style="width:151px;" class="column-headers-background">Z</th></tr></thead><tbody><tr style="height: 19px"><th id="2089702959R0" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">1</div></th><td class="s0">Assignment</td><td class="s0">Exercise</td><td class="freezebar-cell"></td><td class="s0">Instructions</td><td class="s0" dir="ltr">Credit</td><td class="s0" dir="ltr">Is biology oriented (words)?</td><td class="s0" dir="ltr">Is biology oriented (context)?</td><td class="s0">Output type</td><td class="s0">Who attempted</td><td class="s0">Date attempted</td><td class="s0">ChatGPT version</td><td class="s0">ChatGPT solution</td><td class="s0">Number of tests</td><td class="s0">Number of passed tests</td><td class="s0">Number of ChatGPT iterations</td><td class="s0">Comments</td><td class="s0" dir="ltr">Minor differences in output</td><td class="s0" dir="ltr">At least one logic error</td><td class="s0" dir="ltr">At least one exception occurred</td><td class="s0" dir="ltr">Passed but missed the spirit of the prompt</td><td class="s0" dir="ltr">At least one data file</td><td class="s0" dir="ltr">Prompt too long</td><td class="s0" dir="ltr">We tried a shortened version of the prompt</td><td class="s0" dir="ltr">We clarified the prompt as a result of the ChatGPT interaction</td><td class="s0" dir="ltr">We asked ChatGPT to try again using the original prompt</td><td class="s0" dir="ltr">ChatGPT used technique not covered at this point</td><td class="s0" dir="ltr">ChatGPT provided a solution that was off base</td></tr><tr><th style="height:3px;" class="freezebar-cell freezebar-horizontal-handle"></th><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td><td class="freezebar-cell"></td></tr><tr style="height: 19px"><th id="2089702959R1" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">2</div></th><td class="s1">01 - Declaring and Converting Variables</td><td class="s1">01 - Declare and print a variable</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please declare a variable that stores a value of &quot;Hello, World!&quot;. Then print the value of this variable.<br><br>You can name the variable whatever you prefer. However, be sure to use either camel casing or underscore syntax.<br><br>Below is the expected output:<br><br>    Hello, World!</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">message = &quot;Hello, World!&quot;<br>print(message)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">It generated a version using camel casing and a version using underscore syntax.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R2" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">3</div></th><td class="s1">01 - Declaring and Converting Variables</td><td class="s1">02 - Use numeric variables</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please declare a variable that stores an integer value of &quot;2&quot; and another variable that stores a float value of the same number. Print the values of these variables (in the same order they were mentioned above). Then print the types of these variables (in the same order they were mentioned above).<br><br>Important: When declaring the variables, be sure to use proper variable names as well as camel casing or underscore syntax.<br><br>Below is the expected output:<br><br>    2<br>    2.0<br>    &lt;class &#39;int&#39;&gt;<br>    &lt;class &#39;float&#39;&gt;</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">intValue = 2<br>floatValue = 2.0<br><br>print(intValue)<br>print(floatValue)<br>print(type(intValue))<br>print(type(floatValue))</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">It generated a version using camel casing and a version using underscore syntax. </td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R3" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">4</div></th><td class="s1">01 - Declaring and Converting Variables</td><td class="s1">03 - Fix a greeting</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>The following code is supposed to indicate that Bioinformatics is exciting, but it results in an error message. Please correct the code so that it works properly.<br><br>    greeting = Bioinformatics is exciting!<br>    print greeting<br><br>Below is the expected output:<br><br>    Bioinformatics is exciting!</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">greeting = &quot;Bioinformatics is exciting!&quot;<br>print(greeting)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">It explained why the provided code was incorrect. It generated a version with double quotes and one with single quotes.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R4" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">5</div></th><td class="s1">01 - Declaring and Converting Variables</td><td class="s1">04 - Change between variable types</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please declare a variable that stores a string value of &quot;8460.2&quot;. Print the value of this variable. Then convert the value to a float and print that value. Finally, convert the float value to an integer and print that value.<br><br>Important: Be sure to use proper variable names.<br><br>Below is the expected output:<br><br>    8460.2<br>    8460.2<br>    8460</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">stringValue = &quot;8460.2&quot;<br>print(stringValue)<br><br>floatValue = float(stringValue)<br>print(floatValue)<br><br>intValue = int(floatValue)<br>print(intValue)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">It generated a version that used camel casing and one that used underscore syntax. I told it, &quot;Thank you! There&#39;s no need to provide a solution in camel casing and underscore syntax. You can just use underscore syntax.&quot; Not sure if it will remember that feedback.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R5" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">6</div></th><td class="s1">01 - Declaring and Converting Variables</td><td class="s1">05 - Concatenate a string</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Declare a variable called &quot;numLions&quot; that stores an integer value of &quot;42&quot;. Declare another variable called &quot;numTigers&quot; that stores an integer value of &quot;17&quot;. Declare yet another variable called &quot;country&quot; that stores a string value of &quot;South Africa&quot;. Print a message that indicates the following: &quot;The number of lions in South Africa is 42. The number of tigers in South Africa is 17.&quot;<br><br>Important: Your code must use these variables, and the first two variables must be declared as integers (not strings).<br><br>Below is the expected output:<br><br>    The number of lions in South Africa is 42. The number of tigers in South Africa is 17.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">numLions = 42<br>numTigers = 17<br>country = &quot;South Africa&quot;<br><br>message = &quot;The number of lions in &quot; + country + &quot; is &quot; + str(numLions) + &quot;.&quot; + &quot; The number of tigers in &quot; + country + &quot; is &quot; + str(numTigers) + &quot;.&quot;<br><br>print(message)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">3</td><td class="s1" dir="ltr">The first two solutions resulted in an extra space character before each period. After ChatGPT changed its strategy to use string concatenation, it passed.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 19px"><th id="2089702959R6" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">7</div></th><td class="s1">01 - Declaring and Converting Variables</td><td class="s1">06 - Sum different types of numbers</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please declare a variable that stores an integer value of &quot;2&quot; and another variable that stores a float value of the same number. Add these two numbers together and print the sum. Then print the type of the summed result.<br><br>Important: Be sure to use proper variable names.<br><br>Below is the expected output:<br><br>    4.0<br>    &lt;class &#39;float&#39;&gt;</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">integer_value = 2<br>float_value = 2.0<br>sum_result = integer_value + float_value<br><br>print(sum_result)<br>print(type(sum_result))</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R7" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">8</div></th><td class="s1">01 - Declaring and Converting Variables</td><td class="s1">07 - Fix a calculation</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>The following code results in an error.<br><br>    abc = 123<br>    def = 456<br>    ghi = 789<br>    print(abc + def + ghi)<br><br>First run the code as it is shown above so you can see what error message this code gives. Then fix the code so that it calculates the sum of the three integers.<br><br>Below is the expected output:<br><br>    1368</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">abc = 123<br>abc_def = 456<br>ghi = 789<br>print(abc + abc_def + ghi)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">It explained why the provided code resulted in an error.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R8" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">9</div></th><td class="s1">01 - Declaring and Converting Variables</td><td class="s1">08 - Fix multiple errors</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>The following code has some errors.<br><br>    morningTemp = 42.4<br>    afternoonTemp = 69.1<br>    nightTemp = 49.4<br><br>    print(morninTemp)<br>    print(afternonnTemp)<br>    print(niteTemp)<br><br>First run the code as it is shown above so you can see what error message the code gives. Then fix the code so that it prints the three temperatures.<br><br>Below is the expected output:<br><br>    42.4<br>    69.1<br>    49.4</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">morningTemp = 42.4<br>afternoonTemp = 69.1<br>nightTemp = 49.4<br><br>print(morningTemp)<br>print(afternoonTemp)<br>print(nightTemp)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">It explained why the provided code resulted in an error.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R9" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">10</div></th><td class="s1">01 - Declaring and Converting Variables</td><td class="s1">09 - Print some temperatures</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Consider the following code.<br><br>    morningTemp = 42.4<br><br>Write Python code that declares the &quot;morningTemp&quot; variable as shown above and then prints output like the following: &quot;The temperature in the morning was XX&quot; (your code should replace &quot;XX&quot; with the value of the variable).<br><br>Then print a statement similar to what you printed above, except this time indicate that the temperature is 51.8.<br><br>Then print a statement similar to what you printed above, except this time indicate that the temperature is 24.53123.<br><br>Important: your print statement should be exactly the same each time. You should change the value of &quot;morningTemp&quot; but not the print statement.<br><br>Below is the expected output:<br><br>    The temperature in the morning was 42.4<br>    The temperature in the morning was 51.8<br>    The temperature in the morning was 24.53123</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">morningTemp = 42.4<br>print(&quot;The temperature in the morning was&quot;, morningTemp)<br><br>morningTemp = 51.8<br>print(&quot;The temperature in the morning was&quot;, morningTemp)<br><br>morningTemp = 24.53123<br>print(&quot;The temperature in the morning was&quot;, morningTemp)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R10" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">11</div></th><td class="s1">02 - Performing Mathematical Calculations</td><td class="s1">01 - Evaluate a dark situation</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose that five turkeys enter a dark alley. Ten minutes later, thirteen turtles enter the same dark alley. Twenty minutes later, three Siamese cats enter the alley. All are having a delightful time until two rabid dogs enter the dark alley.<br><br>Create one Python variable for each type of animal and set values according to the numbers listed above. Use an expression to calculate the total number of animals in the alley. This expression must use no numbers, only variables. Store the total in a variable called &quot;totalAnimals&quot; and print its value.<br><br>Now suppose that the dogs kill three of the turkeys, eight of the turtles, and all of the cats (morbid, I know!). Use an expression to subtract the number of each type of animal that has been killed. As you write the expression, subtract the numbers that were killed from the variables you created earlier. Store this value in a variable called &quot;survivingAnimals&quot; and print its value. (This number should include the number of dogs.)</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">turkeys = 5<br>turtles = 13<br>cats = 3<br>dogs = 2<br><br>totalAnimals = turkeys + turtles + cats + dogs<br>print(totalAnimals)<br><br>turkeys -= 3<br>turtles -= 8<br>cats -= cats<br><br>survivingAnimals = turkeys + turtles + cats + dogs<br>print(survivingAnimals)</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">5</td><td class="s1" dir="ltr">The first solution tried to use a variable that did not exist. I asked it twice to address this problem, but it gave very similar solutions, and the error continued. On the fourth try, I asked ChatGPT to try a different solution. It did and almost was correct, but it included some extra wording in the output. I asked it to remove that, and it did so correctly.</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 19px"><th id="2089702959R11" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">12</div></th><td class="s1">02 - Performing Mathematical Calculations</td><td class="s1">02 - How many worms in the last box</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>An ecologist is studying the effects of ambient temperature on nematode worm development. Her strategy is to place the worms in small boxes and then to place these boxes under different levels of heat exposure in her lab. In total, she has 341 worms. Write Python code that performs the calculations described below and prints each answer.<br><br>If exactly 26 worms can fit in a single box, what is the minimum number of boxes she should buy so that there will be enough room for all the worms?<br><br>Assuming that she places 26 worms in all the boxes except the last one, how many worms will be in the last box?</td><td class="s1" dir="ltr">The data originated from https://www.programmingforbiologists.org.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">worms = 341<br>boxes = worms // 26 + (worms % 26 != 0)<br>print(boxes)<br><br>lastBoxWorms = worms % 26 or 26<br>print(lastBoxWorms)</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">It did the calculations correctly but included some additional text in the output. I asked it to correct this, and it did.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 19px"><th id="2089702959R12" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">13</div></th><td class="s1">02 - Performing Mathematical Calculations</td><td class="s1">03 - Convert from pounds to kilograms</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>The following code converts a mass in kilograms to a mass in grams and then prints out the resulting value.<br><br>    massKilograms = 2.62<br>    massGrams = massKilograms  1000<br>    print(massGrams)<br><br>Create a variable called &quot;massPounds&quot; and assign it a value of 3.5 (about the size of a Desert Cottontail Rabbit – Sylvilagus audubonii).<br><br>Then write code to convert this value to kilograms (we are serious scientists after all). There are approximately 2.2046 lbs in a kilogram, so divide &quot;massPounds&quot; by 2.2046 and store this value in a new variable called &quot;massKilogramsConverted&quot;. Print the value of &quot;massKilogramsConverted&quot; (rounded to two decimal places).</td><td class="s1" dir="ltr">The data originated from https://www.programmingforbiologists.org.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">massPounds = 3.5<br>massKilogramsConverted = massPounds / 2.2046<br>print(&quot;{:.2f}&quot;.format(massKilogramsConverted))</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">It did the calculations correctly but included some additional text in the output. I asked it to correct this, and it did.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td></tr><tr style="height: 19px"><th id="2089702959R13" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">14</div></th><td class="s1">02 - Performing Mathematical Calculations</td><td class="s1">04 - Calculate biomass</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>The goal of this exercise is to calculate the total biomass in kilograms of 3 White-throated Woodrats (Neotoma albigula). The total biomass is simply the sum of the biomass of all individuals, but in this case we only know that the average size of a single individual is 250 grams.<br><br>1. Create a variable called &quot;grams&quot; and assign it the mass of a single Neotoma albigula.<br>2. Create a variable called &quot;number&quot; and assign it the number of individuals.<br>3. Create a variable called &quot;biomass&quot; and assign it a value that is the product of &quot;grams&quot; and &quot;number&quot;.<br>4. Convert the units of &quot;biomass&quot; into kilograms (there are 1000 grams in a kilogram) and assign this value to a new variable called &quot;kilograms&quot;. The value of this variable should be a float.<br>5. Print a description of what you found. The format of the output should be &quot;The total mass is X kilograms.&quot; However, &quot;X&quot; should be replaced with the actual value.<br><br>Below is the expected output:<br><br>    The total mass is 0.75 kilograms.</td><td class="s1" dir="ltr">The data originated from https://www.programmingforbiologists.org.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">grams = 250<br>number = 3<br>biomass = grams * number<br>kilograms = biomass / 1000<br>print(&quot;The total mass is {:.2f} kilograms.&quot;.format(kilograms))</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R14" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">15</div></th><td class="s1">02 - Performing Mathematical Calculations</td><td class="s1">05 - Use built-in math functions</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Use the Python built-in functions &quot;abs()&quot;, &quot;round()&quot;, &quot;pow()&quot;, &quot;int()&quot;, &quot;float()&quot;, and &quot;str()&quot; to print an answer to each of the following. (A built-in function is one that you don’t need to import a module to use.) <br><br>1. The absolute value of -15.5.<br>2. 3.8 rounded to the nearest integer using standard rounding.<br>3. 4.483847 rounded to one decimal place.<br>4. Take the square (not the square root) of 9.99999. [Make sure to use one of the built-in functions.]<br><br>Below is the expected output:<br><br>    15.5<br>    4<br>    4.5<br>    99.99980000010001</td><td class="s1" dir="ltr">The data originated from https://www.programmingforbiologists.org.</td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">print(abs(-15.5))<br>print(round(3.8))<br>print(round(4.483847, 1))<br>print(pow(9.99999, 2))</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R15" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">16</div></th><td class="s1">02 - Performing Mathematical Calculations</td><td class="s1">06 - Calculate the area of a crop circle</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose that you measure the radius of a crop circle to be 250 feet. Write Python code that declares a variable for the radius and then calculates and prints the area of that crop circle, rounded to 2 decimal places. The format of this output should be &quot;The area of the crop circle is X.XX feet.&quot; (Replace &quot;X.XX&quot; with the actual value.) Your code must use the &quot;math&quot; module to determine the value of pi.<br><br>Hint(s):<br><br>    This page explains the formula for the area of a circle.<br><br>Below is the expected output:<br><br>    The area of the crop circle is 196349.54 feet.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import math<br><br>radius = 250<br><br>area = math.pi * radius**2<br><br>print(&quot;The area of the crop circle is {:.2f} feet.&quot;.format(area))</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">It did not need the hint (which was unavailable because the link did not copy over).</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R16" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">17</div></th><td class="s1">02 - Performing Mathematical Calculations</td><td class="s1">07 - Evaluate planter box</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Use built-in functions and/or the &quot;math&quot; module to print the answer to the following question.<br><br>Suppose you are doing research on sunflowers and have a square planter box that has an area of 82 meters. How long is one side of this box, rounded to two decimal places?</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import math<br><br>planter_area = 82<br>side_length = math.sqrt(planter_area)<br>side_length = round(side_length, 2)<br><br>print(side_length)</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R17" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">18</div></th><td class="s1">02 - Performing Mathematical Calculations</td><td class="s1">08 - Estimate number of species</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Use built-in functions and/or the &quot;math&quot; module to print the answer to the following question.<br><br>The number of species in a region can be estimated based on the area of that region using the equation &quot;species = 3.25 + 0.25  log(area)&quot;. (The &quot;log&quot; function in this equation represents the natural logarithm.)<br><br>Please write Python code that answers the following.<br><br>For an area that has a width of 8.25 meters and a length of 40.96 meters, what is the estimated number of species in that area, rounded to the nearest integer?</td><td class="s1" dir="ltr">The idea for this exercise originated from https://www.programmingforbiologists.org.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import math<br><br>width = 8.25<br>length = 40.96<br>area = width * length<br>species = 3.25 + 0.25 * math.log(area)<br>species_rounded = round(species)<br><br>print(int(species_rounded))</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R18" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">19</div></th><td class="s1">02 - Performing Mathematical Calculations</td><td class="s1">09 - Evaluate Net Primary Productivity</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>The following code calculates the total net primary productivity (NPP) per day for two ecological sites based on the grams of carbon produced per square meter per day, and the total area of the sites, and prints them out.<br><br>    site1GramsCarbonPerSquareMeterPerDay = 5<br>    site2GramsCarbonPerSquareMeterPerDay = 2.3<br><br>    site1AreaMetersSquared = 200<br>    site2AreaMetersSquared = 450<br><br>    site1NppPerDay = site1GramsCarbonPerSquareMeterPerDay  site1AreaMetersSquared <br>    site2NppPerDay = site2GramsCarbonPerSquareMeterPerDay  site2AreaMetersSquared<br><br>    print(site1NppPerDay)<br>    print(site2NppPerDay)<br><br>Modify and extend this code so that it calculates the following values and prints them in order:<br><br>1. The sum of the total daily NPP for the two sites combined.<br>2. The absolute difference between the total daily NPP for the two sites.<br>3. The total NPP for an entire year (365.25 days, due to leap year) for the two sites combined.<br><br>Here is some code to get your started:<br><br>    site1GramsCarbonPerSquareMeterPerDay = 5<br>    site2GramsCarbonPerSquareMeterPerDay = 2.3<br>    <br>    site1AreaMetersSquared = 200<br>    site2AreaMetersSquared = 450<br>    <br>    site1NppPerDay = site1GramsCarbonPerSquareMeterPerDay  site1AreaMetersSquared<br>    site2NppPerDay = site2GramsCarbonPerSquareMeterPerDay  site2AreaMetersSquared<br><br>Below is the expected output:<br><br>    2035.0<br>    35.0<br>    743283.75</td><td class="s1" dir="ltr">The data originated from https://www.programmingforbiologists.org.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">site1GramsCarbonPerSquareMeterPerDay = 5<br>site2GramsCarbonPerSquareMeterPerDay = 2.3<br><br>site1AreaMetersSquared = 200<br>site2AreaMetersSquared = 450<br><br>site1NppPerDay = site1GramsCarbonPerSquareMeterPerDay * site1AreaMetersSquared<br>site2NppPerDay = site2GramsCarbonPerSquareMeterPerDay * site2AreaMetersSquared<br><br># The sum of the total daily NPP for the two sites combined.<br>totalNppPerDay = site1NppPerDay + site2NppPerDay<br>print(totalNppPerDay)<br><br># The absolute difference between the total daily NPP for the two sites.<br>differenceNppPerDay = abs(site1NppPerDay - site2NppPerDay)<br>print(differenceNppPerDay)<br><br># The total NPP for an entire year (365.25 days, due to leap year) for the two sites combined.<br>totalNppPerYear = totalNppPerDay * 365.25<br>print(totalNppPerYear)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 19px"><th id="2089702959R19" style="height: 19px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 19px">20</div></th><td class="s1">03 - Conditional Statements</td><td class="s1">01 - Fix problematic code</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Below is a Python program that has indentation mistakes. Change the code so that it will set the &quot;title&quot; variable based on the value of the &quot;gender&quot; variable and then use the &quot;title&quot; variable to print the final message in a way that addresses the person properly. Note, you will not need to change the order of the lines, just the indentation.<br><br>Note: You can use tabs or spaces (4 spaces = tab), but make sure to be consistent on all lines.<br><br>        title = &quot;?&quot;<br>                gender = &quot;M&quot;<br><br>        if gender == &quot;M&quot;:<br>        title = &quot;Sir&quot;<br>                else:<br>        title = &quot;Ma&#39;am&quot;<br><br>                                print(&quot;Pardon me, &quot; + title + &quot;. Do you have any Grey Poupon?&quot;)<br><br>Below is the expected output:<br><br>    Pardon me, Sir. Do you have any Grey Poupon?</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">title = &quot;?&quot;<br>gender = &quot;M&quot;<br><br>if gender == &quot;M&quot;:<br>    title = &quot;Sir&quot;<br>else:<br>    title = &quot;Ma&#39;am&quot;<br><br>print(&quot;Pardon me, &quot; + title + &quot;. Do you have any Grey Poupon?&quot;)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">Initially, ChatGPT did not format its code using a code block, which would have caused a problem when copying and pasting. I asked it to format it as code, and it did so.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R20" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">21</div></th><td class="s1">03 - Conditional Statements</td><td class="s1">02 - Check gas meter - Part A</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write Python code that does the following:<br><br>1. Declare a variable called &quot;gasMeter&quot; and set the value to 2.<br>2. Write an &quot;if&quot; statement that evaluates whether the value of &quot;gasMeter&quot; is less than or equal to 2. If it is, print a message telling the driver, &quot;Go fill up at a gas station.&quot;</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">gasMeter = 2<br>if gasMeter &lt;= 2:<br>    print(&quot;Go fill up at a gas station.&quot;)</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">Initially, ChatGPT did not format its code using a code block, which would have caused a problem when copying and pasting. I asked it to format it as code, and it did so.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R21" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">22</div></th><td class="s1">03 - Conditional Statements</td><td class="s1">03 - Check gas meter - Part B</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write Python code that does the following:<br><br>1. Declare a variable called &quot;gasMeter&quot; and set the value to 6.<br>2. Write an &quot;if&quot; statement that evaluates whether the value of &quot;gasMeter&quot; is less than or equal to 2. If it is, print a message telling the driver, &quot;Go fill up at a gas station.&quot; Otherwise, print a message telling the driver to &quot;Take it easy.&quot;</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">gasMeter = 6<br>if gasMeter &lt;= 2:<br>    print(&quot;Go fill up at a gas station.&quot;)<br>else:<br>    print(&quot;Take it easy.&quot;)</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">Initially, ChatGPT did not format its code using a code block, which would have caused a problem when copying and pasting. I asked it to format it as code, and it did so.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R22" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">23</div></th><td class="s1">03 - Conditional Statements</td><td class="s1">04 - Check gas meter - Part C</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write Python code that does the following:<br><br>1. Declare a variable called &quot;gasMeter&quot; and set the value to 1.<br>2. Declare a variable called &quot;stationNearby&quot; to False.<br><br>Then write nested &quot;if/else&quot; statements that do the following:<br><br>3. If the value of &quot;gasMeter&quot; is less than or equal to 2, print a message telling the driver, &quot;Your vehicle is low on gasoline.&quot;<br>4. If the value of &quot;gasMeter&quot; is less than or equal to 2 and there is a station nearby, also print a message telling the driver to &quot;There is a station nearby.&quot;<br>5. If the value of &quot;gasMeter&quot; is less than or equal to 2 and there is not a station nearby, tell the drive, &quot;You will soon be stranded on the side of the road.&quot;<br>6. If the value of &quot;gasMeter&quot; is greater than 2, tell the driver to &quot;Take it easy.&quot;</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">gasMeter = 1<br>stationNearby = False<br><br>if gasMeter &lt;= 2:<br>    print(&quot;Your vehicle is low on gasoline.&quot;)<br>    if stationNearby:<br>        print(&quot;There is a station nearby.&quot;)<br>    else:<br>        print(&quot;You will soon be stranded on the side of the road.&quot;)<br>else:<br>    print(&quot;Take it easy.&quot;)</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R23" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">24</div></th><td class="s1">03 - Conditional Statements</td><td class="s1">05 - Evaluate a person&#39;s age</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Below is the Python code you examined in the previous exercise. Please rework the code so that it only uses one &quot;if&quot; statement and one &quot;else&quot; statement. It may use multiple &quot;elif&quot; statements.<br><br>    age = 64<br><br>    if age &lt; 18:<br>        print(&quot;You are just a baby!&quot;)<br>    else:<br>        if age &lt; 25:<br>            print(&quot;Your brain still is not fully developed!&quot;)<br>        else:<br>            if age &lt; 30:<br>                print(&quot;You are over a quarter century old!&quot;)<br>            else:<br>                if age &lt; 50:<br>                    print(&quot;You are not half a century old yet!&quot;)<br>                else:<br>                    if age &lt; 60:<br>                        print(&quot;You are over half a century old!&quot;)<br>                    else:<br>                        print(&quot;You are just hitting your prime!&quot;)<br><br>Below is the expected output:<br><br>    You are just hitting your prime!</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">age = 64<br><br>if age &lt; 18:<br>    print(&quot;You are just a baby!&quot;)<br>elif age &lt; 25:<br>    print(&quot;Your brain still is not fully developed!&quot;)<br>elif age &lt; 30:<br>    print(&quot;You are over a quarter century old!&quot;)<br>elif age &lt; 50:<br>    print(&quot;You are not half a century old yet!&quot;)<br>elif age &lt; 60:<br>    print(&quot;You are over half a century old!&quot;)<br>else:<br>    print(&quot;You are just hitting your prime!&quot;)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R24" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">25</div></th><td class="s1">04 - Basic Functions</td><td class="s1">01 - Add two numbers</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a Python function called &quot;addNumbers&quot; that calculates the sum of two numeric values. The function should accept two arguments (the two numbers to be summed) and return the sum of these values.<br><br>Important: You only need to create the function. You do not need to invoke the function. CodeBuddy will invoke the function for you (using various different arguments). These are called &quot;tests.&quot; After performing each test, CodeBuddy will print what is returned from your function. Near the bottom of this page, you will see the tests that CodeBuddy uses to test your function. After submitting your code, you will see the output of each test.<br><br>Here is some code to get your started:<br><br>    # Here is some code to get you started:<br>    def addNumbers(number1, number2):<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addNumbers(2, 2))<br><br>Below is the expected output:<br><br>    4<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addNumbers(0, 0))<br><br>Below is the expected output:<br><br>    0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addNumbers(-1, 5001))<br><br>Below is the expected output:<br><br>    5000<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addNumbers(-5000.0, 5000.0))<br><br>Below is the expected output:<br><br>    0.0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addNumbers(5.5, 1))<br><br>Below is the expected output:<br><br>    6.5</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def addNumbers(number1, number2):<br>    return number1 + number2</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R25" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">26</div></th><td class="s1">04 - Basic Functions</td><td class="s1">02 - Add three numbers</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a Python function called &quot;addThreeNumbers&quot; that accepts 3 arguments. When this function is invoked, each of the parameter values will be either float or integer values. Your function should calculate and return the sum of these numbers.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addThreeNumbers(2, 2, 2))<br><br>Below is the expected output:<br><br>    6<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addThreeNumbers(-10, 10, 10.0))<br><br>Below is the expected output:<br><br>    10.0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addThreeNumbers(-1.0, 1.0, 5.5))<br><br>Below is the expected output:<br><br>    5.5</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def addThreeNumbers(num1, num2, num3):<br>    return num1 + num2 + num3</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R26" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">27</div></th><td class="s1">04 - Basic Functions</td><td class="s1">03 - Triple take</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;tripleTake&quot; that accepts one argument: a string. Your function should build a string that repeats the input string three times (with a period and a space between them and a period at the very end).<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(tripleTake(&quot;Biology is the study of living organisms&quot;))<br><br>Below is the expected output:<br><br>    Biology is the study of living organisms. Biology is the study of living organisms. Biology is the study of living organisms.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(tripleTake(&quot;I am learning to program in Python&quot;))<br><br>Below is the expected output:<br><br>    I am learning to program in Python. I am learning to program in Python. I am learning to program in Python.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def tripleTake(inputString):<br>    return inputString + &quot;. &quot; + inputString + &quot;. &quot; + inputString + &quot;.&quot;</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">ChatGPT put an extra space and an extra period at the end of the output. This was fixed on the second iteration.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R27" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">28</div></th><td class="s1">04 - Basic Functions</td><td class="s1">04 - Should I eat</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a Python function called &quot;shouldIEat&quot; that tells you whether you should eat, depending on how hungry you are. This function should accept a single argument. The value of that argument will be a number between 1 and 10. If your hunger level is greater than or equal to 8 and less than or equal to 10, the function should return, &quot;Go eat!&quot; If your hunger level is greater than or equal to 3 and less than 8, it should return, &quot;Don&#39;t eat!&quot; If your hunger level is greater than or equal to 1 and less than 3, it should return, &quot;Take a nap!&quot; If your hunger level is greater than 10 or less than 1, it should return, &quot;Invalid input.&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(shouldIEat(9.5))<br><br>Below is the expected output:<br><br>    Go eat!<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(shouldIEat(8.0))<br><br>Below is the expected output:<br><br>    Go eat!<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(shouldIEat(7.9))<br><br>Below is the expected output:<br><br>    Don&#39;t eat!<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(shouldIEat(3.0))<br><br>Below is the expected output:<br><br>    Don&#39;t eat!<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(shouldIEat(1.1))<br><br>Below is the expected output:<br><br>    Take a nap!<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(shouldIEat(11))<br><br>Below is the expected output:<br><br>    Invalid input.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(shouldIEat(-2))<br><br>Below is the expected output:<br><br>    Invalid input.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def shouldIEat(hunger_level):<br>    if hunger_level &gt;= 1 and hunger_level &lt;= 10:<br>        if hunger_level &gt;= 8:<br>            return &quot;Go eat!&quot;<br>        elif hunger_level &gt;= 3:<br>            return &quot;Don&#39;t eat!&quot;<br>        else:<br>            return &quot;Take a nap!&quot;<br>    else:<br>        return &quot;Invalid input.&quot;</td><td class="s2">7</td><td class="s2" dir="ltr">7</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R28" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">29</div></th><td class="s1">04 - Basic Functions</td><td class="s1">05 - Nematode worm development</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>An ecologist is studying the effects of ambient temperature on nematode worm development. Her strategy is to place the worms in small boxes and then to place these boxes under different levels of heat exposure in her lab. Her goal is to determine the minimum number of boxes that she should buy so that there will enough room for all the worms. Different boxes hold different numbers of worms. Assume that she will fill an entire box before moving to the next one. However, the last box may not be full, if there are not enough worms to fill it.<br><br>Please write a Python function that allows researchers to calculate the number of boxes that they will need, depending on how many worms they have. Name the function &quot;calculateWormBoxes&quot;. It should accept two arguments: 1) the number of worms and 2) the number of worms that can fit in a single box. Calculate the number of boxes that will be needed and return this number (an integer).<br><br>Hint(s):<br><br>    Don&#39;t forget to import the math module.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calculateWormBoxes(100, 20))<br><br>Below is the expected output:<br><br>    5<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calculateWormBoxes(337, 15))<br><br>Below is the expected output:<br><br>    23<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calculateWormBoxes(1026, 47))<br><br>Below is the expected output:<br><br>    22</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import math<br><br>def calculateWormBoxes(numWorms, wormsPerBox):<br>    return math.ceil(numWorms / wormsPerBox)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R29" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">30</div></th><td class="s1">04 - Basic Functions</td><td class="s1">06 - Salt Flats</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose you are studying mammal species in the Utah Salt Flats. You capture a variety of species, weigh each of them, and record this information in a lab notebook. When you get back to the university, you need to convert those measurements from pounds to kilograms. One kilogram is equivalent to 2.2046 pounds.<br><br>Please write a Python function called &quot;convertPoundsToKilograms&quot; that performs this calculation. It should accept two arguments: 1) the name of the species, and 2) the number of pounds. Return something like the following: &quot;Sylvilagus audubonii weighed 1.588 kg&quot;. (Please round the number of kilograms to three decimal places.)<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(convertPoundsToKilograms(&quot;Sylvilagus audubonii&quot;, 3.50))<br><br>Below is the expected output:<br><br>    Sylvilagus audubonii weighed 1.588 kg<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(convertPoundsToKilograms(&quot;Neotoma albigula&quot;, 15.45))<br><br>Below is the expected output:<br><br>    Neotoma albigula weighed 7.008 kg</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def convertPoundsToKilograms(species, pounds):<br>    kilograms = round(pounds / 2.2046, 3)<br>    return species + &quot; weighed &quot; + str(kilograms) + &quot; kg&quot;</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R30" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">31</div></th><td class="s1">04 - Basic Functions</td><td class="s1">07 - Predict population growth</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose you are working in a research laboratory with E. coli (bacteria) that grow in culture. As you keep the bacteria in proper growth conditions, they multiply at a rate of 20% each day. Suppose that other scientists in the same laboratory are working with other types of bacteria, and these grow at different rates than the E. coli. <br><br>Please write a function that allows researchers to predict the size of their bacterial population after 3 days. This function should be called &quot;predictPopulationGrowth&quot; and should accept three arguments: 1) the name of the bacterial species, 2) the number of bacteria at the beginning of the experiment, and 3) the growth rate per day. This function should calculate the expected growth after 3 days. This number should be rounded to the nearest whole number and converted to an integer value. The function should return a statement such as: &quot;After 3 days, there will be 16345 E. coli&quot;.<br><br>Hint(s):<br><br>    To perform this calculation, multiply the original value by one plus the growth rate. Multiply the resulting value by one plus the growth rate. Then repeat this process one more time.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(predictPopulationGrowth(&quot;E. coli&quot;, 10, 0.20))<br><br>Below is the expected output:<br><br>    After 3 days, there will be 17 E. coli<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(predictPopulationGrowth(&quot;Yersinia pestis&quot;, 42847, 0.80))<br><br>Below is the expected output:<br><br>    After 3 days, there will be 249884 Yersinia pestis<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(predictPopulationGrowth(&quot;Actinobacteria&quot;, 9947, 0.0125))<br><br>Below is the expected output:<br><br>    After 3 days, there will be 10325 Actinobacteria</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def predictPopulationGrowth(species, population, rate):<br>    for i in range(3):<br>        population = population * (1 + rate)<br>    return &quot;After 3 days, there will be &quot; + str(int(round(population))) + &quot; &quot; + species</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R31" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">32</div></th><td class="s1">04 - Basic Functions</td><td class="s1">08 - Odd or even</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a Python function called &quot;oddOrEven&quot; that indicates whether an integer is odd or even. This function should accept a single argument that is a number. If the number is odd, the function should return &quot;odd&quot;. If the number if even, it should return &quot;even&quot;. However, the number must be a whole number to be considered odd or even. If it is not a whole number, your function should return &quot;Not a whole number&quot;.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(oddOrEven(3))<br><br>Below is the expected output:<br><br>    odd<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(oddOrEven(4))<br><br>Below is the expected output:<br><br>    even<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(oddOrEven(-3))<br><br>Below is the expected output:<br><br>    odd<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(oddOrEven(-4))<br><br>Below is the expected output:<br><br>    even<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(oddOrEven(5.25))<br><br>Below is the expected output:<br><br>    Not a whole number</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def oddOrEven(num):<br>    if type(num) != int:<br>        return &quot;Not a whole number&quot;<br>    elif num % 2 == 0:<br>        return &quot;even&quot;<br>    else:<br>        return &quot;odd&quot;</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R32" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">33</div></th><td class="s1">04 - Basic Functions</td><td class="s1">09 - Maximum milk production</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Let&#39;s say you are tracking the amount of milk produced by each of three cows. Each day you want to know the maximum amount of milk produced by any of the cows in the group. <br><br>Please write a Python function called &quot;getMaxMilk&quot; that accepts three arguments (&quot;cow1&quot;, &quot;cow2&quot;, and &quot;cow3&quot;), which are each numbers that represent the amount of milk produced by the cows. Your function should figure out the maximum amount of milk produced and then return the maximum value.<br><br>If there is a tie, just have it return any of the values.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(getMaxMilk(5.2, 3.3, 4.4))<br><br>Below is the expected output:<br><br>    5.2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(getMaxMilk(0.9, 8.8, 1.2))<br><br>Below is the expected output:<br><br>    8.8<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(getMaxMilk(1.0, 1.0, 1.0))<br><br>Below is the expected output:<br><br>    1.0</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def getMaxMilk(cow1, cow2, cow3):<br>    return max(cow1, cow2, cow3)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">ChatGPT&#39;s solution is much simpler than the instructor&#39;s. In the course, we do not teach how to use pass multiple arguments to the max() function. The instructor&#39;s solution deliberately uses conditional statements under the belief that this solution would be more intuitive to students.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R33" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">34</div></th><td class="s1">04 - Basic Functions</td><td class="s1">10 - Find stop codon</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>As cellular RNA is translated into proteins, the cellular machinery processes the RNA in triplets (codons). Each codon is translated into an amino acid. There is a specific set of rules that determine which amino acid corresponds to which codon. For example, the codon &quot;CTT&quot; would be translated into a &quot;Leucine&quot; amino acid. This process continues until the cellular machine encounters a &quot;stop&quot; codon, which indicates that the translation process should be halted.<br><br>Examine this table. Identify the three codons that are stop codons. Then write a Python function called &quot;isStopCodon&quot; that accepts a single argument: a string that contains the codon sequence. Your function should return a Boolean value that indicates whether the specified codon is a stop codon.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isStopCodon(&quot;ATA&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isStopCodon(&quot;GTC&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isStopCodon(&quot;CCC&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isStopCodon(&quot;TAA&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isStopCodon(&quot;AAA&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isStopCodon(&quot;AAA&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isStopCodon(&quot;CGT&quot;))<br><br>Below is the expected output:<br><br>    False</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def isStopCodon(codon):<br>    stop_codons = [&quot;TAA&quot;, &quot;TAG&quot;, &quot;TGA&quot;]<br>    return codon in stop_codons</td><td class="s2">7</td><td class="s2" dir="ltr">7</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">ChatGPT&#39;s solution uses the &quot;in&quot; operator, which is more succinct than the instructor&#39;s solution. However, students have not learned about lists or the &quot;in&quot; operator at this point in the course.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R34" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">35</div></th><td class="s1">05 - Lists</td><td class="s1">01 - Is list empty</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;isListEmpty&quot; that accepts a single argument: a list. Write this function so that it returns a Boolean value that indicates whether the list is empty.<br><br>Hint(s):<br><br>    You can use the &quot;len&quot; function to count the number of elements in a list.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isListEmpty([2, 8, 7]))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isListEmpty([&quot;cat&quot;, &quot;dog&quot;, &quot;parrot&quot;, &quot;camel&quot;, &quot;horse&quot;, &quot;penguin&quot;, &quot;owl&quot;, &quot;tiger&quot;, &quot;chimpanzee&quot;]))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isListEmpty([7.75, 3.32]))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isListEmpty([]))<br><br>Below is the expected output:<br><br>    True</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def isListEmpty(list):<br>    return len(list) == 0</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R35" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">36</div></th><td class="s1">05 - Lists</td><td class="s1">02 - Return third element</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Declare a function called &quot;returnThirdElement&quot;. This function should accept one argument: a list. If the list has at least 3 elements in it, your function should return the third element. If it includes fewer than 3 elements, it should return, &quot;There are fewer than 3 elements in this list.&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(returnThirdElement([2, 8, 7]))<br><br>Below is the expected output:<br><br>    7<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(returnThirdElement([&quot;cat&quot;, &quot;dog&quot;, &quot;parrot&quot;, &quot;camel&quot;, &quot;horse&quot;, &quot;penguin&quot;, &quot;owl&quot;, &quot;tiger&quot;, &quot;chimpanzee&quot;]))<br><br>Below is the expected output:<br><br>    parrot<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(returnThirdElement([7.75, 3.32]))<br><br>Below is the expected output:<br><br>    There are fewer than 3 elements in this list.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(returnThirdElement([]))<br><br>Below is the expected output:<br><br>    There are fewer than 3 elements in this list.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def returnThirdElement(list):<br>    if len(list) &gt;= 3:<br>        return list[2]<br>    else:<br>        return &quot;There are fewer than 3 elements in this list.&quot;</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R36" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">37</div></th><td class="s1">05 - Lists</td><td class="s1">03 - Build a zoo</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Write a function called &quot;buildZoo&quot; that accepts 3 arguments. Each of these arguments will be a list of animal names. Your function should combine these lists into a single list and then return a statement like the following: &quot;The zoo has 12 animals.&quot;<br><br>Important: please do not use the &quot;extend()&quot; function with these lists.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    fish = [&quot;shark&quot;, &quot;piranha&quot;]<br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    reptiles = [&quot;alligator&quot;, &quot;crocodile&quot;, &quot;gecko&quot;, &quot;iguana&quot;]<br>    <br>    print(buildZoo(mammals, birds, fish))<br><br>Below is the expected output:<br><br>    The zoo has 15 animals.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    fish = [&quot;shark&quot;, &quot;piranha&quot;]<br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    reptiles = [&quot;alligator&quot;, &quot;crocodile&quot;, &quot;gecko&quot;, &quot;iguana&quot;]<br>    <br>    print(buildZoo(birds, amphibians, reptiles))<br><br>Below is the expected output:<br><br>    The zoo has 15 animals.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    fish = [&quot;shark&quot;, &quot;piranha&quot;]<br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    reptiles = [&quot;alligator&quot;, &quot;crocodile&quot;, &quot;gecko&quot;, &quot;iguana&quot;]<br>    <br>    print(buildZoo(reptiles, birds, mammals))<br><br>Below is the expected output:<br><br>    The zoo has 17 animals.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def buildZoo(list1, list2, list3):<br>    zoo = list1 + list2 + list3<br>    animal_count = len(zoo)<br>    if animal_count == 0:<br>        return &quot;The zoo is empty.&quot;<br>    else:<br>        return f&quot;The zoo has {animal_count} animals.&quot;</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R37" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">38</div></th><td class="s1">05 - Lists</td><td class="s1">04 - Check animal type</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Write a function called &quot;checkAnimalType&quot;. This function should accept 4 arguments: 1) a list of mammals, 2) a list of birds, 3) a list of amphibians, and 4) the name of an animal that you want to check against these lists. Your function should return a string value that indicates what type of animal it is (mammal, bird, or amphibian). If the animal is none of these types, the function should return &quot;No match&quot;.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    <br>    print(checkAnimalType(mammals, birds, amphibians, &quot;ostrich&quot;))<br><br>Below is the expected output:<br><br>    bird<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    <br>    print(checkAnimalType(mammals, birds, amphibians, &quot;lion&quot;))<br><br>Below is the expected output:<br><br>    mammal<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    <br>    print(checkAnimalType(mammals, birds, amphibians, &quot;emu&quot;))<br><br>Below is the expected output:<br><br>    bird<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    <br>    print(checkAnimalType(mammals, birds, amphibians, &quot;ewok&quot;))<br><br>Below is the expected output:<br><br>    No match<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    <br>    print(checkAnimalType(mammals, birds, amphibians, &quot;badger&quot;))<br><br>Below is the expected output:<br><br>    No match</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def checkAnimalType(mammals, birds, amphibians, animal):<br>    if animal in mammals:<br>        return &quot;mammal&quot;<br>    elif animal in birds:<br>        return &quot;bird&quot;<br>    elif animal in amphibians:<br>        return &quot;amphibian&quot;<br>    else:<br>        return &quot;No match&quot;<br></td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R38" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">39</div></th><td class="s1">05 - Lists</td><td class="s1">05 - Calculate average</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;calculateAverage&quot; that accepts a single argument: a list that contains numeric values. Your function should return the average of these values.<br><br>Hint(s):<br><br>    One way to calculate the average is to first calculate the sum and then divide by the number of elements in the list.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calculateAverage([1, 2, 3]))<br><br>Below is the expected output:<br><br>    2.0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calculateAverage([4.56382, 2.91838, 3.92918, 105.21928, -10.1982, -0.0231]))<br><br>Below is the expected output:<br><br>    17.734893333333332<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calculateAverage([0]))<br><br>Below is the expected output:<br><br>    0.0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calculateAverage([0, 0, 0, 0, 0.0005]))<br><br>Below is the expected output:<br><br>    0.0001</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def calculateAverage(numericValues):<br>    return sum(numericValues) / len(numericValues)</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R39" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">40</div></th><td class="s1">05 - Lists</td><td class="s1">06 - Get first and last elements</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;getFirstAndLast&quot; that accepts one argument: a list. Your function should retrieve the first and last elements from the specified list and create a new list that contains these two elements. Your function should return this new list.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    <br>    print(getFirstAndLast(mammals))<br><br>Below is the expected output:<br><br>    [&#39;rhino&#39;, &#39;panda&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    <br>    print(getFirstAndLast(birds))<br><br>Below is the expected output:<br><br>    [&#39;parrot&#39;, &#39;peacock&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    fish = [&quot;shark&quot;, &quot;piranha&quot;]<br>    <br>    print(getFirstAndLast(fish))<br><br>Below is the expected output:<br><br>    [&#39;shark&#39;, &#39;piranha&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    <br>    print(getFirstAndLast(amphibians))<br><br>Below is the expected output:<br><br>    [&#39;frog&#39;, &#39;mantella&#39;]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def getFirstAndLast(lst):<br>    return [lst[0], lst[-1]]</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R40" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">41</div></th><td class="s1">05 - Lists</td><td class="s1">07 - Get all but first and last</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;getAllButFirstAndLast&quot; that accepts one argument: a list. Your function should remove the first and last elements from the specified list and return the list with the remaining items.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    <br>    print(getAllButFirstAndLast(mammals))<br><br>Below is the expected output:<br><br>    [&#39;elephant&#39;, &#39;lion&#39;, &#39;tiger&#39;, &#39;hippo&#39;, &#39;cheetah&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    <br>    print(getAllButFirstAndLast(birds))<br><br>Below is the expected output:<br><br>    [&#39;flamingo&#39;, &#39;ostrich&#39;, &#39;owl&#39;, &#39;emu&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    fish = [&quot;shark&quot;, &quot;piranha&quot;]<br>    <br>    print(getAllButFirstAndLast(fish))<br><br>Below is the expected output:<br><br>    []<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    <br>    print(getAllButFirstAndLast(amphibians))<br><br>Below is the expected output:<br><br>    [&#39;toad&#39;, &#39;newt&#39;, &#39;salamander&#39;]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def getAllButFirstAndLast(lst):<br>    return lst[1:-1]</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R41" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">42</div></th><td class="s1">05 - Lists</td><td class="s1">08 - Update zoo</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;updateZoo&quot;. This function should accept three arguments: 1) a list of animals, 2) an animal to remove from the zoo, and 3) an animal to replace it with. The function should remove the requested animal and replace it with the animal that has been requested to be added.<br><br>Important: the new animal should end up in the same position in the list as the animal that was removed. (Think of this as a zoo where one animal&#39;s cage is cleaned out and a new animal moves in.)<br><br>Finally, your function should return the modified list.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    <br>    print(updateZoo(mammals, &quot;hippo&quot;, &quot;honey badger&quot;))<br><br>Below is the expected output:<br><br>    [&#39;rhino&#39;, &#39;elephant&#39;, &#39;lion&#39;, &#39;tiger&#39;, &#39;honey badger&#39;, &#39;cheetah&#39;, &#39;panda&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    <br>    print(updateZoo(birds, &quot;owl&quot;, &quot;falcon&quot;))<br><br>Below is the expected output:<br><br>    [&#39;parrot&#39;, &#39;flamingo&#39;, &#39;ostrich&#39;, &#39;falcon&#39;, &#39;emu&#39;, &#39;peacock&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    fish = [&quot;shark&quot;, &quot;piranha&quot;]<br>    <br>    print(updateZoo(fish, &quot;piranha&quot;, &quot;clownfish&quot;))<br><br>Below is the expected output:<br><br>    [&#39;shark&#39;, &#39;clownfish&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    <br>    print(updateZoo(amphibians, &quot;frog&quot;, &quot;mudpuppy&quot;))<br><br>Below is the expected output:<br><br>    [&#39;mudpuppy&#39;, &#39;toad&#39;, &#39;newt&#39;, &#39;salamander&#39;, &#39;mantella&#39;]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def updateZoo(animals, animalToRemove, animalToAdd):<br>    index = animals.index(animalToRemove)<br>    animals.pop(index)<br>    animals.insert(index, animalToAdd)<br>    return animals</td><td class="s2" dir="ltr">6</td><td class="s2" dir="ltr">6</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R42" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">43</div></th><td class="s1">05 - Lists</td><td class="s1">09 - Clone animals</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;cloneAnimals&quot;. This function should accept two arguments: 1) a list of animals and 2) the number of times that the animals should be cloned. The function should create the specified number of copies of the animals and add them to the original list. The function should then sort the entire list and return that list.<br><br>Note: If you clone your animals zero times, your list should be the same as the original list, except that it will be sorted. If you clone your animals one time, your new list will have two copies of each animal, etc.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    <br>    print(cloneAnimals(mammals, 1))<br><br>Below is the expected output:<br><br>    [&#39;cheetah&#39;, &#39;cheetah&#39;, &#39;elephant&#39;, &#39;elephant&#39;, &#39;hippo&#39;, &#39;hippo&#39;, &#39;lion&#39;, &#39;lion&#39;, &#39;panda&#39;, &#39;panda&#39;, &#39;rhino&#39;, &#39;rhino&#39;, &#39;tiger&#39;, &#39;tiger&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    <br>    print(cloneAnimals(birds, 2))<br><br>Below is the expected output:<br><br>    [&#39;emu&#39;, &#39;emu&#39;, &#39;emu&#39;, &#39;flamingo&#39;, &#39;flamingo&#39;, &#39;flamingo&#39;, &#39;ostrich&#39;, &#39;ostrich&#39;, &#39;ostrich&#39;, &#39;owl&#39;, &#39;owl&#39;, &#39;owl&#39;, &#39;parrot&#39;, &#39;parrot&#39;, &#39;parrot&#39;, &#39;peacock&#39;, &#39;peacock&#39;, &#39;peacock&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    fish = [&quot;shark&quot;, &quot;piranha&quot;]<br>    <br>    print(cloneAnimals(fish, 5))<br><br>Below is the expected output:<br><br>    [&#39;piranha&#39;, &#39;piranha&#39;, &#39;piranha&#39;, &#39;piranha&#39;, &#39;piranha&#39;, &#39;piranha&#39;, &#39;shark&#39;, &#39;shark&#39;, &#39;shark&#39;, &#39;shark&#39;, &#39;shark&#39;, &#39;shark&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    <br>    print(cloneAnimals(amphibians, 0))<br><br>Below is the expected output:<br><br>    [&#39;frog&#39;, &#39;mantella&#39;, &#39;newt&#39;, &#39;salamander&#39;, &#39;toad&#39;]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def cloneAnimals(animals, num_of_clones):<br>    if num_of_clones == 0:<br>        return sorted(animals)<br>    else:<br>        new_list = animals.copy()<br>        for i in range(num_of_clones):<br>            new_list += animals<br>        return sorted(new_list)</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">5</td><td class="s1" dir="ltr">The first four solutions had logic errors. I tried to help it understand where it was going wrong and encouraged it to ensure the output would match the expected output. On the fifth attempt, I simplified the prompt, and it worked. That may have been just due to stochasticity, but it may have been easier for ChatGPT to interpret the core requirements.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">No</td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R43" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">44</div></th><td class="s1">05 - Lists</td><td class="s1">10 - Average last three elements</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;averageLastThree&quot;. This function should accept a single argument: a list that contains numeric values. The function should calculate the average of the last three elements in the list and return that value.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(averageLastThree([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]))<br><br>Below is the expected output:<br><br>    8.0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(averageLastThree([9, 8, 7, 6, 5, 4, 3, 2, 1, 0]))<br><br>Below is the expected output:<br><br>    1.0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(averageLastThree([0, 0, 0, 0, 0, 0, 0, 1]))<br><br>Below is the expected output:<br><br>    0.3333333333333333<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(averageLastThree([98.183, 800.0, 561.238]))<br><br>Below is the expected output:<br><br>    486.4736666666667</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def averageLastThree(lst):<br>    if len(lst) &lt; 3:<br>        return sum(lst) / len(lst)<br>    else:<br>        return sum(lst[-3:]) / 3</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R44" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">45</div></th><td class="s1">05 - Lists</td><td class="s1">11 - Reverse last three elements</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;reverseLastThree&quot;. This function should accept a single argument: 1) a list. Your function should create a new list that contains the last three elements in the original list in reverse order. Your function should then return this new list.<br><br>Important: if the original list contains fewer than three elements, your function should return whatever elements are present (in reverse order).<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(reverseLastThree([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]))<br><br>Below is the expected output:<br><br>    [9, 8, 7]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(reverseLastThree([9, 8, 7, 6, 5, 4, 3, 2, 1, 0]))<br><br>Below is the expected output:<br><br>    [0, 1, 2]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(reverseLastThree([0, 0, 0, 0, 0, 0, 0, 1]))<br><br>Below is the expected output:<br><br>    [1, 0, 0]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(reverseLastThree([98.183, 800.0, 561.238]))<br><br>Below is the expected output:<br><br>    [561.238, 800.0, 98.183]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(reverseLastThree([-2.1, 0.2819238]))<br><br>Below is the expected output:<br><br>    [0.2819238, -2.1]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def reverseLastThree(lst):<br>    if len(lst) &gt;= 3:<br>        return lst[-3:][::-1]<br>    else:<br>        return lst[::-1]</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">5</td><td class="s1" dir="ltr">The first four solutions had logic errors. It was including more than 3 elements in the list (when the original list had more than three elements). The fifth solution was slightly different, and it worked.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R45" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">46</div></th><td class="s1">05 - Lists</td><td class="s1">12 - Retrieve even values</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;retrieveEvenValues&quot;. This function should accept a single argument: a list. The function should create a new list that contains the values from the original list that have an even index number starting at 2 (for example, 2, 4, 6, ...) and then return that new list.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(retrieveEvenValues([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]))<br><br>Below is the expected output:<br><br>    [2, 4, 6, 8]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(retrieveEvenValues([9, 8, 7, 6, 5, 4, 3, 2, 1, 0]))<br><br>Below is the expected output:<br><br>    [7, 5, 3, 1]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(retrieveEvenValues([0, 0, 0, 0, 0, 0, 0, 1]))<br><br>Below is the expected output:<br><br>    [0, 0, 0]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(retrieveEvenValues([98.183, 800.0, 561.238]))<br><br>Below is the expected output:<br><br>    [561.238]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def retrieveEvenValues(list):<br>  return [list[i] for i in range(2, len(list), 2)]</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">ChatGPT&#39;s original solution used a list comprehension. I briefly explain those to students, but most would not be comfortable with using them. (The solution in my saved ChatGPT results does not use a list comprehension. There was a systems error, and it lost the original solution.)</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R46" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">47</div></th><td class="s1">05 - Lists</td><td class="s1">13 - Use list helpers</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;useListHelpers&quot;. This function should accept a single argument: a list. The function should perform the following tasks:<br><br>1. Create a new list that contains the values from the original list that have an even index number starting at 2 (for example, 2, 4, 6, ...).<br>2. Use the &quot;insert()&quot; function to add an element with value &quot;beginning&quot; at the beginning of the new list.<br>3. Use the &quot;append()&quot; function to add an element with value &quot;end&quot; to the end of the new list.<br>4. Use the &quot;pop()&quot; function remove the 3rd element (index = 2) and then use the &quot;insert()&quot; function to add it to the beginning of the new list.<br>5. Return the new list.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(useListHelpers([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]))<br><br>Below is the expected output:<br><br>    [4, &#39;beginning&#39;, 2, 6, 8, &#39;end&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(useListHelpers([9, 8, 7, 6, 5, 4, 3, 2, 1, 0]))<br><br>Below is the expected output:<br><br>    [5, &#39;beginning&#39;, 7, 3, 1, &#39;end&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(useListHelpers([0, 0, 0, 0, 0, 0, 0, 1]))<br><br>Below is the expected output:<br><br>    [0, &#39;beginning&#39;, 0, 0, &#39;end&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(useListHelpers([98.183, 800.0, 561.238]))<br><br>Below is the expected output:<br><br>    [&#39;end&#39;, &#39;beginning&#39;, 561.238]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def useListHelpers(lst):<br>    new_list = [lst[i] for i in range(2, len(lst), 2)]<br>    new_list.insert(0, &quot;beginning&quot;)<br>    new_list.append(&quot;end&quot;)<br>    popped = new_list.pop(2)<br>    new_list.insert(0, popped)<br>    return new_list</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R47" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">48</div></th><td class="s1">05 - Lists</td><td class="s1">14 - Collect garbage</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;collectGarbage&quot;. This function should accept a single argument: a list. The function should perform the following tasks:<br><br>1. Use the &quot;remove()&quot; function to remove the first element from the list that has a value of &quot;garbage&quot;. Hint: first check to make sure there is at least one element in the list with a value of &quot;garbage&quot;.<br>2. Return the number of remaining elements in the list that have a value of &quot;garbage&quot;. Hint: use the &quot;count()&quot; function.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(collectGarbage([&quot;garbage&quot;, &quot;can&quot;, &quot;litter&quot;, &quot;garbage&quot;, &quot;refuse&quot;, &quot;scrap&quot;, &quot;junk&quot;, &quot;garbage&quot;]))<br><br>Below is the expected output:<br><br>    2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(collectGarbage([&quot;litter&quot;, &quot;refuse&quot;, &quot;scrap&quot;, &quot;junk&quot;, &quot;garbage&quot;]))<br><br>Below is the expected output:<br><br>    0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(collectGarbage([&quot;litter&quot;, &quot;refuse&quot;]))<br><br>Below is the expected output:<br><br>    0</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def collectGarbage(lst):<br>    if &quot;garbage&quot; in lst:<br>        lst.remove(&quot;garbage&quot;)<br>    return lst.count(&quot;garbage&quot;)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R48" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">49</div></th><td class="s1">05 - Lists</td><td class="s1">15 - Find garbage</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;findGarbage&quot;. This function should accept a single argument: a list. The function should use the &quot;index()&quot; function to identify the index of the first element in the list that has a value of &quot;garbage&quot;. If no element in the list has a value of &quot;garbage&quot;, your function should return &quot;not found&quot;. Otherwise, your function should return the index.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findGarbage([&quot;garbage&quot;, &quot;can&quot;, &quot;litter&quot;, &quot;garbage&quot;, &quot;refuse&quot;, &quot;scrap&quot;, &quot;junk&quot;, &quot;garbage&quot;]))<br><br>Below is the expected output:<br><br>    0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findGarbage([&quot;litter&quot;, &quot;refuse&quot;, &quot;scrap&quot;, &quot;junk&quot;, &quot;garbage&quot;]))<br><br>Below is the expected output:<br><br>    4<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findGarbage([&quot;litter&quot;, &quot;refuse&quot;]))<br><br>Below is the expected output:<br><br>    not found</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def findGarbage(lst):<br>    if &quot;garbage&quot; in lst:<br>        return lst.index(&quot;garbage&quot;)<br>    else:<br>        return &quot;not found&quot;</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R49" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">50</div></th><td class="s1">05 - Lists</td><td class="s1">16 - Build nested list 1</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;buildNestedList&quot;. This function should accept no arguments. Your function should perform the following tasks:<br><br>1. Declare a list variable called &quot;list1&quot;. This list should contain two values (in this order): &quot;dog&quot; and &quot;cat&quot;.<br>2. Declare a list variable called &quot;list2&quot;. This list should contain three values (in this order): &quot;sheep&quot;, &quot;horse&quot;, and &quot;marmoset&quot;.<br>3. Declare a list variable called &quot;list3&quot;. Append &quot;list1&quot; and &quot;list2&quot; (one at a time) to &quot;list3&quot;.<br>4. Return a value that indicates the length of &quot;list3&quot;.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildNestedList())<br><br>Below is the expected output:<br><br>    2</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def buildNestedList():<br>    list1 = [&quot;dog&quot;, &quot;cat&quot;]<br>    list2 = [&quot;sheep&quot;, &quot;horse&quot;, &quot;marmoset&quot;]<br>    list3 = []<br>    list3.append(list1)<br>    list3.append(list2)<br>    return len(list3)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R50" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">51</div></th><td class="s1">05 - Lists</td><td class="s1">17 - Build nested list 2</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;buildNestedList2&quot;. This function should accept no arguments. Your function should perform the following tasks:<br><br>1. Declare a list variable called &quot;list1&quot;. This list should contain two values (in this order): &quot;dog&quot; and &quot;cat&quot;.<br>2. Declare a list variable called &quot;list2&quot;. This list should contain three values (in this order): &quot;sheep&quot;, &quot;horse&quot;, and &quot;marmoset&quot;.<br>3. Declare a list variable called &quot;list3&quot;. Append &quot;list1&quot; and &quot;list2&quot; (in this order) to &quot;list3&quot;.<br>4. Return a value that indicates the length of the second element (index = 1) within &quot;list3&quot;.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildNestedList2())<br><br>Below is the expected output:<br><br>    3</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def buildNestedList2():<br>    list1 = [&quot;dog&quot;, &quot;cat&quot;]<br>    list2 = [&quot;sheep&quot;, &quot;horse&quot;, &quot;marmoset&quot;]<br>    list3 = [list1, list2]<br>    return len(list3[1])</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R51" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">52</div></th><td class="s1">05 - Lists</td><td class="s1">18 - Build nested list 3</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;buildNestedList3&quot;. This function should accept no arguments. Your function should perform the following tasks:<br><br>1. Declare a list variable called &quot;list1&quot;. This list should contain two values (in this order): &quot;dog&quot; and &quot;cat&quot;.<br>2. Declare a list variable called &quot;list2&quot;. This list should contain three values (in this order): &quot;sheep&quot;, &quot;horse&quot;, and &quot;marmoset&quot;.<br>3. Declare a list variable called &quot;list3&quot;. Append &quot;list1&quot; and &quot;list2&quot; (in this order) to &quot;list3&quot;.<br>4. Return a value that indicates the value of the 0th element within the 0th element of &quot;list3&quot;.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildNestedList3())<br><br>Below is the expected output:<br><br>    dog</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def buildNestedList3():<br>    list1 = [&quot;dog&quot;, &quot;cat&quot;]<br>    list2 = [&quot;sheep&quot;, &quot;horse&quot;, &quot;marmoset&quot;]<br>    list3 = []<br>    list3.append(list1)<br>    list3.append(list2)<br>    return list3[0][0]</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R52" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">53</div></th><td class="s1">06 - &quot;for&quot; Loops</td><td class="s1">01 - Use the range function</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;useRangeFunction&quot; that accepts a single argument: an integer value. Your function should use the &quot;range()&quot; function to create a string. That string should contain a sequence of numbers starting at zero and ending at the number specified in the argument. For example, if the argument were 3, your function would return &quot;1 2 3 &quot;. (It includes a space at the end of the string.)<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(useRangeFunction(3))<br><br>Below is the expected output:<br><br>    0 1 2 3<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(useRangeFunction(5))<br><br>Below is the expected output:<br><br>    0 1 2 3 4 5<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(useRangeFunction(10))<br><br>Below is the expected output:<br><br>    0 1 2 3 4 5 6 7 8 9 10</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def useRangeFunction(n):<br>    return &quot; &quot;.join(str(x) for x in range(n+1))</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R53" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">54</div></th><td class="s1">06 - &quot;for&quot; Loops</td><td class="s1">02 - Use the enumerate function</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;useEnumerateFunction&quot; that accepts a single argument: a list. Your function should use the &quot;enumerate()&quot; function to create a string. For each element in the list, your function should add the index of that element, followed by an &quot;=&quot; followed by the value of that element. Your function should then return the string. For example, if the list argument were &quot;[&quot;dog&quot;, &quot;cat&quot;, &quot;cow&quot;]&quot;, your function would return &quot;0=dog 1=cat 2=cow &quot;. (It includes a space at the end of the string.)<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(useEnumerateFunction([&quot;dog&quot;, &quot;cat&quot;, &quot;cow&quot;]))<br><br>Below is the expected output:<br><br>    0=dog 1=cat 2=cow<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(useEnumerateFunction([&quot;rabbit&quot;, &quot;pig&quot;, &quot;sheep&quot;, &quot;deer&quot;, &quot;elk&quot;]))<br><br>Below is the expected output:<br><br>    0=rabbit 1=pig 2=sheep 3=deer 4=elk<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(useEnumerateFunction([&quot;gorilla&quot;, &quot;chimpanzee&quot;, &quot;orangutan&quot;, &quot;human&quot;, &quot;bonobo&quot;, &quot;monkey&quot;]))<br><br>Below is the expected output:<br><br>    0=gorilla 1=chimpanzee 2=orangutan 3=human 4=bonobo 5=monkey</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def useEnumerateFunction(lst):<br>    return &quot; &quot;.join(f&quot;{index}={value}&quot; for index, value in enumerate(lst))</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R54" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">55</div></th><td class="s1">06 - &quot;for&quot; Loops</td><td class="s1">03 - Be overly friendly</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;beOverlyFriendly&quot; that accepts a single argument: an integer value. This value will indicate the number of times that your function should say &quot;Hello!&quot; Within the function, use a &quot;for&quot; loop that creates a string that repeats this greeting the specified number of times. (Include a space between each greeting and a space at the end.) Your function should return the string that it created.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(beOverlyFriendly(3))<br><br>Below is the expected output:<br><br>    Hello! Hello! Hello!<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(beOverlyFriendly(5))<br><br>Below is the expected output:<br><br>    Hello! Hello! Hello! Hello! Hello!<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(beOverlyFriendly(10))<br><br>Below is the expected output:<br><br>    Hello! Hello! Hello! Hello! Hello! Hello! Hello! Hello! Hello! Hello!</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def beOverlyFriendly(n):<br>    return &quot; &quot;.join(&quot;Hello!&quot; for i in range(n))</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R55" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">56</div></th><td class="s1">06 - &quot;for&quot; Loops</td><td class="s1">04 - Square a list of numbers</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;squareList&quot; that accepts a single argument: a list of numbers. Your function should use a &quot;for&quot; loop to square each numeric value and return a new list with these squared values. For example, if your list contained the values &quot;2&quot; and &quot;3&quot;, the new list would contain the values &quot;4&quot; and &quot;9&quot;.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(squareList([2, 3]))<br><br>Below is the expected output:<br><br>    [4, 9]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(squareList([5, 1, 8]))<br><br>Below is the expected output:<br><br>    [25, 1, 64]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(squareList([-3.3, 6.9, -0.00000000021]))<br><br>Below is the expected output:<br><br>    [10.889999999999999, 47.61000000000001, 4.41e-20]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def squareList(numbers):<br>    squared_numbers = []<br>    for num in numbers:<br>        squared_numbers.append(num**2)<br>    return squared_numbers</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R56" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">57</div></th><td class="s1">06 - &quot;for&quot; Loops</td><td class="s1">05 - Truncate list</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;truncateList&quot; that accepts two arguments: 1) a list and 2) the number of items to remove from the end of that list. Your function should use a loop to remove the specified number of items from the end of the list. You could use &quot;slicing&quot; to do this; however, to strengthen your understanding of loops, please use a loop instead. Your function should return the truncated list.<br><br>Hint(s):<br><br>    You might consider using the &quot;pop()&quot; function in combination with the &quot;range()&quot; function within your loop.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    mammals = [&quot;rhino&quot;, &quot;elephant&quot;, &quot;lion&quot;, &quot;tiger&quot;, &quot;hippo&quot;, &quot;cheetah&quot;, &quot;panda&quot;]<br>    <br>    print(truncateList(mammals, 4))<br><br>Below is the expected output:<br><br>    [&#39;rhino&#39;, &#39;elephant&#39;, &#39;lion&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    birds = [&quot;parrot&quot;, &quot;flamingo&quot;, &quot;ostrich&quot;, &quot;owl&quot;, &quot;emu&quot;, &quot;peacock&quot;]<br>    <br>    print(truncateList(birds, 1))<br><br>Below is the expected output:<br><br>    [&#39;parrot&#39;, &#39;flamingo&#39;, &#39;ostrich&#39;, &#39;owl&#39;, &#39;emu&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    fish = [&quot;shark&quot;, &quot;piranha&quot;]<br>    <br>    print(truncateList(fish, 2))<br><br>Below is the expected output:<br><br>    []<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    amphibians = [&quot;frog&quot;, &quot;toad&quot;, &quot;newt&quot;, &quot;salamander&quot;, &quot;mantella&quot;]<br>    <br>    print(truncateList(amphibians, 0))<br><br>Below is the expected output:<br><br>    [&#39;frog&#39;, &#39;toad&#39;, &#39;newt&#39;, &#39;salamander&#39;, &#39;mantella&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    reptiles = [&quot;alligator&quot;, &quot;crocodile&quot;, &quot;gecko&quot;, &quot;iguana&quot;]<br>    <br>    print(truncateList(reptiles, 3))<br><br>Below is the expected output:<br><br>    [&#39;alligator&#39;]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def truncateList(lst, n):<br>    for i in range(n):<br>        lst.pop()<br>    return lst</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R57" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">58</div></th><td class="s1">06 - &quot;for&quot; Loops</td><td class="s1">06 - Build even-numbered list</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;buildEvenNumberList&quot; that accepts 2 arguments: 1) a starting number and 2) an ending number. The values of these arguments will be integers. Your function should build a list that contains all the even integers between the specified starting number and the specified ending number. If the starting number is even, it should be included in the list. The ending number should never be included in the list. At the end of your function, return the even-number list.<br><br>Although you might be able to accomplish this task with list slicing, please solve this task using a loop so you can gain additional practice using loops.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildEvenNumberList(0, 11))<br><br>Below is the expected output:<br><br>    [0, 2, 4, 6, 8, 10]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildEvenNumberList(1, 10))<br><br>Below is the expected output:<br><br>    [2, 4, 6, 8]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildEvenNumberList(-5, 12))<br><br>Below is the expected output:<br><br>    [-4, -2, 0, 2, 4, 6, 8, 10]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildEvenNumberList(5, 6))<br><br>Below is the expected output:<br><br>    []<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildEvenNumberList(1, 101))<br><br>Below is the expected output:<br><br>    [2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def buildEvenNumberList(start, end):<br>    even_number_list = []<br>    for number in range(start, end):<br>        if number % 2 == 0:<br>            even_number_list.append(number)<br>    return even_number_list</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R58" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">59</div></th><td class="s1">06 - &quot;for&quot; Loops</td><td class="s1">07 - Intersect lists</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;intersectLists&quot; that accepts two arguments. Both of these arguments will be lists. Your function should create a new list that contains the elements that are present in both of the lists that were input to the function. Finally, your function should sort the list and return it. (If no values overlap between the two lists, your function should return an empty list.)<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    cancer = [&quot;BRCA1&quot;, &quot;BRCA2&quot;, &quot;EGFR&quot;, &quot;TP53&quot;, &quot;APC&quot;, &quot;FGFR&quot;, &quot;ERBB2&quot;, &quot;MYC&quot;]<br>    alzheimers = [&quot;APOE&quot;, &quot;ROS&quot;, &quot;CDK5&quot;, &quot;LRP&quot;]<br>    heartDisease = [&quot;RARG&quot;, &quot;TIMP1&quot;, &quot;RAC2&quot;, &quot;SCO1&quot;, &quot;ROS&quot;, &quot;TP53&quot;, &quot;SIRT1&quot;, &quot;HADHB&quot;]<br>    diabetes = [&quot;HFE&quot;, &quot;TP53&quot;, &quot;CDK5&quot;, &quot;GAA&quot;, &quot;CRP&quot;, &quot;LMNA&quot;, &quot;EGFR&quot;, &quot;LRP&quot;, &quot;ROS&quot;]<br>    <br>    print(intersectLists(cancer, alzheimers))<br><br>Below is the expected output:<br><br>    []<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    cancer = [&quot;BRCA1&quot;, &quot;BRCA2&quot;, &quot;EGFR&quot;, &quot;TP53&quot;, &quot;APC&quot;, &quot;FGFR&quot;, &quot;ERBB2&quot;, &quot;MYC&quot;]<br>    alzheimers = [&quot;APOE&quot;, &quot;ROS&quot;, &quot;CDK5&quot;, &quot;LRP&quot;]<br>    heartDisease = [&quot;RARG&quot;, &quot;TIMP1&quot;, &quot;RAC2&quot;, &quot;SCO1&quot;, &quot;ROS&quot;, &quot;TP53&quot;, &quot;SIRT1&quot;, &quot;HADHB&quot;]<br>    diabetes = [&quot;HFE&quot;, &quot;TP53&quot;, &quot;CDK5&quot;, &quot;GAA&quot;, &quot;CRP&quot;, &quot;LMNA&quot;, &quot;EGFR&quot;, &quot;LRP&quot;, &quot;ROS&quot;]<br>    <br>    print(intersectLists(cancer, heartDisease))<br><br>Below is the expected output:<br><br>    [&#39;TP53&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    cancer = [&quot;BRCA1&quot;, &quot;BRCA2&quot;, &quot;EGFR&quot;, &quot;TP53&quot;, &quot;APC&quot;, &quot;FGFR&quot;, &quot;ERBB2&quot;, &quot;MYC&quot;]<br>    alzheimers = [&quot;APOE&quot;, &quot;ROS&quot;, &quot;CDK5&quot;, &quot;LRP&quot;]<br>    heartDisease = [&quot;RARG&quot;, &quot;TIMP1&quot;, &quot;RAC2&quot;, &quot;SCO1&quot;, &quot;ROS&quot;, &quot;TP53&quot;, &quot;SIRT1&quot;, &quot;HADHB&quot;]<br>    diabetes = [&quot;HFE&quot;, &quot;TP53&quot;, &quot;CDK5&quot;, &quot;GAA&quot;, &quot;CRP&quot;, &quot;LMNA&quot;, &quot;EGFR&quot;, &quot;LRP&quot;, &quot;ROS&quot;]<br>    <br>    print(intersectLists(cancer, diabetes))<br><br>Below is the expected output:<br><br>    [&#39;EGFR&#39;, &#39;TP53&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    cancer = [&quot;BRCA1&quot;, &quot;BRCA2&quot;, &quot;EGFR&quot;, &quot;TP53&quot;, &quot;APC&quot;, &quot;FGFR&quot;, &quot;ERBB2&quot;, &quot;MYC&quot;]<br>    alzheimers = [&quot;APOE&quot;, &quot;ROS&quot;, &quot;CDK5&quot;, &quot;LRP&quot;]<br>    heartDisease = [&quot;RARG&quot;, &quot;TIMP1&quot;, &quot;RAC2&quot;, &quot;SCO1&quot;, &quot;ROS&quot;, &quot;TP53&quot;, &quot;SIRT1&quot;, &quot;HADHB&quot;]<br>    diabetes = [&quot;HFE&quot;, &quot;TP53&quot;, &quot;CDK5&quot;, &quot;GAA&quot;, &quot;CRP&quot;, &quot;LMNA&quot;, &quot;EGFR&quot;, &quot;LRP&quot;, &quot;ROS&quot;]<br>    <br>    print(intersectLists(alzheimers, heartDisease))<br><br>Below is the expected output:<br><br>    [&#39;ROS&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    cancer = [&quot;BRCA1&quot;, &quot;BRCA2&quot;, &quot;EGFR&quot;, &quot;TP53&quot;, &quot;APC&quot;, &quot;FGFR&quot;, &quot;ERBB2&quot;, &quot;MYC&quot;]<br>    alzheimers = [&quot;APOE&quot;, &quot;ROS&quot;, &quot;CDK5&quot;, &quot;LRP&quot;]<br>    heartDisease = [&quot;RARG&quot;, &quot;TIMP1&quot;, &quot;RAC2&quot;, &quot;SCO1&quot;, &quot;ROS&quot;, &quot;TP53&quot;, &quot;SIRT1&quot;, &quot;HADHB&quot;]<br>    diabetes = [&quot;HFE&quot;, &quot;TP53&quot;, &quot;CDK5&quot;, &quot;GAA&quot;, &quot;CRP&quot;, &quot;LMNA&quot;, &quot;EGFR&quot;, &quot;LRP&quot;, &quot;ROS&quot;]<br>    <br>    print(intersectLists(alzheimers, diabetes))<br><br>Below is the expected output:<br><br>    [&#39;CDK5&#39;, &#39;LRP&#39;, &#39;ROS&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    cancer = [&quot;BRCA1&quot;, &quot;BRCA2&quot;, &quot;EGFR&quot;, &quot;TP53&quot;, &quot;APC&quot;, &quot;FGFR&quot;, &quot;ERBB2&quot;, &quot;MYC&quot;]<br>    alzheimers = [&quot;APOE&quot;, &quot;ROS&quot;, &quot;CDK5&quot;, &quot;LRP&quot;]<br>    heartDisease = [&quot;RARG&quot;, &quot;TIMP1&quot;, &quot;RAC2&quot;, &quot;SCO1&quot;, &quot;ROS&quot;, &quot;TP53&quot;, &quot;SIRT1&quot;, &quot;HADHB&quot;]<br>    diabetes = [&quot;HFE&quot;, &quot;TP53&quot;, &quot;CDK5&quot;, &quot;GAA&quot;, &quot;CRP&quot;, &quot;LMNA&quot;, &quot;EGFR&quot;, &quot;LRP&quot;, &quot;ROS&quot;]<br>    <br>    print(intersectLists(heartDisease, diabetes))<br><br>Below is the expected output:<br><br>    [&#39;ROS&#39;, &#39;TP53&#39;]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def intersectLists(list1, list2):<br>    result = list(set(list1) &amp; set(list2))<br>    result.sort()<br>    return result</td><td class="s2">6</td><td class="s2" dir="ltr">6</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">ChatGPT&#39;s solution uses set logic. This is a succinct way to solve the exercise, but I do not teach sets in the course, so the instructor solution uses a different approach.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R59" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">60</div></th><td class="s1">06 - &quot;for&quot; Loops</td><td class="s1">08 - Evaluate toad family size</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Imagine that you are studying toads who live in the Florida Everglades. You look at many different species and find that when there are at least 5 members in a toad family, that family will be more likely to survive than families with fewer than 5 toads. Now you want to evaluate whether this finding is true in other parts of Florida. You travel to different locations, and you observe family sizes and survival rates.<br><br>Please write a function called &quot;getProportionLargeFamilies&quot; that accepts a single argument: a list of toad-family sizes for specific regions of Florida. Your function should calculate the proportion of families who had at least 5 members in their family and return that value. To calculate the proportion, first identify the number of families with at least 5 members and then divide that number by the total number of families.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    locationA = [6, 4, 4, 2, 9, 14, 12, 2, 3, 8]<br>    <br>    print(getProportionLargeFamilies(locationA))<br><br>Below is the expected output:<br><br>    0.5<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    locationB = [2, 12, 5, 4, 19, 13, 6, 8, 10, 2, 2, 3, 9, 8, 7]<br>    <br>    print(getProportionLargeFamilies(locationB))<br><br>Below is the expected output:<br><br>    0.6666666666666666<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    locationC = [8, 9, 3, 6, 2, 11, 10, 8, 5, 3, 2, 6, 6]<br>    <br>    print(getProportionLargeFamilies(locationC))<br><br>Below is the expected output:<br><br>    0.6923076923076923</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def getProportionLargeFamilies(families):<br>    count = 0<br>    for family in families:<br>        if family &gt;= 5:<br>            count += 1<br>    return count / len(families)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R60" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">61</div></th><td class="s1">06 - &quot;for&quot; Loops</td><td class="s1">09 - Calculate mortality rate</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Consider the example from the previous exercise regarding family sizes of toads. This time write a function called &quot;calculateMortalityRate&quot; that calculates the total number of toads that would perish at a specific location, assuming that families with fewer than 5 toads would be eaten by predators. Your function should accept a single argument: a list of toad-family sizes. Your function should return a float value that indicates the proportion of toads, across all the families at that location, that would be eaten by predators.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    locationA = [6, 4, 4, 2, 9, 14, 12, 2, 3, 8]<br>    <br>    print(calculateMortalityRate(locationA))<br><br>Below is the expected output:<br><br>    0.234375<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    locationB = [2, 12, 5, 4, 19, 13, 6, 8, 10, 2, 2, 3, 9, 8, 7]<br>    <br>    print(calculateMortalityRate(locationB))<br><br>Below is the expected output:<br><br>    0.11818181818181818<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    locationC = [8, 9, 3, 6, 2, 11, 10, 8, 5, 3, 2, 6, 6]<br>    <br>    print(calculateMortalityRate(locationC))<br><br>Below is the expected output:<br><br>    0.12658227848101267</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def calculateMortalityRate(family_sizes):<br>    total_toads = sum(family_sizes)<br>    families_with_less_than_5 = [size for size in family_sizes if size &lt; 5]<br>    toads_lost = sum(families_with_less_than_5)<br>    mortality_rate = toads_lost / total_toads<br>    return mortality_rate</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R61" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">62</div></th><td class="s1">06 - &quot;for&quot; Loops</td><td class="s1">10 - Find indices</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;findIndices&quot; that accepts two arguments: 1) a list and 2) a value. Your function should loop through this list and identify all the values in the list that match the specified value. It should create a new list that contains the indices for these matching values and return that list. If there are no matching values, it should return an empty list.<br><br>Hint(s):<br><br>    You might consider using the &quot;enumerate()&quot; function.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findIndices([&quot;abc&quot;, &quot;def&quot;, &quot;abc&quot;, &quot;xyz&quot;], &quot;abc&quot;))<br><br>Below is the expected output:<br><br>    [0, 2]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findIndices([True, True, True, False, True, False, True], True))<br><br>Below is the expected output:<br><br>    [0, 1, 2, 4, 6]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findIndices([1, 2, 3, 4, 5], 0))<br><br>Below is the expected output:<br><br>    []<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findIndices([], &quot;abc&quot;))<br><br>Below is the expected output:<br><br>    []</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def findIndices(lst, value):<br>    indices = []<br>    for i, item in enumerate(lst):<br>        if item == value:<br>            indices.append(i)<br>    return indices</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R62" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">63</div></th><td class="s1">06 - &quot;for&quot; Loops</td><td class="s1">11 - Profile bison</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;profileBison&quot; that accepts a single argument: a list. Each element in that list will be a list. Each value in each of the &quot;inner&quot; lists will represent the weight (in kilograms) of an individual bison from a herd on Antelope Island (near the Great Salt Lake). Your function should calculate the average weight of each bison herd (rounded to 1 decimal place). Your function should return a list that contains a float value for each herd.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(profileBison([[232.7, 298.1, 283.6, 355.9], [428.2, 119.8, 112.4]]))<br><br>Below is the expected output:<br><br>    [292.6, 220.1]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(profileBison([[292.4, 501.3, 158.6, 199.2, 494.6], [321.2, 366.7], [403.6, 197.9, 123.3]]))<br><br>Below is the expected output:<br><br>    [329.2, 343.9, 241.6]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def profileBison(bison_weights):<br>    herd_averages = []<br>    for herd in bison_weights:<br>        average = round(sum(herd) / len(herd), 1)<br>        herd_averages.append(average)<br>    return herd_averages</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R63" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">64</div></th><td class="s1">06 - &quot;for&quot; Loops</td><td class="s1">12 - Build a protein</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>For this exercise, you will need to write two functions. The first function should be called &quot;translateCodon&quot; and should accept a single argument: a string of 3 nucleotides (a codon). Depending on which codon is used as input, the function should return a letter that represents the amino acid corresponding to this codon. For simplicity, your function will only need to handle 6 codon-to-amino-acid conversions, as shown below:<br><br> CTT = L (Leucine)<br> ATC = I (Isoleucine)<br> GTA = V (Valine)<br> TCT = S (Serine)<br> ACA = T (Threonine)<br> GCC = A (Alanine)<br><br>The second function should be called &quot;buildProtein&quot; and should accept a single argument: a list of string values that contain codon sequences. Your &quot;buildProtein&quot; function should invoke your &quot;translateCodon&quot; function to convert each codon sequence to an amino acid and then build a new list that contains the amino-acid letters corresponding to these codons. Finally, your &quot;buildProtein&quot; function should return this list of amino acids.<br><br>When testing your code, the autograding system will only invoke your &quot;buildProtein&quot; function; it will not invoke your &quot;translateCodon&quot; function directly.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildProtein([&quot;CTT&quot;, &quot;ATC&quot;, &quot;TCT&quot;, &quot;ACA&quot;]))<br><br>Below is the expected output:<br><br>    [&#39;L&#39;, &#39;I&#39;, &#39;S&#39;, &#39;T&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildProtein([&quot;ACA&quot;, &quot;GCC&quot;, &quot;CTT&quot;, &quot;CTT&quot;]))<br><br>Below is the expected output:<br><br>    [&#39;T&#39;, &#39;A&#39;, &#39;L&#39;, &#39;L&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildProtein([&quot;TCT&quot;, &quot;ACA&quot;, &quot;ATC&quot;, &quot;CTT&quot;, &quot;CTT&quot;]))<br><br>Below is the expected output:<br><br>    [&#39;S&#39;, &#39;T&#39;, &#39;I&#39;, &#39;L&#39;, &#39;L&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildProtein([&quot;ATC&quot;, &quot;ACA&quot;]))<br><br>Below is the expected output:<br><br>    [&#39;I&#39;, &#39;T&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(buildProtein([&quot;GTA&quot;, &quot;GTA&quot;, &quot;TCT&quot;, &quot;GCC&quot;, &quot;ACA&quot;, &quot;ATC&quot;, &quot;GCC&quot;, &quot;GTA&quot;, &quot;CTT&quot;]))<br><br>Below is the expected output:<br><br>    [&#39;V&#39;, &#39;V&#39;, &#39;S&#39;, &#39;A&#39;, &#39;T&#39;, &#39;I&#39;, &#39;A&#39;, &#39;V&#39;, &#39;L&#39;]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 11, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def translateCodon(codon):<br>    if codon == &quot;CTT&quot;:<br>        return &quot;L&quot;<br>    elif codon == &quot;ATC&quot;:<br>        return &quot;I&quot;<br>    elif codon == &quot;GTA&quot;:<br>        return &quot;V&quot;<br>    elif codon == &quot;TCT&quot;:<br>        return &quot;S&quot;<br>    elif codon == &quot;ACA&quot;:<br>        return &quot;T&quot;<br>    elif codon == &quot;GCC&quot;:<br>        return &quot;A&quot;<br><br>def buildProtein(codons):<br>    amino_acids = []<br>    for codon in codons:<br>        amino_acids.append(translateCodon(codon))<br>    return amino_acids</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R64" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">65</div></th><td class="s1">07 - Problem Solving</td><td class="s1">01 - Count unique animals</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose that you work for the forest service and your job is to observe wildlife. You make some observations and then want to summarize the data using Python code.<br><br>Please write a function called &quot;countUnique&quot; that accepts one argument. This will be a list that indicates the types of animals that you observed on a given trip. It may have duplicate values. For example, if the first two animals you observed were deer, the first two elements in the list would be &quot;deer&quot;.<br><br>Your function should return the number of unique animals that you observed on a given trip.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    trip1 = [&quot;rabbit&quot;, &quot;lizard&quot;, &quot;deer&quot;, &quot;turkey&quot;, &quot;rabbit&quot;, &quot;deer&quot;, &quot;rabbit&quot;]<br>    <br>    print(countUnique(trip1))<br><br>Below is the expected output:<br><br>    4<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    trip2 = [&quot;lizard&quot;, &quot;lizard&quot;, &quot;rabbit&quot;, &quot;rabbit&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;muskrat&quot;, &quot;elk&quot;, &quot;lizard&quot;, &quot;deer&quot;]<br>    <br>    print(countUnique(trip2))<br><br>Below is the expected output:<br><br>    5<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    trip3 = [&quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;]<br>    <br>    print(countUnique(trip3))<br><br>Below is the expected output:<br><br>    1</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def countUnique(trip):<br>    unique_animals = set(trip)<br>    return len(unique_animals)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">ChatGPT uses set logic, which is a succinct way to accomplish this problem. However, for simplicity, we do not teach set logic in this course.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R65" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">66</div></th><td class="s1">07 - Problem Solving</td><td class="s1">02 - Number of repeated animal sightings</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose that you work for the forest service and your job is to observe wildlife. You make some observations and then want to summarize the data using Python code.<br><br>Please write a function called &quot;countSightings&quot; that accepts one argument. This will be a list that indicates the types of animals that you observed on a given trip. It may have duplicate values. For example, if the first two animals you observed were deer, the first two elements in the list would be &quot;deer&quot;.<br><br>Your function should return the number of times you observed the same type of animal consecutively (twice in a row) on a given trip. If you observed the same animal 3 times in a row, that would count as 2 times that you observed that animal consecutively. If you observed the same animal 4 times in a row, that would count as 3 times that you observed that animal consecutively.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    trip1 = [&quot;rabbit&quot;, &quot;lizard&quot;, &quot;deer&quot;, &quot;turkey&quot;, &quot;rabbit&quot;, &quot;deer&quot;, &quot;rabbit&quot;]<br>    <br>    print(countSightings(trip1))<br><br>Below is the expected output:<br><br>    0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    trip2 = [&quot;lizard&quot;, &quot;lizard&quot;, &quot;rabbit&quot;, &quot;rabbit&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;muskrat&quot;, &quot;elk&quot;, &quot;lizard&quot;, &quot;deer&quot;]<br>    <br>    print(countSightings(trip2))<br><br>Below is the expected output:<br><br>    3<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    trip3 = [&quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;]<br>    <br>    print(countSightings(trip3))<br><br>Below is the expected output:<br><br>    9</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def countSightings(trip):<br>    count = 0<br>    for i in range(len(trip) - 1):<br>        if trip[i] == trip[i + 1]:<br>            count += 1<br>    return count</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R66" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">67</div></th><td class="s1">07 - Problem Solving</td><td class="s1">03 - Calculate average weight of rabbits</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose that you work for the forest service and your job is to observe wildlife. You make some observations and then want to summarize the data using Python code.<br><br>Please write a function called &quot;calcRabbitAve&quot; that accepts two arguments. The first argument will be a list that indicates the types of animals that you observed on one of the trips. This list will have some duplicate values. For example, if the first two animals you observed were deer, the first two elements in that list would be &quot;deer&quot;. The second argument will be a list that indicates the weight of each animal that you observed. These will be float values. The first (0th) element in this list will be the weight of the first animal that you observed. The second element in this list will be the weight of the second animal that you observed, etc.<br><br>Your function should return the average weight of the rabbits that you observed on a given trip.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animals1 = [&quot;rabbit&quot;, &quot;lizard&quot;, &quot;deer&quot;, &quot;turkey&quot;, &quot;rabbit&quot;, &quot;deer&quot;, &quot;rabbit&quot;]<br>    <br>    weights1 = [5.2, 3.4, 80.1, 25.4, 5.9, 89.6, 7.4]<br>    <br>    print(calcRabbitAve(animals1, weights1))<br><br>Below is the expected output:<br><br>    6.166666666666667<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animals2 = [&quot;lizard&quot;, &quot;lizard&quot;, &quot;rabbit&quot;, &quot;rabbit&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;muskrat&quot;, &quot;elk&quot;, &quot;lizard&quot;, &quot;deer&quot;]<br>    <br>    weights2 = [1.4, 1.7, 4.4, 4.9, 77.2, 52.8, 13.0, 200.9, 0.4, 99.8]<br>    <br>    print(calcRabbitAve(animals2, weights2))<br><br>Below is the expected output:<br><br>    4.65</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def calcRabbitAve(animals, weights):<br>    total_weight = 0<br>    count = 0<br>    for i in range(len(animals)):<br>        if animals[i] == &#39;rabbit&#39;:<br>            total_weight += weights[i]<br>            count += 1<br>    return total_weight / count if count != 0 else 0</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">ChatGPT&#39;s solution uses an inline if/else statement, but we do not teach that technique in the course.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R67" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">68</div></th><td class="s1">07 - Problem Solving</td><td class="s1">04 - Find lightest animal</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose that you work for the forest service and your job is to observe wildlife. You make some observations and then want to summarize the data using Python code.<br><br>Please write a function called &quot;findLightest&quot; that accepts two arguments. The first argument will be a list that indicates the types of animals that you observed on one of the trips. This list will have some duplicate values. For example, if the first two animals you observed were deer, the first two elements in that list would be &quot;deer&quot;. The second argument will be a list that indicates the weight of each animal that you observed. These will be float values. The first (0th) element in this list will be the weight of the first animal that you observed. The second element in this list will be the weight of the second animal that you observed, etc.<br><br>Your function should return the name of the individual animal that has the lowest weight on a given trip.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animals1 = [&quot;rabbit&quot;, &quot;lizard&quot;, &quot;deer&quot;, &quot;turkey&quot;, &quot;rabbit&quot;, &quot;deer&quot;, &quot;rabbit&quot;]<br>    <br>    weights1 = [5.2, 3.4, 80.1, 25.4, 5.9, 89.6, 7.4]<br>    <br>    print(findLightest(animals1, weights1))<br><br>Below is the expected output:<br><br>    lizard<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animals2 = [&quot;lizard&quot;, &quot;lizard&quot;, &quot;rabbit&quot;, &quot;rabbit&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;muskrat&quot;, &quot;elk&quot;, &quot;lizard&quot;, &quot;deer&quot;]<br>    <br>    weights2 = [1.4, 1.7, 4.4, 4.9, 77.2, 52.8, 13.0, 200.9, 0.4, 99.8]<br>    <br>    print(findLightest(animals2, weights2))<br><br>Below is the expected output:<br><br>    lizard<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animals3 = [&quot;mosquito&quot;, &quot;bear&quot;, &quot;mouse&quot;]<br>    <br>    weights3 = [0.1, 491.2, 2.7]<br>    <br>    print(findLightest(animals3, weights3))<br><br>Below is the expected output:<br><br>    mosquito</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def findLightest(animals, weights):<br>    min_weight = float(&#39;inf&#39;)<br>    min_animal = None<br>    for i in range(len(weights)):<br>        if weights[i] &lt; min_weight:<br>            min_weight = weights[i]<br>            min_animal = animals[i]<br>    return min_animal</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R68" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">69</div></th><td class="s1">07 - Problem Solving</td><td class="s1">05 - Count heavy deer</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose that you work for the forest service and your job is to observe wildlife. You make some observations and then want to summarize the data using Python code.<br><br>Please write a function called &quot;countHeavyDeer&quot; that accepts two arguments. The first argument will be a list that indicates the types of animals that you observed on one of the trips. This list will have some duplicate values. For example, if the first two animals you observed were deer, the first two elements in that list would be &quot;deer&quot;. The second argument will be a list that indicates the weight of each animal that you observed. These will be float values. The first (0th) element in this list will be the weight of the first animal that you observed. The second element in this list will be the weight of the second animal that you observed, etc.<br><br>Your function should return the number of times you observed a deer that weighed more than 80.0 (kilograms).<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animals1 = [&quot;rabbit&quot;, &quot;lizard&quot;, &quot;deer&quot;, &quot;turkey&quot;, &quot;rabbit&quot;, &quot;deer&quot;, &quot;rabbit&quot;]<br>    <br>    weights1 = [5.2, 3.4, 80.1, 25.4, 5.9, 89.6, 7.4]<br>    <br>    print(countHeavyDeer(animals1, weights1))<br><br>Below is the expected output:<br><br>    2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animals2 = [&quot;lizard&quot;, &quot;lizard&quot;, &quot;rabbit&quot;, &quot;rabbit&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;muskrat&quot;, &quot;elk&quot;, &quot;lizard&quot;, &quot;deer&quot;]<br>    <br>    weights2 = [1.4, 1.7, 4.4, 4.9, 77.2, 52.8, 13.0, 200.9, 0.4, 99.8]<br>    <br>    print(countHeavyDeer(animals2, weights2))<br><br>Below is the expected output:<br><br>    1<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animals3 = [&quot;mosquito&quot;, &quot;bear&quot;, &quot;mouse&quot;]<br>    <br>    weights3 = [0.1, 491.2, 2.7]<br>    <br>    print(countHeavyDeer(animals3, weights3))<br><br>Below is the expected output:<br><br>    0</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def countHeavyDeer(animals, weights):<br>    count = 0<br>    for i in range(len(animals)):<br>        if animals[i] == &quot;deer&quot; and weights[i] &gt; 80.0:<br>            count += 1<br>    return count</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R69" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">70</div></th><td class="s1">07 - Problem Solving</td><td class="s1">06 - Categorize animals by weight</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Below are the average weights of 5 types of small mammals:<br><br> Squirrel: 1-1.5 lbs<br> Ferret: 2-4 lbs<br> Rabbit: 5-7 lbs<br> Cat: 8-10 lbs<br> Small dog: 11-20 lbs<br><br>Here is a list of the weights of some examples of the above mentioned animals.<br><br>    animalWeights = [3.5, 15, 8.3, 22, 1.2, 4, 6, 17.5, 4.5]<br><br>Write a function called &quot;findAnimalsByWeight&quot; that accepts a list such as &quot;animalWeights&quot;, checks each weight in the list, returns a list that contains the name of the animal that most likely corresponds to each weight. For example, an animal that weighs 3.5 lbs. is probably a ferret, and an animal that weights 15 lbs. is probably a small dog, etc.<br><br>If the weight does not fit into any of the ranges listed above, return &quot;None&quot;.<br><br>(It might be useful to create a second function that accepts a given weight and prints what animal the weight corresponds to.)<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findAnimalsByWeight([99, -1, 13.3838, 11.0001, 10.9999]))<br><br>Below is the expected output:<br><br>    [None, None, &#39;Small dog&#39;, &#39;Small dog&#39;, None]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findAnimalsByWeight([9, -1, 13.3838, 11.0001, 10.9999]))<br><br>Below is the expected output:<br><br>    [&#39;Cat&#39;, None, &#39;Small dog&#39;, &#39;Small dog&#39;, None]</td><td class="s1" dir="ltr">Thanks to Ashlyn Powell who helped create this exercise.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def findAnimalByWeight(weight):<br>    if weight &gt;= 1 and weight &lt;= 1.5:<br>        return &#39;Squirrel&#39;<br>    elif weight &gt;= 2 and weight &lt;= 4:<br>        return &#39;Ferret&#39;<br>    elif weight &gt;= 5 and weight &lt;= 7:<br>        return &#39;Rabbit&#39;<br>    elif weight &gt;= 8 and weight &lt;= 10:<br>        return &#39;Cat&#39;<br>    elif weight &gt;= 11 and weight &lt;= 20:<br>        return &#39;Small dog&#39;<br>    else:<br>        return None<br><br>def findAnimalsByWeight(weights):<br>    return [findAnimalByWeight(weight) for weight in weights]</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">ChatGPTs&#39; first solution returned &quot;None&quot; rather than None. I asked it to adjust for this, and it worked on the second try.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R70" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">71</div></th><td class="s1">07 - Problem Solving</td><td class="s1">07 - Count flowers by color</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Below is a list of strings that represent flower colors:<br><br>    flowerColors = [ &#39;pink&#39;, &#39;red&#39;, &#39;red&#39;, &#39;blue&#39;, &#39;white&#39;, &#39;pink&#39;, &#39;white&#39;, &#39;red&#39;, &#39;purple&#39;, &#39;white&#39;, &#39;red&#39;, &#39;blue&#39;]<br><br>Write a function called &quot;printFlowerSummary&quot; that accepts a single argument. This will be a list that contains strings representing flower names. Your function should calculate the total number of times that each color of flower was observed, as well as the total number of flowers. Finally, it should print a summary of these counts. For example, for the above list, it would print the following summary:<br><br>    2 pink flower(s)<br>    4 red flower(s)<br>    2 blue flower(s)<br>    3 white flower(s)<br>    1 purple flower(s)<br>    There are 12 flowers total<br><br>For this exercise, there are five possible flower colors: pink, red, blue, white, and purple. Your code should print a count for each color. If no flowers were observed for a particular color, indicate that there were 0.<br><br>Note: This is one of the few exercises in which your function is not fruitful. In other words, it does not return anything. Instead it prints the output.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    printFlowerSummary([ &#39;pink&#39;, &#39;red&#39;, &#39;red&#39;, &#39;blue&#39;, &#39;white&#39;, &#39;pink&#39;, &#39;white&#39;, &#39;red&#39;, &#39;purple&#39;, &#39;white&#39;, &#39;red&#39;, &#39;blue&#39;])<br><br>Below is the expected output:<br><br>    2 pink flower(s)<br>    4 red flower(s)<br>    2 blue flower(s)<br>    3 white flower(s)<br>    1 purple flower(s)<br>    There are 12 flowers total<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    printFlowerSummary([ &#39;pink&#39;, &#39;red&#39;, &quot;white&quot;]  50 + [&quot;blue&quot;])<br><br>Below is the expected output:<br><br>    50 pink flower(s)<br>    50 red flower(s)<br>    1 blue flower(s)<br>    50 white flower(s)<br>    0 purple flower(s)<br>    There are 151 flowers total</td><td class="s1" dir="ltr">Thanks to Ashlyn Powell who helped create this exercise.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def printFlowerSummary(flowerList):<br>    flowerColors = [&#39;pink&#39;, &#39;red&#39;, &#39;blue&#39;, &#39;white&#39;, &#39;purple&#39;]<br>    flowerCounts = {}<br>    totalFlowers = 0<br>    <br>    for color in flowerColors:<br>        flowerCounts[color] = flowerList.count(color)<br>        totalFlowers += flowerCounts[color]<br>    <br>    for color in flowerColors:<br>        if flowerCounts[color] == 0:<br>            print(f&#39;0 {color} flower(s)&#39;)<br>        else:<br>            print(f&#39;{flowerCounts[color]} {color} flower(s)&#39;)<br>    <br>    print(f&#39;There are {totalFlowers} flowers total&#39;)</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">4</td><td class="s1" dir="ltr">ChatGPT&#39;s first solution printed either &quot;flower&quot; or &quot;flowers&quot; depending on how many there were. However, the prompt asked it to always print &quot;flower(s)&quot;. I asked it to fix this. After 3 more attempts, it produced a correct solution. However, I realized that the prompt was not fully clear. It did not clearly state that the function should look for 5 specific colors and print 0 if none were observed for a particular color. I clarified that in the prompt, and ChatGPT adjusted its solution accordingly. Because the prompt was not clear, I did not penalize ChatGPT for this.<br><br>The solution uses dictionaries and string formatting, but we have not covered them at this point in the course. </td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr"></td></tr><tr style="height: 20px"><th id="2089702959R71" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">72</div></th><td class="s1">07 - Problem Solving</td><td class="s1">08 - Summarize safari animals</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose you went on a safari trip in Africa and wrote down every animal that you saw, in the order that you saw them. Now you want to condense your list so that you only have each animal written down once.<br><br>Please write a function called &quot;summarizeSafari&quot; that accepts a single argument. It will be a list that contains animal names. Your function should return a sorted list indicating the unique animals that you saw.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(summarizeSafari([&#39;lion&#39;, &#39;giraffe&#39;, &#39;zebra&#39;, &#39;lion&#39;, &#39;elephant&#39;, &#39;cheetah&#39;, &#39;lion&#39;, &#39;zebra&#39;, &#39;elephant&#39;, &#39;zebra&#39;, &#39;giraffe&#39;, &#39;lion&#39;, &#39;cheetah&#39;]))<br><br>Below is the expected output:<br><br>    [&#39;cheetah&#39;, &#39;elephant&#39;, &#39;giraffe&#39;, &#39;lion&#39;, &#39;zebra&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(summarizeSafari([&#39;warthog&#39;]  32 + [&#39;hippo&#39;]  15 + [&#39;elephant&#39;]  7 + [&#39;lion&#39;]  5 + [&#39;pigeon&#39;]  502))<br><br>Below is the expected output:<br><br>    [&#39;elephant&#39;, &#39;hippo&#39;, &#39;lion&#39;, &#39;pigeon&#39;, &#39;warthog&#39;]</td><td class="s1" dir="ltr">Thanks to Ashlyn Powell who helped create this exercise.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def summarizeSafari(animals):<br>    unique_animals = set(animals)<br>    return sorted(list(unique_animals))</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">ChatGPT uses set logic, which is a succinct way to accomplish this problem. However, for simplicity, we do not teach set logic in this course.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R72" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">73</div></th><td class="s1">07 - Problem Solving</td><td class="s1">09 - Check Tic Tac Toe board</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;didWinGame&quot; that accepts one argument: a nested list. This nested list will consist of an &quot;outer&quot; list that has 3 &quot;inner&quot; lists within it. Each inner list will have 3 string values (each will be either &quot;X&quot; or &quot;O&quot;). Below is an example:<br><br>    [[&quot;X&quot;, &quot;O&quot;, &quot;X&quot;], [&quot;O&quot;, &quot;O&quot;, &quot;X&quot;], [&quot;O&quot;, &quot;X&quot;, &quot;X&quot;]] <br><br>You could think of this as being like a Tic-Tac-Toe board. To illustrate this better, we can change the formatting:<br><br>    [[&quot;X&quot;, &quot;O&quot;, &quot;X&quot;],<br>     [&quot;O&quot;, &quot;O&quot;, &quot;X&quot;],<br>     [&quot;O&quot;, &quot;X&quot;, &quot;X&quot;]] <br><br>In the example above, the &quot;X&quot; player would win Tic-Tac-Toe because she had 3 consecutive &quot;X&quot; values (going vertical in the third column).<br><br>Your function should return a Boolean value that indicates whether there are 3 consecutive &quot;X&quot; values in the specified nested list. The 3 consecutive &quot;X&quot; values could be either vertical or horizontal. For simplicity, only focus on the &quot;X&quot; values (ignoring whether there are 3 consecutive &quot;O&quot; values). Also, for simplicity, ignore the fact that there could be 3 consecutive &quot;X&quot; values along the diagonals.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    game1 = [[&quot;X&quot;, &quot;O&quot;, &quot;X&quot;],<br>             [&quot;O&quot;, &quot;O&quot;, &quot;X&quot;],<br>             [&quot;O&quot;, &quot;X&quot;, &quot;X&quot;]]<br>    <br>    print(didWinGame(game1))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    game2 = [[&quot;O&quot;, &quot;O&quot;, &quot;X&quot;],<br>             [&quot;X&quot;, &quot;O&quot;, &quot;O&quot;],<br>             [&quot;O&quot;, &quot;X&quot;, &quot;X&quot;]]<br>    <br>    print(didWinGame(game2))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    game3 = [[&quot;O&quot;, &quot;X&quot;, &quot;O&quot;],<br>             [&quot;O&quot;, &quot;O&quot;, &quot;X&quot;],<br>             [&quot;X&quot;, &quot;X&quot;, &quot;X&quot;]]<br>    <br>    print(didWinGame(game3))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    game4 = [[&quot;O&quot;, &quot;X&quot;, &quot;O&quot;],<br>             [&quot;O&quot;, &quot;X&quot;, &quot;X&quot;],<br>             [&quot;X&quot;, &quot;O&quot;, &quot;O&quot;]]<br>    <br>    print(didWinGame(game4))<br><br>Below is the expected output:<br><br>    False</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def didWinGame(game):<br>    for row in game:<br>        if row == [&quot;X&quot;, &quot;X&quot;, &quot;X&quot;]:<br>            return True<br><br>    for i in range(3):<br>        if game[0][i] == &quot;X&quot; and game[1][i] == &quot;X&quot; and game[2][i] == &quot;X&quot;:<br>            return True<br><br>    return False</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R73" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">74</div></th><td class="s1">08 - DNA Analysis</td><td class="s1">01 - Check bird DNA sequences</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>A research team studying birds in Indonesia collected DNA samples from 2 different species of birds. As they analyzed their data, they discovered specific sequences that were unique to each of the two species. Unfortunately, a computer glitch mixed up some of their collected data.<br><br>Write a function called &quot;checkDNA&quot; that accepts one argument (a list). This list will contain multiple strings; each string will represent a nucleotide (A, C, G, or T).<br><br>Your function should perform two actions:<br><br>1. Check that the list has valid nucleotide values (only A, C, G, and T). If the list has any invalid value, return &quot;The input was invalid&quot;.<br>2. If the sequence is valid, your function should evaluate whether the sequence came from species 1 or species 2. The reference sequence for species 1 is &quot;ACTGAGGTC&quot;. The reference sequence for species 2 is &quot;ACTTAGGTC&quot;. Return &quot;Species 1&quot; if the sequence came from species 1. Return &quot;Species 2&quot; if the sequence came from species 2. Return &quot;No match&quot; if the sequence came from neither species.<br><br>Hint(s):<br><br>    If you use the &quot;==&quot; operator with two lists, it equates to &quot;True&quot; if all of the elements in the two lists have the same value (and if the sizes of the two lists are identical).<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkDNA([&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;]))<br><br>Below is the expected output:<br><br>    Species 2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkDNA([&quot;A&quot;, &quot;C&quot;, &quot;Q&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;]))<br><br>Below is the expected output:<br><br>    The input was invalid<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkDNA([&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;]))<br><br>Below is the expected output:<br><br>    No match<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkDNA([&quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;]))<br><br>Below is the expected output:<br><br>    No match<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkDNA([&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;]))<br><br>Below is the expected output:<br><br>    Species 1<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkDNA([&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;C&quot;]))<br><br>Below is the expected output:<br><br>    No match<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkDNA([&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;, &quot;Q&quot;, &quot;C&quot;, &quot;T&quot;]))<br><br>Below is the expected output:<br><br>    The input was invalid</td><td class="s1" dir="ltr">Thanks to Eric Upton-Rowley who helped create this exercise.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def checkDNA(dna):<br>    valid_nucleotides = [&quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;]<br>    reference_seq_1 = &quot;ACTGAGGTC&quot;<br>    reference_seq_2 = &quot;ACTTAGGTC&quot;<br><br>    for nucleotide in dna:<br>        if nucleotide not in valid_nucleotides:<br>            return &quot;The input was invalid&quot;<br>    <br>    if dna == list(reference_seq_1):<br>        return &quot;Species 1&quot;<br>    elif dna == list(reference_seq_2):<br>        return &quot;Species 2&quot;<br>    else:<br>        return &quot;No match&quot;</td><td class="s2">7</td><td class="s2" dir="ltr">7</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R74" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">75</div></th><td class="s1">08 - DNA Analysis</td><td class="s1">02 - Identify DNA Variants</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>DNA sequences differ between individuals. Thus, although there is a &quot;reference&quot; human genome, your personal genome differs from the reference in many places. Some locations of DNA have frequent variation within a population. In this problem, you will be provided with short DNA sequences representing the reference genome and another sequence representing an individual person&#39;s DNA. Your goal is to find the positions that differ between the reference sequence and the individual&#39;s sequence.<br><br>You should create a function called &quot;compareSequences&quot;. It should accept two lists. Both of these lists will contain multiple strings. The length of the two lists will be the same. Each string in the lists will contain &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, or &quot;T&quot;, representing nucleotides. The first list will represent a reference DNA sequence. The second list will represent the DNA sequence of a non-reference individual.<br><br>Your function should return a list. Elements of this list should be the indices (positions) of each nucleotide in which the two sequences differ from each other. For example, if the reference allele at position 6 were a &quot;T&quot; and the individual had an &quot;A&quot;, you would include 6 in the list that you return.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    referenceDNA = [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;C&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;C&quot;, &quot;C&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;]<br>    individualDNA = [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;C&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;G&quot;, &quot;C&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;]<br>    <br>    print(compareSequences(referenceDNA, individualDNA))<br><br>Below is the expected output:<br><br>    [4, 12, 27, 33, 36]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    referenceDNA = [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;C&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;C&quot;, &quot;C&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;]<br>    individualDNA = [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;, &quot;C&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;C&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;]<br>    <br>    print(compareSequences(referenceDNA, individualDNA))<br><br>Below is the expected output:<br><br>    [1, 5, 12, 15, 19, 27, 30, 31, 32, 33, 36]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def compareSequences(referenceDNA, individualDNA):<br>    result = []<br>    for i in range(len(referenceDNA)):<br>        if referenceDNA[i] != individualDNA[i]:<br>            result.append(i)<br>    return result</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R75" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">76</div></th><td class="s1">08 - DNA Analysis</td><td class="s1">03 - Identify somatic mutations</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Cancer cells contain many DNA variants that differ from the reference sequence. Some of these mutations are present in all of a person&#39;s cells (germline mutations) and others are specific to tumor cells (somatic mutations). Somatic mutations are typically what cause tumors to arise. To find somatic mutations (and better understand how to treat tumors), we compare an individual&#39;s germline DNA to a reference genome; we also compare the DNA sequence of the tumor to the reference.<br><br>This exercise builds on the previous one. The function (&quot;compareSequences&quot;) you wrote in the previous exercise finds differences between two DNA sequences. You should copy-and-paste that function to here.<br><br>Your goal in this exercise is to find somatic mutations. Please write a new function called &quot;findSomaticMutations&quot;. It should accept three arguments. Each of these arguments will be lists. The first list will contain string values representing the reference DNA. The second list will contain string values representing the germline DNA. The third list will contain string values representing the tumor (somatic) DNA. Your new function should use the &quot;compareSequences&quot; function to identify the positions of germline mutations (germline vs. reference) and somatic mutations (somatic vs. reference). Then it should identify positions that differ between the germline and somatic mutations. Your function should return a sorted list of these positions.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    referenceDNA = [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;C&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;C&quot;, &quot;C&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;]<br>    germlineDNA =  [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;C&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;G&quot;, &quot;C&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;]<br>    somaticDNA =   [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;C&quot;, &quot;A&quot;, &quot;G&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;G&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;]<br>    <br>    print(findSomaticMutations(referenceDNA, germlineDNA, somaticDNA))<br><br>Below is the expected output:<br><br>    [6, 11, 26, 37]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    referenceDNA = [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;C&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;C&quot;, &quot;C&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;]<br>    germlineDNA =  [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;, &quot;C&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;C&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;]<br>    somaticDNA =   [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;, &quot;C&quot;, &quot;C&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;, &quot;G&quot;, &quot;T&quot;, &quot;A&quot;, &quot;C&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;]<br>    <br>    print(findSomaticMutations(referenceDNA, germlineDNA, somaticDNA))<br><br>Below is the expected output:<br><br>    [11, 25, 28, 35, 41]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def findSomaticMutations(referenceDNA, germlineDNA, somaticDNA):<br>    def compareSequences(reference, sequence):<br>        mutations = []<br>        for i in range(len(reference)):<br>            if reference[i] != sequence[i]:<br>                mutations.append(i)<br>        return mutations<br><br>    germlineMutations = compareSequences(referenceDNA, germlineDNA)<br>    somaticMutations = compareSequences(referenceDNA, somaticDNA)<br>    somaticOnlyMutations = [i for i in somaticMutations if i not in germlineMutations]<br>    return sorted(somaticOnlyMutations)</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">7</td><td class="s1" dir="ltr">ChatGPT&#39;s first solution did not produce the expected output. I reminded it of the expected output, and it <span style="font-weight:bold;">insisted</span> that the expected output was incorrect and proceded to say that the code was correct because it was producing a sequence of prime numbers, even though the prompt had nothing to do with prime numbers.<br><br>The third solution was correct, but the outputs were not sorted correctly. I asked it to fix this, and it offered a solution about how to sort a list of numbers, rather than modifying its previous solution. I asked it to incorporate the sorting logic into the previous solution. On attempt #5, it gave me a solution related to somatic mutations that did not address the original problem, and the function name was incorrect. To double check, I asked it to change the function name, but the solution was still invalid, and it appeared to be in a different programming language.<br><br>At this point, I modified the prompt to indicate that the outputs should be sorted and gave it the full prompt. Then it produced a passing solution.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td></tr><tr style="height: 20px"><th id="2089702959R76" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">77</div></th><td class="s1">08 - DNA Analysis</td><td class="s1">04 - Quantify shared alleles</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;quantifySharedAlleles&quot;. This function should accept three arguments. This argument will be a list. Each element of that list with be a list (resulting in a nested list). The inner lists will contain strings, each representing a nucleotide. Each of these lists contains the DNA sequence of a person in a population. The second argument will be an integer; it will refer to the position of one person (sub-list) from the population. The third argument will be another integer; it will also refer to the position of a person (sub-list) from the population.<br><br>Your function should retrieve the list corresponding to the second argument and assign it to a variable. Your function should retrieve the list corresponding to the third argument and assign it to a variable. Your function should then calculate the number of alleles that are identical for those two individuals and return that number.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    populationSequences = [<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;T&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]]<br>    <br>    print(quantifySharedAlleles(populationSequences, 0, 1))<br><br>Below is the expected output:<br><br>    12<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    populationSequences = [<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;T&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]]<br>    <br>    print(quantifySharedAlleles(populationSequences, 1, 3))<br><br>Below is the expected output:<br><br>    10<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    populationSequences = [<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;T&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]]<br>    <br>    print(quantifySharedAlleles(populationSequences, 5, 9))<br><br>Below is the expected output:<br><br>    11<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    populationSequences = [<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;T&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]]<br>    <br>    print(quantifySharedAlleles(populationSequences, 7, 13))<br><br>Below is the expected output:<br><br>    11</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def quantifySharedAlleles(populationSequences, person1, person2):<br>    person1_seq = populationSequences[person1]<br>    person2_seq = populationSequences[person2]<br>    count = 0<br>    for i in range(len(person1_seq)):<br>        if person1_seq[i] == person2_seq[i]:<br>            count += 1<br>    return count</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R77" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">78</div></th><td class="s1">08 - DNA Analysis</td><td class="s1">05 - Summarize unique nucleotides</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;summarizeNucleotides&quot;. This function should accept three arguments. This argument will be a list. Each element of that list with be a list (resulting in a nested list). The inner lists will contain strings, each representing a nucleotide. Each of these lists contains the DNA sequence of a person in a population. The second argument will be an integer; it will refer to the position of one person (sub-list) from the population. The third argument will be another integer; it will also refer to the position of a person (sub-list) from the population.<br><br>Your function should retrieve the list corresponding to the second argument and assign it to a variable. Your function should retrieve the list corresponding to the third argument and assign it to a variable. Your function should then identify which nucleotides are observed at least once in both individuals and return a list of those nucleotides in sorted order.<br><br>Consider this simple example:<br><br>    populationSequences = [[&quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;]]<br><br>The first individual has at least one A, T, and G (but no C). The second individual has at least one A and G (but no C or T). Your function would return &quot;[&quot;A&quot;, &quot;G&quot;]&quot; because both individuals have at least one of those nucleotides. (The first individual has a T, but the second one does not.)<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    populationSequences = [[&quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;]]<br>    <br>    print(summarizeNucleotides(populationSequences, 0, 1))<br><br>Below is the expected output:<br><br>    [&#39;A&#39;, &#39;G&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    populationSequences = [[&quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;A&quot;, &quot;G&quot;, &quot;T&quot;],<br>                           [&quot;T&quot;, &quot;C&quot;, &quot;A&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;]]<br>    <br>    print(summarizeNucleotides(populationSequences, 0, 1))<br><br>Below is the expected output:<br><br>    [&#39;A&#39;, &#39;G&#39;, &#39;T&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    populationSequences = [[&quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;A&quot;, &quot;G&quot;, &quot;T&quot;],<br>                           [&quot;T&quot;, &quot;C&quot;, &quot;A&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;]]<br>    <br>    print(summarizeNucleotides(populationSequences, 4, 2))<br><br>Below is the expected output:<br><br>    [&#39;A&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    populationSequences = [[&quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;A&quot;, &quot;G&quot;, &quot;T&quot;],<br>                           [&quot;T&quot;, &quot;C&quot;, &quot;A&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;A&quot;, &quot;A&quot;, &quot;A&quot;],<br>                           [&quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;]]<br>    <br>    print(summarizeNucleotides(populationSequences, 3, 5))<br><br>Below is the expected output:<br><br>    [&#39;A&#39;, &#39;C&#39;, &#39;G&#39;, &#39;T&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    populationSequences = [[&quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;],<br>                           [&quot;T&quot;, &quot;C&quot;, &quot;A&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;],<br>                           [&quot;A&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;]]<br>    <br>    print(summarizeNucleotides(populationSequences, 2, 4))<br><br>Below is the expected output:<br><br>    [&#39;A&#39;, &#39;C&#39;, &#39;T&#39;]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def summarizeNucleotides(populationSequences, i, j):<br>    individual1 = populationSequences[i]<br>    individual2 = populationSequences[j]<br>    nucleotides = set(individual1) &amp; set(individual2)<br>    return sorted(list(nucleotides))</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">ChatGPT uses set logic, which is a succinct way to accomplish this problem. However, for simplicity, we do not teach set logic in this course.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R78" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">79</div></th><td class="s1">08 - DNA Analysis</td><td class="s1">06 - Calculate minor allele frequency</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Geneticists seek to understand the extent to which DNA mutations have spread throughout a population. By comparing a large group of people to a reference sequence, they can calculate the fraction of people who have a non-reference nucleotide. This number is often called the &quot;minor allele frequency&quot; (MAF). If an allele is relatively common, it is less likely to cause a disease (otherwise, lots of people in the population would have the disease).<br><br>Please create a function called &quot;calculateMAF&quot;. This function should accept three arguments. The first argument will be a list. It will contain string values, each representing a nucleotide. This is the reference DNA sequence. The second argument will be a list. Each element of that list with be a list (resulting in a nested list). The inner lists will contain strings, each representing a nucleotide. Each of these lists contains the DNA sequence of a person in a population. The third argument will be an integer that refers to a specific position within the reference sequence (and the population sequences).<br><br>Your function should return the proportion of non-reference alleles at the specified position. (Technically, the MAF references to the frequency of one particular non-reference allele at a particular position. But this exercise is simplified so that you are calculating the frequency of any non-reference allele at a particular position.)<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    referenceSequence = [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]<br>    <br>    populationSequences = [<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;T&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]]<br>    <br>    print(calculateMAF(referenceSequence, populationSequences, 3))<br><br>Below is the expected output:<br><br>    0.05<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    referenceSequence = [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]<br>    <br>    populationSequences = [<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;T&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]]<br>    <br>    print(calculateMAF(referenceSequence, populationSequences, 5))<br><br>Below is the expected output:<br><br>    0.0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    referenceSequence = [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]<br>    <br>    populationSequences = [<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;T&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]]<br>    <br>    print(calculateMAF(referenceSequence, populationSequences, 7))<br><br>Below is the expected output:<br><br>    0.2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    referenceSequence = [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]<br>    <br>    populationSequences = [<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;T&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]]<br>    <br>    print(calculateMAF(referenceSequence, populationSequences, 9))<br><br>Below is the expected output:<br><br>    0.25<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    referenceSequence = [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]<br>    <br>    populationSequences = [<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;T&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]]<br>    <br>    print(calculateMAF(referenceSequence, populationSequences, 12))<br><br>Below is the expected output:<br><br>    0.05</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def calculateMAF(referenceSequence, populationSequences, position):<br>    reference_allele = referenceSequence[position]<br>    non_reference_alleles = [sequence[position] for sequence in populationSequences if sequence[position] != reference_allele]<br>    return len(non_reference_alleles) / len(populationSequences)</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R79" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">80</div></th><td class="s1">08 - DNA Analysis</td><td class="s1">07 - Find most variable genome position</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Building on the previous exercise, your next step is to write code that calculates the minor allele frequency (MAF) for every position in a DNA sequence, instead of just one specific position.<br><br>Copy your &quot;calculateMAF&quot; function from the previous exercise. As described below, you will need to create a second function that invokes the &quot;calculateMAF&quot; function.<br><br>Please write a function called &quot;findMostVariablePosition&quot; that accepts two arguments. The first argument will be a list. It will contain string values, each representing a nucleotide. This is the reference DNA sequence. The second argument will be a list. Each element of that list with be a list (resulting in a nested list). The inner lists will contain strings, each representing a nucleotide. Each of these lists contains the DNA sequence of a person in a population.<br><br>Your function should identify which nucleotide position has the highest minor allele frequency (the highest percentage of non-reference alleles). Your function should return the position (index) that has the highest minor allele frequency.<br><br>Important: When your function is tested, the system will invoke your &quot;findMostVariablePosition&quot; function. It will not invoke your &quot;calculateMAF&quot; function directly. Your &quot;findMostVariablePosition&quot; function should invoke your &quot;calculateMAF&quot; function when making calculations.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    referenceSequence = [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]<br>    <br>    populationSequences = [<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;T&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;C&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;C&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;G&quot;, &quot;G&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;],<br>    [&quot;A&quot;, &quot;A&quot;, &quot;T&quot;, &quot;A&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;G&quot;, &quot;C&quot;, &quot;T&quot;, &quot;T&quot;, &quot;T&quot;, &quot;A&quot;]]<br>    <br>    print(findMostVariablePosition(referenceSequence, populationSequences))<br><br>Below is the expected output:<br><br>    9</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def calculateMAF(reference, sample):<br>    count = 0<br>    for i in range(len(reference)):<br>        if reference[i] != sample[i]:<br>            count += 1<br>    return count / len(reference)<br><br>def findMostVariablePosition(reference, population):<br>    mafs = []<br>    for i in range(len(reference)):<br>        position_maf = 0<br>        for sample in population:<br>            position_maf += calculateMAF(reference[i:i+1], sample[i:i+1])<br>        mafs.append(position_maf / len(population))<br>    return mafs.index(max(mafs))</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">3</td><td class="s1" dir="ltr">The first solution&#39;s output did not match the expected output. I explained this. It gave me a solution that did not address the original prompt. I asked it to try again and it succeeded.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td></tr><tr style="height: 20px"><th id="2089702959R80" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">81</div></th><td class="s1">09 - Strings</td><td class="s1">01 - Remove middle of string</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;removeMiddleOfString&quot; that accepts a single argument: a string. Your function should retain the first 2 and last 2 characters of the string and return that value. Any characters in the middle should be removed. For example, if the input string were &quot;peanut&quot;, your function would return &quot;peut&quot;.<br><br>If the input string contains 4 or fewer characters, the entire input string should be returned.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(removeMiddleOfString(&quot;alligator&quot;))<br><br>Below is the expected output:<br><br>    alor<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(removeMiddleOfString(&quot;crocodile&quot;))<br><br>Below is the expected output:<br><br>    crle<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(removeMiddleOfString(&quot;giraffe&quot;))<br><br>Below is the expected output:<br><br>    gife<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(removeMiddleOfString(&quot;lion&quot;))<br><br>Below is the expected output:<br><br>    lion<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(removeMiddleOfString(&quot;s&quot;))<br><br>Below is the expected output:<br><br>    s</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def removeMiddleOfString(inputString):<br>    length = len(inputString)<br>    if length &lt;= 4:<br>        return inputString<br>    return inputString[0:2] + inputString[-2:]</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R81" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">82</div></th><td class="s1">09 - Strings</td><td class="s1">02 - Count upper-case characters</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;countUpperCase&quot; that accepts a single argument: a string. Your function should indicate how many characters in that string are upper-case characters. For example, if the string were &quot;Kevin Worthen&quot;, the function would return 2.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(countUpperCase(&quot;Kevin Worthen&quot;))<br><br>Below is the expected output:<br><br>    2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(countUpperCase(&quot;Old McDonald&quot;))<br><br>Below is the expected output:<br><br>    3<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(countUpperCase(&quot;BIOLOGY&quot;))<br><br>Below is the expected output:<br><br>    7<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(countUpperCase(&quot;lizard&quot;))<br><br>Below is the expected output:<br><br>    0</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def countUpperCase(string):<br>    count = 0<br>    for char in string:<br>        if char.isupper():<br>            count += 1<br>    return count</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R82" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">83</div></th><td class="s1">09 - Strings</td><td class="s1">03 - Add zeroes to string</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;addZeroes&quot; that accepts two arguments. The first argument will be a string. The second argument will be an integer. Even though the first argument is a string, it will contain a numeric value (for example, &quot;33&quot;). Your function should add zeroes to the beginning of this string so that the length of that string is equal to the specified integer argument. For example, if the input arguments were &quot;33&quot; and 5, your function would return &quot;00033&quot;.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addZeroes(&quot;33&quot;, 5))<br><br>Below is the expected output:<br><br>    00033<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addZeroes(&quot;48&quot;, 15))<br><br>Below is the expected output:<br><br>    000000000000048<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addZeroes(&quot;1&quot;, 2))<br><br>Below is the expected output:<br><br>    01<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addZeroes(&quot;999&quot;, 3))<br><br>Below is the expected output:<br><br>    999</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def addZeroes(string, n):<br>    zeroes = n - len(string)<br>    return &quot;0&quot; * zeroes + string if zeroes &gt; 0 else string</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R83" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">84</div></th><td class="s1">09 - Strings</td><td class="s1">04 - Cut DNA</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Restriction enzymes are used in laboratories to remove a piece of DNA from a chromosome so that it can be used for molecular cloning. These enzymes recognize specific restriction sites in the DNA, which can be used as starting and stopping points for cutting the DNA.<br><br>Please write a function called &quot;cutDNA&quot; that accepts three arguments: 1) a string that represents a DNA sequence within a chromosome, 2) a string that represents the first restriction site, and 3) a string that represents the second restriction site. Your function should return the DNA sequence in between the two restriction sites (the part that would be cut from the DNA sequence).<br><br>Suppose your chromosome sequence (first argument) were &quot;ACTTAATTTACCAGGAGCTAGGGACCATTACACGTGGATCGAAAGGCGCATTA&quot;, your first restriction site were &quot;AGGGACC&quot;, and your second restriction site were &quot;GGCGCAT&quot;. Your function would return &quot;ATTACACGTGGATCGAAA&quot;.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    chromsomalDNA = &quot;ACTTAATTTACCAGGAGCTAGGGACCATTACACGTGGATCGAAAGGCGCATTA&quot;<br>    restrictionSite1 = &quot;AGGGACC&quot;<br>    restrictionSite2 = &quot;GGCGCAT&quot;<br>    <br>    print(cutDNA(chromsomalDNA, restrictionSite1, restrictionSite2))<br><br>Below is the expected output:<br><br>    ATTACACGTGGATCGAAA<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    chromsomalDNA = &quot;TATACCTAAAGGCGTACAGGAATTTTTAAAACCCGGAGACTCTGAGATAG&quot;<br>    restrictionSite1 = &quot;AATTTTTAA&quot;<br>    restrictionSite2 = &quot;GAGAT&quot;<br>    <br>    print(cutDNA(chromsomalDNA, restrictionSite1, restrictionSite2))<br><br>Below is the expected output:<br><br>    AACCCGGAGACTCT</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def cutDNA(chromosomalDNA, restrictionSite1, restrictionSite2):<br>    start = chromosomalDNA.find(restrictionSite1) + len(restrictionSite1)<br>    end = chromosomalDNA.find(restrictionSite2)<br>    return chromosomalDNA[start:end]</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R84" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">85</div></th><td class="s4">09 - Strings</td><td class="s4">05 - Is CpG island</td><td class="freezebar-cell"></td><td class="s4" dir="ltr">Here is a new Python programming task.<br><br>The human genome consists of 3 billion A, C, G, and T nucleotides. Some parts of the genome are known as &quot;promoter&quot; regions. These regions typically lie upstream of genes and regulate gene expression. Promoter regions typically contain &quot;CpG islands,&quot; which have a relatively high number of CG pairs (a C nucleotide followed by a G nucleotide). If 10% or more of the DNA in a region consists of these CG pairs, that region may be classified as a &quot;CpG&quot; island.<br><br>Please write a function called &quot;isCpgIsland&quot; that accepts a single argument: a string. This string will contain a DNA sequence. Your function should calculate the percentage of nucleotides in the DNA sequence that are CG pairs. For example, if the DNA sequence were ACGACGAATT, the percentage would be 40. Return a Boolean value that indicates whether the DNA sequence is a CpG island (*greater than* 10%).<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isCpgIsland(&quot;ATACATCGGCCGCACGTAATCGCGAATA&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isCpgIsland(&quot;GCGGGCATATAGACTAGCTAGCATTAACGGCAT&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(isCpgIsland(&quot;CTAGATGACCATACTATCCAGCAGCTGCAGGAGCTATA&quot;))<br><br>Below is the expected output:<br><br>    False</td><td class="s4" dir="ltr"></td><td class="s4" dir="ltr">Yes</td><td class="s4" dir="ltr">Yes</td><td class="s4">text</td><td class="s4" dir="ltr">SRP</td><td class="s5" dir="ltr">February 13, 2023</td><td class="s4" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s4" dir="ltr">def isCpgIsland(sequence):<br>    cg_pairs = 0<br>    for i in range(len(sequence) - 1):<br>        if sequence[i:i+2] == &quot;CG&quot;:<br>            cg_pairs += 1<br>    return cg_pairs / len(sequence) &gt;= 0.1</td><td class="s5">3</td><td class="s5" dir="ltr">2</td><td class="s5" dir="ltr">10</td><td class="s4" dir="ltr">I tried all sorts of ways to get it to produce a passing solution. However, when the number of CG pairs was exactly 10% of the pairs, it caused problems. It often used &gt;= rather than &gt; in its logic. I tried emphasizing that it should be greater than rather than greater than or equal to. But it could not figure this out.<br><br>The solution here is the last solution.</td><td class="s4"></td><td class="s4" dir="ltr">Yes</td><td class="s4"></td><td class="s4"></td><td class="s4" dir="ltr">No</td><td class="s4"></td><td class="s4"></td><td class="s4"></td><td class="s4" dir="ltr">Yes</td><td class="s4"></td><td class="s4"></td></tr><tr style="height: 20px"><th id="2089702959R85" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">86</div></th><td class="s1">09 - Strings</td><td class="s1">06 - Mask sequencing reads</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Current DNA sequencers produce data called &quot;sequencing reads.&quot; Each sequencing read is a string of characters (A, C, G, or T) that represents a stretch of nucleotides that has been sequenced. Alongside each of these sequencing reads is another string of the same length as the sequencing read. This second string contains quality scores for each nucleotide in the sequencing read. Suppose that quality scores of 2 or lower indicate poor quality (the DNA sequencer was unable to accurately determine which nucleotide it was). For example, if the sequencing read were &quot;ATACAGA&quot; and the quality string were &quot;5234891&quot;, the 2nd and 7th nucleotides would be considered poor quality.<br><br>Please write a function called &quot;maskSequencingRead&quot; that accepts two arguments. Each of these arguments will be strings. The first string will be a sequencing read. The second string will contain quality scores associated with that read. Your function should return a new string that is a modified version of the sequencing read. To do this, your function should check the sequencing read and quality scores and replace low-quality nucleotides with &quot;N&quot;. For example, if the sequencing read were &quot;ATACAGA&quot; and the quality string were &quot;5234891&quot;, your function would return &quot;ANACAGN&quot;.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(maskSequencingRead(&quot;ATACAGA&quot;, &quot;5234891&quot;))<br><br>Below is the expected output:<br><br>    ANACAGN<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(maskSequencingRead(&quot;GACGACGTTACGGCAAACCA&quot;, &quot;99998798889589594211&quot;))<br><br>Below is the expected output:<br><br>    GACGACGTTACGGCAAANNN<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(maskSequencingRead(&quot;GCCGGGATAGCGACAGAAAC&quot;, &quot;12121233111212113579&quot;))<br><br>Below is the expected output:<br><br>    NNNNNNATNNNNNNNNAAAC</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def maskSequencingRead(sequencingRead, qualityScores):<br>    maskedRead = &quot;&quot;<br>    for i in range(len(sequencingRead)):<br>        if int(qualityScores[i]) &lt;= 2:<br>            maskedRead += &quot;N&quot;<br>        else:<br>            maskedRead += sequencingRead[i]<br>    return maskedRead<br></td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R86" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">87</div></th><td class="s1">09 - Strings</td><td class="s1">07 - Book-end words</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;bookendWords&quot; that accepts a single argument: a string. This string will be a phrase in the English language. Your function should check each word in the phrase. If a word is longer than two characters, your function should convert the first and last letters of the word to upper-case characters. Your function should return this modified string.<br><br>Note: Words will be separated by spaces. Do not worry about punctuation for this exercise.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(bookendWords(&quot;the farmer in the dell&quot;))<br><br>Below is the expected output:<br><br>    ThE FarmeR in ThE DelL<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(bookendWords(&quot;the cat in the hat&quot;))<br><br>Below is the expected output:<br><br>    ThE CaT in ThE HaT<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(bookendWords(&quot;bioinformatics is fun&quot;))<br><br>Below is the expected output:<br><br>    BioinformaticS is FuN</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def bookendWords(phrase):<br>    words = phrase.split()<br>    modified_words = []<br>    for word in words:<br>        if len(word) &gt; 2:<br>            modified_word = word[0].upper() + word[1:-1] + word[-1].upper()<br>            modified_words.append(modified_word)<br>        else:<br>            modified_words.append(word)<br>    return &#39; &#39;.join(modified_words)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R87" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">88</div></th><td class="s1">09 - Strings</td><td class="s1">08 - Question everything</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;questionEverything&quot; that accepts a single argument: a string. Your function should create a new string where all upper-case characters have been changed to lower-case characters and all lower-case characters have been changed to upper-case characters. It should also replace any non-alphanumeric characters (other than spaces) with a &quot;?&quot; character. For example, if your input string were &quot;Hello, World!&quot;, your function would return &quot;hELLO? wORLD?&quot;<br><br>Hint(s):<br><br>    Use the &quot;upper()&quot;, &quot;lower()&quot;, &quot;isupper()&quot;, &quot;islower()&quot;, and &quot;isalnum()&quot; functions.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(questionEverything(&quot;Hello, World!&quot;))<br><br>Below is the expected output:<br><br>    hELLO? wORLD?<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(questionEverything(&quot;Single-step method of RNA isolation by acid guanidinium thiocyanate-phenol-chloroform extraction.&quot;))<br><br>Below is the expected output:<br><br>    sINGLE?STEP METHOD OF rna ISOLATION BY ACID GUANIDINIUM THIOCYANATE?PHENOL?CHLOROFORM EXTRACTION?<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(questionEverything(&quot;A simple method, for the isolation and purification of total lipides, from animal tissues.&quot;))<br><br>Below is the expected output:<br><br>    a SIMPLE METHOD? FOR THE ISOLATION AND PURIFICATION OF TOTAL LIPIDES? FROM ANIMAL TISSUES?<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(questionEverything(&quot;Clustal W: improving the sensitivity of progressive multiple sequence alignment through sequence weighting, position-specific gap penalties and weight matrix choice.&quot;))<br><br>Below is the expected output:<br><br>    cLUSTAL w? IMPROVING THE SENSITIVITY OF PROGRESSIVE MULTIPLE SEQUENCE ALIGNMENT THROUGH SEQUENCE WEIGHTING? POSITION?SPECIFIC GAP PENALTIES AND WEIGHT MATRIX CHOICE?<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(questionEverything(&quot;Basic local alignment search tool.&quot;))<br><br>Below is the expected output:<br><br>    bASIC LOCAL ALIGNMENT SEARCH TOOL?</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def questionEverything(inputString):<br>    result = &quot;&quot;<br>    for char in inputString:<br>        if char.isalnum():<br>            if char.isupper():<br>                result += char.lower()<br>            else:<br>                result += char.upper()<br>        else:<br>            if char == &quot; &quot;:<br>                result += char<br>            else:<br>                result += &quot;?&quot;<br>    return result</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R88" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">89</div></th><td class="s1">09 - Strings</td><td class="s1">09 - Does string start or end with a particular string</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;startsOrEndsWith&quot; that accepts three arguments: 1) a list of string values, 2) a list of &quot;starting&quot; string values, and 3) a list of &quot;ending&quot; string values. Your function should return a new list that includes only the elements from the input list (argument #1) that start with any of the specified starting values (argument #2) or that end with any of the specified ending values (argument #3). If a string starts with one of the specified starting values and ends with one of the specified ending values, that string should only be appended once to the new list.<br><br>Your function should sort the new list and then return it.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animalNames = [&quot;lion&quot;, &quot;deer&quot;, &quot;rat&quot;, &quot;mouse&quot;, &quot;honey badger&quot;, &quot;elephant&quot;, &quot;blue whale&quot;, &quot;antelope&quot;, &quot;trout&quot;, &quot;great white shark&quot;, &quot;grouper&quot;, &quot;salmon&quot;, &quot;guppy&quot;, &quot;bone fish&quot;, &quot;skink&quot;, &quot;gecko&quot;, &quot;viper&quot;, &quot;scheltopusik&quot;]<br>    <br>    print(startsOrEndsWith(animalNames, [&quot;l&quot;, &quot;d&quot;], [&quot;n&quot;, &quot;y&quot;]))<br><br>Below is the expected output:<br><br>    [&#39;deer&#39;, &#39;guppy&#39;, &#39;lion&#39;, &#39;salmon&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animalNames = [&quot;lion&quot;, &quot;deer&quot;, &quot;rat&quot;, &quot;mouse&quot;, &quot;honey badger&quot;, &quot;elephant&quot;, &quot;blue whale&quot;, &quot;antelope&quot;, &quot;trout&quot;, &quot;great white shark&quot;, &quot;grouper&quot;, &quot;salmon&quot;, &quot;guppy&quot;, &quot;bone fish&quot;, &quot;skink&quot;, &quot;gecko&quot;, &quot;viper&quot;, &quot;scheltopusik&quot;]<br>    <br>    print(startsOrEndsWith(animalNames, [&quot;g&quot;], [&quot;n&quot;]))<br><br>Below is the expected output:<br><br>    [&#39;gecko&#39;, &#39;great white shark&#39;, &#39;grouper&#39;, &#39;guppy&#39;, &#39;lion&#39;, &#39;salmon&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animalNames = [&quot;lion&quot;, &quot;deer&quot;, &quot;rat&quot;, &quot;mouse&quot;, &quot;honey badger&quot;, &quot;elephant&quot;, &quot;blue whale&quot;, &quot;antelope&quot;, &quot;trout&quot;, &quot;great white shark&quot;, &quot;grouper&quot;, &quot;salmon&quot;, &quot;guppy&quot;, &quot;bone fish&quot;, &quot;skink&quot;, &quot;gecko&quot;, &quot;viper&quot;, &quot;scheltopusik&quot;]<br>    <br>    print(startsOrEndsWith(animalNames, [&quot;a&quot;, &quot;b&quot;, &quot;c&quot;, &quot;d&quot;], [&quot;e&quot;, &quot;f&quot;, &quot;g&quot;, &quot;h&quot;]))<br><br>Below is the expected output:<br><br>    [&#39;antelope&#39;, &#39;blue whale&#39;, &#39;bone fish&#39;, &#39;deer&#39;, &#39;mouse&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animalNames = [&quot;lion&quot;, &quot;deer&quot;, &quot;rat&quot;, &quot;mouse&quot;, &quot;honey badger&quot;, &quot;elephant&quot;, &quot;blue whale&quot;, &quot;antelope&quot;, &quot;trout&quot;, &quot;great white shark&quot;, &quot;grouper&quot;, &quot;salmon&quot;, &quot;guppy&quot;, &quot;bone fish&quot;, &quot;skink&quot;, &quot;gecko&quot;, &quot;viper&quot;, &quot;scheltopusik&quot;]<br>    <br>    print(startsOrEndsWith(animalNames, [&quot;q&quot;, &quot;x&quot;, &quot;z&quot;], [&quot;m&quot;, &quot;c&quot;, &quot;d&quot;]))<br><br>Below is the expected output:<br><br>    []</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def startsOrEndsWith(strings, starts, ends):<br>    result = []<br>    for string in strings:<br>        if any(string.startswith(start) for start in starts) or any(string.endswith(end) for end in ends):<br>            result.append(string)<br>    result = sorted(set(result))<br>    return result<br></td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R89" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">90</div></th><td class="s1">09 - Strings</td><td class="s1">10 - Describe genome</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;describeGenome&quot; that accepts two arguments. The first argument will be a string and will be the (non-scientific) name of a species. The second argument will be an integer and will indicate how many pairs of chromosomes are present in each normal cell for this species.<br><br>Your function should return a string that indicates how many pairs of chromosomes are present in the specified genome. For example, for humans, your function would return, &quot;The human genome has 23 pairs of chromosomes.&quot;<br><br>Hint(s):<br><br>    Use f-string(s).<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(&quot;fruit fly&quot;, 4))<br><br>Below is the expected output:<br><br>    The fruit fly genome has 4 pairs of chromosomes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(&quot;meadow vole&quot;, 21))<br><br>Below is the expected output:<br><br>    The meadow vole genome has 21 pairs of chromosomes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(&quot;Stalked Adder’s tongue&quot;, 1260))<br><br>Below is the expected output:<br><br>    The Stalked Adder’s tongue genome has 1260 pairs of chromosomes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(&quot;platypus&quot;, 99))<br><br>Below is the expected output:<br><br>    The platypus genome has 99 pairs of chromosomes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(&quot;Swainson&#39;s Thrush&quot;, 41))<br><br>Below is the expected output:<br><br>    The Swainson&#39;s Thrush genome has 41 pairs of chromosomes.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def describeGenome(species, chromosome_pairs):<br>    return f&quot;The {species} genome has {chromosome_pairs} pairs of chromosomes.&quot;<br><br>print(describeGenome(&quot;fruit fly&quot;, 4))<br>print(describeGenome(&quot;meadow vole&quot;, 21))<br>print(describeGenome(&quot;Stalked Adder’s tongue&quot;, 1260))<br>print(describeGenome(&quot;platypus&quot;, 99))<br>print(describeGenome(&quot;Swainson&#39;s Thrush&quot;, 41))<br></td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The solution included code for testing the function. I ignored that.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R90" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">91</div></th><td class="s1">09 - Strings</td><td class="s1">11 - Calculate average chromosome size</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;calcAveChromSize&quot; that accepts three arguments. The first argument will be a string and will be the (non-scientific) name of a species. The second argument will be an integer and will indicate how many pairs of chromosomes are present in each normal cell for this species. The third argument will be the overall size (in megabases) of the genome for this species.<br><br>Your function should return a string that indicates the average size (in megabases) for whichever species was provided. The average size should be rounded to 2 decimal places, as shown in the expected output.<br><br>Hint(s):<br><br>    Use string formatting to construct each string.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calcAveChromSize(&quot;fruit fly&quot;, 4, 127.2))<br><br>Below is the expected output:<br><br>    The fruit fly genome has a total of 8 chromosomes, which average 15.90 megabases in size.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calcAveChromSize(&quot;meadow vole&quot;, 21, 2258.81))<br><br>Below is the expected output:<br><br>    The meadow vole genome has a total of 42 chromosomes, which average 53.78 megabases in size.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calcAveChromSize(&quot;Stalked Adder’s tongue&quot;, 1260, 6829))<br><br>Below is the expected output:<br><br>    The Stalked Adder’s tongue genome has a total of 2520 chromosomes, which average 2.71 megabases in size.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calcAveChromSize(&quot;platypus&quot;, 99, 1980))<br><br>Below is the expected output:<br><br>    The platypus genome has a total of 198 chromosomes, which average 10.00 megabases in size.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calcAveChromSize(&quot;Swainson&#39;s Thrush&quot;, 41, 1132))<br><br>Below is the expected output:<br><br>    The Swainson&#39;s Thrush genome has a total of 82 chromosomes, which average 13.80 megabases in size.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def calcAveChromSize(species, num_pairs, genome_size):<br>    total_chromosomes = num_pairs * 2<br>    average_size = genome_size / total_chromosomes<br>    return f&quot;The {species} genome has a total of {total_chromosomes} chromosomes, which average {average_size:.2f} megabases in size.&quot;</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">The first solution showed one decimal place when a number had a zero at the end (e.g., &quot;15.9&quot; instead of &quot;15.90&quot;. I asked it to fix this, and it did.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R91" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">92</div></th><td class="s1">09 - Strings</td><td class="s1">12 - Find longest word</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;findLongestWord&quot; that accepts a single argument: a string. This string will be an English sentence. Your function should identify the longest word in the sentence, convert it to lower case, and return it. If multiple words tie for the longest, your function should return the following: &quot;unable to determine longest word due to a tie&quot;.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findLongestWord(&quot;The Origin of Species&quot;))<br><br>Below is the expected output:<br><br>    species<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findLongestWord(&quot;On the Perpetuation of Varieties and Species by Natural Means of Selection&quot;))<br><br>Below is the expected output:<br><br>    perpetuation<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findLongestWord(&quot;The Correlation Between Relatives on the Supposition of Mendelian Inheritance&quot;))<br><br>Below is the expected output:<br><br>    unable to determine longest word due to a tie<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findLongestWord(&quot;The physical aspects of the living cell&quot;))<br><br>Below is the expected output:<br><br>    physical<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findLongestWord(&quot;De humani corporis fabrica libri septem&quot;))<br><br>Below is the expected output:<br><br>    corporis<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findLongestWord(&quot;Cat hit dog&quot;))<br><br>Below is the expected output:<br><br>    unable to determine longest word due to a tie</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def findLongestWord(sentence):<br>    words = sentence.split()<br>    longest_word = &quot;&quot;<br>    for word in words:<br>        if len(word) &gt; len(longest_word):<br>            longest_word = word<br>    longest_words = [word for word in words if len(word) == len(longest_word)]<br>    if len(longest_words) &gt; 1:<br>        return &quot;unable to determine longest word due to a tie&quot;<br>    else:<br>        return longest_word.lower()</td><td class="s2">6</td><td class="s2" dir="ltr">6</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">The first solution produced incorrect outputs for two of the tests. I gave some details about what the inputs and outputs were when it failed, and it produced a correct solution.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R92" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">93</div></th><td class="s1">10 - Dictionaries</td><td class="s1">01 - Create dictionary of animals</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;createAnimalDict&quot;. Your function should accept no arguments.<br><br>Your function should create a dictionary. Each key in the dictionary should be the name of an adult animal. Each value in the dictionary should be the term used to describe the baby of a given animal. Your dictionary should contain the following (adult) animal names: dog, cat, horse, pig, bear, and alligator. The baby names for each of these are puppy, kitten, foal, piglet, cub, and hatchling, respectively. Your function should return this dictionary.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(createAnimalDict())<br><br>Below is the expected output:<br><br>    {&#39;dog&#39;: &#39;puppy&#39;, &#39;cat&#39;: &#39;kitten&#39;, &#39;horse&#39;: &#39;foal&#39;, &#39;pig&#39;: &#39;piglet&#39;, &#39;bear&#39;: &#39;cub&#39;, &#39;alligator&#39;: &#39;hatchling&#39;}</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def createAnimalDict():<br>    animal_dict = {<br>        &#39;dog&#39;: &#39;puppy&#39;,<br>        &#39;cat&#39;: &#39;kitten&#39;,<br>        &#39;horse&#39;: &#39;foal&#39;,<br>        &#39;pig&#39;: &#39;piglet&#39;,<br>        &#39;bear&#39;: &#39;cub&#39;,<br>        &#39;alligator&#39;: &#39;hatchling&#39;<br>    }<br>    return animal_dict</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R93" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">94</div></th><td class="s1">10 - Dictionaries</td><td class="s1">02 - Get baby name</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;getBabyName&quot;. This function should accept two arguments: 1) a dictionary and 2) a string. The dictionary will specify adult animal names as the keys. The values will be the respective baby names (as in the dictionary you created in the previous exercise). The second argument will be one of the adult animal names.<br><br>Your function should return a string indicating the baby name corresponding to the specified adult name (as shown in the expected output). If there are no matches for the adult name in the dictionary, return &quot;No match&quot;.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animalDict = {&quot;dog&quot;: &quot;puppy&quot;, &quot;cat&quot;: &quot;kitten&quot;, &quot;horse&quot;: &quot;foal&quot;, &quot;pig&quot;: &quot;piglet&quot;, &quot;bear&quot;: &quot;cub&quot;, &quot;alligator&quot;: &quot;hatchling&quot;}<br>    <br>    print(getBabyName(animalDict, &quot;dog&quot;))<br><br>Below is the expected output:<br><br>    The baby name for dog is puppy.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animalDict = {&quot;dog&quot;: &quot;puppy&quot;, &quot;cat&quot;: &quot;kitten&quot;, &quot;horse&quot;: &quot;foal&quot;, &quot;pig&quot;: &quot;piglet&quot;, &quot;bear&quot;: &quot;cub&quot;, &quot;alligator&quot;: &quot;hatchling&quot;}<br>    <br>    print(getBabyName(animalDict, &quot;cat&quot;))<br><br>Below is the expected output:<br><br>    The baby name for cat is kitten.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animalDict = {&quot;dog&quot;: &quot;puppy&quot;, &quot;cat&quot;: &quot;kitten&quot;, &quot;horse&quot;: &quot;foal&quot;, &quot;pig&quot;: &quot;piglet&quot;, &quot;bear&quot;: &quot;cub&quot;, &quot;alligator&quot;: &quot;hatchling&quot;}<br>    <br>    print(getBabyName(animalDict, &quot;horse&quot;))<br><br>Below is the expected output:<br><br>    The baby name for horse is foal.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animalDict = {&quot;dog&quot;: &quot;puppy&quot;, &quot;cat&quot;: &quot;kitten&quot;, &quot;horse&quot;: &quot;foal&quot;, &quot;pig&quot;: &quot;piglet&quot;, &quot;bear&quot;: &quot;cub&quot;, &quot;alligator&quot;: &quot;hatchling&quot;}<br>    <br>    print(getBabyName(animalDict, &quot;pig&quot;))<br><br>Below is the expected output:<br><br>    The baby name for pig is piglet.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animalDict = {&quot;dog&quot;: &quot;puppy&quot;, &quot;cat&quot;: &quot;kitten&quot;, &quot;horse&quot;: &quot;foal&quot;, &quot;pig&quot;: &quot;piglet&quot;, &quot;bear&quot;: &quot;cub&quot;, &quot;alligator&quot;: &quot;hatchling&quot;}<br>    <br>    print(getBabyName(animalDict, &quot;bear&quot;))<br><br>Below is the expected output:<br><br>    The baby name for bear is cub.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animalDict = {&quot;dog&quot;: &quot;puppy&quot;, &quot;cat&quot;: &quot;kitten&quot;, &quot;horse&quot;: &quot;foal&quot;, &quot;pig&quot;: &quot;piglet&quot;, &quot;bear&quot;: &quot;cub&quot;, &quot;alligator&quot;: &quot;hatchling&quot;}<br>    <br>    print(getBabyName(animalDict, &quot;alligator&quot;))<br><br>Below is the expected output:<br><br>    The baby name for alligator is hatchling.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animalDict = {&quot;dog&quot;: &quot;puppy&quot;, &quot;cat&quot;: &quot;kitten&quot;, &quot;horse&quot;: &quot;foal&quot;, &quot;pig&quot;: &quot;piglet&quot;, &quot;bear&quot;: &quot;cub&quot;, &quot;alligator&quot;: &quot;hatchling&quot;}<br>    <br>    print(getBabyName(animalDict, &quot;crocodile&quot;))<br><br>Below is the expected output:<br><br>    No match</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def getBabyName(d, animal):<br>    if animal in d:<br>        return &quot;The baby name for &quot; + animal + &quot; is &quot; + d[animal] + &quot;.&quot;<br>    else:<br>        return &quot;No match&quot;</td><td class="s2">7</td><td class="s2" dir="ltr">7</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R94" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">95</div></th><td class="s1">10 - Dictionaries</td><td class="s1">03 - Merge dictionaries</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;mergeDictionaries&quot; that accepts two arguments. Each of these arguments will be dictionary variables. Your function should use for loop(s) to merge these two dictionaries, creating a new dictionary that contains the keys and values from both of these dictionaries. Finally, your function should return the merged dictionary.<br><br>Important: Create a completely new dictionary rather than modifying one of the dictionaries that you are merging. To verify this, the tests will print the original dictionaries as well as the merged one.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    merged = mergeDictionaries(habitats1, habitats2)<br>    <br>    print(&quot;Original dictionaries:&quot;)<br>    print(habitats1)<br>    print(habitats2)<br>    <br>    print(&quot;Merged dictionary&quot;)<br>    print(merged)<br><br>Below is the expected output:<br><br>    Original dictionaries:<br>    {&#39;duck&#39;: &#39;pond&#39;, &#39;horse&#39;: &#39;field&#39;, &#39;cow&#39;: &#39;pasture&#39;}<br>    {&#39;deer&#39;: &#39;forest&#39;, &#39;elk&#39;: &#39;forest&#39;, &#39;wolf&#39;: &#39;forest&#39;}<br>    Merged dictionary<br>    {&#39;duck&#39;: &#39;pond&#39;, &#39;horse&#39;: &#39;field&#39;, &#39;cow&#39;: &#39;pasture&#39;, &#39;deer&#39;: &#39;forest&#39;, &#39;elk&#39;: &#39;forest&#39;, &#39;wolf&#39;: &#39;forest&#39;}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    merged = mergeDictionaries(habitats1, habitats3)<br>    <br>    print(&quot;Original dictionaries:&quot;)<br>    print(habitats1)<br>    print(habitats3)<br>    <br>    print(&quot;Merged dictionary&quot;)<br>    print(merged)<br><br>Below is the expected output:<br><br>    Original dictionaries:<br>    {&#39;duck&#39;: &#39;pond&#39;, &#39;horse&#39;: &#39;field&#39;, &#39;cow&#39;: &#39;pasture&#39;}<br>    {&#39;student&#39;: &#39;library&#39;, &#39;trout&#39;: &#39;lake&#39;, &#39;shark&#39;: &#39;ocean&#39;}<br>    Merged dictionary<br>    {&#39;duck&#39;: &#39;pond&#39;, &#39;horse&#39;: &#39;field&#39;, &#39;cow&#39;: &#39;pasture&#39;, &#39;student&#39;: &#39;library&#39;, &#39;trout&#39;: &#39;lake&#39;, &#39;shark&#39;: &#39;ocean&#39;}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    merged = mergeDictionaries(habitats2, habitats3)<br>    <br>    print(&quot;Original dictionaries:&quot;)<br>    print(habitats2)<br>    print(habitats3)<br>    <br>    print(&quot;Merged dictionary&quot;)<br>    print(merged)<br><br>Below is the expected output:<br><br>    Original dictionaries:<br>    {&#39;deer&#39;: &#39;forest&#39;, &#39;elk&#39;: &#39;forest&#39;, &#39;wolf&#39;: &#39;forest&#39;}<br>    {&#39;student&#39;: &#39;library&#39;, &#39;trout&#39;: &#39;lake&#39;, &#39;shark&#39;: &#39;ocean&#39;}<br>    Merged dictionary<br>    {&#39;deer&#39;: &#39;forest&#39;, &#39;elk&#39;: &#39;forest&#39;, &#39;wolf&#39;: &#39;forest&#39;, &#39;student&#39;: &#39;library&#39;, &#39;trout&#39;: &#39;lake&#39;, &#39;shark&#39;: &#39;ocean&#39;}</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def mergeDictionaries(dict1, dict2):<br>    merged = {}<br>    for key, value in dict1.items():<br>        merged[key] = value<br>    for key, value in dict2.items():<br>        if key not in merged:<br>            merged[key] = value<br>    return merged<br></td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R95" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">96</div></th><td class="s1">10 - Dictionaries</td><td class="s1">04 - Merge and sum dictionaries</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;mergeAndSumDictionaries&quot; that accepts two arguments. Both of these arguments will be dictionaries. Your function should create a new dictionary that merges these two dictionaries. When both dictionaries have an element with the same key, the value in the new dictionary should be the sum of the values from the original two dictionaries.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # Declare the test dictionaries<br>    dictA = {&quot;A&quot;: 1, &quot;B&quot;: 9, &quot;C&quot;: 4}<br>    dictB = {&quot;A&quot;: 4, &quot;B&quot;: -7, &quot;Q&quot;: 3}<br>    dictC = {&quot;X&quot;: 4, &quot;Y&quot;: 4, &quot;Z&quot;: 4, &quot;A&quot;: 4}<br>    <br>    print(mergeAndSumDictionaries(dictA, dictB))<br><br>Below is the expected output:<br><br>    {&#39;A&#39;: 5, &#39;B&#39;: 2, &#39;C&#39;: 4, &#39;Q&#39;: 3}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # Declare the test dictionaries<br>    dictA = {&quot;A&quot;: 1, &quot;B&quot;: 9, &quot;C&quot;: 4}<br>    dictB = {&quot;A&quot;: 4, &quot;B&quot;: -7, &quot;Q&quot;: 3}<br>    dictC = {&quot;X&quot;: 4, &quot;Y&quot;: 4, &quot;Z&quot;: 4, &quot;A&quot;: 4}<br>    <br>    print(mergeAndSumDictionaries(dictA, dictC))<br><br>Below is the expected output:<br><br>    {&#39;A&#39;: 5, &#39;B&#39;: 9, &#39;C&#39;: 4, &#39;X&#39;: 4, &#39;Y&#39;: 4, &#39;Z&#39;: 4}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # Declare the test dictionaries<br>    dictA = {&quot;A&quot;: 1, &quot;B&quot;: 9, &quot;C&quot;: 4}<br>    dictB = {&quot;A&quot;: 4, &quot;B&quot;: -7, &quot;Q&quot;: 3}<br>    dictC = {&quot;X&quot;: 4, &quot;Y&quot;: 4, &quot;Z&quot;: 4, &quot;A&quot;: 4}<br>    <br>    print(mergeAndSumDictionaries(dictB, dictC))<br><br>Below is the expected output:<br><br>    {&#39;A&#39;: 8, &#39;B&#39;: -7, &#39;Q&#39;: 3, &#39;X&#39;: 4, &#39;Y&#39;: 4, &#39;Z&#39;: 4}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # Declare the test dictionaries<br>    dictA = {&quot;A&quot;: 1, &quot;B&quot;: 9, &quot;C&quot;: 4}<br>    dictB = {&quot;A&quot;: 4, &quot;B&quot;: -7, &quot;Q&quot;: 3}<br>    dictC = {&quot;X&quot;: 4, &quot;Y&quot;: 4, &quot;Z&quot;: 4, &quot;A&quot;: 4}<br>    <br>    print(mergeAndSumDictionaries(dictB, dictA))<br><br>Below is the expected output:<br><br>    {&#39;A&#39;: 5, &#39;B&#39;: 2, &#39;Q&#39;: 3, &#39;C&#39;: 4}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # Declare the test dictionaries<br>    dictA = {&quot;A&quot;: 1, &quot;B&quot;: 9, &quot;C&quot;: 4}<br>    dictB = {&quot;A&quot;: 4, &quot;B&quot;: -7, &quot;Q&quot;: 3}<br>    dictC = {&quot;X&quot;: 4, &quot;Y&quot;: 4, &quot;Z&quot;: 4, &quot;A&quot;: 4}<br>    <br>    print(mergeAndSumDictionaries(dictC, dictA))<br><br>Below is the expected output:<br><br>    {&#39;X&#39;: 4, &#39;Y&#39;: 4, &#39;Z&#39;: 4, &#39;A&#39;: 5, &#39;B&#39;: 9, &#39;C&#39;: 4}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # Declare the test dictionaries<br>    dictA = {&quot;A&quot;: 1, &quot;B&quot;: 9, &quot;C&quot;: 4}<br>    dictB = {&quot;A&quot;: 4, &quot;B&quot;: -7, &quot;Q&quot;: 3}<br>    dictC = {&quot;X&quot;: 4, &quot;Y&quot;: 4, &quot;Z&quot;: 4, &quot;A&quot;: 4}<br>    <br>    print(mergeAndSumDictionaries(dictC, dictB))<br><br>Below is the expected output:<br><br>    {&#39;X&#39;: 4, &#39;Y&#39;: 4, &#39;Z&#39;: 4, &#39;A&#39;: 8, &#39;B&#39;: -7, &#39;Q&#39;: 3}</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def mergeAndSumDictionaries(dict1, dict2):<br>    result = {}<br>    for key in set(dict1.keys()) | set(dict2.keys()):<br>        result[key] = dict1.get(key, 0) + dict2.get(key, 0)<br>    return result</td><td class="s2">6</td><td class="s2" dir="ltr">6</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The auto-grader said the solution&#39;s output did not match the expected output, but that was an artefact due to dictionaries not having an inherent order, so I marked it as correct.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R96" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">97</div></th><td class="s1">10 - Dictionaries</td><td class="s1">05 - Does dictionary contain particular keys</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;containsAnyOfTheseKeys&quot; that accepts two arguments: 1) a dictionary variable and 2) a list variable. Your function should return a Boolean value that indicates whether the dictionary contains any key in the specified list.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    print(containsAnyOfTheseKeys(habitats1, [&quot;duck&quot;, &quot;wolf&quot;, &quot;bird&quot;]))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    print(containsAnyOfTheseKeys(habitats1, [&quot;mouse&quot;, &quot;wolf&quot;, &quot;bird&quot;]))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    print(containsAnyOfTheseKeys(habitats2, [&quot;shark&quot;, &quot;horse&quot;, &quot;trout&quot;]))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    print(containsAnyOfTheseKeys(habitats2, [&quot;shark&quot;, &quot;deer&quot;, &quot;trout&quot;]))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    print(containsAnyOfTheseKeys(habitats3, [&quot;shark&quot;, &quot;deer&quot;, &quot;trout&quot;]))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    print(containsAnyOfTheseKeys(habitats3, [&quot;manatee&quot;]))<br><br>Below is the expected output:<br><br>    False</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def containsAnyOfTheseKeys(dictionary, keys):<br>    for key in keys:<br>        if key in dictionary:<br>            return True<br>    return False<br></td><td class="s2">6</td><td class="s2" dir="ltr">6</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R97" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">98</div></th><td class="s1">10 - Dictionaries</td><td class="s1">06 - Does dictionary contain key/value pair</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;containsKeyValuePair&quot; that accepts three arguments: 1) a dictionary variable, 2) a key, and 3) a value. Your function should return a Boolean value that indicates whether the dictionary contains a key/value pair that corresponds with the key and value specified in the arguments.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    print(containsKeyValuePair(habitats1, &quot;duck&quot;, &quot;pond&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    print(containsKeyValuePair(habitats1, &quot;horse&quot;, &quot;field&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    print(containsKeyValuePair(habitats2, &quot;deer&quot;, &quot;forrest&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    print(containsKeyValuePair(habitats2, &quot;wolf&quot;, &quot;forest&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    habitats1 = {&quot;duck&quot;: &quot;pond&quot;, &quot;horse&quot;: &quot;field&quot;, &quot;cow&quot;: &quot;pasture&quot;}<br>    habitats2 = {&quot;deer&quot;: &quot;forest&quot;, &quot;elk&quot;: &quot;forest&quot;, &quot;wolf&quot;: &quot;forest&quot;}<br>    habitats3 = {&quot;student&quot;: &quot;library&quot;, &quot;trout&quot;: &quot;lake&quot;, &quot;shark&quot;: &quot;ocean&quot;}<br>    <br>    print(containsKeyValuePair(habitats3, &quot;lake&quot;, &quot;shark&quot;))<br><br>Below is the expected output:<br><br>    False</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def containsKeyValuePair(dictionary, key, value):<br>    return key in dictionary and dictionary[key] == value<br></td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R98" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">99</div></th><td class="s1">10 - Dictionaries</td><td class="s1">07 - Match numbers to squares</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;matchNumbersToSquares&quot; that accepts a single argument: an integer value. Your function should build a dictionary that contains the same number of keys as indicated by the argument. The keys should start at 1 and increment up to (and including) the value of the argument. The value corresponding to each key should be the square of the key. For example, if this function were invoked as below:<br><br>    matchNumbersToSquares(3)<br><br>Your function would return the following dictionary:<br><br>    {1: 1, 2: 4, 3: 9}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    result = matchNumbersToSquares(3)<br>    <br>    print(result)<br><br>Below is the expected output:<br><br>    {1: 1, 2: 4, 3: 9}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    result = matchNumbersToSquares(5)<br>    <br>    print(result)<br><br>Below is the expected output:<br><br>    {1: 1, 2: 4, 3: 9, 4: 16, 5: 25}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    result = matchNumbersToSquares(10)<br>    <br>    print(result)<br><br>Below is the expected output:<br><br>    {1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64, 9: 81, 10: 100}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    result = matchNumbersToSquares(20)<br>    <br>    print(result)<br><br>Below is the expected output:<br><br>    {1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64, 9: 81, 10: 100, 11: 121, 12: 144, 13: 169, 14: 196, 15: 225, 16: 256, 17: 289, 18: 324, 19: 361, 20: 400}</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def matchNumbersToSquares(n):<br>    return {i: i**2 for i in range(1, n+1)}<br></td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">ChatGPT&#39;s solution uses the dictionary comprehension technique.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R99" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">100</div></th><td class="s1">10 - Dictionaries</td><td class="s1">08 - Sum dictionary values</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;sumDictionaryValues&quot; that accepts a single argument: a dictionary variable. The keys of this dictionary will be strings. The values of this dictionary will each be a list of numbers. Your function should create a new dictionary. The keys of the new dictionary should be the same as the keys of the original dictionary. The values of the new dictionary should be the sum of the respective values in the the original lists.<br><br>The tests will print the original dictionary to make sure that it has not been modified.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    theDict = {&quot;A&quot;: [1, 2, 3], &quot;B&quot;: [9, -4, 2], &quot;C&quot;: [3, 99, 1]}<br>    <br>    result = sumDictionaryValues(theDict)<br>    <br>    print(&quot;Input dictionary:&quot;)<br>    print(theDict)<br>    print(&quot;Output dictionary:&quot;)<br>    print(result)<br><br>Below is the expected output:<br><br>    Input dictionary:<br>    {&#39;A&#39;: [1, 2, 3], &#39;B&#39;: [9, -4, 2], &#39;C&#39;: [3, 99, 1]}<br>    Output dictionary:<br>    {&#39;A&#39;: 6, &#39;B&#39;: 7, &#39;C&#39;: 103}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    theDict = {&quot;D&quot;: [1, 2, 10], &quot;E&quot;: [-2, -4, 8], &quot;F&quot;: [100000, 0, 1]}<br>    <br>    result = sumDictionaryValues(theDict)<br>    <br>    print(&quot;Input dictionary:&quot;)<br>    print(theDict)<br>    print(&quot;Output dictionary:&quot;)<br>    print(result)<br><br>Below is the expected output:<br><br>    Input dictionary:<br>    {&#39;D&#39;: [1, 2, 10], &#39;E&#39;: [-2, -4, 8], &#39;F&#39;: [100000, 0, 1]}<br>    Output dictionary:<br>    {&#39;D&#39;: 13, &#39;E&#39;: 2, &#39;F&#39;: 100001}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    theDict = {&quot;G&quot;: [-1, -2, 3, 0, 4], &quot;H&quot;: [-11, -4, 15], &quot;I&quot;: [1]}<br>    <br>    result = sumDictionaryValues(theDict)<br>    <br>    print(&quot;Input dictionary:&quot;)<br>    print(theDict)<br>    print(&quot;Output dictionary:&quot;)<br>    print(result)<br><br>Below is the expected output:<br><br>    Input dictionary:<br>    {&#39;G&#39;: [-1, -2, 3, 0, 4], &#39;H&#39;: [-11, -4, 15], &#39;I&#39;: [1]}<br>    Output dictionary:<br>    {&#39;G&#39;: 4, &#39;H&#39;: 0, &#39;I&#39;: 1}</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def sumDictionaryValues(theDict):<br>    newDict = {}<br>    for key, value in theDict.items():<br>        newDict[key] = sum(value)<br>    return newDict<br></td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R100" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">101</div></th><td class="s1">10 - Dictionaries</td><td class="s1">09 - Count animals</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose you have traveled to the Uinta mountain range for a research project. You sit in various locations to observe animals throughout the day. At the end of your trip, your task is to tally how many times you have seen each animal for the entire trip across all the locations. Please write a function called &quot;countAnimals&quot; that accepts a single argument: a list of string values that represent the animals you observed. This list may contain duplicates of animals that you observed. Your function should create a dictionary that indicates how many times you observed each animal. The keys of your dictionary should be animal names. The values of your dictionary should be the number of sightings for those animals. For example, if you had observed 2 deer, 2 ducks, and a bear, your dictionary would be something like this: &quot;{&quot;deer&quot;: 2, &quot;duck&quot;: 2, &quot;bear&quot;: 1}&quot;. After creating your dictionary and populating these values, please return the dictionary.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animals = [&quot;deer&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;duck&quot;, &quot;deer&quot;, &quot;moose&quot;, &quot;rabbit&quot;, &quot;rabbit&quot;, &quot;deer&quot;, &quot;deer&quot;, &quot;bear&quot;]<br>    <br>    print(countAnimals(animals))<br><br>Below is the expected output:<br><br>    {&#39;deer&#39;: 6, &#39;duck&#39;: 1, &#39;moose&#39;: 1, &#39;rabbit&#39;: 2, &#39;bear&#39;: 1}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animals = [&quot;deer&quot;, &quot;snake&quot;, &quot;rabbit&quot;, &quot;duck&quot;, &quot;moose&quot;, &quot;bear&quot;]<br>    <br>    print(countAnimals(animals))<br><br>Below is the expected output:<br><br>    {&#39;deer&#39;: 1, &#39;snake&#39;: 1, &#39;rabbit&#39;: 1, &#39;duck&#39;: 1, &#39;moose&#39;: 1, &#39;bear&#39;: 1}<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animals = [&quot;deer&quot;]  10 + [&quot;moose&quot;]  5 + [&quot;mouse&quot;]  732<br>    <br>    print(countAnimals(animals))<br><br>Below is the expected output:<br><br>    {&#39;deer&#39;: 10, &#39;moose&#39;: 5, &#39;mouse&#39;: 732}</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def countAnimals(animals):<br>    animal_count = {}<br>    for animal in animals:<br>        if animal in animal_count:<br>            animal_count[animal] += 1<br>        else:<br>            animal_count[animal] = 1<br>    return animal_count<br></td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R101" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">102</div></th><td class="s1">10 - Dictionaries</td><td class="s1">10 - Convert dictionary to list</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;convertDictionaryToList&quot; that accepts a single argument: a dictionary variable. Your function should build and return a nested list variable that contains the contents of the dictionary, sorted by key.<br><br>For example, if the input dictionary were the following:<br><br>    {&quot;cat&quot;: 3, &quot;badger&quot;: 5, &quot;parrot&quot;: 1}<br><br>Your return value would be:<br><br>    [[&quot;badger&quot;, 5], [&quot;cat&quot;, 3], [&quot;parrot&quot;, 1]]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    animalWeights = { &quot;dog&quot;: 45.2, &quot;cat&quot;: 22.9, &quot;monkey&quot;: 33.3, &quot;turkey&quot;: 28.8}<br>    <br>    print(convertDictionaryToList(animalWeights))<br><br>Below is the expected output:<br><br>    [[&#39;cat&#39;, 22.9], [&#39;dog&#39;, 45.2], [&#39;monkey&#39;, 33.3], [&#39;turkey&#39;, 28.8]]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    treeHeights = { &quot;cactus&quot;: 15.4, &quot;grass&quot;: 2.1, &quot;sunflower&quot;: 18.4, &quot;pine&quot;: 118.7}<br>    <br>    print(convertDictionaryToList(treeHeights))<br><br>Below is the expected output:<br><br>    [[&#39;cactus&#39;, 15.4], [&#39;grass&#39;, 2.1], [&#39;pine&#39;, 118.7], [&#39;sunflower&#39;, 18.4]]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def convertDictionaryToList(d):<br>    return [[k, v] for k, v in sorted(d.items())]</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">ChatGPT&#39;s solution used a list comprehension to generate the nested list. Many students would find this difficult to understand.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R102" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">103</div></th><td class="s1">10 - Dictionaries</td><td class="s1">11 - Get nested maximum</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose you had weather data for various cities in various years and that you wanted to identify the maximum temperature across all the cities and years.<br><br>Please write a function called &quot;getNestedMax&quot; that accepts a single argument: a dictionary variable. Each key of this dictionary will be a string (city name). Each value will be a (nested) dictionary. Each key in the nested dictionaries will be an integer (year). Each value in the nested dictionaries will be a number (temperature). Your function should check all the nested dictionaries and identify the maximum temperature across the nested dictionaries. Your function should return the overall maximum.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    rainCityNested = {&quot;Omaha&quot;: {2013: 13.2, 2014: 14.5}, &quot;Miami&quot;: {2013: 88.4, 2014: 75.3}}<br>    <br>    print(getNestedMax(rainCityNested))<br><br>Below is the expected output:<br><br>    88.4<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    rainCityNested = {&quot;Provo&quot;: {2013: 15.2, 2014: 25.4}, &quot;Dallas&quot;: {2013: 34.8, 2014: 40.2}}<br>    <br>    print(getNestedMax(rainCityNested))<br><br>Below is the expected output:<br><br>    40.2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    rainCityNested = {&quot;Moab&quot;: {2013: 35.2, 2014: 35.4, 2015: 37.7}, &quot;Kanab&quot;: {2013: 34.8, 2014: 36.2, 2015: 39.4}, &quot;Green River&quot;: {2013: 36.4, 2014: 38.3, 2015: 38.2}}<br>    <br>    print(getNestedMax(rainCityNested))<br><br>Below is the expected output:<br><br>    39.4</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def getNestedMax(dictionary):<br>    max_temp = float(&quot;-inf&quot;)<br>    for city_dict in dictionary.values():<br>        for temp in city_dict.values():<br>            max_temp = max(max_temp, temp)<br>    return max_temp<br></td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R103" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">104</div></th><td class="s1">10 - Dictionaries</td><td class="s1">12 - Identify bacterial colonies to keep</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>In a lab, researchers often seek to grow bacterial colonies. If bacteria have not grown to 500 CFUs (colony-forming units) after 3 days, the scientists will throw away the bacterial samples. Please write a function called &quot;identifyColoniesToKeep&quot;. This function should accept two arguments. Both of these arguments will be dictionaries that will contain information about the bacterial colonies. In the first dictionary, the keys will be the names of bacterial species. The values in this dictionary will be the current number of CFUs for each species. In the second dictionary, the keys will also be names of bacterial species. The dictionary values will indicate the growth rate for each of these species.<br><br>For example, if the growth rate for Alysiella crassa were 0.5, that would indicate that the number of bacteria would increase by 50% each day. Assuming that all bacterial cultures you are currently growing in the lab have three more days to grow, write code that will identify which bacteria will not be thrown out. Your function should return a list that contains the names of these bacteria. Before returning the list, sort the names in alphabetical order.<br><br>Hint(s):<br><br>    The math on this problem might be a little tricky. First, calculate the number of bacteria after one day as the original number multiplied by one plus the growth rate. Next, take that number and multiply it by one plus the growth rate; that is the number of bacteria after the second day. Then, take that number and multiply it by one plus the growth rate; that is the number of bacteria after the third day.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    currentCFUs1 = {&quot;Alysiella crassa&quot;: 200, &quot;Asaia bogorensis&quot;: 50, &quot;Alishewanella fetalis&quot;: 450}<br>    <br>    growthRates1 = {&quot;Alysiella crassa&quot;: 0.5, &quot;Asaia bogorensis&quot;: 0.99, &quot;Alishewanella fetalis&quot;: 0.03}<br>    <br>    print(identifyColoniesToKeep(currentCFUs1, growthRates1))<br><br>Below is the expected output:<br><br>    [&#39;Alysiella crassa&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    currentCFUs2 = {&quot;Streptococcus mutans&quot;: 10, &quot;Lactococcus lactis&quot;: 400, &quot;Actinomyces israelii&quot;: 350, &quot;Azotobacter vinelandii&quot;: 5}<br>    <br>    growthRates2 = {&quot;Lactococcus lactis&quot;: 0.05, &quot;Streptococcus mutans&quot;: 2.5, &quot;Azotobacter vinelandii&quot;: 5.0, &quot;Actinomyces israelii&quot;: 1.00}<br>    <br>    print(identifyColoniesToKeep(currentCFUs2, growthRates2))<br><br>Below is the expected output:<br><br>    [&#39;Actinomyces israelii&#39;, &#39;Azotobacter vinelandii&#39;]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def identifyColoniesToKeep(currentCFUs, growthRates):<br>    coloniesToKeep = []<br>    for species, CFUs in currentCFUs.items():<br>        growthFactor = 1 + growthRates[species]<br>        for i in range(3):<br>            CFUs *= growthFactor<br>        if CFUs &gt;= 500:<br>            coloniesToKeep.append(species)<br>    coloniesToKeep.sort()<br>    return coloniesToKeep</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R104" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">105</div></th><td class="s1">11 - Reading Text Files</td><td class="s1">01 - Add integers from file</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;addIntegersFromFile&quot; that accepts a single argument: the name of a text file (either &quot;numbers1.txt&quot; or &quot;numbers2.txt&quot;, see below). These files contain multiple numbers, each on a separate line. Your function should read each line in the specified text file, convert the values to float values, and then sum the numbers. Your function should return the sum.<br><br>Hint(s):<br><br>    Watch out for newline characters.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addIntegersFromFile(&quot;numbers1.txt&quot;))<br><br>Below is the expected output:<br><br>    6.0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addIntegersFromFile(&quot;numbers2.txt&quot;))<br><br>Below is the expected output:<br><br>    2863.08<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers1.txt<br><br>File contents:<br><br>    2<br>    2<br>    2<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers2.txt<br><br>File contents:<br><br>    106<br>    22<br>    4.2<br>    5<br>    -192<br>    2829<br>    88.88</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def addIntegersFromFile(file_name):<br>    with open(file_name) as f:<br>        numbers = f.readlines()<br>    <br>    sum = 0<br>    for number in numbers:<br>        sum += float(number.strip())<br>        <br>    return sum</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R105" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">106</div></th><td class="s1">11 - Reading Text Files</td><td class="s1">02 - Count &quot;ing&quot; words</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;countIngWords&quot; that accepts a single argument: the name of a text file (one of the files listed below). Each file contains a word on each line. Your function should count how many words end with &quot;ing&quot; and return that number. Your function should count a word as ending with &quot;ing&quot; even if there is a period or a comma after it.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(countIngWords(&quot;words1.txt&quot;))<br><br>Below is the expected output:<br><br>    2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(countIngWords(&quot;words2.txt&quot;))<br><br>Below is the expected output:<br><br>    6<br><br>Below is a description of a data file that is available for testing.<br><br>File name: words1.txt<br><br>File contents:<br><br>    Doing<br>    biology<br>    research<br>    is<br>    very<br>    fun<br>    especially<br>    when<br>    you<br>    are<br>    writing<br>    programs<br>    in<br>    Python.<br><br>Below is a description of a data file that is available for testing.<br><br>File name: words2.txt<br><br>File contents:<br><br>    Eating<br>    too<br>    much<br>    candy<br>    is<br>    being<br>    naughty,<br>    unless<br>    you<br>    are<br>    eating<br>    candy<br>    that<br>    has<br>    been<br>    fortified<br>    with<br>    micronutrients.<br>    But<br>    that&#39;s<br>    being<br>    a<br>    little<br>    optimistic<br>    because<br>    candy<br>    makers<br>    don&#39;t<br>    sing,<br>    while<br>    they<br>    wear<br>    bling.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def countIngWords(file_name):<br>    with open(file_name) as f:<br>        words = f.readlines()<br>    <br>    count = 0<br>    for word in words:<br>        word = word.strip().rstrip(&#39;,.&#39;)<br>        if word.endswith(&quot;ing&quot;):<br>            count += 1<br>        <br>    return count</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R106" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">107</div></th><td class="s1">11 - Reading Text Files</td><td class="s1">03 - Calculate BMI</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Please write a function called &quot;calculateBMI&quot; that accepts a single argument: the name of a text file. This text file will contain three columns, separated by tabs. The first item in each row will contain a person&#39;s name. The second item in each row will contain that person&#39;s height. The third item in each row will contain that person&#39;s weight. In addition, the first line in this text file will be a header, containing column names.<br><br>Your function should parse through this file and calculate each person&#39;s Body Mass Index (BMI). To properly calculate BMI, you will need to convert the heights from inches to meters. To do this, multiply the heights by 0.025. Also convert the weights from pounds to kilograms. To do this, multiply the weights by 0.45. Finally, calculate BMI by dividing the weight (in kilograms) by the squared value of the height (in meters). Your function should return the name and BMI of the person with the smallest BMI. Your return value should be a string formatted like this: &quot;George: 21&quot;. (Round the person&#39;s BMI to the nearest integer.)<br><br>Note: These are approximate calculations. In reality, we should account for whether a person is male or female, etc. But you don&#39;t need to worry about that for this exercise.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calculateBMI(&quot;data1.txt&quot;))<br><br>Below is the expected output:<br><br>    Sasha: 19<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calculateBMI(&quot;data2.txt&quot;))<br><br>Below is the expected output:<br><br>    Jenica: 20<br><br>Below is a description of a data file that is available for testing.<br><br>File name: data1.txt<br><br>File contents:<br><br>    Name        Height        Weight<br>    Bill        68        175<br>    Harry        79        200<br>    Fernando        71        220<br>    Sunil        66        350<br>    Sasha        73        137<br><br>Below is a description of a data file that is available for testing.<br><br>File name: data2.txt<br><br>File contents:<br><br>    Name        Height        Weight<br>    Becca        63        125<br>    Sue        59        169<br>    Maria        63        180<br>    Jenica        74        150<br>    Miiko        62        121</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def calculateBMI(file_name):<br>    with open(file_name, &#39;r&#39;) as f:<br>        lines = f.readlines()<br>        <br>    header = lines[0].strip().split(&#39;\t&#39;)<br>    name_index = header.index(&quot;Name&quot;)<br>    height_index = header.index(&quot;Height&quot;)<br>    weight_index = header.index(&quot;Weight&quot;)<br>    <br>    min_bmi = float(&quot;inf&quot;)<br>    min_bmi_person = &quot;&quot;<br>    for line in lines[1:]:<br>        person_data = line.strip().split(&#39;\t&#39;)<br>        name = person_data[name_index]<br>        height = float(person_data[height_index]) * 0.025<br>        weight = float(person_data[weight_index]) * 0.45<br>        bmi = weight / (height ** 2)<br>        if bmi &lt; min_bmi:<br>            min_bmi = bmi<br>            min_bmi_person = name<br>    <br>    return f&quot;{min_bmi_person}: {int(min_bmi + 0.5)}&quot;</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R107" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">108</div></th><td class="s1">11 - Reading Text Files</td><td class="s1">04 - Calculate GC content</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;calculateGCContent&quot; that accepts a single argument: the name of a text file. This text file will be in FASTA format. Each entry in the FASTA file represents the DNA sequence of a chromosome. Each entry starts with a &quot;&gt;&quot; character and may be split across multiple lines in the file. Your function should calculate the overall proportion of nucleotides that are &quot;G&quot; or &quot;C&quot; nucleotides across all chromosomes. Please round your answer to two decimal places.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calculateGCContent(&quot;DNA1.fa&quot;))<br><br>Below is the expected output:<br><br>    0.5<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calculateGCContent(&quot;DNA2.fa&quot;))<br><br>Below is the expected output:<br><br>    0.47<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA1.fa<br><br>File contents:<br><br>    &gt;chr1<br>    AGTGTCCTTAAGACTTAAGATAACGCCTCTCGTAGACTCG<br>    ACAAAGTCGACTCCGCCTGCTCAAATCGACACCTTGGTGA<br>    &gt;chr2<br>    TCACAACAATGAGACGTGGCCAGAAGTACGACATTGTGAA<br>    GTGTAGGTCTGAAAACATTATCCGCATCAGGCCT<br>    &gt;chr3<br>    CCAGAAGCTACAGGCTTGAGTTGCCTGGAGATTCAGTTTG<br>    GCGAGCATCCCACCAGTGCTTACTAGCATACCGTCTGTAT<br>    CCTAGTGCATGAG<br>    &gt;chr4<br>    GCAAAAAACATTTTTCCGCTAACTAACTACTTGCGACGCG<br>    TGAACCATAAGCGGAGATGTACCTC<br>    &gt;chr5<br>    AAACAGTTATCTGTTAACCCCCCGTCTTTTTACCGGCCGC<br>    CGGAACGCAACTAACCCCTGTCGGAGCCCAAAGATCGTCA<br>    CCGTGCCTCCTAAGGCGGTGGCACTACGGAAAGTC<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA2.fa<br><br>File contents:<br><br>    &gt;chr1<br>    TACCTCCAGGGGGCATCCTCCCCCCCAATTCGAAACACAT<br>    CGTAGCCCCTGGCACTACCTATGTGTGTCAATTCGGAGAA<br>    GAGAGATTCACGAAAAAAAAGTCTGGACTCAACTAGGATC<br>    ACACATTCGGCTACAGATACCAAAAAAAAAAAAAAAA<br>    &gt;chr2<br>    CTCGTCTCAGCGAAGTATAGGAGACACCCCAGTATTTGTG<br>    CGAAACAGTTATAAGAGGAACGCGACTACGACCTAGAGTG<br>    TAAGCCCGGTTAGTCATAATATGCAGAAGAGAATCAATTG<br>    ATTCCGGGCTTCTTCTTTAATAACATGAGTGGTTCCATAG<br>    ATATAGGTCCGCCTAGGTTAACTCGCCAGTAGAGAGCAAC<br>    GTATGGCAGCACATTCCGATCTTGATCAACGGCCTGCCGA<br>    ACTTCCGACCAGTACGATGCTAGTTCAAATCCCCAGGTGG<br>    TATAGTCTTATCTGCACGTCAAA<br>    &gt;chr3<br>    AAATTTTCACCATTGAGGCACCACCTTCTCGTCGCTGCGC<br>    GCTCTGCTCGCTTCGGCTAAAAATTCGCGCAATACATTCG<br>    CTACAGATACCAAA<br>    &gt;chr4<br>    GTCTCTGCTGTCCGGTAGGCAGGGCTTTGACTGCTGGATG<br>    GTGAATTACATGCCCGATTGTACTCACGTCCACTCAAGCG<br>    CTTGAAAGACTGGGCAATATGATCCCGTATATTGCGTAAT<br>    CTCGGTGGGTCAGGACTCCTAAAATGCTCATGGTAAGTTG<br>    TAGACGGGGGAAATCCTACAACCACCC<br>    &gt;chr5<br>    CTATCTTACCAGGAGCTAGCTTACTAGCCCTTTAAAGGAA<br>    AAATTGGCACGCGGAGACAGACAG</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def calculateGCContent(fileName):<br>    with open(fileName, &#39;r&#39;) as file:<br>        data = file.read()<br>        <br>    sequences = data.split(&#39;&gt;&#39;)[1:]<br>    sequences = [seq.split(&#39;\n&#39;, 1)[1].replace(&#39;\n&#39;, &#39;&#39;) for seq in sequences]<br>    <br>    gc_count = 0<br>    total_count = 0<br>    for seq in sequences:<br>        gc_count += seq.count(&#39;G&#39;) + seq.count(&#39;C&#39;)<br>        total_count += len(seq)<br>        <br>    return round(gc_count / total_count, 2)</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">ChatGPT&#39;s first solution returned numbers that were too high. I asked it to try again, and it was successful on the second try. I believe the error on the first try was that it ignored the first character on every line, not just the first line of each contig.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R108" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">109</div></th><td class="s1">11 - Reading Text Files</td><td class="s1">05 - Find longest chromosome</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;findLongestChromosome&quot; that accepts a single argument: the name of a text file. This text file will be in FASTA format. Each entry in the FASTA file represents the DNA sequence of a chromosome. Each entry starts with a &quot;&gt;&quot; character and may be split across multiple lines in the file. Your function should return the name (for example, &quot;chr2&quot;) of the longest chromosome (i.e., the one with the most nucleotides).<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findLongestChromosome(&quot;DNA1.fa&quot;))<br><br>Below is the expected output:<br><br>    chr5<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findLongestChromosome(&quot;DNA2.fa&quot;))<br><br>Below is the expected output:<br><br>    chr2<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA1.fa<br><br>File contents:<br><br>    &gt;chr1<br>    AGTGTCCTTAAGACTTAAGATAACGCCTCTCGTAGACTCG<br>    ACAAAGTCGACTCCGCCTGCTCAAATCGACACCTTGGTGA<br>    &gt;chr2<br>    TCACAACAATGAGACGTGGCCAGAAGTACGACATTGTGAA<br>    GTGTAGGTCTGAAAACATTATCCGCATCAGGCCT<br>    &gt;chr3<br>    CCAGAAGCTACAGGCTTGAGTTGCCTGGAGATTCAGTTTG<br>    GCGAGCATCCCACCAGTGCTTACTAGCATACCGTCTGTAT<br>    CCTAGTGCATGAG<br>    &gt;chr4<br>    GCAAAAAACATTTTTCCGCTAACTAACTACTTGCGACGCG<br>    TGAACCATAAGCGGAGATGTACCTC<br>    &gt;chr5<br>    AAACAGTTATCTGTTAACCCCCCGTCTTTTTACCGGCCGC<br>    CGGAACGCAACTAACCCCTGTCGGAGCCCAAAGATCGTCA<br>    CCGTGCCTCCTAAGGCGGTGGCACTACGGAAAGTC<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA2.fa<br><br>File contents:<br><br>    &gt;chr1<br>    TACCTCCAGGGGGCATCCTCCCCCCCAATTCGAAACACAT<br>    CGTAGCCCCTGGCACTACCTATGTGTGTCAATTCGGAGAA<br>    GAGAGATTCACGAAAAAAAAGTCTGGACTCAACTAGGATC<br>    ACACATTCGGCTACAGATACCAAAAAAAAAAAAAAAA<br>    &gt;chr2<br>    CTCGTCTCAGCGAAGTATAGGAGACACCCCAGTATTTGTG<br>    CGAAACAGTTATAAGAGGAACGCGACTACGACCTAGAGTG<br>    TAAGCCCGGTTAGTCATAATATGCAGAAGAGAATCAATTG<br>    ATTCCGGGCTTCTTCTTTAATAACATGAGTGGTTCCATAG<br>    ATATAGGTCCGCCTAGGTTAACTCGCCAGTAGAGAGCAAC<br>    GTATGGCAGCACATTCCGATCTTGATCAACGGCCTGCCGA<br>    ACTTCCGACCAGTACGATGCTAGTTCAAATCCCCAGGTGG<br>    TATAGTCTTATCTGCACGTCAAA<br>    &gt;chr3<br>    AAATTTTCACCATTGAGGCACCACCTTCTCGTCGCTGCGC<br>    GCTCTGCTCGCTTCGGCTAAAAATTCGCGCAATACATTCG<br>    CTACAGATACCAAA<br>    &gt;chr4<br>    GTCTCTGCTGTCCGGTAGGCAGGGCTTTGACTGCTGGATG<br>    GTGAATTACATGCCCGATTGTACTCACGTCCACTCAAGCG<br>    CTTGAAAGACTGGGCAATATGATCCCGTATATTGCGTAAT<br>    CTCGGTGGGTCAGGACTCCTAAAATGCTCATGGTAAGTTG<br>    TAGACGGGGGAAATCCTACAACCACCC<br>    &gt;chr5<br>    CTATCTTACCAGGAGCTAGCTTACTAGCCCTTTAAAGGAA<br>    AAATTGGCACGCGGAGACAGACAG</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def findLongestChromosome(filename):<br>    with open(filename, &#39;r&#39;) as f:<br>        lines = f.readlines()<br><br>    longest_chromosome = &quot;&quot;<br>    max_length = 0<br>    current_chromosome = &quot;&quot;<br>    current_length = 0<br><br>    for line in lines:<br>        if line.startswith(&quot;&gt;&quot;):<br>            if current_length &gt; max_length:<br>                longest_chromosome = current_chromosome<br>                max_length = current_length<br>            current_chromosome = line[1:].strip()<br>            current_length = 0<br>        else:<br>            current_length += len(line.strip())<br><br>    if current_length &gt; max_length:<br>        longest_chromosome = current_chromosome<br>        max_length = current_length<br><br>    return longest_chromosome</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R109" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">110</div></th><td class="s1">11 - Reading Text Files</td><td class="s1">06 - Read animal names into a dictionary</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Please write a function called &quot;readIntoDict&quot; that accepts one argument: the name of a text file. This file will contain two columns of text, separated by tabs. The first column will indicate the adult name of an animal (for example, &quot;dog&quot;). The second column will indicate the term used to describe babies of that animal (for example, &quot;puppy&quot;).<br><br>Your function should read this file and store the data in a dictionary. Each adult animal name should be a key, and the respective baby names should be values. Your function should return the dictionary.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(readIntoDict(&quot;animal_names.txt&quot;))<br><br>Below is the expected output:<br><br>    {&#39;dog&#39;: &#39;puppy&#39;, &#39;sheep&#39;: &#39;lamb&#39;, &#39;cat&#39;: &#39;kitten&#39;, &#39;bear&#39;: &#39;cub&#39;, &#39;lion&#39;: &#39;cub&#39;}<br><br>Below is a description of a data file that is available for testing.<br><br>File name: animal_names.txt<br><br>File contents:<br><br>    dog        puppy<br>    sheep        lamb<br>    cat        kitten<br>    bear        cub<br>    lion        cub</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def readIntoDict(file_name):<br>    animal_dict = {}<br>    with open(file_name, &#39;r&#39;) as file:<br>        for line in file:<br>            parts = line.strip().split(&#39;\t&#39;)<br>            animal_dict[parts[0]] = parts[1]<br>    return animal_dict</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R110" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">111</div></th><td class="s1">11 - Reading Text Files</td><td class="s1">07 - Make animals younger</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Please write a function called &quot;makeYounger&quot; that accepts two arguments:<br><br>1. The name of a text file. This file will contain two columns of text, separated by tabs. The first column will indicate the adult name of an animal (for example, &quot;dog&quot;). The second column will indicate the term used to describe babies of that animal (for example, &quot;puppy&quot;).<br><br>2. The name of a text file. This file will contain three paragraphs that describe happenings with various animals. (It is not tab-delimited.)<br><br>Your function should read the first file into a dictionary (reuse the code you created in the previous exercise).  Your function should read the second file into a string and then find instances of the adult animal names in the text. When it finds one of the adult names, your code should replace it with the corresponding baby name. Finally, your function should return the modified version of the text.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(makeYounger(&quot;animal_names.txt&quot;, &quot;story.txt&quot;))<br><br>Below is the expected output:<br><br>    Once upon a time, a little puppy walked into a beautiful meadow in the mountains. She was so happy to bask in the sunshine and frolic in the flowers. Along came a kitten. Normally, a kitten does not get along with a puppy. But this time they had a grand time, smelling flowers, chasing rabbits, and digging holes.<br>    <br>    Soon the kitten and puppy lost track of time and had lost their way. Fortunately, a lamb came along and gave them directions to return to the meadow where they had been. But on their way back, they encountered a cub. The cub was quite hungry and was looking for some kind of mammal to feed to her children. The kitten and puppy scurried away as quickly as they could, and the cub was forced to compete with a cub for whatever scarce resources they could find.<br>    <br>    Moral of the story: if you are a puppy, don&#39;t play with a kitten.<br><br>Below is a description of a data file that is available for testing.<br><br>File name: animal_names.txt<br><br>File contents:<br><br>    dog        puppy<br>    sheep        lamb<br>    cat        kitten<br>    bear        cub<br>    lion        cub<br><br>Below is a description of a data file that is available for testing.<br><br>File name: story.txt<br><br>File contents:<br><br>    Once upon a time, a little dog walked into a beautiful meadow in the mountains. She was so happy to bask in the sunshine and frolic in the flowers. Along came a cat. Normally, a cat does not get along with a dog. But this time they had a grand time, smelling flowers, chasing rabbits, and digging holes.<br>    <br>    Soon the cat and dog lost track of time and had lost their way. Fortunately, a sheep came along and gave them directions to return to the meadow where they had been. But on their way back, they encountered a bear. The bear was quite hungry and was looking for some kind of mammal to feed to her children. The cat and dog scurried away as quickly as they could, and the bear was forced to compete with a lion for whatever scarce resources they could find.<br>    <br>    Moral of the story: if you are a dog, don&#39;t play with a cat.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def makeYounger(file1, file2):<br>    def readIntoDict(file1):<br>        with open(file1, &#39;r&#39;) as f:<br>            animal_dict = {}<br>            for line in f:<br>                animal, baby = line.strip().split(&#39;\t&#39;)<br>                animal_dict[animal] = baby<br>            return animal_dict<br><br>    animal_dict = readIntoDict(file1)<br><br>    with open(file2, &#39;r&#39;) as f:<br>        story = f.read()<br>        for animal, baby in animal_dict.items():<br>            story = story.replace(animal, baby)<br>        return story</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R111" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">112</div></th><td class="s1">11 - Reading Text Files</td><td class="s1">08 - Find high blood pressure measurements</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Suppose that, as part of a biology class project, students were asked to measure their systolic blood pressure every class period. They saved this information into a tab-separated-value (TSV) file called &quot;systolic.txt&quot;. In this file, each row represents a date, and each column represents a student. The header line of the file contains student names for all columns (other than the first column).<br><br>Write a function called &quot;calcNumDays&quot; that accepts a single argument: an integer. This integer will represent a day of the month. For example, if the integer were &quot;16&quot;, it would refer to the date &quot;16-Jan&quot; in the file. This number might also refer to a day of the month that is not specified in the file (example: &quot;1&quot;).<br><br>Your function should read &quot;systolic.txt&quot;. It should ignore any blood-pressure values before the specified day of the month. For the remaining days, your function should calculate the number of times a blood-pressure reading was 140 or higher. Your function should return this number.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calcNumDays(1))<br><br>Below is the expected output:<br><br>    12<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calcNumDays(6))<br><br>Below is the expected output:<br><br>    12<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calcNumDays(12))<br><br>Below is the expected output:<br><br>    9<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calcNumDays(18))<br><br>Below is the expected output:<br><br>    6<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calcNumDays(24))<br><br>Below is the expected output:<br><br>    4<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(calcNumDays(31))<br><br>Below is the expected output:<br><br>    0<br><br>Below is a description of a data file that is available for testing.<br><br>File name: systolic.txt<br><br>File contents:<br><br>    Date        Sarah        John        Sam        Rachel        June        Thomas<br>    6-Jan        120        135        111        105        140        133<br>    8-Jan        121        130        111        110        142        135<br>    10-Jan        115        145        112        103        138        128<br>    12-Jan        116        131        109        109        140        139<br>    14-Jan        122        129        110        105        141        129<br>    16-Jan        118        135        111        106        143        135<br>    18-Jan        110        139        117        103        137        134<br>    20-Jan        120        141        108        105        140        132<br>    22-Jan        117        133        110        110        139        128<br>    24-Jan        119        138        111        102        141        127<br>    26-Jan        120        146        111        110        142        131<br>    28-Jan        121        135        110        105        137        133<br>    30-Jan        125        126        106        104        140        134</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def calcNumDays(day):<br>    count = 0<br>    with open(&quot;systolic.txt&quot;) as file:<br>        lines = file.readlines()<br>        headers = lines[0].strip().split(&quot;\t&quot;)<br>        for line in lines[1:]:<br>            data = line.strip().split(&quot;\t&quot;)<br>            if int(data[0].split(&quot;-&quot;)[0]) &gt;= day:<br>                for i in range(1, len(headers)):<br>                    if int(data[i]) &gt;= 140:<br>                        count += 1<br>    return count</td><td class="s2">6</td><td class="s2" dir="ltr">6</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R112" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">113</div></th><td class="s1">12 - Writing Text Files</td><td class="s1">01 - Add numbers and save</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;addNumbersAndSave&quot; that accepts two arguments. The first argument will be the name of a text file that contains numeric values. Each numeric value will be on a separate line. The second argument will be the name of a text file that your function must create. Your function should read each line in the first text file, convert the values to float values, and sum the numbers. Your function should round the overall sum to one decimal place and save it to a file at the location specified in the second argument. Your function does not need to return a value.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    addNumbersAndSave(&quot;numbers1.txt&quot;, &quot;sum1.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;sum1.txt&quot;):<br>        with open(&quot;sum1.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;sum1.txt does not exist.&quot;)<br><br>Below is the expected output:<br><br>    6.0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    addNumbersAndSave(&quot;numbers2.txt&quot;, &quot;sum2.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;sum2.txt&quot;):<br>        with open(&quot;sum2.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;sum2.txt does not exist.&quot;)<br><br>Below is the expected output:<br><br>    2863.1<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers1.txt<br><br>File contents:<br><br>    2<br>    2<br>    2<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers2.txt<br><br>File contents:<br><br>    106<br>    22<br>    4.2<br>    5<br>    -192<br>    2829<br>    88.88</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def addNumbersAndSave(inputFileName, outputFileName):<br>    with open(inputFileName) as inputFile:<br>        numbers = [float(line.strip()) for line in inputFile]<br>        sumOfNumbers = round(sum(numbers), 1)<br>    <br>    with open(outputFileName, &quot;w&quot;) as outputFile:<br>        outputFile.write(str(sumOfNumbers))</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R113" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">114</div></th><td class="s1">12 - Writing Text Files</td><td class="s1">02 - Add items in multiple files</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;addItemsInMultipleFiles&quot; that accepts two arguments. The first argument will be a list. This list will contain one or more string values. Each of these values will be the path to a text file that contains numeric values. Each number in the text files will be on a separate line. The second argument will be the name of an output file that your function should create.<br><br>Your function should loop through the list of files and calculate the sum of the numeric values across all of the files. Your function should then round the sum to one decimal place and save the sum to the output file. Your function does not need to return a value.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    fileList1 = [&quot;numbers1.txt&quot;, &quot;numbers2.txt&quot;]<br>    <br>    addItemsInMultipleFiles(fileList1, &quot;sum1.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;sum1.txt&quot;):<br>        with open(&quot;sum1.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;sum1.txt does not exist.&quot;)<br><br>Below is the expected output:<br><br>    69.2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    fileList2 = [&quot;numbers3.txt&quot;, &quot;numbers4.txt&quot;, &quot;numbers5.txt&quot;]<br>    <br>    addItemsInMultipleFiles(fileList2, &quot;sum2.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;sum2.txt&quot;):<br>        with open(&quot;sum2.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;sum2.txt does not exist.&quot;)<br><br>Below is the expected output:<br><br>    2461.5<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    fileList1 = [&quot;numbers1.txt&quot;, &quot;numbers2.txt&quot;]<br>    fileList2 = [&quot;numbers3.txt&quot;, &quot;numbers4.txt&quot;, &quot;numbers5.txt&quot;]<br>    fileList3 = fileList1 + fileList2<br>    <br>    addItemsInMultipleFiles(fileList3, &quot;sum3.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;sum3.txt&quot;):<br>        with open(&quot;sum3.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;sum3.txt does not exist.&quot;)<br><br>Below is the expected output:<br><br>    2530.7<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers1.txt<br><br>File contents:<br><br>    2<br>    2<br>    2<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers2.txt<br><br>File contents:<br><br>    106<br>    22<br>    4.2<br>    5<br>    -192<br>    29.1<br>    88.88<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers3.txt<br><br>File contents:<br><br>    13.12<br>    22.27<br>    282.42<br>    83.91<br>    12.02<br>    -0.212348297<br>    0.0000005<br>    500<br>    400<br>    -193<br>    831<br>    182<br>    388<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers4.txt<br><br>File contents:<br><br>    2<br>    2<br>    2<br>    2<br>    2<br>    2<br>    3<br>    3<br>    3<br>    3<br>    3<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers5.txt<br><br>File contents:<br><br>    51<br>    18<br>    28<br>    740<br>    -924</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def addItemsInMultipleFiles(fileList, outputFile):<br>    sum = 0<br>    for file in fileList:<br>        with open(file, &#39;r&#39;) as f:<br>            for line in f:<br>                sum += float(line)<br>    with open(outputFile, &#39;w&#39;) as f:<br>        f.write(f&quot;{round(sum, 1)}\n&quot;)<br></td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R114" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">115</div></th><td class="s1">12 - Writing Text Files</td><td class="s1">03 - Sum parallel files</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;sumParallelFiles&quot; that accepts three arguments. The first two arguments will be paths to input text files. The third argument will be a path to an output text file that your function should create. The two input files will contain numeric values. Each number will be on a separate line. Your function should loop through the input files, add the respective values on each line, and write the sum of each line to the output file.<br><br>For example, if the values in the first input file were:<br><br> 2<br> 2<br> 1<br><br>...and the values in the second input file were:<br><br> 4<br> 0<br> -1<br><br>...the values in the output file would be:<br><br> 6<br> 2<br> 0<br><br>Your function does not need to return a value.<br><br>Hint(s):<br><br>    Loop through each of the input files and store the text for each line in lists. Then loop through these lists, sum the values at identical indices within the lists, and write the sums to the output file.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    sumParallelFiles(&quot;numbers1.txt&quot;, &quot;numbers2.txt&quot;, &quot;sum12.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;sum12.txt&quot;):<br>        with open(&quot;sum12.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;sum12.txt does not exist.&quot;)<br><br>Below is the expected output:<br><br>    6.0<br>    2.0<br>    0.0<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    sumParallelFiles(&quot;numbers3.txt&quot;, &quot;numbers4.txt&quot;, &quot;sum34.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;sum34.txt&quot;):<br>        with open(&quot;sum34.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;sum34.txt does not exist.&quot;)<br><br>Below is the expected output:<br><br>    6762.0<br>    4520589.0<br>    384889.0<br>    -1820119.0<br>    382122.0<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers1.txt<br><br>File contents:<br><br>    2<br>    2<br>    1<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers2.txt<br><br>File contents:<br><br>    4<br>    0<br>    -1<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers3.txt<br><br>File contents:<br><br>    2832<br>    -449<br>    1<br>    2819<br>    383<br><br>Below is a description of a data file that is available for testing.<br><br>File name: numbers4.txt<br><br>File contents:<br><br>    3930<br>    4521038<br>    384888<br>    -1822938<br>    381739</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def sumParallelFiles(file1, file2, output_file):<br>    with open(file1) as f1, open(file2) as f2:<br>        lines1 = f1.readlines()<br>        lines2 = f2.readlines()<br>        lines_sum = [str(float(x) + float(y)) + &#39;\n&#39; for x, y in zip(lines1, lines2)]<br>        <br>    with open(output_file, &#39;w&#39;) as f:<br>        f.writelines(lines_sum)<br></td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R115" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">116</div></th><td class="s1">12 - Writing Text Files</td><td class="s1">04 - Calculate product</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Please write a function called &quot;calcProduct&quot; that accepts two arguments. The first argument will be the path to a file that contains numeric values. The second argument will be a path to an output file that your function will need to create. Each line of the input file (first argument) will contain three float values, separated by tabs.<br><br>Your function should read each line from the input file and calculate the product of the values on that line (multiple them all together). Then your function should write &quot;The product is X&quot; to the output file, substituting &quot;X&quot; with the actual product value, rounded to 2 decimal places. When writing to the output file, use an f-string. Your function does not need to return a value.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    calcProduct(&quot;data1.txt&quot;, &quot;output1.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;output1.txt&quot;):<br>        with open(&quot;output1.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;output1.txt does not exist.&quot;)<br><br>Below is the expected output:<br><br>    The product is 10.00<br>    The product is -20.00<br>    The product is 0.00<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    calcProduct(&quot;data2.txt&quot;, &quot;output2.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;output2.txt&quot;):<br>        with open(&quot;output2.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;output2.txt does not exist.&quot;)<br><br>Below is the expected output:<br><br>    The product is 8230205.76<br>    The product is 20606760.45<br>    The product is 9998.10<br>    The product is 236299.67<br>    The product is -12.42<br><br>Below is a description of a data file that is available for testing.<br><br>File name: data1.txt<br><br>File contents:<br><br>    2.0        5.0        1.0<br>    4.0        5.0        -1.0<br>    828291828371.2        923819873241.9        0.0<br><br>Below is a description of a data file that is available for testing.<br><br>File name: data2.txt<br><br>File contents:<br><br>    111.11        222.22        333.33<br>    111.22        333.44        555.66<br>    95.4321        1.89        55.4321<br>    58.238        211.412        19.1923<br>    -1.2        2.3        4.5</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def calcProduct(inputFile, outputFile):<br>    with open(inputFile, &#39;r&#39;) as file:<br>        with open(outputFile, &#39;w&#39;) as output:<br>            for line in file:<br>                values = line.strip().split(&#39;\t&#39;)<br>                product = 1<br>                for value in values:<br>                    product *= float(value)<br>                output.write(f&quot;The product is {product:.2f}\n&quot;)</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R116" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">117</div></th><td class="s1">12 - Writing Text Files</td><td class="s1">05 - Reformat animal data</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Please write a function called &quot;reformatAnimalData&quot; that accepts two arguments. The first argument will be the path to a file that contains information about animal sightings in U.S. national parks. The second argument will be a path to a file that your function will need to create.<br><br>The first line of the input file (first argument) will contain a header with the following column names: Location, Animal, and NumSightings. Each line in the file will indicate how many times a specific animal has been seen in a specific geographical location. These values (and the column names) will be separated by tabs. Your function should read each line from the input file and translate it to an English-language description. For example, if one line in the input file indicates that 45 Grizzly bears had been seen in Yellowstone National Park, you would write &quot;45 Grizzly bears have been seen in Yellowstone National Park&quot;. However, if a certain animal had been observed exactly one time, your output should say something like, &quot;1 Grizzly bear has been seen in Yellowstone National Park&quot;.<br><br>When writing to the output file, use a string formatter. Your function does not need to return a value.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    reformatAnimalData(&quot;data1.txt&quot;, &quot;output1.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;output1.txt&quot;):<br>        with open(&quot;output1.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;output1.txt does not exist.&quot;)<br><br>Below is the expected output:<br><br>    42 Black bears have been seen in Yellowstone National Park<br>    23 Grizzly bears have been seen in Yellowstone National Park<br>    128 Brown bears have been seen in Yellowstone National Park<br>    105 Black bears have been seen in Yosemite National Park<br>    33 Grizzly bears have been seen in Yosemite National Park<br>    1 Brown bear has been seen in Yosemite National Park<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    reformatAnimalData(&quot;data2.txt&quot;, &quot;output2.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;output2.txt&quot;):<br>        with open(&quot;output2.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;output2.txt does not exist.&quot;)<br><br>Below is the expected output:<br><br>    18 Black bears have been seen in Zion National Park<br>    42 Grizzly bears have been seen in Zion National Park<br>    3 Brown bears have been seen in Zion National Park<br>    15 Black bears have been seen in Bryce Canyon National Park<br>    1 Grizzly bear has been seen in Bryce Canyon National Park<br>    33 Brown bears have been seen in Bryce Canyon National Park<br><br>Below is a description of a data file that is available for testing.<br><br>File name: data1.txt<br><br>File contents:<br><br>    Location        Animal        NumSightings<br>    Yellowstone National Park        Black bear        42<br>    Yellowstone National Park        Grizzly bear        23<br>    Yellowstone National Park        Brown bear        128<br>    Yosemite National Park        Black bear        105<br>    Yosemite National Park        Grizzly bear        33<br>    Yosemite National Park        Brown bear        1<br><br>Below is a description of a data file that is available for testing.<br><br>File name: data2.txt<br><br>File contents:<br><br>    Location        Animal        NumSightings<br>    Zion National Park        Black bear        18<br>    Zion National Park        Grizzly bear        42<br>    Zion National Park        Brown bear        3<br>    Bryce Canyon National Park        Black bear        15<br>    Bryce Canyon National Park        Grizzly bear        1<br>    Bryce Canyon National Park        Brown bear        33</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def reformatAnimalData(input_file, output_file):<br>    with open(input_file, &#39;r&#39;) as file:<br>        lines = file.readlines()[1:]<br>    <br>    with open(output_file, &#39;w&#39;) as file:<br>        for line in lines:<br>            location, animal, num_sightings = line.strip().split(&#39;\t&#39;)<br>            animal = animal.split(&#39; &#39;)[0]<br>            num_sightings = int(num_sightings)<br>            <br>            if num_sightings == 1:<br>                line = &quot;{} {} bear has been seen in {}&quot;.format(num_sightings, animal, location)<br>            else:<br>                line = &quot;{} {} bears have been seen in {}&quot;.format(num_sightings, animal, location)<br>                <br>            file.write(line + &#39;\n&#39;)</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">3</td><td class="s1" dir="ltr">The output of the first solution included an extra word. I asked it to fix this and gave it another example of the expected output. The second solution was closer but added an extra &quot;s&quot; after &quot;bears&quot;. I asked it to fix the problem and gave it another example of the expected output. Then it passed.</td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R117" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">118</div></th><td class="s1">12 - Writing Text Files</td><td class="s1">06 - Transcribe DNA to RNA</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Transcription is the process whereby cells use DNA as a template to generate RNA. These RNA molecules are then exported from the nucleus, and they are typically used for translation (creating proteins). In practice, many species have introns in their genes, which are removed before translation; but for this problem, assume that the DNA is from a bacterial species and thus that there are no introns. When DNA molecules are transcribed into RNA, thymine (T) nucleotides are converted to uracil (U) nucleotides.<br><br>A file format commonly used to store DNA and RNA sequence information is the FASTA format. FASTA files consist of a header line followed by one or more sequence lines. FASTA files often have multiple header lines; each header line is followed by at least one sequence line. Header lines always start with a right angle bracket (&gt;). Below is an example.<br><br>    &gt;Gene_One<br>    ACTTACGGTACAATACGGACTAGCGACAGAAC<br>    ACTAAAGACAGATTGACGAGAGAAGTAGAATT<br>    TAGATAGACGATTTACCGGGAGATA<br>    &gt;Gene_Two<br>    GAGGAGACTCTCGATGGCGGACTAGACTAGAC<br>    AGACCCTTAGACGATTAGGTACTAGATCCCTT<br>    AAAAATTTACCCCT<br><br>Please write a function called transcribe that accepts two arguments. The first argument will be a string and will be the name of a FASTA file that contains DNA sequences. The second argument will be a string and will be the name of a FASTA file that your function will create. Your function should 1) read the specified input file, 2) &quot;transcribe&quot; the DNA to RNA, and 3) write the corresponding RNA sequences to the output file in FASTA format, including the same headers and newline characters. The output (RNA) should look the same as the input (DNA), except that the T nucleotides will be replaced with U nucleotides.<br><br>The header lines may contain the letter &quot;T&quot; in them. So make sure not to &quot;transcribe&quot; the header line. For simplicity, don&#39;t worry about the fact that DNA is double stranded.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    transcribe(&quot;DNA1.fasta&quot;, &quot;DNA1_transcribed.fasta&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;DNA1_transcribed.fasta&quot;):<br>        with open(&quot;DNA1_transcribed.fasta&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;No output file exists at DNA1_transcribed.fasta.&quot;)<br><br>Below is the expected output:<br><br>    &gt;Gene1<br>    UGCUACAGGUGGCAAGAGUUGUAUAUUGGUAUAUAGACGA<br>    CUGGACGGCCUUUAUAGCCUUACUAGGAGCAGUAGAAGCU<br>    UAUGCCCAAUUGAAUCCU<br>    &gt;Gene2<br>    AACUCCGUCUGAUCACCGUCAUGGAAGGGCGGUAUAGGAU<br>    AUGGUCGCUCCGCCGUGCAGCGGCUACGGGUC<br>    &gt;Gene3<br>    CUUUAAAAGCCGCACUAUGCGAUAUCGAGGAACAGGAGUG<br>    CCUGCAUCACAAACAGAUCUCUACUCUUUGAAGUUAGCGC<br>    UUGUAGCACAAAAUCUAGACAGAAGUAGGCAGCCGAAGAA<br>    AGCGUUCCCUUUUCCGACACGGGCGGAUCGCUAGCAAGUG<br>    CUUAACAGAGAAUGUGGGACGGGUUCUAUCUGUCAUAAGG<br>    GA<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    transcribe(&quot;DNA2.fasta&quot;, &quot;DNA2_transcribed.fasta&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;DNA2_transcribed.fasta&quot;):<br>        with open(&quot;DNA2_transcribed.fasta&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;No output file exists at DNA2_transcribed.fasta.&quot;)<br><br>Below is the expected output:<br><br>    &gt;TRANSCRIPT1<br>    UCGACGGCGGGAUGCACUGGCCGUGUCGCGGAA<br>    GGGCAUGUUUGCUCUAAAGCUACCAGGCCGGAG<br>    CCAAGAUGUGUCGUGGGUAUCUAAUUUCAUCUA<br>    GUUUCAUGUUCGGUGGAAACGCAG<br>    &gt;TRANSCRIPT2<br>    UGACUCUGGUUCGAUUAGUGAUGGGUAUCCACU<br>    CGCUUUCAUGAAUCAUCAGGAUCCGUGUGCAGU<br>    UUCGAUCCUCGUUAGCUAUGAAACGGGUAGAGC<br>    AGAGGAUGUAAAUCAGAUCCAUACGGCCCCUAA<br>    CCCGCAGGAAAUCUCCGA<br>    &gt;TRANSCRIPT3<br>    ACGCCUAGCCAGCUAGACGGUAGUAUAAUUGGU<br>    GAGGUUCCCAGGAUCACAUGAUCAAGGUUGGUU<br>    UGGAUUCCUCCACCAAAGCUGACUUUACAAAAC<br>    ACGCUGAUGCGCUCCAUCUUAAUUCAACAGCUC<br>    UGAACCCCUAAUGGGAUAACUUCUCUUCGCACU<br>    CCUCCUAUCCUGACGACUCCCCUUAAUAUCUCC<br>    CAGACAGAAUUGAUUUCAUACG<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA1.fasta<br><br>File contents:<br><br>    &gt;Gene1<br>    TGCTACAGGTGGCAAGAGTTGTATATTGGTATATAGACGA<br>    CTGGACGGCCTTTATAGCCTTACTAGGAGCAGTAGAAGCT<br>    TATGCCCAATTGAATCCT<br>    &gt;Gene2<br>    AACTCCGTCTGATCACCGTCATGGAAGGGCGGTATAGGAT<br>    ATGGTCGCTCCGCCGTGCAGCGGCTACGGGTC<br>    &gt;Gene3<br>    CTTTAAAAGCCGCACTATGCGATATCGAGGAACAGGAGTG<br>    CCTGCATCACAAACAGATCTCTACTCTTTGAAGTTAGCGC<br>    TTGTAGCACAAAATCTAGACAGAAGTAGGCAGCCGAAGAA<br>    AGCGTTCCCTTTTCCGACACGGGCGGATCGCTAGCAAGTG<br>    CTTAACAGAGAATGTGGGACGGGTTCTATCTGTCATAAGG<br>    GA<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA2.fasta<br><br>File contents:<br><br>    &gt;TRANSCRIPT1<br>    TCGACGGCGGGATGCACTGGCCGTGTCGCGGAA<br>    GGGCATGTTTGCTCTAAAGCTACCAGGCCGGAG<br>    CCAAGATGTGTCGTGGGTATCTAATTTCATCTA<br>    GTTTCATGTTCGGTGGAAACGCAG<br>    &gt;TRANSCRIPT2<br>    TGACTCTGGTTCGATTAGTGATGGGTATCCACT<br>    CGCTTTCATGAATCATCAGGATCCGTGTGCAGT<br>    TTCGATCCTCGTTAGCTATGAAACGGGTAGAGC<br>    AGAGGATGTAAATCAGATCCATACGGCCCCTAA<br>    CCCGCAGGAAATCTCCGA<br>    &gt;TRANSCRIPT3<br>    ACGCCTAGCCAGCTAGACGGTAGTATAATTGGT<br>    GAGGTTCCCAGGATCACATGATCAAGGTTGGTT<br>    TGGATTCCTCCACCAAAGCTGACTTTACAAAAC<br>    ACGCTGATGCGCTCCATCTTAATTCAACAGCTC<br>    TGAACCCCTAATGGGATAACTTCTCTTCGCACT<br>    CCTCCTATCCTGACGACTCCCCTTAATATCTCC<br>    CAGACAGAATTGATTTCATACG</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def transcribe(input_file, output_file):<br>    with open(input_file, &#39;r&#39;) as f:<br>        data = f.readlines()<br><br>    rna_sequences = []<br>    for line in data:<br>        if line[0] == &quot;&gt;&quot;:<br>            rna_sequences.append(line)<br>        else:<br>            rna_sequences.append(line.replace(&quot;T&quot;, &quot;U&quot;))<br><br>    with open(output_file, &#39;w&#39;) as f:<br>        f.writelines(rna_sequences)</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">5</td><td class="s1" dir="ltr">The first solution transcribed the sequences correctly but did not format the output file correctly in FASTA format. I gave it another example of how the output file should look. It just repeated that output back to me. Next it gave me a new solution, but the function name did not match what was requested in the prompt. I asked it to change the function name, but the parameters did not match the prompt. I gave it the original prompt, and it succeeded.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td></tr><tr style="height: 20px"><th id="2089702959R118" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">119</div></th><td class="s1">12 - Writing Text Files</td><td class="s1">07 - Find most common animal</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Please write a function called &quot;findMostCommonAnimal&quot; that accepts two arguments. The first argument will be the path to a file that contains information about animal sightings in U.S. national parks. The second argument will be a path to a file that your function will need to create.<br><br>The first line of the input file (first argument) will contain a header with the following column names: Location, Animal, and NumSightings. Each line in the file will indicate how many times a specific animal has been observed in a specific geographical location. These values (and the column names) will be separated by tabs. Your function should identify which animal was observed most frequently on average, across the national parks. Your function should write a statement to the output file such as, &quot;The most commonly observed animal was X&quot;, but should replace &quot;X&quot; with the actual animal name.<br><br>When writing to the output file, use f-string(s). Your function does not need to return a value.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    findMostCommonAnimal(&quot;data1.txt&quot;, &quot;output1.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;output1.txt&quot;):<br>        with open(&quot;output1.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;No output file exists at output1.txt.&quot;)<br><br>Below is the expected output:<br><br>    The most commonly observed animal was Black bear<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    findMostCommonAnimal(&quot;data2.txt&quot;, &quot;output2.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;output2.txt&quot;):<br>        with open(&quot;output2.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;No output file exists at output2.txt.&quot;)<br><br>Below is the expected output:<br><br>    The most commonly observed animal was Grizzly bear<br><br>Below is a description of a data file that is available for testing.<br><br>File name: data1.txt<br><br>File contents:<br><br>    Location        Animal        NumSightings<br>    Yellowstone National Park        Black bear        42<br>    Yellowstone National Park        Grizzly bear        23<br>    Yellowstone National Park        Brown bear        128<br>    Yosemite National Park        Black bear        105<br>    Yosemite National Park        Grizzly bear        33<br>    Yosemite National Park        Brown bear        1<br><br>Below is a description of a data file that is available for testing.<br><br>File name: data2.txt<br><br>File contents:<br><br>    Location        Animal        NumSightings<br>    Zion National Park        Black bear        18<br>    Zion National Park        Grizzly bear        42<br>    Zion National Park        Brown bear        3<br>    Bryce Canyon National Park        Black bear        15<br>    Bryce Canyon National Park        Grizzly bear        1<br>    Bryce Canyon National Park        Brown bear        33</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def findMostCommonAnimal(inputFile, outputFile):<br>    animals = {}<br>    with open(inputFile) as file:<br>        # skip the header line<br>        next(file)<br>        for line in file:<br>            location, animal, num = line.strip().split(&#39;\t&#39;)<br>            if animal not in animals:<br>                animals[animal] = []<br>            animals[animal].append(int(num))<br><br>    avg_sightings = {}<br>    for animal, sightings in animals.items():<br>        avg_sightings[animal] = sum(sightings) / len(sightings)<br><br>    most_common = max(avg_sightings, key=avg_sightings.get)<br>    with open(outputFile, &#39;w&#39;) as file:<br>        file.write(f&quot;The most commonly observed animal was {most_common}&quot;)</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R119" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">120</div></th><td class="s1">13 - Advanced Functions &amp; Additional Practice</td><td class="s1">01 - Check parameters</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;checkIfGreater&quot; that accepts two arguments. Both values will be integers. Your function should evaluate whether the first argument value is greater than the value of the second argument. The names of the arguments should be &quot;int1&quot; and &quot;int2&quot;, respectively. The default values of these arguments should both be 0. The tests will check your code by using keyword arguments to make sure it is functioning correctly, so be sure to name your arguments correctly.<br><br>Your function should return &quot;True&quot; if the value of &quot;int1&quot; is greater than the value of &quot;int2&quot;; otherwise, it should return &quot;False&quot;. Your function should also check the argument values to make sure they are integers. If either of the values is not an integer, your function should return &quot;The input values were invalid.&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkIfGreater(4, 3))<br>    print(checkIfGreater(2, 5))<br><br>Below is the expected output:<br><br>    True<br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkIfGreater(int1 = 1, int2 = 4))<br>    print(checkIfGreater(int2 = 2, int1 = 5))<br><br>Below is the expected output:<br><br>    False<br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkIfGreater(2, int2 = 5))<br>    print(checkIfGreater(int2 = 5))<br><br>Below is the expected output:<br><br>    False<br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkIfGreater(int1 = 3))<br>    print(checkIfGreater(int2 = 3, int1 = 3))<br><br>Below is the expected output:<br><br>    True<br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkIfGreater())<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkIfGreater(int1 = 1, int2 = &quot;4&quot;))<br>    print(checkIfGreater(int2 = 6, int1 = &quot;2&quot;))<br><br>Below is the expected output:<br><br>    The input values were invalid.<br>    The input values were invalid.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def checkIfGreater(int1=0, int2=0):<br>    if not (isinstance(int1, int) and isinstance(int2, int)):<br>        return &quot;The input values were invalid.&quot;<br>    return int1 &gt; int2</td><td class="s2">6</td><td class="s2" dir="ltr">6</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R120" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">121</div></th><td class="s1">13 - Advanced Functions &amp; Additional Practice</td><td class="s1">02 - Describe genome</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;describeGenome&quot; that accepts two arguments. The first argument will be a string; it should be called &quot;genome&quot; and should have a default value of &quot;human&quot;. The second argument will be an integer; it should be called &quot;numChromosomePairs&quot; and should have a default value of &quot;23&quot;. The tests will check your code by using keyword arguments to make sure it is functioning correctly, so be sure to name your arguments correctly.<br><br>Your function should use the &quot;genome&quot; and &quot;numChromosomePairs&quot; values to return a string that indicates how many pairs of chromosomes are present in the specified genome. For example, if the default values are used, your function would return, &quot;The human genome has 23 pairs of chromosomes.&quot; Or if the &quot;genome&quot; argument were &quot;fruit fly&quot; and the &quot;numChromosomes&quot; argument were 4, your function would return, &quot;The fruit fly genome has 4 pairs of chromosomes.&quot;<br><br>Your function should use an f-string.<br><br>Hint(s):<br><br>    Hint: Use string formatting to construct the string.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome())<br><br>Below is the expected output:<br><br>    The human genome has 23 pairs of chromosomes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(&quot;fruit fly&quot;, 4))<br><br>Below is the expected output:<br><br>    The fruit fly genome has 4 pairs of chromosomes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(&quot;meadow vole&quot;))<br><br>Below is the expected output:<br><br>    The meadow vole genome has 23 pairs of chromosomes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(&quot;Stalked Adder’s tongue&quot;, 1260))<br><br>Below is the expected output:<br><br>    The Stalked Adder’s tongue genome has 1260 pairs of chromosomes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(numChromosomePairs = 4, genome = &quot;fruit fly&quot;))<br><br>Below is the expected output:<br><br>    The fruit fly genome has 4 pairs of chromosomes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(numChromosomePairs = 23))<br><br>Below is the expected output:<br><br>    The human genome has 23 pairs of chromosomes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(genome = &quot;meadow vole&quot;))<br><br>Below is the expected output:<br><br>    The meadow vole genome has 23 pairs of chromosomes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(numChromosomePairs = 24))<br><br>Below is the expected output:<br><br>    The human genome has 24 pairs of chromosomes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(describeGenome(genome=&quot;Stalked Adder&#39;s tongue&quot;, numChromosomePairs=1260))<br><br>Below is the expected output:<br><br>    The Stalked Adder&#39;s tongue genome has 1260 pairs of chromosomes.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def describeGenome(genome=&quot;human&quot;, numChromosomePairs=23):<br>    return f&quot;The {genome} genome has {numChromosomePairs} pairs of chromosomes.&quot;</td><td class="s2">9</td><td class="s2" dir="ltr">9</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R121" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">122</div></th><td class="s1">13 - Advanced Functions &amp; Additional Practice</td><td class="s1">03 - Get temperature extremes</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;getTemperatureExtremes&quot; that accepts a single argument: a list of float values. These numbers will indicate noontime temperatures (in degrees Celsius) observed at a specific geographic location on the first day of each month of a specific year. Your function should return two float values: 1) the lowest temperature in the list and 2) the highest temperature in the list.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    cairoMin, cairoMax = getTemperatureExtremes([15.4, 18.9, 26.7, 26.9, 33.5, 43.2, 37.8, 31.4, 33.8, 29.9, 23.4, 19.2])<br>    <br>    print(cairoMin)<br>    print(cairoMax)<br><br>Below is the expected output:<br><br>    15.4<br>    43.2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    stockholmMin, stockholmMax = getTemperatureExtremes([-2.3, -8.9, 5.4, 6.9, 15.5, 20.2, 32.8, 25.7, 27.4, 19.9, 15.0, 4.2])<br>    <br>    print(stockholmMin)<br>    print(stockholmMax)<br><br>Below is the expected output:<br><br>    -8.9<br>    32.8</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def getTemperatureExtremes(temperatures):<br>    return min(temperatures), max(temperatures)<br></td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R122" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">123</div></th><td class="s1">13 - Advanced Functions &amp; Additional Practice</td><td class="s1">04 - Check argument types</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;checkArgumentTypes&quot; that accepts four different arguments. The names of these arguments should be &quot;myString&quot;, &quot;myFloat&quot;, &quot;myInteger&quot;, and &quot;myBoolean&quot;. Your function should check that each of the values of these arguments are the correct type. The value of &quot;myString&quot; should be a string; the value of &quot;myFloat&quot; should be a float, etc. If all of the values are the correct type, your function should return a string that states, &quot;All of the input values were valid.&quot; Otherwise, your function should return a message that indicates which argument value(s) were invalid and what the invalid value(s) were.<br><br>For example, if an integer value of 3 were specified for &quot;myString&quot;, your error message would state, &quot;The value of myString [3] is not a String.&quot; If multiple values are invalid, your error message should indicate this. For example: &quot;The value of myString [3] is not a String.\nThe value of myBoolean [7] is not a Boolean.&quot; The order of these messages should coincide with the following order: String, Float, Integer, Boolean.<br><br>The tests will check your code using keyword arguments to make sure it is functioning correctly, so be sure to name your arguments correctly.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkArgumentTypes(&quot;hello&quot;, 3.0, 4, True))<br><br>Below is the expected output:<br><br>    All of the input values were valid.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkArgumentTypes(&quot;hello&quot;, 3, &quot;4&quot;, True))<br><br>Below is the expected output:<br><br>    The value of myFloat [3] is not a Float.<br>    The value of myInteger [4] is not an Integer.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkArgumentTypes(myFloat = 7.0, myBoolean = False, myString = &quot;hi&quot;, myInteger = 40))<br><br>Below is the expected output:<br><br>    All of the input values were valid.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(checkArgumentTypes(myBoolean = &quot;hello&quot;, myFloat = 8, myString = &quot;hi&quot;, myInteger = 40.0))<br><br>Below is the expected output:<br><br>    The value of myFloat [8] is not a Float.<br>    The value of myInteger [40.0] is not an Integer.<br>    The value of myBoolean [hello] is not a Boolean.</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def checkArgumentTypes(myString, myFloat, myInteger, myBoolean):<br>    error_message = []<br><br>    if not isinstance(myString, str):<br>        error_message.append(f&quot;The value of myString [{myString}] is not a String.&quot;)<br><br>    if not isinstance(myFloat, float):<br>        error_message.append(f&quot;The value of myFloat [{myFloat}] is not a Float.&quot;)<br><br>    if not isinstance(myInteger, int):<br>        error_message.append(f&quot;The value of myInteger [{myInteger}] is not an Integer.&quot;)<br><br>    if not isinstance(myBoolean, bool):<br>        error_message.append(f&quot;The value of myBoolean [{myBoolean}] is not a Boolean.&quot;)<br><br>    if error_message:<br>        return &quot;\n&quot;.join(error_message)<br>    else:<br>        return &quot;All of the input values were valid.&quot;</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R123" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">124</div></th><td class="s1">13 - Advanced Functions &amp; Additional Practice</td><td class="s1">05 - Calculate CD4 cells increase over baseline</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Researchers counted CD4 cells in HIV patients and assessed whether these counts increased after a year of treatment with a particular drug. The before and after counts are shown in &quot;cd4.txt&quot;, a data file with tab-separated values.<br><br>Please write a function called &quot;countDiffsGreaterThan&quot; that accepts two arguments: 1) the path to the data file and 2) a numeric value (a threshold). Calculate the number of times that the number of CD4 cells increased by more than the specified threshold and return this number.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(countDiffsGreaterThan(&quot;cd4.txt&quot;, 0))<br><br>Below is the expected output:<br><br>    18<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(countDiffsGreaterThan(&quot;cd4.txt&quot;, 1))<br><br>Below is the expected output:<br><br>    6<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(countDiffsGreaterThan(&quot;cd4.txt&quot;, 2))<br><br>Below is the expected output:<br><br>    2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(countDiffsGreaterThan(&quot;cd4.txt&quot;, 3))<br><br>Below is the expected output:<br><br>    0<br><br>Below is a description of a data file that is available for testing.<br><br>File name: cd4.txt<br><br>File contents:<br><br>    baseline        oneyear<br>    2.12        2.47<br>    4.35        4.61<br>    3.39        5.26<br>    2.51        3.02<br>    4.04        6.36<br>    5.1        5.93<br>    3.77        3.93<br>    3.35        4.09<br>    4.1        4.88<br>    3.35        3.81<br>    4.15        4.74<br>    3.56        3.29<br>    3.39        5.55<br>    1.88        2.82<br>    2.56        4.23<br>    2.96        3.23<br>    2.49        2.56<br>    3.03        4.31<br>    2.66        4.37<br>    3        2.4</td><td class="s1" dir="ltr"><span style="text-decoration:underline;-webkit-text-decoration-skip:none;text-decoration-skip-ink:none;color:#1155cc;"><a target="_blank" href="https://vincentarelbundock.github.io/Rdatasets/doc/boot/cd4.html">https://vincentarelbundock.github.io/Rdatasets/doc/boot/cd4.html</a></span><br>DiCiccio, T.J. and Efron B. (1996) Bootstrap confidence intervals (with Discussion). Statistical Science, 11, 189–228.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def countDiffsGreaterThan(file_path, threshold):<br>    with open(file_path, &quot;r&quot;) as f:<br>        lines = f.readlines()[1:]  # skip the header line<br>    <br>    count = 0<br>    for line in lines:<br>        baseline, oneyear = map(float, line.strip().split(&quot;\t&quot;))<br>        diff = oneyear - baseline<br>        if diff &gt; threshold:<br>            count += 1<br>    <br>    return count</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R124" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">125</div></th><td class="s1">13 - Advanced Functions &amp; Additional Practice</td><td class="s1">06 - Find average blood pressure per person - Part A</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Suppose that, as part of a biology class project, students were asked to measure their systolic blood pressure every class period. They saved this information into a tab-separated-value (TSV) file called &quot;systolic.txt&quot;. In this file, each row represents a date, and each column represents a student. The header line of the file contains student names for all columns (other than the first column).<br><br>Your goal is to find the average blood pressure of each student for a specific date ranges. However, this problem is a bit challenging, so we will do it in two steps. This is the first step.<br><br>Write a function called &quot;readData&quot; that accepts a single argument: an integer. This integer will represent a day of the month. For example, if the integer were &quot;16&quot;, it would refer to the date &quot;16-Jan&quot; in the file. This number might also refer to a day of the month that is not specified in the file (example: &quot;1&quot;).<br><br>Your function should read &quot;systolic.txt&quot;. It should ignore any blood-pressure values before the specified day of the month. Your function should evaluate the data for the remaining days. To get started, read the header line and store the students&#39; names in a list. Second, create a dictionary. Each key in the dictionary should be a student&#39;s name. Each value in the dictionary should be a list of blood-pressure values for that student. Before you start reading through the data line by line, create a key in the dictionary for each student and specify its value as an empty list. Then as you loop through each line in the file, add the data points to the dictionary.<br><br>As you loop through each line of the file, you have to figure out which data point corresponds to which student. If you created a list with students&#39; names (see above), then as you are looping through the file, you can convert each line of the file to a list that is &quot;parallel&quot; to the list of students&#39; names. This will make it easier to match the names with the data values.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # We will use a for loop to print the output so it is more readable.<br>    for key, value in readData(0).items():<br>        print(f&quot;{key}: {value}&quot;)<br><br>Below is the expected output:<br><br>    Sarah: [120, 121, 115, 116, 122, 118, 110, 120, 117, 119, 120, 121, 125]<br>    John: [135, 130, 145, 131, 129, 135, 139, 141, 133, 138, 146, 135, 126]<br>    Sam: [111, 111, 112, 109, 110, 111, 117, 108, 110, 111, 111, 110, 106]<br>    Rachel: [105, 110, 103, 109, 105, 106, 103, 105, 110, 102, 110, 105, 104]<br>    June: [140, 142, 138, 140, 141, 143, 137, 140, 139, 141, 142, 137, 140]<br>    Thomas: [133, 135, 128, 139, 129, 135, 134, 132, 128, 127, 131, 133, 134]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # We will use a for loop to print the output so it is more readable.<br>    for key, value in readData(6).items():<br>        print(f&quot;{key}: {value}&quot;)<br><br>Below is the expected output:<br><br>    Sarah: [120, 121, 115, 116, 122, 118, 110, 120, 117, 119, 120, 121, 125]<br>    John: [135, 130, 145, 131, 129, 135, 139, 141, 133, 138, 146, 135, 126]<br>    Sam: [111, 111, 112, 109, 110, 111, 117, 108, 110, 111, 111, 110, 106]<br>    Rachel: [105, 110, 103, 109, 105, 106, 103, 105, 110, 102, 110, 105, 104]<br>    June: [140, 142, 138, 140, 141, 143, 137, 140, 139, 141, 142, 137, 140]<br>    Thomas: [133, 135, 128, 139, 129, 135, 134, 132, 128, 127, 131, 133, 134]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # We will use a for loop to print the output so it is more readable.<br>    for key, value in readData(12).items():<br>        print(f&quot;{key}: {value}&quot;)<br><br>Below is the expected output:<br><br>    Sarah: [116, 122, 118, 110, 120, 117, 119, 120, 121, 125]<br>    John: [131, 129, 135, 139, 141, 133, 138, 146, 135, 126]<br>    Sam: [109, 110, 111, 117, 108, 110, 111, 111, 110, 106]<br>    Rachel: [109, 105, 106, 103, 105, 110, 102, 110, 105, 104]<br>    June: [140, 141, 143, 137, 140, 139, 141, 142, 137, 140]<br>    Thomas: [139, 129, 135, 134, 132, 128, 127, 131, 133, 134]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # We will use a for loop to print the output so it is more readable.<br>    for key, value in readData(18).items():<br>        print(f&quot;{key}: {value}&quot;)<br><br>Below is the expected output:<br><br>    Sarah: [110, 120, 117, 119, 120, 121, 125]<br>    John: [139, 141, 133, 138, 146, 135, 126]<br>    Sam: [117, 108, 110, 111, 111, 110, 106]<br>    Rachel: [103, 105, 110, 102, 110, 105, 104]<br>    June: [137, 140, 139, 141, 142, 137, 140]<br>    Thomas: [134, 132, 128, 127, 131, 133, 134]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # We will use a for loop to print the output so it is more readable.<br>    for key, value in readData(24).items():<br>        print(f&quot;{key}: {value}&quot;)<br><br>Below is the expected output:<br><br>    Sarah: [119, 120, 121, 125]<br>    John: [138, 146, 135, 126]<br>    Sam: [111, 111, 110, 106]<br>    Rachel: [102, 110, 105, 104]<br>    June: [141, 142, 137, 140]<br>    Thomas: [127, 131, 133, 134]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # We will use a for loop to print the output so it is more readable.<br>    for key, value in readData(31).items():<br>        print(f&quot;{key}: {value}&quot;)<br><br>Below is the expected output:<br><br>    Sarah: []<br>    John: []<br>    Sam: []<br>    Rachel: []<br>    June: []<br>    Thomas: []<br><br>Below is a description of a data file that is available for testing.<br><br>File name: systolic.txt<br><br>File contents:<br><br>    Date        Sarah        John        Sam        Rachel        June        Thomas<br>    6-Jan        120        135        111        105        140        133<br>    8-Jan        121        130        111        110        142        135<br>    10-Jan        115        145        112        103        138        128<br>    12-Jan        116        131        109        109        140        139<br>    14-Jan        122        129        110        105        141        129<br>    16-Jan        118        135        111        106        143        135<br>    18-Jan        110        139        117        103        137        134<br>    20-Jan        120        141        108        105        140        132<br>    22-Jan        117        133        110        110        139        128<br>    24-Jan        119        138        111        102        141        127<br>    26-Jan        120        146        111        110        142        131<br>    28-Jan        121        135        110        105        137        133<br>    30-Jan        125        126        106        104        140        134</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def readData(day):<br>    students = []<br>    data = {}<br>    with open(&quot;systolic.txt&quot;, &quot;r&quot;) as file:<br>        header = file.readline().strip().split(&quot;\t&quot;)[1:]<br>        students = [student for student in header]<br>        for student in students:<br>            data[student] = []<br>        for line in file:<br>            line = line.strip().split(&quot;\t&quot;)<br>            date = int(line[0].split(&quot;-&quot;)[0])<br>            if date &gt;= day:<br>                for i in range(len(students)):<br>                    data[students[i]].append(int(line[i+1]))<br>    return data</td><td class="s2">6</td><td class="s2" dir="ltr">6</td><td class="s2" dir="ltr">3</td><td class="s1" dir="ltr">The first two solutions gave me errors. I explained this and re-gave it the full prompt; it eventually succeeded.</td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R125" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">126</div></th><td class="s1">13 - Advanced Functions &amp; Additional Practice</td><td class="s1">07 - Find average blood pressure per person - Part B</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>This exercise builds on the previous one. First, copy and paste the &quot;readData&quot; function that you just created.<br><br>Then write a function called &quot;saveAverages&quot; that accepts two arguments: an integer and a string. The integer will represent a day of the month. For example, if the integer were &quot;16&quot;, it would refer to the date &quot;16-Jan&quot; in the file. This number might also refer to a day of the month that is not specified in the file (example: &quot;1&quot;). The second argument (string) will be the name of a text file that your function should create.<br><br>Your &quot;saveAverages&quot; function should invoke your &quot;readData&quot; function, relaying the argument indicating the day of the month. With the resulting dictionary, your function should create a tab-delimited text file. This file should have two columns. The first column should store the names of students. The second column should store each person&#39;s average blood pressure (rounded to one decimal place). The rows should be sorted based on the names in the first column. If no measurements are available, store &quot;Not available&quot; in the second column. Your function does not need to return anything.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    saveAverages(0, &quot;output.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;output.txt&quot;):<br>        with open(&quot;output.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;No output file exists at output.txt.&quot;)<br><br>Below is the expected output:<br><br>    Student        Average_BP<br>    John        135.6<br>    June        140.0<br>    Rachel        105.9<br>    Sam        110.5<br>    Sarah        118.8<br>    Thomas        132.2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    saveAverages(6, &quot;output.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;output.txt&quot;):<br>        with open(&quot;output.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;No output file exists at output.txt.&quot;)<br><br>Below is the expected output:<br><br>    Student        Average_BP<br>    John        135.6<br>    June        140.0<br>    Rachel        105.9<br>    Sam        110.5<br>    Sarah        118.8<br>    Thomas        132.2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    saveAverages(12, &quot;output.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;output.txt&quot;):<br>        with open(&quot;output.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;No output file exists at output.txt.&quot;)<br><br>Below is the expected output:<br><br>    Student        Average_BP<br>    John        135.3<br>    June        140.0<br>    Rachel        105.9<br>    Sam        110.3<br>    Sarah        118.8<br>    Thomas        132.2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    saveAverages(18, &quot;output.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;output.txt&quot;):<br>        with open(&quot;output.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;No output file exists at output.txt.&quot;)<br><br>Below is the expected output:<br><br>    Student        Average_BP<br>    John        136.9<br>    June        139.4<br>    Rachel        105.6<br>    Sam        110.4<br>    Sarah        118.9<br>    Thomas        131.3<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    saveAverages(24, &quot;output.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;output.txt&quot;):<br>        with open(&quot;output.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;No output file exists at output.txt.&quot;)<br><br>Below is the expected output:<br><br>    Student        Average_BP<br>    John        136.2<br>    June        140.0<br>    Rachel        105.2<br>    Sam        109.5<br>    Sarah        121.2<br>    Thomas        131.2<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    saveAverages(31, &quot;output.txt&quot;)<br>    <br>    # This code will read your file and print its contents so it can be matched against the expected output.<br>    if os.path.exists(&quot;output.txt&quot;):<br>        with open(&quot;output.txt&quot;) as outputFile:<br>            for line in outputFile:<br>                print(line, end=&quot;&quot;)<br>    else:<br>        print(&quot;No output file exists at output.txt.&quot;)<br><br>Below is the expected output:<br><br>    Student        Average_BP<br>    John        Not available<br>    June        Not available<br>    Rachel        Not available<br>    Sam        Not available<br>    Sarah        Not available<br>    Thomas        Not available<br><br>Below is a description of a data file that is available for testing.<br><br>File name: systolic.txt<br><br>File contents:<br><br>    Date        Sarah        John        Sam        Rachel        June        Thomas<br>    6-Jan        120        135        111        105        140        133<br>    8-Jan        121        130        111        110        142        135<br>    10-Jan        115        145        112        103        138        128<br>    12-Jan        116        131        109        109        140        139<br>    14-Jan        122        129        110        105        141        129<br>    16-Jan        118        135        111        106        143        135<br>    18-Jan        110        139        117        103        137        134<br>    20-Jan        120        141        108        105        140        132<br>    22-Jan        117        133        110        110        139        128<br>    24-Jan        119        138        111        102        141        127<br>    26-Jan        120        146        111        110        142        131<br>    28-Jan        121        135        110        105        137        133<br>    30-Jan        125        126        106        104        140        134</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def readData(day):<br>    students = []<br>    data = {}<br>    with open(&quot;systolic.txt&quot;, &quot;r&quot;) as file:<br>        header = file.readline().strip().split(&quot;\t&quot;)[1:]<br>        students = [student for student in header]<br>        for student in students:<br>            data[student] = []<br>        for line in file:<br>            line = line.strip().split(&quot;\t&quot;)<br>            date = int(line[0].split(&quot;-&quot;)[0])<br>            if date &gt;= day:<br>                for i in range(len(students)):<br>                    data[students[i]].append(int(line[i+1]))<br>    return data<br><br><br>def saveAverages(day, filename):<br>    data = readData(day)<br>    with open(filename, &#39;w&#39;) as file:<br>        file.write(&quot;Student\tAverage_BP\n&quot;)<br>        for student in sorted(data.keys()):<br>            if len(data[student]) == 0:<br>                file.write(student + &quot;\tNot available\n&quot;)<br>            else:<br>                avg_bp = round(sum(data[student])/len(data[student]), 1)<br>                file.write(student + &quot;\t&quot; + str(avg_bp) + &quot;\n&quot;)</td><td class="s2">6</td><td class="s2" dir="ltr">6</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The initial solution did not pass because ChatGPT did not include the readData function from the previous problem in its output. At first, I tried to get it to do that, but it didn&#39;t understand. But I didn&#39;t want to penalize ChatGPT for this error because it was more of a logistical issue. When I copied and pasted the readData function into the autograder, ChatGPT&#39;s solution passed.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R126" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">127</div></th><td class="s1">13 - Advanced Functions &amp; Additional Practice</td><td class="s1">08 - Calculate average gene expression</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Idiopathic pulmonary fibrosis (IPF) is a disease that causes lung tissue to form scars. This scarring prevents the person from breathing properly and eventually may cause lung failure. For most patients, physicians cannot determine a specific cause, and there is currently no cure. Accordingly, there is an urgent need for researchers to understand more about IPF.<br><br>Researchers at the University of Nevada School of Medicine performed a study to examine 115 people who had either been diagnosed with IPF or who were healthy controls. They used gene-expression microarrays to profile blood cells from each person. In hopes of understanding more about the disease, the researchers searched for genes that are expressed differently between the two groups (IPF vs. healthy). The researchers posted their raw data online. For this exercise, you will use simplified versions of the data.<br><br>Each line (not including the header) contains data for a single patient. The first column indicates whether each patient was healthy or had IPF. The remaining columns contain gene-expression values for specific genes. The gene identifiers (for example, &quot;ENSG00000159579&quot;) are from the Ensembl database.<br><br>The purpose of this exercise is to give you practice writing programs that use multiple functions. Please write functions with the following names:<br><br> &quot;convertStringsToFloats&quot;<br> &quot;retrieveValues&quot;<br> &quot;calculateAverage&quot;<br> &quot;getAverageGeneExpression&quot;<br><br>The &quot;convertStringsToFloats&quot; function should accept a single argument: a list of string values (these will be numeric values in string form). Your function should convert these strings to floats and create a new list that contains the floats. Lastly, your function should return the list of float values.<br><br>The &quot;retrieveValues&quot; function should accept three arguments: 1) the name of one of the gene-expression data files, 2) an Ensembl gene identifier, and 3) a phenotype (either &quot;Healthy&quot; or &quot;IPF&quot;). This function should iterate through the specified data file and retrieve expression values for the specified gene, but only for patients who match the specified phenotype. As you retrieve values from the file, store them (as string values) in a list and return that list.<br><br>The &quot;calculateAverage&quot; function should accept a single argument: a list of float values. This function should calculate and return the average of these values.<br><br>The &quot;getAverageGeneExpression&quot; function should accepts two arguments: 1) the name of one of the data files and 2) an Ensembl gene identifier. This function should return two values: 1) the average expression for the specified gene for the &quot;Healthy&quot; phenotype and 2) the average expression for the specified gene for the &quot;IPF&quot; phenotype. To obtain these values, invoke the &quot;retrieveValues&quot;, &quot;convertStringsToFloats&quot;, and &quot;calculateAverage&quot; functions from within the &quot;getAverageGeneExpression&quot; function.<br><br>The test code will invoke each of these functions.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # We will first check to make sure the helper functions work<br>    print(convertStringsToFloats([&quot;5.1&quot;, &quot;2.3&quot;, &quot;4.4&quot;]))<br>    print(convertStringsToFloats([&quot;9.9&quot;, &quot;15.4&quot;, &quot;1.0&quot;]))<br><br>Below is the expected output:<br><br>    [5.1, 2.3, 4.4]<br>    [9.9, 15.4, 1.0]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # We will first check to make sure the helper functions work<br>    print(retrieveValues(&quot;GeneExpression1.txt&quot;, &quot;ENSG00000159579&quot;, &quot;IPF&quot;))<br>    print(retrieveValues(&quot;GeneExpression1.txt&quot;, &quot;ENSG00000159579&quot;, &quot;Healthy&quot;))<br>    print(retrieveValues(&quot;GeneExpression2.txt&quot;, &quot;ENSG00000260630&quot;, &quot;IPF&quot;))<br>    print(retrieveValues(&quot;GeneExpression2.txt&quot;, &quot;ENSG00000260630&quot;, &quot;Healthy&quot;))<br><br>Below is the expected output:<br><br>    [&#39;0.720443048666667&#39;, &#39;1.07926641116667&#39;, &#39;0.878639459166667&#39;, &#39;0.965804671833333&#39;, &#39;0.650749944&#39;, &#39;0.763675397833333&#39;, &#39;0.852651406&#39;, &#39;1.02372743&#39;, &#39;0.790027802333333&#39;, &#39;0.975916547666667&#39;, &#39;0.9513219925&#39;, &#39;0.963846274666667&#39;, &#39;0.840522914666667&#39;, &#39;0.913611431166667&#39;, &#39;0.7294963545&#39;, &#39;0.869973905&#39;, &#39;0.579357269833333&#39;, &#39;0.803148642333333&#39;, &#39;0.899024269666667&#39;, &#39;0.882625347666667&#39;, &#39;0.876259577&#39;, &#39;0.9961458805&#39;, &#39;0.597356628833333&#39;, &#39;0.941969501833333&#39;, &#39;1.02417957983333&#39;, &#39;0.800099906166667&#39;, &#39;0.920928301833333&#39;, &#39;0.887203205&#39;, &#39;0.7415509635&#39;, &#39;0.918489527166667&#39;, &#39;0.735468456833333&#39;, &#39;0.85071585&#39;, &#39;0.931519249&#39;, &#39;0.829510049333333&#39;, &#39;1.01348211833333&#39;, &#39;0.717858250166667&#39;, &#39;0.5559225695&#39;, &#39;0.774489637833333&#39;, &#39;0.587487637333333&#39;, &#39;0.684169514833333&#39;, &#39;0.711292474333333&#39;, &#39;0.729910574&#39;, &#39;0.576948023833333&#39;, &#39;0.619548850833333&#39;, &#39;0.6084259785&#39;, &#39;0.739188058&#39;, &#39;0.633424281666667&#39;, &#39;0.761322311166667&#39;, &#39;0.836358715833333&#39;, &#39;0.6757913565&#39;, &#39;0.682284854333333&#39;, &#39;0.8334994795&#39;, &#39;0.675695947&#39;, &#39;0.858684911833333&#39;, &#39;0.815437001&#39;, &#39;0.7703783005&#39;, &#39;0.781148020666667&#39;, &#39;0.698799648&#39;, &#39;0.8354274325&#39;]<br>    [&#39;0.901418872166667&#39;, &#39;0.852689667&#39;, &#39;0.936391221166667&#39;, &#39;0.867417970166667&#39;, &#39;0.846749336666667&#39;, &#39;0.689366527&#39;, &#39;0.845154432&#39;, &#39;0.908416151666667&#39;, &#39;0.887684473166667&#39;, &#39;1.01546799933333&#39;, &#39;0.856701790833333&#39;, &#39;0.730538818&#39;, &#39;0.924559968833333&#39;, &#39;0.8006456655&#39;, &#39;0.7037035845&#39;, &#39;0.933686961333333&#39;, &#39;0.9048968025&#39;, &#39;0.999194695666667&#39;, &#39;0.858187995166667&#39;, &#39;0.945059782166667&#39;, &#39;0.970030449166667&#39;, &#39;0.949303858666667&#39;, &#39;0.849962371333333&#39;, &#39;0.944543867833333&#39;, &#39;0.723879957833333&#39;, &#39;0.78488339&#39;, &#39;0.744153883333333&#39;, &#39;0.872937969333333&#39;, &#39;0.835714428666667&#39;, &#39;0.779100297666667&#39;, &#39;0.918804511&#39;, &#39;0.925546609666667&#39;, &#39;0.9188464015&#39;, &#39;0.978587744&#39;, &#39;0.920691954&#39;]<br>    [&#39;-0.309696331428571&#39;, &#39;-0.182498225714286&#39;, &#39;-0.247980797857143&#39;, &#39;-0.0858778371428572&#39;, &#39;-0.223409471428571&#39;, &#39;-0.138069455&#39;, &#39;-0.202700226428571&#39;, &#39;-0.133333627857143&#39;, &#39;-0.231426917142857&#39;, &#39;-0.208714587857143&#39;, &#39;-0.255090378571429&#39;, &#39;-0.20140226&#39;, &#39;-0.242353669285714&#39;, &#39;-0.151470535&#39;, &#39;-0.303789417142857&#39;, &#39;-0.242421695&#39;, &#39;-0.221999645714286&#39;, &#39;-0.0679122935714286&#39;, &#39;-0.121675904285714&#39;, &#39;-0.161224825&#39;, &#39;-0.103521242142857&#39;, &#39;-0.194807208571429&#39;, &#39;-0.0343955064285714&#39;, &#39;-0.350856658571429&#39;, &#39;-0.200068457857143&#39;, &#39;-0.221881428571429&#39;, &#39;-0.225046937142857&#39;, &#39;-0.126005958571429&#39;, &#39;-0.346330067857143&#39;, &#39;-0.179361309285714&#39;, &#39;-0.28350006&#39;, &#39;-0.265007768571429&#39;, &#39;-0.219416427857143&#39;, &#39;-0.0915580107142857&#39;, &#39;-0.137512372142857&#39;, &#39;-0.328470062142857&#39;, &#39;-0.152487093571429&#39;, &#39;-0.170406159285714&#39;, &#39;-0.356719008571429&#39;, &#39;-0.225270512142857&#39;, &#39;-0.346393447142857&#39;, &#39;-0.239670892142857&#39;, &#39;-0.315177131428571&#39;, &#39;-0.286572312857143&#39;, &#39;-0.189694412857143&#39;, &#39;-0.228502737857143&#39;, &#39;-0.267193097857143&#39;, &#39;-0.146369179285714&#39;, &#39;-0.254121916428571&#39;, &#39;-0.280042720714286&#39;, &#39;-0.176163881428571&#39;, &#39;-0.228484878571429&#39;, &#39;-0.102995643571429&#39;, &#39;-0.294260162142857&#39;, &#39;-0.279911625&#39;, &#39;-0.418529934285714&#39;, &#39;-0.262981955714286&#39;, &#39;-0.217279261428571&#39;, &#39;-0.376431605714286&#39;]<br>    [&#39;-0.108228932142857&#39;, &#39;-0.227582886428571&#39;, &#39;-0.234746420714286&#39;, &#39;-0.17906784&#39;, &#39;-0.208843281428571&#39;, &#39;-0.366101175&#39;, &#39;-0.241616709285714&#39;, &#39;-0.168552873571429&#39;, &#39;-0.237061819285714&#39;, &#39;-0.247392382142857&#39;, &#39;-0.142178458571429&#39;, &#39;-0.292672374285714&#39;, &#39;-0.243790525714286&#39;, &#39;-0.194076037142857&#39;, &#39;-0.222449616428571&#39;, &#39;-0.229420534285714&#39;, &#39;-0.250636569285714&#39;, &#39;-0.228571152857143&#39;, &#39;-0.109752674285714&#39;, &#39;-0.210930146428571&#39;, &#39;-0.192667648571429&#39;, &#39;-0.200293036428571&#39;, &#39;-0.171904693571429&#39;, &#39;-0.217204077857143&#39;, &#39;-0.195743385&#39;, &#39;-0.175639716428571&#39;, &#39;-0.362762323571429&#39;, &#39;-0.12970407&#39;, &#39;-0.240267706428571&#39;, &#39;-0.233506495&#39;, &#39;-0.255999148571429&#39;, &#39;-0.229160798571429&#39;, &#39;-0.232046761428571&#39;, &#39;-0.212818317142857&#39;, &#39;-0.174095747857143&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # We will first check to make sure the helper functions work<br>    print(calculateAverage([8.1, 0.03, 100.0]))<br>    print(calculateAverage([9.9, 150.29, 19.02]))<br><br>Below is the expected output:<br><br>    36.04333333333333<br>    59.73666666666667<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # This function will invoke the getAverageGeneExpression function and print the output<br>    def testAndCheck(dataFilePath, geneID):<br>        healthy, ipf = getAverageGeneExpression(dataFilePath, geneID)<br>        print(f&quot;{geneID} and Healthy: {healthy}&quot;)<br>        print(f&quot;{geneID} and IPF: {ipf}&quot;)<br>    <br>    testAndCheck(&quot;GeneExpression1.txt&quot;, &quot;ENSG00000159579&quot;)<br>    testAndCheck(&quot;GeneExpression1.txt&quot;, &quot;ENSG00000072506&quot;)<br>    testAndCheck(&quot;GeneExpression1.txt&quot;, &quot;ENSG00000182013&quot;)<br><br>Below is the expected output:<br><br>    ENSG00000159579 and Healthy: 0.8721405831095237<br>    ENSG00000159579 and IPF: 0.8027492058559322<br>    ENSG00000072506 and Healthy: 0.932900728217143<br>    ENSG00000072506 and IPF: 1.0286842186779661<br>    ENSG00000182013 and Healthy: -0.13102794278571434<br>    ENSG00000182013 and IPF: -0.15293962288740914<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    # This function will invoke the getAverageGeneExpression function and print the output<br>    def testAndCheck(dataFilePath, geneID):<br>        healthy, ipf = getAverageGeneExpression(dataFilePath, geneID)<br>        print(f&quot;{geneID} and Healthy: {healthy}&quot;)<br>        print(f&quot;{geneID} and IPF: {ipf}&quot;)<br>    <br>    testAndCheck(&quot;GeneExpression2.txt&quot;, &quot;ENSG00000170369&quot;)<br>    testAndCheck(&quot;GeneExpression2.txt&quot;, &quot;ENSG00000161681&quot;)<br>    testAndCheck(&quot;GeneExpression2.txt&quot;, &quot;ENSG00000260630&quot;)<br><br>Below is the expected output:<br><br>    ENSG00000170369 and Healthy: 0.32694815965714286<br>    ENSG00000170369 and IPF: 0.30755882427118647<br>    ENSG00000161681 and Healthy: -0.16983490472023813<br>    ENSG00000161681 and IPF: -0.15705856525070627<br>    ENSG00000260630 and Healthy: -0.2162138953061224<br>    ENSG00000260630 and IPF: -0.22129571437046008<br><br>Below is a description of a data file that is available for testing.<br><br>File name: GeneExpression1.txt<br><br>File contents:<br><br>    Status        ENSG00000159579        ENSG00000072506        ENSG00000076351        ENSG00000182013<br>    Healthy        0.901418872166667        1.0453540916        -0.0625931783333333        -0.0955378835714286<br>    Healthy        0.852689667        1.086381914        0.0176642633333333        -0.122901103571429<br>    Healthy        0.936391221166667        1.0534241872        -0.0134109266666667        -0.238341259642857<br>    Healthy        0.867417970166667        0.8648321492        -0.101131300555556        -0.184759876785714<br>    Healthy        0.846749336666667        1.14715143        0.06090428        -0.116575965<br>    Healthy        0.689366527        0.8291884252        -0.0519287227777778        -0.137364478571429<br>    Healthy        0.845154432        1.0593610368        -0.0354302383333333        -0.0731305946428571<br>    Healthy        0.908416151666667        0.852182536        -0.0301351261111111        -0.15024738<br>    Healthy        0.887684473166667        0.9728788796        -0.107534338333333        -0.167761058571429<br>    Healthy        1.01546799933333        0.8561614732        0.0173485572222222        -0.1496827525<br>    Healthy        0.856701790833333        0.997844604        -0.04300791        -0.138005396071429<br>    Healthy        0.730538818        0.8026989732        -0.036602265        -0.156036952142857<br>    Healthy        0.924559968833333        0.979618708        0.078872005        -0.160277516071429<br>    Healthy        0.8006456655        0.8350474788        -0.0163587705555556        -0.1376621975<br>    Healthy        0.7037035845        0.880491632        -0.106547226666667        -0.162186849642857<br>    Healthy        0.933686961333333        1.1800993076        0.0884679672222222        -0.228896463214286<br>    Healthy        0.9048968025        0.8085201172        -0.0349399588888889        -0.119709941785714<br>    Healthy        0.999194695666667        1.0480830812        0.134185595555556        -0.131954875<br>    Healthy        0.858187995166667        0.7057345348        -0.0507829066666667        -0.13122804<br>    Healthy        0.945059782166667        1.0495981688        -0.0134510788888889        -0.121611294285714<br>    Healthy        0.970030449166667        1.1538201612        0.124746165        0.127027653214286<br>    Healthy        0.949303858666667        0.951182588        0.0493951061111111        -0.134408855357143<br>    Healthy        0.849962371333333        0.8543440908        -0.010503        -0.193122773214286<br>    Healthy        0.944543867833333        1.1286812156        -0.0817085294444445        -0.139713988214286<br>    Healthy        0.723879957833333        0.7400547208        -0.00547655222222222        -0.0768035935714286<br>    Healthy        0.78488339        0.6353006924        -0.200923601111111        -0.155326211071429<br>    Healthy        0.744153883333333        0.9008497412        -0.07053703        -0.0970768496428571<br>    Healthy        0.872937969333333        0.7996452684        -0.0233825588888889        -0.0762655957142857<br>    Healthy        0.835714428666667        0.7961986512        -0.0903501288888889        -0.144891903214286<br>    Healthy        0.779100297666667        0.7040681408        -0.01613057        -0.127893135714286<br>    Healthy        0.918804511        0.91016764000000006        -0.142328028888889        -0.0612514378571429<br>    Healthy        0.925546609666667        0.9745345532        0.0156268922222222        -0.170578621785714<br>    Healthy        0.9188464015        1.0738166176        0.0721684811111111        -0.144212220357143<br>    Healthy        0.978587744        0.9888923184        -0.0125857005555556        -0.164609526785714<br>    Healthy        0.920691954        0.9853163596        0.083731555        -0.102979059642857<br>    IPF        0.720443048666667        0.7338941568        9.84291111111114e-4        -0.128031595<br>    IPF        1.07926641116667        1.0300255204        0.075187275        -0.137688708214286<br>    IPF        0.878639459166667        0.8317262244        -0.130544552777778        -0.211162568571429<br>    IPF        0.965804671833333        1.043208338        0.01524487        -0.150379857857143<br>    IPF        0.650749944        0.532386004        -0.145610513888889        -0.119939463928571<br>    IPF        0.763675397833333        0.7091208648        -0.101152641666667        -0.122578410714286<br>    IPF        0.852651406        0.9704170852        0.0354188777777778        -0.191742822857143<br>    IPF        1.02372743        1.0860226916        -0.137424879444444        -0.134854125714286<br>    IPF        0.790027802333333        0.7476323872        -0.0991699511111111        -0.114474491428571<br>    IPF        0.975916547666667        0.9325066804        0.0303812661111111        -0.154771296785714<br>    IPF        0.9513219925        0.989518078        -0.166272451666667        -0.132009367142857<br>    IPF        0.963846274666667        0.8902793864        -0.0497895433333333        -0.176004594285714<br>    IPF        0.840522914666667        0.8192871796        -0.0922461683333333        -0.137833540714286<br>    IPF        0.913611431166667        0.9060389612        0.0506074005555555        -0.147349152142857<br>    IPF        0.7294963545        1.000832964        -0.142571270555556        -0.1457513175<br>    IPF        0.869973905        0.852444982        0.0193721011111111        -0.141593401071429<br>    IPF        0.579357269833333        0.9716875372        -0.051304025        -0.0610283596428571<br>    IPF        0.803148642333333        0.8946018148        -0.0425929161111111        -0.0671016092857143<br>    IPF        0.899024269666667        0.9185384152        -0.049246655        -0.123704763571429<br>    IPF        0.882625347666667        0.948106238        0.0465554        -0.166330613928571<br>    IPF        0.876259577        0.9131600172        -0.0527110466666667        -0.122539193214286<br>    IPF        0.9961458805        0.917440732        0.0227106677777778        -0.1473017175<br>    IPF        0.597356628833333        0.9407905148        -0.127197023333333        -0.281749252142857<br>    IPF        0.941969501833333        0.9450952256        0.0468287966666667        -0.124209095<br>    IPF        1.02417957983333        0.8916196848        0.0415853033333333        -0.182984304285714<br>    IPF        0.800099906166667        0.9388567496        -0.00173395833333334        -0.118866164285714<br>    IPF        0.920928301833333        0.8400791536        -0.0519121433333333        -0.139109325<br>    IPF        0.887203205        0.814396888        0.024027095        -0.1497251825<br>    IPF        0.7415509635        0.9221027608        -0.00304966722222222        -0.137841917142857<br>    IPF        0.918489527166667        0.8478946768        -0.0882695794444444        -0.148749078571429<br>    IPF        0.735468456833333        0.746920894        -0.178810784444444        -0.109697335357143<br>    IPF        0.85071585        0.9315748868        -0.101232737222222        -0.240460188928571<br>    IPF        0.931519249        0.9556234008        -0.0877214077777778        -0.0978784289285714<br>    IPF        0.829510049333333        0.9356840136        -0.167647637222222        -0.178547441428571<br>    IPF        1.01348211833333        0.9435583292        0.15070622        -0.1232065325<br>    IPF        0.717858250166667        1.2329865768        0.043022135        -0.141169659642857<br>    IPF        0.5559225695        0.9824713028        -0.00738506833333333        -0.230898773928571<br>    IPF        0.774489637833333        1.3324162752        0.149371057777778        -0.121583604642857<br>    IPF        0.587487637333333        0.9522709916        0.0214489638888889        -0.224078802142857<br>    IPF        0.684169514833333        1.2418631156        0.113565009444444        -0.176494091071429<br>    IPF        0.711292474333333        1.1715367356        -0.0389653472222222        -0.145883320357143<br>    IPF        0.729910574        1.2919526944        0.0406765461111111        -0.1960149575<br>    IPF        0.576948023833333        1.04785525        -0.019661200555555602        -0.163267704285714<br>    IPF        0.619548850833333        1.2166143008        0.0767119833333333        -0.153001948928571<br>    IPF        0.6084259785        1.1175399972        -0.0644907466666667        -0.170358809285714<br>    IPF        0.739188058        1.265155878        0.154262353333333        -0.2232026825<br>    IPF        0.633424281666667        1.249297868        -0.162023042777778        -0.160127986785714<br>    IPF        0.761322311166667        1.2572963856        0.149726498333333        -0.1518233175<br>    IPF        0.836358715833333        1.1939343144        0.126403649444444        -0.128558856428571<br>    IPF        0.6757913565        1.212361594        0.0171236027777778        -0.219700829642857<br>    IPF        0.682284854333333        1.3787991304        0.111369551111111        -0.175493336071429<br>    IPF        0.8334994795        1.5197838864        0.0997885744444444        -0.208016201071429<br>    IPF        0.675695947        1.3516450676        -0.00706929777777779        -0.0737236085714286<br>    IPF        0.858684911833333        1.1634479912        -0.0486958494444444        -0.169946564642857<br>    IPF        0.815437001        1.1576340604        -0.0319780072222222        -0.173870832857143<br>    IPF        0.7703783005        1.215186202        0.0385871327777778        -0.125840873214286<br>    IPF        0.781148020666667        1.3363902        -0.05590485        -0.119963378928571<br>    IPF        0.698799648        1.2859848704        -0.0380738138888889        -0.140019709642857<br>    IPF        0.8354274325        1.2248707768        -0.0110229611111111        -0.163202683571429<br><br>Below is a description of a data file that is available for testing.<br><br>File name: GeneExpression2.txt<br><br>File contents:<br><br>    Status        ENSG00000170369        ENSG00000161681        ENSG00000260630        ENSG00000115307        ENSG00000153560        ENSG00000099783        ENSG00000083635        ENSG00000248360        ENSG00000269741        ENSG00000219073        ENSG00000185477        ENSG00000111536        ENSG00000136425        ENSG00000254835        ENSG00000198399        ENSG00000010322        ENSG00000169188        ENSG00000128654        ENSG00000164167        ENSG00000124721        ENSG00000127415        ENSG00000176485        ENSG00000196772        ENSG00000138161        ENSG00000065427        ENSG00000274750        ENSG00000180992        ENSG00000106333        ENSG00000261371        ENSG00000134755        ENSG00000205771        ENSG00000244752        ENSG00000085274        ENSG00000184385        ENSG00000189132        ENSG00000131831        ENSG00000168060        ENSG00000138160        ENSG00000027075        ENSG00000106853        ENSG00000163530        ENSG00000138650        ENSG00000170502        ENSG00000119865        ENSG00000226887        ENSG00000158764        ENSG00000116406        ENSG00000189169        ENSG00000132612        ENSG00000118298        ENSG00000136868        ENSG00000035681        ENSG00000133665        ENSG00000104889        ENSG00000087448        ENSG00000177051        ENSG00000243775        ENSG00000175646        ENSG00000062650        ENSG00000170802        ENSG00000154767        ENSG00000160505        ENSG00000139209        ENSG00000221953        ENSG00000159208        ENSG00000221878        ENSG00000255559        ENSG00000122591        ENSG00000172264        ENSG00000153823        ENSG00000100092        ENSG00000214380        ENSG00000184083        ENSG00000153233        ENSG00000149090        ENSG00000280414        ENSG00000204703        ENSG00000102271        ENSG00000151650        ENSG00000175920        ENSG00000099338        ENSG00000163959        ENSG00000143321        ENSG00000085662        ENSG00000173567        ENSG00000138642        ENSG00000179262        ENSG00000130005        ENSG00000185085        ENSG00000136450        ENSG00000125775        ENSG00000109610        ENSG00000175348        ENSG00000173818        ENSG00000126890        ENSG00000197125        ENSG00000166435        ENSG00000213760        ENSG00000124784        ENSG00000099960<br>    Healthy        0.346757602        -0.174518154479167        -0.108228932142857        0.993645367777778        1.02375798453333        1.66558000895522        0.20972439375        -0.190990445263158        0.7867176        -0.188769945        0.7601168636        -0.332861712727273        -0.179423339        -0.417491693333333        1.12235049923567        0.530160854380952        0.2581987134375        0.343646615        0.959725795        -0.154088635841584        0.12543940625        0.642487489        -0.24001035125        -0.0417506764444444        1.22429580082353        0.6199598575        0.329241928888889        0.0139117104        2.58933184388889        0.383468468591549        -0.1036352625        0.10724995        0.918787524583333        0.163495432857143        0.21177917125        -0.00641235625        -0.0625360903278689        -0.030448744691358        1.6399680515625        -0.140580364230769        -0.368561024444444        -0.0738890358730159        0.223735242432432        0.0455100231818182        -0.160047194347826        -0.266950949642857        1.27392101263736        -0.03404918        0.973685015        -0.103104792        1.196865259        1.22214349808989        -0.185203761515152        0.270724282727273        0.302460385555556        0.113290378095238        -0.112950541666667        -0.0925370685714286        1.43359292823529        1.241639853125        0.812508078641975        -0.22181350375        -0.205935675762712        -0.104997686666667        -0.0487296363636364        -0.100336031428571        -0.172426758888889        1.2756737892        -0.097253304375        0.441575950277778        0.397299853571429        0.100865412        0.0233730834246575        -0.221040782142857        -0.206588772391304        -0.0220289458333333        -0.163241129        -0.184009954230769        0.1269512375        0.154358077272727        0.0654918435869565        -0.0309875029032258        1.037277333125        0.074024324        -0.117555903428571        0.600334866282051        0.76699535        0.0334089890625        0.121851682        2.03864926645161        0.0372988716666667        -0.232393518        0.895285604516129        0.01433155275        -0.391907055        -0.242843601538462        0.140652152666667        -0.27465252        0.764035159473684        -0.082768680625<br>    Healthy        0.356500394        -0.161735866770833        -0.227582886428571        1.1013562812963        0.958431344266667        1.74575628156716        0.29370186625        -0.193523703157895        0.30023352        -0.10104922        0.5938258408        -0.348633742727273        -0.169633582        -0.35198374        0.99662136388535        0.503620751809524        0.268481875625        0.3642224678125        1.07959624142857        -0.151249570825083        0.0667248904166667        0.68183934        -0.30390552125        -0.0689791144444444        1.44320716435294        0.7157277525        0.525291082222222        -0.0754882604        2.61502041814815        0.455301346056338        -0.19494601        -0.162424532142857        0.778948571875        0.0861112935714286        0.06464488        -0.063918333125        -0.0860352562295082        -0.0708332183950617        1.2119544465625        -0.0778439319230769        -0.301880673333333        -0.105248322222222        0.279567273513514        0.0265917086363636        -0.134490842608696        -0.255536307142857        1.21385066692308        -0.04934389        0.9610362375        -0.162784205333333        1.120301911        1.19260041764045        -0.213064128181818        0.185783529393939        0.179620263888889        0.0486329152380952        -0.227866623333333        -0.115317645        1.39804520811765        1.284099318125        0.680236019506173        -0.26472482        -0.214223422711864        -0.112811595833333        -0.126534638409091        -0.182422342857143        -0.13019585        1.1124663262        -0.18910746125        0.684363483888889        0.387161068214286        0.034888108        0.0247396347945205        -0.232548076571429        -0.23865471326087        -0.06887642125        -0.153988891        -0.171812899807692        0.131374819375        0.173203133030303        -0.0889097843478261        -0.0171494477419355        1.0542583528125        0.132337925        -0.0910714294285714        0.463496392692308        0.667663151282051        0.054374680625        0.113942256        2.06948690870968        0.0319832129166667        -0.161800864        0.940962987741935        0.0368894095        -0.345791173        -0.173506808461538        0.0653697277333333        -0.2515425025        0.655838233333333        -0.10904323<br>    Healthy        0.253053196        -0.1608140796875        -0.234746420714286        1.36766454888889        1.22959223493333        1.6145675288806        0.23156811625        -0.154911018421053        0.001465175        -0.191992781        0.767249118        -0.332130297727273        -0.1558193195        -0.269563543333333        1.23629676630573        0.75173987647619        0.450806596875        0.402687691875        1.00012756857143        -0.177604520693069        0.301888603333333        0.539453451        -0.29755323625        -0.122153299555556        1.25027229082353        0.4451595925        0.551149574444444        0.1057094576        2.42688847        0.338486487605634        -0.036206385        0.00698041928571428        0.928432507083333        -0.0371892721428571        0.08990331        -0.125580075625        -4.05823114754101e-4        -0.0116817590123457        1.9197938815625        -0.110930815769231        -0.172173048888889        -0.167861114444444        0.277692827027027        -0.0463091981818182        -0.10364119        -0.240951431071429        1.30755896692308        0.0892468655555556        1.1972410075        -0.0854643183333333        0.890941022        1.21466797224719        -0.212152043333333        0.34891750030303        0.314121334444444        0.274626358571429        -0.238867458333333        -0.0628528035714286        1.38526361070588        1.4465148465625        0.792639531604938        -0.277256770714286        -0.170604457627119        -0.0363703533333333        -0.0760004802272727        -0.161203588571429        -0.190463115555556        1.1960485294        -0.113290229375        0.294676568888889        0.623604905357143        0.064068246        0.101625702191781        -0.228188884714286        -0.227199078913043        0.0572050204166667        -0.233173965        -0.0902616830769231        0.238056065        0.217234224848485        -0.0687503759782609        0.0462727912903226        1.27758343625        0.033579787        -0.0661425274285714        0.65073835051282        0.96154568        0.1298855340625        0.29332837        1.9258131383871        0.0578079216666667        -0.0713428705        1.04373549935484        0.106660731        -0.389627101        -0.299206089230769        0.6665902812        -0.3377424875        0.777697717894737        -0.09070101875<br>    Healthy        0.291168824        -0.214762561770833        -0.17906784        1.08623973648148        1.0792355996        1.69535385970149        0.17193059375        -0.117556268947368        0.517725615        -0.042994625        0.6378130548        -0.377371593181818        -0.2571701315        -0.396653956666667        1.33536117636943        0.532123248476191        0.2385669734375        0.3060323790625        0.826938717142857        -0.16243777356435599        0.0572935479166667        0.334573332        -0.433977575        -0.00596412555555555        1.24586859141176        0.728016145        0.359246341111111        0.0409488036        2.52179200333333        0.260984332394366        -0.00748596000000001        -0.06810473        0.984608805833333        0.0590286692857143        0.18525224875        -0.01276214375        -0.0587388901639344        -0.0696681809876543        1.4355508971875        -0.143415021923077        -0.232426743333333        -0.100689893492063        0.260334420810811        0.103126180454545        -0.13411560826087        -0.296782898928571        1.35185940351648        -0.192145243333333        0.883813235        -0.167162824333333        1.076400012        1.25678800382022        -0.25196927        0.167127321515152        0.314142718888889        0.0619312066666667        -0.399968125        -0.107997950714286        1.48234864658824        1.4451323375        0.787810382222222        -0.26079312375        -0.200227866949153        -0.145114806666667        -0.116847677272727        -0.155115595714286        -0.162667954444444        1.2399019286        -0.0158054825        0.5410380375        0.443510295        0.117775468        0.0367996615068493        -0.244699900285714        -0.209521868478261        -0.0592024779166667        -0.223043054        -0.212218364807692        0.212940640625        0.141153421515152        -0.10677155923913        0.0676968883870968        1.011923493125        -0.110778453        -0.056653368        0.641681878717949        0.709708893076923        -0.0382145375        0.042265442        1.92106102935484        0.103384511666667        -0.2211811585        0.695037273225806        -0.04583648175        -0.583641791        -0.268034353846154        0.538267832533333        -0.343655975        0.826769613157895        -0.05825890375<br>    Healthy        0.36357048        -0.120674191666667        -0.208843281428571        1.03592398055556        0.870399964133333        1.93606941253731        0.3589965575        -0.167238060526316        0.7043955        -0.117213627        0.7471522336        -0.268151695909091        -0.153926218        -0.20034166        0.908571675732484        0.568105250666667        0.3176476565625        0.340726815625        1.09965263357143        -0.18701834679868        0.14306197875        0.849455433        -0.27728298375        0.00356569666666667        1.43618045294118        0.6573796675        0.414124585555556        -0.005838812        2.50166228611111        0.399194312112676        0.0165586375        -0.351286731428571        0.686134221875        -0.04895674        0.06903035        -0.036466743125        -0.150297263606557        -0.085652020617284        1.7688339884375        -0.08005961        -0.285033098888889        -0.0969431285714286        0.242383048378378        0.0648121404545455        -0.107860243913043        -0.274200811071429        0.993129066923077        0.136495248888889        0.9170885525        -0.110343781        0.907259303        1.18796065539326        -0.216748057575758        0.169613512121212        0.319888007777778        0.198891736190476        -0.230979808333333        -0.235229054285714        1.50373212270588        1.1131451490625        0.679821310864198        -0.261619760357143        -0.22075726661017        -0.03169217        -0.108221572045455        0.01389991        -0.201875314444444        0.9965241734        -0.113491320625        0.526056638611111        0.462029115178571        0.117097824        0.0745193310958904        -0.239640270857143        -0.273837436304348        -0.04245830125        -0.134952537        -0.150869515384615        0.16711859        0.0839297681818182        -0.062206909673913        -0.0294534887096774        0.9356250753125        0.189770786        -0.119502913571429        0.417247788333333        0.789222308205128        0.087243855625        0.119010034        2.24680954935484        0.01744740375        -0.220081239        1.04846050870968        5.96929250000001e-4        -0.470914323        -0.314841553846154        0.133500616666667        -0.3850960825        0.57910340245614        -0.1528377825<br>    Healthy        0.268518826        -0.230150447083333        -0.366101175        1.24204881777778        1.04471567013333        1.76590341373134        -0.1185285725        -0.0986433115789474        0.00855644999999999        -0.25037999        0.7326153132        -0.325622696818182        -0.1347375995        -0.126538143333333        1.13675119299363        0.650298453809524        0.3098704596875        0.4484580171875        1.21586994214286        -0.164997194653465        0.0383535316666667        0.5823257        -0.35875289375        0.033302348        1.21860727941176        0.6904176225        0.498889185555556        -0.0617726148        2.28998670592593        0.393588969295775        0.06720431        -0.0359097228571428        0.851162787708333        0.0158391007142857        -0.00542449        -0.13570578        -0.0896537847540984        -0.0826587933333333        1.2335668384375        -0.059184005        -0.168506883333333        -0.0938464817460317        0.279623527027027        0.0492581286363636        -0.107687085652174        -0.285235420714286        1.39461745703297        0.0103739144444444        0.6610446725        -0.161514776666667        0.974584213        0.926475679101124        -0.322571637878788        0.133130203030303        0.216473536666667        0.0997798852380952        -0.222735505        -0.0656302735714286        1.10675925447059        1.4341418896875        0.762691779753086        -0.2503006075        -0.22148395440678        -0.255536965        -0.143717550909091        -0.200295377142857        -0.263938214444444        1.000270654        -0.1856174825        0.569621966111111        0.369030993214286        0.149046156        -0.0443002482191781        -0.211903178857143        -0.243350921304348        -0.129842240833333        -0.118011473        -0.194275378653846        0.014649766875        0.0758343524242424        -0.0802793916304348        -0.00737397838709677        1.126555679375        -0.1900047        -0.103599930571429        0.674010471025641        0.512733508461538        -0.0349183046875        -0.033020567        1.57843629870968        0.0604984158333333        -0.1270406945        0.903700428387097        0.01121044975        -0.657989685        -0.207333413076923        0.0922771910666667        -0.32683917        0.869072251929825        -0.054370970625<br>    Healthy        0.35872791        -0.119725247708333        -0.241616709285714        1.1257917162963        0.969110221466667        1.75199433320896        0.1551679        -0.0995196784210526        0.676672555        -0.216746933        0.6432686264        -0.312076558181818        -0.247610108        -0.19849657        1.02434842433121        0.600043064761905        0.29984540125        0.335386854375        1.17671679928571        -0.176008421320132        0.0981247195833333        0.7621251925        -0.34261090375        0.00778539177777778        1.44841412164706        0.73454446        0.403766166666667        -0.0610369596        2.47134176037037        0.496049294225352        -0.0417363375        -0.106606522142857        0.825828390208333        0.0370644714285714        0.32133517125        -0.0991504275        -0.0829155316393443        -0.106740891975309        1.47347493875        -0.104045513846154        -0.28286011        -0.100128019365079        0.438629388918919        -0.00982937636363636        -0.0734229730434783        -0.233022854642857        1.23659660857143        0.19763393        0.88526479        -0.158959958        1.061476943        1.06714904393258        -0.193892538181818        0.24080740030303        0.382560346666667        0.131303615238095        -0.284052941666667        -0.245216495        1.35867240705882        1.410140999375        0.631960024814815        -0.278490351964286        -0.266443043220339        -0.1159919325        -0.115567894090909        -0.32285518        -0.26595017        1.1251305384        -0.015780359375        0.377910265277778        0.494105655        0.053478572        0.0137009167123288        -0.275449821285714        -0.231747430217391        -0.0395196341666667        -0.1953441        -0.199086344807692        0.086675705        0.108928692424242        -0.0692496227173913        -0.0462666703225806        1.02874561375        0.038112303        -0.0949701208571429        0.465836403589744        0.601288373333333        -0.021568083125        0.185339423        2.10729910774194        0.0423623545833333        -0.1715166225        0.898349867419355        0.00293152799999999        -0.440821454        -0.228330847692308        0.0748618241333333        -0.02439218        0.651860672105263        -0.07256586875<br>    Healthy        0.340868194        -0.1573187196875        -0.168552873571429        1.06678700962963        1.34016872453333        1.66134403626866        0.3762336975        -0.0979810842105263        1.029417995        -0.300317874        0.9036126752        -0.318452289545455        -0.185442329        -0.363889306666667        1.22203377216561        0.641668646857143        0.270407484375        0.3298129025        0.831528434285714        -0.148994528514851        0.217826011458333        0.1762033955        -0.31714108875        0.0225166926666667        1.22233541247059        0.5614682        0.3188224        -0.0055674824        2.07261472666667        0.0305861292957746        -0.0337500125        0.106723210714286        0.8491103        0.0375724535714286        -0.06065037875        -0.161164504375        -0.0297345078688525        -0.0916141108641975        1.8382981990625        -0.150253473461538        -0.359655965555556        -0.113236367777778        0.117880032972973        0.0596115990909091        -0.142067710869565        -0.278041582142857        1.34831035879121        0.114141348888889        0.7010320575        -0.129520213        0.448593289        1.33686432764045        -0.224359236060606        0.0887772148484848        0.305790372222222        0.0328147452380952        -0.298907328333333        -0.0827835207142857        1.42916470141176        1.432839301875        0.776576978518518        -0.229271981607143        -0.225030452542373        -0.0927849708333334        -0.0476376306818182        -0.118907878571429        -0.273607218888889        1.1467312732        -0.111056365625        0.379521525277778        0.466516529821429        0.115378116        0.102594733835616        -0.230541474571429        -0.116758671086957        -0.0818317966666667        -0.26967451        -0.195637567115385        0.11161723875        0.212804946060606        -0.0666527870652174        -0.0425005838709677        1.10876396        -0.09087346        -0.11846254        0.573407577820513        0.786984337179487        0.106957305625        0.230409426        2.06272019419355        0.0103363758333333        -0.1374745195        0.760689949354839        0.032613467        -0.583048788        -0.216147775384615        0.169631714533333        -0.2335578        0.674638460877193        -0.082275248125<br>    Healthy        0.401934518        -0.1850600265625        -0.237061819285714        1.01580596388889        1.01287794226667        1.73045596365672        0.17893881125        -0.135094614736842        0.22830114        -0.162699884        0.709548602        -0.346844040454545        -0.205983528        -0.329395413333333        1.0427946343949        0.570675134095238        0.322154869375        0.2851922390625        0.961541688571429        -0.187647959405941        0.101037128333333        0.49755606        -0.35702362625        -0.01841985        1.32817586976471        0.76639426        0.45498165        0.0399365608        2.55561053333333        0.39595592056338        -0.0248294175        -0.191055620714286        0.643315649375        0.111061558571429        0.19364938125        -0.136175076875        -0.121396303770492        -0.0922232130864198        1.626044495625        -0.12981458        -0.388132556666667        -0.11734252904761901        0.0816290767567567        0.0669520559090909        -0.103677</td><td class="s1" dir="ltr">The data came from here: <span style="text-decoration:underline;-webkit-text-decoration-skip:none;text-decoration-skip-ink:none;color:#1155cc;"><a target="_blank" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE38958.">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE38958.</a></span></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def convertStringsToFloats(stringValues):<br>    floatValues = [float(value) for value in stringValues]<br>    return floatValues<br><br>def retrieveValues(fileName, geneId, phenotype):<br>    expressionValues = []<br>    with open(fileName) as file:<br>        lines = file.readlines()<br>        header = lines[0]<br>        geneIndex = header.split().index(geneId)<br>        for line in lines[1:]:<br>            data = line.split()<br>            if data[0] == phenotype:<br>                expressionValues.append(data[geneIndex])<br>    return expressionValues<br><br>def calculateAverage(values):<br>    average = sum(values) / len(values)<br>    return average<br><br>def getAverageGeneExpression(fileName, geneId):<br>    healthyValues = retrieveValues(fileName, geneId, &quot;Healthy&quot;)<br>    healthyValues = convertStringsToFloats(healthyValues)<br>    healthyAverage = calculateAverage(healthyValues)<br><br>    ipfValues = retrieveValues(fileName, geneId, &quot;IPF&quot;)<br>    ipfValues = convertStringsToFloats(ipfValues)<br>    ipfAverage = calculateAverage(ipfValues)<br><br>    return (healthyAverage, ipfAverage)</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R127" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">128</div></th><td class="s1">14 - Regular Expressions 1</td><td class="s1">01 - Does string contain pattern</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;containsTextPattern&quot; that accepts two arguments. The first argument will be the name of a text file that contains part of a classic biology paper. The second argument will be a string. Your function should read the specified input file and use the &quot;re&quot; module to evaluate whether the specified string (second argument) is found on any line of the file. Your function should return a Boolean value to indicate this.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(containsTextPattern(&quot;Gurdon.txt&quot;, r&quot;embryology&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(containsTextPattern(&quot;Gurdon.txt&quot;, r&quot;transplantation experiments in Xenopus&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(containsTextPattern(&quot;Kornberg.txt&quot;, r&quot;The papers were not accepted because of the term DNA in the title.&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(containsTextPattern(&quot;Kornberg.txt&quot;, r&quot;Biochimica et Biophysica Acta&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(containsTextPattern(&quot;Mendel.txt&quot;, r&quot;Vegetable&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(containsTextPattern(&quot;Mendel.txt&quot;, r&quot;differentiation&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Mendel.txt<br><br>File contents:<br><br>    EXPERIMENTS IN PLANT HYBRIDIZATION (1865)<br>    <br>    GREGOR MENDEL<br>    <br>    Read at the February 8th, and March 8th, 1865, meetings<br>    of the Brünn Natural History Society<br>    <br>    INTRODUCTORY REMARKS<br>    <br>    EXPERIENCE OF ARTIFICIAL FERTILIZATION, such as is effected with ornamental plants in order to obtain new variations in color, has led to the experiments which will here be discussed. The striking regularity with which the same hybrid forms always reappeared whenever fertilization took place between the same species induced further experiments to be undertaken, the object of which was to follow up the developments of the hybrids in their progeny.<br>    <br>    To this object numerous careful observers, such as Kölreuter, Gärtner, Herbert, Lecoq, Wichura and others, have devoted a part of their lives with inexhaustible perseverance. Gärtner especially in his work Die Bastarderzeugung im Pflanzenreiche [The Production of Hybrids in the Vegetable Kingdom], has recorded very valuable observations; and quite recently Wichura published the results of some profound investigations into the hybrids of the Willow. That, so far, no generally applicable law governing the formation and development of hybrids has been successfully formulated can hardly be wondered at by anyone who is acquainted with the extent of the task, and can appreciate the difficulties with which experiments of this class have to contend. A final decision can only be arrived at when we shall have before us the results of detailed experiments made on plants belonging to the most diverse orders.<br>    <br>    Those who survey the work done in this department will arrive at the conviction that among all the numerous experiments made, not one has been carried out to such an extent and in such a way as to make it possible to determine the number of different forms under which the offspring of the hybrids appear, or to arrange these forms with certainty according to their separate generations, or definitely to ascertain their statistical relations.<br>    <br>    It requires indeed some courage to undertake a labor of such far-reaching extent; this appears, however, to be the only right way by which we can finally reach the solution of a question the importance of which cannot be overestimated in connection with the history of the evolution of organic forms.<br>    <br>    The paper now presented records the results of such a detailed experiment. This experiment was practically confined to a small plant group, and is now, after eight years’ pursuit, concluded in all essentials. Whether the plan upon which the separate experiments were conducted and carried out was the best suited to attain the desired end is left to the friendly decision of the reader.<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Kornberg.txt<br><br>File contents:<br><br>    The early history of DNA polymerase: a commentary by Arthur Kornberg<br>    <br>    Department of Biochemistry, Stanford University School of Medicine, Stanford, CA (U.S.A.)<br>    <br>    on &#39;Enzymic synthesis of deoxyribonucleic acid&#39; by A. Kornberg, I.R. Lehman, M.J. Bessman and E.S. Simms Biochim. Biophys. Acta 21 (1956) 197-198<br>    <br>    The brief publication [1] in Biochimica et Biophysica Acta (BBA) entitled &quot;Enzymic Synthesis of Deoxyribonucleic Acid&quot;, submitted May 2, 1956, was the first such report of DNA synthesis, aside from an abstract which appeared in Federation Proceedings in March of that year [2]. In the BBA paper, we (see photograph, p. 55) showed that the incorporation of [a4C]thymidine into an acid-soluble, DNase-sensitive, alkali-stable product depended on partially purified enzyme fractions, ATP and DNA which was converted to an &#39;active primer&#39;. We expressed the brash hope of assembling &#39;genetically specific DNA&#39;.<br>    <br>    Two years later, we submitted two papers to the Journal of Biological Chemistry describing the 2000-fold purification of DNA polymerase and the enzymatic synthesis of the four deoxyribonucleoside triphosphates which the enzyme assembled into long chains. The papers were titled: &#39;The Enzymatic Synthesis of DNA.  I. Preparation of Substrates and Partial Purification of an Enzyme from Escheriehia coli. II. General Properties of the Reaction.&#39; The papers were not accepted because of the term &#39;DNA&#39; in the title. The consensus among the many referees after a protracted review was that the papers would be acceptable if we used the term &quot;polydeoxyribonucleotide&quot; to identify the synthetic product; the use of &#39;DNA&#39; could be justified only with a demonstration of genetic activity. To us this seemed unreasonable, because few if any DNA papers in the Journal had been required to fulfill this criterion. Fortunately, with accession of John Edsall to the editorship, the papers were accepted with DNA in the title [3,4]. <br>    <br>    [Full text available here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Gurdon.txt<br><br>File contents:<br><br>    The Developmental Capacity of Nuclei taken from Intestinal Epithelium Cells of Feeding Tadpoles<br>    <br>    by J. B. GURDON<br>    <br>    From the Embryology Laboratory, Department of Zoology, Oxford<br>    <br>    WITH ONE PLATE<br>    <br>    An important problem in embryology is whether the differentiation of cells depends upon a stable restriction of the genetic information contained in their nuclei. The technique of nuclear transplantation has shown to what extent the nuclei of differentiating cells can promote the formation of different cell types (e.g. King &amp; Briggs, 1956; Gurdon, 1960c). Yet no experiments have so far been published on the transplantation of nuclei from fully differentiated normal cells. This is partly because it is difficult to obtain meaningful results from such experiments. The small amount of cytoplasm in differentiated cells renders their nuclei susceptible to damage through exposure to the saline medium, and this makes it difficult to assess the significance of the abnormalities resulting from their transplantation. It is, however, very desirable to know the developmental capacity of such nuclei, since any nuclear changes which are necessarily involved in cellular differentiation must have already taken place in cells of this kind.<br>    <br>    The experiments described below are some attempts to transplant nuclei from fully differentiated cells. Many of these nuclei gave abnormal results after transplantation, and several different kinds of experiments have been carried out to determine the cause and significance of these abnormalities.<br>    <br>    The donor cells used for these experiments were intestinal epithelium cells of feeding tadpoles. This is the final stage of differentiation of many of the endoderm cells whose nuclei have already been studied by means of nuclear transplantation experiments in Xenopus. The results to be described here may therefore be regarded as an extension of those previously obtained from differentiating endoderm cells (Gurdon, 1960c).<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def containsTextPattern(fileName, pattern):<br>    with open(fileName, &quot;r&quot;) as file:<br>        contents = file.read()<br>        return bool(re.search(pattern, contents))</td><td class="s2">6</td><td class="s2" dir="ltr">6</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R128" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">129</div></th><td class="s1">14 - Regular Expressions 1</td><td class="s1">02 - Has &quot;X&quot; word</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;hasXWord&quot; that accepts one argument. This argument will be the name of a text file that contains part of a classic biology paper. Your function should read the specified file and use the &quot;re&quot; module to evaluate whether the file contains any word that starts with X. For simplicity, focus on words that are preceded by space characters (don&#39;t worry about words at the beginning of a line). Your function should return a Boolean value to indicate whether the file contains any word that starts with X.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(hasXWord(&quot;Gurdon.txt&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(hasXWord(&quot;Kornberg.txt&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(hasXWord(&quot;Mendel.txt&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Mendel.txt<br><br>File contents:<br><br>    EXPERIMENTS IN PLANT HYBRIDIZATION (1865)<br>    <br>    GREGOR MENDEL<br>    <br>    Read at the February 8th, and March 8th, 1865, meetings<br>    of the Brünn Natural History Society<br>    <br>    INTRODUCTORY REMARKS<br>    <br>    EXPERIENCE OF ARTIFICIAL FERTILIZATION, such as is effected with ornamental plants in order to obtain new variations in color, has led to the experiments which will here be discussed. The striking regularity with which the same hybrid forms always reappeared whenever fertilization took place between the same species induced further experiments to be undertaken, the object of which was to follow up the developments of the hybrids in their progeny.<br>    <br>    To this object numerous careful observers, such as Kölreuter, Gärtner, Herbert, Lecoq, Wichura and others, have devoted a part of their lives with inexhaustible perseverance. Gärtner especially in his work Die Bastarderzeugung im Pflanzenreiche [The Production of Hybrids in the Vegetable Kingdom], has recorded very valuable observations; and quite recently Wichura published the results of some profound investigations into the hybrids of the Willow. That, so far, no generally applicable law governing the formation and development of hybrids has been successfully formulated can hardly be wondered at by anyone who is acquainted with the extent of the task, and can appreciate the difficulties with which experiments of this class have to contend. A final decision can only be arrived at when we shall have before us the results of detailed experiments made on plants belonging to the most diverse orders.<br>    <br>    Those who survey the work done in this department will arrive at the conviction that among all the numerous experiments made, not one has been carried out to such an extent and in such a way as to make it possible to determine the number of different forms under which the offspring of the hybrids appear, or to arrange these forms with certainty according to their separate generations, or definitely to ascertain their statistical relations.<br>    <br>    It requires indeed some courage to undertake a labor of such far-reaching extent; this appears, however, to be the only right way by which we can finally reach the solution of a question the importance of which cannot be overestimated in connection with the history of the evolution of organic forms.<br>    <br>    The paper now presented records the results of such a detailed experiment. This experiment was practically confined to a small plant group, and is now, after eight years’ pursuit, concluded in all essentials. Whether the plan upon which the separate experiments were conducted and carried out was the best suited to attain the desired end is left to the friendly decision of the reader.<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Kornberg.txt<br><br>File contents:<br><br>    The early history of DNA polymerase: a commentary by Arthur Kornberg<br>    <br>    Department of Biochemistry, Stanford University School of Medicine, Stanford, CA (U.S.A.)<br>    <br>    on &#39;Enzymic synthesis of deoxyribonucleic acid&#39; by A. Kornberg, I.R. Lehman, M.J. Bessman and E.S. Simms Biochim. Biophys. Acta 21 (1956) 197-198<br>    <br>    The brief publication [1] in Biochimica et Biophysica Acta (BBA) entitled &quot;Enzymic Synthesis of Deoxyribonucleic Acid&quot;, submitted May 2, 1956, was the first such report of DNA synthesis, aside from an abstract which appeared in Federation Proceedings in March of that year [2]. In the BBA paper, we (see photograph, p. 55) showed that the incorporation of [a4C]thymidine into an acid-soluble, DNase-sensitive, alkali-stable product depended on partially purified enzyme fractions, ATP and DNA which was converted to an &#39;active primer&#39;. We expressed the brash hope of assembling &#39;genetically specific DNA&#39;.<br>    <br>    Two years later, we submitted two papers to the Journal of Biological Chemistry describing the 2000-fold purification of DNA polymerase and the enzymatic synthesis of the four deoxyribonucleoside triphosphates which the enzyme assembled into long chains. The papers were titled: &#39;The Enzymatic Synthesis of DNA.  I. Preparation of Substrates and Partial Purification of an Enzyme from Escheriehia coli. II. General Properties of the Reaction.&#39; The papers were not accepted because of the term &#39;DNA&#39; in the title. The consensus among the many referees after a protracted review was that the papers would be acceptable if we used the term &quot;polydeoxyribonucleotide&quot; to identify the synthetic product; the use of &#39;DNA&#39; could be justified only with a demonstration of genetic activity. To us this seemed unreasonable, because few if any DNA papers in the Journal had been required to fulfill this criterion. Fortunately, with accession of John Edsall to the editorship, the papers were accepted with DNA in the title [3,4]. <br>    <br>    [Full text available here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Gurdon.txt<br><br>File contents:<br><br>    The Developmental Capacity of Nuclei taken from Intestinal Epithelium Cells of Feeding Tadpoles<br>    <br>    by J. B. GURDON<br>    <br>    From the Embryology Laboratory, Department of Zoology, Oxford<br>    <br>    WITH ONE PLATE<br>    <br>    An important problem in embryology is whether the differentiation of cells depends upon a stable restriction of the genetic information contained in their nuclei. The technique of nuclear transplantation has shown to what extent the nuclei of differentiating cells can promote the formation of different cell types (e.g. King &amp; Briggs, 1956; Gurdon, 1960c). Yet no experiments have so far been published on the transplantation of nuclei from fully differentiated normal cells. This is partly because it is difficult to obtain meaningful results from such experiments. The small amount of cytoplasm in differentiated cells renders their nuclei susceptible to damage through exposure to the saline medium, and this makes it difficult to assess the significance of the abnormalities resulting from their transplantation. It is, however, very desirable to know the developmental capacity of such nuclei, since any nuclear changes which are necessarily involved in cellular differentiation must have already taken place in cells of this kind.<br>    <br>    The experiments described below are some attempts to transplant nuclei from fully differentiated cells. Many of these nuclei gave abnormal results after transplantation, and several different kinds of experiments have been carried out to determine the cause and significance of these abnormalities.<br>    <br>    The donor cells used for these experiments were intestinal epithelium cells of feeding tadpoles. This is the final stage of differentiation of many of the endoderm cells whose nuclei have already been studied by means of nuclear transplantation experiments in Xenopus. The results to be described here may therefore be regarded as an extension of those previously obtained from differentiating endoderm cells (Gurdon, 1960c).<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def hasXWord(file):<br>    with open(file, &quot;r&quot;) as f:<br>        text = f.read()<br>        pattern = r&quot;\sX\w+&quot;<br>        return bool(re.search(pattern, text))</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R129" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">130</div></th><td class="s1">14 - Regular Expressions 1</td><td class="s1">03 - Has consecutive uppercase words</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;hasConsecutiveUpperCaseWords&quot; that accepts one argument. This argument will be the name of a text file that contains part of a classic biology paper. Your function should read the specified file and use the &quot;re&quot; module to evaluate whether the file contains two consecutive words where the letters are all upper case (for example, &quot;GREGOR MENDEL&quot;). Your function should return a Boolean value to indicate this.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(hasConsecutiveUpperCaseWords(&quot;Gurdon.txt&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(hasConsecutiveUpperCaseWords(&quot;Kornberg.txt&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(hasConsecutiveUpperCaseWords(&quot;Mendel.txt&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Mendel.txt<br><br>File contents:<br><br>    EXPERIMENTS IN PLANT HYBRIDIZATION (1865)<br>    <br>    GREGOR MENDEL<br>    <br>    Read at the February 8th, and March 8th, 1865, meetings<br>    of the Brünn Natural History Society<br>    <br>    INTRODUCTORY REMARKS<br>    <br>    EXPERIENCE OF ARTIFICIAL FERTILIZATION, such as is effected with ornamental plants in order to obtain new variations in color, has led to the experiments which will here be discussed. The striking regularity with which the same hybrid forms always reappeared whenever fertilization took place between the same species induced further experiments to be undertaken, the object of which was to follow up the developments of the hybrids in their progeny.<br>    <br>    To this object numerous careful observers, such as Kölreuter, Gärtner, Herbert, Lecoq, Wichura and others, have devoted a part of their lives with inexhaustible perseverance. Gärtner especially in his work Die Bastarderzeugung im Pflanzenreiche [The Production of Hybrids in the Vegetable Kingdom], has recorded very valuable observations; and quite recently Wichura published the results of some profound investigations into the hybrids of the Willow. That, so far, no generally applicable law governing the formation and development of hybrids has been successfully formulated can hardly be wondered at by anyone who is acquainted with the extent of the task, and can appreciate the difficulties with which experiments of this class have to contend. A final decision can only be arrived at when we shall have before us the results of detailed experiments made on plants belonging to the most diverse orders.<br>    <br>    Those who survey the work done in this department will arrive at the conviction that among all the numerous experiments made, not one has been carried out to such an extent and in such a way as to make it possible to determine the number of different forms under which the offspring of the hybrids appear, or to arrange these forms with certainty according to their separate generations, or definitely to ascertain their statistical relations.<br>    <br>    It requires indeed some courage to undertake a labor of such far-reaching extent; this appears, however, to be the only right way by which we can finally reach the solution of a question the importance of which cannot be overestimated in connection with the history of the evolution of organic forms.<br>    <br>    The paper now presented records the results of such a detailed experiment. This experiment was practically confined to a small plant group, and is now, after eight years’ pursuit, concluded in all essentials. Whether the plan upon which the separate experiments were conducted and carried out was the best suited to attain the desired end is left to the friendly decision of the reader.<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Kornberg.txt<br><br>File contents:<br><br>    The early history of DNA polymerase: a commentary by Arthur Kornberg<br>    <br>    Department of Biochemistry, Stanford University School of Medicine, Stanford, CA (U.S.A.)<br>    <br>    on &#39;Enzymic synthesis of deoxyribonucleic acid&#39; by A. Kornberg, I.R. Lehman, M.J. Bessman and E.S. Simms Biochim. Biophys. Acta 21 (1956) 197-198<br>    <br>    The brief publication [1] in Biochimica et Biophysica Acta (BBA) entitled &quot;Enzymic Synthesis of Deoxyribonucleic Acid&quot;, submitted May 2, 1956, was the first such report of DNA synthesis, aside from an abstract which appeared in Federation Proceedings in March of that year [2]. In the BBA paper, we (see photograph, p. 55) showed that the incorporation of [a4C]thymidine into an acid-soluble, DNase-sensitive, alkali-stable product depended on partially purified enzyme fractions, ATP and DNA which was converted to an &#39;active primer&#39;. We expressed the brash hope of assembling &#39;genetically specific DNA&#39;.<br>    <br>    Two years later, we submitted two papers to the Journal of Biological Chemistry describing the 2000-fold purification of DNA polymerase and the enzymatic synthesis of the four deoxyribonucleoside triphosphates which the enzyme assembled into long chains. The papers were titled: &#39;The Enzymatic Synthesis of DNA.  I. Preparation of Substrates and Partial Purification of an Enzyme from Escheriehia coli. II. General Properties of the Reaction.&#39; The papers were not accepted because of the term &#39;DNA&#39; in the title. The consensus among the many referees after a protracted review was that the papers would be acceptable if we used the term &quot;polydeoxyribonucleotide&quot; to identify the synthetic product; the use of &#39;DNA&#39; could be justified only with a demonstration of genetic activity. To us this seemed unreasonable, because few if any DNA papers in the Journal had been required to fulfill this criterion. Fortunately, with accession of John Edsall to the editorship, the papers were accepted with DNA in the title [3,4]. <br>    <br>    [Full text available here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Gurdon.txt<br><br>File contents:<br><br>    The Developmental Capacity of Nuclei taken from Intestinal Epithelium Cells of Feeding Tadpoles<br>    <br>    by J. B. GURDON<br>    <br>    From the Embryology Laboratory, Department of Zoology, Oxford<br>    <br>    WITH ONE PLATE<br>    <br>    An important problem in embryology is whether the differentiation of cells depends upon a stable restriction of the genetic information contained in their nuclei. The technique of nuclear transplantation has shown to what extent the nuclei of differentiating cells can promote the formation of different cell types (e.g. King &amp; Briggs, 1956; Gurdon, 1960c). Yet no experiments have so far been published on the transplantation of nuclei from fully differentiated normal cells. This is partly because it is difficult to obtain meaningful results from such experiments. The small amount of cytoplasm in differentiated cells renders their nuclei susceptible to damage through exposure to the saline medium, and this makes it difficult to assess the significance of the abnormalities resulting from their transplantation. It is, however, very desirable to know the developmental capacity of such nuclei, since any nuclear changes which are necessarily involved in cellular differentiation must have already taken place in cells of this kind.<br>    <br>    The experiments described below are some attempts to transplant nuclei from fully differentiated cells. Many of these nuclei gave abnormal results after transplantation, and several different kinds of experiments have been carried out to determine the cause and significance of these abnormalities.<br>    <br>    The donor cells used for these experiments were intestinal epithelium cells of feeding tadpoles. This is the final stage of differentiation of many of the endoderm cells whose nuclei have already been studied by means of nuclear transplantation experiments in Xenopus. The results to be described here may therefore be regarded as an extension of those previously obtained from differentiating endoderm cells (Gurdon, 1960c).<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def hasConsecutiveUpperCaseWords(filename):<br>    with open(filename, &#39;r&#39;) as file:<br>        text = file.read()<br>        words = text.split()<br>        <br>    for i in range(len(words) - 1):<br>        if re.fullmatch(r&#39;\b[A-Z]+\b&#39;, words[i]) and re.fullmatch(r&#39;\b[A-Z]+\b&#39;, words[i + 1]):<br>            return True<br>    <br>    return False</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R130" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">131</div></th><td class="s1">14 - Regular Expressions 1</td><td class="s1">04 - Find &quot;S&quot; words before a period</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Please write a function called &quot;findSWordsBeforePeriod&quot; that accepts two arguments. The first argument will be the name of a text file that contains part of a classic biology paper. The second argument will be the name of a text file that your function will need to create.<br><br>Your function should read the specified input file and use the &quot;re&quot; module to identify any words that are preceded by a space, end with the letter &quot;s&quot;, and are followed by a period. Your function should write each of these words, including the &quot;s&quot; at the end of each identified word, to the output file on a separate line. Before you write these words, trim off the trailing period.<br><br>Hint(s):<br><br>    Use the &quot;findall&quot; function.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    findSWordsBeforePeriod(&quot;Gurdon.txt&quot;, &quot;Gurdon_Output.txt&quot;)<br>    findSWordsBeforePeriod(&quot;Kornberg.txt&quot;, &quot;Kornberg_Output.txt&quot;)<br>    findSWordsBeforePeriod(&quot;Mendel.txt&quot;, &quot;Mendel_Output.txt&quot;)<br>    <br>    def checkOutputFile(outputFilePath):<br>        if os.path.exists(outputFilePath):<br>            with open(outputFilePath) as outputFile:<br>                print(&quot;{}:&quot;.format(outputFilePath))<br>                for line in outputFile:<br>                    print(&quot;  &quot; + line.rstrip(&quot;\n&quot;))<br>                print(&quot;&quot;)<br>        else:<br>            print(&quot;No output file exists at {}.&quot;.format(outputFilePath))<br>    <br>    checkOutputFile(&quot;Gurdon_Output.txt&quot;)<br>    checkOutputFile(&quot;Kornberg_Output.txt&quot;)<br>    checkOutputFile(&quot;Mendel_Output.txt&quot;)<br><br>Below is the expected output:<br><br>    Gurdon_Output.txt:<br>      cells<br>      experiments<br>      cells<br>      abnormalities<br>      tadpoles<br>      Xenopus<br>    <br>    Kornberg_Output.txt:<br>      Biophys<br>      chains<br>    <br>    Mendel_Output.txt:<br>      orders<br>      relations<br>      forms<br>      essentials<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Mendel.txt<br><br>File contents:<br><br>    EXPERIMENTS IN PLANT HYBRIDIZATION (1865)<br>    <br>    GREGOR MENDEL<br>    <br>    Read at the February 8th, and March 8th, 1865, meetings<br>    of the Brünn Natural History Society<br>    <br>    INTRODUCTORY REMARKS<br>    <br>    EXPERIENCE OF ARTIFICIAL FERTILIZATION, such as is effected with ornamental plants in order to obtain new variations in color, has led to the experiments which will here be discussed. The striking regularity with which the same hybrid forms always reappeared whenever fertilization took place between the same species induced further experiments to be undertaken, the object of which was to follow up the developments of the hybrids in their progeny.<br>    <br>    To this object numerous careful observers, such as Kölreuter, Gärtner, Herbert, Lecoq, Wichura and others, have devoted a part of their lives with inexhaustible perseverance. Gärtner especially in his work Die Bastarderzeugung im Pflanzenreiche [The Production of Hybrids in the Vegetable Kingdom], has recorded very valuable observations; and quite recently Wichura published the results of some profound investigations into the hybrids of the Willow. That, so far, no generally applicable law governing the formation and development of hybrids has been successfully formulated can hardly be wondered at by anyone who is acquainted with the extent of the task, and can appreciate the difficulties with which experiments of this class have to contend. A final decision can only be arrived at when we shall have before us the results of detailed experiments made on plants belonging to the most diverse orders.<br>    <br>    Those who survey the work done in this department will arrive at the conviction that among all the numerous experiments made, not one has been carried out to such an extent and in such a way as to make it possible to determine the number of different forms under which the offspring of the hybrids appear, or to arrange these forms with certainty according to their separate generations, or definitely to ascertain their statistical relations.<br>    <br>    It requires indeed some courage to undertake a labor of such far-reaching extent; this appears, however, to be the only right way by which we can finally reach the solution of a question the importance of which cannot be overestimated in connection with the history of the evolution of organic forms.<br>    <br>    The paper now presented records the results of such a detailed experiment. This experiment was practically confined to a small plant group, and is now, after eight years’ pursuit, concluded in all essentials. Whether the plan upon which the separate experiments were conducted and carried out was the best suited to attain the desired end is left to the friendly decision of the reader.<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Kornberg.txt<br><br>File contents:<br><br>    The early history of DNA polymerase: a commentary by Arthur Kornberg<br>    <br>    Department of Biochemistry, Stanford University School of Medicine, Stanford, CA (U.S.A.)<br>    <br>    on &#39;Enzymic synthesis of deoxyribonucleic acid&#39; by A. Kornberg, I.R. Lehman, M.J. Bessman and E.S. Simms Biochim. Biophys. Acta 21 (1956) 197-198<br>    <br>    The brief publication [1] in Biochimica et Biophysica Acta (BBA) entitled &quot;Enzymic Synthesis of Deoxyribonucleic Acid&quot;, submitted May 2, 1956, was the first such report of DNA synthesis, aside from an abstract which appeared in Federation Proceedings in March of that year [2]. In the BBA paper, we (see photograph, p. 55) showed that the incorporation of [a4C]thymidine into an acid-soluble, DNase-sensitive, alkali-stable product depended on partially purified enzyme fractions, ATP and DNA which was converted to an &#39;active primer&#39;. We expressed the brash hope of assembling &#39;genetically specific DNA&#39;.<br>    <br>    Two years later, we submitted two papers to the Journal of Biological Chemistry describing the 2000-fold purification of DNA polymerase and the enzymatic synthesis of the four deoxyribonucleoside triphosphates which the enzyme assembled into long chains. The papers were titled: &#39;The Enzymatic Synthesis of DNA.  I. Preparation of Substrates and Partial Purification of an Enzyme from Escheriehia coli. II. General Properties of the Reaction.&#39; The papers were not accepted because of the term &#39;DNA&#39; in the title. The consensus among the many referees after a protracted review was that the papers would be acceptable if we used the term &quot;polydeoxyribonucleotide&quot; to identify the synthetic product; the use of &#39;DNA&#39; could be justified only with a demonstration of genetic activity. To us this seemed unreasonable, because few if any DNA papers in the Journal had been required to fulfill this criterion. Fortunately, with accession of John Edsall to the editorship, the papers were accepted with DNA in the title [3,4]. <br>    <br>    [Full text available here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Gurdon.txt<br><br>File contents:<br><br>    The Developmental Capacity of Nuclei taken from Intestinal Epithelium Cells of Feeding Tadpoles<br>    <br>    by J. B. GURDON<br>    <br>    From the Embryology Laboratory, Department of Zoology, Oxford<br>    <br>    WITH ONE PLATE<br>    <br>    An important problem in embryology is whether the differentiation of cells depends upon a stable restriction of the genetic information contained in their nuclei. The technique of nuclear transplantation has shown to what extent the nuclei of differentiating cells can promote the formation of different cell types (e.g. King &amp; Briggs, 1956; Gurdon, 1960c). Yet no experiments have so far been published on the transplantation of nuclei from fully differentiated normal cells. This is partly because it is difficult to obtain meaningful results from such experiments. The small amount of cytoplasm in differentiated cells renders their nuclei susceptible to damage through exposure to the saline medium, and this makes it difficult to assess the significance of the abnormalities resulting from their transplantation. It is, however, very desirable to know the developmental capacity of such nuclei, since any nuclear changes which are necessarily involved in cellular differentiation must have already taken place in cells of this kind.<br>    <br>    The experiments described below are some attempts to transplant nuclei from fully differentiated cells. Many of these nuclei gave abnormal results after transplantation, and several different kinds of experiments have been carried out to determine the cause and significance of these abnormalities.<br>    <br>    The donor cells used for these experiments were intestinal epithelium cells of feeding tadpoles. This is the final stage of differentiation of many of the endoderm cells whose nuclei have already been studied by means of nuclear transplantation experiments in Xenopus. The results to be described here may therefore be regarded as an extension of those previously obtained from differentiating endoderm cells (Gurdon, 1960c).<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def findSWordsBeforePeriod(inputFilePath, outputFilePath):<br>    with open(inputFilePath, &quot;r&quot;) as inputFile, open(outputFilePath, &quot;w&quot;) as outputFile:<br>        text = inputFile.read()<br>        words = re.findall(r&quot;\s(\w+)s\.&quot;, text)<br>        for word in words:<br>            outputFile.write(word + &quot;s\n&quot;)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">4</td><td class="s1" dir="ltr">The first solution was correct but didn&#39;t write the &quot;s&quot; at the end of each word in the output file. I asked it to fix this, but it gave me a solution to re-read the file and modify it rather than modifying the original solution. I asked it to try again, and it did the same things as the first time. After that, I tweaked the prompt to clarify that an &quot;s&quot; should be present at the end of each identified word, and it passed.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R131" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">132</div></th><td class="s1">14 - Regular Expressions 1</td><td class="s1">05 - Find integers in parentheses</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;findIntegersInParentheses&quot; that accepts one argument. This argument will be the name of a text file that contains part of a classic biology paper. Your function should read the specified file and use the &quot;re&quot; module to identify any cases where an integer is surrounded by parentheses (for example, &quot;(1956)&quot;). Your function should return a list that includes these integer values without the parentheses. Convert each value in the list to a string before returning it. If no match is found, your function should return an empty list.<br><br>Hint(s):<br><br>    You will need to &quot;escape&quot; the parentheses.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findIntegersInParentheses(&quot;Gurdon.txt&quot;))<br><br>Below is the expected output:<br><br>    []<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findIntegersInParentheses(&quot;Kornberg.txt&quot;))<br><br>Below is the expected output:<br><br>    [&#39;1956&#39;]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findIntegersInParentheses(&quot;Mendel.txt&quot;))<br><br>Below is the expected output:<br><br>    [&#39;1865&#39;]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Mendel.txt<br><br>File contents:<br><br>    EXPERIMENTS IN PLANT HYBRIDIZATION (1865)<br>    <br>    GREGOR MENDEL<br>    <br>    Read at the February 8th, and March 8th, 1865, meetings<br>    of the Brünn Natural History Society<br>    <br>    INTRODUCTORY REMARKS<br>    <br>    EXPERIENCE OF ARTIFICIAL FERTILIZATION, such as is effected with ornamental plants in order to obtain new variations in color, has led to the experiments which will here be discussed. The striking regularity with which the same hybrid forms always reappeared whenever fertilization took place between the same species induced further experiments to be undertaken, the object of which was to follow up the developments of the hybrids in their progeny.<br>    <br>    To this object numerous careful observers, such as Kölreuter, Gärtner, Herbert, Lecoq, Wichura and others, have devoted a part of their lives with inexhaustible perseverance. Gärtner especially in his work Die Bastarderzeugung im Pflanzenreiche [The Production of Hybrids in the Vegetable Kingdom], has recorded very valuable observations; and quite recently Wichura published the results of some profound investigations into the hybrids of the Willow. That, so far, no generally applicable law governing the formation and development of hybrids has been successfully formulated can hardly be wondered at by anyone who is acquainted with the extent of the task, and can appreciate the difficulties with which experiments of this class have to contend. A final decision can only be arrived at when we shall have before us the results of detailed experiments made on plants belonging to the most diverse orders.<br>    <br>    Those who survey the work done in this department will arrive at the conviction that among all the numerous experiments made, not one has been carried out to such an extent and in such a way as to make it possible to determine the number of different forms under which the offspring of the hybrids appear, or to arrange these forms with certainty according to their separate generations, or definitely to ascertain their statistical relations.<br>    <br>    It requires indeed some courage to undertake a labor of such far-reaching extent; this appears, however, to be the only right way by which we can finally reach the solution of a question the importance of which cannot be overestimated in connection with the history of the evolution of organic forms.<br>    <br>    The paper now presented records the results of such a detailed experiment. This experiment was practically confined to a small plant group, and is now, after eight years’ pursuit, concluded in all essentials. Whether the plan upon which the separate experiments were conducted and carried out was the best suited to attain the desired end is left to the friendly decision of the reader.<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Kornberg.txt<br><br>File contents:<br><br>    The early history of DNA polymerase: a commentary by Arthur Kornberg<br>    <br>    Department of Biochemistry, Stanford University School of Medicine, Stanford, CA (U.S.A.)<br>    <br>    on &#39;Enzymic synthesis of deoxyribonucleic acid&#39; by A. Kornberg, I.R. Lehman, M.J. Bessman and E.S. Simms Biochim. Biophys. Acta 21 (1956) 197-198<br>    <br>    The brief publication [1] in Biochimica et Biophysica Acta (BBA) entitled &quot;Enzymic Synthesis of Deoxyribonucleic Acid&quot;, submitted May 2, 1956, was the first such report of DNA synthesis, aside from an abstract which appeared in Federation Proceedings in March of that year [2]. In the BBA paper, we (see photograph, p. 55) showed that the incorporation of [a4C]thymidine into an acid-soluble, DNase-sensitive, alkali-stable product depended on partially purified enzyme fractions, ATP and DNA which was converted to an &#39;active primer&#39;. We expressed the brash hope of assembling &#39;genetically specific DNA&#39;.<br>    <br>    Two years later, we submitted two papers to the Journal of Biological Chemistry describing the 2000-fold purification of DNA polymerase and the enzymatic synthesis of the four deoxyribonucleoside triphosphates which the enzyme assembled into long chains. The papers were titled: &#39;The Enzymatic Synthesis of DNA.  I. Preparation of Substrates and Partial Purification of an Enzyme from Escheriehia coli. II. General Properties of the Reaction.&#39; The papers were not accepted because of the term &#39;DNA&#39; in the title. The consensus among the many referees after a protracted review was that the papers would be acceptable if we used the term &quot;polydeoxyribonucleotide&quot; to identify the synthetic product; the use of &#39;DNA&#39; could be justified only with a demonstration of genetic activity. To us this seemed unreasonable, because few if any DNA papers in the Journal had been required to fulfill this criterion. Fortunately, with accession of John Edsall to the editorship, the papers were accepted with DNA in the title [3,4]. <br>    <br>    [Full text available here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Gurdon.txt<br><br>File contents:<br><br>    The Developmental Capacity of Nuclei taken from Intestinal Epithelium Cells of Feeding Tadpoles<br>    <br>    by J. B. GURDON<br>    <br>    From the Embryology Laboratory, Department of Zoology, Oxford<br>    <br>    WITH ONE PLATE<br>    <br>    An important problem in embryology is whether the differentiation of cells depends upon a stable restriction of the genetic information contained in their nuclei. The technique of nuclear transplantation has shown to what extent the nuclei of differentiating cells can promote the formation of different cell types (e.g. King &amp; Briggs, 1956; Gurdon, 1960c). Yet no experiments have so far been published on the transplantation of nuclei from fully differentiated normal cells. This is partly because it is difficult to obtain meaningful results from such experiments. The small amount of cytoplasm in differentiated cells renders their nuclei susceptible to damage through exposure to the saline medium, and this makes it difficult to assess the significance of the abnormalities resulting from their transplantation. It is, however, very desirable to know the developmental capacity of such nuclei, since any nuclear changes which are necessarily involved in cellular differentiation must have already taken place in cells of this kind.<br>    <br>    The experiments described below are some attempts to transplant nuclei from fully differentiated cells. Many of these nuclei gave abnormal results after transplantation, and several different kinds of experiments have been carried out to determine the cause and significance of these abnormalities.<br>    <br>    The donor cells used for these experiments were intestinal epithelium cells of feeding tadpoles. This is the final stage of differentiation of many of the endoderm cells whose nuclei have already been studied by means of nuclear transplantation experiments in Xenopus. The results to be described here may therefore be regarded as an extension of those previously obtained from differentiating endoderm cells (Gurdon, 1960c).<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def findIntegersInParentheses(filename):<br>    with open(filename, &#39;r&#39;) as file:<br>        contents = file.read()<br>        pattern = re.compile(r&#39;\((\d+)\)&#39;)<br>        matches = pattern.findall(contents)<br>        return [str(int) for int in matches]</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R132" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">133</div></th><td class="s4">14 - Regular Expressions 1</td><td class="s4">06 - Find words that start with a vowel</td><td class="freezebar-cell"></td><td class="s4">Here is a new Python programming task.<br><br>Please write a function called &quot;findVowelStartWords&quot; that accepts two arguments. The first argument will be the name of a text file that contains part of a classic biology paper. The second argument will be the name of a text file that your function will need to create.<br><br>Your function should read the specified input file and use the &quot;re&quot; module to identify any multi-letter words that start with an upper-case vowel. Your function should write each of these words to the output file on a separate line (not including any white space before or after the word). For simplicity, focus on words that are preceded and followed by a space character (don&#39;t worry about words at the beginning of a line or that are followed by a punctuation mark).<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    findVowelStartWords(&quot;Gurdon.txt&quot;, &quot;Gurdon_Output.txt&quot;)<br>    findVowelStartWords(&quot;Kornberg.txt&quot;, &quot;Kornberg_Output.txt&quot;)<br>    findVowelStartWords(&quot;Mendel.txt&quot;, &quot;Mendel_Output.txt&quot;)<br>    <br>    def checkOutputFile(outputFilePath):<br>        if os.path.exists(outputFilePath):<br>            with open(outputFilePath) as outputFile:<br>                print(&quot;{}:&quot;.format(outputFilePath))<br>                for line in outputFile:<br>                    print(&quot;  &quot; + line.rstrip(&quot;\n&quot;))<br>                print(&quot;&quot;)<br>        else:<br>            print(&quot;No output file exists at {}.&quot;.format(outputFilePath))<br>    <br>    checkOutputFile(&quot;Gurdon_Output.txt&quot;)<br>    checkOutputFile(&quot;Kornberg_Output.txt&quot;)<br>    checkOutputFile(&quot;Mendel_Output.txt&quot;)<br><br>Below is the expected output:<br><br>    Gurdon_Output.txt:<br>      Intestinal<br>      Embryology<br>      ONE<br>      It<br>    <br>    Kornberg_Output.txt:<br>      Arthur<br>      University<br>      Acta<br>      Acta<br>      In<br>      ATP<br>      Enzymatic<br>      Enzyme<br>      Escheriehia<br>      Edsall<br>    <br>    Mendel_Output.txt:<br>      IN<br>      OF<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Mendel.txt<br><br>File contents:<br><br>    EXPERIMENTS IN PLANT HYBRIDIZATION (1865)<br>    <br>    GREGOR MENDEL<br>    <br>    Read at the February 8th, and March 8th, 1865, meetings<br>    of the Brünn Natural History Society<br>    <br>    INTRODUCTORY REMARKS<br>    <br>    EXPERIENCE OF ARTIFICIAL FERTILIZATION, such as is effected with ornamental plants in order to obtain new variations in color, has led to the experiments which will here be discussed. The striking regularity with which the same hybrid forms always reappeared whenever fertilization took place between the same species induced further experiments to be undertaken, the object of which was to follow up the developments of the hybrids in their progeny.<br>    <br>    To this object numerous careful observers, such as Kölreuter, Gärtner, Herbert, Lecoq, Wichura and others, have devoted a part of their lives with inexhaustible perseverance. Gärtner especially in his work Die Bastarderzeugung im Pflanzenreiche [The Production of Hybrids in the Vegetable Kingdom], has recorded very valuable observations; and quite recently Wichura published the results of some profound investigations into the hybrids of the Willow. That, so far, no generally applicable law governing the formation and development of hybrids has been successfully formulated can hardly be wondered at by anyone who is acquainted with the extent of the task, and can appreciate the difficulties with which experiments of this class have to contend. A final decision can only be arrived at when we shall have before us the results of detailed experiments made on plants belonging to the most diverse orders.<br>    <br>    Those who survey the work done in this department will arrive at the conviction that among all the numerous experiments made, not one has been carried out to such an extent and in such a way as to make it possible to determine the number of different forms under which the offspring of the hybrids appear, or to arrange these forms with certainty according to their separate generations, or definitely to ascertain their statistical relations.<br>    <br>    It requires indeed some courage to undertake a labor of such far-reaching extent; this appears, however, to be the only right way by which we can finally reach the solution of a question the importance of which cannot be overestimated in connection with the history of the evolution of organic forms.<br>    <br>    The paper now presented records the results of such a detailed experiment. This experiment was practically confined to a small plant group, and is now, after eight years’ pursuit, concluded in all essentials. Whether the plan upon which the separate experiments were conducted and carried out was the best suited to attain the desired end is left to the friendly decision of the reader.<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Kornberg.txt<br><br>File contents:<br><br>    The early history of DNA polymerase: a commentary by Arthur Kornberg<br>    <br>    Department of Biochemistry, Stanford University School of Medicine, Stanford, CA (U.S.A.)<br>    <br>    on &#39;Enzymic synthesis of deoxyribonucleic acid&#39; by A. Kornberg, I.R. Lehman, M.J. Bessman and E.S. Simms Biochim. Biophys. Acta 21 (1956) 197-198<br>    <br>    The brief publication [1] in Biochimica et Biophysica Acta (BBA) entitled &quot;Enzymic Synthesis of Deoxyribonucleic Acid&quot;, submitted May 2, 1956, was the first such report of DNA synthesis, aside from an abstract which appeared in Federation Proceedings in March of that year [2]. In the BBA paper, we (see photograph, p. 55) showed that the incorporation of [a4C]thymidine into an acid-soluble, DNase-sensitive, alkali-stable product depended on partially purified enzyme fractions, ATP and DNA which was converted to an &#39;active primer&#39;. We expressed the brash hope of assembling &#39;genetically specific DNA&#39;.<br>    <br>    Two years later, we submitted two papers to the Journal of Biological Chemistry describing the 2000-fold purification of DNA polymerase and the enzymatic synthesis of the four deoxyribonucleoside triphosphates which the enzyme assembled into long chains. The papers were titled: &#39;The Enzymatic Synthesis of DNA.  I. Preparation of Substrates and Partial Purification of an Enzyme from Escheriehia coli. II. General Properties of the Reaction.&#39; The papers were not accepted because of the term &#39;DNA&#39; in the title. The consensus among the many referees after a protracted review was that the papers would be acceptable if we used the term &quot;polydeoxyribonucleotide&quot; to identify the synthetic product; the use of &#39;DNA&#39; could be justified only with a demonstration of genetic activity. To us this seemed unreasonable, because few if any DNA papers in the Journal had been required to fulfill this criterion. Fortunately, with accession of John Edsall to the editorship, the papers were accepted with DNA in the title [3,4]. <br>    <br>    [Full text available here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Gurdon.txt<br><br>File contents:<br><br>    The Developmental Capacity of Nuclei taken from Intestinal Epithelium Cells of Feeding Tadpoles<br>    <br>    by J. B. GURDON<br>    <br>    From the Embryology Laboratory, Department of Zoology, Oxford<br>    <br>    WITH ONE PLATE<br>    <br>    An important problem in embryology is whether the differentiation of cells depends upon a stable restriction of the genetic information contained in their nuclei. The technique of nuclear transplantation has shown to what extent the nuclei of differentiating cells can promote the formation of different cell types (e.g. King &amp; Briggs, 1956; Gurdon, 1960c). Yet no experiments have so far been published on the transplantation of nuclei from fully differentiated normal cells. This is partly because it is difficult to obtain meaningful results from such experiments. The small amount of cytoplasm in differentiated cells renders their nuclei susceptible to damage through exposure to the saline medium, and this makes it difficult to assess the significance of the abnormalities resulting from their transplantation. It is, however, very desirable to know the developmental capacity of such nuclei, since any nuclear changes which are necessarily involved in cellular differentiation must have already taken place in cells of this kind.<br>    <br>    The experiments described below are some attempts to transplant nuclei from fully differentiated cells. Many of these nuclei gave abnormal results after transplantation, and several different kinds of experiments have been carried out to determine the cause and significance of these abnormalities.<br>    <br>    The donor cells used for these experiments were intestinal epithelium cells of feeding tadpoles. This is the final stage of differentiation of many of the endoderm cells whose nuclei have already been studied by means of nuclear transplantation experiments in Xenopus. The results to be described here may therefore be regarded as an extension of those previously obtained from differentiating endoderm cells (Gurdon, 1960c).<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]</td><td class="s4" dir="ltr"></td><td class="s4" dir="ltr">Yes</td><td class="s4" dir="ltr">No</td><td class="s4">text</td><td class="s4" dir="ltr">SRP</td><td class="s5" dir="ltr">February 13, 2023</td><td class="s4" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s4" dir="ltr">import re<br><br>def findVowelStartWords(input_file, output_file):<br>    vowels = &#39;AEIOU&#39;<br>    pattern = re.compile(r&#39;\b[%s][a-zA-Z]+[^\s\W]\b&#39; % vowels)<br>    <br>    with open(input_file, &#39;r&#39;) as f_in, open(output_file, &#39;w&#39;) as f_out:<br>        text = f_in.read()<br>        words = re.findall(pattern, text)<br>        f_out.write(&#39;\n&#39;.join(words))</td><td class="s5">1</td><td class="s5" dir="ltr">0</td><td class="s5" dir="ltr">10</td><td class="s4" dir="ltr">I tried many ways to get it to pass the test, but it could not figure out that it needed to ignore words that were preceded by a space or followed by a space or punctuation mark. I tried shortening the prompt, but that still didn&#39;t help.</td><td class="s4"></td><td class="s4" dir="ltr">Yes</td><td class="s4"></td><td class="s4" dir="ltr"></td><td class="s4" dir="ltr">Yes</td><td class="s4" dir="ltr"></td><td class="s4" dir="ltr">Yes</td><td class="s4"></td><td class="s4"></td><td class="s4" dir="ltr">Yes</td><td class="s4"></td></tr><tr style="height: 20px"><th id="2089702959R133" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">134</div></th><td class="s1">14 - Regular Expressions 1</td><td class="s1">07 - Find compound words</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;findCompoundWords&quot; that accepts two arguments. The first argument will be the name of a text file that contains part of a classic biology paper. The second argument will be the name of a text file that your function will need to create.<br><br>Your function should read the specified input file and use the &quot;re&quot; module to identify any compound words. These are words that have been joined by a hyphen (dash), such as &quot;student-athlete&quot; or &quot;acid-soluble&quot;. Both words should have only letters, no numbers. The compound words do not necessarily need to be surrounded by spaces. Your function should write each of these compound words to the output file on a separate line.<br><br>Hint(s):<br><br>    Your regular expression will need to use an escape character.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    findCompoundWords(&quot;Gurdon.txt&quot;, &quot;Gurdon_Output.txt&quot;)<br>    findCompoundWords(&quot;Kornberg.txt&quot;, &quot;Kornberg_Output.txt&quot;)<br>    findCompoundWords(&quot;Mendel.txt&quot;, &quot;Mendel_Output.txt&quot;)<br>    <br>    def checkOutputFile(outputFilePath):<br>        if os.path.exists(outputFilePath):<br>            with open(outputFilePath) as outputFile:<br>                print(&quot;{}:&quot;.format(outputFilePath))<br>                for line in outputFile:<br>                    print(&quot;  &quot; + line.rstrip(&quot;\n&quot;))<br>                print(&quot;&quot;)<br>        else:<br>            print(&quot;No output file exists at {}.&quot;.format(outputFilePath))<br>    <br>    checkOutputFile(&quot;Gurdon_Output.txt&quot;)<br>    checkOutputFile(&quot;Kornberg_Output.txt&quot;)<br>    checkOutputFile(&quot;Mendel_Output.txt&quot;)<br><br>Below is the expected output:<br><br>    Gurdon_Output.txt:<br>    <br>    Kornberg_Output.txt:<br>      acid-soluble<br>      DNase-sensitive<br>      alkali-stable<br>    <br>    Mendel_Output.txt:<br>      far-reaching<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Mendel.txt<br><br>File contents:<br><br>    EXPERIMENTS IN PLANT HYBRIDIZATION (1865)<br>    <br>    GREGOR MENDEL<br>    <br>    Read at the February 8th, and March 8th, 1865, meetings<br>    of the Brünn Natural History Society<br>    <br>    INTRODUCTORY REMARKS<br>    <br>    EXPERIENCE OF ARTIFICIAL FERTILIZATION, such as is effected with ornamental plants in order to obtain new variations in color, has led to the experiments which will here be discussed. The striking regularity with which the same hybrid forms always reappeared whenever fertilization took place between the same species induced further experiments to be undertaken, the object of which was to follow up the developments of the hybrids in their progeny.<br>    <br>    To this object numerous careful observers, such as Kölreuter, Gärtner, Herbert, Lecoq, Wichura and others, have devoted a part of their lives with inexhaustible perseverance. Gärtner especially in his work Die Bastarderzeugung im Pflanzenreiche [The Production of Hybrids in the Vegetable Kingdom], has recorded very valuable observations; and quite recently Wichura published the results of some profound investigations into the hybrids of the Willow. That, so far, no generally applicable law governing the formation and development of hybrids has been successfully formulated can hardly be wondered at by anyone who is acquainted with the extent of the task, and can appreciate the difficulties with which experiments of this class have to contend. A final decision can only be arrived at when we shall have before us the results of detailed experiments made on plants belonging to the most diverse orders.<br>    <br>    Those who survey the work done in this department will arrive at the conviction that among all the numerous experiments made, not one has been carried out to such an extent and in such a way as to make it possible to determine the number of different forms under which the offspring of the hybrids appear, or to arrange these forms with certainty according to their separate generations, or definitely to ascertain their statistical relations.<br>    <br>    It requires indeed some courage to undertake a labor of such far-reaching extent; this appears, however, to be the only right way by which we can finally reach the solution of a question the importance of which cannot be overestimated in connection with the history of the evolution of organic forms.<br>    <br>    The paper now presented records the results of such a detailed experiment. This experiment was practically confined to a small plant group, and is now, after eight years’ pursuit, concluded in all essentials. Whether the plan upon which the separate experiments were conducted and carried out was the best suited to attain the desired end is left to the friendly decision of the reader.<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Kornberg.txt<br><br>File contents:<br><br>    The early history of DNA polymerase: a commentary by Arthur Kornberg<br>    <br>    Department of Biochemistry, Stanford University School of Medicine, Stanford, CA (U.S.A.)<br>    <br>    on &#39;Enzymic synthesis of deoxyribonucleic acid&#39; by A. Kornberg, I.R. Lehman, M.J. Bessman and E.S. Simms Biochim. Biophys. Acta 21 (1956) 197-198<br>    <br>    The brief publication [1] in Biochimica et Biophysica Acta (BBA) entitled &quot;Enzymic Synthesis of Deoxyribonucleic Acid&quot;, submitted May 2, 1956, was the first such report of DNA synthesis, aside from an abstract which appeared in Federation Proceedings in March of that year [2]. In the BBA paper, we (see photograph, p. 55) showed that the incorporation of [a4C]thymidine into an acid-soluble, DNase-sensitive, alkali-stable product depended on partially purified enzyme fractions, ATP and DNA which was converted to an &#39;active primer&#39;. We expressed the brash hope of assembling &#39;genetically specific DNA&#39;.<br>    <br>    Two years later, we submitted two papers to the Journal of Biological Chemistry describing the 2000-fold purification of DNA polymerase and the enzymatic synthesis of the four deoxyribonucleoside triphosphates which the enzyme assembled into long chains. The papers were titled: &#39;The Enzymatic Synthesis of DNA.  I. Preparation of Substrates and Partial Purification of an Enzyme from Escheriehia coli. II. General Properties of the Reaction.&#39; The papers were not accepted because of the term &#39;DNA&#39; in the title. The consensus among the many referees after a protracted review was that the papers would be acceptable if we used the term &quot;polydeoxyribonucleotide&quot; to identify the synthetic product; the use of &#39;DNA&#39; could be justified only with a demonstration of genetic activity. To us this seemed unreasonable, because few if any DNA papers in the Journal had been required to fulfill this criterion. Fortunately, with accession of John Edsall to the editorship, the papers were accepted with DNA in the title [3,4]. <br>    <br>    [Full text available here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Gurdon.txt<br><br>File contents:<br><br>    The Developmental Capacity of Nuclei taken from Intestinal Epithelium Cells of Feeding Tadpoles<br>    <br>    by J. B. GURDON<br>    <br>    From the Embryology Laboratory, Department of Zoology, Oxford<br>    <br>    WITH ONE PLATE<br>    <br>    An important problem in embryology is whether the differentiation of cells depends upon a stable restriction of the genetic information contained in their nuclei. The technique of nuclear transplantation has shown to what extent the nuclei of differentiating cells can promote the formation of different cell types (e.g. King &amp; Briggs, 1956; Gurdon, 1960c). Yet no experiments have so far been published on the transplantation of nuclei from fully differentiated normal cells. This is partly because it is difficult to obtain meaningful results from such experiments. The small amount of cytoplasm in differentiated cells renders their nuclei susceptible to damage through exposure to the saline medium, and this makes it difficult to assess the significance of the abnormalities resulting from their transplantation. It is, however, very desirable to know the developmental capacity of such nuclei, since any nuclear changes which are necessarily involved in cellular differentiation must have already taken place in cells of this kind.<br>    <br>    The experiments described below are some attempts to transplant nuclei from fully differentiated cells. Many of these nuclei gave abnormal results after transplantation, and several different kinds of experiments have been carried out to determine the cause and significance of these abnormalities.<br>    <br>    The donor cells used for these experiments were intestinal epithelium cells of feeding tadpoles. This is the final stage of differentiation of many of the endoderm cells whose nuclei have already been studied by means of nuclear transplantation experiments in Xenopus. The results to be described here may therefore be regarded as an extension of those previously obtained from differentiating endoderm cells (Gurdon, 1960c).<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def findCompoundWords(input_file, output_file):<br>    with open(input_file, &#39;r&#39;) as f_in:<br>        text = f_in.read()<br><br>    pattern = r&#39;\b[a-zA-Z]+-[a-zA-Z]+\b&#39;<br>    compound_words = re.findall(pattern, text)<br><br>    with open(output_file, &#39;w&#39;) as f_out:<br>        for word in compound_words:<br>            f_out.write(word + &#39;\n&#39;)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">5</td><td class="s1" dir="ltr">The first few solutions matched compound words that had numbers in them. But the prompt asked them to ignore those words. I repeated the prompt a couple of times, but it continued to miss that part. Later I simplified the prompt to not include the tests, and it passed.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R134" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">135</div></th><td class="s1">14 - Regular Expressions 1</td><td class="s1">08 - Find three consecutive vowels</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;findThreeConsecutiveVowels&quot; that accepts two arguments. The first argument will be the name of a text file that contains part of a classic biology paper. The second argument will be the name of a text file that your function will need to create.<br><br>Your function should read the specified input file and use the &quot;re&quot; module to identify words that have three consecutive vowels in them. Your function should write each of these words to the output file on a separate line. Make sure to strip out any white space surrounding the words, before writing them to the file.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    findThreeConsecutiveVowels(&quot;Gurdon.txt&quot;, &quot;Gurdon_Output.txt&quot;)<br>    findThreeConsecutiveVowels(&quot;Kornberg.txt&quot;, &quot;Kornberg_Output.txt&quot;)<br>    findThreeConsecutiveVowels(&quot;Mendel.txt&quot;, &quot;Mendel_Output.txt&quot;)<br>    <br>    def checkOutputFile(outputFilePath):<br>        if os.path.exists(outputFilePath):<br>            with open(outputFilePath) as outputFile:<br>                print(&quot;{}:&quot;.format(outputFilePath))<br>                for line in outputFile:<br>                    print(&quot;  &quot; + line.rstrip(&quot;\n&quot;))<br>                print(&quot;&quot;)<br>        else:<br>            print(&quot;No output file exists at {}.&quot;.format(outputFilePath))<br>    <br>    checkOutputFile(&quot;Gurdon_Output.txt&quot;)<br>    checkOutputFile(&quot;Kornberg_Output.txt&quot;)<br>    checkOutputFile(&quot;Mendel_Output.txt&quot;)<br><br>Below is the expected output:<br><br>    Gurdon_Output.txt:<br>      previously<br>    <br>    Kornberg_Output.txt:<br>    <br>    Mendel_Output.txt:<br>      acquainted<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Mendel.txt<br><br>File contents:<br><br>    EXPERIMENTS IN PLANT HYBRIDIZATION (1865)<br>    <br>    GREGOR MENDEL<br>    <br>    Read at the February 8th, and March 8th, 1865, meetings<br>    of the Brünn Natural History Society<br>    <br>    INTRODUCTORY REMARKS<br>    <br>    EXPERIENCE OF ARTIFICIAL FERTILIZATION, such as is effected with ornamental plants in order to obtain new variations in color, has led to the experiments which will here be discussed. The striking regularity with which the same hybrid forms always reappeared whenever fertilization took place between the same species induced further experiments to be undertaken, the object of which was to follow up the developments of the hybrids in their progeny.<br>    <br>    To this object numerous careful observers, such as Kölreuter, Gärtner, Herbert, Lecoq, Wichura and others, have devoted a part of their lives with inexhaustible perseverance. Gärtner especially in his work Die Bastarderzeugung im Pflanzenreiche [The Production of Hybrids in the Vegetable Kingdom], has recorded very valuable observations; and quite recently Wichura published the results of some profound investigations into the hybrids of the Willow. That, so far, no generally applicable law governing the formation and development of hybrids has been successfully formulated can hardly be wondered at by anyone who is acquainted with the extent of the task, and can appreciate the difficulties with which experiments of this class have to contend. A final decision can only be arrived at when we shall have before us the results of detailed experiments made on plants belonging to the most diverse orders.<br>    <br>    Those who survey the work done in this department will arrive at the conviction that among all the numerous experiments made, not one has been carried out to such an extent and in such a way as to make it possible to determine the number of different forms under which the offspring of the hybrids appear, or to arrange these forms with certainty according to their separate generations, or definitely to ascertain their statistical relations.<br>    <br>    It requires indeed some courage to undertake a labor of such far-reaching extent; this appears, however, to be the only right way by which we can finally reach the solution of a question the importance of which cannot be overestimated in connection with the history of the evolution of organic forms.<br>    <br>    The paper now presented records the results of such a detailed experiment. This experiment was practically confined to a small plant group, and is now, after eight years’ pursuit, concluded in all essentials. Whether the plan upon which the separate experiments were conducted and carried out was the best suited to attain the desired end is left to the friendly decision of the reader.<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Kornberg.txt<br><br>File contents:<br><br>    The early history of DNA polymerase: a commentary by Arthur Kornberg<br>    <br>    Department of Biochemistry, Stanford University School of Medicine, Stanford, CA (U.S.A.)<br>    <br>    on &#39;Enzymic synthesis of deoxyribonucleic acid&#39; by A. Kornberg, I.R. Lehman, M.J. Bessman and E.S. Simms Biochim. Biophys. Acta 21 (1956) 197-198<br>    <br>    The brief publication [1] in Biochimica et Biophysica Acta (BBA) entitled &quot;Enzymic Synthesis of Deoxyribonucleic Acid&quot;, submitted May 2, 1956, was the first such report of DNA synthesis, aside from an abstract which appeared in Federation Proceedings in March of that year [2]. In the BBA paper, we (see photograph, p. 55) showed that the incorporation of [a4C]thymidine into an acid-soluble, DNase-sensitive, alkali-stable product depended on partially purified enzyme fractions, ATP and DNA which was converted to an &#39;active primer&#39;. We expressed the brash hope of assembling &#39;genetically specific DNA&#39;.<br>    <br>    Two years later, we submitted two papers to the Journal of Biological Chemistry describing the 2000-fold purification of DNA polymerase and the enzymatic synthesis of the four deoxyribonucleoside triphosphates which the enzyme assembled into long chains. The papers were titled: &#39;The Enzymatic Synthesis of DNA.  I. Preparation of Substrates and Partial Purification of an Enzyme from Escheriehia coli. II. General Properties of the Reaction.&#39; The papers were not accepted because of the term &#39;DNA&#39; in the title. The consensus among the many referees after a protracted review was that the papers would be acceptable if we used the term &quot;polydeoxyribonucleotide&quot; to identify the synthetic product; the use of &#39;DNA&#39; could be justified only with a demonstration of genetic activity. To us this seemed unreasonable, because few if any DNA papers in the Journal had been required to fulfill this criterion. Fortunately, with accession of John Edsall to the editorship, the papers were accepted with DNA in the title [3,4]. <br>    <br>    [Full text available here: http://tetrad.ucsf.edu/classicpapers.]<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Gurdon.txt<br><br>File contents:<br><br>    The Developmental Capacity of Nuclei taken from Intestinal Epithelium Cells of Feeding Tadpoles<br>    <br>    by J. B. GURDON<br>    <br>    From the Embryology Laboratory, Department of Zoology, Oxford<br>    <br>    WITH ONE PLATE<br>    <br>    An important problem in embryology is whether the differentiation of cells depends upon a stable restriction of the genetic information contained in their nuclei. The technique of nuclear transplantation has shown to what extent the nuclei of differentiating cells can promote the formation of different cell types (e.g. King &amp; Briggs, 1956; Gurdon, 1960c). Yet no experiments have so far been published on the transplantation of nuclei from fully differentiated normal cells. This is partly because it is difficult to obtain meaningful results from such experiments. The small amount of cytoplasm in differentiated cells renders their nuclei susceptible to damage through exposure to the saline medium, and this makes it difficult to assess the significance of the abnormalities resulting from their transplantation. It is, however, very desirable to know the developmental capacity of such nuclei, since any nuclear changes which are necessarily involved in cellular differentiation must have already taken place in cells of this kind.<br>    <br>    The experiments described below are some attempts to transplant nuclei from fully differentiated cells. Many of these nuclei gave abnormal results after transplantation, and several different kinds of experiments have been carried out to determine the cause and significance of these abnormalities.<br>    <br>    The donor cells used for these experiments were intestinal epithelium cells of feeding tadpoles. This is the final stage of differentiation of many of the endoderm cells whose nuclei have already been studied by means of nuclear transplantation experiments in Xenopus. The results to be described here may therefore be regarded as an extension of those previously obtained from differentiating endoderm cells (Gurdon, 1960c).<br>    <br>    [See full text here: http://tetrad.ucsf.edu/classicpapers.]</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def findThreeConsecutiveVowels(inputFile, outputFile):<br>    with open(inputFile, &#39;r&#39;) as f:<br>        text = f.read()<br>    <br>    words = re.findall(r&#39;\b\w+\b&#39;, text)<br>    <br>    vowels = &#39;aeiouAEIOU&#39;<br>    three_consecutive_vowel_words = []<br>    <br>    for word in words:<br>        for i in range(len(word) - 2):<br>            if word[i] in vowels and word[i + 1] in vowels and word[i + 2] in vowels:<br>                three_consecutive_vowel_words.append(word)<br>                break<br>    <br>    with open(outputFile, &#39;w&#39;) as f:<br>        for word in three_consecutive_vowel_words:<br>            f.write(word + &quot;\n&quot;)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">3</td><td class="s1" dir="ltr">The first solution identified no matches. I tried to explain this, but it didn&#39;t understand and gave a second solution that did not address the prompt. I gave it the original prompt, which passed the tests. However, it did not follow the true spirit of the prompt because it used an if statement to identify words with three consecutive vowels, rather than using a regular expression for this.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td></tr><tr style="height: 20px"><th id="2089702959R135" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">136</div></th><td class="s1">15 - Regular Expressions 2</td><td class="s1">01 - Cut DNA</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Restriction enzymes are used in laboratories to remove a piece of DNA from a chromosome so that it can be used for molecular cloning. These enzymes recognize specific restriction sites in the DNA, which can be used as starting and stopping points for cutting the DNA.<br><br>Please write a function called &quot;cutDNA&quot; that accepts three arguments: 1) a string that represents a DNA sequence on a chromosome, 2) a string that represents the first restriction site, and 3) a string that represents the second restriction site. Your function should return the DNA sequence in between the two restriction sites that will be cut from the first DNA sequence.<br><br>Your function must use the &quot;re&quot; module to accomplish this task.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    chromsomalDNA = &quot;ACTTAATTTACCAGGAGCTAGGGACCATTACACGTGGATCGAAAGGCGCATTA&quot;<br>    restrictionSite1 = r&quot;AGGGACC&quot;<br>    restrictionSite2 = r&quot;GGCGCAT&quot;<br>    <br>    print(cutDNA(chromsomalDNA, restrictionSite1, restrictionSite2))<br><br>Below is the expected output:<br><br>    ATTACACGTGGATCGAAA<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    <br>    chromsomalDNA = &quot;TATACCTAAAGGCGTACAGGAATTTTTAAAACCCGGAGACTCTGAGATAG&quot;<br>    restrictionSite1 = r&quot;AATTTTT&quot;<br>    restrictionSite2 = r&quot;GAGATAG&quot;<br>    <br>    print(cutDNA(chromsomalDNA, restrictionSite1, restrictionSite2))<br><br>Below is the expected output:<br><br>    AAAACCCGGAGACTCT<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    chromsomalDNA = &quot;AATTTTTCCAAGCAGCTAGACGACGAGAGATAG&quot;<br>    restrictionSite1 = r&quot;AATTTTT&quot;<br>    restrictionSite2 = r&quot;GAGATAG&quot;<br>    <br>    print(cutDNA(chromsomalDNA, restrictionSite1, restrictionSite2))<br><br>Below is the expected output:<br><br>    CCAAGCAGCTAGACGACGA</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def cutDNA(chromosomalDNA, restrictionSite1, restrictionSite2):<br>    pattern = restrictionSite1 + &quot;(.*?)&quot; + restrictionSite2<br>    result = re.search(pattern, chromosomalDNA)<br>    return result.group(1)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R136" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">137</div></th><td class="s1">15 - Regular Expressions 2</td><td class="s1">02 - Question everything</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;questionEverything&quot; that accepts a single argument: a string. Your function should return a new string where all non-alphabetical characters (other than spaces) have been replaced with a &quot;?&quot; character. Your solution should use the &quot;re&quot; module to do this.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(questionEverything(&quot;Hello, World!&quot;))<br><br>Below is the expected output:<br><br>    Hello? World?<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(questionEverything(&quot;Single-step method of RNA isolation by acid guanidinium thiocyanate-phenol-chloroform extraction.&quot;))<br><br>Below is the expected output:<br><br>    Single?step method of RNA isolation by acid guanidinium thiocyanate?phenol?chloroform extraction?<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(questionEverything(&quot;A simple method, for the isolation and purification of total lipides, from animal tissues.&quot;))<br><br>Below is the expected output:<br><br>    A simple method? for the isolation and purification of total lipides? from animal tissues?<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(questionEverything(&quot;Clustal W: improving the sensitivity of progressive multiple sequence alignment through sequence weighting, position-specific gap penalties and weight matrix choice.&quot;))<br><br>Below is the expected output:<br><br>    Clustal W? improving the sensitivity of progressive multiple sequence alignment through sequence weighting? position?specific gap penalties and weight matrix choice?</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def questionEverything(string):<br>    return re.sub(r&#39;[^a-zA-Z\s]&#39;, &#39;?&#39;, string)</td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R137" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">138</div></th><td class="s1">15 - Regular Expressions 2</td><td class="s1">03 - Find Simple Sequence Repeats</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Simple Sequence Repeats (SSRs) are regions of DNA that repeat multiple times. You are interested in a particular SSR that consists of a series of &quot;AT&quot; nucleotides that repeat 3 or 4 times in the middle of a DNA sequence. The AT repeat region is always preceded and followed by either a G or a C nucleotide. You have found that this SSR is important in determining the phenotype of a specific fish population. However, with more than 4 repeats, a different phenotype occurs, and you are only interested in finding species where &quot;AT&quot; is repeated 3 or 4 times in their genome.<br><br>Please create a function called &quot;findSSR&quot; that accepts a single argument. This argument will be a string that contains a DNA sequence. If that DNA sequence has an &quot;AT&quot; that repeats 3 or 4 times and is preceded and followed by either a G or C nucleotide, your function should return True. Otherwise, your function should return False.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findSSR(&quot;ATACGCGATGC&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findSSR(&quot;ATACGATATATCG&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findSSR(&quot;ATATCGATATATATATCGC&quot;))<br><br>Below is the expected output:<br><br>    False<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findSSR(&quot;CGATATCGATATATATGCGATGC&quot;))<br><br>Below is the expected output:<br><br>    True<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(findSSR(&quot;ATATATCGATACTGACCGGATATCATATAT&quot;))<br><br>Below is the expected output:<br><br>    False</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def findSSR(dna):<br>    pattern = r&#39;(G|C)(AT){3,4}(G|C)&#39;<br>    match = re.search(pattern, dna)<br>    if match:<br>        return True<br>    return False</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R138" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">139</div></th><td class="s1">15 - Regular Expressions 2</td><td class="s1">04 - Delete gene name</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;deleteGeneName&quot; that accepts two arguments. The first argument will be a string of gene names separated by commas (for example, &quot;BRCA1,BRCA2,TRMT61A,TRMT61B&quot;). The second argument will be a string that contains one of those gene names. Your function must use &quot;re.sub&quot; to remove the specified gene name (second argument) from the overall list of gene names (first argument) and remove any extra commas that remain. Your function should return the new string of gene names without the gene name that was just deleted.<br><br>Hint(s):<br><br>    You may want to invoke &quot;re.sub&quot; multiple times, and it is possible to do this with very simple regular expression(s).<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(deleteGeneName(&quot;BRCA1,BRCA2,TRMT61A,TRMT61B&quot;,&quot;BRCA1&quot;))<br><br>Below is the expected output:<br><br>    BRCA2,TRMT61A,TRMT61B<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(deleteGeneName(&quot;TMZ1,CO,GCM3,HEP&quot;,&quot;HEP&quot;))<br><br>Below is the expected output:<br><br>    TMZ1,CO,GCM3<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(deleteGeneName(&quot;OMM,PHE,H1B&quot;,&quot;PHE&quot;))<br><br>Below is the expected output:<br><br>    OMM,H1B</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def deleteGeneName(gene_names, gene_to_delete):<br>    gene_names = gene_names.split(&#39;,&#39;)<br>    gene_names = [gene for gene in gene_names if gene != gene_to_delete]<br>    return &#39;,&#39;.join(gene_names)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">3</td><td class="s1" dir="ltr">The first two attempts did not include any commas in the output for two of the tests. I asked it to try a different approach, and it succeeded.<br><br>The solution does not use a regular expression, so it does not really match the spirit of the prompt.</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R139" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">140</div></th><td class="s1">15 - Regular Expressions 2</td><td class="s1">05 - Add comma</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose you went camping for a year and counted a lot of animals, keeping track of each sighting in your field book. After you finished putting the counts in a text file, you realized that, for species with counts greater than 1000, you failed to put a comma before the last three digits. Now you want to add the comma for every species that you observed at least 1000 times. (Assume that you never counted more than 999,999 individuals of a given species.)<br><br>Please create a function called &quot;addComma&quot; that accepts a single argument: a string representation of one of these numbers followed by the species name (for example, &quot;1298 Bison&quot;). If the number of observations is greater than or equal to 1000, your function should return a string that includes a comma before the last three digits in the animal count (for example, &quot;1,298 bison&quot;). If the number of observations is less than 1000, your function should return the original string. Recommendation: Use the &quot;re.sub&quot; function for this.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addComma(&quot;1298 Bison&quot;))<br><br>Below is the expected output:<br><br>    1,298 Bison<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addComma(&quot;98122 Canada Geese&quot;))<br><br>Below is the expected output:<br><br>    98,122 Canada Geese<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addComma(&quot;312 Tarantula&quot;))<br><br>Below is the expected output:<br><br>    312 Tarantula<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(addComma(&quot;2 Ghosts&quot;))<br><br>Below is the expected output:<br><br>    2 Ghosts</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def addComma(string):<br>    match = re.match(r&quot;(\d+)\s+(.+)&quot;, string)<br>    if not match:<br>        return string<br><br>    count, species = match.groups()<br>    if int(count) &lt; 1000:<br>        return string<br><br>    return &quot;{:,} {}&quot;.format(int(count), species)<br></td><td class="s2">4</td><td class="s2" dir="ltr">4</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R140" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">141</div></th><td class="s4">15 - Regular Expressions 2</td><td class="s4">06 - Switch column order</td><td class="freezebar-cell"></td><td class="s4" dir="ltr">Here is a new Python programming task.<br><br>Suppose you are running a service to match hospital patients to potential blood donors. You have created a tab-delimited text file that indicates a person&#39;s name in the first column and their blood type in the second column. When you encounter a new patient at the hospital who needs a blood transfusion, you want to be able to search for people who match that person&#39;s blood type. As you have begin building that text file, you realize that you made a mistake. You have listed each person&#39;s name first and then their blood type. To make it easier for you to search, you would like to have the blood type first and then their name.<br><br>Please write a function called &quot;switchColumnOrder&quot; that accepts two arguments: 1) the name of an input file and 2) the name of a new file that your function will create. Your function should read the specified input file, switch the columns, and write the output to the new file.<br><br>Your function must use the &quot;re&quot; module to accomplish this task.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    switchColumnOrder(&quot;Transfusions1.txt&quot;, &quot;Transfusions1_Output.txt&quot;)<br>    switchColumnOrder(&quot;Transfusions2.txt&quot;, &quot;Transfusions2_Output.txt&quot;)<br>    <br>    def checkOutputFile(outputFilePath):<br>        if os.path.exists(outputFilePath):<br>            with open(outputFilePath) as outputFile:<br>                print(&quot;{}:&quot;.format(outputFilePath))<br>                for line in outputFile:<br>                    print(&quot;  &quot; + line.rstrip(&quot;\n&quot;))<br>                print(&quot;&quot;)<br>        else:<br>            print(&quot;No output file exists at {}.&quot;.format(outputFilePath))<br>    <br>    checkOutputFile(&quot;Transfusions1_Output.txt&quot;)<br>    checkOutputFile(&quot;Transfusions2_Output.txt&quot;)<br><br>Below is the expected output:<br><br>    Transfusions1_Output.txt:<br>      O+        Angeline Wheeler<br>      A+        Jamari Cordova<br>      A-        Iris Pierce<br>      B-        Odin Roy<br>      O+        Santiago Duran<br>      O+        Johan Sanders<br>      O-        Zion Haas<br>      AB+        Rachel Hayden<br>      O-        Walter Tran<br>      O+        Samantha Mann<br>    <br>    Transfusions2_Output.txt:<br>      O+        Grady Steele<br>      A+        Alvaro Gardner<br>      AB-        Sawyer Cooper<br>      O+        Raven Russell<br>      O-        Jeramiah Price<br>      B+        Lucy Cabrera<br>      B-        Maleah Sims<br>      A-        Skylar Villa<br>      O-        Jaidyn Figueroa<br>      O+        Joshua Hogan<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Transfusions1.txt<br><br>File contents:<br><br>    Angeline Wheeler        O+<br>    Jamari Cordova        A+<br>    Iris Pierce        A-<br>    Odin Roy        B-<br>    Santiago Duran        O+<br>    Johan Sanders        O+<br>    Zion Haas        O-<br>    Rachel Hayden        AB+<br>    Walter Tran        O-<br>    Samantha Mann        O+<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Transfusions2.txt<br><br>File contents:<br><br>    Grady Steele        O+<br>    Alvaro Gardner        A+<br>    Sawyer Cooper        AB-<br>    Raven Russell        O+<br>    Jeramiah Price        O-<br>    Lucy Cabrera        B+<br>    Maleah Sims        B-<br>    Skylar Villa        A-<br>    Jaidyn Figueroa        O-<br>    Joshua Hogan        O+</td><td class="s4" dir="ltr">Names were generated randomly via <span style="text-decoration:underline;-webkit-text-decoration-skip:none;text-decoration-skip-ink:none;color:#1155cc;"><a target="_blank" href="https://www.randomlists.com/random-names.">https://www.randomlists.com/random-names.</a></span></td><td class="s4" dir="ltr">Yes</td><td class="s4" dir="ltr">Yes</td><td class="s4">text</td><td class="s4" dir="ltr">SRP</td><td class="s5" dir="ltr">February 13, 2023</td><td class="s4" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s4" dir="ltr">import re<br><br>def switchColumnOrder(inputFileName, outputFileName):<br>    with open(inputFileName, &#39;r&#39;) as inputFile:<br>        lines = inputFile.readlines()<br>        with open(outputFileName, &#39;w&#39;) as outputFile:<br>            for line in lines:<br>                match = re.search(r&#39;(\w+)\s+(\w+\+|\w+-)&#39;, line)<br>                bloodType = match.group(2)<br>                name = match.group(1)<br>                outputFile.write(f&#39;{bloodType}    {name}\n&#39;)</td><td class="s5">1</td><td class="s5" dir="ltr">0</td><td class="s5" dir="ltr">10</td><td class="s4" dir="ltr">ChatGPT got very close on most of the attempts. But several times I got a Python error when executing the code. Other times, it would show people&#39;s blood type but not + or -. Sometimes it included only a person&#39;s first or last name but not both.</td><td class="s4"></td><td class="s4" dir="ltr">Yes</td><td class="s4" dir="ltr">Yes</td><td class="s4" dir="ltr"></td><td class="s4" dir="ltr">Yes</td><td class="s4" dir="ltr"></td><td class="s4" dir="ltr">Yes</td><td class="s4"></td><td class="s4" dir="ltr">Yes</td><td class="s4" dir="ltr">Yes</td><td class="s4"></td></tr><tr style="height: 20px"><th id="2089702959R141" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">142</div></th><td class="s1">15 - Regular Expressions 2</td><td class="s1">07 - Convert CSV to TSV</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Biologists commonly use two file formats to store data: comma-separated values (CSV) and tab-separated values (TSV). Most people use &quot;.csv&quot; as the file extension for CSV files and &quot;.tsv&quot; or &quot;.txt&quot; as the extension for TSV files.<br><br>One disadvantage of CSV files is that some of the values may have commas in them. For example, the first column might be a person&#39;s full name and might contain values like &quot;Doe, John&quot; or &quot;Fonda, Jane&quot;. When parsing such a file, you may have difficulty differentiating between commas that are part of a person&#39;s name versus commas that are used to separate different columns. Thus when people create CSV files, they often surround the values with double-quotation marks (see example below). In contrast, when using TSV files, these quotations marks are not needed.<br><br>Please write a function called &quot;convertCSVToTSV&quot; that accepts two arguments. The first argument will be the name of a CSV file that contains comma-separated values. The second argument will be the name of a TSV file that your function will need to create. Your function should do the following:<br><br>1. Read the specified input file (one line at a time).<br><br>2. Use the &quot;re&quot; module to replace commas with tabs, but only commas that are surrounded by double-quotation marks on each side.<br><br>3. Use the &quot;re&quot; module to remove the double-quotation marks that remain at the beginning and ending of the line.<br><br>4. Write the modified line to the output file.<br><br>Your function does not need to return anything.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    convertCSVToTSV(&quot;example1.csv&quot;, &quot;example1_output.tsv&quot;)<br>    convertCSVToTSV(&quot;example2.csv&quot;, &quot;example2_output.tsv&quot;)<br>    <br>    def checkOutputFile(outputFilePath):<br>        if os.path.exists(outputFilePath):<br>            with open(outputFilePath) as outputFile:<br>                print(&quot;{}:&quot;.format(outputFilePath))<br>                for line in outputFile:<br>                    print(&quot;  &quot; + line.rstrip(&quot;\n&quot;))<br>                print(&quot;&quot;)<br>        else:<br>            print(&quot;No output file exists at {}.&quot;.format(outputFilePath))<br>    <br>    checkOutputFile(&quot;example1_output.tsv&quot;)<br>    checkOutputFile(&quot;example2_output.tsv&quot;)<br><br>Below is the expected output:<br><br>    example1_output.tsv:<br>      Person        Project Name        Hours Per Week<br>      Doe, John        Bacteria        23<br>      Fonda, Jane        Mammals        109<br>      Smith, Jack        Reptiles        7<br>      <br>    <br>    example2_output.tsv:<br>      Person        Project Name        Hours Per Week<br>      Mouse, Mickey        Cooking        34<br>      Mouse, Minnie        Painting        58<br>      Duck, Donald        Cleaning        2<br>      Duck, Daisy        Weeding        4<br><br>Below is a description of a data file that is available for testing.<br><br>File name: example1.csv<br><br>File contents:<br><br>    &quot;Person&quot;,&quot;Project Name&quot;,&quot;Hours Per Week&quot;<br>    &quot;Doe, John&quot;,&quot;Bacteria&quot;,&quot;23&quot;<br>    &quot;Fonda, Jane&quot;,&quot;Mammals&quot;,&quot;109&quot;<br>    &quot;Smith, Jack&quot;,&quot;Reptiles&quot;,&quot;7&quot;<br><br>Below is a description of a data file that is available for testing.<br><br>File name: example2.csv<br><br>File contents:<br><br>    &quot;Person&quot;,&quot;Project Name&quot;,&quot;Hours Per Week&quot;<br>    &quot;Mouse, Mickey&quot;,&quot;Cooking&quot;,&quot;34&quot;<br>    &quot;Mouse, Minnie&quot;,&quot;Painting&quot;,&quot;58&quot;<br>    &quot;Duck, Donald&quot;,&quot;Cleaning&quot;,&quot;2&quot;<br>    &quot;Duck, Daisy&quot;,&quot;Weeding&quot;,&quot;4&quot;</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def convertCSVToTSV(inputFile, outputFile):<br>    with open(inputFile, &quot;r&quot;) as input_file:<br>        with open(outputFile, &quot;w&quot;) as output_file:<br>            for line in input_file:<br>                line = re.sub(r&#39;&quot;,&quot;&#39;, &#39;\t&#39;, line)<br>                line = re.sub(r&#39;^&quot;|&quot;$&#39;, &#39;&#39;, line)<br>                output_file.write(line)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R142" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">143</div></th><td class="s1">16 - Python Scripts &amp; &quot;while&quot; Loops</td><td class="s1">01 - Create a script that adds three numbers</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please create code that uses &quot;sys.argv&quot; to accept three arguments. Each of these arguments will be numbers. However, when you retrieve via &quot;sys.argv&quot;, they will be string values, so you will need to convert them to integers. Sum the three numbers and then print the sum.<br><br>FYI: The code that you enter in the box below will be placed in a Python script called &quot;code.py&quot; and then that script will be executed (behind the scenes) at the command line as shown in the test code.<br><br>Here is some code to get your started:<br><br>    # Here is some code to get you started...<br>    <br>    import sys<br>    <br>    number1 = sys.argv[1]<br>    number2 = sys.argv[2]<br>    number3 = sys.argv[3]<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    # These commands will be executed at the command line.<br>    <br>    python code.py &quot;2&quot; &quot;2&quot; &quot;2&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;4&quot; &quot;3&quot; &quot;1&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;-4&quot; &quot;3&quot; &quot;0&quot;</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br><br>number1 = int(sys.argv[1])<br>number2 = int(sys.argv[2])<br>number3 = int(sys.argv[3])<br><br>sum = number1 + number2 + number3<br><br>print(sum)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">The first try did the calculations correctly but printed the answer as a sentence rather than just printing the sum.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R143" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">144</div></th><td class="s1">16 - Python Scripts &amp; &quot;while&quot; Loops</td><td class="s1">02 - Count metadata lines in VCF file</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>For this problem, you will work with DNA variant data for some Ebolavirus genomes. The variant information is stored in VCF files.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept one argument. This argument will be a string and will be the name of one of the VCF files. Your script should count the number of &quot;metadata&quot; lines in the specified VCF file, not including the header line that contains the column names. Your script should then print this number.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972917.vcf&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972918.vcf&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972919.vcf&quot;<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972919.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972919.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        6219        A        C        NO PASS        GP<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972917.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972917.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1007        G        A        NO PASS        NP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        6314        C        T        PASS        GP<br>    KJ660346        16981        A        G        NO PASS        L<br>    KJ660346        17798        A        G        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972918.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972918.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1346        T        C        NO PASS        NP<br>    KJ660346        1755        T        C        NO PASS        NP<br>    KJ660346        1849        T        C        PASS        NP<br>    KJ660346        2553        A        G        NO PASS        NP<br>    KJ660346        4737        G        A        NO PASS        VP40<br>    KJ660346        4753        T        A        NO PASS        VP40<br>    KJ660346        6226        T        C        NO PASS        GP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        7783        T        G        NO PASS        GP<br>    KJ660346        7912        G        A        PASS        GP<br>    KJ660346        7930        T        C        NO PASS        GP<br>    KJ660346        8145        A        G        NO PASS        VP30<br>    KJ660346        8178        A        C        NO PASS        VP30<br>    KJ660346        8778        C        T        NO PASS        VP30<br>    KJ660346        8928        A        C        PASS        VP30<br>    KJ660346        9532        T        G        NO PASS        VP24<br>    KJ660346        9829        TAAAAAT        TAAAAAAT        NO PASS        VP24<br>    KJ660346        10232        A        G        PASS        VP24<br>    KJ660346        10752        G        A        PASS        VP24<br>    KJ660346        11980        GAAAAAAT        GAAAAAAAT        NO PASS        L<br>    KJ660346        15377        A        G        NO PASS        L<br>    KJ660346        16774        A        G        NO PASS        L<br>    KJ660346        17142        T        C        PASS        L<br>    KJ660346        18345        A        G        PASS        L</td><td class="s1" dir="ltr">The data were generated as part of this study: <span style="text-decoration:underline;-webkit-text-decoration-skip:none;text-decoration-skip-ink:none;color:#1155cc;"><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/25214632.">https://pubmed.ncbi.nlm.nih.gov/25214632.</a></span></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br><br># Get the file name from command line argument<br>filename = sys.argv[1]<br><br># Counter for metadata lines<br>metadata_count = 0<br><br># Open the file and count the metadata lines<br>with open(filename, &#39;r&#39;) as f:<br>    for line in f:<br>        if line.startswith(&#39;##&#39;):<br>            metadata_count += 1<br>        elif line.startswith(&#39;#&#39;):<br>            break<br><br># Print the number of metadata lines<br>print(metadata_count)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R144" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">145</div></th><td class="s1">16 - Python Scripts &amp; &quot;while&quot; Loops</td><td class="s1">03 - Count columns in VCF file</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>For this problem, you will evaluate VCF files from the Ebolavirus project.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept one argument. This argument will be a string and will be the name of one of the VCF files. Your script should count the number of header columns in the specified VCF file. Your script should then print this number.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972917.vcf&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972918.vcf&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972919.vcf&quot;<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972917.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972917.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1007        G        A        NO PASS        NP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        6314        C        T        PASS        GP<br>    KJ660346        16981        A        G        NO PASS        L<br>    KJ660346        17798        A        G        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972918.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972918.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1346        T        C        NO PASS        NP<br>    KJ660346        1755        T        C        NO PASS        NP<br>    KJ660346        1849        T        C        PASS        NP<br>    KJ660346        2553        A        G        NO PASS        NP<br>    KJ660346        4737        G        A        NO PASS        VP40<br>    KJ660346        4753        T        A        NO PASS        VP40<br>    KJ660346        6226        T        C        NO PASS        GP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        7783        T        G        NO PASS        GP<br>    KJ660346        7912        G        A        PASS        GP<br>    KJ660346        7930        T        C        NO PASS        GP<br>    KJ660346        8145        A        G        NO PASS        VP30<br>    KJ660346        8178        A        C        NO PASS        VP30<br>    KJ660346        8778        C        T        NO PASS        VP30<br>    KJ660346        8928        A        C        PASS        VP30<br>    KJ660346        9532        T        G        NO PASS        VP24<br>    KJ660346        9829        TAAAAAT        TAAAAAAT        NO PASS        VP24<br>    KJ660346        10232        A        G        PASS        VP24<br>    KJ660346        10752        G        A        PASS        VP24<br>    KJ660346        11980        GAAAAAAT        GAAAAAAAT        NO PASS        L<br>    KJ660346        15377        A        G        NO PASS        L<br>    KJ660346        16774        A        G        NO PASS        L<br>    KJ660346        17142        T        C        PASS        L<br>    KJ660346        18345        A        G        PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972919.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972919.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        6219        A        C        NO PASS        GP</td><td class="s1" dir="ltr">The data were generated as part of this study: <span style="text-decoration:underline;-webkit-text-decoration-skip:none;text-decoration-skip-ink:none;color:#1155cc;"><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/25214632.">https://pubmed.ncbi.nlm.nih.gov/25214632.</a></span></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br><br>filename = sys.argv[1]<br><br>with open(filename, &#39;r&#39;) as f:<br>    for line in f:<br>        if line.startswith(&#39;#CHROM&#39;):<br>            header = line.strip().split()<br>            print(len(header))<br>            break</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R145" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">146</div></th><td class="s1">16 - Python Scripts &amp; &quot;while&quot; Loops</td><td class="s1">04 - Count variants in VCF file</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>For this problem, you will evaluate VCF files from the Ebolavirus project.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept one argument. This argument will be a string and will be the name of one of the VCF files. Your script should count the number of variants in the specified VCF file. Your script should then print this number.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972917.vcf&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972918.vcf&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972919.vcf&quot;<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972917.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972917.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1007        G        A        NO PASS        NP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        6314        C        T        PASS        GP<br>    KJ660346        16981        A        G        NO PASS        L<br>    KJ660346        17798        A        G        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972918.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972918.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1346        T        C        NO PASS        NP<br>    KJ660346        1755        T        C        NO PASS        NP<br>    KJ660346        1849        T        C        PASS        NP<br>    KJ660346        2553        A        G        NO PASS        NP<br>    KJ660346        4737        G        A        NO PASS        VP40<br>    KJ660346        4753        T        A        NO PASS        VP40<br>    KJ660346        6226        T        C        NO PASS        GP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        7783        T        G        NO PASS        GP<br>    KJ660346        7912        G        A        PASS        GP<br>    KJ660346        7930        T        C        NO PASS        GP<br>    KJ660346        8145        A        G        NO PASS        VP30<br>    KJ660346        8178        A        C        NO PASS        VP30<br>    KJ660346        8778        C        T        NO PASS        VP30<br>    KJ660346        8928        A        C        PASS        VP30<br>    KJ660346        9532        T        G        NO PASS        VP24<br>    KJ660346        9829        TAAAAAT        TAAAAAAT        NO PASS        VP24<br>    KJ660346        10232        A        G        PASS        VP24<br>    KJ660346        10752        G        A        PASS        VP24<br>    KJ660346        11980        GAAAAAAT        GAAAAAAAT        NO PASS        L<br>    KJ660346        15377        A        G        NO PASS        L<br>    KJ660346        16774        A        G        NO PASS        L<br>    KJ660346        17142        T        C        PASS        L<br>    KJ660346        18345        A        G        PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972919.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972919.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        6219        A        C        NO PASS        GP</td><td class="s1" dir="ltr">The data were generated as part of this study: <span style="text-decoration:underline;-webkit-text-decoration-skip:none;text-decoration-skip-ink:none;color:#1155cc;"><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/25214632.">https://pubmed.ncbi.nlm.nih.gov/25214632.</a></span></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br><br>filename = sys.argv[1]<br><br>with open(filename, &quot;r&quot;) as f:<br>    variants = 0<br>    for line in f:<br>        if not line.startswith(&quot;#&quot;):<br>            variants += 1<br><br>print(variants)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">3</td><td class="s1" dir="ltr">The first try did the calculations correctly but printed &quot;Number of variants: &quot; and then the count. I asked it to print just the count, but it gave me a completely irrelevant answer (perhaps a bug in the system). I told ChatGPT that the answer was irrelevant and gave it the original prompt and ask it to just print the count with no additional text.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td></tr><tr style="height: 20px"><th id="2089702959R146" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">147</div></th><td class="s1">16 - Python Scripts &amp; &quot;while&quot; Loops</td><td class="s1">05 - Count passing GP variants in VCF file</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>For this problem, you will evaluate VCF files from the Ebolavirus project.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept one argument. This argument will be a string and will be the name of one of the VCF files. Your script should count the number of variants in the GP gene that have a value of &quot;PASS&quot; in the FILTER column within the specified VCF file. Your script should then print this number.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972917.vcf&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972918.vcf&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972919.vcf&quot;<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972917.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972917.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1007        G        A        NO PASS        NP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        6314        C        T        PASS        GP<br>    KJ660346        16981        A        G        NO PASS        L<br>    KJ660346        17798        A        G        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972918.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972918.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1346        T        C        NO PASS        NP<br>    KJ660346        1755        T        C        NO PASS        NP<br>    KJ660346        1849        T        C        PASS        NP<br>    KJ660346        2553        A        G        NO PASS        NP<br>    KJ660346        4737        G        A        NO PASS        VP40<br>    KJ660346        4753        T        A        NO PASS        VP40<br>    KJ660346        6226        T        C        NO PASS        GP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        7783        T        G        NO PASS        GP<br>    KJ660346        7912        G        A        PASS        GP<br>    KJ660346        7930        T        C        NO PASS        GP<br>    KJ660346        8145        A        G        NO PASS        VP30<br>    KJ660346        8178        A        C        NO PASS        VP30<br>    KJ660346        8778        C        T        NO PASS        VP30<br>    KJ660346        8928        A        C        PASS        VP30<br>    KJ660346        9532        T        G        NO PASS        VP24<br>    KJ660346        9829        TAAAAAT        TAAAAAAT        NO PASS        VP24<br>    KJ660346        10232        A        G        PASS        VP24<br>    KJ660346        10752        G        A        PASS        VP24<br>    KJ660346        11980        GAAAAAAT        GAAAAAAAT        NO PASS        L<br>    KJ660346        15377        A        G        NO PASS        L<br>    KJ660346        16774        A        G        NO PASS        L<br>    KJ660346        17142        T        C        PASS        L<br>    KJ660346        18345        A        G        PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972919.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972919.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        6219        A        C        NO PASS        GP</td><td class="s1" dir="ltr">The data were generated as part of this study: <span style="text-decoration:underline;-webkit-text-decoration-skip:none;text-decoration-skip-ink:none;color:#1155cc;"><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/25214632.">https://pubmed.ncbi.nlm.nih.gov/25214632.</a></span></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br><br>filename = sys.argv[1]<br><br>with open(filename, &quot;r&quot;) as f:<br>    count = 0<br>    for line in f:<br>        if line.startswith(&quot;#&quot;):<br>            continue<br>        line = line.strip().split(&quot;\t&quot;)<br>        if line[4] == &quot;PASS&quot; and line[5] == &quot;GP&quot;:<br>            count += 1<br><br>print(count)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">6</td><td class="s1" dir="ltr">At first it got an error message because it did not ignore the lines that start with &quot;#&quot;. I went back and forth a few times until it provided a correct solution.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R147" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">148</div></th><td class="s1">16 - Python Scripts &amp; &quot;while&quot; Loops</td><td class="s1">06 - Summarize number of variants per chromosome in VCF file</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>For this problem, you will evaluate VCF files from the Ebolavirus project.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept two arguments. The first argument will be a string and will be the name of one of the VCF files. The second argument will be a string and will be the name of an output file that your script should create. Your script should count the number of variants per chromosome in the input file and save this information to the output file. Consider only variants with a &quot;FILTER&quot; value of &quot;PASS&quot;. Only output chromosomes that have at least one variant that passes the quality filter (no zero values).<br><br>As you write each line to the output file, first write the chromosome name (for example, 1, 2, or X), then a tab character, then the number of variants on that chromosome, followed by a newline character.<br><br>The order in which you write the chromosome information to the output file does not matter.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972917.vcf&quot; &quot;SRR1972917_Output.txt&quot;<br>    <br>    echo &quot;SRR1972917_Output.txt&quot;<br>    cat &quot;SRR1972917_Output.txt&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972918.vcf&quot; &quot;SRR1972918_Output.txt&quot;<br>    <br>    echo &quot;SRR1972918_Output.txt&quot;<br>    cat &quot;SRR1972918_Output.txt&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972919.vcf&quot; &quot;SRR1972919_Output.txt&quot;<br>    <br>    echo &quot;SRR1972919_Output.txt&quot;<br>    cat &quot;SRR1972919_Output.txt&quot;<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972917.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972917.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1007        G        A        NO PASS        NP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        6314        C        T        PASS        GP<br>    KJ660346        16981        A        G        NO PASS        L<br>    KJ660346        17798        A        G        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972918.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972918.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1346        T        C        NO PASS        NP<br>    KJ660346        1755        T        C        NO PASS        NP<br>    KJ660346        1849        T        C        PASS        NP<br>    KJ660346        2553        A        G        NO PASS        NP<br>    KJ660346        4737        G        A        NO PASS        VP40<br>    KJ660346        4753        T        A        NO PASS        VP40<br>    KJ660346        6226        T        C        NO PASS        GP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        7783        T        G        NO PASS        GP<br>    KJ660346        7912        G        A        PASS        GP<br>    KJ660346        7930        T        C        NO PASS        GP<br>    KJ660346        8145        A        G        NO PASS        VP30<br>    KJ660346        8178        A        C        NO PASS        VP30<br>    KJ660346        8778        C        T        NO PASS        VP30<br>    KJ660346        8928        A        C        PASS        VP30<br>    KJ660346        9532        T        G        NO PASS        VP24<br>    KJ660346        9829        TAAAAAT        TAAAAAAT        NO PASS        VP24<br>    KJ660346        10232        A        G        PASS        VP24<br>    KJ660346        10752        G        A        PASS        VP24<br>    KJ660346        11980        GAAAAAAT        GAAAAAAAT        NO PASS        L<br>    KJ660346        15377        A        G        NO PASS        L<br>    KJ660346        16774        A        G        NO PASS        L<br>    KJ660346        17142        T        C        PASS        L<br>    KJ660346        18345        A        G        PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972919.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972919.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        6219        A        C        NO PASS        GP</td><td class="s1" dir="ltr">The data were generated as part of this study: <span style="text-decoration:underline;-webkit-text-decoration-skip:none;text-decoration-skip-ink:none;color:#1155cc;"><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/25214632.">https://pubmed.ncbi.nlm.nih.gov/25214632.</a></span></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br><br>def count_variants(input_file, output_file):<br>    with open(input_file, &#39;r&#39;) as f:<br>        contents = f.readlines()<br>    <br>    chrom_counts = {}<br>    for line in contents:<br>        if line.startswith(&quot;#&quot;) or &#39;NO PASS&#39; in line:<br>            continue<br>        <br>        chrom, _, _, _, _, _ = line.strip().split(&quot;\t&quot;)<br>        chrom_counts[chrom] = chrom_counts.get(chrom, 0) + 1<br>    <br>    with open(output_file, &#39;w&#39;) as f:<br>        for chrom, count in chrom_counts.items():<br>            f.write(f&quot;{chrom}\t{count}\n&quot;)<br><br>input_file = sys.argv[1]<br>output_file = sys.argv[2]<br>count_variants(input_file, output_file)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">7</td><td class="s1" dir="ltr">It got confused by the extra header lines. At times it wanted to try to help me debug the solution, but I insisted that it provide me a complete solution and continued interacting with it until it did. Sometimes its solutions were quite different from one iteration to the next.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R148" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">149</div></th><td class="s1">16 - Python Scripts &amp; &quot;while&quot; Loops</td><td class="s1">07 - Identify most common reference allele in VCF file</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>For this problem, you will evaluate VCF files from the Ebolavirus project.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept one argument. This argument will be a string and will be the name of one of the VCF files. The fourth column in the VCF files is the alternate allele for each variant. Your script should identify the alternate allele that is most common within the specified VCF file. Consider only variants with a FILTER value of &quot;PASS&quot;. Your script should print the name of the VCF file, followed by a colon and a space, followed by the alternate allele that is most common in the specified file (for example, &quot;SRR1972917.vcf: T&quot;). If there are no passing variants, the alternate allele should be listed as &quot;N/A&quot; (for example, &quot;SRR1972919.vcf: N/A&quot;).<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972917.vcf&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972918.vcf&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SRR1972919.vcf&quot;<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972917.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972917.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1007        G        A        NO PASS        NP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        6314        C        T        PASS        GP<br>    KJ660346        16981        A        G        NO PASS        L<br>    KJ660346        17798        A        G        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972918.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972918.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1346        T        C        NO PASS        NP<br>    KJ660346        1755        T        C        NO PASS        NP<br>    KJ660346        1849        T        C        PASS        NP<br>    KJ660346        2553        A        G        NO PASS        NP<br>    KJ660346        4737        G        A        NO PASS        VP40<br>    KJ660346        4753        T        A        NO PASS        VP40<br>    KJ660346        6226        T        C        NO PASS        GP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        7783        T        G        NO PASS        GP<br>    KJ660346        7912        G        A        PASS        GP<br>    KJ660346        7930        T        C        NO PASS        GP<br>    KJ660346        8145        A        G        NO PASS        VP30<br>    KJ660346        8178        A        C        NO PASS        VP30<br>    KJ660346        8778        C        T        NO PASS        VP30<br>    KJ660346        8928        A        C        PASS        VP30<br>    KJ660346        9532        T        G        NO PASS        VP24<br>    KJ660346        9829        TAAAAAT        TAAAAAAT        NO PASS        VP24<br>    KJ660346        10232        A        G        PASS        VP24<br>    KJ660346        10752        G        A        PASS        VP24<br>    KJ660346        11980        GAAAAAAT        GAAAAAAAT        NO PASS        L<br>    KJ660346        15377        A        G        NO PASS        L<br>    KJ660346        16774        A        G        NO PASS        L<br>    KJ660346        17142        T        C        PASS        L<br>    KJ660346        18345        A        G        PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972919.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972919.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        6219        A        C        NO PASS        GP</td><td class="s1" dir="ltr">The data were generated as part of this study: <span style="text-decoration:underline;-webkit-text-decoration-skip:none;text-decoration-skip-ink:none;color:#1155cc;"><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/25214632.">https://pubmed.ncbi.nlm.nih.gov/25214632.</a></span></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 9, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>from collections import Counter<br><br>filename = sys.argv[1]<br>alleles = []<br><br>with open(filename, &#39;r&#39;) as f:<br>    for line in f:<br>        if line.startswith(&quot;#&quot;):<br>            continue<br>        fields = line.strip().split(&quot;\t&quot;)<br>        if fields[4] == &quot;PASS&quot;:<br>            alleles.append(fields[3])<br><br>if len(alleles) == 0:<br>    print(f&quot;{filename}: N/A&quot;)<br>else:<br>    most_common = Counter(alleles).most_common(1)<br>    print(f&quot;{filename}: {most_common[0][0]}&quot;)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">It used collections.Counter, which would be unfamiliar to students in this course. But it worked.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R149" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">150</div></th><td class="s1">16 - Python Scripts &amp; &quot;while&quot; Loops</td><td class="s1">08 - Identify most frequently mutated sample in VCF</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>For this problem, you will evaluate 60 VCF files from the Ebolavirus project.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a string and will be a wildcard path (for example, &quot;.vcf&quot;) that indicates where the VCF files are stored. Your script should use the &quot;glob&quot; module to identify these file names and iterate through them (see this tutorial).<br><br>Your script should loop through the VCF files and identify which VCF file has the largest number of variants that have a FILTER value of &quot;PASS&quot;. Your script should print the name of this VCF file.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;.vcf&quot;<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972917.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972917.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1007        G        A        NO PASS        NP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        6314        C        T        PASS        GP<br>    KJ660346        16981        A        G        NO PASS        L<br>    KJ660346        17798        A        G        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972918.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972918.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1346        T        C        NO PASS        NP<br>    KJ660346        1755        T        C        NO PASS        NP<br>    KJ660346        1849        T        C        PASS        NP<br>    KJ660346        2553        A        G        NO PASS        NP<br>    KJ660346        4737        G        A        NO PASS        VP40<br>    KJ660346        4753        T        A        NO PASS        VP40<br>    KJ660346        6226        T        C        NO PASS        GP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        7783        T        G        NO PASS        GP<br>    KJ660346        7912        G        A        PASS        GP<br>    KJ660346        7930        T        C        NO PASS        GP<br>    KJ660346        8145        A        G        NO PASS        VP30<br>    KJ660346        8178        A        C        NO PASS        VP30<br>    KJ660346        8778        C        T        NO PASS        VP30<br>    KJ660346        8928        A        C        PASS        VP30<br>    KJ660346        9532        T        G        NO PASS        VP24<br>    KJ660346        9829        TAAAAAT        TAAAAAAT        NO PASS        VP24<br>    KJ660346        10232        A        G        PASS        VP24<br>    KJ660346        10752        G        A        PASS        VP24<br>    KJ660346        11980        GAAAAAAT        GAAAAAAAT        NO PASS        L<br>    KJ660346        15377        A        G        NO PASS        L<br>    KJ660346        16774        A        G        NO PASS        L<br>    KJ660346        17142        T        C        PASS        L<br>    KJ660346        18345        A        G        PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972919.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972919.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        6219        A        C        NO PASS        GP<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972920.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972920.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        6190        T        C        NO PASS        GP<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972921.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972921.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        321        A        G        NO PASS        NP<br>    KJ660346        969        C        T        NO PASS        NP<br>    KJ660346        3186        A        G        NO PASS        VP35<br>    KJ660346        3667        C        G        NO PASS        VP35<br>    KJ660346        3905        T        C        NO PASS        VP35<br>    KJ660346        4153        C        T        NO PASS        VP40<br>    KJ660346        5076        A        G        NO PASS        VP40<br>    KJ660346        6425        T        C        NO PASS        GP<br>    KJ660346        6581        T        C        NO PASS        GP<br>    KJ660346        6605        T        G        NO PASS        GP<br>    KJ660346        7912        G        A        NO PASS        GP<br>    KJ660346        8002        T        G        NO PASS        GP<br>    KJ660346        8928        A        C        PASS        VP30<br>    KJ660346        9507        TAA        TA        NO PASS        VP24<br>    KJ660346        10109        C        T        NO PASS        VP24<br>    KJ660346        10218        G        A        PASS        VP24<br>    KJ660346        10232        A        G        NO PASS        VP24<br>    KJ660346        10752        G        A        PASS        VP24<br>    KJ660346        16982        C        A        NO PASS        L<br>    KJ660346        18345        A        G        PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972922.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972922.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        6726        A        G        NO PASS        GP<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972923.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972923.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972924.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972924.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972925.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972925.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        5621        A        G        NO PASS        GP<br>    KJ660346        5849        T        C        PASS        GP<br>    KJ660346        5855        CAA        CA        NO PASS        GP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        8928        A        C        PASS        VP30<br>    KJ660346        10291        C        T        NO PASS        VP24<br>    KJ660346        18524        T        C        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972926.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972926.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972927.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972927.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972928.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972928.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        2639        G        A        NO PASS        NP<br>    KJ660346        2643        T        A        NO PASS        NP<br>    KJ660346        3752        A        C        NO PASS        VP35<br>    KJ660346        3758        T        G        NO PASS        VP35<br>    KJ660346        3773        G        A        NO PASS        VP35<br>    KJ660346        3790        C        A        NO PASS        VP35<br>    KJ660346        4160        CTT        CT        NO PASS        VP40<br>    KJ660346        4489        T        G        NO PASS        VP40<br>    KJ660346        4496        G        T        NO PASS        VP40<br>    KJ660346        4539        A        C        NO PASS        VP40<br>    KJ660346        4775        T        G        NO PASS        VP40<br>    KJ660346        4895        T        G        NO PASS        VP40<br>    KJ660346        6283        C        T        PASS        GP<br>    KJ660346        6469        CGG        CG        NO PASS        GP<br>    KJ660346        6605        T        G        NO PASS        GP<br>    KJ660346        7289        T        C        NO PASS        GP<br>    KJ660346        7526        A        G        NO PASS        GP<br>    KJ660346        7533        C        G        NO PASS        GP<br>    KJ660346        7536        G        C        NO PASS        GP<br>    KJ660346        8158        TTGA        TA        NO PASS        VP30<br>    KJ660346        9187        T        C        NO PASS        VP30<br>    KJ660346        10752        G        A        NO PASS        VP24<br>    KJ660346        13890        GAGGG        TAGGC        NO PASS        L<br>    KJ660346        15296        TAA        TA        NO PASS        L<br>    KJ660346        15963        G        A        NO PASS        L<br>    KJ660346        16212        A        C        NO PASS        L<br>    KJ660346        16238        C        A        NO PASS        L<br>    KJ660346        16633        G        C        NO PASS        L<br>    KJ660346        18276        TAAAAAAC        TAAAAAAAC        NO PASS        L<br>    KJ660346        18345        A        G        PASS        L<br>    KJ660346        18613        GTTTTTTA        GTTTTTTTA        NO PASS        L<br>    KJ660346        18633        C        T        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972929.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972929.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972930.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972930.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972931.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972931.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972932.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972932.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        NO PASS        NP<br>    KJ660346        1849        T        C        NO PASS        NP<br>    KJ660346        4895        T        G        NO PASS        VP40<br>    KJ660346        5738        T        C        NO PASS        GP<br>    KJ660346        8928        A        C        NO PASS        VP30<br>    KJ660346        9839        G        T        NO PASS        VP24<br>    KJ660346        10478        T        G        NO PASS        VP24<br>    KJ660346        13196        A        T        NO PASS        L<br>    KJ660346        13444        G        A        NO PASS        L<br>    KJ660346        14439        T        G        NO PASS        L<br>    KJ660346        14447        A        C        NO PASS        L<br>    KJ660346        14456        A        C        NO PASS        L<br>    KJ660346        14461        A        C        NO PASS        L<br>    KJ660346        16225        T        G        NO PASS        L<br>    KJ660346        16243        A        C        NO PASS        L<br>    KJ660346        16247        C        T        NO PASS        L<br>    KJ660346        16248        A        C        NO PASS        L<br>    KJ660346        17142        T        C        NO PASS        L<br>    KJ660346        17878        ACC        AC        NO PASS        L<br>    KJ660346        18345        A        G        NO PASS        L<br>    KJ660346        18911        G        A        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972933.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972933.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972934.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972934.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        NO PASS        NP<br>    KJ660346        5449        T        C        NO PASS        VP40<br>    KJ660346        8105        C        T        NO PASS        VP30<br>    KJ660346        8614        C        T        NO PASS        VP30<br>    KJ660346        8928        A        C        NO PASS        VP30<br>    KJ660346        10218        G        A        NO PASS        VP24<br>    KJ660346        10232        A        G        NO PASS        VP24<br>    KJ660346        10752        G        A        NO PASS        VP24<br>    KJ660346        10801        A        T        NO PASS        VP24<br>    KJ660346        11014        G        A        NO PASS        VP24<br>    KJ660346        11414        T        A        NO PASS        L<br>    KJ660346        12919        C        A        NO PASS        L<br>    KJ660346        13767        T        G        NO PASS        L<br>    KJ660346        15963        G        A        PASS        L<br>    KJ660346        17142        T        C        NO PASS        L<br>    KJ660346        18345        A        G        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972935.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972935.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        7912        G        A        NO PASS        GP<br>    KJ660346        14009        T        G        NO PASS        L<br>    KJ660346        14021        A        G        NO PASS        L<br>    KJ660346        14028        T        G        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972936.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972936.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972937.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972937.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972938.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972938.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        2893        A        G        NO PASS        VP35<br>    KJ660346        17142        T        C        NO PASS        L<br>    KJ660346        17760        T        C        NO PASS        L<br>    KJ660346        18920        A        G        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972939.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972939.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972940.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972940.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        7912        G        A        NO PASS        GP<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972941.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972941.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        2861        T        C        NO PASS        VP35<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972942.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972942.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972943.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972943.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972944.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972944.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        13911        A        C        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972945.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972945.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972946.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972946.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972947.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972947.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972948.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972948.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        1179        C        A        NO PASS        NP<br>    KJ660346        1192        T        G        NO PASS        NP<br>    KJ660346        1266        C        T        NO PASS        NP<br>    KJ660346        1722        G        T        NO PASS        NP<br>    KJ660346        3451        A        C        NO PASS        VP35<br>    KJ660346        3567        A        C        NO PASS        VP35<br>    KJ660346        4214        G        A        NO PASS        VP40<br>    KJ660346        5862        A        C        NO PASS        GP<br>    KJ660346        5971        A        C        NO PASS        GP<br>    KJ660346        6503        A        C        NO PASS        GP<br>    KJ660346        8928        A        C        NO PASS        VP30<br>    KJ660346        9277        G        T        NO PASS        VP30<br>    KJ660346        10218        G        A        NO PASS        VP24<br>    KJ660346        12699        A        C        NO PASS        L<br>    KJ660346        12748        A        T        NO PASS        L<br>    KJ660346        13217        G        T        NO PASS        L<br>    KJ660346        13506        T        G        NO PASS        L<br>    KJ660346        13508        AGT        AGG        NO PASS        L<br>    KJ660346        13517        C        T        NO PASS        L<br>    KJ660346        13590        A        C        NO PASS        L<br>    KJ660346        14824        T        A        NO PASS        L<br>    KJ660346        14841        T        A        NO PASS        L<br>    KJ660346        16356        C        A        NO PASS        L<br>    KJ660346        16357        C        T        NO PASS        L<br>    KJ660346        16358        TC        TT        NO PASS        L<br>    KJ660346        16375        C        A        NO PASS        L<br>    KJ660346        16795        A        G        NO PASS        L<br>    KJ660346        16943        C        A        NO PASS        L<br>    KJ660346        17142        T        C        NO PASS        L<br>    KJ660346        17329        C        A        NO PASS        L<br>    KJ660346        17351        T        G        NO PASS        L<br>    KJ660346        17653        G        A        NO PASS        L<br>    KJ660346        17740        T        G        NO PASS        L<br>    KJ660346        17781        A        G        NO PASS        L<br>    KJ660346        17874        A        C        NO PASS        L<br>    KJ660346        17928        T        G        NO PASS        L<br>    KJ660346        17987        T        C        NO PASS        L<br>    KJ660346        18345        A        G        PASS        L<br>    KJ660346        18911        G        A        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972949.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972949.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        686        T        C        NO PASS        NP<br>    KJ660346        800        C        T        NO PASS        NP<br>    KJ660346        1179        C        A        NO PASS        NP<br>    KJ660346        2422        G        T        NO PASS        NP<br>    KJ660346        3056        C        T        NO PASS        VP35<br>    KJ660346        3254        C        G        NO PASS        VP35<br>    KJ660346        4295        C        T        NO PASS        VP40<br>    KJ660346        4435        CAT        CT        NO PASS        VP40<br>    KJ660346        4441        C        T        NO PASS        VP40<br>    KJ660346        4496        G        T        NO PASS        VP40<br>    KJ660346        4565        T        G        NO PASS        VP40<br>    KJ660346        4730        C        A        NO PASS        VP40<br>    KJ660346        4777        T        C        NO PASS        VP40<br>    KJ660346        5214        G        A        NO PASS        VP40<br>    KJ660346        5315        A        G        NO PASS        VP40<br>    KJ660346        5405        G        C        NO PASS        VP40<br>    KJ660346        5447        G        T        NO PASS        VP40<br>    KJ660346        6063        T        G        NO PASS        GP<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        8002        T        G        NO PASS        GP<br>    KJ660346        9720        C        T        NO PASS        VP24<br>    KJ660346        10218        G        A        PASS        VP24<br>    KJ660346        10349        C        G        NO PASS        VP24<br>    KJ660346        10769        A        C        NO PASS        VP24<br>    KJ660346        11513        GAAAAAC        GAAAAAAC        PASS        L<br>    KJ660346        11782        G        T        NO PASS        L<br>    KJ660346        11851        C        A        NO PASS        L<br>    KJ660346        12228        A        G        NO PASS        L<br>    KJ660346        12874        G        T        NO PASS        L<br>    KJ660346        12905        A        C        NO PASS        L<br>    KJ660346        13846        A        C        NO PASS        L<br>    KJ660346        14019        T        C        NO PASS        L<br>    KJ660346        14895        C        G        NO PASS        L<br>    KJ660346        15306        A        C        NO PASS        L<br>    KJ660346        15312        G        C        NO PASS        L<br>    KJ660346        15313        G        A        NO PASS        L<br>    KJ660346        15405        C        T        NO PASS        L<br>    KJ660346        15801        C        T        PASS        L<br>    KJ660346        15963        G        A        PASS        L<br>    KJ660346        16813        C        A        NO PASS        L<br>    KJ660346        16855        C        A        NO PASS        L<br>    KJ660346        17142        T        C        PASS        L<br>    KJ660346        18081        T        C        PASS        L<br>    KJ660346        18911        G        A        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972950.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972950.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972951.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972951.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1849        T        C        NO PASS        NP<br>    KJ660346        3945        C        T        PASS        VP35<br>    KJ660346        4570        A        T        NO PASS        VP40<br>    KJ660346        4722        A        G        NO PASS        VP40<br>    KJ660346        5965        G        A        NO PASS        GP<br>    KJ660346        5996        C        A        NO PASS        GP<br>    KJ660346        6149        T        C        NO PASS        GP<br>    KJ660346        6598        C        T        NO PASS        GP<br>    KJ660346        7203        ATA        GTT        NO PASS        GP<br>    KJ660346        7212        CC        GG        NO PASS        GP<br>    KJ660346        7526        A        G        NO PASS        GP<br>    KJ660346        7577        A        C        NO PASS        GP<br>    KJ660346        7588        C        A        NO PASS        GP<br>    KJ660346        8002        T        G        NO PASS        GP<br>    KJ660346        8736        A        G        PASS        VP30<br>    KJ660346        10218        G        A        NO PASS        VP24<br>    KJ660346        10545        T        C        NO PASS        VP24<br>    KJ660346        14886        A        G        NO PASS        L<br>    KJ660346        15006        T        A        NO PASS        L<br>    KJ660346        15510        C        T        PASS        L<br>    KJ660346        16794        C        T        NO PASS        L<br>    KJ660346        17928        T        G        NO PASS        L<br>    KJ660346        18304        T        C        PASS        L<br>    KJ660346        18673        T        C        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972952.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972952.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972953.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972953.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972954.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972954.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        17226        G        T        NO PASS        L<br>    KJ660346        17415        A        C        NO PASS        L<br>    KJ660346        18609        T        A        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972955.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972955.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        301        T        C        PASS        NP<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1849        T        C        PASS        NP<br>    KJ660346        4200        A        G        PASS        VP40<br>    KJ660346        4329        T        A        NO PASS        VP40<br>    KJ660346        5849        T        C        PASS        GP<br>    KJ660346        6283        C        T        PASS        GP<br>    KJ660346        8928        A        C        PASS        VP30<br>    KJ660346        10218        G        A        PASS        VP24<br>    KJ660346        10229        TAAAAAAT        TAAAAAT        PASS        VP24<br>    KJ660346        11851        C        A        NO PASS        L<br>    KJ660346        14019        T        C        PASS        L<br>    KJ660346        14590        T        C        PASS        L<br>    KJ660346        15963        G        A        PASS        L<br>    KJ660346        17142        T        C        PASS        L<br>    KJ660346        17848        G        A        PASS        L<br>    KJ660346        18706        T        C        PASS        L<br>    KJ660346        18911        G        A        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972956.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972956.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        800        C        T        PASS        NP<br>    KJ660346        1365        T        C        NO PASS        NP<br>    KJ660346        1395        A        C        NO PASS        NP<br>    KJ660346        1849        T        C        PASS        NP<br>    KJ660346        2620        T        A        NO PASS        NP<br>    KJ660346        2627        T        A        NO PASS        NP<br>    KJ660346        2632        T        C        NO PASS        NP<br>    KJ660346        2688        G        T        NO PASS        NP<br>    KJ660346        3565        T        A        NO PASS        VP35<br>    KJ660346        4489        T        G        NO PASS        VP40<br>    KJ660346        4895        T        C        NO PASS        VP40<br>    KJ660346        5482        A        C        NO PASS        GP<br>    KJ660346        5509        T        A        NO PASS        GP<br>    KJ660346        5603        C        T        PASS        GP<br>    KJ660346        5652        A        G        NO PASS        GP<br>    KJ660346        6283        C        T        PASS        GP<br>    KJ660346        7526        A        G        NO PASS        GP<br>    KJ660346        7530        G        A        NO PASS        GP<br>    KJ660346        7542        A        G        NO PASS        GP<br>    KJ660346        7882        G        T        NO PASS        GP<br>    KJ660346        8928        A        C        PASS        VP30<br>    KJ660346        9305        C        A        NO PASS        VP30<br>    KJ660346        10218        G        A        NO PASS        VP24<br>    KJ660346        10729        A        G        NO PASS        VP24<br>    KJ660346        12655        G        T        NO PASS        L<br>    KJ660346        12660        G        C        NO PASS        L<br>    KJ660346        12887        C        T        NO PASS        L<br>    KJ660346        13916        T        G        NO PASS        L<br>    KJ660346        14019        T        C        NO PASS        L<br>    KJ660346        14045        T        A        NO PASS        L<br>    KJ660346        14274        A        G        NO PASS        L<br>    KJ660346        14550        G        C        NO PASS        L<br>    KJ660346        14675        C        A        NO PASS        L<br>    KJ660346        15647        T        A        NO PASS        L<br>    KJ660346        15963        G        A        NO PASS        L<br>    KJ660346        16231        T        C        NO PASS        L<br>    KJ660346        17928        T        G        NO PASS        L<br>    KJ660346        18911        G        A        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972957.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972957.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        2533        T        G        NO PASS        NP<br>    KJ660346        3008        TTTT        CTTC        PASS        VP35<br>    KJ660346        3222        C        A        NO PASS        VP35<br>    KJ660346        3689        T        C        NO PASS        VP35<br>    KJ660346        3816        G        A        NO PASS        VP35<br>    KJ660346        3862        T        G        NO PASS        VP35<br>    KJ660346        7824        G        C        NO PASS        GP<br>    KJ660346        8371        A        G        PASS        VP30<br>    KJ660346        8811        C        T        NO PASS        VP30<br>    KJ660346        9319        G        A        NO PASS        VP30<br>    KJ660346        9991        C        A        NO PASS        VP24<br>    KJ660346        11851        C        A        NO PASS        L<br>    KJ660346        14411        A        G        NO PASS        L<br>    KJ660346        14430        A        C        NO PASS        L<br>    KJ660346        15963        G        A        PASS        L<br>    KJ660346        16188        T        G        NO PASS        L<br>    KJ660346        16757        C        G        NO PASS        L<br>    KJ660346        17142        T        C        NO PASS        L<br>    KJ660346        18911        G        A        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972958.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972958.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        3385        T        G        NO PASS        VP35<br>    KJ660346        4211        A        G        NO PASS        VP40<br>    KJ660346        5491        G        A        NO PASS        GP<br>    KJ660346        5509        T        C        NO PASS        GP<br>    KJ660346        5515        C        A        NO PASS        GP<br>    KJ660346        6372        G        T        NO PASS        GP<br>    KJ660346        9137        T        A        NO PASS        VP30<br>    KJ660346        14019        T        C        NO PASS        L<br>    KJ660346        15710        A        C        NO PASS        L<br>    KJ660346        15801        C        T        NO PASS        L<br>    KJ660346        15963        G        A        NO PASS        L<br>    KJ660346        17142        T        C        NO PASS        L<br>    KJ660346        17329        C        A        NO PASS        L<br>    KJ660346        17534        G        T        NO PASS        L<br>    KJ660346        17704        A        C        NO PASS        L<br>    KJ660346        17758        G        T        NO PASS        L<br>    KJ660346        17807        A        C        NO PASS        L<br>    KJ660346        18081        T        C        PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972959.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972959.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        6283        C        T        NO PASS        GP<br>    KJ660346        9303        T        C        NO PASS        VP30<br>    KJ660346        9619        T        C        PASS        VP24<br>    KJ660346        10218        G        A        PASS        VP24<br>    KJ660346        15963        G        A        PASS        L<br>    KJ660346        17142        T        C        NO PASS        L<br>    KJ660346        18627        T        C        NO PASS        L<br>    KJ660346        18795        A        G        PASS        L<br>    KJ660346        18911        G        A        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972960.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972960.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        17142        T        C        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972961.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972961.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        3884        T        G        NO PASS        VP35<br>    KJ660346        9253        T        G        NO PASS        VP30<br>    KJ660346        9727        A        C        NO PASS        VP24<br>    KJ660346        12338        T        C        NO PASS        L<br>    KJ660346        12885        A        C        NO PASS        L<br>    KJ660346        15101        T        C        NO PASS        L<br>    KJ660346        18112        G        T        NO PASS        L<br>    KJ660346        18254        T        A        NO PASS        L<br>    KJ660346        18625        T        C        NO PASS        L<br>    KJ660346        18848        A        C,T        NO PASS        L<br><br>Below is a description of a data file that is available for testing.<br><br>File name: SRR1972962.vcf<br><br>File contents:<br><br>    ##fileformat=VCFv4.2<br>    ##fileDate=20170411<br>    ##source=freeBayes v1.1.0-3-g961e5f3<br>    ##reference=refs/KJ660346.fa<br>    ##contig=&lt;ID=KJ660346,length=18959&gt;<br>    ##phasing=none<br>    ##commandline=&quot;freebayes/bin/freebayes -p 1 -f refs/KJ660346.fa bam/SRR1972962.bam&quot;<br>    #CHROM        POS        REF        ALT        FILTER        GENE<br>    KJ660346        629        C        A        NO PASS        NP<br>    KJ660346        800        C        T        NO PASS        NP<br>    KJ660346        1202        T        C        PASS        NP<br>    KJ660346        1849        T        C        PASS        NP<br>    KJ660346        2072        A        C        NO PASS        NP<br>    KJ660346        2141        A        C        NO PASS        NP<br>    KJ660346        2163        T        A        NO PASS        NP<br>    KJ660346        2719        C        T        NO PASS        VP35<br>    KJ660346        3008        TTTT        CTTC        NO PASS        VP35<br>    KJ660346        4895        T        C        NO PASS        VP40<br>    KJ660346        5051        G        A        NO PASS        VP40<br>    KJ660346        5276        T        C        PASS        VP40<br>    KJ660346        5753        T        C        NO PASS        GP<br>    KJ660346        5899        C        A        NO</td><td class="s1" dir="ltr">The data were generated as part of this study: <span style="text-decoration:underline;-webkit-text-decoration-skip:none;text-decoration-skip-ink:none;color:#1155cc;"><a target="_blank" href="https://pubmed.ncbi.nlm.nih.gov/25214632.">https://pubmed.ncbi.nlm.nih.gov/25214632.</a></span></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import glob<br><br>def main():<br>    # get the argument from the command line<br>    file_pattern = sys.argv[1]<br><br>    # find all the VCF files using the file pattern<br>    file_names = glob.glob(file_pattern)<br><br>    # initialize the variables to keep track of the file with the largest number of &quot;PASS&quot; variants<br>    max_pass_count = 0<br>    max_pass_file = &quot;&quot;<br><br>    # loop through each file<br>    for file_name in file_names:<br>        pass_count = 0<br>        with open(file_name, &quot;r&quot;) as f:<br>            # loop through each line in the file<br>            for line in f:<br>                # skip the header lines<br>                if line.startswith(&quot;#&quot;):<br>                    continue<br><br>                # split the line into columns<br>                fields = line.strip().split()<br>                # increment the count if the FILTER value is &quot;PASS&quot;<br>                if fields[4] == &quot;PASS&quot;:<br>                    pass_count += 1<br><br>        # check if the current file has more &quot;PASS&quot; variants than the previous files<br>        if pass_count &gt; max_pass_count:<br>            max_pass_count = pass_count<br>            max_pass_file = file_name<br><br>    # print the name of the file with the largest number of &quot;PASS&quot; variants<br>    print(max_pass_file)<br><br>if __name__ == &quot;__main__&quot;:<br>    main()</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">3</td><td class="s1" dir="ltr">This exercise uses a LOT of example data files. Thus the length of the full prompt exceeded ChatGPT&#39;s limit. I shortened it to include only three data files.<br><br>ChatGPT&#39;s first solution did not work. It displayed an empty string for the max file. I explained this and asked ChatGPT to try again. It gave me some code that did not address the prompt. Then I gave it the original prompt (this time with 5 example files), and it passed. I&#39;m not sure if having more files made a difference or if it merely passed due to an additional iteration.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td></tr><tr style="height: 20px"><th id="2089702959R150" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">151</div></th><td class="s1">16 - Python Scripts &amp; &quot;while&quot; Loops</td><td class="s1">09 - Predict species recovery</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>The goal of many conservation biologists is to save endangered species from extinction. They hope that as numbers of the species increase, genetic diversity will increase, and the species will be less susceptible to genetic drift.<br><br>Suppose you are a conservation biologist, and you want to predict how long it will take for various species to reach a point of recovery.<br><br>Please create a Python script that uses &quot;sys.argv&quot; to accept two arguments. The first argument will be an integer value (you will need to convert it from a string to an integer). The second argument will also be an integer value. The first value indicates how many of a given species are known to exist currently. The second value is an estimate of how many species are needed for the species to be endangered no longer (the &quot;target&quot; value). Assume that the number of species will double each year due to some remarkable conservation efforts.<br><br>Please use a &quot;while&quot; loop to calculate how many years it will be until the species has recovered. Print this number.<br><br>Think about this only in whole years. For example, suppose that there are currently 100 of a species, and the target is 550. After year 1, the number would be 200. After year 2, the number would be 400. After year 3, the number would be 800. In this case, the answer would be 3.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;100&quot; &quot;550&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;350&quot; &quot;15000&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;745&quot; &quot;2700000&quot;</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br><br>current_population = int(sys.argv[1])<br>target_population = int(sys.argv[2])<br><br>years = 0<br>while current_population &lt; target_population:<br>    current_population *= 2<br>    years += 1<br><br>print(years)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">On the first try, it did the calculations correctly but printed too much text in the output. I asked it to modify the solution, and it did so successfully.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">No</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R151" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">152</div></th><td class="s1">16 - Python Scripts &amp; &quot;while&quot; Loops</td><td class="s1">10 - Reverse complement DNA</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>In many organisms, the genome consists of double-stranded DNA. Two DNA strands are bound together by base-pairing complementary nucleotides (A and T nucleotides pair with each other; G and C nucleotides pair with each other). One strand is called the &quot;sense&quot; strand and the other is called the &quot;antisense&quot; strand. Both strands have a 5&#39; (pronounced five prime) end and a 3&#39; end. Transcription starts at the 5&#39; end and moves toward the 3&#39; end. For the sense strand, if you were reading from left to right, the 5&#39; end would be on the left side. For the antisense strand, the 5&#39; end would be on the right side. Below is an example for a small part of the genome:<br><br>    5&#39; AATGTGCCGATG 3&#39; (sense strand)<br>       ||||||||||||<br>    3&#39; TTACACGGCTAC 5&#39; (antisense strand)<br><br>When storing information about DNA sequences in a FASTA file, the standard approach is to store the sense strand only (it would be redundant to store both). For example, if you were storing the above sequence, your FASTA file would look something like this:<br><br>    &gt;Sequence1<br>    AATGTGCCGATG<br><br>For some research applications, it is helpful to create a FASTA file that contains the &quot;reverse complement&quot; of the sense strand. This would be the antisense strand, but the 5&#39; end would be on the left side, and the 3&#39; end would be on the right side. For example, if you reverse-complemented the above FASTA file, it would look something like this:<br><br>    &gt;Sequence1_ReverseComplement<br>    CATCGGCACATT<br><br>Please note that the output contains the same antisense strand that you saw in the first example above (TTACACGGCTAC). However, the sequence is in reverse order.<br><br>Please write a Python script that accepts two arguments. The first argument will be the name of a FASTA file that contains DNA sequences. The second argument will be the name of a FASTA file that your function will need to create. Your function should 1) read the specified input file, 2) perform a reverse complement of each DNA sequence in the input file, and 3) write the headers and reverse-complemented DNA to the output file. Your function should also append &quot;_ReverseComplement&quot; to the end of each header line.<br><br>For simplicity, the FASTA files will contain only one line of DNA sequence per header line. However, the lengths of the sequences will vary.<br><br>Hint(s):<br><br>     The &quot;replace()&quot; and &quot;re.sub()&quot; functions are not your friends on this problem.<br>     There is no &quot;reverse()&quot; function for strings in Python.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;DNA1.fasta&quot; &quot;ReverseComplement1.fasta&quot;<br>    <br>    cat &quot;ReverseComplement1.fasta&quot;<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;DNA2.fasta&quot; &quot;ReverseComplement2.fasta&quot;<br>    <br>    cat &quot;ReverseComplement2.fasta&quot;<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA1.fasta<br><br>File contents:<br><br>    &gt;Sequence1<br>    ACTAGCAGACCGACGACGAGACGACGAG<br>    &gt;Sequence2<br>    AACCGGAACGAGAGACGGGTTAGCAGATAGACGAGATCGAGA<br>    &gt;Sequence3<br>    ACCGGAGACTTTAGCGAGACGAGTCGAGACGATGAGA<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA2.fasta<br><br>File contents:<br><br>    &gt;Sequence1<br>    AATGCCTGCATCCGACGTCTCTGTAAATTT<br>    &gt;Sequence2<br>    TACTGAGCCGAGTGCGACCGTGCTCTTAGAACGAA<br>    &gt;Sequence3<br>    GAAGGAGTTCAACGAACCCGCTACC<br>    &gt;Sequence4<br>    ATATGGCACCGCAGGTTGCC</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br><br>def reverse_complement(dna):<br>    complement = {&#39;A&#39;: &#39;T&#39;, &#39;T&#39;: &#39;A&#39;, &#39;C&#39;: &#39;G&#39;, &#39;G&#39;: &#39;C&#39;}<br>    return &quot;&quot;.join(complement[base] for base in reversed(dna))<br><br>def main(input_file, output_file):<br>    with open(input_file, &#39;r&#39;) as f:<br>        lines = f.readlines()<br><br>    with open(output_file, &#39;w&#39;) as f:<br>        dna = &quot;&quot;<br>        header = &quot;&quot;<br>        for line in lines:<br>            if line[0] == &quot;&gt;&quot;:<br>                if dna != &quot;&quot;:<br>                    f.write(header + reverse_complement(dna) + &#39;\n&#39;)<br>                header = line.strip() + &quot;_ReverseComplement\n&quot;<br>                dna = &quot;&quot;<br>            else:<br>                dna = line.strip()<br>        if dna != &quot;&quot;:<br>            f.write(header + reverse_complement(dna) + &#39;\n&#39;)<br><br>if __name__ == &#39;__main__&#39;:<br>    main(sys.argv[1], sys.argv[2])</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">I suspect the model has seen this type of problem before.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R152" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">153</div></th><td class="s1">16 - Python Scripts &amp; &quot;while&quot; Loops</td><td class="s1">11 - Classify protein sequences</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Suppose you are given tab-delimited text files that contain information about viral proteins. These text files contain two columns of data. The first column contains protein identifiers (for example, &quot;001R&quot; or &quot;VP31&quot;). The second column contains an amino-acid sequence for each protein. The files contain no header lines.<br><br>Your task is to classify the proteins according to size. Proteins with fewer than 500 amino acids should be classified as &quot;short&quot;. Proteins with 500-999 amino acids should be classified as &quot;medium&quot;. Proteins with 1000+ amino acids should be classified as &quot;long&quot;.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept two arguments. The first argument will be a string and will be the name of an input file. The second argument will be a string and will be the name of a text file that your code should create. Your code should loop through the specified input file, calculate the length of each protein sequence, and write the protein identifier and length classification to the output file (in the same order as the input file). Below is an example that shows how the first few lines of the output file should look.<br><br>        001R        short<br>        002L        short<br>        003R        medium<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;Uniprot_Protein_Sequences_FRG3G.tsv&quot; &quot;FRG3G_Length_Classifications.tsv&quot;<br>    <br>    # This part is necessary to validate the contents of the output files<br>    echo FRG3G_Length_Classifications.tsv<br>    cat FRG3G_Length_Classifications.tsv<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;Uniprot_Protein_Sequences_IIV3.tsv&quot; &quot;IIV3_Length_Classifications.tsv&quot;<br>    <br>    # This part is necessary to validate the contents of the output files<br>    echo IIV3_Length_Classifications.tsv<br>    cat IIV3_Length_Classifications.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Uniprot_Protein_Sequences_IIV3.tsv<br><br>File contents:<br><br>    002R        MAFSAEDVLKEYDRRRRMEALLLSLYYPNDRKLLDYKEWSPPRVQVECPKAPVEWNNPPSEKGLIVGHFSGIKYKGEKAQASEVDVNKMCCWVSKFKDAMRRYQGIQTCKIPGKVLSDLDAKIKAYNLTVEGVEGFVRYSRVTKQHVAAFLKELRHSKQYENVNLIHYILTDKRVDIQHLEKDLVKDFKALVESAHRMRQGHMINVKYILYQLLKKHGHGPDGPDILTVKTGSKGVLYDDSFRKIYTDLGWKFTPLMSIIGATRLQNDKSDTYSAGPCYAGGCSAFTPRGTCGKDWDLGEQTCASGFCTSQPLCARIKKTQVCGLRYSSKGKDPLVSAEWDSRGAPYVRCTYDADLIDTQAQVDQFVSMFGESPSLAERYCMRGVKNTAGELVSRVSSDADPAGGWCRKWYSAHRGPDQDAALGSFCIKNPGAADCKCINRASDPVYQKVKTLHAYPDQCWYVPCAADVGELKMGTQRDTPTNCPTQVCQIVFNMLDDGSVTMDDVKNTINCDFSKYVPPPPPPKPTPPTPPTPPTPPTPPTPPTPPTPRPVHNRKVMFFVAGAVLVAILISTVRW<br>    003L        MASNTVSAQGGSNRPVRDFSNIQDVAQFLLFDPIWNEQPGSIVPWKMNREQALAERYPELQTSEPSEDYSGPVESLELLPLEIKLDIMQYLSWEQISWCKHPWLWTRWYKDNVVRVSAITFEDFQREYAFPEKIQEIHFTDTRAEEIKAILETTPNVTRLVIRRIDDMNYNTHGDLGLDDLEFLTHLMVEDACGFTDFWAPSLTHLTIKNLDMHPRWFGPVMDGIKSMQSTLKYLYIFETYGVNKPFVQWCTDNIETFYCTNSYRYENVPRPIYVWVLFQEDEWHGYRVEDNKFHRRYMYSTILHKRDTDWVENNPLKTPAQVEMYKFLLRISQLNRDGTGYESDSDPENEHFDDESFSSGEEDSSDEDDPTWAPDSDDSDWETETEEEPSVAARILEKGKLTITNLMKSLGFKPKPKKIQSIDRYFCSLDSNYNSEDEDFEYDSDSEDDDSDSEDDC<br>    005L        MYQAINPCPQSWYGSPQLEREIVCKMSGAPHYPNYYPVHPNALGGAWFDTSLNARSLTTTPSLTTCTPPSLAACTPPTSLGMVDSPPHINPPRRIGTLCFDFGSAKSPQRCECVASDRPSTTSNTAPDTYRLLITNSKTRKNNYGTCRLEPLTYGIMARPLLGKTSSVRRRLESLSACSIFFFLRKFCQKMASLVFLNSPVYQMSNILLTERRQVDRAMGGSDDDGVMVVALSPSDFKTVLGSALLAVERDMVHVVPKYLQTPGILHDMLVLLTPIFGEALSVDMSGATDVMVQQIATAGFVDVDPLHSSVSWKDNVSCPVALLAVSNAVRTMMGQPCQVTLIIDVGTQNILRDLVNLPVEMSGDLQVMAYTKDPLGKVPAVGVSVFDSGSVQKGDAHSVGAPDGLVSFHTHPVSSAVELNYHAGWPSNVDMSSLLTMKNLMHVVVAEEGLWTMARTLSMQRLTKVLTDAEKDVMRAAAFNLFLPLNELRVMGTKDSNNKSLKTYFEVFETFTIGALMKHSGVTPTAFVDRRWLDNTIYHMGFIPWGRDMRFVVEYDLDGTNPFLNTVPTLMSVKRKAKIQEMFDNMVSRMVTSMNAKYDTDQGVGRMLFLGTIGLAVVVGGLMAYGYYYDGKTPSSGTSFHTASPSFSSRYRY<br>    007R        MRYTVLIALQGALLLLLLIDDGQGQSPYPYPGMPCNSSRQCGLGTCVHSRCAHCSSDGTLCSPEDPTMVWPCCPESSCQLVVGLPSLVNHYNCLPNQCTDSSQCPGGFGCMTRRSKCELCKADGEACNSPYLDWRKDKECCSGYCHTEARGLEGVCIDPKKIFCTPKNPWQLAPYPPSYHQPTTLRPPTSLYDSWLMSGFLVKSTTAPSTQEEEDDYMQNPLPEVMSPEHDKRTTTPMSKEANKFIRELDKKPGDLAVVSDFVKRNTGKRLPIGKRSNLYVRICDLSGTIYMGETFILESWEELYLPEPTKMEVLGTLESCCGIPPFPEWIVMVGEDQCVYAYGDEEILLFAYSVKQLVEEGIQETGISYKYPDDISDVDEEVLQQDEEIQKIRKKTREFVDKDAQEFQDFLNSLDASLLSMDSLNEVCYEQIKGTFYKGLFGDFPLIVDKKTGCFNATKLCVLGGKRFVDWNKTLRSKKLIQYYETRCDIKTESLLYEIKGDNNDEITKQITGTYLPKEFILDIASWISVEFYDKCNNIIINYFVNEYKTMDKKTLQSKINEVEEKMQKLLNEKEEELQEKNDKIDELILFSKRMEEDRKKDREMMIKQEKMLRELGIHLEDVSSQNNELIEKVDEQVEQNAVLNFKIDNIQNKLEIAVEDRAPQPKQNLKRERFILLKRNDDYYPYYTIRAQDINARSALKRQKNLYNEVSVLLDLTCHPNSKTLYVRVKDELKQKGVVFNLCKVSISNSKINEEELIKAMETINDEKRDVMYKMYFLKDQKFSLSGTIRINDKTQSEYGSVWCPGLSITGLHHDAIDHNMFEEMETEIIEYLGPWVQAEYRRIKGMRSIKPLRCCNAHGRHVSQEYGRCTLLLFREKLFLQTGLVCNKQCNAPNNDGAESKHHGIHHGSRGALALRGAGVHLLASAALGPRVLAGLVPTGRSVQGSVGQCGRVAQIGRARDVAARKQESYCEK<br>    008L        MEAKNITIDNTTYNFFKFYNINQPLTNLKYLNSERLCFSNAVMGKIVDDASTITITYHRVYFGISGPKPRQVADLGEYYDVNELLNYDTYTKTQEFAQKYNSLVKPTIDAKNWSGNELVLLVGNEWYCKTFGKAGSKNVFLYNMIPTIYRDEPQHQEQILKKFMFFNATKNVEQNPNFLDNVPEEYYHLLLPKSWVEKNLSDKYRKIMETEHKPLVFSCEPAFSFGLCRNTQDKNESYQLSLCLYEREKPRDAEIVWAAKYDELAAMVRDYLKKTPEFKKYRSFISCMKGLSWKNNEIGDKDGPKLYPKVIFNRKKGEFVTIFTKDDDVEPETIEDPRTILDRRCVVQAALRLESVFVHNKVAIQLRINDVLISEWKEASSKPQPLILRRHRFTKPSSSVAKSTSPSLRNSGSDESDLNQSDSDKEDERVVPVPKTKRIVKTVKLPN<br>    011L        MSFKVYDPIAELIATQFPTSNPDLQIINNDVLVVSPHKITLPMGPQNAGDVTNKAYVDQAVMSAAVPVASSTTVGTIQMAGDLEGSSGTNPIIAANKITLNKLQKIGPKMVIGNPNSDWNNTQEIELDSSFRIVDNRLNAGIVPISSTDPNKSNTVIPAPQQNGLFYLDSSGRVWVWAEHYYKCITPSRYISKWMGVGDFQELTVGQSVMWDSGRPSIETVSTQGLEVEWISSTNFTLSSLYLIPIVVKVTICIPLLGQPDQMAKFVLYSVSSAQQPRTGIVLTTDSSRSSAPIVSEYITVNWFEPKSYSVQLKEVNSDSGTTVTICSDKWLANPFLDCWITIEEVGMDTSPYDFLKLYPWLSRGEADKGTLLDAFPGETFEQSLASDVAMRRAVQDDPAFGHQKLVETFLSEDTPYRELLLFHAPGTGKTCTVVSVAERAKEKGLTRGCIVLARGAALLRNFLHELVFNCGTGGRYIPEGYADMGDQERTRKMRKAVSSYYQFRTYETFAKSVATMSAEAIRARYDRFVIVMDEVHHLRSVQAEGVNTYSAISRFLRTVRGCVKMLLTGTPMTNEPGELADVLNLILPQDKTIRPEDGIFSNSGDLLKPDELAERVRGRVSYLKAARPDAGLTFAGEVLGGTGMTHLRLVRLEMSAFQSDAYASAWDQDAGDRNIFSNSRQCSLAVMPDRRWGSAAEARNPSQVRRMAGQNLAEYSVKYDYLVRVASSSPKTFAYCEYVNGSGLSLLSDILLANGWRRATGRETTPGKRFALLTASQKNIHKIVQRFNHEDNVDGAYISLLLGSRVVAEGLTFKEVRHTVILTPHWNYTETAQAIARSWRAGSHDRLKARGEAVAVTVHRLVAVPRGRDTPRSIDSDMYAVSEVKDKRIKAVERILMTSAADCSLLRSRNLYPSEFDGSRECEYGRCAYRCSNVSVEPGPLPALLGASAAEAVAQVRLDGGGDPAIMKVDMSTLWAEVTAGRRYVNRWGDGAVLRAEGGRLELSAPYGSSEEGRWGDFYKTRNLCYAKMDQDHLRADDLRDSLPQEVEELLTVSPVETIGETASAMPQEVATAILMACVQARADGKTLNVVRRDALLDFYKGFYAMGPSGWTVWLHARGANAKVYDGRRWNPADEDTLEFLAARSAKFTDTRIGYYGLYNPNLKDFCIRDVTQGKRDKVDLRKLTVGRRCVDWDQRTLVHIVARLMKIDGRRDFMPHATLREMRELAEQDPLHEPSDLTSKEACRRFLFWTQKGDNKFRRQDICKAMEKWFIENDLMEDNFDCGHQHKRRGKFAMIKLFCVLAAFISINSACQSSHQQREEFTVATYHSSSICTTYCYSNCVVASQHKGLNVESYTCDKPDPYGRETVCKCTLIKCHDIMKMDTDCRHWIVLASVPVLTVLAFKGEGALALAGLLVMAAVAMYRDRTEKKYSAARAPSPIAGHKTAYVTDPSAFAAGTVPVYPAPSNMGSDRFEGWVGGVLTGVGSSHLDHRKFAERQLVDRREKMVGYGWTKSFF<br>    013L        MMESPKYKKSTCSVTNLGGTCILPQKGATAPKAKDVSPELLVNKMDNLCQDWARTRNEYNKVHIEQAPTDSYFGVVHSHTPKKKYTSRDSDSEPEATSTRRSATAQRAANLKSSPVDQWSTTPPQPQPQPAAPTVKKTCASSPPAALSVKRTCTSPPPPPVLIDDDTGEDAFYDTNDPDIFYDIENGVSELETEGPKRPVYYQRNIRYPIDGSVPQESEQWYDPIDDEFLASSGDVVSLEPSPIAAFQPTPPKTVQFVPMPEEIIVPPPPPPKTVVDEGVQAMPYTVDQMIQTDFEESPLLANVNLRTIPIEEVNPNFSPVLMQDMVRDSFVFGTVAQRVMASQRVKQFFKELIEQDVSLAGRMCMDSGSPQLNLYNSLMGVKLLYRWRSSTTFYRAIVPEIDEPVQVMQDVLSSSEWAKFDSQAGIPPKMVYIHYKLLNDLVKTLICPNFQLTHAALVCVDCRPEAVGSDGLQDGRQRRCSNLVSEYHEMTLEDLFNTIKPADLNAKNIILSVLFQMLYAVATVQKQFGMGGLFANADSVHVRRIQPGGFWHYTVNGLRYSVPNYGYLVILTNFTDVVNYRPDFATTRYFGRRQAKVVPTRNWYKFVPFTTRYRPFVTVDPITQAKTTAYAPNPPTEGITINEFYKDSSDLRPSVPVDLNDMITFPVPEFHLTICRLFSFFSKFYDSNFIGNDPFVRNLVDRYSQPFEFPDVYWPEDGVSRVLACYTIEEIYPNWVDGDTDYVIESYNLDMTSVKTIAMLAMLVIVAALIYMGYRTFTSMQSKLNELESRVNAPQLRPPVMSPIVPLNFIESEDLDKELDMCAKLVEMAFGPVNADSPPLTAEEKESAVEKLVGSKPFPALKKKYHDKVPAQDPKYCLFSFVEVLPSCDIKAAGAEEMCSCCIKRRRGQVFGVACVRGTAHTLAKAKQKADKLVGDYDSVHVVQTCHVGRPFPLVSSGMAQETVAPSAMEAAEAAMDAKSAEKRKERMRQKLEMRKREQEIKARNRKLLEDPSCDPDAEEETDLERYATLRVKTTCLLENAKNASAQIKEYLASMRKSAEAVVAMEAADPTLVENYPGLIRDSRAKMGVSKQDTEAFLKMSSFDCLTAASELETMGF<br>    022L        MYYRDQYGNVKYAPEGMGPHHAASSSHHSAQHHHMTKENFSMDDVHSWFEKYKMWFLYALILALIFGVFMWWSKYNHDKKRSLNTASIFYMANSVAFSSMTWYSPLASDNLYDICVDKVHNRVLCLCHSFGCCTNAVVIWILPSFDEFTPQTLSCKGPMETLVQAYLDIQGKIAEFRREIKALRVEEKAITANLFEAMGEAGVESIRISEDRYLVAEEKPKRTRSKQQFYQAAEGEGFTQEDVDRLMSLSRGAVTGSSSNVKIRKSAPARNEEDDDGMEQVPIKEMRLSDLRPNNKSIDTDLGGTKLVVIGKPGSGKSTLIKALLDSKRHIIPCAVVISGSEEANGFYKGVVPDLFIYHQFSPSIIDRIHRRQVKAKAEMGSKKSWLLVVIDDCMDNAKMFNDKEVRALFKNGRHWNVLVVIANQYVMDLTPDLRSSVDGVFLFRENNVTYRDKTYANFASVVPKKLYPTVMETVCQNYRCMFIDNTKATDNWHDSVFWYKAPYSKSAVAPFGARSYWKYACSKTGEEMPAVFDNVKILGDLLLKELPEAGEALVTYGGKDGPSDNEDGPSDDEDGPSDDEEGLSKDGVSEYYQSDLDDMETMSDYSKEVSEALSALRGELSALSAAISNTVRAGSYSAPVAKDCKAGHCDSKAVLKSLSRSARDLDSAVEAVSSNCEWASSGYGKQIARALRDDAVRVKREVESTRDAVDVVTPSCCVQGLAEEAGKLSEMAAVYRCMATVFETADSHGVREMLAKVDGLKQTMSGFKRLLGKTAEIDGLSDSVIRLGRSIGEVLPATEGKAMRDLVKQCERLNGLVVDGSRKVEEQCSKLRDMASQSYVVADLASQYDVLGGKAQEALSASDALEQAAAVALRAKAAADAVAKSLDSLDVKKLDRLLEQASAVSGLLAKKNDLDAVVTSLAGLEALVAKKDELYKICAAVNSVDKSKLELLNVKPDRLKSLTEQTVVVSQMTTALATFNEDKLDSVLGKYMQMHRFLGMATQLKLMSDSLAEFQPAKMAQMAAAASQLKDFLTDQTVSRLEKVSAAVDATDVTKYASAFSDGGMVSDMTKAYETVKAFAAVVNSLDSKKLKLVAECAKKMQNSKTDMCAALWAVTGLVLNVAVRFALEPFKESMGQGWHTAARVAVNGAIVLALADRLSDSPVTMTLFVMALSASPEMATNYCDEFERNPTRNPRTGRTIKRGGPVFRALERECSDGAARVFPAAAVRGAAAARAASPRVAAASPCPEFARDPTRNPRTGRPIKRGGPVFRALERECADYGGASPRRVSPARAFPNRRVSPARRQSPAEAAEASPCPEFARDPTRNPRTGRTIKRGGPTYRALEAECADYGRLSPIRSPWSDWSSTGLSPFRSHMRKSPARRSPARRSPARRSLARYTEHLTSDSETEVDYDARNVIRSQVGPGGVCERFAADPTRNPVTGSPLSRNDPLYTDLMEICKGYPDTPLTKSLTGEGTDDDTCEAFCRDPTRNPVTGQKMRRNGIEYQMFAEECDCSGISRPSGVSRTSGTSGSSGSSASSRPPNSFEAPGASSRPPNSFEASGAARVPGTPSVSRGEPRWMSSISTRHNYDESNPMSVAFRLRHVKDIRKFLRTVRPGRSGFCATDKGGWLGSAAVSDNVIGQGSWGSVHMVKFRDFPEEFVVKEAVLMSVSEKHRYKPTVVWDEWAAGSVPDEVVVNNMVTEIAATGMTPFVPLTAGAGACDSCNPQLLEKAAKVTKCYLQAMEAADFSLDRVLPTMSPDQAASALAQILLGLQSLQTTLGIMHNDIKAHNILVKRVPPGGYWKVTDSFNGQVFYIPNEGYLCMLADYGVVRLVKPAVGMDTLYGTRNARFVPRDVGRWGKGAGTEYVVTPIRSKISVVVRGGRFVGVEPNKAVRYWKNTDTSKVGDVITTNNVFYMGYDIEPDMQVQLDDTNSFPVWESRGDVADCVRTFVGGKRASQPGFHRLFYKKTGSAWEKAAETVAKQNPLFSGFTLDGSGLKYIRAATACAYIFPGMAVPRPGEREIESFTMMLQNYAIVLGMAVAVAIWYFFKIEEEAPPGPNPPKPDPPKPDPPKMHMPKKKPHWMDPHLTGSQTVQYSRNRSMGDPIRGDLPIIPRDDGWFSTAANPAHTLHAGALSMIAPASTGGGLTVNKLISAYADKGNAMSGRHNSPSYYGSSMETIVLVPRQDQETFSDSRPVLDGDLMLEFLENKIRHPVRRRQPRVVPVTSSDPEVVDDEDDEDQSDDSDEERQRLYFQYMVLKRMYPTEVIPEMTTYSNVAIMREKYKLLTRRLSLDKHINEWKKYIIVGMCIMELVMTKLNFDASGFARYQIKSLGAYDQLLAEMADKYYEATPQSSVEMRLMTTMGMNMAVFMLGKLLGGQMDFLGLLENAFGSSS<br>    023R        MSFVHKLPTFYTAGVGAIIGGLSLRFNGAKFLSDWYINKYNDSVPAWSLQTCHWAGIALYCVGWVTLASVIYLKHRDNSILKGSILSCIVISAVWSILEYNQDMFVSNPKLPLISCAMLVSSLAALVALKYHIKDIFTILGAAIIIILAEYVVLPYQRQYNIVDGIGLPLLLLGFFILYQVFSVPNPSTPTGVMVPKPEDEWDIEMAPLNHRDRQVPESELENVKMRVSQTSWIVSRMLEYPRGGFFYSTDMACMMEGLAEELAGGHKDEVLIVSGRNGDDEVFKEFPNVRAADGLKGPNSIDPETKLVLIIDVSPTAISNALAATLQEFLIPVWVFCNHTRTLTASVTRRLGYKLWPKGTYTPYICEKAGVSEVVTYNQPESEKFVAFMSAARQIMDKRKSKKTMQELAFLPHLAFAEIAMEGDQEMTPTLTAKKVSDIKDEQVNELASAMFRTGKLSHLDMLSVPDCVYSCGEALKREVAKAKANRERFVVALRNAQYKKYTAGLLEAGTPVKTFTEVIKNWGAYDTIFLPMGVDWTYTGGSNLIRMMMTPGSHKTVTFVPESDDVHEFCHNKPTVNTMGVESAATGLAAELNRRWRRDNPVDAS<br>    025R        MGSYMLFDSLIKLVENRNPLNHEQKLWLIDVINNTLNLEGKEKLYSLLIVHNKQQTKIYDPKEPFYDIEKIPVQLQLVWYEFTKMHLKSQNEDRRRKMSLYAGRSPMWQYLPILLMTMISQLEWTVAAVKRYPAGGFITGDKLSRVFEALPWRVAVVSDEPEKYEGFPILTEEDPAVFEDADCILFAVSDPKCVTGAMKSVFMASSKTAWVVYDGTETRATVRSWMRRLWRAETYVPLLTHRGFVTDVCVYSQPDSERYVSVMTATAHFYSNRLEVLEEMAFVPHLAYAKLAMGRYTVLDGCMSVKGSADVAPLNRSMWFLTAAAIPHGEIDTDSLFSDPGAVYSCGSALREALGSLPEGSTSVVAVRNSSYRKYVRGILGPNFRVETFTNVVKTWGVYDYVLLPMGISDSYKQGRDLMEKLEMPGGHRVVTFAPENYTVNEVHLNRPLKYAIKRMDLITPMVLRHVSLNK<br>    VF395        MLQNVIKELYDKGVLIFLDPNNYVTNNVAENPEIFANRDKFLEMYQDPIKIGWIFCYTPSLEEVLSTIKEDTCCLIEFGFNSDTEKKACETGRLLINHLVRHKFMAHWDENAIKNHKISTVITAKDLPESIQDLIVDYAPTTSTNPEQREMFIVKNLNFNFLNVISNFLPNKMLENVFIELYERGILLFLDVDSYCKNSDDADLNILKGRKEFLKYFDEPLEKGYVFTFNTDIHKLYKEIEEESCAVISYGYESDTDDKADDVGRNFVEILNKNGYITDWTEEVKDSKTIKIVIDENDIPNFKLNESPYLLNLTNDDLFENKNPLIQLIEDEPFPTTKNHNNDKRETNDKDDQQLKLNKMKEIEEKTIAFIDKNSLTKSQIEEYCTQQIINLDAIDTTDEIVKCRRKETIQNLQQVSLPTLKCSKCLKLYKQKKSYEKHIEKCN<br>    VF396        MNSRHVGQDSTTRDWYRDPKTIAWISDAVMEKTLGVARRPIRVSDKVILNVLDSFLQNNRPAIGHGCGVTNNPSYVTCDKYCPSLAPIETNPVVVRVVEFISQYIRDEYSFQEAGKQWSAWPSDRIQKHSQIKVRDRRPIPFQFNMNFMSGSGYAAFAPTEAYGDLFSENTFSFIAKQVMDITKNSHLGRPYLVPNPRIKEVMDSIWSNFRPNTGDIVTRYAINNENQYRYINYSADIIKQVIEVIVNDIMNYVDITKNNNTFSVWNTLSGANSRGLRHHAPIKINEKRANPLQFNMNF<br>    VF401        MSNFKFYDPLSELMSGQLGGGSITGDMTVVAPSKISQSEPPTDPTDLVNKQYVDDAISNGAFLPLVGGTLLGKVYQPMDPSAPTELVNKKYVDSVATGGGPVDVYLPLAGGTLTGPAKYASPPTTDDQLVNKQYVDGQVTALGSGPFLPLTGGALTGQVVQPSAPVAANNLVNKQYVDDTITTQISTVTSNQNSTFLPLAGGTVTGPINYASPPTANEQLANKQYVDGQVTALGSGPFLPLAGGTVTGPIVQTAPATAQNHLVTKDYVDGVATVSDATTAAKGVVQLAGDLSGVAAAPTIAAGVVTNAKLANLTGPSHLKGSNETSAAATDISLGPNLTIGGGGVLDVDSSKIPTIPVQVAQGGTGATTLTGYVKGNGTSPLTTVAKIPVQDVNGSVLTVNGSAPDANGNVATALSNVLTGLDANRPTVIPASGTMYVVSGDDPTVNGKSYISNGTNWLTINTASTSNDTRYVLKGGDTMGGDLSFPSTTKLTLDAAPVNATDAVNKQYVDTQMAGATVANATTTAPGIVQLAGDLSGTATVPKIANAVVSNQKLTPGTSGTLKGTDTTGAVADVTLGSGLTISGTTLSVDAASVPKAGSSQFGTVQFNATSGDLEPSGANSGIALVKSGAINNVKLANFSGPSRLKGSSSLSSTPDDISLGTGLTMTGTVLSVDQSTLTNVLPLSGGTMTGPIVQPADPTSDTQLANKKYVDTAIGSVGATKYLALTGGAVTGTIQSPAPTAASDLTNKEYVDNKVTALGSGPFLPLSGGVVTGVVSQAALPTANEDLANKQYVDQKVATVSGGTAASPATTTSLGTIQLGGDLAGVGSEASAPVITSAAITNNKLKPGGSGTLKGTDTAGAISDVILGPSMSMTNGNQLNAAISVLSGSNPNVTAPTDRPSTANVLYLGTDGGVWVWNGTSYVSPTGSSFKFIKSTTKYTIPLTTAPSTSLQLTDYNIAVLPGQTCRASYMIRYQTGDASYGVSFGFSGPVAGDFFFGHSYMLTTNPGIATSINAVSWAITNMTSLLGVTDTSTPSSNLGTTDDNQYATAMITVEFTNNGTASKTLVVLFNRDLSNYPNVTQRVVGGSVEYRMYMFKFNIDNPLHKLHKIQSKVSEINTEGGSGGGDGVDEIKTINPATETSLGVVQLSGVLKGTATNPELSFNCVKSNHLSEECVQYNHIQKLSKASLLGQTTKSGGNITEVELNSGLILENGKLSLDLPSIITLSDVMATKSELGVITLGDGGDLNGSPNVNLLENGKISNDPVIGLNKVTYPKIQKVTGKSLLGNSSIFDDNVEEIKIKNGLVLEDSSLSLDVNTLPQASSTTFGVIKLSGDLSGNASSPTIKNQSITSYKLAPLPKHTILGSLDNTLSPTPISLHKSLGVVNEQLKVNPKEIQSTTDSLGTIMLSGDLTGTATNPEISEKSITYSKMQSVNGTKVLLGNLSGTGNISEIEVGDSLILKNGFLDVDLDKLHEESILKVESGGTGNSVLPLGYLKGKEKDPVESVQKIPVTDIFPNVVGSVNGVFPSKPGGNISLIFSNVTTGILSEIPSRQVNGTIYIISGDPTPTNNGRTFISDGTNWNEVTNHLGSTDARYVQLAGSTMSEDASLVFPSTSNIILNQTSFSNKDAVTKEYVDEIKINDATVSSKGILELSGDFDSLSTADNPIIKSATPTVKGKIQLAGDFDPLSTSSLPLIKAASTTSRGKIQLSGDFDPLSTSTNPIIKSATVHSEGKIQLSGDFDSKSTSQFPLIKEASEVHFGKIKLAGDLSGNADAPEIAHNAITYSKIQQASQKSIIGNAINGATDIGEVVIGNGLMLSSGGSLPTVGTLSVDSNTLPKATLNSYGSIQLNGDFNPLSTATAPMIGEGKITYKKMQQISSQRKLLGSASLPIGIQPIQEISVGPSLTYSGAGNNILNASTSFFSGTNPNTSPPTDRPTTSNLLYIGVDGSVWMWNGVHYSKGPGALNVLKSKNLYTIGPLGTPSPLSMSDFDIKVSAGQRIKIKYILNFQSTGGAEYAPSFGWNGVAQEDFFQASIIGFFSSSNTSSSFSTIYQTNTSFTSSSRQVTGGANLFALTSTNADNNLLAWNVGTPEQQLGGIGGQAIPIYITAYYENNTNKETILEIVFNRDLKTQTGQTIQIAGGVADYVTY<br>    VF414        MAAKSKKILLQNLNSFVLDSILAIKDGEETPQSDASLVANLRKQWESSEFQKKMAVEVFGIKGKNLTPGPKRNKSAYMFFCQDMRQNIVADNPDCKPHQIMSLLGCKWRELTTKQKSQYYKQAADDKERYLDDKELEKRRNKTPSKLSSYFLFCEDERPIVKKEFPNMSTKKVTAECGKRWNEMKINDPKKYKYYVEKAAKMEDTTFLEGANLAGITTLMNNLHINEQANLEELEKQVMGKQQSFPTDHFDEELNGLAKSLGINFNDPEFSLDSPHSVISKKPSGRGRDKVHGGIRRDSVCTDSICSDSVCSGSIRSGSIRSGSIRNGSIRSGSVRDDSVRSGKTRRGLACNSSSRNDRGYSLSTHRKKYAESEASQKTAFSKRDRKNHYAESEYSEKSIKPSTKQVDRLINHLRSNGDPNSFYKKDHDYERKTKLVKLEKINMLLTYLGNEQISTDDIKIPTIDSSMQEIDDVIEMLTLRNVGIRYSSIAEEILIGLARGLEIVFDGTREIPFLNYRPDYTGLHNTFMIKLFKMRYETSQVVGNLVQNMSPLSKICLELGPSLLLYPALIRTKHKASEDLYNLLQKGPEDPFTAYNEIHETLKKNNKMEDTTFLEGANLAGITTLMNNLHINEQANLEELEKQVMGKQQNFPTDHFDEELNGLAKSLGINFNDPEFSLDSAHSVISKKPSRGGSDMISRDGLRHDGLRRDSVCTDSICSESVCSGSMRSGSMRHGSIRNGSVRDGSVRSGRTRRGPARSSSSRNDRIHSYSLSTHRKKYAESEASQKTVISKRDRKHHYAESEYSEKSIKPSTKQVDRLINHLRSNGDPNSFYKKEHDYDRKTKLVKLEKINMLLTYLGNEQISTDGIKIPTIDSSMQEIDDAIEMLTLRNVGIRYSSIAEEILIGLARGLEIVFDGTREIPFLNYRPDYTGLHNTFMIKLFKMRYETSQVVGNFVQNMSPLSKICLELGPSLLLYPALIRTKHKASEDLYNLLQKGPEDPFTAYNEIHETLKKNNKMEDTTFLEGANLAGITTLMNNLHINEQANLEELEKQVMGKQQSFPTDHFDEELNGLAKSLGINFNDPEFSLDSAHSVISKKPSGGGSIRGDLCRHSMRGDKISHEKIHSGIRRDSVCSDSVCSDSIRSGSVRGSSMHDGSTCDGSACDGSVRSGRTRRGPARSSSSRHDRIHGYSLSTHRKKYAESEASQKTVISKRDRKNHYAESEYSEKSIKPSTKQVDRLINHLRSNGDPNSFYKKEHDYDRKTKLVKLERINMLLTYLTNEQISTDGIKIPTIDSSMQEIDDAIEMLTLRNVGIRYSSIAEEILIGLARGLEIVFDGTREIPFLNYRPDYTGLHNTFMIKLFKMRYETSQVVGNLVQNMSPLSKICLELGPSLLLYPALIRTKHKASEDLYNLLQKGPEDPFTAYNEIHETLKKNNKMEDTTFLEGANLAGITTLMNNLHINEQANLEELEKQVMGKQQSFPTDHFDEELNGLAKSLGINFNDPEFSLDSAHSVISKKPSGRGSDKVHGGIRRDSVCTDSICSDSVCSGSIRSGSIRSGSIRDGSIRDGSIRSGNIRDGSVRSSKTRRGPARNSSSRNDRGYSLSTHRKKYAESEASQKTAISKRDRKNHYAESEYSEKSIKPSTKQVDRLINHLRSNGDPNSFYKKEHDYERKTKLVKLEKINMLLTYLGNEQISTDDIKIPTIDSSMQEIDDVIEMLTLRNVGIRYSSIAEEILIGLARGLEIVFDGTREIPFLNYRPDYTGLHNTFMIKLFKMRYETSQVVGNLVQNMSPLSKICLELGPSLLLYPALIRTKHKASEDLYNLLQKGPEDPFTAYNEIHETLKKNNKMEDTTFLEGANLAGITTLMNNLHINEQANLEELEKQVMGKQQSFPTDHFDEELNGLAKSLGINFNDPEFSLDAAHSVISKKPSGRGSDKVHGGIRRDSVCTDSICSDSVCSGSIRSGSIRSGSIRDGSIRDGSIRSGNIRDGSVRSSKTRRGPARNSSSRNDRGYSLSTHRKKYAESEASQKTAISKRDRKNHYAESEYSEKSIKPSTKQVDRLINHLRSNGDPNSFYKKEHDYERKTKLVKLEKINMLLTYLGNEQISTDDIKIPTIDSSMQEIDDVIEMLTLRNVGIRYSSIAEEILIGLARGLEIVFDGTREIPFLNYRPDYTGLHNTFMIKLFKMRYETSQVVGNLVQNMSPLSKICLELGPSLLLYPALIRTKHKASEDLYNLLQKGPEDPFTAYNEIHETLKKNNK<br>    VF415        MYFSTQIVNNKEKIMDTIFCNKGCCSLQTIKKKKPKVCDPNERYPFEKRKAGVFVECGSKFLLVQSYNDCWGIPKGHMEAYDHSPKTCAERELKEETGLEVKLTDAHLFRILLDNYYIYKISIPSVDQVDLSALPQLDSTGIGWVDMECAFDLNLTVLTQKILMALSCNMKVLKKFIYNYKLINLLKMNKIKCAKGCCILNFTNEIHHFKNKNISKNSHRESAGIAIIVDENMDKKILITQSYNNLWGVPKGKKESNETLLECASREVVEESGIKVDVSSLKSCEEIIFIPNYDKKLTIHIFKYFIPFVDYISCSMNNLCLKDLHDDSTGFGWINLKCLNEITKAKTIKLNSLTKYILRKI<br>    VF436        MSGGLLSLESALRTCKVNTELAARLQSHRTIGPASEKVCPVWNGLDAAGRQVCPDSFANKSPGCHSPMDRLHVENEISRPQYISYLSYGSQGVYGNEYGTIGGHRAAYPDNSIYGISGSFGQQLTSHVKPSMTECSCVRLGTQSQCVAGTSHNTGLSYKTLNGYYMSGGYTSLESSIRTCKVNTGNADRIESDRFLGFPDKKTCPPFLGTDLTGRAICPDSFMTKSAGCNTPEDRIYIENTVSRPHYYEYINLSPRGMLNDETIEGFGFGQQESAYIREGFGTNGTSAQYGSNGNGGNGSCGLNNYRAPVNLNGWPRAGYGIGPEDIDDLPGDLRFNATQPGPFNNMPLWAFSVVPREYGMPYSQLYGQNAEDQRLKQALWARAEIMGQDPIVPGPSRYGYRMMGDYMYTHVDVVGIAEASAALYVQKDRDRYLDVLTTIENFIYQHKCIITGESAHLLFLKKNIYLYEFYSNNVAEHSKALATLLYKLDPEYLTRYTVLITKIPNHWYVINVDQREFVRLYAIPAVKQHLPIPILPFYCTSALTHQELFCLGPELQLIQIYSKLCNPNFVEEWPTLLDYEKSMRMLFLEQFPRRLEIAGGKKEEEKHESIIKKIILEMVSTRQRIVVGGYIQKNLYNHVLKNRNRLQLITSLDIYEEKDIIQQFCDSNGLKIKIRINNPLLPTNPELRRLTIYFNHNNDDDQSYLIVDMYNTGSYELVPTNQINTLDGSFLIGTPFVQARFLLVEIWVLMLIAQQTKKDTKKIIQFFINQYEMLMNSPWPSMEALFPSSSKRYLGNYVDPNALIKWAQLKLKRIPPFYPGKPDEESCMYTHVDVVGIAEASAALYVQKDRDRYLDVLTAIENFIYQHKCIITGESAHLLFLKKSIYLYECYSNNVAEHSKALATLLYKLDPEYLTRYTVLITKVPNQSYVINVDQREFVRLYAIPAVKEHLPIPILPFHCTSALTQQELFCLGPELQLIQIYSKLCNPNFVEEWPTLLDYEKNMRTLFLEQFSQRLEMTGGKKEEEEKHKSIIKKIILEMVSTRQRIVVGGYIQKNLYNHVVKNRNRLQLITSLNIYEEKEIIQQFCDSNGLKIKIRINNPLLPTNPELRRLTIYFNNNDDQSYLIVDMYNTGSYELVPTNEVNTLDGSFLIGTPFVQARFLLVEIWVLMLIAQQTKKDTKKIIQFFINQYETLMNGPWPSMEALFPSSSKRYLGNYVDPNALIKWAQLKLKRIPLFYPGRPDEEEMYTHVDVVGIAEASAALYVQKDRDRYLDVLTTIENFIYQHKCILTGESAHLLFLKKSIYLYEFYSDNVAEHSKALATLLYNLDPEYLTRYTVLITKIPNHWYVINVDQREFVRLYAIPAVKQHLPIPILPFHCTSALTQQDLFCLGPELQLIQIYSKLCNPNFVEEWPTLLDYEKNMRTLFLEQFPEKLEKTGGKKEKEEKHESIIKKIILEMVSTHQRIVVGGYIQKNLYNHVIKNRNRLQLITSLNIYEEKDIIQQFCDSNGLKIKICVNNPLLPTNPELRRLTIYFNNNDDQSYLIVDMYNTGSYELVPTNQINTLDNSFLIGTPFVQARFLLVEIWVLMLIAQQTKKDTKKIIQFFINQYETLMNSPWPSIEALFPSSSKRYLGNYVDPNALIKWAQLKLKRIPPFYPGRPDEESCMYTHVDVVGIAEASAALYVQKDRDRYSDVLTTIENFIYQHKCIITGESAHLLFLKKNIYLYEFYSNNVAEHSKALATLLYKLDPEYLTRYTVLITKIPNHWYVINVDQREFVRLYAIPAVKQHLPIPILPFYCTSALTQQELFCLGPELQLIQIYSKLCNPNFVEEWPTLLDYEKSMRTLFLEQFPQRLEMTGGKKEEEEKHESIIKKIILEMVSTRQRIVVGGYIQKNLYNHVLKNRNRLQLITSLNIYEEKDIIQQFCDSNGLKIKIRINNPLLPTNPELRRLTIYFNNTNDDDQSYLIVDMYNTGSYELVPTNQINTLDGSFLIGTPFVQARFLLVEIWVLMLIAQQTKKDTKKIIQFFINQYEMLMNSPWPSMEALFPSSSKRYLGNYVDPNALIKWAQLKLKRIPPFYPGKPDEESCMYTHVDVVGIAEASAALYVQKDRDRYSDVLTTIENFIYQHKCIITGESAHLLFLKKNIYLYEFYSNNVAEHSKALATLLYKLDPEYLTRYTVLITKIPNHWYVINVDQREFVRLYAIPAVKQHLPIPILPFYCTSALTQQELFCLGPELQLIQIYSKLCNPNFVEEWPTLLDYEKSMRTLFLEQFPQRLEMTGGKKEEEEKHESIIKKIILEMVSTRQRIVVGGYIQKNLYNHVLKNRNRLQLITSLNIYEEKDIIQQFCDSNGLKIKIRINNPLLPTNPELRRLTIYFNNTNDDDQSYLIVDMYNTGSYELVPTNQINTLDGSFLIGTPFVQARFLLVEIWVLMLIAQQTKKDTKKIIQFFINQYEMLMNSPWPSMEALFPSSSKRYLGNYVDPNALIKWAQLKLKRIPPFYPGKPDEKSCMLWRNEITEFMDQLSKYSQEILKTFKQLRPSEYKQYNEFLTQVTPLLQKTSEKIPELVDHIFNYLDNVEKICELLVNASSIIISSKIREQVKHGMSFSYKADLDSLADILSQKQYVLMHLSKNIAAEYFNTCLNQGKSKLDLKAASVFYSSRPRTASSAELYRKMLYAYGSPQEINYYTEKARNKTLDVEESDSMAIIERTARHNLSLMHPLEAMGLTFGATNTDADPEDLKDKTVINLTLPQATESITYHLKSLMQLKKVSTASGLNTNILKAFDNIISTPVKKNKMASKLAPGMDVVFTSDNGKTFFTKNILSKNMLAGPKERVFAYNNLISNLNNSCFIQNHNDFLRQQDSWPFYDAHNFTNKFLMQPIFSGQTRPRLQGAMEAAHVETHLTAFLQSIQPSRPQDPSVLASPKLSALILNMLWRNEITEFMDQLSKYSQEILKTFKQLRPSEYKQYNEFLTQVTPLLQKTPEKIPEEVDHIFDYLDNVEKICELLVHASSIIISSKIREQVKHGMSFSYKTDLDSLAGILSQKQYVLIHLSKNIAAHYFNTCLNQGKSRLDLKAASVFYNSRPRTASSAELYRKMLYAYGSLQEINYYTEKARNKTLDVEESDSMATIERTARHNLSLMHPLEAMGLTFGATNTDADPEDLKDKTVINLTLPQATESVTYHLNSLMQLKKVSTTSGLNTNILKAFDNIISTPVKKNKMASKLAPGMDVVFTSDNGKTFFTKNVLSKNMLAGPKERVFAYNNLISNLNNSCFIQNHNDFLRQQDSWPFYDAHNFTNKFLMQPIFSGQTRPRLQGAMEAAHVETHLTAFLQSIQPSRPQDPSILASPKLSALILNMLWRNEITEFMDQLSKYSQEILKTFKQLRPSEYKQYNEFLTQVTSLLQKTPEKIPELVDHIFNYLDNVEKICELLVNASSIIISSKIREQVKHGMSFSYKADLDSLADVLSQKQYVLMNLSKNIAAQYFNTCLKQGKSRLDLKAASVFYNSRPRTASSAELYRKMLYAYGSLQEINYYTEKARNKTLDVEESDSMAIIERTARHNLSLMHPLEAMGLTFGATNTDADPEDLKNKTVINLTLPQATESITYHLKSLMQLKKVSTASGLNTNILKAFDNIISTPVKKNKMASKLAPGMDVVFTSDNGKTFFTKNILSKNMLAGPKERVFAYNNLINNLNNSCFIQNHTDFLKQQDSWPFYDAHNFTNKFLMQPIFSGQTRPRLQGAMEAAHVETHLTAFLQSIQPSRPQDPSVLASPKLSALILNMLWRNEITEFMDQLSKYSQEILKTFKQLRPSEYKQYNEFLTQVTPLLQKTPEKIPELVDHIFNYLDNVEKICELLVNASSIIISSKIREQVKHGMSFSYKADLDSLADILSQKQYVLMHLSKNIAAQYFNTCLNQGKSKLDLKAASVFYSSRPRTASSAELYRKMLYAYGSPQEINYYTEKARNKTLDVEENDSMAIIERTARHNLSLMHPLEAMGLTFGATNTDADPEDLKDKTVINLTLPQATESITYHLKSLMQLKKVSTASGLNTNILKAFDNIISTPVKKNKMASKLAPGMDVVFTSDNGKTFFTKNILSKNMLAGPKERVFAYNNLISNLNNSCFIQNHNDFLRQQDSWPFYDAHNFTNKFLMQPIFSGQTRPRLQGAMEAAHVETHLTAFLQSIQPSRPQDPSVLASPKLSALILNMLWRNEITEFMDQLSKYSQEILKTFKQLRPSEYKQYNEFLTQVTPLLQKTPDKIPELVDHIFNYLDNVEKICELLVNASSIIISSKIREQVKHGMSFSYKADLDSLADILIQKQYVLMHLSKNIAAQYFNTCLNQGKSKLDLKAASVFYSSRPRTASSAELYRKMLYAYGSPQEINYYTEKARNKTLDVEESDSMAIIERTARHNLSLMHPLEAMGLTFGATNTDADPEDLKDKTVINLTLPQATESITYHLKSLMQLKKVSTASGLNTNILKAFDNIISTPVKKNKMASKLAPGMDVVFTSDNGKTFFTKNILSKNMLAGPKERVFAYNNLISNLNNSCFIQNHNDFLRQQDSWPFYDAHNFTNKFLMQPIFSGQTRPRLQGAMEAAHVETHLTAFLQSIQPSRPQDPSVLASPKLSALILNMSNYYYYYGGGRYDWLKTVEPTNFLKIGLPYQAHPLHLQHQATTTPPSILEKFKRADILLNEVKAEMDPLMLQPETEKKLYQILGSIDMFKGLRKKVEFTYNAQIVTNAWLKMYELLNTMNFNNTSQAFCNCELPGGFISAINHFNYTMMHYPTFNWVASSLYPSSETDALEDHYGLYQCNPDNWLMQSPLLKKNVDYNDGDVTIASNVKNLALRATQRLTPIHLYTADGGINVGHDYNKQEELNLKLHFGQALTGLLSLSKGGNMILKHYTLNHAFTLSLICVFSHFFEELYITKPTSSRPTNSETYIVGKNRLRLFTPKEEQILLKRLEFFNDTPLVDLSLYQNLLESIYFAVETIHLKQQIEFLNFGMKCYRHFYNKIKLLNEYLAPKKKIFQDRWRVLNKLYVLEKKHKLKLCAPQGSVAMSNYYYYYGGGRYDWLKTVEPTNFLKIGLPYQAHPLHLQHQATTPPSILEKFKRADILLNEVKSEMDPLMLQPETEKKLYQVLGSIDMFKGLRKKVEYTYNAQIVTNAWLKMYELLNTMNFNNTSQAFCNCELPGGFISAINHFNHTMMHYPTFNWVASSLYPSSETDALEDHYGLYQCNPDNWLMQSPLLKKNMDYNNGDVTIASNVKNLALKATQKLTPIHLYTADGGINVGHDYNKQEELNLKLHFGQALTGLLSLSKGGNMILKHYTLNHAFTLSLICVFSHFFEELYITKPTSSRPSNSETYIVGKNRLRLFTPKEEQVLLKRLEFFNDTPLVDLSLYQNLLESIYFAVETIHLKQQIEFLNFGMKCYRHFYNKIKLLNEYLAPKKKIFQDRWRVLNKLYVLEKKHKLKLCTSFQGSVAMSNYYYYYGGGRYDWLKTVEPTNFLKIGLPYQAHPLHLQHQATTPPSILEKFKRAEILLNEVKAEMDPLMLQPETEKKLYQILGSIDMFKGLRKKVEFTYNAQIVTNAWLKMYELLNTMNFNNTSQAFCNCELPGGFISAINHFNHTMMHYPSFNWVASSLYPSSETDALEDHYGLYQCNPDNWLMQSPLLKKNMDYNNGDVTIASNVKNLALRATQRLTPIHLYTADGGINVGHDYNKQEELNLKLHFGQALTGLLSLSKGGNMILKHYTLNHAFTLSLICVFSHFFEELYITKPTSSRPTNSETYIVGKNRLRLFTPKEEQVLLKRLEFFNDTPLVDLSLYQNLLESVYFAVETIHLKQQIEFLNFGMKCYRHFYNKIKLLNEYLAPKKKIFQDRWRVLNKLYVLEKKHKLKLCAPLQGSVAMSNYYYYYGGGRYDWLKTVEPTNFLKIGLPYQAHPLHLQHQATTPPSILEKFKRADILLNEVKAEMDPLMLQPETEKKLFQILSSIDMFKGLRKKVEFTYNAQIVTNAWLKMYELLNTMNFNNTSQAFCNCELPGGFISAINHFNYTMMHYPTFNWVASSLYPSSETDALEDHYGLYQCNPDNWLMQSPLLKKNMDYNNGDVTIASNVKNLALRATQRLTPIHLYTADGGINVGHDYNKQEELNLKLHFGQALTGLLSLSKGGNMILKHYTLNHAFTLSLICVFSHFFEELYITKPTSSRPTNSETYIVGKNRLRLFTPKEEQVLLKRLEFFNDTPLVDLSLYQNLLESVYFAVETIHLKQQIEFLNFGMKCYRHFYNKIKLLNDYLAPKKKIFQDRWRVLNKLYVLEKKHKLKLCAASQGSVAMSNYYYYYGGGRYDWLKTVEPTNFLKIGLPYQAHPLHLQHQTTTPPSILEKFKRADILLNEVKAEMDPLMLQPETEKKLFQILSSIDMFKGLRKKVEFTYNAQIVTNAWLKMYELLNTMNFNNTSQAFCNCELPGGFISAINHFNYTMMHYPTFNWVASSLYPSSETDALEDHYGLYQCNPDNWLMQSPLLKKNMDYNNGDVTIASNVKNLALRATQRLTPIHLYTADGGINVGHDYNKQEELNLKLHFGQALTGLLSLSKGGNMILKHYTLNHAFTLSLICVFSHFFEELYITKPTSSRPTNSETYIVGKNRLRLFTPKEEQVLLKRLEFFNDTPLVDLSLYQNLLESVYFAVETIHLKQQIEFLNFGMKCYRHFYNKIKLLNDYLAPKKKIFQDRWRVLNKLYVLEKKHKLKLCAASQGSVA<br>    VF439        MKKFTHYIVEQIVNKYNKMFQCKMNKIPKNQSITHFKTVFDLYEKIGIQEVFFNIKPTGIYIYTTYDNSIHVETIFSNNFFLEYMCEKDAIFTLKVNAMKVRNLVNTDSLEMGIRSAQDAPLVIRMSTKTIDFEKKIAIKKTQSYKLPELIDVTPLTIKSADFLEFCKSINGGKYTLSIKTKKSEMVNGVLQRGEIIFEAENSRIVIRSETPIESEFEGEFKSEYFSSLRKVTKFNTTLKIYLCPDHPLIFETTIGPKEKDKVIVWIKSLKQMEEEYHLQKSITMSLEKDEEVVIFEAYLAGSANIFKNLIELLSQVAPIKKSDNSNKSIKQAFFKITKNGIYINIDHQSDILINISLDADKFGSYKYNFSLPELHIGITLDIIKDALKNVKKSEGVALCIRKKTFQTMPNEMSFSISSMDNNNASRGFVVKFNIVQNININAMLDGQELIEMRSNQFLGLCKDMGGSKKQIKVIVHNDSVVFSCNMVDIATKWLSFPIEEEVFQSSIHEKLYFLNHFKSEHIKTITKLATFDDTIKMSCVDDSIVFKSYITRCTPTKARHLPTEYIGKINIWVKAEPKKEISDDEDDL<br>    VF453        MKEQEIIAELSHSYFNHILREVRPFLEYFNDDFLDSSLTAPIDSELLERLFFMAAITSLDVSLPFNFSDSDVVKSHRKHRWLTSVQSFGSKSKQGIVGKSRLFNAFTVVVKRAKNSKFDEITMRDFCVGINLNKILRQAPFFVRTLGGFQHKNQFNIVTEFVDGRTLKTFLQSKRSSWMDFLNIFFQILLGLEIAQNRLNFSHYDLHTDNIILVPVERSFTVSLYGSNYTVKHDYRPVMIDFGLSSVHTKGKTLGQTNLENKGIFGHMSPGYDIYVFLLFCIDVVQSTNLSIYKGITDLLGFFNSKTNISMDLLTNNHIQSLEKGVSNLVPYQFISYIRDKFSPYLNVDIGPQKMSIDRCLGQKPMFLRLKHLLDRDDELEPLNSPLLKKGFVKTLVNNIKVYYWYREKMVLESDQTAQLIETDKEILDNLIDDLELKIVRKGSDKPQITVEQKNLFFMALEYYNFIRELRLEESSEFYKTWSKRFKKTWVCRNIFSQLDCVIREERLTKCPTAGATLEKMAECLSKKIKQEETKLKKKNDTVSDIIYIIDKIREDNIMKYSCLEPYISSLIINIERGQLKLTKLEKQLTREEYLQLMSYIFYGCICITKETSTVAPTVENVDEKMHKRNFVKMNKWLTNLESFGNLSKQGDVYKGYFLGKEVVLKKPKKNFFLENTLKDYFNGIKCINSLRMECPMFAYTLGILTIETPSERKKLDEIWLVTEFIKGKTLKEMLKCNELSFKEFLDIFAQILVALEIGQNKYKFCHYDLHTDNVIIVPITNNYTCHVHVYDINVKTKFMPVMIDYGMSSSSLEKDISIGQHKIEDNGIFPYLFPGYDPYVFLLFCRDVSNVSIKSGINHLFKFYGDIDHIDYVSTLKNNTGRQTPLLFLEYIIKTYPQLSMDFTPRTIMTKSLLFKTPSMMMADLFLNPNEKESISSCFNNELNNGFIYFMLECTRNKLWFGKDKNMLTMSSYKKMISHDLKFLQKFISFVHEEKNNWSERIDDLQCAVNLLFLMKQLKIYDFKPYREWIDKFEASEMYSWYKSNKPYIDRCQRLKMSNESFPETLENLLSTLQTKQQNAIQSEVIEWLHSFCETFHLKIHCHKQFIPSGEKKRAKIPAQEIQGNTQPSHHVHRVVLSRAQPVKAQESLLTTMCNGLVLDANTWTCLAIPPPAPFQQATRQVQHFYRNNFYEVVPIQDGTLLTIYYWDDPEHGPSWCLASTHGYDVSNYCWIGDKTFAELVYELLQQHSTCDVTLEKNKTRGTRLFFNNLNPDYCYTIGIRHHNLQPLIYDPQNIWAIQSTNLKTLKTVYPEYYGYIGIPGIQSQVPELPQYDLPYLIRSYKTAMNQAKNAIKNGKKDKEYFNYGYLLISRAPAITKSTSNVLLKSPLLVFLQKSVYQKKHNISNSQRLEFIILQNYLMQHFRDHFIALFPQYISYYTKYQNMLNMIIHSIATKDKDHPFAGAVVKKVLEDIENAENIIDHTTIQNYAHQSKYAMLYLSIISHFMSNESFPETLENLLSTLQTKQQNAIQSEVIKWLHSFCETFHLKIHCHKQFIPSGEKKWPKIPTQETQENTQPPHHVHRVVLSRAQPLKVQESLLTTMCNGLVLDANTWTCLAVPPPAPFQQVTRQVQHYYRNKFYEVAAIQDGTLLTIYYWDDPEHGPSWCLASTHGYDVSNYCWIGDKTFAELVYELLQQHSTCNVTLEKNKTRGTRLFFNELNRDYCYTIGIRHHNLQPLIHDPQNIWAIQSTNLKTLKTVYPEYYGYVGIPGIQSQVPELPQFELPYLIRSYRTAMNQAKNAIKNGKKEKEYFNYGYLLISRAPAITKSISIVLLKSPLLVFLQKSVYQKKYNISSSLRLEFIILQNYLMQHFRDNFIALFPQYISYYMKYQNMLNMIIHSIAIKDKDHPFAGAVVKKVLEDIENAENIIDHTTIQNYAYQSKYAMLYLSIITHFMSNESFPETLENLLSTLQTKQQNAIQSEVIEWLHSFCETFHLKIHCHKQFIPSGEKKWAKIPTQETQENTQPHVQRVVLSRAQPLKVQESLLTTMCNGLVLDANTWTCLAVPPPAPFQQVTRQIQHYYRNNFYEVVPIQDGTLLTIYYWDDPEHGPSWCLASTHGYDVSNYCWIGDKTFAELVYELLQQHSTCDVTLEKKQTRGTRLFFNNLNRDYCYTIGIRHHNLQPLIYDPQNIWAIQTTNLKTLKTVYPEYHGYVGIPGIQSQVPELPQYELPYLIRSYKTAMNQAKNAIKNGKKDKGYFNYGYLLVSRAPAITKSISTVLLKSPLLVFLQKSVYQKKHNLTSSLRLEFIILQNYLMQHFRDNFIALFPQYISYYTKYQNMLNMIIHSIAIKDIDHPFAGTVVKKVLEDIENAENIIDHTTIQNYVHQSKYAMLYLSIISHFMSNESFPETLENLLSTLQTKQQNAIQSEVIEWLHNFCETFHLKIHCHKQFIPSGEKKRAKIPAQETQGNTQPSHHVHRIVLSRAQPVKAQESLLTTMCNGLVLDANTWTCLAIPPPAPFQQATRQVQHFYRNNFYEVVPIQDGTLLTIYYWDDPEHGPSWCLASTHGYDVSNYCWIGDKTFAELVYELLQQHSTCDVTLEKNKTRGTRLFFNNLNPDYCYTIGIRHHNLQPLIYDPQNIWAIQSTNLKTLKTVYPEYYGYIGIPGIQSQVPELPQYDLPYLIRSYKTAMNQAKNAIKNGKKDKGYFNYGYLLISRAPAITKSTSNVLLKSPLLVFLQKSVYQKKHNISNSQRLEFIILQNYLMQHFRDHFIALFPQYISYYTKYQNMLNMIINSIATKDKDHPFAGAVVKKVLEDIENAENIIDHTTIQNYVHQSKYAMLYLSIISHFMSNESFPETLENLLSTLQTKQQNAIQSEVIEWLHNFCETFHLKIHCHKQFIPSGEKKRAKIPAQETQGNTQPSHHVHRIVLSRAQPVKAQESLLTTMCNGLVLDANTWTCLAIPPPAPFQQATRQVQHFYRNNFYEVVPIQDGTLLTIYYWDDPEHGPSWCLASTHGYDVSNYCWIGDKTFAELVYELLQQHSTCDVTLEKNKTRGTRLFFNNLNPDYCYTIGIRHHNLQPLIYDPQNIWAIQSTNLKTLKTVYPEYYGYIGIPGIQSQVPELPQYDLPYLIRSYKTAMNQAKNAIKNGKKDKGYFNYGYLLISRAPAITKSTSNVLLKSPLLVFLQKSVYQKKHNISNSQRLEFIILQNYLMQHFRDHFIALFPQYISYYTKYQNMLNMIINSIATKDKDHPFAGAVVKKVLEDIENAENIIDHTTIQNYVHQSKYAMLYLSIISHF<br>    VF466        MSYNNPANTKGYLVAPIRYLEYKDFNPDGTLKHFRNKTCLVMVQASFCGHCSSAKPAFQNFARKHPEIVCLTVQGDDPVSDSIDRLVKLISTIKPSFQGYPDYLLFRNGVLVPREIDGRSEADLEKFVRMYQQKYFEKPVYYLQRNDFDDNGNLIVPELRNKKVIIMIQANYCGHCTNAKGDYYKAAKYIKELEKNGGTSYKNKVVFATIQADGDEEGEKELNQILDKIKPTFVGFPDYVLYVNGKRIEDDGPPGRNFNNIVNYVMG<br>    VF467        MGRQVTPIYPRTNGTIQPVNFPIRNMEPPNHSLQSAGFQIPPPDAQFPRYHAAAPHHPRVEAAAPSCLDVARHVESCPICSRIHDTDKTLYVLVIVGLTILCFLLVKRILKLMNMPGKTPIEKIVSLSAKPFKVSRGSPRDDKERSYENTFRTPIPYNCPDVSTHLSDCPVCSNLYGKEKLLYIFLGAMIVIIFLVIKNQLN<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Uniprot_Protein_Sequences_FRG3G.tsv<br><br>File contents:<br><br>    001R        MAFSAEDVLKEYDRRRRMEALLLSLYYPNDRKLLDYKEWSPPRVQVECPKAPVEWNNPPSEKGLIVGHFSGIKYKGEKAQASEVDVNKMCCWVSKFKDAMRRYQGIQTCKIPGKVLSDLDAKIKAYNLTVEGVEGFVRYSRVTKQHVAAFLKELRHSKQYENVNLIHYILTDKRVDIQHLEKDLVKDFKALVESAHRMRQGHMINVKYILYQLLKKHGHGPDGPDILTVKTGSKGVLYDDSFRKIYTDLGWKFTPL<br>    002L        MSIIGATRLQNDKSDTYSAGPCYAGGCSAFTPRGTCGKDWDLGEQTCASGFCTSQPLCARIKKTQVCGLRYSSKGKDPLVSAEWDSRGAPYVRCTYDADLIDTQAQVDQFVSMFGESPSLAERYCMRGVKNTAGELVSRVSSDADPAGGWCRKWYSAHRGPDQDAALGSFCIKNPGAADCKCINRASDPVYQKVKTLHAYPDQCWYVPCAADVGELKMGTQRDTPTNCPTQVCQIVFNMLDDGSVTMDDVKNTINCDFSKYVPPPPPPKPTPPTPPTPPTPPTPPTPPTPPTPRPVHNRKVMFFVAGAVLVAILISTVRW<br>    003R        MASNTVSAQGGSNRPVRDFSNIQDVAQFLLFDPIWNEQPGSIVPWKMNREQALAERYPELQTSEPSEDYSGPVESLELLPLEIKLDIMQYLSWEQISWCKHPWLWTRWYKDNVVRVSAITFEDFQREYAFPEKIQEIHFTDTRAEEIKAILETTPNVTRLVIRRIDDMNYNTHGDLGLDDLEFLTHLMVEDACGFTDFWAPSLTHLTIKNLDMHPRWFGPVMDGIKSMQSTLKYLYIFETYGVNKPFVQWCTDNIETFYCTNSYRYENVPRPIYVWVLFQEDEWHGYRVEDNKFHRRYMYSTILHKRDTDWVENNPLKTPAQVEMYKFLLRISQLNRDGTGYESDSDPENEHFDDESFSSGEEDSSDEDDPTWAPDSDDSDWETETEEEPSVAARILEKGKLTITNLMKSLGFKPKPKKIQSIDRYFCSLDSNYNSEDEDFEYDSDSEDDDSDSEDDCMYQAINPCPQSWYGSPQLEREIVCKMSGAPHYPNYYPVHPNALGGAWFDTSLNARSLTTTPSLTTCTPPSLAACTPPTSLGMVDSPPHINPPRRIGTLCFDFGSAKSPQRCECVASDRPSTTSNTAPDTYRLLITNSKTRKNNYGTCRLEPLTYGI<br>    004R        MARPLLGKTSSVRRRLESLSACSIFFFLRKFCQKMASLVFLNSPVYQMSNILLTERRQVDRAMGGSDDDGVMVVALSPSDFKTVLGSALLAVERDMVHVVPKYLQTPGILHDMLVLLTPIFGEALSVDMSGATDVMVQQIATAGFVDVDPLHSSVSWKDNVSCPVALLAVSNAVRTMMGQPCQVTLIIDVGTQNILRDLVNLPVEMSGDLQVMAYTKDPLGKVPAVGVSVFDSGSVQKGDAHSVGAPDGLVSFHTHPVSSAVELNYHAGWPSNVDMSSLLTMKNLMHVVVAEEGLWTMARTLSMQRLTKVLTDAEKDVMRAAAFNLFLPLNELRVMGTKDSNNKSLKTYFEVFETFTIGALMKHSGVTPTAFVDRRWLDNTIYHMGFIPWGRDMRFVVEYDLDGTNPFLNTVPTLMSVKRKAKIQEMFDNMVSRMVTS<br>    005R        MNAKYDTDQGVGRMLFLGTIGLAVVVGGLMAYGYYYDGKTPSSGTSFHTASPSFSSRYRYMRYTVLIALQGALLLLLLIDDGQGQSPYPYPGMPCNSSRQCGLGTCVHSRCAHCSSDGTLCSPEDPTMVWPCCPESSCQLVVGLPSLVNHYNCLPNQCTDSSQCPGGFGCMTRRSKCELCKADGEACNSPYLDWRKDKECCSGYCHTEARGLEGVCIDPKKIFCTPKNPWQLAPYPPSYHQPTTLRPPTSLYDSWLMSGFLVKSTTAPSTQEEEDDY<br>    RNC        MFPHVTIVEKPGCGNCVTLKRALKEKSVPHTVMPYVPGVPGSYPIATVGGQRVGYREITKALAMDMFTTSKCFKPFRYNWAEDIRTKHEKVHWVETEINLRDDVTDWKSGVMTPGEKAFVSCVLRLFTQLDVEVGALYLDHLTQVFRCNEIRNMLTSFACREAIHQKAYSLLNDTLGMPDCEYQAFADYGEMTDKLDFMREMDCSTLPGLARSLLRAVLNEGVSLFAAFVMLLTFQRSGKMKGMGKVVEWSVRDENIHVEGVSKLFETLNSEYPWVVTKDLIELFKEDAERVAELEEAFVDLAFQECGPVCGLEPKQVKDYVLYVINLRAANMGLPKLIERAPEACPIPWVKWLIDGADLTNFFENHVTEYETGGLTGSWSYEPLDIMSGPPSHEIGMLYSLAPSVDMKSENIEDLPSESIPTTNHCASNYFYAPECPDIGHLRALSIMNRWTETEFVISDDLRDVSMLTEEERYFYRFLFTFLSAADDLVNLNIDNLLGLFSQKDIHHYYFEQECIEAVHSRAYSIIQLMLFDNDSLARAEYVKTSLNSPAIQSKLDWLNKRVIECSSIAEKYILMILIEGIFFSASFAAIAYLRINNLFVVTCQINNLISRDEAIHVEASCCIFKNYLMGPKPSIGRVQELFKEAVQIECAFLRTAAPRNSQLLDVEAICSYVRYSADRLLKELDMPPIYNEPKPTADFPLALMTASNNTNFFERRNTAYSGSVSNDLMDSAAPALSPALTALTDQSATADLAIQIPKCPDPERYFYTSQCPDINHLRSLSILNRWLETELVFVGDEEDVSKLSEGELSFYRFLFAFLSAADDLVTENLGGLSGLFEQKDILHYYVEQECIEVVHSRVYNIIQLVLFHNNDQARREYVAGTINHPAIRAKVDWLEARVRECASVPEKFILMILIEGIFFAASFAAIAYLRTNNLLRVTCQSNDLISRDEAVHTTASCYIYNNYLGGHAKPPPDRVYGLFRQAVEIEIGFIRSQAPTDSHILSPAALAAIENYVRFSADRLLGLIHMKPLFSAPPPDASFPLSLMSTDKHTNFFECRSTSYAGAVVNDLMDSAAPALSPALTAHTGHSATADLAIQIPKCPDPERYFYTSQCPDINHLRSLSILNRWLETELVFVGDEEDVSKLSEGELSFYRFLFAFLSAADDLVTENLGGLSGLFEQKDILHYYVEQECIEVAHSRVYNIIQLVLFHNNDQARREYVAGTINHPAIRAKVDWLEARVRECASVPEKFILMILIEGIFFAASFAAIAYLRTNNLLRVTCQSNDLISRDEAVHTTASCYIYNNYLGGHAKPPPDRVYGLFRQAVEIEIGFIRSQAPTDSHILSPAALAAIENYVRFSADRLLGLIHMKPLFSAPPPDASFPLSLMSTDKHTNFFECRSTSYAGAVVNDLMDPAVSPASTDPLDTHASGAGAAPIPVCPTPERYFYTSQCPDINHLRSLSILNRWLETELVFVGDEEDVSKLSEGELGFYRFLFAFLSAADDLVTENLGGLSGLFEQKDILHYYVEQECIEVVHSRVYNIIQLVLFHNNDQARRAYVARTINHPAIRVKVDWLEARVRECDSIPEKFILMILIEGVFFAASFAAIAYLRTNNLLRVTCQSNDLISRDEAVHTTASCYIYNNYLGGHAKPEAARVYRLFREAVDIEIGFIRSQAPTDSSILSPGALAAIENYVRFSADRLLGLIHMQPLYSAPAPDASFPLSLMSTDKHTNFFECRSTSYAGAVVNDLMDPAVSPASTDPLDTHASGAGAAPIPVCPTPERYFYTSQCPDINHLRSLSILNRWLETELVFVGDEEDVSKLSEGELGFYRFLFAFLSAADDLVTENLGGLSGLFEQKDILHYYVEQECIEVVHSRVYNIIQLVLFHNNDQARRAYVARTINHPAIRVKVDWLEARVRECDSIPEKFILMILIEGVFFAASFAAIAYLRTNNLLRVTCQSNDLISRDEAVHTTASCYIYNNYLGGHAKPEAARVYRLFREAVDIEIGFIRSQAPTDSSILSPGALAAIENYVRFSADRLLGLIHMQPLYSAPAPDASFPLSLMSTDKHTNFFECRSTSYAGAVVNDLMDSVDRFLYTSDHDGFLALTRETWQNRWFPSQIPLHLDVSHVHQLSDADKDFYRFLFTFLGMAERLVNFNIEDLVTNFNSHDVAHYYAEQVAMENIHGVVYANILKIFFNNSRGELLAYAAKIILDPALCEKLEWLHSRVRKATTKAQKILMFLIVEGIYFISSFYSISLCRVRGIMPGVCLANDYISRDELLHTKAASMLYNTMVEIQEKPSHAEVQALFREAVNIEIRFIQAKAAGVTLVNVVDIRRFLEATADRILRDIFVPPIFGTPPPEACPLSYTGSFKAVNFFERDNSDYTTCVTDDLMEYEHLWLTAYGCLAVVGAVVVYCAIKQSLVTNWWFYGSIYPQVKLMADRTTFKPFEYPGCYDKWDTHEHAHWSFKELNMQDDVHDWHNRLSEAEKTFLVQILRF</td><td class="s1" dir="ltr">The sequences in these data files were extracted from the UniProt database: http://www.uniprot.org.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br><br># get input and output file names from sys.argv<br>input_file = sys.argv[1]<br>output_file = sys.argv[2]<br><br># open the input file and read its contents<br>with open(input_file, &quot;r&quot;) as f:<br>    lines = f.readlines()<br><br># open the output file<br>with open(output_file, &quot;w&quot;) as f:<br>    for line in lines:<br>        # extract the protein identifier and amino-acid sequence<br>        identifier, sequence = line.strip().split(&quot;\t&quot;)<br>        # calculate the length of the sequence<br>        length = len(sequence)<br>        # classify the protein based on its length<br>        if length &lt; 500:<br>            classification = &quot;short&quot;<br>        elif length &gt;= 500 and length &lt;= 999:<br>            classification = &quot;medium&quot;<br>        else:<br>            classification = &quot;long&quot;<br>        # write the results to the output file<br>        f.write(&quot;{}\t{}\n&quot;.format(identifier, classification))</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The first data file is quite large. Thus the length of the full prompt exceeded ChatGPT&#39;s limit. I shortened it to include only the second data file.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R153" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">154</div></th><td class="s1">16 - Python Scripts &amp; &quot;while&quot; Loops</td><td class="s1">12 - Summarize microorganisms</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Suppose you are given tab-delimited text files that contain information about proteins in a specific type of microorganism (virus, bacteria, fungi). These text files contain three columns of data. The first column contains species names (for example, &quot;AKT8 murine leukemia virus&quot;). The second column contains protein identifiers (for example, &quot;001R&quot; or &quot;VP31&quot;). The third column contains amino-acid sequences. The file also includes a header line with column names.<br><br>Your task is to summarize the different types of microorganisms according to the sizes of the proteins. Proteins with fewer than 500 amino acids should be classified as &quot;short&quot;. Proteins with 500-999 amino acids should be classified as &quot;medium&quot;. Proteins with 1000+ amino acids should be classified as &quot;long&quot;.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept four arguments. The first three arguments will be the names of input files, as described above, that contain information for viruses, bacteria, and fungi, respectively. The fourth argument will be the name of a text file that your code should create. For each input file, your code should calculate the number of short, medium, and long proteins across all species for that type of microorganism. Finally, your code should write each combination of microorganism type and length classification to the output file.<br><br>Hint(s):<br><br>    You might consider creating a Python function that summarizes data from one of the input files. Your script could then invoke this function rather than having to repeat similar code for each file.<br>    <br>    Make sure to remove the newline character from the end of each line or your counts may be slightly off.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;Uniprot_Protein_Sequences_Viruses.tsv&quot; &quot;Uniprot_Protein_Sequences_Bacteria.tsv&quot; &quot;Uniprot_Protein_Sequences_Fungi.tsv&quot; &quot;Microorganism_Summary.tsv&quot;<br>    <br>    cat Microorganism_Summary.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Uniprot_Protein_Sequences_Bacteria.tsv<br><br>File contents:<br><br>    Species        ProteinID        Sequence<br>    Acaryochloris marina         ACCA        MAVSDRKLQLLDFEKPLAELEDRIEQIRSLSEQNGVDVTDQIAQLEGRAEQLRQEIFSSLTPMQELQLARHPRRPSTLDYIHAISDEWMELHGDRRGYDDPAIVGGVGRIGGQPVLMLGHQKGRDTKDNVARNFGMPFPSGYRKAMRLMDHADRFGLPIISFIDTPAAWAGLEAEQFGQGEAIALNLREMFRLDVPIICTVIGEGGSGGALGIGVGDRLLMFEHSIYSVAPPEACAAILWRDAQEGPQAAEALKITATDLQELGIIDQILPEPPGGAHVNPIKAANIIKTAILSNLEELWRLSPQERRQQRYHKFRNMGIFSQLP<br>    Acaryochloris marina         ACCD        MSLFDWFANRRKSHPVTKAQPEREISDGLWSKCESCGALTYTKDLRANQMVCLECGHHLRVTSDERIAQLIDAETWVAMDNNLVACDPLQFRDRKAYCDRIQEYQSKTGLQDAVQTGVGELDGLPLALGVMDFRFMGGSMGSVVGEKLTRLIEHATQNRLPLVIVCASGGARMQEGMLSLMQMAKTSAALDQHRTNRLLYIPVLTHPTTGGVTASFAMLGDMILAEPKATIGFAGRRVVEQTLREKLPDDFQTAEYLLSHGFVDAIVPRPELKRTLAQLIRIHAQMPMASDVSPAAVPVAAAVE<br>    Acaryochloris marina         ACKA        MKILVLNAGSSSHKCCLYAIEGPLPDHPAPPLWEAQLDWHDPNHASLKVKTKRENLEEDLSDVSRADALSHLLTTLWHGPTQVIQMAQEINVVGHRVVHGGQDYQSSVVVTSEVKTAIADLIPLAPNHNPANLEGIKLIEQLLGNVRQVAVFDTAFHSHLPEVAAIYPGPYDWQQQGIRRYGFHGISHQYCTQRTAEILGRGVDRLIICHLGNGASLTAVKQGQSIDTSMGFTPLEGLMMGTRSGSIDPGILIHLMRQGYDADQLDHMLNKASGLKGISGVSHDLREIEQAIAQDNAQAKLARDLYLHRLKACLGSMLMSLGGVDVLVFTGGIGEHSASVRAETCEALAFLGLKLDPILNRKSPVDQDIAAVDSKVRVLVIKTQEDWAIAQSCYEQCC<br>    Acaryochloris marina         ACP        MSESEIFGKVKDIVAEQLSVDADKVTPEASFQNDLDADSLDVVEMVMALEEEFDIEIPDEAAEEILTVQAAVDFIAGKVKA<br>    Acaryochloris marina         ACYP        MMPNIICLKAVISGKVQGVGYRYSTRAKAQSLGLVGWVRNLPDGRVEAMAEGERTQVDKLIEWFKQGPPAAEVSKVDVDEESLGEFRAFEILR<br>    Acaryochloris marina         AMPA        MKIQAIETVAADWSGDYLGIGFFEDSAAIDAGLTQLDQSLQDILQDLVSETEFQGKVGEQPWTRIAGPGSIRKVMLIGLGSPDAFTLDSLRQAAAAFAKAAQKQKAASAGLQLPSWDDDAAASTQAMAEGIELALHQEIRFKSDPDAKKPGEFPQEVHLLGLANQAAAIEKAQQICAGVILTRELVAAPANVVTPSALAETAAQIAEDHGLTLEVLEREECEAQGMGAFLGVSLASELPPKFIHLTYKPSGTPRRKLAIVGKGVTFDSGGLNLKVGGSGIETMKMDMAGSGATLGAAKAIGQLKPDVEVHFIVAAAENMISGHALHPGDILTASNGKTIEINNTDAEGRLTLADALVFAEKQGVDAIVDLATLTGACIVALGNDIAGMWTPEDSLAEELSQASEQAGEKFWRMPLEEKYFEGLKSPIADMKNTGPRPGGSITAALFLKQYIENTPWAHLDVAGPVWTEKENGYLNVGATGFAVRTLVNWVMG<br>    Acaryochloris marina         ANMK        MHVIGLMSGTSVDGIDAALVEISGRELDLQVDLIAAQTYPYPDALRNQILAVCRGEPLSIEALAGLDDAIAMQFAQAAQSIQAEATLTAELIGSHGQTVFHRPPQGNLGYTLQIGRGAVIAHQTGINTVSNFRVADMAAGGQGAPLVSKLDICLLSHPEFYRCVQNIGGIGNVTYLPPLTDASQLGMGVKGWDTGPGNVLMDLAVAHFSKGELTYDADGAWAAQGNPCQPLIEEWLRHPFFQAPPPKSTGRELFSPDYLQTCLQTAASYQLSPADMLATLTDFTAATIVYNYDQFLPHPPDQVVLCGGGSRNGYLRQCLQQRLGGSTVLTTDDVGLNSDAKEAIAFAVLAYWQQLQIPGNVPAVTGASGPILLGELHLV<br>    Acaryochloris marina         APT        MDLKSLIRDIPDFPKPGILFRDITTLLQNPDGLRYVIDQFTEDYREQSIDYVVGIESRGFIFGAPLAYQLGAGFVPVRKPGKLPAAIHAQEYELEYGTDRLEIHQDACPAGSRVLVVDDLLATGGTAAATAQLLAPLNCTLVGFGFIIELTGLAGRQKLPDCHINALVEY<br>    Acaryochloris marina         ARGB        MFNASDRVQVLSEALPYIQSFAGRTVVVKYGGAAMKDSTLKATVIRDVVFMACVGLRPVLVHGGGPEINLWLEKLGIEPQFMKGLRVTDAPTMDVVEMVLVGRVNKELVSLINQAGGSAVGVCGKDGKLLTARPQGESGIGFVGEVASVNPQLIHNLLDGGHIPIVSSVATDEEGQVYNINADTVAGELAAALDAEKLILLTDTPGILQDYKDATSLIHKLDIRQARELIAAEVVAGGMIPKVTCCIRSLAQGIQAAHIIDGRAPHALLLEIFTDSGIGTMITA<br>    Acaryochloris marina         ARGC        MGSQEDIPVGIIGASGYGGVQLTRLLLEHPHVHLAYLGGDSSAGRPFADLYPHLGFKQDLKVEAIDLDKVVERTQVVFLALPNGLAAEMAPTLVECGCKVLDLSADYRFTNLDTYEDWYGTKRQDRDLASTAVYGLPELYRQGISTASLVGCPGCFPTASLLALAPLLKQGLIDPNSAIIDAKTGTSGGGRQAKTHLLLAEANGSIGPYGVAHHRHTPEIEQVSSDLAGREVVVQFTPHLAPMTRGILATVYATMRDPGLVREDLITIFSAFYRNSPWVKVLPGSTYPFTKWAYGTNLCYLGIEVDQRTGRVIVISAIDNLMKGQAGQAVQCLNLMMGWPETMGLPELTFYP<br>    Acaryochloris marina         ARLY        MTSSPVPPTNSSDSSDSKVWSQRFESALNPVIAAFNASIHFDIALIEYDLTGSQAHVQMLAHSGIISQAEADAIDQGLEAIRQEYRQGQFQPGIEAEDIHFAVENRLIELIGDTGKKLHTGRSRNDQVGTDLRLYLHDQILQVQTLIRNWQQALVILASDHVQTLIPGYTHLQRAQPISLAHHLLAYFEMAERDWQRLREIDRRVNVSPLGLGALAGTPFPIDRQYTADKLGFADLYRNSLDGVSDRDFAIEFLCAASLIMVHLSRFSEEVILWASEEFGFIKLKDSCATGSSIMPQKKNPDVPELVRGKSGRVFGHLQSLLVVMKGLPLAYNKDLQEDKEAIFDGVTTIKACLEAMTILVAEGLEFQTPRLESAVGEDFSNATDVADYLSKQGVPFREAYNIVGAVVKTCLSQNKLLKDLTLEEWQQFHPQFADDIYSAISPRQVVAVRNSFGGTGFTQVQNALEQAKTLLSQSK<br>    Acetobacter aceti        ACOCT        MTETTPAKPKGPFDGLLVIDLTHVLNGPFGTTILTDLGARTIKIEPPGHGDDTRTYGPYVGDQSLYFSFVNRGKESIVLNLKDEGDRAIFLEMVRKADVLAENFRPGVMDRLGFNYEELAKINPRLIYASSSGFGQTGPLAHYPAYDTIVQAMSGIMMATGFPDGPPTRVGGTSLSDLCGGVFMFCGIASALYARERTGKGAHIDVSMFDGTLAFLQHALMCWSATGKAPARIGNRHPYMAPFDVFQAQDKPFVICCGNDHLFKALCDVIGAPELATDPRFVENHDRMANNDALKAALEKALSKQPAAHWLDVIHKAGVPVGPLLDVAEAANLPQTAARNMLIKSGGVMMPGNPVKISGYDDPHERPGAPKLDEQGAALRKEFAAPEAK<br>    Acetobacter pasteurianus         AMPN        MRRLLTLTTLLAGVPLAAPVCAEQVFSFQRAAGQLPKTVVPVSYGINISTDIDNLKLTGQETIQVDVRTPTEDVTLNQAGLHLAGAVLDNGVKATITQDDAAETATLHFPAKVSKGAHTLVITYSGPILKTPNGIYVDDYTAPSGETKRMLVTQFEVADARRMFPGWDEPAFKATFQLNVTLPKEAVAVSNMPVTQSTPEGTSQKRVSFATTPRMSTYLLALVAGDMKSVQGQADGTPLAVYAPSGLEEQGEYALHASEKILPYYNNYFGVKYPLPQMDMVAIPGNYQAGAMENWGLLTYIDNVLLFDPPNSTPRTRELIYEVVAHEMAHQWSGDLVTMGWWDNIWLNEGFASWM<br>    Acholeplasma laidlawii        AMGDS        MRIGIFSEAYLPLISGVVTSVVNLKEGLEALGHEVYVITPIPSKDKFENDPSVIRIPGWVIPRKSLKGFRLVLFVKRYVRKMRKLKLDVVHIHTEFSMGKLGLAVAKKERIPSVYTLHTSYQDYTHYVSKLLTRFAPNAAKKLAGKINNQYTKNCHMTIVPTKKIYDKMIRLKHDGEFTIIPSGINLKPFYKSSYTSEQVQALKDKLGIRNDEFVAILVARIAKEKSIGDLVEAFVEFYKSYPNSRFIIIGDGPDKPVLDKLIDSKKASKYINTLGFVKNAEVGLYYQIADVFLNASTTETQGLTYVEALAASLPIIVRYDDVFDAFVEDGKNGIFFNKNEELVKHLIHIRQNPEILGTLSKNAEISTKPYAKEVYAKSCETLYLDLIDKNNKKLNKK<br>    Acholeplasma laidlawii         ACCD        MNHLLDERKRRLSQFHEIIKKRPIETRPVDIPDGVFSQCEQCNSAIYNKDLEHNYEVCPYCSYHFKINAVKRLKYTLDEDTFKPLFENITSKNPLGMPEYEEKLDKGMRMAKMNEAFLAGTGKIDGQNLAIGVLDSYFMMGSMGSVVGERITRLIEHAAKHNLPLVIFSASGGARMQEGILSLMQMAKTSGALNLLDEKGLVYISVMTNPTTGGVAASFASLGDINIAEKSSLIGFAGARVIKQTIGQDLPAGFQTDAFQLQKGQVDLVVLRSQLKKTLSTLLKLHAKKVTS<br>    Acholeplasma laidlawii         ACKA        MKILSVNAGSSSLKFQLLQMPEEFEIASGLVERIGNKNAEFKIKFNGSSKTQSDLVVLDHSVAVDLVIKGLLENKVIASLDEIQGVGHRVVQGGQLFKESVVIDDDVVIKIASLNDLAPLHNPANIIGINAFRKAISNVIHVAVFDTTFHQTMPEENFLYGTPYEWYTKHGVRKYGFHGTSHQYVSELAHARLGKDHSRIIVAHIGNGASITAVRDGKSIDTSMGLTPLEGFPTGTRSGNVDPAIFQLMQQKEGYSLEKTLNELNKNSGYLGISGVSHDSRDIIAAAEGGNKRAQLALDIQAKRIVDYIASYYVLLGGLDALVFTAGIGENAKEVRKLICDRLGVLGIKLDDNKNNSRGDKEISSDASNVKIWVIPTNEEVMIARDVLRLQNK<br>    Acholeplasma laidlawii         ACPS        MIHAIGTDLVELERIKSIGIDRFKDKILNEDEKNEYAKINHENRKLTYLAGRFAVKESLFKCFKAGDKTANYKDFSVLNDSVGAPYVVSKHTSDFVVHITISHTNLYAIAFVVLETKV<br>    Acholeplasma laidlawii         APT        MDLKAHIAAVKDFPKEGILFRDITPLMLDGKAFKYASDQFTEFARSKKADLIVGPEARGFIFGCPVAVNLGVGFAPVRKPGKLPRASVTVSYDLEYGSNSLSLHEDAVKPGQRVVIIDDLLATGGTMQATVELVEKLGGIVVGLAFLIELDDLEGRKLLKNYDVKTLINY<br>    Achromobacter lyticus        ACLY        AQVGTGPGGNQKIGQYEYGSGGRPFLDVAQSGSTYTFNTTNLTVNLNHGTSGSTAYSYTGPRNTINGAYSPLNDAHYFGRDYWTPSTNFNQGGQGVRQAAADLGYSTADVIDAFRQVGV<br>    Achromobacter lyticus        API        MKRICGSLLLLGLSISAALAAPASRPAAFDYANLSSVDKVALRTMPAVDVAKAKAEDLQRDKRGDIPRFALAIDVDMTPQNSGAWEYTADGQFAVWRQRVRSEKALSLNFGFTDYYMPAGGRLLVYPATQAPAGDRGLISQYDASNNNSARQLWTAVVPGAEAVIEAVIPRDKVGEFKLRLTKVNHDYVGFGPLARRLAAASGEKGVSGSCNIDVVCPEGDGRRDIIRAVGAYSKSGTLACTGSLVNNTANDRKMYFLTAHHCGMGTASTAASIVVYWNYQNSTCRAPNTPASGANGDGSMSQTQSGSTVKATYATSDFTLLELNNAANPAFNLFWAGWDRRDQNYPGAIAIHHPNVAEKRISNSTSPTSFVAWGGGAGTTHLNVQWQPSGGVTEPGSSGSPIYSPEKRVLGQLHGGPSSCSATGTNRSDQYGRVFTSWTGGGAAASRLSDWLDPASTGAQFIDGLDSGGGTPNTPPVANFTSTTSGLTATFTDSSTDSDGSIASRSWNFGDGSTSTATNPSKTYAAAGTYTVTLTVTDNGGATNTKTGSVTVSGGPGAQTYTNDTDVAIPDNATVESPITVSGRTGNGSATTPIQVTIYHTYKSDLKVDLVAPDGTVYNLHNRTGGSAHNIIQTFTKDLSSEAAQRAPGSCG<br>    Achromobacter obae        ACLR        MTKALYDRDGAAIGNLQKLRFFPLAISGGRGARLIEENGRELIDLSGAWGAASLGYGHPAIVAAVSAAAANPAGATILSASNAPAVTLAERLLASFPGEGTHKIWFGHSGSDANEAAYRAIVKATGRSGVIAFAGAYHGCTVGSMAFSGHSVQADAAKADGLILLPYPDPYRPYRNDPTGDAILTLLTEKLAAVPAGSIGAAFIEPIQSDGGLIVPPDGFLRKFADICRAHGILVVCDEVKVGLARSGRLHCFEHEGFVPDILVLGKGLGGGLPLSAVIAPAEILDCASAFAMQTLHGNPISAAAGLAVLETIDRDDLPAMAERKGRLLRDGLSELAKRHPLIGDIRGRGLACGMELVCDRQSREPARAETAKLIYRAYQLGLVVYYVGMNGNVLEFTPPLTITETDIHKALDLLDRAFSELSAVSNEEIAQFAGW<br>    Acidimicrobium ferrooxidans         ARC        MPRDETPEREHAEQQSRQALEEEVRELRRRVEEAPSRVRALEEKLLEVSGALAQTQAKNEKLTFTLQQAREHIQNLREEVEKLTQPPSAYGVYLAANQDGTADVFTTGRKMRVAVHPEIDLAAVRVGQEVVLNESFAIVSVRGSDVIGEVATVKDVLDDGSRVILLGRADEERVAQVAGHLQHPPLRVGDSVMVDPRSAMVLERLPRPEVSELALEEVPDITYHDIGGLDRQIEEIQDAVELPFLYRELFGDYRLPAPKGILLYGPPGCGKTLIAKAVANSLAKKVEQVSGRSVRSYFLNVKGPELLNKYVGETERQIRLIFQRAREKAEEGVPVIVFFDEMDSLFRTRGSGISSDMESTVVPQLLAEIDGVEALRNVIVIGASNREDLIDPAILRPGRLDVKIKIERPDEQAAREIFARYLTPEVPIDAGEVTTLGGGDAEKAIAVMIERTVERMYATSDENRFLEVTYQNGEKEILYFKDFSSGAMIENIVRRAKKLAIKREIAGGSRGICLDDLLASIAKEFKEHEDLPNTTNPDDWAKISGRKGERIVFMRILLHEEEGQTGGRAIERVATGQYL<br>    Acidiphilium cryptum         ACCA        MRHFLDFEKPVAELEAKIEELRRMTDPGELNIAEEVTLLSDKAERQLRTLYGRLSPWQKTQVARHPERPKARDVIAGLITEFTPLAGDRGFGEDAAIVAGPGRFKGEPVMVIAIEKGWDLESRLKHNFGSPRPEGYRKARRLIEMAGRFGLPVLSFVDTSGAYPGVDAEARGQAEAIARGIDACLAAPVPFIATIIGEGGSGGAIAIAAADTVLMFEHAIYSVISPEGCAAILWEDRANAAQAAEAMKITAQDLKRLGIIDRIVPEPLGGAHRDPQAAITALGAAIAETLPGLAALAPEALRAKRREKFLAIGQTLPV<br>    Acidiphilium cryptum         ACP        MSEVAEKVKKIVVEHLGVDEAKVTPEASFIDDLGADSLDTVELVMAFEEAFGVEIPEDAAEKIGTVKDAIDYIEKKKAE<br>    Acidiphilium cryptum         ACPS        MIIGIGSDLCDIRRVERTLERFGERFIQRVFTETERRRAARRPAQFAATLAKRFAAKEACAKALGTGFRRGVFMSDLGVVNLPSGQPTLALTGGAAERLAALTPPGRSAFVHLSLTDEYPYAYAHVIIEARDGA<br>    Acidiphilium cryptum         ACYP        MIAKHLILSGRVQGVGFRDWMVTRARRLALAGWVRNRADGTLEALVAGDAPAVEELLRACRRGPPLADVTDIVETFAEPPAEPGFVKRATG<br>    Acidiphilium cryptum         ADC        MDAATIRETAFAMPLTSPGFPKGPYRFVNREYMIITYRTDRAALEKIVPEPLELADDVVKYEFIRMPDSTGFGDYTESGQVIPVRYKGKPGVYVHSMFLNDHPPIAGGREIWGFPKKLARPVLEVEIDTLVGTLDYGRIRVATGTMGYKHHTLDHDPVLSSMHEPNFLLKIIPHVDGSPRICELVRYHMSDIVLKGAWTGPAALALYPHALAPVADLPVREVVAASHILADMTLDLGEVVHDYLA<br>    Acidiphilium cryptum         ALR        MNHPTLTIDPGAVAANWQALAARHTGEVAGVVKADAYGLGAALIAPALAAAGCRSFFVATLDEALSLRALLPAARIGVLNGCPPGEEGTMRARELRPVLGSLAACARWRGAAGDDKAPPSFLHLDTGMNRLGLDRDEAAHLIAHPTLLDGLGLTHVMTHLVAAEEPDASANAAQRERFAAFAARFPWLRTSLANSSGLFLGPGFGSDLARPGYALYGGNPTPTAANPMRPAIALAAPILQIRAIASGETVGYNGTWRAARPSRIATIGVGYADGLPRSLSNRLTARWQGKIIPVAGRLSMDLTTFDVTDHPDIAPGDVIELIGPGHDIDAFAAEAGTIGYEILTSLGPRYRRVVKSV<br>    Acidiphilium cryptum         ANMK        MHAIGLMSGTSLDGVDAAVIETDGERVFGTGRSATLFYEDGLRARLRALLDAAPGLAPDDPELLSLDWLLADRHAEAVAMVASRADVVGFHGQTILHDPDRRRTWQIGDAAHLAHKTRLPVVHDFRSADVAAGGQGAPLAPLFHAALAEALEKPLLVVNIGGVANITWLGPTGEVLACDTGPGNGPLDDWVRLHTGAAFDADGALAAAGRVDQARLTRLMDDPWFARPAPKSLDRLQFSARVAAATEGLSAADGAALLVAFTAAAIAAAPLPAPPRRVLVAGGGRHNAAIMDALRRHFAAPVEPVEAVGWDGDALEAQCFGFLAVRALRGLPLSLPETTGVPAPLAGGRIVRPGPAAE<br>    Acidiphilium cryptum         APT        MNLKDHIRGIPDFPKPGILFYDISTLIRHADAWQVAMGRLAKVVRAHHPDLLAGVESRGFILAAPLALKLGCGFIMLRKRGKLPGATVGYDYDLEYGQDRIEIQADAVQPGQRVVVVDDLLATGGTMAAGIKLLRKVGAEVPAAAALIELAFLKGRDRLDVPFEALVSYDQ<br>    Acidiphilium cryptum         ARGB        MTISDRHADAAEQARIVALALPYLRRYAGATIVVKYGGHAMGEESLAEEFGRDIALLKQVGINPVVVHGGGPQINAMLKRLAIQSTFVDGLRVTDAAMVEVVEMVLAGTVNKMVAGLINRAGALAVGICGKDGGLIRARKLTRTRRDPEGAIERALDLGFVGEPEHVDVRVIHALTGAGLIPVVAPVGVGADGATYNINADTVAGAIAGALGANRLLMLTDVAGVLDRNGKLIPDLSIAEVEALRADGTISGGMIPKVETCIEAVRQGVKGATIVDGRVPHACLLELFTEGGIGTLIHA<br>    Acidiphilium cryptum         ARLY        MTNSGADQAGTQPSRLQWGGRFAAGPASVMQAINASIGFDQRLWREDIEGSLAHAGMLAAMGIISAEDEAAIRAGLAEIAEAIAEGSFTFDPALEDIHTNIEAWLVARIGEAGRRLHTARSRNDQVATDFRLWVRNAIDSLDGQVRELMRALADRAAEHAATIMPGFTHLQTAQPVTLGHHLLAYVEMLSRDRGRLADARRRLNECPLGAAALAGTTFPIDRSMTAAALGFDRPTANSLDSVSDRDFALEFLSALSILAMHLSRFAEEIIIWTSAPYRFIRLSDAYTTGSSIMPQKRNPDAAELARAKAGRIFGALTGLLAVMKGLPLAYAKDMQEDKEPVFDAADAAELCLAAMTGMVRDMMPDVLRMRSVAGADFSTATDLADYLVRDLGLPFRTAHHVTGRIVSEAESRGLELVALPLSVMQEVEPRITEAVYDVLTIEASVAARRTLGGTAPDNVARAAARWQETLA<br>    Acidithiobacillus ferrooxidans         ACCA        MKISYLDFEQSVAELDAKVEELHALNQPGIADDISRLEVKARKELQRIYSKLGAWQTVQVARHPQRPYTMDYVQALFTEVQVLAGDRAFADDQAIIGGLARFNGQPIVWMGHQKGRDTKEKIQRNFGMPRPEGYRKALRLLRLAERFALPVFTFIDTPGAYPGIGAEERGQSEAIARNLAVMSDLAVPIICTVIGEGGSGGALAIGVGDRMLMLEYGVYSVISPEGCASILWKNAAMAAEAAETLGITARRLQGMGLVDEVLAEPLGGAHRDPEAVFALARERFAHHLQELQAMDTSKLLTTRYERLMHYGVVGAA<br>    Acidithiobacillus ferrooxidans         ACP        MDNVADRVKKVVVEQLGVNEDEVTNEASFVDDLGADSLDTVELVMALEEEFDCEIPDEEAEKIATVQQAIDYVSAHIPAKDA<br>    Acidithiobacillus ferrooxidans         ACPS        MIVGMGTDIVAVERMARLHARFGERLTERLLGPLEVADMPVEAAAGAAFLARRFAAKEATFKALGSGMTNGMRWMDVQVGHDSGGRPQLVLGGRAQQLLRGLGDGVRSWLSISDERRYAMAVVVLERGG<br>    Acidithiobacillus ferrooxidans         ALR        MTRPIVADISAAALRHNVAVVREHAPRARIMAAVKANAYGHGVTLCAPVLAEAGVDAFAVASLEEAEALHALGLERPICLLGGPFDADEVSVAAERAYLLVIHEQRQLQWLETRAADAALRLFIKVDTGMHRLGFAPERLPALFAALQRHPRWEVLGLMSHLARSDTPDDPFNRQQAGVFADAIAHVGHVTAGQHSLANSGGVLALPFTHQYWVRPGLMLYGLSPFAGRRGSEIGLQPVLSWRSAVVAIRELGPGDWLGYGAAWQAPARCRVGVVAAGYGDGYPRHLGCGAPVTVAGQTTRTLARVSMDMLFVDLTAVEADIGAPVVLMGAGGPPLESLAAELGTISYEMSCRMQMRVPRRLVS<br>    Acidithiobacillus ferrooxidans         AMPA        MEIAVQCQNPTTDNSDCVVVGIYEGGILSPAATLVDLASGGALRALLDTGDFTGDCGDTQLLYQVPGMAAARVLVLGLGSHGKVKDSQFRKAALAAARALQGARVGRASLHLLDTPVIRRSAPACAKILVQAVADAEYHFDRHKKPADAPQRPISELQLSISENDTAALAELQGAVAEAQATARAVAWTRDMANEPGNICTPTWLAEQAEAMAGRLGIKSTILGPDAMEALGMHLLLGVAHGSRQPPRLIILEYRGGAENQAPIVLVGKGITFDAGGISLKPADKMDEMKYDMCGGASALAAIQAAAELQLPLNIVTVVPASENLPDGQATKPGDIHRSMNGLSVEVVNTDAEGRLILADTLTYVERFEPDVVIDMATLTGACIIALGHQTAAVMGNHEGLVHDLIAAGKESMDRVWELPLFEEYQEQLKSPVADLSNVGGRPAGTITAACFLSRFTENYRWAHLDIAGVAWKSGEHKGATGRPVPLLVEYLLRRARQVA<br>    Acidithiobacillus ferrooxidans         ANMK        MRGLGLMSGTSADGVDAAVLDLAVAGCGGYRGVFTHAFEPALRTEVLAANGPLGVEAMAQLDRRLGACYARVAREAVDRLGPVDFIALHGQTIRHQPRGEPGFTLQIGAAADIAVATGLTVVHDFRRTDVAAGGEGAPLVPPFHQYCFQDEQPRLVLNLGGMANVTWLPGTGDPRPLLAFDCGPGNVLMDAAVELCSAGQATCDVDGRLAAAGQCDVVRLEEWLDHVFFRQAPPKSTGRETFGMPLVTRWWSSWRGSAADFLATLTALTAESVARAVRAWTPGAAEMLVFGGGAENQALMQALQDAMKETRVLHGGRHSGIPGQALEALAFAWLGSQCLLGKRLDLERVTGAQHPMILGNILPGDNWPDLLVQLSQQPEITAREGPYRALRSV<br>    Acidithiobacillus ferrooxidans         APAG        MEDQPPTEIQISVETRYLPEQSSPEQEHFAFAYQITMQNNGPQTAQLLSRHWIITDAEGHVQEVKGPGVVGEQPTLQPGQRFRYTSGSVLSTPVGSMHGTFEWVSDTGESFVVPIPAFRLAAATVFH<br>    Acidithiobacillus ferrooxidans         ARGC        MIRAGIVGGTGYTGVELLRLLLPHPEVELVAITSRAEAGQRVDEHFPNLRGHCNLSYTAPDPAILGTLDVVFFATPHGVAMDSAPKLLAEGVRIIDLGADFRLPDAEVFAQWYGMAHRAPEVLGEACYGLPEYYRTKISEARLIANPGCYPTAVILGLAPLLAEGLLQEDTLIADCKSGVSGAGRSAKVGLILPETADSVSAYGVGGHRHRPEIEAVLSDISGTPLELQFTPHLMPMIRGIHATLYGRLAQPMSDAALQDLFATRYASEPFVDVLPFGSHPATRSVRGANMCLIAVHQPRPGQVVVLSVIDNLVKGAAGQAIQNMNRLFSLAEDAGLMQIALLP<br>    Acidobacterium capsulatum         AHPD        MPIDTVMETLPGYAKDIKLNYSTLVRQNTELTPQQLWGTVVASAIATRNDTLTAAALEDGAQHLSEQALEAARIAAALMSMNNIFYRFQHLSSNEKYATMPARLRMNGMRTHGVEPVDFELWSLAVSAINGCGKCVDSHEKVLREKGAGEDLVLAAVRVASVIHAIGAVLDAEKVHQPEAAMA<br>    Acidobacterium capsulatum         AMPA        MKTTLVSTPLAQLETELLAVFATDTAPAADMANGTQASPQVQLLTRDAALDAAVKTILASGDFKAEANETLLIHAPQGMAARRLLLVGAGKQARFTVHALRKAAGSAVRAARAKNIREATLAIPQSQGLDVTATARALGHGAAVADFDPDFYRSDRKDKSLQSLTLALPEATDANAAEAGLREGIALGESQNLTRTLVNEPGNRLTPTLLGEQAKKMCAEQGLRCQVYSSEKLHELKMGSFWSVTQGSDEPPALIVMEYTPEGAAEGPVLGLVGKGITFDSGGLSLKPADSMEKMKYDMAGAAAMIGAMRAIALLKPRIKVISVICSAENMPSGKAQKPGDVQISMIGKSIEVLNTDAEGRLVLADGLAYAKQLGATHLIDAATLTGAVMVALGGVNAGVFCNDEEAWQHFEAALGQSGEKFWRLPLDEEYREMLRSPIADIKNVGGRYGGASTAAMFLKEFVGDTPWVHLDIAGTAWMDEAKPWMSSGPSGIAMPSIVEWVRSFAR<br>    Acidobacterium capsulatum         APT        MQPTHSVNCEPLKSLVRTVPDFPKPGILFYDITTLLKDRQGFAKLIDALAEHYIGKNIDLVLGIEARGFIFGPALAYRLNAGFVPVRKPRKLPARTVRVTYDLEYGSDTLEIHEDAIEPGQRIVLVDDLLATGGTMEATVKLVRQLGGEIAGLAFAVELDFLKGRERFPDLDVFSLLHYSE<br>    Acidobacterium capsulatum         ARGB        MKYVVKLGGAALENPEIFMACARAVADLVKDGHQVALVHGGGVQLTRTLKQLGKQSEFIAGLRVTDAETRDAALMVLSGRVNKSLVAALGSLGQAAMGLSGGDGLIFRARKKRTVPDLGFVGEIVASDPRWLEAIWKMNAVPVISSIALGFDGEYYNVNADEMAAACAAACRADALVFLTDVPGVRGADGTIMRWLTVDQIPVLSQTEVISGGMLPKLGACREALLNGVKRVRILPAEAAHVLPDLCSARVTDGTEVMAS<br>    Acidobacterium capsulatum         ARGR        MSKHERHNAIRELIAQRPIASQDELRRKLVRRGFDVTQATLSRDIHELRIYKGPGGYALPNGNGHDEQDDLPDVDEVMSSFGLKVKQAQNQLVLVTTAGSAQPVALAIDHEDWPEVVGTLAGDDTVLIICPDQKRATVLAERLEKIIG<br>    Acidobacterium capsulatum         ARLY        MKMWSGRFREPLDPAFDHWQRSLQFDWQLLPEEVAASKAHALALEAAGVLTAGEREALHNALDLVTSRFHAPDGSGPSWVMSNQEAEDIHHFVELQLVATVGDLGLKLHTGRSRNEQIATDLRLFVRSRAQMLQAYLGTWAEILVARAQQMGNAAMPAYTHLQRAEPVLVAHWLLAYAEMLLRDASRLEDCVRRLNYCPLGSGAVAGATLALDRGIASQALNFAAPTANSMDATSDRDFVLEFLQALTGIALHASRFAEEITLYATAEFGFVDLPEAYSTGSSAMPQKKNPDLTELVRAKVGRINGAAQAVTLLLKGLPLAYNKDMQETQEPLFQATIATHQMLHLLAKFTHALQFRTDHMQAACESGFLNAMAAATYLVHKGIPFRKAHEIVGHAVRLGLDKGCELAGLSLDELRSLSPEFGADFYDAVTLESTLDCHDVLGGTARAQVQASLEAMQRRTGDLVNARGRIDLGLSTVAEVSHADS<br>    Acidothermus cellulolyticus         ACP        MALSKEEILTGLAEIINEVAGIPTSDVQPEKSFTDDLDVDSLSMVEVVVAAEEKFGVRIPDDEVKNLKTVGDAVDYIANATVSA<br>    Acidothermus cellulolyticus         ACYP        MTDVARPDVQACEQVRLTAWVRGRVQGVGFRWWVRARALELGLTGVARNLPDSRVEVVAEGPRERCAELLELLSGVPRHGRPGFVAGVTAEWSTARGGYSGFTQA<br>    Acidothermus cellulolyticus         APT        MPPADIGELVAAHIRDIPDYPSPGIVFKDITPLLAEPAVFAAVVDALVRDYQHGTVDKVVGIEARGFIVAAPVAYHLGAGFVPMRKKGKLPYRTLETSYALEYGTATIEVHVDAFRPGDRVLVVDDVLATGGTAAAALELIDQASAKAVGLSFLIELSALGGRSRLPASMVRSLLRV<br>    Acidothermus cellulolyticus         ARAA        MTDLPYPEYECWFLTGSQHLYGEDVLSAVARQSQAIVEALNAAGLPVRLVWKPVLTDATGIRRMCSEASATDACIGVIAWMHTFSPAKAWINGLLALRKPLLHLHTQANLTLPWSTIDMDFMNLNQAAHGDREFGYVAARLAIPRKIVTGHFSDPDVVRDIAAWQRAAAGLADLRSTRLVRFGDTMRNVAVTDGDRVEAQIRLGSAIETYGVHDLGVRVDAVAESDVDALVDRYLADYDMAPELTIGGARHESLRYAAKLELALRSFLHDGRFTAFTTNFEDLGPLRQLPGIAVQRLMADGFGFGAEGDWKTALLVRAVKTMSRGLPGGTSFMEDYTYHLEPSGRLVLGAHMLEVCPTLTSATPRCEIHPLLMGGREDPVRLVFTADPAPAVIVGLCDMGDRLRLVANTADLVAPPEPLPRLPVARAVWQPHPELKTAATAWIAAGGPHHTALSTAVSAREIRDFARMAGLELVLIDEHTALDAALDRLWAIEQTRASRPW<br>    Acidothermus cellulolyticus         ARC        MASREDRDAANARVIQLQQQVERLEEEVASLRRRLAESPRGSRSLEERLAAAQATIAGLTSQNERLVATLKEARDQIVALREEVDRLAQPPSGYGIFLGRHEDGTVDVFTGGRKLRVTVSPSVDVDSLRKGQEVMLNEALNVVRAMSFERQGEVVMFKELLDGDRALVIAHADEERVVMLADSLADTPLRVGDSLLIEPRSGFAYERIPKAEVEELVLEEVPDVDYSKIGGLSAQIEQIRDAIELPYLYPELFKEHKLRPPKGVLLYGPPGCGKTLIAKAVANSLAKKVAEVTGKPEGKSFFLNIKGPELLNKYVGETERHIRLVFQRAREKASGGWPVIVFFDEMDSIFRTRGSGVSSDVENTIVPQLLSEIDGVEGLENVIVIGASNREDMIDPAILRPGRLDVKIKIERPDAEAARDIFSKYITPDLPLHPDDLAEHGGSREATVQAMIQHAVERMYAETEENRFLEVTYANGDKEVLYFKDFNSGAMIQNIVDRAKKMAIKDFLETGQKGLRIQHLMAACYDEFKENEDLPNTTNPDDWARISGKKGERIVYIRTLISGKQGTEAGRSIDTIANTGQYL<br>    Acidothermus cellulolyticus         ARGB        MSRDLTAVQAKAATLVEALPWLATFHGRTVVIKYGGHAMIDPGLAASFAADVVFLRYAGLRPVVVHGGGPQIDAMLRRLGIEPVFAAGLRVTTPETMDVVRMVLVGQVQREVVGLLNAHGPFAVGVSGEDGRLFTAVRRSAVVDGHDVDLGLVGEITEVNPTVIQSLLDDGHVPVVSSIAGGVDSDGRRCVYNVNADTAAAALAVALGAEKLVVLTDVEGLFADWPNTDEVISRLTADELEKLLPGLAAGMIPKMEACLRAVRGGVAAAHVLDGRVPHAILLEIFTDEGVGTMVVP<br>    Acidothermus cellulolyticus         ARGC        MGARVAVAGASGYSGGELLRLIAAHPELELAVATAASHAGQPIGAVHPHLVDLADQVFAPTDPAVLGDADVVFLALPHGQSAAIAATLPESALVVDIGADFRLRDPAAWAKFYGGTHAGTWTYGLPELPGARAEIAKSRRIANPGCYVTAATLALAPLFAAHLVDPDDVVIVAASGTSGAGRSLKPNLLASEVMGSLAAYKVGGVHQHTPEIEQNLSQAAGEPVTVSFTPILAPLPRGILATCTARPKAGVDGDAIRAALANAYHDEPFVHLLPPDVWPQTAATLGSNSVHLQAAFDEAARRVVVVAALDNLTKGAAGQAIQNANIALGFPETTGLTRSGVAP<br>    Acidothermus cellulolyticus         ARLY        MSEPSAAVGQRPGGESAPAHRLWGGRFTAGPAESVAALSRSIDVDWRLARYDLRASKAHARVLAAAGLLDADELAQLLAALDELDRACASGTFQPAAADEDVHTALERALLERLGPLGGKLRAGRSRNDQIATDLRLYLRDHVRLLVRRLVDLQWALVDQAERHLDLPAPGMTHLQHAQPVLFAHHLLAHVAAFARDVDRFRDWDRRAAICPLGAGALAGSSLPLDPHAVAAELGFTAPAANSIDAVSDRDFAAEFAFAAALVGVHLSRLGEEIVLWCSQEFGWVELDDAYATGSSIMPQKKNPDVAELARGKAGRLIGNLTSLLTMLKGLPLAYDRDLQEDKFPVFDSLDTLLLVLPAMTGLVATMRVKADRVAAAAPAGFSLATDVAEALVRRGVPFHEAHQAVGRLVAWCAAHGTDLSAVDDDQLAEISPYFTPENRAEVRTVLSVRGAIGARAAYGGTAPERVAEQLAAVRAALADSAAWADAAP<br>    Acidovorax citrulli         1A1D        MNLQKFPRHALTFGPTPIHPLKRLSAHLGGKVELYAKREDCNSGLAFGGNKTRKLEYLIPEAIEGGYDTLVSIGGIQSNQTRQVAAVAAHLGMKCVLVQENWVNYSDAVYDRVGNIEMSRIMGADVRLDAAGFDIGIRPSWEQAMEDVRRAGGKPFPIPAGCSEHPRGGLGFVAFAEEVRQQEEELGFQFDYIVVCSVTGSTQAGMVVGFAADGRADRVIGIDASAKPEQTREQILRIARNTAELVELGREITDADVVLDTRYGGPEYGLPNEGTLEAIRLCARQEGMLTDPVYEGKSMHGMIDMVRNGEFPAGSRVLYAHLGGVPALNAYSFLFRNG<br>    Acidovorax citrulli         ACEK        MFPRRLDSPAAYGIAQAMMDGFDRHYRLFRAESARAKHRFETSDWSGQQRAQRERIEFYDLRVKECVRRLEKEFSAGAQPMDVWHQVKLHYIGLLVGHRQPELAETFFNSVTTKILHRTHFHNDFIFVRPAISTEYLENDEPGARPTYRAYYPTPESLQETLVRVVDNFQLQGEFEDLARDAGRVAEVMRPRLGPAKWRANFQIQVLSSLFYRNKGAYLVGKVINGFQELAFALPILHGTDGRYVIDAALFGEDDLQMLFSFARAYFMVDMEIPSAYVQFLRSLMPRKPRAELYNALGLAKQGKTLFYRDFLHHLRYSSDRFRIAPGIKGMVMLVFDLPSFPFVFKVIKDYYPPQKDTTREQIKGKYLLVKQHDRVGRMADTLEYSEVAFPRERFEDDLIAEIEKFAPSQLEISDRDGDGNVEVILKHVYIERRMIPLNIYLQEAFDAGPHDARARQQMERSVVEYGNAIKDLVAANIFPGDMLWKNFGVTRNGKVVFYDYDEIEYLTDCHFRRVPAPRNEEDELSGEVWWAVGPRDVFPETFGPFLLGNDAVREVFMRHHADLLDPVFWQSHKERIQAGHMYDVFPYDPERRFGAGTGSPA<br>    Acidovorax citrulli         ACP        MSDIEARVKKIIAEQLGVEESQVTNEKAFVADLGADSLDTVELVMALEDEFGIEIPDEDAEKITTVQNAIDYANTHQKA<br>    Acidovorax citrulli         ALR        MPRPIQATIHTAALQQNLARARAAAPDARVWAVVKANAYGHGIERAFEGLRGADGFALLDLGEAERVRQLGWRGPILLLEGVFEPRDLELCSRLSLWHAVHCDEQIDWLSAHKTHAPHRVFLKMNSGMNRLGFPPARYRAAWARLNALPQVDEISFMTHFSDADGPRGIAHQIDAFCTATQDLPGERTVSNSAATLRHAGGDARVRGDWVRAGIVLYGSAADHPEHTAAHWGLAPTMTLSARIIGTQQLQAGDTVGYGSSFTADRPMTIGVVACGYADGYPRHAPTGTPVLVNGVRTRTVGRVSMDMITVDLTPLHEAGVEAGVGSEATLWGRASCGAVLCIDEVAQAAGTVGYELMCALAPRVPVVVD<br>    Acidovorax citrulli         APAH        MALYCIGDIQGCDAALQRLLDTVDFSPSRDTVYLLGDLVNRGPQSAEVLRRCARGGDSLRALLGNHDLHLLAAAHGARKPSRRDTLAGVLQAPDRDALLDWLKRQPLAREHRLAGERLLMVHAGVLPSWTADATLEHAAEVQARLQGDALPEFLQAMYGNAPDRWSDELSGMDRWRAIVNALTRLRFCTPDGRMDFDSSEAADAAPDGLVPWFDAPGRRTRGTLVAFGHWSTLGWMNRPDLLALDTGCVWGGCLSAVRFGATLAEREHISVRCEQAQAPG<br>    Acidovorax citrulli         APT        MQDLSVNEYLRQYIRTVPDWPAAGVQFRDITPLLQDPKVFRVLIDAFVHRYMDRALRPDVVAGLDARGFILGAVVAYELNVGFVPIRKKGKLPFTTVEETYELEYGSATVELHADAVKPGDRVLLIDDLIATGGTMMAGRRLLERLGATVTEGAAIVDLPELGGSSRLRESGLPLYTLVDFAGH<br>    Acidovorax citrulli         ARGB        MTDLQSIAPRDKAEILAQALPYIRKFHGKTIVIKYGGNAMTDPALQADFAEDVVLLKLVGMNPVVVHGGGPQIETALKRLGKQGHFIQGMRVTDAETMEVVEWVLAGEVQQDIVGLINQAGGKAVGLTGRDGGLIRARKLKMVDKDDPSKEHDIGQVGDIVSIDPSVVKALQDDAFIPVISPLGFGEENESYNINADVVASKLATVLKAEKLMMLTNIPGVLNKSGALIPELTAREIDELVVDGTISGGMLPKIAGAIDAAKSGVNAVHIVDGRVPHAMLLEILTEQAYGTMIRSH<br>    Acidovorax citrulli         ARLY        MSSSPTAQPSQDQLATKAQAWSALFSEPMSDLVKRYTSSVFFDKRLWQADIAGSLAHADMLAAQGIISADDHAAIRRGMATITQEIESGAFEWKLDLEDVHLNIEARLTQLVGDAGKRLHTGRSRNDQVATDVRLWLRGEIDMIADLLKELQRSLVDVAEQNVEVILPGFTHLQVAQPVSFAHHMLAYVEMFARDAERMQDVRRRTNVLPLGSAALAGTTYPLDRERVARTLGMEGVCQNSLDAVSDRDFAIEFTAAASLCMVHVSRLSEELIIWMSQNFGFIRIADRFTTGSSIMPQKKNPDVPELARGKTGRVVGHLMGLITLMKGQPLAYNKDNQEDKEPLFDTVDTLKDTLRIFAEMVGGQRDPATGRKEGGITVNPQAMEQAAQKGYATATDLADYLVKKGLPFRDAHETVAHAVKAATSHGVDLSELPLTVLQGFHPAIGKDVFDALSLRGSLNARNTLGGTAPAQVRTQLARHRARLG<br>    Acidovorax ebreus         1A1D        MNLQKFPRHPLTFGPTPIQPLKRLSQHLGGKVELYAKREDCNSGLAFGGNKTRKLEYLIPEAIAGGYDTLVSIGGIQSNQTRQVAAVAAHLGLKCVLVQENWVNYSDAVYDRVGNIEMSRILGADVRLDAAGFDIGIRPSWEQAMADVKQAGGKPFPIPAGCSEHPYGGLGFVGFAEEVRQQEKELGFRFDYIVVCSVTGSTQAGMVVGFAADGRARRVIGIDASATPEKTHAQILRIAQSTADLVELGQDITAEDVVLDTRYGGPEYGLPNEGTLEAIRLCARQEGMLTDPVYEGKSMHGMIERVRNGEFPAGSKVLYAHLGGVPALNAYSFLFRNG<br>    Acidovorax ebreus         ACP        MSDIEARVKKIIAEQLGVEESQVTNEKAFVADLGADSLDTVELVMALEDEFGIEIPDEDAEKITTVQNAIDYANTHQKA<br>    Acidovorax ebreus         ACPS        MIYGIGTDICDVRRIRASLARHGDRFAEKVLADGELATWRARSARWPERGVRYLATRFSAKEAFSKAIGLGLHMPMTWRHCEVANLPSGQPTIVLHGALQDWFAARGLRCHLSVTDESDYAASFCVVEKD<br>    Acidovorax ebreus         ALR        MPRPILATIHPAAVHHNLERARRAAPDARVWAVVKANAYGHGIERVFEGLRAADGFALLDLAEAERVRALGWRGPILLLEGVFEPRDLELCSRLGLWHAVHCDAQIDWLAAHKTQVPHRVFLKMNSGMNRLGFTPERYRSAWARLNALPQVDEISCMTHFSDADGPRGIAHQVQAFQTATRDLPGERCIANSAALLRHGGDAQVRLDWVRAGIVLYGSAPDHPERRAADWDLQPTMTLASRIIGVQQLQAGDTVGYGSRFTAQGPLGIGVVACGYADGYPRHCDTGTPVLVNGVRTRTIGRVSMDMLAVDLTPVPGAGLGAEVTLWGRAANGAVLPIDEVAQAGGTIGYELMCALAPRVPVVVQD<br>    Acidovorax ebreus         AMPA        MNFELKTLSLAAAAAYKCDLLVVLVPEGFLPGSDVLSTLTAHALKQGDLEVKPGKLLQCYAPAGVAARRVVLLGCGAADAHAVRQAMQALAPSFKLPSVKRVALVFGTSAQRGAIGAAVRAVSDGSYVYTATKTKAEPRALLRVTLGVPDAGAAQPEFATATAVAAGVEFAREWANRPANHATPTLVANAAKTLAKYPGVQCQVLGPTEVAKLGMGAFLAVAQGSDQPLRLVELRYNGAARTQAPVVLVGKGITFDTGGISLKPAAEMDEMKFDMGGAASVLGVFRALAELRPAINVVGLIPACENMPDGKAVKPGDVVTSMSGQTIEILNTDAEGRLVLCDALTYAARFKPAAVVDIATLTGACVVALGGLRSGLFASDDLLAEQLLSAGDAALDPCWRMPLDDEYAEGLKSNFADMANVAGRAGGSITAAKFLQRFVGDTPWAHLDIAGTAWKGGAAKGATGRPVGLLVHYLLERATVSTVKPKQKTRSRKSVA<br>    Acidovorax ebreus         APAH        MALYCVGDIQGCDDAFERLLATIGFSPSRDTLYVLGDLVNRGPHSAAVLRRCITLGDSVRPLLGNHDLHLLAAAYGTRRPSRRDTLQDILLAPDRDEMLEWLRHQPLARRVHHGGGDLLMVHAGVLPQWTAEETLAYAGEVEAVLQSREFAGFLQQMYGNSPDLWSPELQGTDRLRVIVNALTRMRFCSPEGRMDFESTESASEAPPGLVPWFDAPGRRTLNTLIAFGHWSTLGWLDRADVLGLDTGCVWGGCLSAVRFGTTLADRERCHVECPQAQMPGA<br>    Acidovorax ebreus         APT        MTALSVNDYLRQHIRTVPDWPAPGVQFRDITPLLQNPKVFRVLIDAFVHRYMDRALRPDVVAGLDARGFILGAVVAYELNVGFVPIRKKGKLPFTTVEETYELEYGSATVELHADAVQPGDRVLLIDDLIATGGTMMAGKKLLEKLGATVMEGAAIVDLPELGGSARLRASGLPLYTLVDFSGH<br>    Acidovorax ebreus         ARLY        MSSSSASQPSHDQLATKAQAWSALFSEPMSDLVKRYTSSVFFDKRLWQADIAGSLAHAEMLAAQGIISAQDHADIQKGMAQITQEIASGAFEWKLDLEDVHLNIEARLTQLVGDAGKRLHTGRSRNDQVATDVRLWLRGEIDLIEGLLSELQLSLVEVAEQNVEVILPGFTHLQVAQPVSFAHHLLAYVEMFARDAERMRDVRRRVNVLPLGSAALAGTTYPLDRERVAKTLGMEGVCQNSLDGVSDRDFAIEFTAAASLCMVHVSRLSEELIIWMSQNFGFIKIADRFTTGSSIMPQKKNPDVPELARGKTGRVVGHLMGLITLMKGQPLAYNKDNQEDKEPLFDTVDTLKDTLRIFAEMIGGQMNPATGCKDGGITVNAEAMRAAALKGYATATDLADYLVKKGLPFRDAHETVAHAVKAAVSHSVDLSELPLAVLQGFHPAIEKDVFDALSLQGSLNARNTLGGTAPAQVRTQLARHRARLSA<br>    Acidovorax sp.         ACDH        MTQKIKCALIGPGNIGTDLLAKLQRSPVLEPVWMVGIDPESDGLKRAREMGIKTTHEGVDGLIPHMKADGVQIVFDATSAYVHAENSRKVNAQGALMIDLTPAAIGPFCVPPVNLKEHVGKAEMNVNMVTCGGQATIPMVAAVSRVQPVAYGEIVATVSSRSAGPGTRKNIDEFTRTTAGAIEKVGGAQKGKAIIIINPADPPLIMRDTVHCLVEGEPDKEAITRSIHDMLAEVQKYVPGYKLVNGPVFDGNRVSVFLEVEGLGDYLPKYAGNLDIMTAAAARTAEMFAEEILAGKLTLQAA<br>    Acidovorax sp.         ACEK        MFPERLDAPQAYDIAKAMMDGFNRHYRLFRAESARAKHRFETADWHGQQRAQRERIEFYDLRVRECVRRLDKEFNAGALPMDVWHQIKLHYIGLMVNHLQPELAETFFNSVTTKILHRTHFHNDFIFVRPAVSTEYIESDDPGARPTYRAYYPSRDNLHETVVRIVEHCALQRDFENLPRDAGHVVQALQQRLGAVKLRTNFQVQVLSSLFFRNKGAYLVGKVINGYNELPFALPILHGEDGRLLIDAVLFGENDLQMLFSFARAYFMVDMEIPSAYVQFLRSLMPRKPRAELYTALGLAKQGKTLFYRDFLHHLRYSTDKFRIAPGIKGMVMLVFDLPSFPYVFKLIKDQFPAPKDTTREQVQGKYLLVKQHDRVGRMADTLEYSLVAFPRERFSDELIEEIRRHAPSQIEISDRDGDGRQEVIIAHLYIERRMIPLNIHLQECFDTGLDKPEARSALEHAVTEYGNAIKDMVAANIFPGDMLWKNFGITRNGKVVFYDYDEIEYLTDCNFRRVPPPRCEEDEVSGEVWWPVGPHDVFPETFGPFLLGNDSVREAFMRHHADLLDVEFWQSHKERIQAGHLYDVFPYDSARRFRCASPSSFQTQGDST<br>    Acidovorax sp.         ACP        MSDIEARVKKIIAEQLGVEESQVTNEKAFVADLGADSLDTVELVMALEDEFGIEIPDEDAEKITTVQNAIDYANTHQKA<br>    Acidovorax sp.         ACPS        MIYGIGTDICDVRRIRASLARHGDRFAEKVLADGELATWRARSARWPERGVRYLATRFSAKEAFSKAIGLGLHMPMTWRHCEVANLPSGQPTIVLHGALQDWFAARGLRCHLSVTDESDYAASFCVVEKD<br>    Acidovorax sp.         ACYP        MAESAASITRHLRIHGLVQGVYYRKSMTEAARRLGVQGWVRNRQDGTVEALASGAAPAVQALIDWAHEGPPAARVERVEVAEAPACDAQGFEQRETV<br>    Acidovorax sp.         ALR        MPRPILATIHPAAVHHNLERARRAAPDARVWAVVKANAYGHGIERVFEGLRAADGFALLDLAEAERVRALGWRGPILLLEGVFEPRDLELCSRLGLWHAVHCDAQIDWLAAHKTQVPHRVFLKMNSGMNRLGFTPERYRSAWARLNALPQVDEISCMTHFSDADGPRGIAHQVQAFQAATQDLPGERCIANSAALLRHGGDAQVRLDWVRAGIVLYGSAPDHPERRAADWDLQPTMTLASRIIGVQQLQAGDTVGYGSRFTAQGPLGIGIVACGYADGYPRHCDTGTPVLVNGVRTRTIGRVSMDMLAVDLTPVPGAGLGAEVTLWGRAANGAVLPIDEVAQAGGTIGYELMCALAPRVPVVVQD<br>    Acidovorax sp.         AMPA        MNFELKTLSLAASAAHKCDLLVVLVPEGFLPGSDVLSTMVAHALKQGDLEVKPGKLLQCYAPAGVAARRVVLLGCGAADAHAVRQAMQALSPSFKLPSVKRVALVFGTSAQRGAIGAAVRAVADGSYVYTATKTKAEPRALLRVTLGVPNAGAAQPEFATATAVAAGVEFAREWANRPANHATPTLVANAAKTLAKYPGVQCQVLGPAEVAKLGMGAFLAVAQGSDQPLRLVELRYNGAARTQAPVVLVGKGITFDTGGISLKPAAEMDEMKFDMGGAASVLGVFRALAELRPAINVVGLIPACENMPDGKAVKPGDVVTSMSGQTIEILNTDAEGRLVLCDALTYAARFKPAAVVDIATLTGACVVALGGLRSGLFASDDLLAEQLLSAGDAALDPCWRMPLDDEYAEGLKSNFADMANVAGRAGGSITAAKFLQRFVGDTPWAHLDIAGTAWKGGAAKGATGRPVALLVHYLLDQATVSTVKPKQKTRSRKSVA<br>    Acidovorax sp.         APAG        MPKYQFQVQVQVQPQYLPDQSAPDEGVYSFAYTITITNTGDVTAQLISRHWIISDANGHTEQVKGLGVVGQQPLLKPGEAFQYTSGCRLRTASGSMHGSYFCVAEDGEPFTCPIELFVLEAFSPGQPGQPMAARVLH<br>    Acidovorax sp.         APAH        MALYCVGDIQGCDDAFERLLATIGFSPSRDALYVLGDLVNRGPDSAAVLRRCITLGDSVRPLLGNHDLHLLAAAYGTRRPSRRDTLQGILLAPDRDEMLEWLRHQPLARRVHHGGGDLLMVHAGVLPQWTAEETLAYAGEVEAVLQSREFAGFLQQMYGNSPDLWSPELQGTDRLRVIVNALTRMRFCSPEGRMDFESTESASEAPPGLVPWFDAPGRRTLNTLIAFGHWSTLGWLDRADVLGLDTGCVWGGCLSAVRFGTTLADRERYHVECPQAQMPGA<br>    Acidovorax sp.         APT        MTALSVNDYLRQHIRTVPDWPAPGVQFRDITPLLQNPKVFRVLIDAFVHRYMDRALRPDVVAGLDARGFILGAVVAYELNVGFVPIRKKGKLPFTTVEETYELEYGSATVELHADAVQPGDRVLLIDDLIATGGTMMAGKKLLEKLGATVMEGAAIVDLPELGGSERLRASGLPLYTLVDFSGH<br>    Acidovorax sp.         ARGB        MTDLLSIAPRDKAEILAQALPYIRKFHGKTMVIKYGGNAMTDPALQADFAEDVVLLKLVGMNPVVVHGGGPQIEQALNRLGKKGEFIQGMRVTDAETMEVVEWVLAGEVQQDIVGLIHQAGGKAVGLTGRDGGMIRAKKLKMVDNKDPSVEHDVGQVGDIVAIDPSVVKALQDDAFIPVISPIGFGEENESYNINADVVASKLATVLQAEKLVLLTNTPGVLDKEGKLLTDLTAREIDALFADGTISGGMLPKISGALDAAKAGVNAVHIIDGRVPHSMLLEILTDQAYGTMIRSH<br>    Acidovorax sp.         ARLY        MSSSSASQPSHDQLATKAQAWSALFSEPMSDLVKRYTSSVFFDKRLWQADIAGSLAHAEMLAAQGIISAQDHADIQKGMAQITQEIASGAFEWKLDLEDVHLNIEARLTQLVGDAGKRLHTGRSRNDQVATDVRLWLRGEIDLIEGLLSELQLSLVEVAEQNVEVILPGFTHLQVAQPVSFAHHLLAYVEMFARDAERMRDVRRRVNVLPLGSAALAGTTYPLDRERVAKTLGMEGVCQNSLDGVSDRDFAIEFTAAASLCMVHVSRLSEELIIWMSQNFGFIKIADRFTTGSSIMPQKKNPDVPELARGKTGRVVGHLMGLITLMKGQPLAYNKDNQEDKEPLFDTVDTLKDTLRIFAEMIGGQMNPATGCKDGGITVNAEAMRAAALKGYATATDLADYLVKKGLPFRDAHETVAHAVKAAVSHSVDLSELPLAVLQGFHPAIEKDVFDALSLQGSLNARNTLGGTAPAQVRTQLARHRARLSA<br>    Acinetobacter baumannii        AAC6        MNIMPISESQLSDWLALRCLLWPDHEDVHLQEMRQLITQAHRLQLLAYTDTQQAIAMLEASIRYEYVNGTQTSPVAFLEGIFVLPEYRRSGIATGLVQQVEIWAKQFACTEFASDAALDNQISHAMHQALGFHETERVVYFKKNIG<br>    Acinetobacter baumannii        AACC2        MHTRKAITEAIRKLGVQTGDLLMVHASLKAIGPVEGGAETVVAALRSAVGPTGTVMGYASWDRSPYEETLNGARLDDKARRTWPPFDPATAGTYRGFGLLNQFLVQAPGARRSAHPDASMVAVGPLAETLTEPHELGHALGKGSPVERFVRLGGKALLLGAPLNSVTALHYAEAVADIPNKRWVTYEMPMLGRNGEVAWKTASEYDSNGILDCFAIEGKPDAVETIANAYVKLGRHREGVVGFAQCYLFDAQDIVTFGVTYLEKHFGATPIVPAHEAAQRSCEPSG<br>    Acinetobacter baumannii        ABAI        MNIIAGFQNNFSEGLYTKFKSYRYRVFVEYLGWELNCPNNEETRIQFDKVDTAYVVAQDRESNIIGCARLLPTTQPYLLGEIFPQLLNGMPIPCSPEIWELSRFSAVDFSKPPSSSSQAVSSPISIAILQEAINFAREQGAKQLITTSPLGVERLLRAAGFRAHRAGPPMMIDGYSMFACLIDV<br>    Acinetobacter baumannii         4HYPE        MKTVKILDSHTGGEPTRLVLEGFPDLGTGDIESRRKILSEQYDHFRRATMLEPRGNDVLVGALLCKPVNPKASAGVIFFNNTGYLGMCGHGTIGLVASLAHLGKIQVGTHLIETPVGDVEATLHEDHSVSVRNVPAYRYKKAVEVNVEKYGKVTGDIAWGGNWFFLINDHGQRVASDNLDQLTEYAWTVRQALTAQGITGKDGQEIDHIELFASDTEADSKNFVLCPGKAYDRSPCGTGTSAKIACLAADGKLEPGKLWKQASIIGSQFIASYEQAGEYVIPTIRGEAYMSAEATLFMDENDPFAWGIQL<br>    Acinetobacter baumannii         4HYPE        MKTVKILDSHTGGEPTRLVLEGFPDLGTGDMESRRKILSEQYDHFRRATMLEPRGNDVLVGALLCKPVNPKASAGVIFFNNTGYLGMCGHGTIGLVASLAHLGKIQVGTHLIETPVGDVEATLHEDHSVSVRNVPAYRYKKAVEVDVEKYGKVTGDIAWGGNWFFLINDHGQRVASDNLDQLTEYAWTVRQALTAQGITGKDGQEIDHIELFASDTEADSKNFVLCPGKAYDRSPCGTGTSAKIACLAADGKLEPGKLWKQASIIGSQFIASYEQAGEYVIPTIRGEAYMSAEATLFMDENDPFAWGIQL<br>    Acinetobacter baumannii         ACCA        MKKATQSKAWTTVQIARHPERPQFLDYVGEIFTEFDALHGDRLFGDDGAMVGGLARFDGQPVMVIGQHRGRSTREKLKHNFGMCNPEGYRKSQRLLDMAERFNLPVFTFIDTMGAYPGVGAEERGQAEAIATSLAQLSSLKVPVIATVLGEGGSGGALGIGVADRVIMLSHSIYSVISPEGCASILWKTAEKAAQASEALGLTADKLQSLGIVEYVVDEGEGAHLDPERVMQNLKVVLKQALDELLPMDANERCEARYQRLMKFGSENLGMAS<br>    Acinetobacter baumannii         ACCA        MKKATQSKAWTTVQIARHPERPQFLDYVGEIFTEFDALHGDRLFGDDGAMVGGLARFDGQPVMVIGQHRGRSTREKLKHNFGMCNPEGYRKSQRLLDMAERFNLPVFTFIDTMGAYPGVGAEERGQAEAIATSLAQLSSLKVPVIATVLGEGGSGGALGIGVADRVIMLSHSIYSVISPEGCASILWKTAEKAAQASEALGLTADKLQSLGIVEYVVDEGEGAHLDPERVMQNLKVVLKQALDELLPMDANERCEARYQRLMKFGSENLGVAS<br>    Acinetobacter baumannii         ACCD        MNQEVKSGKVLSPSTPWTQRPVPGIEVADEQQTLKATFTEPTIECPECHALVTRTAISFNAYVCPQCDEHLRMKARDRLNWFFDNVVAELGQEFSAKDPLKFVDSKPYPDRMREAQTKTGETEALIAMQGNLNGVDMIACAFEFDFMAGSMGTVVGDRFVKAAELAIEKRQPLICFAASGGARMQEGMLSLMQMARTSAAIQKLKDTGLPYIVVLTHPVYGGVTASLAMLGDIHIAEPKAMIGFAGKRVIEQTVRETLEEPFQRAEYLLDHGVVDQIVHRHALRDTVSRLVSKLMNLP<br>    Acinetobacter baumannii         ACCD        MNQEVKSGKVLSPSTPWTQRPVPGIEVADEQQTLKATFTEPTIECPECHALVTRTAISFNAYVCPQCDEHLRMKARDRLNWFFDNVVAELGQEFSAKDPLKFVDSKPYPDRMREAQTKTGETEALIAMQGNLNGVDMIACAFEFDFMGGSMGTVVGDRFVKAAELAIEKRQPLICFAASGGARMQEGMLSLMQMARTSAAIQKLKDAGLPYIVVLTHPVYGGVTASLAMLGDIHIAEPKAMIGFAGKRVIEQTVRETLEEPFQRAEYLLDHGVVDQIVHRHALRDTVSRLVSKLMNLP<br>    Acinetobacter baumannii         ACP        MSDIEQRVKQAVAEQLGLKAEEIKNEASFMDDLGADSLDLVELVMSFENDFDITIPDEDSNEITTVQSAIDYVTKKLG<br>    Acinetobacter baumannii         ALLC        MATLHAPAFELPEILNTKTNLADARIGAQVIECSDDFFAEAKRMLQFEAPIFVEDKFDDHGKWMDGWESRRKRHAGYDWCIVKLGVSGKISALDIDTTFFTGNYPASASLEACYAPNGDLTGVTWQSILENTELGPSQHHIFMVNNDAIFTHIRLNIFPDGGIARLRVYGDVHIQVTDHEQTLDLLALENGGRVIAYS</td><td class="s1" dir="ltr">The sequences in these data files were extracted from the UniProt database: http://www.uniprot.org.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br><br>def classify_protein_length(filename):<br>    short_count = 0<br>    medium_count = 0<br>    long_count = 0<br>    with open(filename, &#39;r&#39;) as f:<br>        # Skip header line<br>        next(f)<br>        for line in f:<br>            parts = line.rstrip().split(&quot;\t&quot;)<br>            sequence = parts[2]<br>            length = len(sequence)<br>            if length &lt; 500:<br>                short_count += 1<br>            elif length &gt;= 500 and length &lt;= 999:<br>                medium_count += 1<br>            else:<br>                long_count += 1<br>    return (short_count, medium_count, long_count)<br><br>def main():<br>    viruses_file = sys.argv[1]<br>    bacteria_file = sys.argv[2]<br>    fungi_file = sys.argv[3]<br>    output_file = sys.argv[4]<br>    with open(output_file, &#39;w&#39;) as f:<br>        short_count, medium_count, long_count = classify_protein_length(viruses_file)<br>        f.write(&quot;Viruses\tshort\t{}\n&quot;.format(short_count))<br>        f.write(&quot;Viruses\tmedium\t{}\n&quot;.format(medium_count))<br>        f.write(&quot;Viruses\tlong\t{}\n&quot;.format(long_count))<br>        short_count, medium_count, long_count = classify_protein_length(bacteria_file)<br>        f.write(&quot;Bacteria\tshort\t{}\n&quot;.format(short_count))<br>        f.write(&quot;Bacteria\tmedium\t{}\n&quot;.format(medium_count))<br>        f.write(&quot;Bacteria\tlong\t{}\n&quot;.format(long_count))<br>        short_count, medium_count, long_count = classify_protein_length(fungi_file)<br>        f.write(&quot;Fungi\tshort\t{}\n&quot;.format(short_count))<br>        f.write(&quot;Fungi\tmedium\t{}\n&quot;.format(medium_count))<br>        f.write(&quot;Fungi\tlong\t{}\n&quot;.format(long_count))<br><br>if __name__ == &quot;__main__&quot;:<br>    main()</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">5</td><td class="s1" dir="ltr">The data files are quite large. Thus the length of the full prompt exceeded ChatGPT&#39;s limit, and the spreadsheet program could not handle such long text. I shortened it to exclude the data files.<br><br>From the beginning, ChatGPT figured out how to do the calculations correctly. However, the output file was not quite right. I tried explaining what to do, but it didn&#39;t get it quite right and sometimes did quite poorly. However, when I modified the prompt to include the expected output (which had been truncated), it worked.</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R154" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">155</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">01 - Print a cell&#39;s value</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>For this exercise, you will use clinical data for endometrial cancer patients. The data file is tab delimited.<br><br>The rows represent different patients; the columns represent different data points about these patients. For example, there is a column that indicates the country of origin for each patient, their tumor grade, etc. On the far right, you will see data representing the patients&#39; tumor size and number of pregnancies.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be the path to the data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names. Use the &quot;sep&quot; keyword argument to indicate that the values are tab-separated.<br><br>Print the data value from the first row and first column (not counting row names and column names).<br><br>Hint(s):<br><br>    When using &quot;pandas&quot; to read a file, you do not need to use the &quot;with&quot; statement. The pandas package does that for use behind the scenes.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br><br># Get the file path from command line argument<br>file_path = sys.argv[1]<br><br># Read the data file into a pandas DataFrame<br>data = pd.read_csv(file_path, index_col=0, sep=&#39;\t&#39;)<br><br># Print the value from the first row and first column<br>print(data.iloc[0, 0])</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R155" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">156</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">02 - Print another cell value</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a string and will be the name of the data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names.  Also use the &quot;sep&quot; keyword argument to indicate that the values are tab-separated.<br><br>Print the data value from row 20 (index = 19) and column 4 (index = 3), not counting row names and column names.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br><br># Get the file path from command line argument<br>file_path = sys.argv[1]<br><br># Read the data file into a pandas DataFrame<br>data = pd.read_csv(file_path, index_col=0, sep=&#39;\t&#39;)<br><br># Print the value from row 20 (index = 19) and column 4 (index = 3)<br>print(data.iloc[19, 3])</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R156" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">157</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">03 - Count patients</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a path to one of above-mentioned data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names (patient identifiers). Also use the &quot;sep&quot; keyword argument to indicate the the values are tab separated.<br><br>Your script should print the number of patients in this cohort. (There is more than one way to do this.)<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br><br># Get the file path from command line argument<br>file_path = sys.argv[1]<br><br># Read the data file into a pandas DataFrame<br>data = pd.read_csv(file_path, index_col=0, sep=&#39;\t&#39;)<br><br># Print the number of patients in the cohort<br>print(len(data.index))</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.<br><br>The first solution was valid, but it included additional text in the output. After I asked it to exclude this text, ChatGPT&#39;s solution passed.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R157" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">158</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">04 - Extract stage column</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a path to one of above-mentioned data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names (patient identifiers). Also use the &quot;sep&quot; keyword argument to indicate the the values are tab separated.<br><br>Often when we are working with a DataFrame, we want to extract a subset of data for convenience. You can think of this as similar to slicing a Python list. When you slice a list, you get back a list. But when you slice a DataFrame, you get back a DataFrame (or a Series if is only one row or column). For this problem, extract the &quot;FIGO_stage&quot; column and print the first 5 rows of data.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br><br># Get the file path from command line argument<br>file_path = sys.argv[1]<br><br># Read the data file into a pandas DataFrame<br>data = pd.read_csv(file_path, index_col=0, sep=&#39;\t&#39;)<br><br># Extract the &quot;FIGO_stage&quot; column from the DataFrame<br>figo_stage = data[&#39;FIGO_stage&#39;]<br><br># Print the first 5 rows of data from the &quot;FIGO_stage&quot; column<br>print(figo_stage.head(5))</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R158" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">159</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">05 - Extract multiple columns</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a path to one of above-mentioned data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names (patient identifiers). Also use the &quot;sep&quot; keyword argument to indicate the the values are tab separated.<br><br>Extract the FIGO_stage, BMI, and Age columns and print the first 5 rows of that DataFrame.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br><br># Reading the file into a DataFrame<br>filename = sys.argv[1]<br>df = pd.read_csv(filename, index_col=0, sep=&#39;\t&#39;)<br><br># Extracting the columns &quot;FIGO_stage&quot;, &quot;BMI&quot;, and &quot;Age&quot;<br>subset = df[[&quot;FIGO_stage&quot;, &quot;BMI&quot;, &quot;Age&quot;]]<br><br># Printing the first 5 rows of the subset<br>print(subset.head())</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R159" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">160</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">06 - Calculate summary statistics</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a path to one of above-mentioned data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names (patient identifiers). Also use the &quot;sep&quot; keyword argument to indicate the the values are tab separated.<br><br>Print the minimum, mean (average), and maximum age of the patients. Print one per line, in that order, and round the numbers to 1 decimal place (using string formatting).<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br><br># read data file into a DataFrame<br>file_path = sys.argv[1]<br>data_frame = pd.read_csv(file_path, index_col=0, sep=&quot;\t&quot;)<br><br># calculate the minimum, mean, and maximum age<br>min_age = data_frame[&quot;Age&quot;].min()<br>mean_age = data_frame[&quot;Age&quot;].mean()<br>max_age = data_frame[&quot;Age&quot;].max()<br><br># round the numbers to 1 decimal place<br>min_age = round(min_age, 1)<br>mean_age = round(mean_age, 1)<br>max_age = round(max_age, 1)<br><br># print the results<br>print(min_age)<br>print(mean_age)<br>print(max_age)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R160" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">161</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">07 - Summarize stages</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a path to one of above-mentioned data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names (patient identifiers). Also use the &quot;sep&quot; keyword argument to indicate the the values are tab separated.<br><br>The FIGO_Stage column indicates the stage of each patient&#39;s tumor. Print a DataFrame that shows the number of patients at each stage, sorted in decreasing order by the number of patients in each stage.<br><br>Hint(s):<br><br>    There is a function in the &quot;pandas&quot; module that makes this easy.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br><br># Get the file path from the command line arguments<br>file_path = sys.argv[1]<br><br># Read the data file into a DataFrame<br>df = pd.read_csv(file_path, sep=&#39;\t&#39;, index_col=0)<br><br># Get the count of patients in each FIGO stage<br>figo_stage_counts = df[&#39;FIGO_stage&#39;].value_counts()<br><br># Sort the count of patients in decreasing order<br>figo_stage_counts = figo_stage_counts.sort_values(ascending=False)<br><br># Print the resulting DataFrame<br>print(figo_stage_counts)<br></td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R161" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">162</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">08 - Extract patient subset</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a path to one of above-mentioned data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names (patient identifiers). Also use the &quot;sep&quot; keyword argument to indicate the the values are tab separated.<br><br>Create a DataFrame that contains information for patients with the following Patient_ID values: C3L-00006, C3L-00008, C3L-00032. For these patients, extract information for the BMI, Age, Race, Tumor_Site, and Tumor_Size_cm columns. You should be able to do this filtering in one line of code using &quot;.loc&quot; (rather than &quot;.iloc&quot;). Print this DataFrame.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br><br># read the input file<br>file_path = sys.argv[1]<br>df = pd.read_csv(file_path, sep=&#39;\t&#39;, index_col=0)<br><br># extract information for the required patients<br>patients = [&#39;C3L-00006&#39;, &#39;C3L-00008&#39;, &#39;C3L-00032&#39;]<br>df_filtered = df.loc[patients, [&#39;BMI&#39;, &#39;Age&#39;, &#39;Race&#39;, &#39;Tumor_Site&#39;, &#39;Tumor_Size_cm&#39;]]<br><br># print the filtered DataFrame<br>print(df_filtered)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R162" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">163</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">09 - Count by country</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a path to one of above-mentioned data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names (patient identifiers). Also use the &quot;sep&quot; keyword argument to indicate the the values are tab separated.<br><br>How many patients are from the United States? Please write code to determine this and print the answer.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import pandas as pd<br>import sys<br><br># Reading the data file into a DataFrame<br>file_path = sys.argv[1]<br>data = pd.read_csv(file_path, sep=&#39;\t&#39;, index_col=0)<br><br># Counting the number of patients from the United States<br>us_patients = data[data[&#39;Country&#39;] == &#39;United States&#39;].shape[0]<br><br># Printing the number of patients<br>print(us_patients)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.<br><br>The first solution was valid, but it included additional text in the output. After I asked it to exclude this text, ChatGPT&#39;s solution passed.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R163" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">164</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">10 - Summarize diabetes diagnoses</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a path to one of above-mentioned data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names (patient identifiers). Also use the &quot;sep&quot; keyword argument to indicate the the values are tab separated.<br><br>Now you will practice selecting parts of a DataFrame that meet specific condition(s). For example, suppose we wanted to create a DataFrame of patients who have a value of &#39;FIGO grade 2&#39; in the &quot;Histologic_Grade_FIGO&quot; column and/or patients who are over 70 years old. To accomplish this, you can use comparison operators like &quot;==&quot;, ` 70]<br><br>We can combine multiple conditions using logical operators. Sadly, these operators are different in pandas than in regular Python. The &quot;&amp;&quot; character represents &quot;and&quot; and &quot;|&quot; represents &quot;or&quot;. When specifying multiple conditions, you also need to put a set of parentheses around each condition. For example, we could identify all grade 2 patients over 70 using this code:<br><br>    grade2Over70 = clinical_dataframe.loc[(clinical_dataframe[&#39;Histologic_Grade_FIGO&#39;] == &#39;FIGO grade 2&#39;) &amp; (clinical_dataframe[&#39;Age&#39;] &gt; 70)]<br><br>Please write code to print an answer to the following question:<br><br>How many patients from Ukraine are childless women over 50? (A value of &quot;None&quot; in the &quot;Num_full_term_pregnancies&quot; column indicates when a woman had no children.)<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import pandas as pd<br>import sys<br><br># Reading the data file into a DataFrame<br>filename = sys.argv[1]<br>df = pd.read_csv(filename, sep=&#39;\t&#39;, index_col=0)<br><br># Selecting patients from Ukraine who are childless women over 50<br>childless_women_over_50 = df.loc[(df[&#39;Country&#39;] == &#39;Ukraine&#39;) &amp; (df[&#39;Num_full_term_pregnancies&#39;] == &#39;None&#39;) &amp; (df[&#39;Age&#39;] &gt; 50)]<br><br># Printing the number of such patients<br>print(len(childless_women_over_50))</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R164" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">165</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">11 - Perform complex queries - A</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a path to one of above-mentioned data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names (patient identifiers). Also use the &quot;sep&quot; keyword argument to indicate the the values are tab separated.<br><br>For many of the patients, it was recorded whether or not they had been diagnosed with diabetes. For a few patients, the value is missing or &quot;Unknown&quot;. Please write code to examine all of the patients for whom diabetes status is known and calculate the percentage of patients who were diabetic. Please print the answer, rounded to 2 decimal places, followed by a percent sign.<br><br>Hint(s):<br><br>    The answer is between 25.00% and 30.00%.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br><br>def main():<br>    filename = sys.argv[1]<br>    df = pd.read_csv(filename, sep=&#39;\t&#39;, index_col=0)<br>    diabetes_col = df[&#39;Diabetes&#39;]<br>    known_diabetes = diabetes_col[diabetes_col.isin([&#39;Yes&#39;, &#39;No&#39;])]<br>    percent_diabetic = 100 * known_diabetes.value_counts(normalize=True)[&#39;Yes&#39;]<br>    print(&quot;{:.2f}%&quot;.format(percent_diabetic))<br><br>if __name__ == &#39;__main__&#39;:<br>    main()</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">3</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.<br><br>The first solution resulted in an error because it did not have an example of the data file. For the second try, I provided the first few lines of the data file. It overcame the error, but the calculation output was wrong. I informed ChatGPT that there were some missing values in the data file (on rows that I had not provided), and it corrected the solution.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R165" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">166</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">12 - Perform complex queries - B</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a path to one of above-mentioned data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names (patient identifiers). Also use the &quot;sep&quot; keyword argument to indicate the the values are tab separated.<br><br>Please write code to print an answer to the following question:<br><br>For patients who have a BMI higher than 30 or a BMI lower than 10, what is the ratio of patients who have a &quot;Stage II&quot; tumor compares to those who have a &quot;Stage IV&quot; tumor (&quot;tumor_Stage-Pathological&quot; column)?<br><br>Use string formatting to round this number to two decimal places.<br><br>Hint(s):<br><br>    The answer is between 2.00 and 3.00.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import pandas as pd<br>import sys<br><br># Load data file into a pandas dataframe<br>file_path = sys.argv[1]<br>data = pd.read_csv(file_path, sep=&#39;\t&#39;, index_col=0)<br><br># Filter rows with BMI higher than 30 or lower than 10<br>bmi_filter = (data[&#39;BMI&#39;] &gt; 30) | (data[&#39;BMI&#39;] &lt; 10)<br>data = data[bmi_filter]<br><br># Count the number of patients with Stage II and Stage IV tumors<br>stage_ii = (data[&#39;tumor_Stage-Pathological&#39;] == &#39;Stage II&#39;).sum()<br>stage_iv = (data[&#39;tumor_Stage-Pathological&#39;] == &#39;Stage IV&#39;).sum()<br><br># Calculate the ratio of patients with Stage II compared to those with Stage IV<br>ratio = stage_ii / stage_iv<br><br># Format the ratio and print the answer<br>print(&quot;{:.2f}&quot;.format(ratio))</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R166" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">167</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">13 - Perform complex queries - C</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a path to one of above-mentioned data file. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names (patient identifiers). Also use the &quot;sep&quot; keyword argument to indicate the the values are tab separated.<br><br>Please write code to print answers to the following questions (print one per line):<br><br>1. What is the average tumor size (in centimeters) for patients from the United States who are over 60 years old?<br>2. What is the average tumor size (in centimeters) for patients not from the United States who are over 60 years old?<br><br>Use string formatting to round your answers to 2 decimal places.<br><br>Lastly, write a comment describing your ideas on why these values might be so different.<br><br>Hint(s):<br><br>    The first answer is between 4.00 and 5.00.<br>    <br>    The second answer is between 3.00 and 4.00.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br><br># Check if the file path argument is provided<br>if len(sys.argv) &lt; 2:<br>    print(&quot;Please provide a path to the data file as an argument.&quot;)<br>    sys.exit(1)<br><br># Read the data file into a DataFrame<br>file_path = sys.argv[1]<br>df = pd.read_csv(file_path, index_col=0, sep=&#39;\t&#39;)<br><br># Filter the data to only include patients over 60 years old<br>df = df[df[&quot;Age&quot;] &gt; 60]<br><br># Calculate the average tumor size for patients from the United States<br>us_patients = df[df[&quot;Country&quot;] == &quot;United States&quot;]<br>us_tumor_size_avg = round(us_patients[&quot;Tumor_Size_cm&quot;].mean(), 2)<br><br># Calculate the average tumor size for patients not from the United States<br>non_us_patients = df[df[&quot;Country&quot;] != &quot;United States&quot;]<br>non_us_tumor_size_avg = round(non_us_patients[&quot;Tumor_Size_cm&quot;].mean(), 2)<br><br>print(us_tumor_size_avg)<br>print(non_us_tumor_size_avg)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">3</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt.<br><br>The first solution resulted in an error because it did not have an example of the data file. For the second try, I provided the first few lines of the data file. It overcame the error, but it included extra text in the output. I informed ChatGPT about this, and it corrected the solution.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R167" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">168</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">14 - Perform query and save TSV file</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise builds on the previous one.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept two arguments. The first argument will be the name of one of the above-mentioned data files. The second argument will be the name of a file that you code will need to create.<br><br>Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names (patient identifiers). Also use the &quot;sep&quot; keyword argument to indicate the the values are tab separated.<br><br>Write code to filter the data to patients from the United States who are over 60 years old. Save the data to the specified output file (in TSV format).<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv Results.tsv<br>    <br>    cat Results.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import pandas as pd<br>import sys<br><br># Get the input file name and output file name from the command line arguments<br>input_file = sys.argv[1]<br>output_file = sys.argv[2]<br><br># Read the data into a DataFrame<br>df = pd.read_csv(input_file, sep=&#39;\t&#39;, index_col=0)<br><br># Filter the data to patients from the United States who are over 60 years old<br>df = df[(df[&#39;Country&#39;] == &#39;United States&#39;) &amp; (df[&#39;Age&#39;] &gt; 60)]<br><br># Save the filtered data to the output file in TSV format<br>df.to_csv(output_file, sep=&#39;\t&#39;)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt. However, I did provide the first few lines of the file.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R168" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">169</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">15 - Find longest chromosome (Biopython)</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a Python script that accepts a single argument: the name of a text file. This text file will be in FASTA format. Each entry in the FASTA file represents the DNA sequence of a chromosome. Each entry starts with a &quot;&gt;&quot; character and may be split across multiple lines in the file. Your function should use the &quot;biopython&quot; module to parse the FASTA file and identify the name (for example, &quot;chr2&quot;) of the longest chromosome (i.e., the one with the most nucleotides). Please print the name of this chromosome.<br><br>Hint(s):<br><br>    Your code should be considerably shorter than when you solved this problem in a previous exercise without the &quot;biopython&quot; module.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA1.fasta<br>    python code.py DNA2.fasta<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA1.fasta<br><br>File contents:<br><br>    &gt;chr1<br>    AGTGTCCTTAAGACTTAAGATAACGCCTCTCGTAGACTCG<br>    ACAAAGTCGACTCCGCCTGCTCAAATCGACACCTTGGTGA<br>    &gt;chr2<br>    TCACAACAATGAGACGTGGCCAGAAGTACGACATTGTGAA<br>    GTGTAGGTCTGAAAACATTATCCGCATCAGGCCT<br>    &gt;chr3<br>    CCAGAAGCTACAGGCTTGAGTTGCCTGGAGATTCAGTTTG<br>    GCGAGCATCCCACCAGTGCTTACTAGCATACCGTCTGTAT<br>    CCTAGTGCATGAG<br>    &gt;chr4<br>    GCAAAAAACATTTTTCCGCTAACTAACTACTTGCGACGCG<br>    TGAACCATAAGCGGAGATGTACCTC<br>    &gt;chr5<br>    AAACAGTTATCTGTTAACCCCCCGTCTTTTTACCGGCCGC<br>    CGGAACGCAACTAACCCCTGTCGGAGCCCAAAGATCGTCA<br>    CCGTGCCTCCTAAGGCGGTGGCACTACGGAAAGTC<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA2.fasta<br><br>File contents:<br><br>    &gt;chr1<br>    TACCTCCAGGGGGCATCCTCCCCCCCAATTCGAAACACAT<br>    CGTAGCCCCTGGCACTACCTATGTGTGTCAATTCGGAGAA<br>    GAGAGATTCACGAAAAAAAAGTCTGGACTCAACTAGGATC<br>    ACACATTCGGCTACAGATACCAAAAAAAAAAAAAAAA<br>    &gt;chr2<br>    CTCGTCTCAGCGAAGTATAGGAGACACCCCAGTATTTGTG<br>    CGAAACAGTTATAAGAGGAACGCGACTACGACCTAGAGTG<br>    TAAGCCCGGTTAGTCATAATATGCAGAAGAGAATCAATTG<br>    ATTCCGGGCTTCTTCTTTAATAACATGAGTGGTTCCATAG<br>    ATATAGGTCCGCCTAGGTTAACTCGCCAGTAGAGAGCAAC<br>    GTATGGCAGCACATTCCGATCTTGATCAACGGCCTGCCGA<br>    ACTTCCGACCAGTACGATGCTAGTTCAAATCCCCAGGTGG<br>    TATAGTCTTATCTGCACGTCAAA<br>    &gt;chr3<br>    AAATTTTCACCATTGAGGCACCACCTTCTCGTCGCTGCGC<br>    GCTCTGCTCGCTTCGGCTAAAAATTCGCGCAATACATTCG<br>    CTACAGATACCAAA<br>    &gt;chr4<br>    GTCTCTGCTGTCCGGTAGGCAGGGCTTTGACTGCTGGATG<br>    GTGAATTACATGCCCGATTGTACTCACGTCCACTCAAGCG<br>    CTTGAAAGACTGGGCAATATGATCCCGTATATTGCGTAAT<br>    CTCGGTGGGTCAGGACTCCTAAAATGCTCATGGTAAGTTG<br>    TAGACGGGGGAAATCCTACAACCACCC<br>    &gt;chr5<br>    CTATCTTACCAGGAGCTAGCTTACTAGCCCTTTAAAGGAA<br>    AAATTGGCACGCGGAGACAGACAG</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">from Bio import SeqIO<br>import sys<br><br>def main(file_name):<br>    longest_chromosome = &#39;&#39;<br>    max_length = 0<br>    with open(file_name, &#39;r&#39;) as f:<br>        for record in SeqIO.parse(f, &#39;fasta&#39;):<br>            if len(record.seq) &gt; max_length:<br>                max_length = len(record.seq)<br>                longest_chromosome = record.id<br>    print(longest_chromosome)<br><br>if __name__ == &#39;__main__&#39;:<br>    main(sys.argv[1])</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">It must have been trained on BioPython examples.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R169" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">170</div></th><td class="s1">17 - Tabular Data Analysis</td><td class="s1">16 - Calculate GC content (Biopython)</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a Python script that accepts a single argument: the name of a text file. This text file will be in FASTA format. Each entry in the FASTA file represents the DNA sequence of a chromosome. Each entry starts with a &quot;&gt;&quot; character and may be split across multiple lines in the file. Your function should use the &quot;biopython&quot; module (see examples in slides) to calculate the overall proportion of nucleotides that are &quot;G&quot; or &quot;C&quot; nucleotides across all chromosomes. Please round your answer to two decimal places and print the answer. (You will need to use string formatting to make sure there are two decimal places in the output.)<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA1.fasta<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA2.fasta<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA1.fasta<br><br>File contents:<br><br>    &gt;chr1<br>    AGTGTCCTTAAGACTTAAGATAACGCCTCTCGTAGACTCG<br>    ACAAAGTCGACTCCGCCTGCTCAAATCGACACCTTGGTGA<br>    &gt;chr2<br>    TCACAACAATGAGACGTGGCCAGAAGTACGACATTGTGAA<br>    GTGTAGGTCTGAAAACATTATCCGCATCAGGCCT<br>    &gt;chr3<br>    CCAGAAGCTACAGGCTTGAGTTGCCTGGAGATTCAGTTTG<br>    GCGAGCATCCCACCAGTGCTTACTAGCATACCGTCTGTAT<br>    CCTAGTGCATGAG<br>    &gt;chr4<br>    GCAAAAAACATTTTTCCGCTAACTAACTACTTGCGACGCG<br>    TGAACCATAAGCGGAGATGTACCTC<br>    &gt;chr5<br>    AAACAGTTATCTGTTAACCCCCCGTCTTTTTACCGGCCGC<br>    CGGAACGCAACTAACCCCTGTCGGAGCCCAAAGATCGTCA<br>    CCGTGCCTCCTAAGGCGGTGGCACTACGGAAAGTC<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA2.fasta<br><br>File contents:<br><br>    &gt;chr1<br>    TACCTCCAGGGGGCATCCTCCCCCCCAATTCGAAACACAT<br>    CGTAGCCCCTGGCACTACCTATGTGTGTCAATTCGGAGAA<br>    GAGAGATTCACGAAAAAAAAGTCTGGACTCAACTAGGATC<br>    ACACATTCGGCTACAGATACCAAAAAAAAAAAAAAAA<br>    &gt;chr2<br>    CTCGTCTCAGCGAAGTATAGGAGACACCCCAGTATTTGTG<br>    CGAAACAGTTATAAGAGGAACGCGACTACGACCTAGAGTG<br>    TAAGCCCGGTTAGTCATAATATGCAGAAGAGAATCAATTG<br>    ATTCCGGGCTTCTTCTTTAATAACATGAGTGGTTCCATAG<br>    ATATAGGTCCGCCTAGGTTAACTCGCCAGTAGAGAGCAAC<br>    GTATGGCAGCACATTCCGATCTTGATCAACGGCCTGCCGA<br>    ACTTCCGACCAGTACGATGCTAGTTCAAATCCCCAGGTGG<br>    TATAGTCTTATCTGCACGTCAAA<br>    &gt;chr3<br>    AAATTTTCACCATTGAGGCACCACCTTCTCGTCGCTGCGC<br>    GCTCTGCTCGCTTCGGCTAAAAATTCGCGCAATACATTCG<br>    CTACAGATACCAAA<br>    &gt;chr4<br>    GTCTCTGCTGTCCGGTAGGCAGGGCTTTGACTGCTGGATG<br>    GTGAATTACATGCCCGATTGTACTCACGTCCACTCAAGCG<br>    CTTGAAAGACTGGGCAATATGATCCCGTATATTGCGTAAT<br>    CTCGGTGGGTCAGGACTCCTAAAATGCTCATGGTAAGTTG<br>    TAGACGGGGGAAATCCTACAACCACCC<br>    &gt;chr5<br>    CTATCTTACCAGGAGCTAGCTTACTAGCCCTTTAAAGGAA<br>    AAATTGGCACGCGGAGACAGACAG</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>from Bio import SeqIO<br><br>def fasta_GC_content(file_name):<br>    gc_count = 0<br>    total_count = 0<br>    with open(file_name, &quot;r&quot;) as fasta_file:<br>        for record in SeqIO.parse(fasta_file, &quot;fasta&quot;):<br>            gc_count += record.seq.count(&quot;G&quot;) + record.seq.count(&quot;C&quot;)<br>            total_count += len(record.seq)<br>    return round(gc_count/total_count, 2)<br><br>if __name__ == &#39;__main__&#39;:<br>    file_name = sys.argv[1]<br>    proportion = fasta_GC_content(file_name)<br>    print(&quot;%.2f&quot; % proportion)</td><td class="s2">2</td><td class="s2" dir="ltr">2</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">The first solution gave me a deprecation warning, and it printed the output as percentages rather than proportions. I notified ChatGPT, and it rectified both problems.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R170" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">171</div></th><td class="s1">18 - Data Visualization</td><td class="s1">01 - Make a box plot</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercise uses clinical data for endometrial cancer patients. The rows represent different patients; the columns represent different data points about these patients. For example, there is a column that indicates the country of origin for each patient, their tumor grade, etc. On the far right, you will see data representing the patients&#39; tumor size and number of pregnancies.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be the name of the data file.<br><br>Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names. Use the &quot;sep&quot; keyword argument to indicate that the values are tab-separated.<br><br>As your complete the next steps, you may wish to refer to the reading material, slides, and the seaborn documentation.<br><br>Box plots are a great way to investigate relationships between categorical labels and numerical data. A person&#39;s country of origin is an example of a categorical label. The values are strings, and there are a limited number of possible values. Age is an example of a numerical variable. It can have an infinite number of possible values; well, if we&#39;re counting in years, it could have a value anywhere between 0 and maybe 120 (for humans).<br><br>Please create a boxplot that shows the Country categorical variable on the x-axis and the Age numeric variable on the y-axis. Remember, the syntax is slightly different for some types of graphs than for others (the slides show examples).<br><br>You do not need to save the graph to an output file.<br><br>Here is some code to get your started:<br><br>    # We are using the &quot;as&quot; keyword to specify shortcuts for each import<br>    import pandas as pd<br>    import seaborn as sns<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">jpeg</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import pandas as pd<br>import seaborn as sns<br>import sys<br>import matplotlib.pyplot as plt<br><br>file_name = sys.argv[1]<br><br>df = pd.read_csv(file_name, index_col=0, sep=&#39;\t&#39;)<br><br>sns.boxplot(x=&#39;Country&#39;, y=&#39;Age&#39;, data=df)<br><br>plt.show()</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">3</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt. However, I did provide the first few lines of the file.<br><br>On the first try, I received an error message in the auto-grader that it couldn&#39;t find a particular function. I notified ChatGPT, and it fixed this problem. But then it failed to match the expected output because ChatGPT specified a style setting that the auto-grader was not expecting. I asked ChatGPT to remove that, and it worked.</td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R171" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">172</div></th><td class="s1">18 - Data Visualization</td><td class="s1">02 - Make a swarm plot</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercises uses the same data as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be the name of the data file.<br><br>Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names. Use the &quot;sep&quot; keyword argument to indicate that the values are tab-separated.<br><br>In the previous exercise, you created a classic box plot. It had each country as a separate box and showed the average value, as well as the upper and lower quartiles. It would be helpful to see the individual data points. This can be done with a &quot;swarm&quot; plot.<br><br>Please create a swarm plot that shows the Country categorical variable on the x-axis and the Age numeric variable on the y-axis. Use the &quot;swarmplot&quot; function rather than the &quot;catplot&quot; function for this (the slides show an example).<br><br>You do not need to save the graph to an output file.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">jpeg</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br>import seaborn as sns<br>import matplotlib.pyplot as plt<br><br>file_name = sys.argv[1]<br>df = pd.read_csv(file_name, sep=&#39;\t&#39;, index_col=0)<br><br>sns.swarmplot(x=&#39;Country&#39;, y=&#39;Age&#39;, data=df)<br>plt.show()</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt. However, I did provide the first few lines of the file.<br><br>On the first try, I received an error message in the auto-grader that it couldn&#39;t find a particular function. I notified ChatGPT, and it fixed this problem.</td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R172" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">173</div></th><td class="s1">18 - Data Visualization</td><td class="s1">03 - Make a histogram plot</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercises uses the same data as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be the name of the data file.<br><br>Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names. Use the &quot;sep&quot; keyword argument to indicate that the values are tab-separated.<br><br>Histograms are a helpful way to visualize the shape of numeric data. Use &quot;pandas&quot; to filter the data based on the &quot;Sample_Tumor_Normal&quot; column; keep data from patients with a &quot;Tumor&quot; only. For these patients, create a histogram of the &quot;Age&quot; numbers (first extract this information into a &quot;pandas&quot; Series). Also set the y-axis label to &quot;Frequency&quot;.<br><br>The slides show an example of using the &quot;histplot&quot; function. Instead use &quot;distplot&quot;.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">jpeg</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br>import seaborn as sns<br>import matplotlib.pyplot as plt<br><br>filename = sys.argv[1]<br>data = pd.read_csv(filename, sep=&#39;\t&#39;, index_col=0)<br><br># Filter data for tumor patients only<br>tumor_data = data[data[&quot;Sample_Tumor_Normal&quot;] == &quot;Tumor&quot;]<br><br># Plot histogram of age for tumor patients<br>sns.distplot(tumor_data[&quot;Age&quot;])<br>plt.ylabel(&quot;Frequency&quot;)<br>plt.show()</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">4</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt. However, I did provide the first few lines of the file.<br><br>On the first try, I received an error message in the auto-grader that it couldn&#39;t find a particular function that was being used. (The solution also had some extra formatting that I did not request.) I notified ChatGPT, and it fixed the error. On the second try, the auto-grader gave me a message stating that it couldn&#39;t compare the generated image with the expected output because they were different sizes. ChatGPT tried to fix this by specifying image dimensions explicitly. That gave me the same error. I asked it to use the default dimensions, and the code passed. In this final solution, it also removed some of the styles that it had added previously.</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R173" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">174</div></th><td class="s1">18 - Data Visualization</td><td class="s1">04 - Make a scatter plot</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercises uses the same data as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be the name of the data file.<br><br>Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names. Use the &quot;sep&quot; keyword argument to indicate that the values are tab-separated.<br><br>Scatter plots are a good way to visualize relationships between two numeric variables. Let&#39;s examine the relationship between a patient&#39;s age and BMI. Use &quot;pandas&quot; to filter the data based on the &quot;Sample_Tumor_Normal&quot; column; keep data from patients with a &quot;Tumor&quot; only. For these patients, create a scatter plot of &quot;Age&quot; (x-axis) and &quot;BMI&quot; (y-axis). Color the points in the graph based on the values in the &quot;Histologic_type&quot; column. Use the built-in functionality in seaborn to specify the colorblind-friendly palette.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">jpeg</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br>import seaborn as sns<br>import matplotlib.pyplot as plt<br><br># Read the input file<br>input_file = sys.argv[1]<br>df = pd.read_csv(input_file, sep=&#39;\t&#39;, index_col=0)<br><br># Filter the data for tumor samples only<br>tumor_samples = df[df[&quot;Sample_Tumor_Normal&quot;] == &quot;Tumor&quot;]<br><br># Create a scatter plot of age and BMI for tumor samples<br>sns.set_palette(&quot;colorblind&quot;)<br>sns.relplot(x=&quot;Age&quot;, y=&quot;BMI&quot;, hue=&quot;Histologic_type&quot;, data=tumor_samples)<br><br># Show the plot<br>plt.show()</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt. However, I did provide the first few lines of the file.<br><br>On the first try, the auto-grader gave me an error saying that it couldn&#39;t compare the output against the expected output because the image sizes were different. I concluded that the problem might be that this problem was designed to use the relplot function to create a scatterplot rather than the scatterplot function. I had taught relplot because it was simpler. I asked ChatGPT to use that function instead.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R174" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">175</div></th><td class="s1">18 - Data Visualization</td><td class="s1">05 - Make a violin plot</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>This exercises uses the same data as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be the name of the data file.<br><br>Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names. Use the &quot;sep&quot; keyword argument to indicate that the values are tab-separated.<br><br>A violin plot can be used to examine relationships among three variables: tumor stage, tumor focality (whether there is more than one tumor), and patient age. Use &quot;pandas&quot; to filter the data based on the &quot;Sample_Tumor_Normal&quot; column; keep data from patients with a &quot;Tumor&quot; only. Then sort the DataFrame based on the &quot;tumor_Stage-Pathological&quot; column using the sort_values.<br><br>Create a violin plot that shows data from the &quot;tumor_Stage-Pathological&quot; column on the x-axis and the &quot;Age&quot; column on the y-axis. Use the &quot;violinplot&quot; function rather than the &quot;catplot&quot; function for this (the slides show an example). Color the &quot;violins&quot; based on the &quot;Tumor_Focality&quot; column. Use the built-in functionality in seaborn to specify the colorblind-friendly palette. Set the x-axis label to &quot;Tumor stage&quot;.<br><br>Hint(s):<br><br>    The function you use to sort a DataFrame in &quot;pandas&quot; is a fruitful function.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py Endometrial.tsv<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Endometrial.tsv<br><br>File contents:<br><br>    Patient_ID	Sample_ID	Sample_Tumor_Normal	Proteomics_Tumor_Normal	Country	Histologic_Grade_FIGO	Myometrial_invasion_Specify	Histologic_type	Treatment_naive	Tumor_purity	Path_Stage_Primary_Tumor-pT	Path_Stage_Reg_Lymph_Nodes-pN	Clin_Stage_Dist_Mets-cM	Path_Stage_Dist_Mets-pM	tumor_Stage-Pathological	FIGO_stage	LVSI	BMI	Age	Diabetes	Race	Ethnicity	Gender	Tumor_Site	Tumor_Site_Other	Tumor_Focality	Tumor_Size_cm	Num_full_term_pregnancies<br>    C3L-00006	S001	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	1.0	38.88	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.9	1<br>    C3L-00008	S002	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage IV	IA	0.0	39.76	58.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Unifocal	3.5	1<br>    C3L-00032	S003	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	51.19	50.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	4 or more<br>    C3L-00090	S005	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	32.69	75.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	3.5	4 or more<br>    C3L-00098	S006	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA		20.28	63.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior  and Posterior endometrium	Unifocal	6.0	2<br>    C3L-00136	S007	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	55.67	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.5	3<br>    C3L-00137	S008	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	1.0	25.68	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	5.0	2<br>    C3L-00139	S009	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	21.57	83.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	4.0	4 or more<br>    C3L-00143	S010	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1 (FIGO I)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.26	69.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	5.2	2<br>    C3L-00145	S011	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	36.57	59.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	4.7	3<br>    C3L-00156	S012	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.83	56.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-00161	S014	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IB	1.0	68.39	46.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	7.0	2<br>    C3L-00358	S016	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	1.0	26.22	90.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	4.5	Unknown<br>    C3L-00361	S017	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	42.98	64.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	2.7	None<br>    C3L-00362	S018	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	55.86	38.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	13.5	None<br>    C3L-00413	S019	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	42.19	60.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.2	3<br>    C3L-00449	S020	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	27.82	59.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior endometrium	Unifocal	7.0	None<br>    C3L-00563	S021	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	34.72	62.0	Yes	Asian	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.0	1<br>    C3L-00586	S022	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT2 (FIGO II)	pN2 (FIGO IIIC2)	cM0	Staging Incomplete	Stage III	IIIC2	1.0	21.45	50.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	6.0	2<br>    C3L-00601	S023	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	25.03	57.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior	Unifocal	3.5	3<br>    C3L-00605	S024	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	38.54	73.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	2.7	Unknown<br>    C3L-00767	S025	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	65.71	56.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.3	4 or more<br>    C3L-00769	S026	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	46.64	56.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00770	S027	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	45.83	73.0	Yes	White	Not-Hispanic or Latino	Female	Anterior endometrium		Multifocal	2.7	4 or more<br>    C3L-00771	S028	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	1.0	22.86	86.0	No	White	Not-Hispanic or Latino	Female	Posterior endometrium		Multifocal	6.0	Unknown<br>    C3L-00780	S029	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	46.41	69.0	Yes	White	Hispanic or Latino	Female	Other, specify	Posterior and Anterior Endometrium	Unifocal	1.7	2<br>    C3L-00781	S030	Tumor	Tumor	Other_specify	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	71.09	48.0	No	White	Hispanic or Latino	Female	Other, specify	Anterior and Posterior Endometrium	Unifocal	5.5	4 or more<br>    C3L-00905	S031	Tumor	Tumor	United States	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	Staging Incomplete	No pathologic evidence of distant metastasis	Stage III	IIIC1	1.0	44.81	64.0	No	White	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior	Unifocal	4.5	Unknown<br>    C3L-00918	S032	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	43.0	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Both anterior and posterior endometrium	Unifocal	3.0	2<br>    C3L-00921	S033	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	No pathologic evidence of distant metastasis	Stage II	II	1.0	32.32	66.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	3<br>    C3L-00932	S034	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	47.82	67.0	No	White	Not-Hispanic or Latino	Female	Other, specify	left cornu	Unifocal	1.0	3<br>    C3L-00942	S036	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	31.58	64.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	4.5	None<br>    C3L-00946	S037	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	17.64	64.0	No	White	Not reported	Female	Posterior endometrium		Multifocal	3.9	1<br>    C3L-00947	S038	Tumor	Tumor	Other_specify	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.84	71.0	Yes	White	Not reported	Female	Posterior endometrium		Multifocal	1.8	2<br>    C3L-00949	S039	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	1.0	37.69	64.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.5	2<br>    C3L-00961	S040	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.5	59.0	No	White	Not reported	Female	Anterior endometrium		Multifocal	4.5	2<br>    C3L-00963	S041	Tumor	Tumor	Other_specify		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	34.89	59.0	Yes	White	Not reported	Female	Other, specify	along anterior and posterior surface	Unifocal	2.6	1<br>    C3L-01246	S042	Tumor	Tumor	Other_specify		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	39.14	62.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	2.3	1<br>    C3L-01248	S044	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	59.78	42.0	No	White	Not reported	Female	Posterior endometrium		Unifocal	6.3	1<br>    C3L-01249	S045	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.89	65.0	No	White	Not reported	Female	Other, specify	Tumor occupies 75% of endometrial surface	Unifocal	6.5	1<br>    C3L-01252	S046	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	38.41	76.0	Yes	White	Not reported	Female	Posterior endometrium		Unifocal	0.9	4 or more<br>    C3L-01256	S048	Tumor	Tumor	Other_specify	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	0.0	34.37	75.0	Yes	White	Not reported	Female	Anterior endometrium		Unifocal	4.3	4 or more<br>    C3L-01257	S049	Tumor	Tumor	Other_specify	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	17.11	71.0	No	White	Not reported	Female	Other, specify	Tumor involves 75% of endometrial cavity per diagnostic pathology report	Unifocal	8.0	2<br>    C3L-01275	S050	Tumor	Tumor	Other_specify	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage I	IB	1.0	32.06	65.0	No	Not Reported	Not reported	Female	Other, specify	100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED	Unifocal	5.0	Unknown<br>    C3L-01282	S051	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	1.0	31.96	64.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	Unknown<br>    C3L-01304	S053	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	41.44	68.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.7	3<br>    C3L-01307	S054	Tumor	Tumor	United States	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	pM1	Stage IV	IVB	1.0	31.63	74.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	3.5	3<br>    C3L-01311	S055	Tumor	Tumor	United States	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	37.11	55.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	3.0	None<br>    C3L-01312	S056	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	No pathologic evidence of distant metastasis	Stage I	IA	0.0	31.96	56.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and Posterior	Unifocal	4.0	1<br>    C3L-01744	S057	Tumor	Tumor	Other_specify	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	Staging Incomplete	Staging Incomplete	Stage I	IA	0.0	46.45	62.0	No	White	Not reported	Female	Anterior endometrium		Unifocal	2.2	2<br>    C3L-01925	S058	Tumor	Tumor	United States		50 % or more	Serous	YES	Normal	pT3b (FIGO IIIB)	pN1 (FIGO IIIC1)	Staging Incomplete	pM1	Stage IV	IVB	1.0	27.66	65.0	No	White	Not-Hispanic or Latino	Female	Other, specify	Anterior and posterior endometrium	Unifocal	4.5	None<br>    C3N-00151	S059	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	0.0	27.1	60.0	Unknown			Female	Other, specify	Entire uterine cavity	Unifocal		2<br>    C3N-00200	S060	Tumor	Tumor	United States	FIGO grade 2	Not identified	Endometrioid	YES	Normal	pT2 (FIGO II)	pN1 (FIGO IIIC1)	Staging Incomplete	Staging Incomplete	Stage III	IIIC1		46.85	72.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	anterior and posterior endometrial cavity	Multifocal	9.0	4 or more<br>    C3N-00321	S061	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	26.0	64.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.5	2<br>    C3N-00322	S062	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	70.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	2.6	2<br>    C3N-00323	S063	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.0	78.0	No			Female	Other, specify	Uterine cavity	Multifocal	4.0	Unknown<br>    C3N-00324	S064	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.0	66.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	2.3	Unknown<br>    C3N-00326	S065	Tumor	Tumor	Ukraine	FIGO grade 1	Not identified	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	35.0	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	2<br>    C3N-00328	S066	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	31.22	62.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.0	Unknown<br>    C3N-00333	S067	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	31.0	65.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.0	Unknown<br>    C3N-00334	S068	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	32.83	68.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.4	1<br>    C3N-00335	S069	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	29.52	57.0	No			Female	Other, specify	Uterine cavity	Multifocal	2.5	None<br>    C3N-00337	S070	Tumor	Tumor	Ukraine	FIGO grade 3	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	29.07	67.0	No			Female	Other, specify	Entire Uterine Cavity	Multifocal	1.3	1<br>    C3N-00339	S071	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	21.83	45.0	No			Female	Other, specify	Uterine cavity	Multifocal	1.3	Unknown<br>    C3N-00340	S072	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT3a (FIGO IIIA)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	27.0	60.0	No			Female	Other, specify	Uterine cavity	Multifocal	3.5	2<br>    C3N-00377	S073	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	34.0	64.0	Yes			Female	Other, specify	Uterine cavity	Multifocal	1.0	Unknown<br>    C3N-00379	S074	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	36.81	41.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.5	1<br>    C3N-00383	S075	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	46.0	61.0	No			Female	Other, specify	Uterine cavity	Unifocal	4.0	1<br>    C3N-00386	S076	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	27.31	44.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	2.3	2<br>    C3N-00388	S077	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	20.55	59.0	No			Female	Other, specify	Entire uterine cavity.	Multifocal	4.0	1<br>    C3N-00389	S078	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	cM0	Staging Incomplete	Stage I	IB	0.0	17.85	62.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.5	1<br>    C3N-00729	S079	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN0	Staging Incomplete	Staging Incomplete	Stage II	IB	0.0	29.62	86.0	No	White	Not-Hispanic or Latino	Female	Other, specify	both anterior and posterior	Multifocal	4.0	4 or more<br>    C3N-00734	S080	Tumor	Tumor	United States	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	Staging Incomplete	Staging Incomplete	Stage III	IIIA	1.0	38.97	53.0	No	White	Not-Hispanic or Latino	Female	Anterior endometrium		Unifocal	4.0	None<br>    C3N-00743	S081	Tumor	Tumor	United States	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	48.46	53.0	No	Black or African American	Not-Hispanic or Latino	Female	Other, specify	Involves fundus, anterior and posterior walls	Multifocal	3.5	2<br>    C3N-00836	S082	Tumor	Tumor	Ukraine	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM1	Staging Incomplete	Stage I	IA	0.0	30.47	75.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	3<br>    C3N-00847	S083	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	34.53	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	4.3	2<br>    C3N-00848	S084	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	33.65	66.0	Yes			Female	Other, specify	Entire uterine cavity	Multifocal	3.0	None<br>    C3N-00850	S085	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	28.84	65.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.0	1<br>    C3N-00858	S086	Tumor	Tumor	Poland	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage III	IIIB	1.0	36.0	65.0	No			Female	Anterior endometrium		Multifocal	11.0	2<br>    C3N-00866	S087	Tumor	Tumor	United States	FIGO grade 1	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	23.88	77.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Anterior, posterior and fundus	Multifocal	3.0	4 or more<br>    C3N-00880	S088	Tumor	Tumor	Poland	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IA	0.0	27.0	61.0	No			Female	Anterior endometrium		Unifocal	3.2	1<br>    C3N-01003	S090	Tumor	Tumor	Poland	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT3b (FIGO IIIB)	pN0	cM1	pM1	Stage IV	IVB	0.0	31.0	73.0	Yes			Female	Anterior endometrium		Multifocal	3.0	4 or more<br>    C3N-01211	S091	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	1.0	39.2	59.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.4	1<br>    C3N-01212	S092	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pNX	cM0	Staging Incomplete	Stage III	IIIA	1.0	30.48	63.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	2<br>    C3N-01217	S093	Tumor	Tumor	Ukraine	FIGO grade 1	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pNX	cM0	Staging Incomplete	Stage I	IB	0.0	36.2	58.0	No			Female	Other, specify	Endometrium	Multifocal	0.8	None<br>    C3N-01219	S094	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT3a (FIGO IIIA)	pN0	cM0	Staging Incomplete	Stage III	IIIA	0.0	26.14	58.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	1<br>    C3N-01267	S095	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT1b (FIGO IB)	pN1 (FIGO IIIC1)	cM0	Staging Incomplete	Stage III	IIIC1	1.0	30.85	57.0	No			Female	Other, specify	Entire uterine cavity	Multifocal	1.2	2<br>    C3N-01346	S096	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN2 (FIGO IIIC2)	cM0	No pathologic evidence of distant metastasis	Stage III	IIIC2	1.0	34.0	63.0				Female	Anterior endometrium		Unifocal	5.5	1<br>    C3N-01349	S097	Tumor	Tumor	Poland		50 % or more	Serous	YES	Normal	pT1b (FIGO IB)	pN0	cM0	No pathologic evidence of distant metastasis	Stage I	IB	1.0	31.0	77.0	Yes			Female	Anterior endometrium		Multifocal	5.0	4 or more<br>    C3N-01510	S098	Tumor	Tumor	United States	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pNX	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	40.72	53.0	Yes	White	Not-Hispanic or Latino	Female	Other, specify	Bulky tumor involving both anterior and posterior walls	Multifocal	8.5	None<br>    C3N-01520	S099	Tumor	Tumor	Ukraine	FIGO grade 2	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.37	69.0	No			Female	Other, specify	Endometrium	Multifocal	1.0	2<br>    C3N-01521	S100	Tumor	Tumor	Ukraine	FIGO grade 3	under 50 %	Endometrioid	YES	Normal	pT1a (FIGO IA)	pNX	cM0	Staging Incomplete	Stage I	IA	0.0	29.4	75.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	4.2	2<br>    C3N-01537	S101	Tumor	Tumor	Ukraine	FIGO grade 2	50 % or more	Endometrioid	YES	Normal	pT2 (FIGO II)	pN0	cM0	Staging Incomplete	Stage II	II	0.0	35.42	74.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	1.5	1<br>    C3N-01802	S102	Tumor	Tumor	United States		under 50 %	Serous	YES	Normal	pT2 (FIGO II)	pN0	Staging Incomplete	Staging Incomplete	Stage II	II	1.0	24.32	85.0	Yes	Black or African American	Not-Hispanic or Latino	Female	Other, specify	entire uterine cavity	Unifocal	3.8	1<br>    C3N-01825	S103	Tumor	Tumor	Ukraine		under 50 %	Serous	YES	Normal	pT1a (FIGO IA)	pN0	cM0	Staging Incomplete	Stage I	IA	0.0	34.06	70.0	No			Female	Other, specify	Entire uterine cavity	Unifocal	5.0	Unknown<br>    C3L-00006.N	S105	Normal	Adjacent_normal																								<br>    C3L-00361.N	S106	Normal	Adjacent_normal																								<br>    C3L-00563.N	S130	Normal	Myometrium_normal																								<br>    C3L-00586.N	S107	Normal	Adjacent_normal																								<br>    C3L-00601.N	S108	Normal	Adjacent_normal																								<br>    C3L-00605.N	S131	Normal	Myometrium_normal																								<br>    C3L-00769.N	S109	Normal	Adjacent_normal																								<br>    C3L-00770.N	S132	Normal	Myometrium_normal																								<br>    C3L-00771.N	S133	Normal	Myometrium_normal																								<br>    C3L-00930.N	S110	Normal	Adjacent_normal																								<br>    C3L-00932.N	S111	Normal	Adjacent_normal																								<br>    C3L-00947.N	S112	Normal	Adjacent_normal																								<br>    C3L-00963.N	S113	Normal	Adjacent_normal																								<br>    C3L-01246.N	S114	Normal	Adjacent_normal																								<br>    C3L-01249.N	S115	Normal	Adjacent_normal																								<br>    C3L-01252.N	S116	Normal	Adjacent_normal																								<br>    C3L-01256.N	S117	Normal	Adjacent_normal																								<br>    C3L-01257.N	S118	Normal	Adjacent_normal																								<br>    C3L-01282.N	S119	Normal	Adjacent_normal																								<br>    C3L-01304.N	S120	Normal	Adjacent_normal																								<br>    C3L-01307.N	S121	Normal	Adjacent_normal																								<br>    C3L-01311.N	S122	Normal	Adjacent_normal																								<br>    C3L-01744.N	S123	Normal	Adjacent_normal																								<br>    C3N-00200.N	S134	Normal	Myometrium_normal																								<br>    C3N-00333.N	S124	Normal	Adjacent_normal																								<br>    C3N-00383.N	S125	Normal	Adjacent_normal																								<br>    C3N-00729.N	S126	Normal	Adjacent_normal																								<br>    C3N-00858.N	S127	Normal	Adjacent_normal																								<br>    C3N-00866.N	S128	Normal	Adjacent_normal																								<br>    C3N-01211.N	S135	Normal	Myometrium_normal																								<br>    C3N-01346.N	S129	Normal	Adjacent_normal																								<br>    NX1.N	S136	Normal	Enriched_normal																								<br>    NX10.N	S145	Normal	Enriched_normal																								<br>    NX11.N	S146	Normal	Enriched_normal																								<br>    NX12.N	S147	Normal	Enriched_normal																								<br>    NX13.N	S148	Normal	Enriched_normal																								<br>    NX14.N	S149	Normal	Enriched_normal																								<br>    NX15.N	S150	Normal	Enriched_normal																								<br>    NX16.N	S151	Normal	Enriched_normal																								<br>    NX17.N	S152	Normal	Enriched_normal																								<br>    NX18.N	S153	Normal	Enriched_normal																								<br>    NX2.N	S137	Normal	Enriched_normal																								<br>    NX3.N	S138	Normal	Enriched_normal																								<br>    NX4.N	S139	Normal	Enriched_normal																								<br>    NX5.N	S140	Normal	Enriched_normal																								<br>    NX6.N	S141	Normal	Enriched_normal																								<br>    NX7.N	S142	Normal	Enriched_normal																								<br>    NX8.N	S143	Normal	Enriched_normal																								<br>    NX9.N	S144	Normal	Enriched_normal</td><td class="s1" dir="ltr">The data were provided by the National Cancer Institute&#39;s Clinical Proteomic Tumor Analysis Consortium (CPTAC) and the Payne lab: https://payne.byu.edu.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">jpeg</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 10, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br>import seaborn as sns<br>import matplotlib.pyplot as plt<br><br>def main():<br>    # Check if the input file name is provided<br>    if len(sys.argv) &lt; 2:<br>        print(&quot;Error: Please provide the input file name.&quot;)<br>        return<br><br>    # Read the data file<br>    file_name = sys.argv[1]<br>    data = pd.read_csv(file_name, sep=&#39;\t&#39;, index_col=0)<br><br>    # Filter the data based on the Sample_Tumor_Normal column<br>    tumor_data = data[data[&#39;Sample_Tumor_Normal&#39;] == &#39;Tumor&#39;]<br><br>    # Sort the data based on the tumor_Stage-Pathological column<br>    tumor_data.sort_values(&#39;tumor_Stage-Pathological&#39;, inplace=True)<br><br>    # Create a violin plot of the data<br>    sns.violinplot(x=&#39;tumor_Stage-Pathological&#39;, y=&#39;Age&#39;, data=tumor_data, hue=&#39;Tumor_Focality&#39;, palette=&#39;colorblind&#39;)<br><br>    # Set the x-axis label<br>    plt.xlabel(&#39;Tumor stage&#39;)<br><br>    # Show the plot<br>    plt.show()<br><br>if __name__ == &#39;__main__&#39;:<br>    main()</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">The data file was too long, so I excluded it from the prompt. However, I did provide the first few lines of the file.<br><br>Python gave some warnings, but it passed on the first try.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R175" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">176</div></th><td class="s1">19 - Additional Practice</td><td class="s1">01 - Delete duplicate animals</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Suppose you have 20 terabytes of data on the supercomputer, and the computing staff are getting mad at you for storing so much data. You realize that when you first started programming, you accidentally created a giant text file with a list of animal names. Each line contained the name of a different animal, but the animal names were accidentally repeated many times on each line. You want to keep the first instance of each animal and remove all other instances of that animal on the same line. This will help to significantly reduce the size of the data file.<br><br>Please write a function called &quot;deleteDuplicates&quot; that accepts two arguments: 1) a path to the data file that contains duplicates, and 2) a path to the new data file that your function will create. Your function should use the &quot;re&quot; module to accomplish this task. Your function does not need to return anything.<br><br>In this exercise, you&#39;ll get practice working with relatively small files (not terabytes in size). But the same techniques should apply to very large files.<br><br>Hint(s):<br><br>    If needed, use https://regex101.com to figure out a regular expression that will work.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    deleteDuplicates(&quot;Animals1.txt&quot;, &quot;Animals1_Output.txt&quot;)<br>    <br>    # This code checks your output file.<br>    def checkOutputFile(outputFilePath):<br>        if os.path.exists(outputFilePath):<br>            with open(outputFilePath) as outputFile:<br>                for line in outputFile:<br>                    print(line.rstrip(&quot;\n&quot;))<br>        else:<br>            print(f&quot;No output file exists at {outputFilePath}.&quot;)<br>    <br>    checkOutputFile(&quot;Animals1_Output.txt&quot;)<br><br>Below is the expected output:<br><br>    cat<br>    dog<br>    parrot<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    deleteDuplicates(&quot;Animals2.txt&quot;, &quot;Animals2_Output.txt&quot;)<br>    <br>    # This code checks your output file.<br>    def checkOutputFile(outputFilePath):<br>        if os.path.exists(outputFilePath):<br>            with open(outputFilePath) as outputFile:<br>                for line in outputFile:<br>                    print(line.rstrip(&quot;\n&quot;))<br>        else:<br>            print(f&quot;No output file exists at {outputFilePath}.&quot;)<br>    <br>    checkOutputFile(&quot;Animals2_Output.txt&quot;)<br><br>Below is the expected output:<br><br>    beetle<br>    horse<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    import os<br>    <br>    deleteDuplicates(&quot;Animals3.txt&quot;, &quot;Animals3_Output.txt&quot;)<br>    <br>    # This code checks your output file.<br>    def checkOutputFile(outputFilePath):<br>        if os.path.exists(outputFilePath):<br>            with open(outputFilePath) as outputFile:<br>                for line in outputFile:<br>                    print(line.rstrip(&quot;\n&quot;))<br>        else:<br>            print(f&quot;No output file exists at {outputFilePath}.&quot;)<br>    <br>    checkOutputFile(&quot;Animals3_Output.txt&quot;)<br><br>Below is the expected output:<br><br>    elephant<br>    hummingbird<br>    wasp<br>    rat<br>    duck<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Animals1.txt<br><br>File contents:<br><br>    cat cat cat<br>    dog dog dog dog dog dog dog dog<br>    parrot parrot<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Animals2.txt<br><br>File contents:<br><br>    beetle beetle<br>    horse horse horse horse horse horse horse<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Animals3.txt<br><br>File contents:<br><br>    elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant elephant<br>    hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird hummingbird<br>    wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp wasp<br>    rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat rat<br>    duck duck duck duck duck duck</td><td class="s1" dir="ltr">Thanks to Brandon Pickett for conceiving the idea for this exercise.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def deleteDuplicates(file_path, new_file_path):<br>    # Open the input file and the new file for writing<br>    with open(file_path, &#39;r&#39;) as input_file, open(new_file_path, &#39;w&#39;) as output_file:<br>        # Read the input file line by line<br>        for line in input_file:<br>            # Remove all duplicates from the line using regex<br>            line = re.sub(r&#39;\b(\w+)\b(?:\s+\1\b)+&#39;, r&#39;\1&#39;, line)<br>            # Write the line to the new file<br>            output_file.write(line)</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R176" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">177</div></th><td class="s1">19 - Additional Practice</td><td class="s1">02 - Extract species name</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Imagine you have a bunch of text files that you want to rename. When you created the files, you inadvertently added a bunch of seemingly random characters somewhere in the file names. Also embedded in each file name is the name of a species you had been profiling via DNA sequencing. Please write a function called &quot;extractSpeciesName&quot; that accepts a single argument: a string that specifies one of these file names. Your function should extract the species name from the file name. Your function should also extract the file extension from the file name. Then your function should return a concatenation of the species name and the file extension and return this modified file name. For example, if the current name of your file were &quot;Xa869cyccahd9sdt7fas7ey9Mus\_musculus88y89yaAYGH85ABIASUhkjk.txt&quot;, your function would return &quot;Mus\_musculus.txt&quot;.<br><br>Note that the format of the species names will be &quot;Genus_species&quot;. There will be one underscore between the genus and species names in the file name and only one period (right before the file extension). The Genus will always start with a capital letter. The genus and species names will each contain at least 2 characters.<br><br>Hint(s):<br><br>    If needed, use https://regex101.com to figure out a regular expression that will work.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(extractSpeciesName(&quot;Xa869cyccahd9sdt7fas7ey9Mus_musculus88y89yaAYGH85ABIASUhkjk.vcf&quot;))<br><br>Below is the expected output:<br><br>    Mus_musculus.vcf<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(extractSpeciesName(&quot;Homo_sapiens88y89yaAYGH85ABIASUhkjk.fa&quot;))<br><br>Below is the expected output:<br><br>    Homo_sapiens.fa<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    print(extractSpeciesName(&quot;Xa869cyccahd9sdt7fas7ey9Pan_paniscus.fastq&quot;))<br><br>Below is the expected output:<br><br>    Pan_paniscus.fastq</td><td class="s1" dir="ltr">Thanks to Brandon Pickett for conceiving the idea for this exercise.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def extractSpeciesName(filename):<br>    species_name = re.search(&quot;[A-Z][a-z]+_[a-z]+&quot;, filename).group()<br>    file_extension = re.search(&quot;\.[a-z]+$&quot;, filename).group()<br>    return species_name + file_extension</td><td class="s2">3</td><td class="s2" dir="ltr">3</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R177" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">178</div></th><td class="s1">19 - Additional Practice</td><td class="s1">03 - Evaluate Tic-Tac-Toe board</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Please write a function called &quot;evaluateTicTacToe&quot; that accepts one argument: a nested list. This nested list will consist of an &quot;outer&quot; list that has 3 &quot;inner&quot; lists within it. Each inner list will have 3 string values (each will be either &quot;X&quot; or &quot;O&quot;). Below is an example:<br><br>    [[&quot;X&quot;, &quot;O&quot;, &quot;X&quot;], [&quot;O&quot;, &quot;O&quot;, &quot;X&quot;], [&quot;O&quot;, &quot;X&quot;, &quot;X&quot;]] <br><br>You could think of this as being like a Tic-Tac-Toe board. To illustrate this better, we can change the formatting:<br><br>    [[&quot;X&quot;, &quot;O&quot;, &quot;X&quot;],<br>     [&quot;O&quot;, &quot;O&quot;, &quot;X&quot;],<br>     [&quot;O&quot;, &quot;X&quot;, &quot;X&quot;]] <br><br>In the example above, the &quot;X&quot; player would win Tic-Tac-Toe because she had 3 consecutive &quot;X&quot; values (going vertical in the third column).<br><br>Your function should return a String value that indicates the outcome of the game. These outcomes could &quot;X&quot;, &quot;O&quot;, or &quot;Draw&quot;. An &quot;X&quot; would indicate that there were three consecutive X&#39;s. These could be vertical, horizontal, or diagonal in any direction. An &quot;O&quot; would indicate that &quot;O&quot; had won (vertically, horizontally, or diagonally). A &quot;Draw&quot; would indicate that neither &quot;X&quot; nor &quot;O&quot; had won.<br><br>Hint(s):<br><br>    This would be a great opportunity to divide the code into multiple functions. The logic for checking &quot;X&quot; values can basically be identical to the logic for checking &quot;O&quot; values.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    game1 = [[&quot;X&quot;, &quot;O&quot;, &quot;X&quot;],<br>             [&quot;O&quot;, &quot;O&quot;, &quot;X&quot;],<br>             [&quot;O&quot;, &quot;X&quot;, &quot;X&quot;]]<br>    <br>    print(evaluateTicTacToe(game1))<br><br>Below is the expected output:<br><br>    X<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    game2 = [[&quot;O&quot;, &quot;O&quot;, &quot;X&quot;],<br>             [&quot;X&quot;, &quot;O&quot;, &quot;X&quot;],<br>             [&quot;O&quot;, &quot;X&quot;, &quot;O&quot;]]<br>    <br>    print(evaluateTicTacToe(game2))<br><br>Below is the expected output:<br><br>    O<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    game3 = [[&quot;O&quot;, &quot;X&quot;, &quot;O&quot;],<br>             [&quot;O&quot;, &quot;O&quot;, &quot;X&quot;],<br>             [&quot;X&quot;, &quot;X&quot;, &quot;X&quot;]]<br>    <br>    print(evaluateTicTacToe(game3))<br><br>Below is the expected output:<br><br>    X<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    game4 = [[&quot;O&quot;, &quot;O&quot;, &quot;X&quot;],<br>             [&quot;O&quot;, &quot;X&quot;, &quot;X&quot;],<br>             [&quot;X&quot;, &quot;O&quot;, &quot;O&quot;]]<br>    <br>    print(evaluateTicTacToe(game4))<br><br>Below is the expected output:<br><br>    X<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    game5 = [[&quot;X&quot;, &quot;O&quot;, &quot;X&quot;],<br>             [&quot;O&quot;, &quot;O&quot;, &quot;O&quot;],<br>             [&quot;X&quot;, &quot;X&quot;, &quot;O&quot;]]<br>    <br>    print(evaluateTicTacToe(game5))<br><br>Below is the expected output:<br><br>    O<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    game6 = [[&quot;X&quot;, &quot;O&quot;, &quot;O&quot;],<br>             [&quot;O&quot;, &quot;X&quot;, &quot;O&quot;],<br>             [&quot;X&quot;, &quot;X&quot;, &quot;O&quot;]]<br>    <br>    print(evaluateTicTacToe(game6))<br><br>Below is the expected output:<br><br>    O<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    game7 = [[&quot;X&quot;, &quot;O&quot;, &quot;O&quot;],<br>             [&quot;O&quot;, &quot;X&quot;, &quot;X&quot;],<br>             [&quot;X&quot;, &quot;X&quot;, &quot;O&quot;]]<br>    <br>    print(evaluateTicTacToe(game7))<br><br>Below is the expected output:<br><br>    Draw<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following code will be executed after your code:<br><br>    game8 = [[&quot;O&quot;, &quot;X&quot;, &quot;O&quot;],<br>             [&quot;O&quot;, &quot;X&quot;, &quot;X&quot;],<br>             [&quot;X&quot;, &quot;O&quot;, &quot;X&quot;]]<br>    <br>    print(evaluateTicTacToe(game8))<br><br>Below is the expected output:<br><br>    Draw</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">No</td><td class="s1" dir="ltr">No</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">def evaluateTicTacToe(game):<br>    def check_rows(game):<br>        for row in game:<br>            if all(val == &#39;X&#39; for val in row):<br>                return &#39;X&#39;<br>            elif all(val == &#39;O&#39; for val in row):<br>                return &#39;O&#39;<br>        return None<br>    <br>    def check_columns(game):<br>        for col in zip(*game):<br>            if all(val == &#39;X&#39; for val in col):<br>                return &#39;X&#39;<br>            elif all(val == &#39;O&#39; for val in col):<br>                return &#39;O&#39;<br>        return None<br>    <br>    def check_diagonals(game):<br>        if all(val == &#39;X&#39; for val in [game[i][i] for i in range(3)]):<br>            return &#39;X&#39;<br>        elif all(val == &#39;O&#39; for val in [game[i][i] for i in range(3)]):<br>            return &#39;O&#39;<br>        <br>        if all(val == &#39;X&#39; for val in [game[i][2-i] for i in range(3)]):<br>            return &#39;X&#39;<br>        elif all(val == &#39;O&#39; for val in [game[i][2-i] for i in range(3)]):<br>            return &#39;O&#39;<br>        return None<br>    <br>    result = check_rows(game)<br>    if result:<br>        return result<br>    <br>    result = check_columns(game)<br>    if result:<br>        return result<br>    <br>    result = check_diagonals(game)<br>    if result:<br>        return result<br>    <br>    return &quot;Draw&quot;<br></td><td class="s2">8</td><td class="s2" dir="ltr">8</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">No</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R178" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">179</div></th><td class="s1">19 - Additional Practice</td><td class="s1">04 - Find highest expressed gene</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>Idiopathic pulmonary fibrosis (IPF) is a disease that causes lung tissue to form scars. This scarring prevents the person from breathing properly and eventually may cause lung failure. For most patients, physicians cannot determine a specific cause, and there is currently no cure. Accordingly, there is an urgent need for researchers to understand more about IPF.<br><br>Researchers at the University of Nevada School of Medicine performed a study to examine 115 people who had either been diagnosed with IPF or who were healthy controls. They used gene-expression microarrays to profile blood cells from each person. In hopes of understanding more about the disease, the researchers searched for genes that are expressed differently between the two groups (IPF vs. healthy). The researchers posted their raw data online. For this exercise, you will use simplified versions of the data.<br><br>Each line (not including the header) contains data for a single patient. The values are tab-delimited. The first column indicates whether each patient was healthy or had IPF. The remaining columns contain gene-expression values for specific genes. The gene identifiers (for example, &quot;ENSG00000159579&quot;) are from the Ensembl database.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a string and will be a path to one of above-mentioned data files. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names.<br><br>Please print the name of the gene that has the highest average expression across all samples in the data. (You might need to go back and look at the slides to remember the variety of ways that you can work with pandas DataFrames.)<br><br>Hint(s):<br><br>    You can solve this exercise with &quot;pandas&quot;. Or you can do it without using &quot;pandas&quot;. Your code will probably be shorter and simpler if you use &quot;pandas&quot;, but it is up to you.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py GeneExpression1.txt<br>    python code.py GeneExpression2.txt<br><br>Below is a description of a data file that is available for testing.<br><br>File name: GeneExpression1.txt<br><br>File contents:<br><br>    Status        ENSG00000159579        ENSG00000072506        ENSG00000076351        ENSG00000182013<br>    Healthy        0.901418872166667        1.0453540916        -0.0625931783333333        -0.0955378835714286<br>    Healthy        0.852689667        1.086381914        0.0176642633333333        -0.122901103571429<br>    Healthy        0.936391221166667        1.0534241872        -0.0134109266666667        -0.238341259642857<br>    Healthy        0.867417970166667        0.8648321492        -0.101131300555556        -0.184759876785714<br>    Healthy        0.846749336666667        1.14715143        0.06090428        -0.116575965<br>    Healthy        0.689366527        0.8291884252        -0.0519287227777778        -0.137364478571429<br>    Healthy        0.845154432        1.0593610368        -0.0354302383333333        -0.0731305946428571<br>    Healthy        0.908416151666667        0.852182536        -0.0301351261111111        -0.15024738<br>    Healthy        0.887684473166667        0.9728788796        -0.107534338333333        -0.167761058571429<br>    Healthy        1.01546799933333        0.8561614732        0.0173485572222222        -0.1496827525<br>    Healthy        0.856701790833333        0.997844604        -0.04300791        -0.138005396071429<br>    Healthy        0.730538818        0.8026989732        -0.036602265        -0.156036952142857<br>    Healthy        0.924559968833333        0.979618708        0.078872005        -0.160277516071429<br>    Healthy        0.8006456655        0.8350474788        -0.0163587705555556        -0.1376621975<br>    Healthy        0.7037035845        0.880491632        -0.106547226666667        -0.162186849642857<br>    Healthy        0.933686961333333        1.1800993076        0.0884679672222222        -0.228896463214286<br>    Healthy        0.9048968025        0.8085201172        -0.0349399588888889        -0.119709941785714<br>    Healthy        0.999194695666667        1.0480830812        0.134185595555556        -0.131954875<br>    Healthy        0.858187995166667        0.7057345348        -0.0507829066666667        -0.13122804<br>    Healthy        0.945059782166667        1.0495981688        -0.0134510788888889        -0.121611294285714<br>    Healthy        0.970030449166667        1.1538201612        0.124746165        0.127027653214286<br>    Healthy        0.949303858666667        0.951182588        0.0493951061111111        -0.134408855357143<br>    Healthy        0.849962371333333        0.8543440908        -0.010503        -0.193122773214286<br>    Healthy        0.944543867833333        1.1286812156        -0.0817085294444445        -0.139713988214286<br>    Healthy        0.723879957833333        0.7400547208        -0.00547655222222222        -0.0768035935714286<br>    Healthy        0.78488339        0.6353006924        -0.200923601111111        -0.155326211071429<br>    Healthy        0.744153883333333        0.9008497412        -0.07053703        -0.0970768496428571<br>    Healthy        0.872937969333333        0.7996452684        -0.0233825588888889        -0.0762655957142857<br>    Healthy        0.835714428666667        0.7961986512        -0.0903501288888889        -0.144891903214286<br>    Healthy        0.779100297666667        0.7040681408        -0.01613057        -0.127893135714286<br>    Healthy        0.918804511        0.91016764000000006        -0.142328028888889        -0.0612514378571429<br>    Healthy        0.925546609666667        0.9745345532        0.0156268922222222        -0.170578621785714<br>    Healthy        0.9188464015        1.0738166176        0.0721684811111111        -0.144212220357143<br>    Healthy        0.978587744        0.9888923184        -0.0125857005555556        -0.164609526785714<br>    Healthy        0.920691954        0.9853163596        0.083731555        -0.102979059642857<br>    IPF        0.720443048666667        0.7338941568        9.84291111111114e-4        -0.128031595<br>    IPF        1.07926641116667        1.0300255204        0.075187275        -0.137688708214286<br>    IPF        0.878639459166667        0.8317262244        -0.130544552777778        -0.211162568571429<br>    IPF        0.965804671833333        1.043208338        0.01524487        -0.150379857857143<br>    IPF        0.650749944        0.532386004        -0.145610513888889        -0.119939463928571<br>    IPF        0.763675397833333        0.7091208648        -0.101152641666667        -0.122578410714286<br>    IPF        0.852651406        0.9704170852        0.0354188777777778        -0.191742822857143<br>    IPF        1.02372743        1.0860226916        -0.137424879444444        -0.134854125714286<br>    IPF        0.790027802333333        0.7476323872        -0.0991699511111111        -0.114474491428571<br>    IPF        0.975916547666667        0.9325066804        0.0303812661111111        -0.154771296785714<br>    IPF        0.9513219925        0.989518078        -0.166272451666667        -0.132009367142857<br>    IPF        0.963846274666667        0.8902793864        -0.0497895433333333        -0.176004594285714<br>    IPF        0.840522914666667        0.8192871796        -0.0922461683333333        -0.137833540714286<br>    IPF        0.913611431166667        0.9060389612        0.0506074005555555        -0.147349152142857<br>    IPF        0.7294963545        1.000832964        -0.142571270555556        -0.1457513175<br>    IPF        0.869973905        0.852444982        0.0193721011111111        -0.141593401071429<br>    IPF        0.579357269833333        0.9716875372        -0.051304025        -0.0610283596428571<br>    IPF        0.803148642333333        0.8946018148        -0.0425929161111111        -0.0671016092857143<br>    IPF        0.899024269666667        0.9185384152        -0.049246655        -0.123704763571429<br>    IPF        0.882625347666667        0.948106238        0.0465554        -0.166330613928571<br>    IPF        0.876259577        0.9131600172        -0.0527110466666667        -0.122539193214286<br>    IPF        0.9961458805        0.917440732        0.0227106677777778        -0.1473017175<br>    IPF        0.597356628833333        0.9407905148        -0.127197023333333        -0.281749252142857<br>    IPF        0.941969501833333        0.9450952256        0.0468287966666667        -0.124209095<br>    IPF        1.02417957983333        0.8916196848        0.0415853033333333        -0.182984304285714<br>    IPF        0.800099906166667        0.9388567496        -0.00173395833333334        -0.118866164285714<br>    IPF        0.920928301833333        0.8400791536        -0.0519121433333333        -0.139109325<br>    IPF        0.887203205        0.814396888        0.024027095        -0.1497251825<br>    IPF        0.7415509635        0.9221027608        -0.00304966722222222        -0.137841917142857<br>    IPF        0.918489527166667        0.8478946768        -0.0882695794444444        -0.148749078571429<br>    IPF        0.735468456833333        0.746920894        -0.178810784444444        -0.109697335357143<br>    IPF        0.85071585        0.9315748868        -0.101232737222222        -0.240460188928571<br>    IPF        0.931519249        0.9556234008        -0.0877214077777778        -0.0978784289285714<br>    IPF        0.829510049333333        0.9356840136        -0.167647637222222        -0.178547441428571<br>    IPF        1.01348211833333        0.9435583292        0.15070622        -0.1232065325<br>    IPF        0.717858250166667        1.2329865768        0.043022135        -0.141169659642857<br>    IPF        0.5559225695        0.9824713028        -0.00738506833333333        -0.230898773928571<br>    IPF        0.774489637833333        1.3324162752        0.149371057777778        -0.121583604642857<br>    IPF        0.587487637333333        0.9522709916        0.0214489638888889        -0.224078802142857<br>    IPF        0.684169514833333        1.2418631156        0.113565009444444        -0.176494091071429<br>    IPF        0.711292474333333        1.1715367356        -0.0389653472222222        -0.145883320357143<br>    IPF        0.729910574        1.2919526944        0.0406765461111111        -0.1960149575<br>    IPF        0.576948023833333        1.04785525        -0.019661200555555602        -0.163267704285714<br>    IPF        0.619548850833333        1.2166143008        0.0767119833333333        -0.153001948928571<br>    IPF        0.6084259785        1.1175399972        -0.0644907466666667        -0.170358809285714<br>    IPF        0.739188058        1.265155878        0.154262353333333        -0.2232026825<br>    IPF        0.633424281666667        1.249297868        -0.162023042777778        -0.160127986785714<br>    IPF        0.761322311166667        1.2572963856        0.149726498333333        -0.1518233175<br>    IPF        0.836358715833333        1.1939343144        0.126403649444444        -0.128558856428571<br>    IPF        0.6757913565        1.212361594        0.0171236027777778        -0.219700829642857<br>    IPF        0.682284854333333        1.3787991304        0.111369551111111        -0.175493336071429<br>    IPF        0.8334994795        1.5197838864        0.0997885744444444        -0.208016201071429<br>    IPF        0.675695947        1.3516450676        -0.00706929777777779        -0.0737236085714286<br>    IPF        0.858684911833333        1.1634479912        -0.0486958494444444        -0.169946564642857<br>    IPF        0.815437001        1.1576340604        -0.0319780072222222        -0.173870832857143<br>    IPF        0.7703783005        1.215186202        0.0385871327777778        -0.125840873214286<br>    IPF        0.781148020666667        1.3363902        -0.05590485        -0.119963378928571<br>    IPF        0.698799648        1.2859848704        -0.0380738138888889        -0.140019709642857<br>    IPF        0.8354274325        1.2248707768        -0.0110229611111111        -0.163202683571429<br><br>Below is a description of a data file that is available for testing.<br><br>File name: GeneExpression2.txt<br><br>File contents:<br><br>    Status        ENSG00000170369        ENSG00000161681        ENSG00000260630        ENSG00000115307        ENSG00000153560        ENSG00000099783        ENSG00000083635        ENSG00000248360        ENSG00000269741        ENSG00000219073        ENSG00000185477        ENSG00000111536        ENSG00000136425        ENSG00000254835        ENSG00000198399        ENSG00000010322        ENSG00000169188        ENSG00000128654        ENSG00000164167        ENSG00000124721        ENSG00000127415        ENSG00000176485        ENSG00000196772        ENSG00000138161        ENSG00000065427        ENSG00000274750        ENSG00000180992        ENSG00000106333        ENSG00000261371        ENSG00000134755        ENSG00000205771        ENSG00000244752        ENSG00000085274        ENSG00000184385        ENSG00000189132        ENSG00000131831        ENSG00000168060        ENSG00000138160        ENSG00000027075        ENSG00000106853        ENSG00000163530        ENSG00000138650        ENSG00000170502        ENSG00000119865        ENSG00000226887        ENSG00000158764        ENSG00000116406        ENSG00000189169        ENSG00000132612        ENSG00000118298        ENSG00000136868        ENSG00000035681        ENSG00000133665        ENSG00000104889        ENSG00000087448        ENSG00000177051        ENSG00000243775        ENSG00000175646        ENSG00000062650        ENSG00000170802        ENSG00000154767        ENSG00000160505        ENSG00000139209        ENSG00000221953        ENSG00000159208        ENSG00000221878        ENSG00000255559        ENSG00000122591        ENSG00000172264        ENSG00000153823        ENSG00000100092        ENSG00000214380        ENSG00000184083        ENSG00000153233        ENSG00000149090        ENSG00000280414        ENSG00000204703        ENSG00000102271        ENSG00000151650        ENSG00000175920        ENSG00000099338        ENSG00000163959        ENSG00000143321        ENSG00000085662        ENSG00000173567        ENSG00000138642        ENSG00000179262        ENSG00000130005        ENSG00000185085        ENSG00000136450        ENSG00000125775        ENSG00000109610        ENSG00000175348        ENSG00000173818        ENSG00000126890        ENSG00000197125        ENSG00000166435        ENSG00000213760        ENSG00000124784        ENSG00000099960<br>    Healthy        0.346757602        -0.174518154479167        -0.108228932142857        0.993645367777778        1.02375798453333        1.66558000895522        0.20972439375        -0.190990445263158        0.7867176        -0.188769945        0.7601168636        -0.332861712727273        -0.179423339        -0.417491693333333        1.12235049923567        0.530160854380952        0.2581987134375        0.343646615        0.959725795        -0.154088635841584        0.12543940625        0.642487489        -0.24001035125        -0.0417506764444444        1.22429580082353        0.6199598575        0.329241928888889        0.0139117104        2.58933184388889        0.383468468591549        -0.1036352625        0.10724995        0.918787524583333        0.163495432857143        0.21177917125        -0.00641235625        -0.0625360903278689        -0.030448744691358        1.6399680515625        -0.140580364230769        -0.368561024444444        -0.0738890358730159        0.223735242432432        0.0455100231818182        -0.160047194347826        -0.266950949642857        1.27392101263736        -0.03404918        0.973685015        -0.103104792        1.196865259        1.22214349808989        -0.185203761515152        0.270724282727273        0.302460385555556        0.113290378095238        -0.112950541666667        -0.0925370685714286        1.43359292823529        1.241639853125        0.812508078641975        -0.22181350375        -0.205935675762712        -0.104997686666667        -0.0487296363636364        -0.100336031428571        -0.172426758888889        1.2756737892        -0.097253304375        0.441575950277778        0.397299853571429        0.100865412        0.0233730834246575        -0.221040782142857        -0.206588772391304        -0.0220289458333333        -0.163241129        -0.184009954230769        0.1269512375        0.154358077272727        0.0654918435869565        -0.0309875029032258        1.037277333125        0.074024324        -0.117555903428571        0.600334866282051        0.76699535        0.0334089890625        0.121851682        2.03864926645161        0.0372988716666667        -0.232393518        0.895285604516129        0.01433155275        -0.391907055        -0.242843601538462        0.140652152666667        -0.27465252        0.764035159473684        -0.082768680625<br>    Healthy        0.356500394        -0.161735866770833        -0.227582886428571        1.1013562812963        0.958431344266667        1.74575628156716        0.29370186625        -0.193523703157895        0.30023352        -0.10104922        0.5938258408        -0.348633742727273        -0.169633582        -0.35198374        0.99662136388535        0.503620751809524        0.268481875625        0.3642224678125        1.07959624142857        -0.151249570825083        0.0667248904166667        0.68183934        -0.30390552125        -0.0689791144444444        1.44320716435294        0.7157277525        0.525291082222222        -0.0754882604        2.61502041814815        0.455301346056338        -0.19494601        -0.162424532142857        0.778948571875        0.0861112935714286        0.06464488        -0.063918333125        -0.0860352562295082        -0.0708332183950617        1.2119544465625        -0.0778439319230769        -0.301880673333333        -0.105248322222222        0.279567273513514        0.0265917086363636        -0.134490842608696        -0.255536307142857        1.21385066692308        -0.04934389        0.9610362375        -0.162784205333333        1.120301911        1.19260041764045        -0.213064128181818        0.185783529393939        0.179620263888889        0.0486329152380952        -0.227866623333333        -0.115317645        1.39804520811765        1.284099318125        0.680236019506173        -0.26472482        -0.214223422711864        -0.112811595833333        -0.126534638409091        -0.182422342857143        -0.13019585        1.1124663262        -0.18910746125        0.684363483888889        0.387161068214286        0.034888108        0.0247396347945205        -0.232548076571429        -0.23865471326087        -0.06887642125        -0.153988891        -0.171812899807692        0.131374819375        0.173203133030303        -0.0889097843478261        -0.0171494477419355        1.0542583528125        0.132337925        -0.0910714294285714        0.463496392692308        0.667663151282051        0.054374680625        0.113942256        2.06948690870968        0.0319832129166667        -0.161800864        0.940962987741935        0.0368894095        -0.345791173        -0.173506808461538        0.0653697277333333        -0.2515425025        0.655838233333333        -0.10904323<br>    Healthy        0.253053196        -0.1608140796875        -0.234746420714286        1.36766454888889        1.22959223493333        1.6145675288806        0.23156811625        -0.154911018421053        0.001465175        -0.191992781        0.767249118        -0.332130297727273        -0.1558193195        -0.269563543333333        1.23629676630573        0.75173987647619        0.450806596875        0.402687691875        1.00012756857143        -0.177604520693069        0.301888603333333        0.539453451        -0.29755323625        -0.122153299555556        1.25027229082353        0.4451595925        0.551149574444444        0.1057094576        2.42688847        0.338486487605634        -0.036206385        0.00698041928571428        0.928432507083333        -0.0371892721428571        0.08990331        -0.125580075625        -4.05823114754101e-4        -0.0116817590123457        1.9197938815625        -0.110930815769231        -0.172173048888889        -0.167861114444444        0.277692827027027        -0.0463091981818182        -0.10364119        -0.240951431071429        1.30755896692308        0.0892468655555556        1.1972410075        -0.0854643183333333        0.890941022        1.21466797224719        -0.212152043333333        0.34891750030303        0.314121334444444        0.274626358571429        -0.238867458333333        -0.0628528035714286        1.38526361070588        1.4465148465625        0.792639531604938        -0.277256770714286        -0.170604457627119        -0.0363703533333333        -0.0760004802272727        -0.161203588571429        -0.190463115555556        1.1960485294        -0.113290229375        0.294676568888889        0.623604905357143        0.064068246        0.101625702191781        -0.228188884714286        -0.227199078913043        0.0572050204166667        -0.233173965        -0.0902616830769231        0.238056065        0.217234224848485        -0.0687503759782609        0.0462727912903226        1.27758343625        0.033579787        -0.0661425274285714        0.65073835051282        0.96154568        0.1298855340625        0.29332837        1.9258131383871        0.0578079216666667        -0.0713428705        1.04373549935484        0.106660731        -0.389627101        -0.299206089230769        0.6665902812        -0.3377424875        0.777697717894737        -0.09070101875<br>    Healthy        0.291168824        -0.214762561770833        -0.17906784        1.08623973648148        1.0792355996        1.69535385970149        0.17193059375        -0.117556268947368        0.517725615        -0.042994625        0.6378130548        -0.377371593181818        -0.2571701315        -0.396653956666667        1.33536117636943        0.532123248476191        0.2385669734375        0.3060323790625        0.826938717142857        -0.16243777356435599        0.0572935479166667        0.334573332        -0.433977575        -0.00596412555555555        1.24586859141176        0.728016145        0.359246341111111        0.0409488036        2.52179200333333        0.260984332394366        -0.00748596000000001        -0.06810473        0.984608805833333        0.0590286692857143        0.18525224875        -0.01276214375        -0.0587388901639344        -0.0696681809876543        1.4355508971875        -0.143415021923077        -0.232426743333333        -0.100689893492063        0.260334420810811        0.103126180454545        -0.13411560826087        -0.296782898928571        1.35185940351648        -0.192145243333333        0.883813235        -0.167162824333333        1.076400012        1.25678800382022        -0.25196927        0.167127321515152        0.314142718888889        0.0619312066666667        -0.399968125        -0.107997950714286        1.48234864658824        1.4451323375        0.787810382222222        -0.26079312375        -0.200227866949153        -0.145114806666667        -0.116847677272727        -0.155115595714286        -0.162667954444444        1.2399019286        -0.0158054825        0.5410380375        0.443510295        0.117775468        0.0367996615068493        -0.244699900285714        -0.209521868478261        -0.0592024779166667        -0.223043054        -0.212218364807692        0.212940640625        0.141153421515152        -0.10677155923913        0.0676968883870968        1.011923493125        -0.110778453        -0.056653368        0.641681878717949        0.709708893076923        -0.0382145375        0.042265442        1.92106102935484        0.103384511666667        -0.2211811585        0.695037273225806        -0.04583648175        -0.583641791        -0.268034353846154        0.538267832533333        -0.343655975        0.826769613157895        -0.05825890375<br>    Healthy        0.36357048        -0.120674191666667        -0.208843281428571        1.03592398055556        0.870399964133333        1.93606941253731        0.3589965575        -0.167238060526316        0.7043955        -0.117213627        0.7471522336        -0.268151695909091        -0.153926218        -0.20034166        0.908571675732484        0.568105250666667        0.3176476565625        0.340726815625        1.09965263357143        -0.18701834679868        0.14306197875        0.849455433        -0.27728298375        0.00356569666666667        1.43618045294118        0.6573796675        0.414124585555556        -0.005838812        2.50166228611111        0.399194312112676        0.0165586375        -0.351286731428571        0.686134221875        -0.04895674        0.06903035        -0.036466743125        -0.150297263606557        -0.085652020617284        1.7688339884375        -0.08005961        -0.285033098888889        -0.0969431285714286        0.242383048378378        0.0648121404545455        -0.107860243913043        -0.274200811071429        0.993129066923077        0.136495248888889        0.9170885525        -0.110343781        0.907259303        1.18796065539326        -0.216748057575758        0.169613512121212        0.319888007777778        0.198891736190476        -0.230979808333333        -0.235229054285714        1.50373212270588        1.1131451490625        0.679821310864198        -0.261619760357143        -0.22075726661017        -0.03169217        -0.108221572045455        0.01389991        -0.201875314444444        0.9965241734        -0.113491320625        0.526056638611111        0.462029115178571        0.117097824        0.0745193310958904        -0.239640270857143        -0.273837436304348        -0.04245830125        -0.134952537        -0.150869515384615        0.16711859        0.0839297681818182        -0.062206909673913        -0.0294534887096774        0.9356250753125        0.189770786        -0.119502913571429        0.417247788333333        0.789222308205128        0.087243855625        0.119010034        2.24680954935484        0.01744740375        -0.220081239        1.04846050870968        5.96929250000001e-4        -0.470914323        -0.314841553846154        0.133500616666667        -0.3850960825        0.57910340245614        -0.1528377825<br>    Healthy        0.268518826        -0.230150447083333        -0.366101175        1.24204881777778        1.04471567013333        1.76590341373134        -0.1185285725        -0.0986433115789474        0.00855644999999999        -0.25037999        0.7326153132        -0.325622696818182        -0.1347375995        -0.126538143333333        1.13675119299363        0.650298453809524        0.3098704596875        0.4484580171875        1.21586994214286        -0.164997194653465        0.0383535316666667        0.5823257        -0.35875289375        0.033302348        1.21860727941176        0.6904176225        0.498889185555556        -0.0617726148        2.28998670592593        0.393588969295775        0.06720431        -0.0359097228571428        0.851162787708333        0.0158391007142857        -0.00542449        -0.13570578        -0.0896537847540984        -0.0826587933333333        1.2335668384375        -0.059184005        -0.168506883333333        -0.0938464817460317        0.279623527027027        0.0492581286363636        -0.107687085652174        -0.285235420714286        1.39461745703297        0.0103739144444444        0.6610446725        -0.161514776666667        0.974584213        0.926475679101124        -0.322571637878788        0.133130203030303        0.216473536666667        0.0997798852380952        -0.222735505        -0.0656302735714286        1.10675925447059        1.4341418896875        0.762691779753086        -0.2503006075        -0.22148395440678        -0.255536965        -0.143717550909091        -0.200295377142857        -0.263938214444444        1.000270654        -0.1856174825        0.569621966111111        0.369030993214286        0.149046156        -0.0443002482191781        -0.211903178857143        -0.243350921304348        -0.129842240833333        -0.118011473        -0.194275378653846        0.014649766875        0.0758343524242424        -0.0802793916304348        -0.00737397838709677        1.126555679375        -0.1900047        -0.103599930571429        0.674010471025641        0.512733508461538        -0.0349183046875        -0.033020567        1.57843629870968        0.0604984158333333        -0.1270406945        0.903700428387097        0.01121044975        -0.657989685        -0.207333413076923        0.0922771910666667        -0.32683917        0.869072251929825        -0.054370970625<br>    Healthy        0.35872791        -0.119725247708333        -0.241616709285714        1.1257917162963        0.969110221466667        1.75199433320896        0.1551679        -0.0995196784210526        0.676672555        -0.216746933        0.6432686264        -0.312076558181818        -0.247610108        -0.19849657        1.02434842433121        0.600043064761905        0.29984540125        0.335386854375        1.17671679928571        -0.176008421320132        0.0981247195833333        0.7621251925        -0.34261090375        0.00778539177777778        1.44841412164706        0.73454446        0.403766166666667        -0.0610369596        2.47134176037037        0.496049294225352        -0.0417363375        -0.106606522142857        0.825828390208333        0.0370644714285714        0.32133517125        -0.0991504275        -0.0829155316393443        -0.106740891975309        1.47347493875        -0.104045513846154        -0.28286011        -0.100128019365079        0.438629388918919        -0.00982937636363636        -0.0734229730434783        -0.233022854642857        1.23659660857143        0.19763393        0.88526479        -0.158959958        1.061476943        1.06714904393258        -0.193892538181818        0.24080740030303        0.382560346666667        0.131303615238095        -0.284052941666667        -0.245216495        1.35867240705882        1.410140999375        0.631960024814815        -0.278490351964286        -0.266443043220339        -0.1159919325        -0.115567894090909        -0.32285518        -0.26595017        1.1251305384        -0.015780359375        0.377910265277778        0.494105655        0.053478572        0.0137009167123288        -0.275449821285714        -0.231747430217391        -0.0395196341666667        -0.1953441        -0.199086344807692        0.086675705        0.108928692424242        -0.0692496227173913        -0.0462666703225806        1.02874561375        0.038112303        -0.0949701208571429        0.465836403589744        0.601288373333333        -0.021568083125        0.185339423        2.10729910774194        0.0423623545833333        -0.1715166225        0.898349867419355        0.00293152799999999        -0.440821454        -0.228330847692308        0.0748618241333333        -0.02439218        0.651860672105263        -0.07256586875<br>    Healthy        0.340868194        -0.1573187196875        -0.168552873571429        1.06678700962963        1.34016872453333        1.66134403626866        0.3762336975        -0.0979810842105263        1.029417995        -0.300317874        0.9036126752        -0.318452289545455        -0.185442329        -0.363889306666667        1.22203377216561        0.641668646857143        0.270407484375        0.3298129025        0.831528434285714        -0.148994528514851        0.217826011458333        0.1762033955        -0.31714108875        0.0225166926666667        1.22233541247059        0.5614682        0.3188224        -0.0055674824        2.07261472666667        0.0305861292957746        -0.0337500125        0.106723210714286        0.8491103        0.0375724535714286        -0.06065037875        -0.161164504375        -0.0297345078688525        -0.0916141108641975        1.8382981990625        -0.150253473461538        -0.359655965555556        -0.113236367777778        0.117880032972973        0.0596115990909091        -0.142067710869565        -0.278041582142857        1.34831035879121        0.114141348888889        0.7010320575        -0.129520213        0.448593289        1.33686432764045        -0.224359236060606        0.0887772148484848        0.305790372222222        0.0328147452380952        -0.298907328333333        -0.0827835207142857        1.42916470141176        1.432839301875        0.776576978518518        -0.229271981607143        -0.225030452542373        -0.0927849708333334        -0.0476376306818182        -0.118907878571429        -0.273607218888889        1.1467312732        -0.111056365625        0.379521525277778        0.466516529821429        0.115378116        0.102594733835616        -0.230541474571429        -0.116758671086957        -0.0818317966666667        -0.26967451        -0.195637567115385        0.11161723875        0.212804946060606        -0.0666527870652174        -0.0425005838709677        1.10876396        -0.09087346        -0.11846254        0.573407577820513        0.786984337179487        0.106957305625        0.230409426        2.06272019419355        0.0103363758333333        -0.1374745195        0.760689949354839        0.032613467        -0.583048788        -0.216147775384615        0.169631714533333        -0.2335578        0.674638460877193        -0.082275248125<br>    Healthy        0.401934518        -0.1850600265625        -0.237061819285714        1.01580596388889        1.01287794226667        1.73045596365672        0.17893881125        -0.135094614736842        0.22830114        -0.162699884        0.709548602        -0.346844040454545        -0.205983528        -0.329395413333333        1.0427946343949        0.570675134095238        0.322154869375        0.2851922390625        0.961541688571429        -0.187647959405941        0.101037128333333        0.49755606        -0.35702362625        -0.01841985        1.32817586976471        0.76639426        0.45498165        0.0399365608        2.55561053333333        0.39595592056338        -0.0248294175        -0.191055620714286        0.643315649375        0.111061558571429        0.19364938125        -0.136175076875        -0.121396303770492        -0.0922232130864198        1.626044495625        -0.12981458        -0.388132556666667        -0.11734252904761901        0.0816290767567567        0.0669520559090909        -0.103677805217391        -0.249464778214286        1.17755166637363        0.0774416077777778        0.9387029275        -0.162186488        0.949436143        1.27192551853933        -0.253441594848485        0.149889628787879        0.151900162222222        0.00962794142857144        -0.26130156        -0.0522167342857143        1.406156696        1.2157705178125        0.572687715061728        -0.259215161607143        -0.256301706779661        -0.0972494583333333        -0.137834561363636        -0.101546294285714        -0.299491655555556        0.956091459        -0.13162969625        0.403443008055556        0.506213695714286        0.00983690799999999        0.0216972721917808        -0.244642606428571        -0.269545435652174        -0.1961643425        -0.259398225        -0.144730865384615        0.091097360625        0.0516010145454546        -0.0776609788043478        -0.107619010645161        0.8377079953125        -0.253846208        -0.155687464714286        0.480137641666667        0.609412538974359        0.1115907378125        0.183811693        2.05024733354839        0.0117455929166667        -0.1616232015        0.903210301612903        0.0287698005        -0.591567758        -0.33084672        0.400263564533333        -0.3862755325        0.501753579298246        -0.1102317875<br>    Healthy        0.169861974        -0.192109059583333        -0.247392382142857        1.09664566703704        1.0809720412        1.61273233298507        0.18853399625        -0.155112504210526        -0.167939255        -0.256259692        0.7444071968        -0.334069505        -0.214116231        -0.184978676666667        1.14361108515924        0.606064921619048        0.4100925734375        0.2957485259375        0.776762278571429        -0.154524421716172        0.218537005208333        0.2524199185        -0.312979545        0.110657873111111        1.23159175494118        0.6368377175        0.336534238888889        0.019290126        2.40827473425926        0.31559230028169        -0.2514389675        -0.121809594285714        0.801248969791667        0.0813906        0.08103767125        -0.151826415        -0.0965090004918033        -0.0499073551851852        1.5761775503125        -0.0916419215384615        -0.285743126666667        -0.137549803015873        0.128741334324324        0.0455702154545455        -0.0133328834782609        -0.264244396428571        1.19977044747253        0.0368408666666666        0.89531123        -0.097293318        1.081066925        1.22168168988764        -0.238223711515152        0.130675676060606        0.184781945        0.15165049047619        -0.335807926666667        -0.139345836428571        1.40783621270588        1.29817296625        0.688276343580247        -0.24931426375        -0.219485300338983        -0.284221269166667        -0.131414681363636        -0.0732947957142857        -0.14761493        1.17133261        -0.159832739375        0.0806888325        0.538361367142857        -0.016043304        0.12622466260274        -0.273061779142857        -0.227941858043478        -0.0970614354166667        -0.147541733        -0.226388729807692        0.14268790375        -0.0213724836363636        -0.0430323258695652        -0.0560554135483871        0.9371827115625        -0.240519076        -0.128779393428571        0.541752534230769        0.83468502        0.021152974375        0.05571325        1.98190097225806        0.00861847958333333        -0.255737078        0.787643466774194        -0.02320377475        -0.482628168        -0.271555382307692        0.123690218666667        -0.1476981975        0.689866911403509        -0.123453264375<br>    Healthy        0.43504397        -0.120728111666667        -0.142178458571429        1.26637949759259        1.13195481666667        1.79961018223881        0.41691595375        -0.112104933157895        0.946498475        -0.204536312        0.8562266924        -0.384462155454545        -0.235028992        -0.0869302733333333        1.29816510789809        0.68372284847619        0.3404402253125        0.479132803125        1.14603140571429        -0.168984360957096        0.117516225        0.327751653        -0.222949505        0.0794605295555555        1.35397575729412        0.5633061375        0.536983708888889        -0.0705932396        2.34419002833333        0.283019069014085        0.075279195        0.07105178        1.04371469645833        0.02265331        0.01571725875        -0.17221992375        -0.063120071147541        0.00400179160493827        1.688371255        -0.184231103076923        -0.394475766666667        -0.130348347142857        0.373488794054054        -0.0469270822727273        -0.139222091304348        -0.266716997142857        1.43132318384615        0.109666676666667        0.9190997425        -0.178100952666667        1.010300371        1.28268926932584        -0.243207091818182        0.260729698787879        0.356163211666667        0.192525628095238        -0.112050053333333        -0.0713947571428571        1.54044018047059        1.5385647921875        0.896069017037037        -0.231773842321429        -0.20328097779661        0.00215824999999998        -0.0759292220454545        0.00182992714285714        0.0105117655555556        1.2777738632        -0.205880295        0.418918151111111        0.422929457678571        0.2191736        0.10714643260274        -0.235244844571429        -0.174284275217391        0.0582842820833333        -0.237224873        -0.187634161730769        0.12800893875        0.221125073030303        -0.0503188165217391        0.0312370306451613        1.080222861875        0.040697595        -0.102934530571429        0.829462195769231        0.818221373333333        0.1217710921875        0.12236433        2.15807331387097        0.11843517        -0.1209788935        0.828988842258065        0.00305846175000001        -0.300300691        -0.244152443846154        0.220364475866667        -0.25033188        0.946570320350877        -0.0497264725<br>    Healthy        0.285265864        -0.153754146979167        -0.292672374285714        0.884662380740741        0.850419883866667        1.64892010507463        -0.0037355175        -0.176030438421053        0.63710599        -0.228137293        0.6549561968        -0.267422062272727        -0.175434499        -0.609693213333333        0.906649625477707        0.443345882952381        0.1894845946875        0.090080905        0.859696592857143        -0.152625947887789        0.0333661597916667        0.2488016555        -0.43388526125        -0.0967471597777778        1.22513790905882        0.8062835725        0.42165624        -0.0987359132        2.41650254425926        0.0737410711267606        -0.019746735        0.0606299214285714        0.512157061666667        0.0941749564285714        0.028135165        -0.081898571875        -0.116241576885246        -0.169143248765432        1.5719875015625        -0.130265993461538        -0.293822203333333        -0.131158263650794        0.0328491332432432        0.0313462340909091        -0.113748666521739        -0.243782711428571        0.940959754725275        -0.04907212        0.678890965        -0.186596118666667        0.862268237        1.00723832292135        -0.230794094848485        0.0876040124242424        0.294302941666667        0.0139336042857143        -0.407624151666667        -0.203559659285714        1.2499347629411801        1.0115031965625        0.359740411358025        -0.297138202857143        -0.210749071355932        -0.173706244166667        -0.169249741363636        -0.236982524285714        -0.241289348888889        0.8240465954        -0.1081904975        0.201109162222222        0.337972723035714        -0.21681558        -0.114135661917808        -0.239324817857143        -0.235329984347826        -0.245692093333333        -0.08981907        -0.180511790961538        0.081420301875        0.0363860475757576        -0.019031785        -0.0512535541935484        1.140718591875        0.047048728        -0.224048405428571        0.284128110641026        1.12215285641026        0.1096684940625        0.168394055        1.93404597322581        -0.00504589958333333        -0.0824983635        0.92349017483871        -0.00301933675000001        -0.448499587        -0.162081449230769        0.0316389249333333        -0.158479255        0.331516861403509        -0.115227639375<br>    Healthy        0.309599734        -0.158564705208333        -0.243790525714286        1.22586295851852        1.30104737306667        1.59462381932836        0.15645711375        -0.105274672631579        0.087701185        -0.15150655        0.8208965812        -0.375159864090909        -0.271804354        -0.369599316666667        1.26803753617834        0.766858892666667        0.4306443640625        0.4389372328125        0.903202115        -0.151482604026403        0.258155523958333        0.4983373135        -0.21304345        -0.101912644222222        1.19976763047059        0.5546919925        0.422472752222222        0.0846774        2.2634284537037        0.263181737042254        -0.05278329        -0.0804760314285714        0.828571808541667        -0.0307661807142857        -0.07172935625        -0.122975730625        -0.00727928885245902        -0.0999300754320988        1.8733040178125        -0.138716530769231        -0.333366122222222        -0.14555815031746        0.10402939027027        -0.02542354        -0.0998852852173913        -0.371022318571429        1.24713729120879        0.00802474444444445        0.97545554        -0.101442350666667        0.728842278        1.2385468411236        -0.252324970606061        0.209805481212121        0.316371812777778        0.225365492380952        -0.0225245866666667        -0.113539272142857        1.30809496835294        1.3886347609375        0.674398627654321        -0.283633160178571        -0.238579094745763        -0.198397405        -0.139190579090909        0.00860176428571428        -0.205601902222222        1.0362249572        -0.10923031        0.448433406111111        0.754619716428571        0.170722158        0.103454895479452        -0.247657147714286        -0.229874805217391        0.0547064725        -0.265280543        -0.159288761730769        0.287098936875        0.179223250606061        -0.0774978916304348        0.0417288322580645        0.9792281978125        -0.076234569        -0.0690754241428571        0.5744215        0.870281821282051        0.1069387375        0.241613657        1.87782258935484        0.0382666979166667        -0.16623578        0.846881227741936        -0.00912485525        -0.375743441        -0.222166838461538        0.207044689066667        -0.5125698675        0.647585174035088        -0.113247199375<br>    Healthy        0.315755404        -0.159529116770833        -0.194076037142857        1.17631770833333        1.25570775626667        1.6318</td><td class="s1" dir="ltr">The data came from here: <span style="text-decoration:underline;-webkit-text-decoration-skip:none;text-decoration-skip-ink:none;color:#1155cc;"><a target="_blank" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE38958.">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE38958.</a></span></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br><br>def main():<br>    # Get the file path from command line argument<br>    file_path = sys.argv[1]<br>    <br>    # Read the data file into a pandas DataFrame<br>    df = pd.read_csv(file_path, index_col=0, sep=&#39;\t&#39;)<br><br>    # Calculate the average expression for each gene<br>    gene_averages = df.mean(axis=0)<br><br>    # Get the gene with the highest average expression<br>    highest_gene = gene_averages.idxmax()<br><br>    # Print the name of the gene with the highest average expression<br>    print(highest_gene)<br><br>if __name__ == &#39;__main__&#39;:<br>    main()</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">5</td><td class="s1" dir="ltr">I limited the prompt to the first few lines of the data file.<br><br>The first attempts did not recognize that the input file was tab-delimited, so it used the default separator when reading the data using the pandas package. That caused an error, and ChatGPT could not figure it out. It could be justified that the prompt was not fully clear. I modified the prompt to indicate that the file is tab delimited. On the 4th attempt, it produced output that was nearly correct. However, it included extra text in the output. I asked it to fix this, and it did.</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R179" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">180</div></th><td class="s1">19 - Additional Practice</td><td class="s1">05 - Find largest absolute difference</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>This exercise uses the same data file as the previous exercise.<br><br>Please write a Python script that uses &quot;sys.argv&quot; to accept a single argument. This argument will be a string and will be a path to one of the data files. Use the &quot;pandas&quot; module to read the file into a &quot;DataFrame&quot;. When reading the data, you can use the &quot;index_col&quot; keyword argument to indicate that the first (0th) column contains row names. But you can also solve it without specifying a value for &quot;index_col&quot;. It&#39;s up to you.<br><br>Please calculate the following for each gene:<br><br>1. The average expression level for Healthy patients.<br>2. The average expression level for IPF patients.<br>3. The absolute value of the difference in average expression between Healthy and IPF patients.<br><br>Please print the name of the gene that has the highest absolute value of the difference in average expression.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py GeneExpression1.txt<br><br>Below is a description of a data file that is available for testing.<br><br>File name: GeneExpression1.txt<br><br>File contents:<br><br>    Status        ENSG00000159579        ENSG00000072506        ENSG00000076351        ENSG00000182013<br>    Healthy        0.901418872166667        1.0453540916        -0.0625931783333333        -0.0955378835714286<br>    Healthy        0.852689667        1.086381914        0.0176642633333333        -0.122901103571429<br>    Healthy        0.936391221166667        1.0534241872        -0.0134109266666667        -0.238341259642857<br>    Healthy        0.867417970166667        0.8648321492        -0.101131300555556        -0.184759876785714<br>    Healthy        0.846749336666667        1.14715143        0.06090428        -0.116575965<br>    Healthy        0.689366527        0.8291884252        -0.0519287227777778        -0.137364478571429<br>    Healthy        0.845154432        1.0593610368        -0.0354302383333333        -0.0731305946428571<br>    Healthy        0.908416151666667        0.852182536        -0.0301351261111111        -0.15024738<br>    Healthy        0.887684473166667        0.9728788796        -0.107534338333333        -0.167761058571429<br>    Healthy        1.01546799933333        0.8561614732        0.0173485572222222        -0.1496827525<br>    Healthy        0.856701790833333        0.997844604        -0.04300791        -0.138005396071429<br>    Healthy        0.730538818        0.8026989732        -0.036602265        -0.156036952142857<br>    Healthy        0.924559968833333        0.979618708        0.078872005        -0.160277516071429<br>    Healthy        0.8006456655        0.8350474788        -0.0163587705555556        -0.1376621975<br>    Healthy        0.7037035845        0.880491632        -0.106547226666667        -0.162186849642857<br>    Healthy        0.933686961333333        1.1800993076        0.0884679672222222        -0.228896463214286<br>    Healthy        0.9048968025        0.8085201172        -0.0349399588888889        -0.119709941785714<br>    Healthy        0.999194695666667        1.0480830812        0.134185595555556        -0.131954875<br>    Healthy        0.858187995166667        0.7057345348        -0.0507829066666667        -0.13122804<br>    Healthy        0.945059782166667        1.0495981688        -0.0134510788888889        -0.121611294285714<br>    Healthy        0.970030449166667        1.1538201612        0.124746165        0.127027653214286<br>    Healthy        0.949303858666667        0.951182588        0.0493951061111111        -0.134408855357143<br>    Healthy        0.849962371333333        0.8543440908        -0.010503        -0.193122773214286<br>    Healthy        0.944543867833333        1.1286812156        -0.0817085294444445        -0.139713988214286<br>    Healthy        0.723879957833333        0.7400547208        -0.00547655222222222        -0.0768035935714286<br>    Healthy        0.78488339        0.6353006924        -0.200923601111111        -0.155326211071429<br>    Healthy        0.744153883333333        0.9008497412        -0.07053703        -0.0970768496428571<br>    Healthy        0.872937969333333        0.7996452684        -0.0233825588888889        -0.0762655957142857<br>    Healthy        0.835714428666667        0.7961986512        -0.0903501288888889        -0.144891903214286<br>    Healthy        0.779100297666667        0.7040681408        -0.01613057        -0.127893135714286<br>    Healthy        0.918804511        0.91016764000000006        -0.142328028888889        -0.0612514378571429<br>    Healthy        0.925546609666667        0.9745345532        0.0156268922222222        -0.170578621785714<br>    Healthy        0.9188464015        1.0738166176        0.0721684811111111        -0.144212220357143<br>    Healthy        0.978587744        0.9888923184        -0.0125857005555556        -0.164609526785714<br>    Healthy        0.920691954        0.9853163596        0.083731555        -0.102979059642857<br>    IPF        0.720443048666667        0.7338941568        9.84291111111114e-4        -0.128031595<br>    IPF        1.07926641116667        1.0300255204        0.075187275        -0.137688708214286<br>    IPF        0.878639459166667        0.8317262244        -0.130544552777778        -0.211162568571429<br>    IPF        0.965804671833333        1.043208338        0.01524487        -0.150379857857143<br>    IPF        0.650749944        0.532386004        -0.145610513888889        -0.119939463928571<br>    IPF        0.763675397833333        0.7091208648        -0.101152641666667        -0.122578410714286<br>    IPF        0.852651406        0.9704170852        0.0354188777777778        -0.191742822857143<br>    IPF        1.02372743        1.0860226916        -0.137424879444444        -0.134854125714286<br>    IPF        0.790027802333333        0.7476323872        -0.0991699511111111        -0.114474491428571<br>    IPF        0.975916547666667        0.9325066804        0.0303812661111111        -0.154771296785714<br>    IPF        0.9513219925        0.989518078        -0.166272451666667        -0.132009367142857<br>    IPF        0.963846274666667        0.8902793864        -0.0497895433333333        -0.176004594285714<br>    IPF        0.840522914666667        0.8192871796        -0.0922461683333333        -0.137833540714286<br>    IPF        0.913611431166667        0.9060389612        0.0506074005555555        -0.147349152142857<br>    IPF        0.7294963545        1.000832964        -0.142571270555556        -0.1457513175<br>    IPF        0.869973905        0.852444982        0.0193721011111111        -0.141593401071429<br>    IPF        0.579357269833333        0.9716875372        -0.051304025        -0.0610283596428571<br>    IPF        0.803148642333333        0.8946018148        -0.0425929161111111        -0.0671016092857143<br>    IPF        0.899024269666667        0.9185384152        -0.049246655        -0.123704763571429<br>    IPF        0.882625347666667        0.948106238        0.0465554        -0.166330613928571<br>    IPF        0.876259577        0.9131600172        -0.0527110466666667        -0.122539193214286<br>    IPF        0.9961458805        0.917440732        0.0227106677777778        -0.1473017175<br>    IPF        0.597356628833333        0.9407905148        -0.127197023333333        -0.281749252142857<br>    IPF        0.941969501833333        0.9450952256        0.0468287966666667        -0.124209095<br>    IPF        1.02417957983333        0.8916196848        0.0415853033333333        -0.182984304285714<br>    IPF        0.800099906166667        0.9388567496        -0.00173395833333334        -0.118866164285714<br>    IPF        0.920928301833333        0.8400791536        -0.0519121433333333        -0.139109325<br>    IPF        0.887203205        0.814396888        0.024027095        -0.1497251825<br>    IPF        0.7415509635        0.9221027608        -0.00304966722222222        -0.137841917142857<br>    IPF        0.918489527166667        0.8478946768        -0.0882695794444444        -0.148749078571429<br>    IPF        0.735468456833333        0.746920894        -0.178810784444444        -0.109697335357143<br>    IPF        0.85071585        0.9315748868        -0.101232737222222        -0.240460188928571<br>    IPF        0.931519249        0.9556234008        -0.0877214077777778        -0.0978784289285714<br>    IPF        0.829510049333333        0.9356840136        -0.167647637222222        -0.178547441428571<br>    IPF        1.01348211833333        0.9435583292        0.15070622        -0.1232065325<br>    IPF        0.717858250166667        1.2329865768        0.043022135        -0.141169659642857<br>    IPF        0.5559225695        0.9824713028        -0.00738506833333333        -0.230898773928571<br>    IPF        0.774489637833333        1.3324162752        0.149371057777778        -0.121583604642857<br>    IPF        0.587487637333333        0.9522709916        0.0214489638888889        -0.224078802142857<br>    IPF        0.684169514833333        1.2418631156        0.113565009444444        -0.176494091071429<br>    IPF        0.711292474333333        1.1715367356        -0.0389653472222222        -0.145883320357143<br>    IPF        0.729910574        1.2919526944        0.0406765461111111        -0.1960149575<br>    IPF        0.576948023833333        1.04785525        -0.019661200555555602        -0.163267704285714<br>    IPF        0.619548850833333        1.2166143008        0.0767119833333333        -0.153001948928571<br>    IPF        0.6084259785        1.1175399972        -0.0644907466666667        -0.170358809285714<br>    IPF        0.739188058        1.265155878        0.154262353333333        -0.2232026825<br>    IPF        0.633424281666667        1.249297868        -0.162023042777778        -0.160127986785714<br>    IPF        0.761322311166667        1.2572963856        0.149726498333333        -0.1518233175<br>    IPF        0.836358715833333        1.1939343144        0.126403649444444        -0.128558856428571<br>    IPF        0.6757913565        1.212361594        0.0171236027777778        -0.219700829642857<br>    IPF        0.682284854333333        1.3787991304        0.111369551111111        -0.175493336071429<br>    IPF        0.8334994795        1.5197838864        0.0997885744444444        -0.208016201071429<br>    IPF        0.675695947        1.3516450676        -0.00706929777777779        -0.0737236085714286<br>    IPF        0.858684911833333        1.1634479912        -0.0486958494444444        -0.169946564642857<br>    IPF        0.815437001        1.1576340604        -0.0319780072222222        -0.173870832857143<br>    IPF        0.7703783005        1.215186202        0.0385871327777778        -0.125840873214286<br>    IPF        0.781148020666667        1.3363902        -0.05590485        -0.119963378928571<br>    IPF        0.698799648        1.2859848704        -0.0380738138888889        -0.140019709642857<br>    IPF        0.8354274325        1.2248707768        -0.0110229611111111        -0.163202683571429<br><br>Below is a description of a data file that is available for testing.<br><br>File name: GeneExpression2.txt<br><br>File contents:<br><br>    Status        ENSG00000170369        ENSG00000161681        ENSG00000260630        ENSG00000115307        ENSG00000153560        ENSG00000099783        ENSG00000083635        ENSG00000248360        ENSG00000269741        ENSG00000219073        ENSG00000185477        ENSG00000111536        ENSG00000136425        ENSG00000254835        ENSG00000198399        ENSG00000010322        ENSG00000169188        ENSG00000128654        ENSG00000164167        ENSG00000124721        ENSG00000127415        ENSG00000176485        ENSG00000196772        ENSG00000138161        ENSG00000065427        ENSG00000274750        ENSG00000180992        ENSG00000106333        ENSG00000261371        ENSG00000134755        ENSG00000205771        ENSG00000244752        ENSG00000085274        ENSG00000184385        ENSG00000189132        ENSG00000131831        ENSG00000168060        ENSG00000138160        ENSG00000027075        ENSG00000106853        ENSG00000163530        ENSG00000138650        ENSG00000170502        ENSG00000119865        ENSG00000226887        ENSG00000158764        ENSG00000116406        ENSG00000189169        ENSG00000132612        ENSG00000118298        ENSG00000136868        ENSG00000035681        ENSG00000133665        ENSG00000104889        ENSG00000087448        ENSG00000177051        ENSG00000243775        ENSG00000175646        ENSG00000062650        ENSG00000170802        ENSG00000154767        ENSG00000160505        ENSG00000139209        ENSG00000221953        ENSG00000159208        ENSG00000221878        ENSG00000255559        ENSG00000122591        ENSG00000172264        ENSG00000153823        ENSG00000100092        ENSG00000214380        ENSG00000184083        ENSG00000153233        ENSG00000149090        ENSG00000280414        ENSG00000204703        ENSG00000102271        ENSG00000151650        ENSG00000175920        ENSG00000099338        ENSG00000163959        ENSG00000143321        ENSG00000085662        ENSG00000173567        ENSG00000138642        ENSG00000179262        ENSG00000130005        ENSG00000185085        ENSG00000136450        ENSG00000125775        ENSG00000109610        ENSG00000175348        ENSG00000173818        ENSG00000126890        ENSG00000197125        ENSG00000166435        ENSG00000213760        ENSG00000124784        ENSG00000099960<br>    Healthy        0.346757602        -0.174518154479167        -0.108228932142857        0.993645367777778        1.02375798453333        1.66558000895522        0.20972439375        -0.190990445263158        0.7867176        -0.188769945        0.7601168636        -0.332861712727273        -0.179423339        -0.417491693333333        1.12235049923567        0.530160854380952        0.2581987134375        0.343646615        0.959725795        -0.154088635841584        0.12543940625        0.642487489        -0.24001035125        -0.0417506764444444        1.22429580082353        0.6199598575        0.329241928888889        0.0139117104        2.58933184388889        0.383468468591549        -0.1036352625        0.10724995        0.918787524583333        0.163495432857143        0.21177917125        -0.00641235625        -0.0625360903278689        -0.030448744691358        1.6399680515625        -0.140580364230769        -0.368561024444444        -0.0738890358730159        0.223735242432432        0.0455100231818182        -0.160047194347826        -0.266950949642857        1.27392101263736        -0.03404918        0.973685015        -0.103104792        1.196865259        1.22214349808989        -0.185203761515152        0.270724282727273        0.302460385555556        0.113290378095238        -0.112950541666667        -0.0925370685714286        1.43359292823529        1.241639853125        0.812508078641975        -0.22181350375        -0.205935675762712        -0.104997686666667        -0.0487296363636364        -0.100336031428571        -0.172426758888889        1.2756737892        -0.097253304375        0.441575950277778        0.397299853571429        0.100865412        0.0233730834246575        -0.221040782142857        -0.206588772391304        -0.0220289458333333        -0.163241129        -0.184009954230769        0.1269512375        0.154358077272727        0.0654918435869565        -0.0309875029032258        1.037277333125        0.074024324        -0.117555903428571        0.600334866282051        0.76699535        0.0334089890625        0.121851682        2.03864926645161        0.0372988716666667        -0.232393518        0.895285604516129        0.01433155275        -0.391907055        -0.242843601538462        0.140652152666667        -0.27465252        0.764035159473684        -0.082768680625<br>    Healthy        0.356500394        -0.161735866770833        -0.227582886428571        1.1013562812963        0.958431344266667        1.74575628156716        0.29370186625        -0.193523703157895        0.30023352        -0.10104922        0.5938258408        -0.348633742727273        -0.169633582        -0.35198374        0.99662136388535        0.503620751809524        0.268481875625        0.3642224678125        1.07959624142857        -0.151249570825083        0.0667248904166667        0.68183934        -0.30390552125        -0.0689791144444444        1.44320716435294        0.7157277525        0.525291082222222        -0.0754882604        2.61502041814815        0.455301346056338        -0.19494601        -0.162424532142857        0.778948571875        0.0861112935714286        0.06464488        -0.063918333125        -0.0860352562295082        -0.0708332183950617        1.2119544465625        -0.0778439319230769        -0.301880673333333        -0.105248322222222        0.279567273513514        0.0265917086363636        -0.134490842608696        -0.255536307142857        1.21385066692308        -0.04934389        0.9610362375        -0.162784205333333        1.120301911        1.19260041764045        -0.213064128181818        0.185783529393939        0.179620263888889        0.0486329152380952        -0.227866623333333        -0.115317645        1.39804520811765        1.284099318125        0.680236019506173        -0.26472482        -0.214223422711864        -0.112811595833333        -0.126534638409091        -0.182422342857143        -0.13019585        1.1124663262        -0.18910746125        0.684363483888889        0.387161068214286        0.034888108        0.0247396347945205        -0.232548076571429        -0.23865471326087        -0.06887642125        -0.153988891        -0.171812899807692        0.131374819375        0.173203133030303        -0.0889097843478261        -0.0171494477419355        1.0542583528125        0.132337925        -0.0910714294285714        0.463496392692308        0.667663151282051        0.054374680625        0.113942256        2.06948690870968        0.0319832129166667        -0.161800864        0.940962987741935        0.0368894095        -0.345791173        -0.173506808461538        0.0653697277333333        -0.2515425025        0.655838233333333        -0.10904323<br>    Healthy        0.253053196        -0.1608140796875        -0.234746420714286        1.36766454888889        1.22959223493333        1.6145675288806        0.23156811625        -0.154911018421053        0.001465175        -0.191992781        0.767249118        -0.332130297727273        -0.1558193195        -0.269563543333333        1.23629676630573        0.75173987647619        0.450806596875        0.402687691875        1.00012756857143        -0.177604520693069        0.301888603333333        0.539453451        -0.29755323625        -0.122153299555556        1.25027229082353        0.4451595925        0.551149574444444        0.1057094576        2.42688847        0.338486487605634        -0.036206385        0.00698041928571428        0.928432507083333        -0.0371892721428571        0.08990331        -0.125580075625        -4.05823114754101e-4        -0.0116817590123457        1.9197938815625        -0.110930815769231        -0.172173048888889        -0.167861114444444        0.277692827027027        -0.0463091981818182        -0.10364119        -0.240951431071429        1.30755896692308        0.0892468655555556        1.1972410075        -0.0854643183333333        0.890941022        1.21466797224719        -0.212152043333333        0.34891750030303        0.314121334444444        0.274626358571429        -0.238867458333333        -0.0628528035714286        1.38526361070588        1.4465148465625        0.792639531604938        -0.277256770714286        -0.170604457627119        -0.0363703533333333        -0.0760004802272727        -0.161203588571429        -0.190463115555556        1.1960485294        -0.113290229375        0.294676568888889        0.623604905357143        0.064068246        0.101625702191781        -0.228188884714286        -0.227199078913043        0.0572050204166667        -0.233173965        -0.0902616830769231        0.238056065        0.217234224848485        -0.0687503759782609        0.0462727912903226        1.27758343625        0.033579787        -0.0661425274285714        0.65073835051282        0.96154568        0.1298855340625        0.29332837        1.9258131383871        0.0578079216666667        -0.0713428705        1.04373549935484        0.106660731        -0.389627101        -0.299206089230769        0.6665902812        -0.3377424875        0.777697717894737        -0.09070101875<br>    Healthy        0.291168824        -0.214762561770833        -0.17906784        1.08623973648148        1.0792355996        1.69535385970149        0.17193059375        -0.117556268947368        0.517725615        -0.042994625        0.6378130548        -0.377371593181818        -0.2571701315        -0.396653956666667        1.33536117636943        0.532123248476191        0.2385669734375        0.3060323790625        0.826938717142857        -0.16243777356435599        0.0572935479166667        0.334573332        -0.433977575        -0.00596412555555555        1.24586859141176        0.728016145        0.359246341111111        0.0409488036        2.52179200333333        0.260984332394366        -0.00748596000000001        -0.06810473        0.984608805833333        0.0590286692857143        0.18525224875        -0.01276214375        -0.0587388901639344        -0.0696681809876543        1.4355508971875        -0.143415021923077        -0.232426743333333        -0.100689893492063        0.260334420810811        0.103126180454545        -0.13411560826087        -0.296782898928571        1.35185940351648        -0.192145243333333        0.883813235        -0.167162824333333        1.076400012        1.25678800382022        -0.25196927        0.167127321515152        0.314142718888889        0.0619312066666667        -0.399968125        -0.107997950714286        1.48234864658824        1.4451323375        0.787810382222222        -0.26079312375        -0.200227866949153        -0.145114806666667        -0.116847677272727        -0.155115595714286        -0.162667954444444        1.2399019286        -0.0158054825        0.5410380375        0.443510295        0.117775468        0.0367996615068493        -0.244699900285714        -0.209521868478261        -0.0592024779166667        -0.223043054        -0.212218364807692        0.212940640625        0.141153421515152        -0.10677155923913        0.0676968883870968        1.011923493125        -0.110778453        -0.056653368        0.641681878717949        0.709708893076923        -0.0382145375        0.042265442        1.92106102935484        0.103384511666667        -0.2211811585        0.695037273225806        -0.04583648175        -0.583641791        -0.268034353846154        0.538267832533333        -0.343655975        0.826769613157895        -0.05825890375<br>    Healthy        0.36357048        -0.120674191666667        -0.208843281428571        1.03592398055556        0.870399964133333        1.93606941253731        0.3589965575        -0.167238060526316        0.7043955        -0.117213627        0.7471522336        -0.268151695909091        -0.153926218        -0.20034166        0.908571675732484        0.568105250666667        0.3176476565625        0.340726815625        1.09965263357143        -0.18701834679868        0.14306197875        0.849455433        -0.27728298375        0.00356569666666667        1.43618045294118        0.6573796675        0.414124585555556        -0.005838812        2.50166228611111        0.399194312112676        0.0165586375        -0.351286731428571        0.686134221875        -0.04895674        0.06903035        -0.036466743125        -0.150297263606557        -0.085652020617284        1.7688339884375        -0.08005961        -0.285033098888889        -0.0969431285714286        0.242383048378378        0.0648121404545455        -0.107860243913043        -0.274200811071429        0.993129066923077        0.136495248888889        0.9170885525        -0.110343781        0.907259303        1.18796065539326        -0.216748057575758        0.169613512121212        0.319888007777778        0.198891736190476        -0.230979808333333        -0.235229054285714        1.50373212270588        1.1131451490625        0.679821310864198        -0.261619760357143        -0.22075726661017        -0.03169217        -0.108221572045455        0.01389991        -0.201875314444444        0.9965241734        -0.113491320625        0.526056638611111        0.462029115178571        0.117097824        0.0745193310958904        -0.239640270857143        -0.273837436304348        -0.04245830125        -0.134952537        -0.150869515384615        0.16711859        0.0839297681818182        -0.062206909673913        -0.0294534887096774        0.9356250753125        0.189770786        -0.119502913571429        0.417247788333333        0.789222308205128        0.087243855625        0.119010034        2.24680954935484        0.01744740375        -0.220081239        1.04846050870968        5.96929250000001e-4        -0.470914323        -0.314841553846154        0.133500616666667        -0.3850960825        0.57910340245614        -0.1528377825<br>    Healthy        0.268518826        -0.230150447083333        -0.366101175        1.24204881777778        1.04471567013333        1.76590341373134        -0.1185285725        -0.0986433115789474        0.00855644999999999        -0.25037999        0.7326153132        -0.325622696818182        -0.1347375995        -0.126538143333333        1.13675119299363        0.650298453809524        0.3098704596875        0.4484580171875        1.21586994214286        -0.164997194653465        0.0383535316666667        0.5823257        -0.35875289375        0.033302348        1.21860727941176        0.6904176225        0.498889185555556        -0.0617726148        2.28998670592593        0.393588969295775        0.06720431        -0.0359097228571428        0.851162787708333        0.0158391007142857        -0.00542449        -0.13570578        -0.0896537847540984        -0.0826587933333333        1.2335668384375        -0.059184005        -0.168506883333333        -0.0938464817460317        0.279623527027027        0.0492581286363636        -0.107687085652174        -0.285235420714286        1.39461745703297        0.0103739144444444        0.6610446725        -0.161514776666667        0.974584213        0.926475679101124        -0.322571637878788        0.133130203030303        0.216473536666667        0.0997798852380952        -0.222735505        -0.0656302735714286        1.10675925447059        1.4341418896875        0.762691779753086        -0.2503006075        -0.22148395440678        -0.255536965        -0.143717550909091        -0.200295377142857        -0.263938214444444        1.000270654        -0.1856174825        0.569621966111111        0.369030993214286        0.149046156        -0.0443002482191781        -0.211903178857143        -0.243350921304348        -0.129842240833333        -0.118011473        -0.194275378653846        0.014649766875        0.0758343524242424        -0.0802793916304348        -0.00737397838709677        1.126555679375        -0.1900047        -0.103599930571429        0.674010471025641        0.512733508461538        -0.0349183046875        -0.033020567        1.57843629870968        0.0604984158333333        -0.1270406945        0.903700428387097        0.01121044975        -0.657989685        -0.207333413076923        0.0922771910666667        -0.32683917        0.869072251929825        -0.054370970625<br>    Healthy        0.35872791        -0.119725247708333        -0.241616709285714        1.1257917162963        0.969110221466667        1.75199433320896        0.1551679        -0.0995196784210526        0.676672555        -0.216746933        0.6432686264        -0.312076558181818        -0.247610108        -0.19849657        1.02434842433121        0.600043064761905        0.29984540125        0.335386854375        1.17671679928571        -0.176008421320132        0.0981247195833333        0.7621251925        -0.34261090375        0.00778539177777778        1.44841412164706        0.73454446        0.403766166666667        -0.0610369596        2.47134176037037        0.496049294225352        -0.0417363375        -0.106606522142857        0.825828390208333        0.0370644714285714        0.32133517125        -0.0991504275        -0.0829155316393443        -0.106740891975309        1.47347493875        -0.104045513846154        -0.28286011        -0.100128019365079        0.438629388918919        -0.00982937636363636        -0.0734229730434783        -0.233022854642857        1.23659660857143        0.19763393        0.88526479        -0.158959958        1.061476943        1.06714904393258        -0.193892538181818        0.24080740030303        0.382560346666667        0.131303615238095        -0.284052941666667        -0.245216495        1.35867240705882        1.410140999375        0.631960024814815        -0.278490351964286        -0.266443043220339        -0.1159919325        -0.115567894090909        -0.32285518        -0.26595017        1.1251305384        -0.015780359375        0.377910265277778        0.494105655        0.053478572        0.0137009167123288        -0.275449821285714        -0.231747430217391        -0.0395196341666667        -0.1953441        -0.199086344807692        0.086675705        0.108928692424242        -0.0692496227173913        -0.0462666703225806        1.02874561375        0.038112303        -0.0949701208571429        0.465836403589744        0.601288373333333        -0.021568083125        0.185339423        2.10729910774194        0.0423623545833333        -0.1715166225        0.898349867419355        0.00293152799999999        -0.440821454        -0.228330847692308        0.0748618241333333        -0.02439218        0.651860672105263        -0.07256586875<br>    Healthy        0.340868194        -0.1573187196875        -0.168552873571429        1.06678700962963        1.34016872453333        1.66134403626866        0.3762336975        -0.0979810842105263        1.029417995        -0.300317874        0.9036126752        -0.318452289545455        -0.185442329        -0.363889306666667        1.22203377216561        0.641668646857143        0.270407484375        0.3298129025        0.831528434285714        -0.148994528514851        0.217826011458333        0.1762033955        -0.31714108875        0.0225166926666667        1.22233541247059        0.5614682        0.3188224        -0.0055674824        2.07261472666667        0.0305861292957746        -0.0337500125        0.106723210714286        0.8491103        0.0375724535714286        -0.06065037875        -0.161164504375        -0.0297345078688525        -0.0916141108641975        1.8382981990625        -0.150253473461538        -0.359655965555556        -0.113236367777778        0.117880032972973        0.0596115990909091        -0.142067710869565        -0.278041582142857        1.34831035879121        0.114141348888889        0.7010320575        -0.129520213        0.448593289        1.33686432764045        -0.224359236060606        0.0887772148484848        0.305790372222222        0.0328147452380952        -0.298907328333333        -0.0827835207142857        1.42916470141176        1.432839301875        0.776576978518518        -0.229271981607143        -0.225030452542373        -0.0927849708333334        -0.0476376306818182        -0.118907878571429        -0.273607218888889        1.1467312732        -0.111056365625        0.379521525277778        0.466516529821429        0.115378116        0.102594733835616        -0.230541474571429        -0.116758671086957        -0.0818317966666667        -0.26967451        -0.195637567115385        0.11161723875        0.212804946060606        -0.0666527870652174        -0.0425005838709677        1.10876396        -0.09087346        -0.11846254        0.573407577820513        0.786984337179487        0.106957305625        0.230409426        2.06272019419355        0.0103363758333333        -0.1374745195        0.760689949354839        0.032613467        -0.583048788        -0.216147775384615        0.169631714533333        -0.2335578        0.674638460877193        -0.082275248125<br>    Healthy        0.401934518        -0.1850600265625        -0.237061819285714        1.01580596388889        1.01287794226667        1.73045596365672        0.17893881125        -0.135094614736842        0.22830114        -0.162699884        0.709548602        -0.346844040454545        -0.205983528        -0.329395413333333        1.0427946343949        0.570675134095238        0.322154869375        0.2851922390625        0.961541688571429        -0.187647959405941        0.101037128333333        0.49755606        -0.35702362625        -0.01841985        1.32817586976471        0.76639426        0.45498165        0.0399365608        2.55561053333333        0.39595592056338        -0.0248294175        -0.191055620714286        0.643315649375        0.111061558571429        0.19364938125        -0.136175076875        -0.121396303770492        -0.0922232130864198        1.626044495625        -0.12981458        -0.388132556666667        -0.11734252904761901        0.0816290767567567        0.0669520559090909        -0.103677805217391        -0.249464778214286        1.17755166637363        0.0774416077777778        0.9387029275        -0.162186488        0.949436143        1.27192551853933        -0.253441594848485        0.149889628787879        0.151900162222222        0.00962794142857144        -0.26130156        -0.0522167342857143        1.406156696        1.2157705178125        0.572687715061728        -0.259215161607143        -0.256301706779661        -0.0972494583333333        -0.137834561363636        -0.101546294285714        -0.299491655555556        0.956091459        -0.13162969625        0.403443008055556        0.506213695714286        0.00983690799999999        0.0216972721917808        -0.244642606428571        -0.269545435652174        -0.1961643425        -0.259398225        -0.144730865384615        0.091097360625        0.0516010145454546        -0.0776609788043478        -0.107619010645161        0.8377079953125        -0.253846208        -0.155687464714286        0.480137641666667        0.609412538974359        0.1115907378125        0.183811693        2.05024733354839        0.0117455929166667        -0.1616232015        0.903210301612903        0.0287698005        -0.591567758        -0.33084672        0.400263564533333        -0.3862755325        0.501753579298246        -0.1102317875<br>    Healthy        0.169861974        -0.192109059583333        -0.247392382142857        1.09664566703704        1.0809720412        1.61273233298507        0.18853399625        -0.155112504210526        -0.167939255        -0.256259692        0.7444071968        -0.334069505        -0.214116231        -0.184978676666667        1.14361108515924        0.606064921619048        0.4100925734375        0.2957485259375        0.776762278571429        -0.154524421716172        0.218537005208333        0.2524199185        -0.312979545        0.110657873111111        1.23159175494118        0.6368377175        0.336534238888889        0.019290126        2.40827473425926        0.31559230028169        -0.2514389675        -0.121809594285714        0.801248969791667        0.0813906        0.08103767125        -0.151826415        -0.0965090004918033        -0.0499073551851852        1.5761775503125        -0.0916419215384615        -0.285743126666667        -0.137549803015873        0.128741334324324        0.0455702154545455        -0.0133328834782609        -0.264244396428571        1.19977044747253        0.0368408666666666        0.89531123        -0.097293318        1.081066925        1.22168168988764        -0.238223711515152        0.130675676060606        0.184781945        0.15165049047619        -0.335807926666667        -0.139345836428571        1.40783621270588        1.29817296625        0.688276343580247        -0.24931426375        -0.219485300338983        -0.284221269166667        -0.131414681363636        -0.0732947957142857        -0.14761493        1.17133261        -0.159832739375        0.0806888325        0.538361367142857        -0.016043304        0.12622466260274        -0.273061779142857        -0.227941858043478        -0.0970614354166667        -0.147541733        -0.226388729807692        0.14268790375        -0.0213724836363636        -0.0430323258695652        -0.0560554135483871        0.9371827115625        -0.240519076        -0.128779393428571        0.541752534230769        0.83468502        0.021152974375        0.05571325        1.98190097225806        0.00861847958333333        -0.255737078        0.787643466774194        -0.02320377475        -0.482628168        -0.271555382307692        0.123690218666667        -0.1476981975        0.689866911403509        -0.123453264375<br>    Healthy        0.43504397        -0.120728111666667        -0.142178458571429        1.26637949759259        1.13195481666667        1.79961018223881        0.41691595375        -0.112104933157895        0.946498475        -0.204536312        0.8562266924        -0.384462155454545        -0.235028992        -0.0869302733333333        1.29816510789809        0.68372284847619        0.3404402253125        0.479132803125        1.14603140571429        -0.168984360957096        0.117516225        0.327751653        -0.222949505        0.0794605295555555        1.35397575729412        0.5633061375        0.536983708888889        -0.0705932396        2.34419002833333        0.283019069014085        0.075279195        0.07105178        1.04371469645833        0.02265331        0.01571725875        -0.17221992375        -0.063120071147541        0.00400179160493827        1.688371255        -0.184231103076923        -0.394475766666667        -0.130348347142857        0.373488794054054        -0.0469270822727273        -0.139222091304348        -0.266716997142857        1.43132318384615        0.109666676666667        0.9190997425        -0.178100952666667        1.010300371        1.28268926932584        -0.243207091818182        0.260729698787879        0.356163211666667        0.192525628095238        -0.112050053333333        -0.0713947571428571        1.54044018047059        1.5385647921875        0.896069017037037        -0.231773842321429        -0.20328097779661        0.00215824999999998        -0.0759292220454545        0.00182992714285714        0.0105117655555556        1.2777738632        -0.205880295        0.418918151111111        0.422929457678571        0.2191736        0.10714643260274        -0.235244844571429        -0.174284275217391        0.0582842820833333        -0.237224873        -0.187634161730769        0.12800893875        0.221125073030303        -0.0503188165217391        0.0312370306451613        1.080222861875        0.040697595        -0.102934530571429        0.829462195769231        0.818221373333333        0.1217710921875        0.12236433        2.15807331387097        0.11843517        -0.1209788935        0.828988842258065        0.00305846175000001        -0.300300691        -0.244152443846154        0.220364475866667        -0.25033188        0.946570320350877        -0.0497264725<br>    Healthy        0.285265864        -0.153754146979167        -0.292672374285714        0.884662380740741        0.850419883866667        1.64892010507463        -0.0037355175        -0.176030438421053        0.63710599        -0.228137293        0.6549561968        -0.267422062272727        -0.175434499        -0.609693213333333        0.906649625477707        0.443345882952381        0.1894845946875        0.090080905        0.859696592857143        -0.152625947887789        0.0333661597916667        0.2488016555        -0.43388526125        -0.0967471597777778        1.22513790905882        0.8062835725        0.42165624        -0.0987359132        2.41650254425926        0.0737410711267606        -0.019746735        0.0606299214285714        0.512157061666667        0.0941749564285714        0.028135165        -0.081898571875        -0.116241576885246        -0.169143248765432        1.5719875015625        -0.130265993461538        -0.293822203333333        -0.131158263650794        0.0328491332432432        0.0313462340909091        -0.113748666521739        -0.243782711428571        0.940959754725275        -0.04907212        0.678890965        -0.186596118666667        0.862268237        1.00723832292135        -0.230794094848485        0.0876040124242424        0.294302941666667        0.0139336042857143        -0.407624151666667        -0.203559659285714        1.2499347629411801        1.0115031965625        0.359740411358025        -0.297138202857143        -0.210749071355932        -0.173706244166667        -0.169249741363636        -0.236982524285714        -0.241289348888889        0.8240465954        -0.1081904975        0.201109162222222        0.337972723035714        -0.21681558        -0.114135661917808        -0.239324817857143        -0.235329984347826        -0.245692093333333        -0.08981907        -0.180511790961538        0.081420301875        0.0363860475757576        -0.019031785        -0.0512535541935484        1.140718591875        0.047048728        -0.224048405428571        0.284128110641026        1.12215285641026        0.1096684940625        0.168394055        1.93404597322581        -0.00504589958333333        -0.0824983635        0.92349017483871        -0.00301933675000001        -0.448499587        -0.162081449230769        0.0316389249333333        -0.158479255        0.331516861403509        -0.115227639375<br>    Healthy        0.309599734        -0.158564705208333        -0.243790525714286        1.22586295851852        1.30104737306667        1.59462381932836        0.15645711375        -0.105274672631579        0.087701185        -0.15150655        0.8208965812        -0.375159864090909        -0.271804354        -0.369599316666667        1.26803753617834        0.766858892666667        0.4306443640625        0.4389372328125        0.903202115        -0.151482604026403        0.258155523958333        0.4983373135        -0.21304345        -0.101912644222222        1.19976763047059        0.5546919925        0.422472752222222        0.0846774        2.2634284537037        0.263181737042254        -0.05278329        -0.0804760314285714        0.828571808541667        -0.0307661807142857        -0.07172935625        -0.122975730625        -0.00727928885245902        -0.0999300754320988        1.8733040178125        -0.138716530769231        -0.333366122222222        -0.14555815031746        0.10402939027027        -0.02542354        -0.0998852852173913        -0.371022318571429        1.24713729120879        0.00802474444444445        0.97545554        -0.101442350666667        0.728842278        1.2385468411236        -0.252324970606061        0.209805481212121        0.316371812777778        0.225365492380952        -0.0225245866666667        -0.113539272142857        1.30809496835294        1.3886347609375        0.674398627654321        -0.283633160178571        -0.238579094745763        -0.198397405        -0.139190579090909        0.00860176428571428        -0.205601902222222        1.0362249572        -0.10923031        0.448433406111111        0.754619716428571        0.170722158        0.103454895479452        -0.247657147714286        -0.229874805217391        0.0547064725        -0.265280543        -0.159288761730769        0.287098936875        0.179223250606061        -0.0774978916304348        0.0417288322580645        0.9792281978125        -0.076234569        -0.0690754241428571        0.5744215        0.870281821282051        0.1069387375        0.241613657        1.87782258935484        0.0382666979166667        -0.16623578        0.846881227741936        -0.00912485525        -0.375743441        -0.222166838461538        0.207044689066667        -0.5125698675        0.647585174035088        -0.113247199375<br>    Healthy        0.315755404        -0.159529116770833        -0.194076037142857        1.17631770833333        1.25570775626667        1.63186545865672        0.0021537825        -0.0999085605263158        0.192925515        -0.175242923        0.7424397548        -0.316662938181818        -0.216075697        -0.408719073333333        1.36300959993631        0.747616166380952        0.2414984284375        0.4108583365625        1.04403660285714        -0.132508870528053        0.172053627708333        0.549970302        -0.26999182375        -0.0277436673333333        1.31226889129412        0.7227029575        0.572818395555556        0.0639338668        2.33908538203704        0.230366037887324        0.193284915        -0.0431853207142857        0.944353637708333        -0.0319106985714286        0.0166703375        -0.24618402625        -0.102810814754098        -0.151004597037037        1.584770464375        -0.0914347953846154        -0.185335871111111        -0.0743019788888889        0.297492942702703        0.0838367395454545        -0.073391392173913        -0.177456138928571        1.42021653        -0.112752438888889        0.855569895        -0.121487794333333        0.706974381        1.10695410314607        -0.337439866363636        0.118497632424242        0.156028635555556        0.00350154523809524        -0.101646135        -0.161039267857143        1.16577604635294        1.5224440540625        0.797881516419753        -0.245465779107143        -0.214847228644068        -0.0550328558333333        -0.0996543586363636        -0.334431638571429        -0.133334032222222        1.1698203152        -0.13172442375        0.604326944166667        0.468580223571429        0.150938266        -0.07429684301</td><td class="s1" dir="ltr">The data came from here: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE38958.</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br>import pandas as pd<br><br>def main():<br>    # Get the file path from command line argument<br>    file_path = sys.argv[1]<br>    <br>    # Read the data file into a pandas DataFrame<br>    df = pd.read_csv(file_path, index_col=0, sep=&#39;\t&#39;)<br><br>    # Calculate the average expression for each gene<br>    gene_averages = df.mean(axis=0)<br><br>    # Get the gene with the highest average expression<br>    highest_gene = gene_averages.idxmax()<br><br>    # Print the name of the gene with the highest average expression<br>    print(highest_gene)<br><br>if __name__ == &#39;__main__&#39;:<br>    main()</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1" dir="ltr">I limited the prompt to the first few lines of the data file.</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R180" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">181</div></th><td class="s1">19 - Additional Practice</td><td class="s1">06 - Match peptides to genes</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>Researchers are working on ways to use peptides (short sequences of amino acids) to target human cells. It is hoped that these peptides can be used as treatments for certain diseases. For example, if a peptide can effectively target cancer cells, it may be useful in transporting drugs to those cells and killing them, while leaving normal cells intact. Suppose you are working with a few peptides, and you want to produce large quantities of those peptides. You know the amino-acid sequence for each of these peptides, but you don&#39;t know any genes that produce the peptides (within their full protein products). You want to find this out so you can clone the gene and insert it into bacteria to mass produce the peptides.<br><br>Please write a Python script that accepts three arguments: 1) an amino-acid sequence of a peptide (a string), 2) the name of a FASTA file that contains multiple gene sequences, and 3) the name of an output file that your script must create. Each header line in the FASTA file will consist of an angle bracket (&gt;) followed by the name of a gene. Each header line will be followed by one or more lines that contain the DNA sequence for that gene.<br><br>Your script should search the FASTA file for any gene(s) that produce the specified peptide. Your script should write the name of each gene that could have produced the specified peptide (in the same order they were specified in the FASTA file) to the output file.<br><br>For simplicity, you can assume that there are only 4 amino acids: Leucine (L), Arginine (R), Serine (S), and Tyrosine (T). The list below indicates which codons (a sequence of 3 DNA nucleotides) could have coded for each of these amino acids:<br><br> L = CTA, CTC, CTG, or CTT<br> R = AGA or AGG<br> S = AGC or AGT<br> T = ACA, ACC, ACG, or ACT<br><br>For example, if you were given a peptide sequence of RSS, the corresponding DNA sequence could have been any of the following:<br><br>    AGAAGCAGC<br>    AGAAGCAGT<br>    AGAAGTAGT<br>    AGAAGTAGC<br>    AGGAGCAGC<br>    AGGAGCAGT<br>    AGGAGTAGT<br>    AGGAGTAGC<br><br>Your script would need to search the FASTA file for any gene(s) that contains any of these sequences.<br><br>Hint(s):<br><br>    Regular expressions will probably be helpful to you in solving this exercise.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py &quot;SLTRSSRS&quot; &quot;DNA1.fasta&quot; &quot;Matches1.txt&quot;<br>    <br>    echo Matches1.txt<br>    cat Matches1.txt<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA1.fasta<br><br>File contents:<br><br>    &gt;Gene1<br>    TGCTACAGGTGGCAAGAGTTGTATATTGGTATATAGACGA<br>    CTGGACGGCCTTTATAGCCTTACTAGGAGCAGTAGAAGCT<br>    TATGCCCAATTGAATCCT<br>    &gt;Gene2<br>    AACTCCGTCTGATCACCGTCATGGAAGGGCGGTATAGGAT<br>    ATGGTCGCTCCGCCGTGCAGCGGCTACGGGTC<br>    &gt;Gene3<br>    CTTTAAAAGCCGCACTATGCGATATCGAGGAACAGGAGTG<br>    CCTGCATCACAAACAGATCTCTACTCTTTGAAGTTAGCGC<br>    TTGTAGCACAAAATCTAGACAGAAGTAGGCAGCCGAAGAA<br>    AGCGTTCCCTTTTCCGACACGGGCGGATCGCTAGCAAGTG<br>    CTTAACAGAGAATGTGGGACGGGTTCTATCTGTCATAAGG<br>    GA<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA2.fasta<br><br>File contents:<br><br>    &gt;Gene4<br>    CTAGAAGCAACACTTGCTTCACCGTCGGAGTGGAATTCAA<br>    GTCCCGAGCTCGTTCCATTCGGTTAGCCGTGACACGTGAG<br>    CTCAGGTTTAATTTACCTCAGCTCACTCCGACCGTTAGCA<br>    ACGTGGACCAGTGCAGACCACCCAATCCTGAAGCGAATGT<br>    TACCGCGGACCCGATCAAGATACGCGAGGGAAAGTTTAAC<br>    CGTACTTACTCATCACAGCTGGGGCAAGGAGGTCTATGCA<br>    TGCCCCAGAGCCGAGGTCCTTGTGGCCCCTAAGATAAGCA<br>    GTTGCGCAGGACAGCGGTTAAGACCATGACAACACTCGAT<br>    GAATTGAGGAGGGTGCATCCACCACTGAGCTAGTTTGGTA<br>    TGGGTGATATAACGGTGGCG<br>    &gt;Gene5<br>    ACTCTGCTCACGACGCTAAGCCTCACGAGGACGCTCAGTA<br>    GAAGAGTTTTCTGCACACTGGTGATGCTCCAGTCCCTAAA<br>    CTTCAAGTAGGCTGGTCTGCTAGTTTGCCCTAGGGACCTG<br>    ACGCCCTAGGTTGGAGTACGCTGGCACGTAAGCTATCATC<br>    GCGCCGTGGACATTATCCTAATACGTACCGAACAGATGTT<br>    AAGATCGCTAGGAAGTGTACGAGGCCGCACAGTGGGACCC<br>    ATTAGCCCTGATCCAGTGGGACGTTAGTATACCATTCCAT<br>    CGTGTCTGGGATCGTAGATGGAACGGCTCCTCACGTTCAC<br>    CGTGCACAATTTCCCATTCATAGCGCAGTCGCCCTTGCCT<br>    ACGGTTAGAGACGAAATGACGATGAGGCTGGCCATGCATG<br>    CATCTCGGTCCAGGGACGGATTTTCATGCGACAATAGCGG<br>    TCGCACTGACAAGAAGTCCTGTATTAAATCGTGCGATCCA<br>    TCTGTCGCCAGAGCCACGGCATAGCCGTTCATTTAGATTT<br>    ATGTTACCACGCGGAACTGCGGAGGCCATCCAGCAATCGA<br>    GTTAA<br>    &gt;Gene6<br>    TTGGCTATCATAGAGTGGTGTATGACCCGCTTTGCATTTC<br>    GTGAATCTGGGCTGGAAAGGTAAGAACACCACCAAACGCG<br>    CTTGGACGCCTTTTAGTGACAGCATCGCGCATATAGTAGA<br>    GGTGCCACTCAGGTGCGTGCCGGCACTTGACCATTCCGTG<br>    GTACCTCCGTATTGGAAGCCAGGGGACGGGGGTAAATAAG<br>    CCGCGTTTATCCTGTTGAACATTTCAATCTTTGTGAGGGT<br>    ACTGCTGTATTCATATGGTTTCCCGTGTCGGCTTAGAAAA<br>    CCACGGCACTCGGAGGACCAAGTTAGACAAACTCTCGAGT<br>    AGGTTACACTGCGACCGAGTGTCACGCCTGCCTCGAGTCT<br>    TACTAATCACCCTTAGGTAATGAAGGTCAGGGCTTGCCGG<br>    ACAGTGTTCCCGAGGAGGTAACTCTCGTGCCAACTCCGCG<br>    AACTGATGGCCAGAGGAAACACTACTCACTACACTTAGTC<br>    TTACAAGAACACTTAGCAGAAGA</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def dna_to_peptide(dna):<br>    codon_to_amino = {<br>        &#39;CTA&#39;: &#39;L&#39;, &#39;CTC&#39;: &#39;L&#39;, &#39;CTG&#39;: &#39;L&#39;, &#39;CTT&#39;: &#39;L&#39;,<br>        &#39;AGA&#39;: &#39;R&#39;, &#39;AGG&#39;: &#39;R&#39;,<br>        &#39;AGC&#39;: &#39;S&#39;, &#39;AGT&#39;: &#39;S&#39;,<br>        &#39;ACA&#39;: &#39;T&#39;, &#39;ACC&#39;: &#39;T&#39;, &#39;ACG&#39;: &#39;T&#39;, &#39;ACT&#39;: &#39;T&#39;<br>    }<br>    peptide = &#39;&#39;<br>    for i in range(0, len(dna), 3):<br>        codon = dna[i:i+3]<br>        amino = codon_to_amino.get(codon, None)<br>        if amino is None:<br>            return None<br>        peptide += amino<br>    return peptide<br><br>def search_peptide_in_dna(peptide, dna):<br>    peptide_length = len(peptide) * 3<br>    for i in range(len(dna) - peptide_length + 1):<br>        candidate = dna_to_peptide(dna[i:i+peptide_length])<br>        if candidate == peptide:<br>            return True<br>    return False<br><br>def search_peptide(peptide, fasta_file, output_file):<br>    with open(fasta_file, &#39;r&#39;) as f:<br>        lines = f.readlines()<br>    genes = []<br>    dna = &#39;&#39;<br>    gene_name = None<br>    for line in lines:<br>        if line.startswith(&#39;&gt;&#39;):<br>            if gene_name is not None and search_peptide_in_dna(peptide, dna):<br>                genes.append(gene_name)<br>            gene_name = line[1:].strip()<br>            dna = &#39;&#39;<br>        else:<br>            dna += line.strip()<br>    if gene_name is not None and search_peptide_in_dna(peptide, dna):<br>        genes.append(gene_name)<br>    with open(output_file, &#39;w&#39;) as f:<br>        for gene in genes:<br>            f.write(gene + &#39;\n&#39;)<br><br>if __name__ == &#39;__main__&#39;:<br>    import sys<br>    peptide = sys.argv[1]<br>    fasta_file = sys.argv[2]<br>    output_file = sys.argv[3]<br>    search_peptide(peptide, fasta_file, output_file)</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">3</td><td class="s1" dir="ltr">The first solution did not find any matches. I tried to work with ChatGPT to fix it, but it didn&#39;t work out. I gave it the original prompt, and it provided a valid solution.</td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R181" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">182</div></th><td class="s1">19 - Additional Practice</td><td class="s1">07 - Make inducible promoter - Part A</td><td class="freezebar-cell"></td><td class="s1">Here is a new Python programming task.<br><br>When studying a specific gene, you may want to place it under control of an inducible promoter, which will allow you to turn the gene on/off when you want. One step in that process is to identify the translation initiation sequence of that gene. The translation initiation sequence has a sequence of &quot;ATG&quot; (except in extremely rare cases).<br><br>Please write a Python script that accepts one argument via &quot;sys.argv&quot;: the name of a FASTA file. (Below you can see the FASTA files that will be used for testing.)<br><br>Your script should read the specified FASTA file and identify the initiation sequence. To do this, your code should read the DNA sequence, three nucleotides at a time (starting from the beginning), and identify the first in-frame occurrence of &quot;ATG&quot;. Your script should print this location. As you count the locations within the sequence, use zero as the starting location.<br><br>Important: Ignore any &quot;ATG&quot; sequences that are not in frame.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA1.fasta<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA2.fasta<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA3.fasta<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA4.fasta<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA5.fasta<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA1.fasta<br><br>File contents:<br><br>    &gt; contig1<br>    GGTGTCAAACTTTCGGGCCTGCGCTTCAATTTCTCTCCGCCGCCCTTTCGATCATGAATC<br>    AAATGGCGTTCCCACACCAATATGGGAGAGCTCCCAGCTGCCGTAGACACCGATTGTATT<br>    AGGTACTGTGGGGCCTCTGGTAAATTCCACTAACGGAACTAGCATAGTAGGCTAGTATAG<br>    CTCGGGGCGCAGGTATTGTTCTCTTGATAATTGGTACCACAGCGAGTGCGGAAGTAGTCT<br>    ATACGTGATGACGTTAACTAAAGACGCTCTGCAGGAGAGCAACGGTGGTATAGCGGGAGC<br>    TACGAGAAGGATGCGGACAAATGGCGTTCCAGAATCAGTGTATTTGGCGTGTAACACAAG<br>    CACCTTTCTCCCAGGTGACGCGTCGCTCGCGAAGATTTCCAAGTAGCAAA<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA2.fasta<br><br>File contents:<br><br>    &gt; contig2<br>    CTGTCTCCTATATTATGTACGGACGAAGGAATCGAACCGCTGGCTGTACACGTGCTAAGC<br>    GAGCCTCAACGTATGTTTAACATAGTTGTATCGTTGCCCTGTCAGTACCTGCGTCTTACG<br>    AAGGGCGGTCAAGTGACCTTCCGGCCCAACTTTGCAAGAAGAAACGCAATACGTACCTAT<br>    CGCGGATCTATTATCAGCGCATCCAGTTCGGCTGAGCTTGTGAACTACGCTCTAAATACG<br>    CCTGAGACTACCACGAGAGGGTTCTACACTGAACTATAGTCTCCTTTTTAATCTTGAATT<br>    TCCCAAGATCCAGAACCGCATACATTGGTTTTGT<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA3.fasta<br><br>File contents:<br><br>    &gt; contig3<br>    GCCGGATAATTCCAAGGCTACTTCCGCGAATTTCCGACCCGCTGGGCCTGCTGAGTTGGA<br>    CACCTCGCATAGATTGTGACCCCTCTATGACAGATTGGGACACCTGCAGGCTCAGGGTTA<br>    TGAACCTATAAGTGCGATAAGAGCTGGAACTTTAGGAGATATCGTATGTCAATATTTCAT<br>    GCTGAAGCCCAAATGAAACGAGGTACTGACTACGTACTGATAACACGGCTCCACCTCTAC<br>    CTGTGGATTCGCCCTCAAACGGAACCGGTGTTTGACTGCAACTGCTGGTGCATGTCCAGC<br>    TTAGAGCTTAGACACAGCAGTCCATCGCGTCCAGAGTATACCGGATCGCACCTAGGCGAA<br>    CCGCTAATTGTAGATATCGAGATAAAAGCGCATGACACAGGATCAAGGTGTAGTTAGTGT<br>    GTCGGTTGTTACTTTACGTAGACGCAAGTCGCGGTATGTAAGGCGAAGTGTGGCTTTATT<br>    CAGCAGAAAAAATGCATCGCTAAAGCCGAGATTGCGCAAAGGGTGGCAACGGAGCTCAGC<br>    GTGTATTGCGTAGACCATGATAGCCCCCGCCCAGAATTGAGGGAGCTGCCGGCATGCCAG<br>    TACGCTCTATGTCTACAGGCCGTCCTCAACTCCTTATTCTGTACGCGCGGGCTACTTCGG<br>    CCGTGGAGATATTCGGTGGGACGGTTCTACGCGACAAAATCCTAGTTAGACGACAGCACA<br>    AAATCACATCTGGAACTACCTGCAGGCCTTCACCTATCGGCCCATCACTAAACAACCTCT<br>    AAGGACCATAAGGGTAAGTCACTGGAGCTCAGAGCGCTAGCGGATCGACTCCTGTGCAAA<br>    CCGTGTCTGACTGGCGAACATGCCTTGTTAAGCGCATAGGCGGTGTAGGGGTCAACTACT<br>    CTCGCTCTCGTCCTAAGGCCGCGCCTGGTGA<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA4.fasta<br><br>File contents:<br><br>    &gt; contig4<br>    CTCGTCTGGTGTAATACCAACGGCGGCCGCCCGAGCACGGTTCCGCCAAACAGTGGCTTT<br>    TTACAGTTTTAAAGAGACGACTTGGCTGACCTATGGCCTCGGACCTTAGCGTGCTTCGTA<br>    CTCAGTAGCTTGAAGTCTCTACTTTCTTTTCCGAGTATGACTTGTTTAATGGTTCCCACC<br>    CCCCAGAGTTTTCTCCCGGTCTACACTACCGGGGGGCGCAAGGTGGCACGATGGGAGACC<br>    TCCGTATTTTGCCGTTGCGGATCAAAGGTTCAGCATAGACTATCTAAGGGGCCCGTCAGC<br>    AACTGTTCAGTTGCCCTTACGCAATTACTGCCAATTTATTAACTGTAAGTGGCTTAAGGT<br>    AACCCGGACCTCGCCACAATCCGATATCGCTACGTCGAACCTCAGTTGCGCTAGGCTAAG<br>    GGCGGGGCCTGTATTACCTATGCCCAAGGCGGGTCGGGGCTATTTTACACGGTGCTTGAA<br>    CGCCCGATATGCATACGGGTTATTAAGAAACACTTATTCGTAAGTTAGTAAGCGAGGACA<br>    GCTGTCAGCCCGTCATCTCCATTGATCAAGCAGGATCAGATGGTAAGCGCGTCCTGATTT<br>    GCTTACCTATCAGTAGTGTGGTAGCTTCCGGGTAAGGTCACGGTTAGACGAGGACCATTC<br>    ATTGGACCTTCTGTCGCGAAGTGTACGATAACCCGGGGTGCTCCGCGTGCGGTGACGTTT<br>    TCTGTTCTGCCAATAAAAAGAGCAAAAAACTCAGACTATTCGTTTTCTAGCGCGGCAGCT<br>    CGAAAAATGTGTGCTGCATGAGGGCACGCTCGGCGAGGTCACACCTATTTTCTACCGAGA<br>    TGCTTCCGCTCAGGTCCAAATCCCAACATTATTCGATCGTAGACAGTTAGACCACAAG<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA5.fasta<br><br>File contents:<br><br>    &gt; contig5<br>    CGGGAGGGGAATGGTTGGGCGCCAACAAGTGTGGCCACCACATTTCATCTAAAGATCTGC<br>    ACTGTGATTCGGTTAGGTTTGGCGGAGCCCGCTTATCATGTGGAAGGTTACTGCATCCTA<br>    GTTGAGAGGAGAGATGATACCGCGATCCCTTAATACGTTCCAACAATGCGCATCCCGTTG<br>    GCGCCAATTAATTCCCGGGAGCGTAAGTCCGGAATAGGACCGCCCAACGTACGCTGCGGG<br>    CTATTCTCGAGTTCGAAACAGTCGCTTGGCGCGGTGCTTAACGAGCTTGAACCACTGGTC<br>    CTCTTGCAAGAAGTTGCACCGGGATGCGATACAGCCAAAACTGAGAGCTAAGTGGAGCAG<br>    GCTACTTTCCCGTAACTGTACTCAATCGACAACAACACCTTAGGTCAACCGAAACGGAAC<br>    ACGCGGGACCAGCGAGAAAATTCACACCAAGAAAACCGCCTGGTATCTAGACTTGGGGAA<br>    TTCCCAACTTGCTTTGCACGGGATATCTGGGGGCGGGTAATTACAGATGAATAATGATTA<br>    ACGGACTACATAGAGTTTTCTGCGTTGGCCAATTCCTTTAGCCCTCTTGGCTTGGAGCAC<br>    GTCGGTTCATCTCGAAAACACTACTCCAAATCAGACGGGGGTGAGCAGCGCAAGGTCTAA<br>    TGGCAATTAGGGTCAAATGACGGTCCGTACCTCACTTGTTAGGACGGAAAGCCCTAGAAC<br>    ATGAAGGTTACCCGACGACTGCTTGTGCGATATGATAGTTAGACGCATTGGCCTTCGGAG<br>    GACTAATCAAGAGGGCCTTAGGATAGGAATGCTTATCAGTAAACTGCGTTATGGATTGGG<br>    CACCCTCGCATGAGTCTTGACACAGAATCCGCAAGACTCAAAGACTGAAACGACTTGACG<br>    CAACAACCACCGAACC<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA6.fasta<br><br>File contents:<br><br>    &gt; contig6<br>    TCGCAATGAGGGGAAAAATGGCGTTCTATCACTAACAAAGTCTGGATGTACACTTCAGCT<br>    GCCCATAAACGATTAGGTTTAGAACTGATCGACATCTGGGGCGGGAGTCAGGATTAATTG<br>    ACCGTTACTCTAACTCACAACGCTTTCTTCTGGGAGTCTATTCTTCCGCCTTCTAATGGC<br>    TACATGTCCCCAGACGGCTACTATCACGAGCGGTAGCAAATAGAATGTCGGTGATACTAA<br>    TATGTGTCTATGCGGAAGCAATTGTTGCTCGTGAAACTACTATGCGTTCAAATACAACAC<br>    TCAAGCGCGATGAGGTAACGTTTCTTAAAGCATGTACGCTTCCATAAAAGAACCCACCGG<br>    TTGAGGGGTTCTACACTTGCTTCCTTCCTAAATCGGCCAGAGACAGTTTTCTATCTGCTT<br>    ACGCCTTTGGATAAGCGACCCCCGCAGTAAGCTTTCTCATAGGTAAACCCCAGATGGTGA<br>    AGACTCTCGTAAGTCCACCATCCGGTTTGGTAAGGACATTCAGTGTGGTTTCCCAGAGTT<br>    CTATTACGTAGGAACTGCGGCAACTCCCGTGACAACAAGCCTCAAAATAAGACCCCAGAA<br>    GCCGACGCCCCGGTCAAACGGGCAGGCGAGAAAGTTGTTATCGTGCTTACAAACTAGCCA<br>    TGTATAGCCTAATGAAGTGCACAGTCCGCGGCTAAAGCGCCTAATACTCACGCGGAATGC<br>    TGTCGCAACAGCTGCTTAAAACCATGTTAGATAAAAGAAAGCAACTGCCATCAAGCCGAG<br>    AACCTGCCCTGCGATTGTTCGCGTCAGATGGCGTCGCGGGCAGTATAGCCTTCTCCTAGC<br>    CAGTTAATGGTTTCGTCCGCGTGCTCAGGTCACAGGCTGGTCCTGCACGAGGGTATACAG<br>    TGTCCGAAAGGTTCTATGCCGATCACGTATGGTCCATCGACTTAAGCGGTTATTTCACCC<br>    TCCAGGATAGTGATGAGTCGTTAAGCCGGAAGGGAGTGGAGGCTTCCCCCAAGAAGGGGC<br>    TACCACCCTCAGCAGTAAATAGATAGGGCTCCTCGCAAAGTGCAAGACGGCCTCCGCATG<br>    GCTGGTACTTTGCACCGTTCATCTCGTGCTCAAGTCACGGTACGGCCATCAGCCTAAGAT<br>    TAGGAGTAGCTCCGTGAAATGCTTTAACAATGACGCAAACCTCGCAAGCCTGGGATAATT<br>    GGAAGGTGATCCTAGTAGGAGACAAGAGATTACGGAGGAATTTACTCCAAACAGTAACAC<br>    TAAAGTGTAACCAAAGTCCCCTTGACCGGAAAACCTGTGTGCTTTAGAATGGTTATATAT<br>    CTTCAAAATGAGCCTCGGCAACTCTGGTGGCCAGCACGTCTGAGAAACGCCCAAGCGCAG<br>    CGATTAGAGGCATTCCGGGCGACTGTCAACTTGACGGGCATGCGCCATGGTCAGAATAGG<br>    CCGTCCCTCCGCAAATAGTATACAGCTCTTAAAATTACGGCCGTACCGGACAGTTGTTCT<br>    CATGATTTCATGGGCAGCGCTGAGTGCGTGGCTTCCCCGTTAGAAGGGTGTGCTGGTGCC<br>    CGGTTATCATTTTGAGAGTGCTCGAGATGGGGGGATTGTAATTGTTGTCTCGGAGTGTTT<br>    CTCGCCCAAGCCCAAATGGCCCGATTCGATAATGCTATTAGTCTGACATTTTGGATTTCC<br>    ATTCTGAGCCGATCGTACTGGAAAGTTCTAGATGACTCTAATGGAAAAGGCTGCTCCATG<br>    GCCAGGTCTTCTAGTCGGTAACTTAAGAGACCATGCAGAGCTTGCCCGCGGGGCTATATA<br>    GGAGATTAATATGTGTTCTGACGGTGATCTATTACCGACCTCCCTGCGTAACCTTGTTTC<br>    CTACAAAGCTTAACCCGAAACTTGATCGGACCACGCTACAGTCTGTCAGGAGATCTGAGC<br>    CCGGTGCACCTTTATCGCCGGGAACAACCGCTTACGAAGTACCTTTATCGACCAACTTTG<br>    ACCGCTGCATGATAATCCGGGAGAGGCATGCGCTGTTAATTGCGTAGTCGCCTATCTCAT<br>    GGGGAGTGCCGGGCAAAAGGCCTGCTGTGTAAAGTCGATAGTCGTATGGTCCGGAGGTCC<br>    TAGAGCTCAGCGTCCGAGTTCACCTTTGGCACGTAACTTTAACGCATGTAGAGTTTGCTC<br>    CATTGCCCCCCTAGTGCATCCTTTCGAGTAACCTCTCTAGCGTGCACGAGCTTACACAAC<br>    GATGGAACCTGGTGGATAATCAATGGGGTTAGCTTACTCGTGCTAGGGCCATCAACTTAG<br>    ATTTCAATGCCTAACAGGGCCGAGACCATGACCCTTCGAAATGGCGTTCTTGTTTTCTAC<br>    CGGCAAGTTGCGATGGCGAATGGTAACGTAAGTAACCGAATTCAGCCACATGATGGATTA<br>    TTCAGACAATCCCTCATCCGCGTACATAATTATGCACTGGGGTGAATCGAGTTGCGTAGC<br>    CAAGATTGTGTATCTGTTACAACCAATCCGCGGGCAGCTTTCTGATCTAGTTCTTTGTTG<br>    TGAACGCTGTCCGCTCACAACTTATCCCTGGTATGTCCCTTTGTGCCTCGTGATGAAATA<br>    CTGAACTGCGAACCCCAGCAAAAGGGGCGGCGCCCGGCATGCGACGGTCCTTACTGCGCT<br>    CGGGGCAAGGGTGGAACGGCCAAGTTAGTGAAGGCCATTCGCGTAGGTGTCTCCTGTGCG<br>    GCGCCATGCTCCCGCTCCACGTCAGGTCTTGCGACTACAATTCTCTTGGGGGCTTTAGAT<br>    CCGGCTTGTGGACTACCGACAGTGCGTCAGATCTCCGAGCACGACAAGTTTATCTGTTTT<br>    GGGGGGATAGCACCTTGTTGCAAGCTCCACCATGATTGGACCAGTCGTATCGGCGTTGTT<br>    TAGCACAGTTTTCGCAGGTAGGTGAAGGACCCTCAAGACAGTAATCCGGCCCGGACGCCG<br>    GCCATTTTCCCCTCGAAGGTACCGTTGCTAGCGTTGTCCTCGTACGGATAGGGTCGGTTG<br>    ACCGTTTCCCCAGATCCCCCCACATTGCATCGCCGCAATGGGTCGGTGCACGTCGAGGCA<br>    GGCGGCTAGCACCGGGTGTTTCAAATCGCAAATGTAAAGTCAGTAGCACGATGATCGAGT<br>    CCCATCCCGACCCCGTGTTGGTTGGAAAATACCACTATGAATTCACACGTTTTTTTGTTT<br>    CGCCATTTAGTGAAAAACAATTGGGAGTCGAAGAAAGGTATAGGATTGCGTTAGCGGAAA<br>    CGGCGACGTAACTTCGGATCGCTATTATGGGAATGTGGGTGAAATAAAGAGTGTGCAGCA<br>    ACACATGTTTAAAAAAGCAGGATCGATAGAGCGAGCGTGATTTAGGAGATACTAGATTGG<br>    AACCTCCCCCACAGGTCACCACGTCCTCTACCTAAGATTTATATCTCCGCATGGCCCTGA<br>    TCATGGAAG<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA7.fasta<br><br>File contents:<br><br>    &gt; contig7<br>    ACAACCGAGGTCTCGGTGGTCGCGGGACCCGGAGCATGGATACGAACGCGTCTGAGCGTG<br>    ACATCCACGCAGCGCCCGGGAACCTGGGTTCCCGACTATCACATGGGTAATCAAATCGCG<br>    AACATTCTTAAAGTAAGGCCGATGCTGAAGCACTATCCTAACCACGCTAGTGTGCAATGC<br>    GCGGGAAGGCGGAGAAAAAGAATCTTACGTTTATCAGCGTTTTCTTACTCCGGCGACCAA<br>    GCGAGGAGGGGCTGGTTAGCCCCCAGTCTTTAGTTACCTGGCGTCGGATATATTGTCATC<br>    CTAGAGTAACTGTCCTATCGTTGAATAACCCGTACAAGAGATACTGGGTCCTGGTAACCG<br>    TACAAGTTTTCCGACTTTGCGAGACTCTAAATCAGTACTTTGGCAAGAGGGAAGGGCAGT<br>    AAGGGGTCAGCTTTTGAACAGCGCCCGGGCCCAGGTTACGGCCTAAATTAGTGCCAGTGT<br>    CACCGTCGAGAGGGCAGTTCGACTGAAAGCCTAATAAGATACCGTTGAC<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA8.fasta<br><br>File contents:<br><br>    &gt; contig8<br>    TGGGACAGGACCAATCAATTCGCCAAGAGTCTACTTGTCCGTCCGATAGGCAAAAGTTGT<br>    TCGCTGAAATATCTCGGCCATTCAGACGAGTTTTACATAAGTACGTATGTAGAACGCGAT<br>    GATCCGGATTGGAGTGACGCCTGTCTCTTTTCGAAAGACGGAACCGCCGGCGCACAATAA<br>    GCCTGCTACCTGACTGTAGCCTCTGTATCCTCAATATCGGATACAGTCAGGCAGATTGAG<br>    ATCGCGACATATGCAAAGAGTGGGTGACTAAGGACAATGGGAGGAACTGTGGTTCTCACT<br>    TCAAAGCTTGTGTCAGGCCTGCTCAAAAGGTCAAGGAAAATTTCTACCGTTCTCACGATG<br>    ACGAGGGTGGCACTGGCACGGACCACCAATCGCTCTGAACTAGCTTGCAACCCGTCGTAC<br>    CTCTCGGTGACTCTGGGAGCAGCATAGGAGGCCTCGCTTGTGCAATCCTTAAATCGCGAG<br>    AGACATCCGGCGAAGCATGCACTAAGCCTTTCTGAAATGTGATCGGATTTGACTTTAATC<br>    CGGTCACTCTGATCTCATTTTCTAAGACTAACGATTATCGGTTCAAGTTCGTTCCTAACG<br>    AATTTTGCGGGTCGTGCTACTACGTACTAGCTTAAGGACACTACACCTACACCGTCGGTT<br>    TTTAATAGCGACTTATTATGTCCTTGCTAAAGCTCTAATCCCAGTTGTAACAGTGCATAT<br>    TTAGACAGCACATCCGTGAGTAACCGTATGTGAGCAGAATTTAGGACTGAGTCAAGGAAT<br>    GCAAATGGCAAACGGGGTGGTTCCTCGGTGGATCACAAGCGCGACCATTGAAGTTCAGTT<br>    TGTGGCACGCCCCGCGCGCTTCGTAAGGAACGATGGACGGCAGAAGTGTTGAATTTATAG<br>    CACGAGCGAGGTGTGCTGATTCCTCCGTGATTAGTGCACGGAAAATTCGAAGAGTACACT<br>    CTATTTATTTTGTGTCGCTAGATATGCAGATGCTACCGGAGTTCCGACGTTCAACGACTC<br>    CCCCAGGCGCGGAACCCGAACCCGGGACACTTACGGAAGCGTGAGATCCCGGTTGTCTTT<br>    TCAGGTGTCCAATTATGGTTTTTTGGAGGATCGTTATATATCTTGGCCGCACCTCCACTG<br>    GCCTTCGCATACAGTCTGGGACCAGCGTCGCACCGATAGCCGGCCAGGCCCGCTCTAGAT<br>    AACGGCAGATGTGAACCGCTGTAGTTACACTGTCAGTGTCACCAGGGACTAAGCAAATGG<br>    CAATTCCCAATAGGGAGGCACATTGTGAGTTTGACCGGCGTATTGTAAGCTAGCTATCTG<br>    ACCGACAGGCTTAACAACGCAATATGGATACTCTGTTCACGGTCTTTTGGAGCCAATGCC<br>    TCCATTCCCCTGGACCATGTCAGAATACAAGACGTCAGAGACCGAAAAATTGGACCTTAA<br>    ATTATTCCATGGGTCCGCTCTGGGTTTGACGCGACATGTCCAGTTCGCGCGGCGTTCTCT<br>    TTCGCGTCGAGGAGCTCTGCTTAGAATGGACGTAGCCAGAAAAAATTGAGTGGGGCCAGT<br>    ATTAGATTGTTACCGTGGTTCAAGTGAATATCGCGAGTGT<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA9.fasta<br><br>File contents:<br><br>    &gt; contig9<br>    GAGTGTTAAGAGTGCGCAGCATAAAAAAATCGCCGATGCACAAACCAGACTTTTGCGCGG<br>    AGAGCGGCAATGCCACGAGACACAACGGGGCGAGTAGCCTATCGAGCCCCGGCGCCAAGA<br>    AGCTATTCGTTTCGCATGTGATTAAGTTTCTACACTACATTGCGTAGAGGTCTAGTTATT<br>    TGTTAGCCAACGCTAGACTAGCATACCATGGTGTAATTATGGTAGACACCAAAGCGCGAA<br>    CGGAGAGCGTCCAATGTGATCACGAGCCCGGGAAGCAAAACCCGTAGATACAGATTCCGT<br>    ATTGCAAAGGGGGCACTGAATGCCCCCAATCAATCGTAAACGATTCAAGATAGTCCGAGT<br>    CTAATCGGTTGTCGTTGGTATCCCGAA</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import sys<br><br>def find_ATG(seq):<br>    for i in range(0, len(seq), 3):<br>        if seq[i:i+3] == &quot;ATG&quot;:<br>            return i<br>    return -1<br><br>filename = sys.argv[1]<br>with open(filename, &quot;r&quot;) as f:<br>    lines = f.readlines()<br>    seq = &quot;&quot;.join(lines[1:]).replace(&quot;\n&quot;, &quot;&quot;)<br>    print(find_ATG(seq))</td><td class="s2">5</td><td class="s2" dir="ltr">5</td><td class="s2" dir="ltr">2</td><td class="s1" dir="ltr">The first solution added some text to the output that was unnecessary. I asked it to remove this, which it did. The new solution was considerably shorter than the first one.</td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3" dir="ltr">Yes</td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td><td class="s3"></td></tr><tr style="height: 20px"><th id="2089702959R182" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">183</div></th><td class="s4">19 - Additional Practice</td><td class="s4">08 - Make inducible promoter - Part B</td><td class="freezebar-cell"></td><td class="s4" dir="ltr">Here is a new Python programming task.<br><br>Now that you can identify translation initiation sequence, the next step is to identify the location of a stop codon within the gene. A stop codon (&quot;TAG&quot;, &quot;TAA&quot;, or &quot;TGA&quot;) indicates the location where translation should stop.<br><br>Please write a Python script that accepts one argument via &quot;sys.argv&quot;: the name of a FASTA file.<br><br>Your script should read the FASTA file, three nucleotides at a time, and identify the first in-frame occurrence of start codon. Then starting from that location, your code should read the DNA sequence, three nucleotides at a time, and identify the first in-frame occurrence of a stop codon. Your script should print the start location first and then the stop location (on a separate line).<br><br>Important: Ignore any stop sequences that are not in frame.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA1.fasta<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA2.fasta<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA3.fasta<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA4.fasta<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA5.fasta<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA1.fasta<br><br>File contents:<br><br>    &gt; contig1<br>    GGTGTCAAACTTTCGGGCCTGCGCTTCAATTTCTCTCCGCCGCCCTTTCGATCATGAATC<br>    AAATGGCGTTCCCACACCAATATGGGAGAGCTCCCAGCTGCCGTAGACACCGATTGTATT<br>    AGGTACTGTGGGGCCTCTGGTAAATTCCACTAACGGAACTAGCATAGTAGGCTAGTATAG<br>    CTCGGGGCGCAGGTATTGTTCTCTTGATAATTGGTACCACAGCGAGTGCGGAAGTAGTCT<br>    ATACGTGATGACGTTAACTAAAGACGCTCTGCAGGAGAGCAACGGTGGTATAGCGGGAGC<br>    TACGAGAAGGATGCGGACAAATGGCGTTCCAGAATCAGTGTATTTGGCGTGTAACACAAG<br>    CACCTTTCTCCCAGGTGACGCGTCGCTCGCGAAGATTTCCAAGTAGCAAA<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA2.fasta<br><br>File contents:<br><br>    &gt; contig2<br>    CTGTCTCCTATATTATGTACGGACGAAGGAATCGAACCGCTGGCTGTACACGTGCTAAGC<br>    GAGCCTCAACGTATGTTTAACATAGTTGTATCGTTGCCCTGTCAGTACCTGCGTCTTACG<br>    AAGGGCGGTCAAGTGACCTTCCGGCCCAACTTTGCAAGAAGAAACGCAATACGTACCTAT<br>    CGCGGATCTATTATCAGCGCATCCAGTTCGGCTGAGCTTGTGAACTACGCTCTAAATACG<br>    CCTGAGACTACCACGAGAGGGTTCTACACTGAACTATAGTCTCCTTTTTAATCTTGAATT<br>    TCCCAAGATCCAGAACCGCATACATTGGTTTTGT<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA3.fasta<br><br>File contents:<br><br>    &gt; contig3<br>    GCCGGATAATTCCAAGGCTACTTCCGCGAATTTCCGACCCGCTGGGCCTGCTGAGTTGGA<br>    CACCTCGCATAGATTGTGACCCCTCTATGACAGATTGGGACACCTGCAGGCTCAGGGTTA<br>    TGAACCTATAAGTGCGATAAGAGCTGGAACTTTAGGAGATATCGTATGTCAATATTTCAT<br>    GCTGAAGCCCAAATGAAACGAGGTACTGACTACGTACTGATAACACGGCTCCACCTCTAC<br>    CTGTGGATTCGCCCTCAAACGGAACCGGTGTTTGACTGCAACTGCTGGTGCATGTCCAGC<br>    TTAGAGCTTAGACACAGCAGTCCATCGCGTCCAGAGTATACCGGATCGCACCTAGGCGAA<br>    CCGCTAATTGTAGATATCGAGATAAAAGCGCATGACACAGGATCAAGGTGTAGTTAGTGT<br>    GTCGGTTGTTACTTTACGTAGACGCAAGTCGCGGTATGTAAGGCGAAGTGTGGCTTTATT<br>    CAGCAGAAAAAATGCATCGCTAAAGCCGAGATTGCGCAAAGGGTGGCAACGGAGCTCAGC<br>    GTGTATTGCGTAGACCATGATAGCCCCCGCCCAGAATTGAGGGAGCTGCCGGCATGCCAG<br>    TACGCTCTATGTCTACAGGCCGTCCTCAACTCCTTATTCTGTACGCGCGGGCTACTTCGG<br>    CCGTGGAGATATTCGGTGGGACGGTTCTACGCGACAAAATCCTAGTTAGACGACAGCACA<br>    AAATCACATCTGGAACTACCTGCAGGCCTTCACCTATCGGCCCATCACTAAACAACCTCT<br>    AAGGACCATAAGGGTAAGTCACTGGAGCTCAGAGCGCTAGCGGATCGACTCCTGTGCAAA<br>    CCGTGTCTGACTGGCGAACATGCCTTGTTAAGCGCATAGGCGGTGTAGGGGTCAACTACT<br>    CTCGCTCTCGTCCTAAGGCCGCGCCTGGTGA<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA4.fasta<br><br>File contents:<br><br>    &gt; contig4<br>    CTCGTCTGGTGTAATACCAACGGCGGCCGCCCGAGCACGGTTCCGCCAAACAGTGGCTTT<br>    TTACAGTTTTAAAGAGACGACTTGGCTGACCTATGGCCTCGGACCTTAGCGTGCTTCGTA<br>    CTCAGTAGCTTGAAGTCTCTACTTTCTTTTCCGAGTATGACTTGTTTAATGGTTCCCACC<br>    CCCCAGAGTTTTCTCCCGGTCTACACTACCGGGGGGCGCAAGGTGGCACGATGGGAGACC<br>    TCCGTATTTTGCCGTTGCGGATCAAAGGTTCAGCATAGACTATCTAAGGGGCCCGTCAGC<br>    AACTGTTCAGTTGCCCTTACGCAATTACTGCCAATTTATTAACTGTAAGTGGCTTAAGGT<br>    AACCCGGACCTCGCCACAATCCGATATCGCTACGTCGAACCTCAGTTGCGCTAGGCTAAG<br>    GGCGGGGCCTGTATTACCTATGCCCAAGGCGGGTCGGGGCTATTTTACACGGTGCTTGAA<br>    CGCCCGATATGCATACGGGTTATTAAGAAACACTTATTCGTAAGTTAGTAAGCGAGGACA<br>    GCTGTCAGCCCGTCATCTCCATTGATCAAGCAGGATCAGATGGTAAGCGCGTCCTGATTT<br>    GCTTACCTATCAGTAGTGTGGTAGCTTCCGGGTAAGGTCACGGTTAGACGAGGACCATTC<br>    ATTGGACCTTCTGTCGCGAAGTGTACGATAACCCGGGGTGCTCCGCGTGCGGTGACGTTT<br>    TCTGTTCTGCCAATAAAAAGAGCAAAAAACTCAGACTATTCGTTTTCTAGCGCGGCAGCT<br>    CGAAAAATGTGTGCTGCATGAGGGCACGCTCGGCGAGGTCACACCTATTTTCTACCGAGA<br>    TGCTTCCGCTCAGGTCCAAATCCCAACATTATTCGATCGTAGACAGTTAGACCACAAG<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA5.fasta<br><br>File contents:<br><br>    &gt; contig5<br>    CGGGAGGGGAATGGTTGGGCGCCAACAAGTGTGGCCACCACATTTCATCTAAAGATCTGC<br>    ACTGTGATTCGGTTAGGTTTGGCGGAGCCCGCTTATCATGTGGAAGGTTACTGCATCCTA<br>    GTTGAGAGGAGAGATGATACCGCGATCCCTTAATACGTTCCAACAATGCGCATCCCGTTG<br>    GCGCCAATTAATTCCCGGGAGCGTAAGTCCGGAATAGGACCGCCCAACGTACGCTGCGGG<br>    CTATTCTCGAGTTCGAAACAGTCGCTTGGCGCGGTGCTTAACGAGCTTGAACCACTGGTC<br>    CTCTTGCAAGAAGTTGCACCGGGATGCGATACAGCCAAAACTGAGAGCTAAGTGGAGCAG<br>    GCTACTTTCCCGTAACTGTACTCAATCGACAACAACACCTTAGGTCAACCGAAACGGAAC<br>    ACGCGGGACCAGCGAGAAAATTCACACCAAGAAAACCGCCTGGTATCTAGACTTGGGGAA<br>    TTCCCAACTTGCTTTGCACGGGATATCTGGGGGCGGGTAATTACAGATGAATAATGATTA<br>    ACGGACTACATAGAGTTTTCTGCGTTGGCCAATTCCTTTAGCCCTCTTGGCTTGGAGCAC<br>    GTCGGTTCATCTCGAAAACACTACTCCAAATCAGACGGGGGTGAGCAGCGCAAGGTCTAA<br>    TGGCAATTAGGGTCAAATGACGGTCCGTACCTCACTTGTTAGGACGGAAAGCCCTAGAAC<br>    ATGAAGGTTACCCGACGACTGCTTGTGCGATATGATAGTTAGACGCATTGGCCTTCGGAG<br>    GACTAATCAAGAGGGCCTTAGGATAGGAATGCTTATCAGTAAACTGCGTTATGGATTGGG<br>    CACCCTCGCATGAGTCTTGACACAGAATCCGCAAGACTCAAAGACTGAAACGACTTGACG<br>    CAACAACCACCGAACC<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA6.fasta<br><br>File contents:<br><br>    &gt; contig6<br>    TCGCAATGAGGGGAAAAATGGCGTTCTATCACTAACAAAGTCTGGATGTACACTTCAGCT<br>    GCCCATAAACGATTAGGTTTAGAACTGATCGACATCTGGGGCGGGAGTCAGGATTAATTG<br>    ACCGTTACTCTAACTCACAACGCTTTCTTCTGGGAGTCTATTCTTCCGCCTTCTAATGGC<br>    TACATGTCCCCAGACGGCTACTATCACGAGCGGTAGCAAATAGAATGTCGGTGATACTAA<br>    TATGTGTCTATGCGGAAGCAATTGTTGCTCGTGAAACTACTATGCGTTCAAATACAACAC<br>    TCAAGCGCGATGAGGTAACGTTTCTTAAAGCATGTACGCTTCCATAAAAGAACCCACCGG<br>    TTGAGGGGTTCTACACTTGCTTCCTTCCTAAATCGGCCAGAGACAGTTTTCTATCTGCTT<br>    ACGCCTTTGGATAAGCGACCCCCGCAGTAAGCTTTCTCATAGGTAAACCCCAGATGGTGA<br>    AGACTCTCGTAAGTCCACCATCCGGTTTGGTAAGGACATTCAGTGTGGTTTCCCAGAGTT<br>    CTATTACGTAGGAACTGCGGCAACTCCCGTGACAACAAGCCTCAAAATAAGACCCCAGAA<br>    GCCGACGCCCCGGTCAAACGGGCAGGCGAGAAAGTTGTTATCGTGCTTACAAACTAGCCA<br>    TGTATAGCCTAATGAAGTGCACAGTCCGCGGCTAAAGCGCCTAATACTCACGCGGAATGC<br>    TGTCGCAACAGCTGCTTAAAACCATGTTAGATAAAAGAAAGCAACTGCCATCAAGCCGAG<br>    AACCTGCCCTGCGATTGTTCGCGTCAGATGGCGTCGCGGGCAGTATAGCCTTCTCCTAGC<br>    CAGTTAATGGTTTCGTCCGCGTGCTCAGGTCACAGGCTGGTCCTGCACGAGGGTATACAG<br>    TGTCCGAAAGGTTCTATGCCGATCACGTATGGTCCATCGACTTAAGCGGTTATTTCACCC<br>    TCCAGGATAGTGATGAGTCGTTAAGCCGGAAGGGAGTGGAGGCTTCCCCCAAGAAGGGGC<br>    TACCACCCTCAGCAGTAAATAGATAGGGCTCCTCGCAAAGTGCAAGACGGCCTCCGCATG<br>    GCTGGTACTTTGCACCGTTCATCTCGTGCTCAAGTCACGGTACGGCCATCAGCCTAAGAT<br>    TAGGAGTAGCTCCGTGAAATGCTTTAACAATGACGCAAACCTCGCAAGCCTGGGATAATT<br>    GGAAGGTGATCCTAGTAGGAGACAAGAGATTACGGAGGAATTTACTCCAAACAGTAACAC<br>    TAAAGTGTAACCAAAGTCCCCTTGACCGGAAAACCTGTGTGCTTTAGAATGGTTATATAT<br>    CTTCAAAATGAGCCTCGGCAACTCTGGTGGCCAGCACGTCTGAGAAACGCCCAAGCGCAG<br>    CGATTAGAGGCATTCCGGGCGACTGTCAACTTGACGGGCATGCGCCATGGTCAGAATAGG<br>    CCGTCCCTCCGCAAATAGTATACAGCTCTTAAAATTACGGCCGTACCGGACAGTTGTTCT<br>    CATGATTTCATGGGCAGCGCTGAGTGCGTGGCTTCCCCGTTAGAAGGGTGTGCTGGTGCC<br>    CGGTTATCATTTTGAGAGTGCTCGAGATGGGGGGATTGTAATTGTTGTCTCGGAGTGTTT<br>    CTCGCCCAAGCCCAAATGGCCCGATTCGATAATGCTATTAGTCTGACATTTTGGATTTCC<br>    ATTCTGAGCCGATCGTACTGGAAAGTTCTAGATGACTCTAATGGAAAAGGCTGCTCCATG<br>    GCCAGGTCTTCTAGTCGGTAACTTAAGAGACCATGCAGAGCTTGCCCGCGGGGCTATATA<br>    GGAGATTAATATGTGTTCTGACGGTGATCTATTACCGACCTCCCTGCGTAACCTTGTTTC<br>    CTACAAAGCTTAACCCGAAACTTGATCGGACCACGCTACAGTCTGTCAGGAGATCTGAGC<br>    CCGGTGCACCTTTATCGCCGGGAACAACCGCTTACGAAGTACCTTTATCGACCAACTTTG<br>    ACCGCTGCATGATAATCCGGGAGAGGCATGCGCTGTTAATTGCGTAGTCGCCTATCTCAT<br>    GGGGAGTGCCGGGCAAAAGGCCTGCTGTGTAAAGTCGATAGTCGTATGGTCCGGAGGTCC<br>    TAGAGCTCAGCGTCCGAGTTCACCTTTGGCACGTAACTTTAACGCATGTAGAGTTTGCTC<br>    CATTGCCCCCCTAGTGCATCCTTTCGAGTAACCTCTCTAGCGTGCACGAGCTTACACAAC<br>    GATGGAACCTGGTGGATAATCAATGGGGTTAGCTTACTCGTGCTAGGGCCATCAACTTAG<br>    ATTTCAATGCCTAACAGGGCCGAGACCATGACCCTTCGAAATGGCGTTCTTGTTTTCTAC<br>    CGGCAAGTTGCGATGGCGAATGGTAACGTAAGTAACCGAATTCAGCCACATGATGGATTA<br>    TTCAGACAATCCCTCATCCGCGTACATAATTATGCACTGGGGTGAATCGAGTTGCGTAGC<br>    CAAGATTGTGTATCTGTTACAACCAATCCGCGGGCAGCTTTCTGATCTAGTTCTTTGTTG<br>    TGAACGCTGTCCGCTCACAACTTATCCCTGGTATGTCCCTTTGTGCCTCGTGATGAAATA<br>    CTGAACTGCGAACCCCAGCAAAAGGGGCGGCGCCCGGCATGCGACGGTCCTTACTGCGCT<br>    CGGGGCAAGGGTGGAACGGCCAAGTTAGTGAAGGCCATTCGCGTAGGTGTCTCCTGTGCG<br>    GCGCCATGCTCCCGCTCCACGTCAGGTCTTGCGACTACAATTCTCTTGGGGGCTTTAGAT<br>    CCGGCTTGTGGACTACCGACAGTGCGTCAGATCTCCGAGCACGACAAGTTTATCTGTTTT<br>    GGGGGGATAGCACCTTGTTGCAAGCTCCACCATGATTGGACCAGTCGTATCGGCGTTGTT<br>    TAGCACAGTTTTCGCAGGTAGGTGAAGGACCCTCAAGACAGTAATCCGGCCCGGACGCCG<br>    GCCATTTTCCCCTCGAAGGTACCGTTGCTAGCGTTGTCCTCGTACGGATAGGGTCGGTTG<br>    ACCGTTTCCCCAGATCCCCCCACATTGCATCGCCGCAATGGGTCGGTGCACGTCGAGGCA<br>    GGCGGCTAGCACCGGGTGTTTCAAATCGCAAATGTAAAGTCAGTAGCACGATGATCGAGT<br>    CCCATCCCGACCCCGTGTTGGTTGGAAAATACCACTATGAATTCACACGTTTTTTTGTTT<br>    CGCCATTTAGTGAAAAACAATTGGGAGTCGAAGAAAGGTATAGGATTGCGTTAGCGGAAA<br>    CGGCGACGTAACTTCGGATCGCTATTATGGGAATGTGGGTGAAATAAAGAGTGTGCAGCA<br>    ACACATGTTTAAAAAAGCAGGATCGATAGAGCGAGCGTGATTTAGGAGATACTAGATTGG<br>    AACCTCCCCCACAGGTCACCACGTCCTCTACCTAAGATTTATATCTCCGCATGGCCCTGA<br>    TCATGGAAG<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA7.fasta<br><br>File contents:<br><br>    &gt; contig7<br>    ACAACCGAGGTCTCGGTGGTCGCGGGACCCGGAGCATGGATACGAACGCGTCTGAGCGTG<br>    ACATCCACGCAGCGCCCGGGAACCTGGGTTCCCGACTATCACATGGGTAATCAAATCGCG<br>    AACATTCTTAAAGTAAGGCCGATGCTGAAGCACTATCCTAACCACGCTAGTGTGCAATGC<br>    GCGGGAAGGCGGAGAAAAAGAATCTTACGTTTATCAGCGTTTTCTTACTCCGGCGACCAA<br>    GCGAGGAGGGGCTGGTTAGCCCCCAGTCTTTAGTTACCTGGCGTCGGATATATTGTCATC<br>    CTAGAGTAACTGTCCTATCGTTGAATAACCCGTACAAGAGATACTGGGTCCTGGTAACCG<br>    TACAAGTTTTCCGACTTTGCGAGACTCTAAATCAGTACTTTGGCAAGAGGGAAGGGCAGT<br>    AAGGGGTCAGCTTTTGAACAGCGCCCGGGCCCAGGTTACGGCCTAAATTAGTGCCAGTGT<br>    CACCGTCGAGAGGGCAGTTCGACTGAAAGCCTAATAAGATACCGTTGAC<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA8.fasta<br><br>File contents:<br><br>    &gt; contig8<br>    TGGGACAGGACCAATCAATTCGCCAAGAGTCTACTTGTCCGTCCGATAGGCAAAAGTTGT<br>    TCGCTGAAATATCTCGGCCATTCAGACGAGTTTTACATAAGTACGTATGTAGAACGCGAT<br>    GATCCGGATTGGAGTGACGCCTGTCTCTTTTCGAAAGACGGAACCGCCGGCGCACAATAA<br>    GCCTGCTACCTGACTGTAGCCTCTGTATCCTCAATATCGGATACAGTCAGGCAGATTGAG<br>    ATCGCGACATATGCAAAGAGTGGGTGACTAAGGACAATGGGAGGAACTGTGGTTCTCACT<br>    TCAAAGCTTGTGTCAGGCCTGCTCAAAAGGTCAAGGAAAATTTCTACCGTTCTCACGATG<br>    ACGAGGGTGGCACTGGCACGGACCACCAATCGCTCTGAACTAGCTTGCAACCCGTCGTAC<br>    CTCTCGGTGACTCTGGGAGCAGCATAGGAGGCCTCGCTTGTGCAATCCTTAAATCGCGAG<br>    AGACATCCGGCGAAGCATGCACTAAGCCTTTCTGAAATGTGATCGGATTTGACTTTAATC<br>    CGGTCACTCTGATCTCATTTTCTAAGACTAACGATTATCGGTTCAAGTTCGTTCCTAACG<br>    AATTTTGCGGGTCGTGCTACTACGTACTAGCTTAAGGACACTACACCTACACCGTCGGTT<br>    TTTAATAGCGACTTATTATGTCCTTGCTAAAGCTCTAATCCCAGTTGTAACAGTGCATAT<br>    TTAGACAGCACATCCGTGAGTAACCGTATGTGAGCAGAATTTAGGACTGAGTCAAGGAAT<br>    GCAAATGGCAAACGGGGTGGTTCCTCGGTGGATCACAAGCGCGACCATTGAAGTTCAGTT<br>    TGTGGCACGCCCCGCGCGCTTCGTAAGGAACGATGGACGGCAGAAGTGTTGAATTTATAG<br>    CACGAGCGAGGTGTGCTGATTCCTCCGTGATTAGTGCACGGAAAATTCGAAGAGTACACT<br>    CTATTTATTTTGTGTCGCTAGATATGCAGATGCTACCGGAGTTCCGACGTTCAACGACTC<br>    CCCCAGGCGCGGAACCCGAACCCGGGACACTTACGGAAGCGTGAGATCCCGGTTGTCTTT<br>    TCAGGTGTCCAATTATGGTTTTTTGGAGGATCGTTATATATCTTGGCCGCACCTCCACTG<br>    GCCTTCGCATACAGTCTGGGACCAGCGTCGCACCGATAGCCGGCCAGGCCCGCTCTAGAT<br>    AACGGCAGATGTGAACCGCTGTAGTTACACTGTCAGTGTCACCAGGGACTAAGCAAATGG<br>    CAATTCCCAATAGGGAGGCACATTGTGAGTTTGACCGGCGTATTGTAAGCTAGCTATCTG<br>    ACCGACAGGCTTAACAACGCAATATGGATACTCTGTTCACGGTCTTTTGGAGCCAATGCC<br>    TCCATTCCCCTGGACCATGTCAGAATACAAGACGTCAGAGACCGAAAAATTGGACCTTAA<br>    ATTATTCCATGGGTCCGCTCTGGGTTTGACGCGACATGTCCAGTTCGCGCGGCGTTCTCT<br>    TTCGCGTCGAGGAGCTCTGCTTAGAATGGACGTAGCCAGAAAAAATTGAGTGGGGCCAGT<br>    ATTAGATTGTTACCGTGGTTCAAGTGAATATCGCGAGTGT<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA9.fasta<br><br>File contents:<br><br>    &gt; contig9<br>    GAGTGTTAAGAGTGCGCAGCATAAAAAAATCGCCGATGCACAAACCAGACTTTTGCGCGG<br>    AGAGCGGCAATGCCACGAGACACAACGGGGCGAGTAGCCTATCGAGCCCCGGCGCCAAGA<br>    AGCTATTCGTTTCGCATGTGATTAAGTTTCTACACTACATTGCGTAGAGGTCTAGTTATT<br>    TGTTAGCCAACGCTAGACTAGCATACCATGGTGTAATTATGGTAGACACCAAAGCGCGAA<br>    CGGAGAGCGTCCAATGTGATCACGAGCCCGGGAAGCAAAACCCGTAGATACAGATTCCGT<br>    ATTGCAAAGGGGGCACTGAATGCCCCCAATCAATCGTAAACGATTCAAGATAGTCCGAGT<br>    CTAATCGGTTGTCGTTGGTATCCCGAA</td><td class="s4" dir="ltr"></td><td class="s4" dir="ltr">Yes</td><td class="s4" dir="ltr">Yes</td><td class="s4">text</td><td class="s4" dir="ltr">SRP</td><td class="s5" dir="ltr">February 13, 2023</td><td class="s4" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s4" dir="ltr">import sys<br><br>def find_stop_codon(sequence):<br>    stop_codons = [&#39;TAG&#39;, &#39;TAA&#39;, &#39;TGA&#39;]<br>    for i in range(0, len(sequence), 3):<br>        codon = sequence[i:i+3]<br>        if codon in stop_codons:<br>            return i<br>    return -1<br><br>def get_gene_location(sequence):<br>    start_codon = &quot;ATG&quot;<br>    start_index = sequence.find(start_codon)<br>    if start_index == -1:<br>        return -1, -1<br>    end_index = find_stop_codon(sequence[start_index:])<br>    if end_index == -1:<br>        return -1, -1<br>    end_index += start_index<br>    return start_index, end_index<br><br>def main():<br>    file_name = sys.argv[1]<br>    with open(file_name) as f:<br>        lines = f.readlines()<br>        sequence = &quot;&quot;.join(lines[1:]).replace(&quot;\n&quot;, &quot;&quot;)<br>        start, stop = get_gene_location(sequence)<br>        if start == -1:<br>            print(&quot;No gene found&quot;)<br>        else:<br>            print(start)<br>            print(stop)<br><br>if __name__ == &#39;__main__&#39;:<br>    main()</td><td class="s5">5</td><td class="s5" dir="ltr">0</td><td class="s5" dir="ltr">10</td><td class="s4" dir="ltr">ChatGPT came close on each attempt but could not quite figure it out. I tried to guide it in the right direction, but it didn&#39;t quite get there. One problem was that it didn&#39;t look for the start codon in frame. So I clarified the prompt, but it didn&#39;t quite get it. On some attempts, I shortened the prompt in various ways to see if that would help.</td><td class="s4"></td><td class="s4" dir="ltr">Yes</td><td class="s4"></td><td class="s4"></td><td class="s4" dir="ltr">Yes</td><td class="s4"></td><td class="s4"></td><td class="s4" dir="ltr">Yes</td><td class="s4"></td><td class="s4"></td><td class="s4"></td></tr><tr style="height: 20px"><th id="2089702959R183" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">184</div></th><td class="s1">19 - Additional Practice</td><td class="s1">09 - Make inducible promoter - Part C</td><td class="freezebar-cell"></td><td class="s1" dir="ltr">Here is a new Python programming task.<br><br>This exercise builds on the previous ones.<br><br>Restriction enzymes are necessary to create an inducible promoter. They are used to cleave DNA into fragments at specific locations. In this case, we want to shorten the upstream sequence.<br><br>Suppose that your lab has six restriction enzymes on hand. You can see the names and sequences of these enzymes in the tab-separated-value file &quot;RestrictionEnzymes.txt&quot;. Restriction-enzyme sequences consist of DNA nucleotides (A, C, G, or T) combined with placeholders (N, K, M, R, or Y) that each represent multiple nucleotides. For example, K is a placeholder for either G or T, and N is a placeholder for A, C, G, or T. &quot;Placeholders.txt&quot; (a tab-separated-value file) lists all of the placeholders that are used. The values in the second column are separated by commas.<br><br>Please write a Python script that accepts zero arguments. Your code should use information in &quot;RestrictionEnzymes.txt&quot; and &quot;Placeholders.txt&quot; to create a dictionary in which each key is the name of a restriction enzyme and each value is a regular expression that indicates the DNA pattern that corresponds to the restriction enzyme. Your function should print the contents of this dictionary in sorted order, as shown in the expected output.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Placeholders.txt<br><br>File contents:<br><br>    Shortcut        Nucleotides<br>    N        A,C,G,T<br>    K        G,T<br>    M        A,C<br>    R        A,G<br>    Y        C,T<br><br>Below is a description of a data file that is available for testing.<br><br>File name: RestrictionEnzymes.txt<br><br>File contents:<br><br>    Name        Sequence<br>    ApoI        RAATKGCNTTY<br>    EcoRI        GAANNTTMC<br>    HgaI        GAMYCGC<br>    HinFI        GANNTCCC<br>    MspA1I        CMGCKYCCGG<br>    SbfI        CCTGCAGG</td><td class="s1" dir="ltr"></td><td class="s1" dir="ltr">Yes</td><td class="s1" dir="ltr">Yes</td><td class="s1">text</td><td class="s1" dir="ltr">SRP</td><td class="s2" dir="ltr">February 13, 2023</td><td class="s1" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s1" dir="ltr">import re<br><br>def get_placeholder_dict():<br>    placeholders = {}<br>    with open(&quot;Placeholders.txt&quot;) as f:<br>        for line in f.readlines()[1:]:<br>            placeholder, nucleotides = line.strip().split(&quot;\t&quot;)<br>            placeholders[placeholder] = nucleotides.split(&quot;,&quot;)<br>    return placeholders<br><br>def get_restriction_enzymes():<br>    restriction_enzymes = {}<br>    placeholders = get_placeholder_dict()<br>    with open(&quot;RestrictionEnzymes.txt&quot;) as f:<br>        for line in f.readlines()[1:]:<br>            name, sequence = line.strip().split(&quot;\t&quot;)<br>            pattern = sequence<br>            for placeholder, nucleotides in placeholders.items():<br>                pattern = pattern.replace(placeholder, &quot;[&quot; + &quot;&quot;.join(nucleotides) + &quot;]&quot;)<br>            restriction_enzymes[name] = re.compile(pattern)<br>    return restriction_enzymes<br><br>def main():<br>    restriction_enzymes = get_restriction_enzymes()<br>    for name, pattern in sorted(restriction_enzymes.items()):<br>        print(f&quot;{name}: {pattern.pattern}&quot;)<br><br>if __name__ == &quot;__main__&quot;:<br>    main()</td><td class="s2">1</td><td class="s2" dir="ltr">1</td><td class="s2" dir="ltr">1</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1"></td><td class="s1" dir="ltr">Yes</td><td class="s1"></td></tr><tr style="height: 20px"><th id="2089702959R184" style="height: 20px;" class="row-headers-background"><div class="row-header-wrapper" style="line-height: 20px">185</div></th><td class="s4">19 - Additional Practice</td><td class="s4">10 - Make inducible promoter - Part D</td><td class="freezebar-cell"></td><td class="s4" dir="ltr">Here is a new Python programming task.<br><br>Now the goal will be to identify any location(s) upstream of the gene where a restriction enzyme can bind. But you also want to ensure that the restriction enzyme will not bind to a location within the gene itself.<br><br>Please write a Python script that accepts one argument via &quot;sys.argv&quot;: the name of a FASTA file. As in the previous exercise, your code should use information in &quot;RestrictionEnzymes.txt&quot; and &quot;Placeholders.txt&quot; to determine a DNA pattern associated with each enzyme. Your script should read the FASTA file and extract the upstream sequence and the gene sequence. For each restriction enzyme, your code should identify any restriction enzyme that that matches the upstream sequence and does not match the  gene sequence.<br><br>Your script should print the name of any restriction enzyme that meets these criteria. It should also print the sequence from the upstream DNA that matches the restriction enzyme. If there is no matching sequence in the upstream DNA, print &quot;No match&quot; for that enzyme.<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA1.fasta<br><br>Below is the expected output:<br><br>    ApoI: AAATGGCGTTC<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA2.fasta<br><br>Below is the expected output:<br><br>    HgaI: GAACCGC<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA3.fasta<br><br>Below is the expected output:<br><br>    SbfI: CCTGCAGG<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA4.fasta<br><br>Below is the expected output:<br><br>    No match<br><br>Below is a description of a test to verify your code.<br><br>For this test, the following command will be used to execute your script:<br><br>    python code.py DNA5.fasta<br><br>Below is the expected output:<br><br>    EcoRI: GAAGGTTAC<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA1.fasta<br><br>File contents:<br><br>    &gt; contig1<br>    GGTGTCAAACTTTCGGGCCTGCGCTTCAATTTCTCTCCGCCGCCCTTTCGATCATGAATC<br>    AAATGGCGTTCCCACACCAATATGGGAGAGCTCCCAGCTGCCGTAGACACCGATTGTATT<br>    AGGTACTGTGGGGCCTCTGGTAAATTCCACTAACGGAACTAGCATAGTAGGCTAGTATAG<br>    CTCGGGGCGCAGGTATTGTTCTCTTGATAATTGGTACCACAGCGAGTGCGGAAGTAGTCT<br>    ATACGTGATGACGTTAACTAAAGACGCTCTGCAGGAGAGCAACGGTGGTATAGCGGGAGC<br>    TACGAGAAGGATGCGGACAAATGGCGTTCCAGAATCAGTGTATTTGGCGTGTAACACAAG<br>    CACCTTTCTCCCAGGTGACGCGTCGCTCGCGAAGATTTCCAAGTAGCAAA<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA2.fasta<br><br>File contents:<br><br>    &gt; contig2<br>    CTGTCTCCTATATTATGTACGGACGAAGGAATCGAACCGCTGGCTGTACACGTGCTAAGC<br>    GAGCCTCAACGTATGTTTAACATAGTTGTATCGTTGCCCTGTCAGTACCTGCGTCTTACG<br>    AAGGGCGGTCAAGTGACCTTCCGGCCCAACTTTGCAAGAAGAAACGCAATACGTACCTAT<br>    CGCGGATCTATTATCAGCGCATCCAGTTCGGCTGAGCTTGTGAACTACGCTCTAAATACG<br>    CCTGAGACTACCACGAGAGGGTTCTACACTGAACTATAGTCTCCTTTTTAATCTTGAATT<br>    TCCCAAGATCCAGAACCGCATACATTGGTTTTGT<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA3.fasta<br><br>File contents:<br><br>    &gt; contig3<br>    GCCGGATAATTCCAAGGCTACTTCCGCGAATTTCCGACCCGCTGGGCCTGCTGAGTTGGA<br>    CACCTCGCATAGATTGTGACCCCTCTATGACAGATTGGGACACCTGCAGGCTCAGGGTTA<br>    TGAACCTATAAGTGCGATAAGAGCTGGAACTTTAGGAGATATCGTATGTCAATATTTCAT<br>    GCTGAAGCCCAAATGAAACGAGGTACTGACTACGTACTGATAACACGGCTCCACCTCTAC<br>    CTGTGGATTCGCCCTCAAACGGAACCGGTGTTTGACTGCAACTGCTGGTGCATGTCCAGC<br>    TTAGAGCTTAGACACAGCAGTCCATCGCGTCCAGAGTATACCGGATCGCACCTAGGCGAA<br>    CCGCTAATTGTAGATATCGAGATAAAAGCGCATGACACAGGATCAAGGTGTAGTTAGTGT<br>    GTCGGTTGTTACTTTACGTAGACGCAAGTCGCGGTATGTAAGGCGAAGTGTGGCTTTATT<br>    CAGCAGAAAAAATGCATCGCTAAAGCCGAGATTGCGCAAAGGGTGGCAACGGAGCTCAGC<br>    GTGTATTGCGTAGACCATGATAGCCCCCGCCCAGAATTGAGGGAGCTGCCGGCATGCCAG<br>    TACGCTCTATGTCTACAGGCCGTCCTCAACTCCTTATTCTGTACGCGCGGGCTACTTCGG<br>    CCGTGGAGATATTCGGTGGGACGGTTCTACGCGACAAAATCCTAGTTAGACGACAGCACA<br>    AAATCACATCTGGAACTACCTGCAGGCCTTCACCTATCGGCCCATCACTAAACAACCTCT<br>    AAGGACCATAAGGGTAAGTCACTGGAGCTCAGAGCGCTAGCGGATCGACTCCTGTGCAAA<br>    CCGTGTCTGACTGGCGAACATGCCTTGTTAAGCGCATAGGCGGTGTAGGGGTCAACTACT<br>    CTCGCTCTCGTCCTAAGGCCGCGCCTGGTGA<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA4.fasta<br><br>File contents:<br><br>    &gt; contig4<br>    CTCGTCTGGTGTAATACCAACGGCGGCCGCCCGAGCACGGTTCCGCCAAACAGTGGCTTT<br>    TTACAGTTTTAAAGAGACGACTTGGCTGACCTATGGCCTCGGACCTTAGCGTGCTTCGTA<br>    CTCAGTAGCTTGAAGTCTCTACTTTCTTTTCCGAGTATGACTTGTTTAATGGTTCCCACC<br>    CCCCAGAGTTTTCTCCCGGTCTACACTACCGGGGGGCGCAAGGTGGCACGATGGGAGACC<br>    TCCGTATTTTGCCGTTGCGGATCAAAGGTTCAGCATAGACTATCTAAGGGGCCCGTCAGC<br>    AACTGTTCAGTTGCCCTTACGCAATTACTGCCAATTTATTAACTGTAAGTGGCTTAAGGT<br>    AACCCGGACCTCGCCACAATCCGATATCGCTACGTCGAACCTCAGTTGCGCTAGGCTAAG<br>    GGCGGGGCCTGTATTACCTATGCCCAAGGCGGGTCGGGGCTATTTTACACGGTGCTTGAA<br>    CGCCCGATATGCATACGGGTTATTAAGAAACACTTATTCGTAAGTTAGTAAGCGAGGACA<br>    GCTGTCAGCCCGTCATCTCCATTGATCAAGCAGGATCAGATGGTAAGCGCGTCCTGATTT<br>    GCTTACCTATCAGTAGTGTGGTAGCTTCCGGGTAAGGTCACGGTTAGACGAGGACCATTC<br>    ATTGGACCTTCTGTCGCGAAGTGTACGATAACCCGGGGTGCTCCGCGTGCGGTGACGTTT<br>    TCTGTTCTGCCAATAAAAAGAGCAAAAAACTCAGACTATTCGTTTTCTAGCGCGGCAGCT<br>    CGAAAAATGTGTGCTGCATGAGGGCACGCTCGGCGAGGTCACACCTATTTTCTACCGAGA<br>    TGCTTCCGCTCAGGTCCAAATCCCAACATTATTCGATCGTAGACAGTTAGACCACAAG<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA5.fasta<br><br>File contents:<br><br>    &gt; contig5<br>    CGGGAGGGGAATGGTTGGGCGCCAACAAGTGTGGCCACCACATTTCATCTAAAGATCTGC<br>    ACTGTGATTCGGTTAGGTTTGGCGGAGCCCGCTTATCATGTGGAAGGTTACTGCATCCTA<br>    GTTGAGAGGAGAGATGATACCGCGATCCCTTAATACGTTCCAACAATGCGCATCCCGTTG<br>    GCGCCAATTAATTCCCGGGAGCGTAAGTCCGGAATAGGACCGCCCAACGTACGCTGCGGG<br>    CTATTCTCGAGTTCGAAACAGTCGCTTGGCGCGGTGCTTAACGAGCTTGAACCACTGGTC<br>    CTCTTGCAAGAAGTTGCACCGGGATGCGATACAGCCAAAACTGAGAGCTAAGTGGAGCAG<br>    GCTACTTTCCCGTAACTGTACTCAATCGACAACAACACCTTAGGTCAACCGAAACGGAAC<br>    ACGCGGGACCAGCGAGAAAATTCACACCAAGAAAACCGCCTGGTATCTAGACTTGGGGAA<br>    TTCCCAACTTGCTTTGCACGGGATATCTGGGGGCGGGTAATTACAGATGAATAATGATTA<br>    ACGGACTACATAGAGTTTTCTGCGTTGGCCAATTCCTTTAGCCCTCTTGGCTTGGAGCAC<br>    GTCGGTTCATCTCGAAAACACTACTCCAAATCAGACGGGGGTGAGCAGCGCAAGGTCTAA<br>    TGGCAATTAGGGTCAAATGACGGTCCGTACCTCACTTGTTAGGACGGAAAGCCCTAGAAC<br>    ATGAAGGTTACCCGACGACTGCTTGTGCGATATGATAGTTAGACGCATTGGCCTTCGGAG<br>    GACTAATCAAGAGGGCCTTAGGATAGGAATGCTTATCAGTAAACTGCGTTATGGATTGGG<br>    CACCCTCGCATGAGTCTTGACACAGAATCCGCAAGACTCAAAGACTGAAACGACTTGACG<br>    CAACAACCACCGAACC<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA6.fasta<br><br>File contents:<br><br>    &gt; contig6<br>    TCGCAATGAGGGGAAAAATGGCGTTCTATCACTAACAAAGTCTGGATGTACACTTCAGCT<br>    GCCCATAAACGATTAGGTTTAGAACTGATCGACATCTGGGGCGGGAGTCAGGATTAATTG<br>    ACCGTTACTCTAACTCACAACGCTTTCTTCTGGGAGTCTATTCTTCCGCCTTCTAATGGC<br>    TACATGTCCCCAGACGGCTACTATCACGAGCGGTAGCAAATAGAATGTCGGTGATACTAA<br>    TATGTGTCTATGCGGAAGCAATTGTTGCTCGTGAAACTACTATGCGTTCAAATACAACAC<br>    TCAAGCGCGATGAGGTAACGTTTCTTAAAGCATGTACGCTTCCATAAAAGAACCCACCGG<br>    TTGAGGGGTTCTACACTTGCTTCCTTCCTAAATCGGCCAGAGACAGTTTTCTATCTGCTT<br>    ACGCCTTTGGATAAGCGACCCCCGCAGTAAGCTTTCTCATAGGTAAACCCCAGATGGTGA<br>    AGACTCTCGTAAGTCCACCATCCGGTTTGGTAAGGACATTCAGTGTGGTTTCCCAGAGTT<br>    CTATTACGTAGGAACTGCGGCAACTCCCGTGACAACAAGCCTCAAAATAAGACCCCAGAA<br>    GCCGACGCCCCGGTCAAACGGGCAGGCGAGAAAGTTGTTATCGTGCTTACAAACTAGCCA<br>    TGTATAGCCTAATGAAGTGCACAGTCCGCGGCTAAAGCGCCTAATACTCACGCGGAATGC<br>    TGTCGCAACAGCTGCTTAAAACCATGTTAGATAAAAGAAAGCAACTGCCATCAAGCCGAG<br>    AACCTGCCCTGCGATTGTTCGCGTCAGATGGCGTCGCGGGCAGTATAGCCTTCTCCTAGC<br>    CAGTTAATGGTTTCGTCCGCGTGCTCAGGTCACAGGCTGGTCCTGCACGAGGGTATACAG<br>    TGTCCGAAAGGTTCTATGCCGATCACGTATGGTCCATCGACTTAAGCGGTTATTTCACCC<br>    TCCAGGATAGTGATGAGTCGTTAAGCCGGAAGGGAGTGGAGGCTTCCCCCAAGAAGGGGC<br>    TACCACCCTCAGCAGTAAATAGATAGGGCTCCTCGCAAAGTGCAAGACGGCCTCCGCATG<br>    GCTGGTACTTTGCACCGTTCATCTCGTGCTCAAGTCACGGTACGGCCATCAGCCTAAGAT<br>    TAGGAGTAGCTCCGTGAAATGCTTTAACAATGACGCAAACCTCGCAAGCCTGGGATAATT<br>    GGAAGGTGATCCTAGTAGGAGACAAGAGATTACGGAGGAATTTACTCCAAACAGTAACAC<br>    TAAAGTGTAACCAAAGTCCCCTTGACCGGAAAACCTGTGTGCTTTAGAATGGTTATATAT<br>    CTTCAAAATGAGCCTCGGCAACTCTGGTGGCCAGCACGTCTGAGAAACGCCCAAGCGCAG<br>    CGATTAGAGGCATTCCGGGCGACTGTCAACTTGACGGGCATGCGCCATGGTCAGAATAGG<br>    CCGTCCCTCCGCAAATAGTATACAGCTCTTAAAATTACGGCCGTACCGGACAGTTGTTCT<br>    CATGATTTCATGGGCAGCGCTGAGTGCGTGGCTTCCCCGTTAGAAGGGTGTGCTGGTGCC<br>    CGGTTATCATTTTGAGAGTGCTCGAGATGGGGGGATTGTAATTGTTGTCTCGGAGTGTTT<br>    CTCGCCCAAGCCCAAATGGCCCGATTCGATAATGCTATTAGTCTGACATTTTGGATTTCC<br>    ATTCTGAGCCGATCGTACTGGAAAGTTCTAGATGACTCTAATGGAAAAGGCTGCTCCATG<br>    GCCAGGTCTTCTAGTCGGTAACTTAAGAGACCATGCAGAGCTTGCCCGCGGGGCTATATA<br>    GGAGATTAATATGTGTTCTGACGGTGATCTATTACCGACCTCCCTGCGTAACCTTGTTTC<br>    CTACAAAGCTTAACCCGAAACTTGATCGGACCACGCTACAGTCTGTCAGGAGATCTGAGC<br>    CCGGTGCACCTTTATCGCCGGGAACAACCGCTTACGAAGTACCTTTATCGACCAACTTTG<br>    ACCGCTGCATGATAATCCGGGAGAGGCATGCGCTGTTAATTGCGTAGTCGCCTATCTCAT<br>    GGGGAGTGCCGGGCAAAAGGCCTGCTGTGTAAAGTCGATAGTCGTATGGTCCGGAGGTCC<br>    TAGAGCTCAGCGTCCGAGTTCACCTTTGGCACGTAACTTTAACGCATGTAGAGTTTGCTC<br>    CATTGCCCCCCTAGTGCATCCTTTCGAGTAACCTCTCTAGCGTGCACGAGCTTACACAAC<br>    GATGGAACCTGGTGGATAATCAATGGGGTTAGCTTACTCGTGCTAGGGCCATCAACTTAG<br>    ATTTCAATGCCTAACAGGGCCGAGACCATGACCCTTCGAAATGGCGTTCTTGTTTTCTAC<br>    CGGCAAGTTGCGATGGCGAATGGTAACGTAAGTAACCGAATTCAGCCACATGATGGATTA<br>    TTCAGACAATCCCTCATCCGCGTACATAATTATGCACTGGGGTGAATCGAGTTGCGTAGC<br>    CAAGATTGTGTATCTGTTACAACCAATCCGCGGGCAGCTTTCTGATCTAGTTCTTTGTTG<br>    TGAACGCTGTCCGCTCACAACTTATCCCTGGTATGTCCCTTTGTGCCTCGTGATGAAATA<br>    CTGAACTGCGAACCCCAGCAAAAGGGGCGGCGCCCGGCATGCGACGGTCCTTACTGCGCT<br>    CGGGGCAAGGGTGGAACGGCCAAGTTAGTGAAGGCCATTCGCGTAGGTGTCTCCTGTGCG<br>    GCGCCATGCTCCCGCTCCACGTCAGGTCTTGCGACTACAATTCTCTTGGGGGCTTTAGAT<br>    CCGGCTTGTGGACTACCGACAGTGCGTCAGATCTCCGAGCACGACAAGTTTATCTGTTTT<br>    GGGGGGATAGCACCTTGTTGCAAGCTCCACCATGATTGGACCAGTCGTATCGGCGTTGTT<br>    TAGCACAGTTTTCGCAGGTAGGTGAAGGACCCTCAAGACAGTAATCCGGCCCGGACGCCG<br>    GCCATTTTCCCCTCGAAGGTACCGTTGCTAGCGTTGTCCTCGTACGGATAGGGTCGGTTG<br>    ACCGTTTCCCCAGATCCCCCCACATTGCATCGCCGCAATGGGTCGGTGCACGTCGAGGCA<br>    GGCGGCTAGCACCGGGTGTTTCAAATCGCAAATGTAAAGTCAGTAGCACGATGATCGAGT<br>    CCCATCCCGACCCCGTGTTGGTTGGAAAATACCACTATGAATTCACACGTTTTTTTGTTT<br>    CGCCATTTAGTGAAAAACAATTGGGAGTCGAAGAAAGGTATAGGATTGCGTTAGCGGAAA<br>    CGGCGACGTAACTTCGGATCGCTATTATGGGAATGTGGGTGAAATAAAGAGTGTGCAGCA<br>    ACACATGTTTAAAAAAGCAGGATCGATAGAGCGAGCGTGATTTAGGAGATACTAGATTGG<br>    AACCTCCCCCACAGGTCACCACGTCCTCTACCTAAGATTTATATCTCCGCATGGCCCTGA<br>    TCATGGAAG<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA7.fasta<br><br>File contents:<br><br>    &gt; contig7<br>    ACAACCGAGGTCTCGGTGGTCGCGGGACCCGGAGCATGGATACGAACGCGTCTGAGCGTG<br>    ACATCCACGCAGCGCCCGGGAACCTGGGTTCCCGACTATCACATGGGTAATCAAATCGCG<br>    AACATTCTTAAAGTAAGGCCGATGCTGAAGCACTATCCTAACCACGCTAGTGTGCAATGC<br>    GCGGGAAGGCGGAGAAAAAGAATCTTACGTTTATCAGCGTTTTCTTACTCCGGCGACCAA<br>    GCGAGGAGGGGCTGGTTAGCCCCCAGTCTTTAGTTACCTGGCGTCGGATATATTGTCATC<br>    CTAGAGTAACTGTCCTATCGTTGAATAACCCGTACAAGAGATACTGGGTCCTGGTAACCG<br>    TACAAGTTTTCCGACTTTGCGAGACTCTAAATCAGTACTTTGGCAAGAGGGAAGGGCAGT<br>    AAGGGGTCAGCTTTTGAACAGCGCCCGGGCCCAGGTTACGGCCTAAATTAGTGCCAGTGT<br>    CACCGTCGAGAGGGCAGTTCGACTGAAAGCCTAATAAGATACCGTTGAC<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA8.fasta<br><br>File contents:<br><br>    &gt; contig8<br>    TGGGACAGGACCAATCAATTCGCCAAGAGTCTACTTGTCCGTCCGATAGGCAAAAGTTGT<br>    TCGCTGAAATATCTCGGCCATTCAGACGAGTTTTACATAAGTACGTATGTAGAACGCGAT<br>    GATCCGGATTGGAGTGACGCCTGTCTCTTTTCGAAAGACGGAACCGCCGGCGCACAATAA<br>    GCCTGCTACCTGACTGTAGCCTCTGTATCCTCAATATCGGATACAGTCAGGCAGATTGAG<br>    ATCGCGACATATGCAAAGAGTGGGTGACTAAGGACAATGGGAGGAACTGTGGTTCTCACT<br>    TCAAAGCTTGTGTCAGGCCTGCTCAAAAGGTCAAGGAAAATTTCTACCGTTCTCACGATG<br>    ACGAGGGTGGCACTGGCACGGACCACCAATCGCTCTGAACTAGCTTGCAACCCGTCGTAC<br>    CTCTCGGTGACTCTGGGAGCAGCATAGGAGGCCTCGCTTGTGCAATCCTTAAATCGCGAG<br>    AGACATCCGGCGAAGCATGCACTAAGCCTTTCTGAAATGTGATCGGATTTGACTTTAATC<br>    CGGTCACTCTGATCTCATTTTCTAAGACTAACGATTATCGGTTCAAGTTCGTTCCTAACG<br>    AATTTTGCGGGTCGTGCTACTACGTACTAGCTTAAGGACACTACACCTACACCGTCGGTT<br>    TTTAATAGCGACTTATTATGTCCTTGCTAAAGCTCTAATCCCAGTTGTAACAGTGCATAT<br>    TTAGACAGCACATCCGTGAGTAACCGTATGTGAGCAGAATTTAGGACTGAGTCAAGGAAT<br>    GCAAATGGCAAACGGGGTGGTTCCTCGGTGGATCACAAGCGCGACCATTGAAGTTCAGTT<br>    TGTGGCACGCCCCGCGCGCTTCGTAAGGAACGATGGACGGCAGAAGTGTTGAATTTATAG<br>    CACGAGCGAGGTGTGCTGATTCCTCCGTGATTAGTGCACGGAAAATTCGAAGAGTACACT<br>    CTATTTATTTTGTGTCGCTAGATATGCAGATGCTACCGGAGTTCCGACGTTCAACGACTC<br>    CCCCAGGCGCGGAACCCGAACCCGGGACACTTACGGAAGCGTGAGATCCCGGTTGTCTTT<br>    TCAGGTGTCCAATTATGGTTTTTTGGAGGATCGTTATATATCTTGGCCGCACCTCCACTG<br>    GCCTTCGCATACAGTCTGGGACCAGCGTCGCACCGATAGCCGGCCAGGCCCGCTCTAGAT<br>    AACGGCAGATGTGAACCGCTGTAGTTACACTGTCAGTGTCACCAGGGACTAAGCAAATGG<br>    CAATTCCCAATAGGGAGGCACATTGTGAGTTTGACCGGCGTATTGTAAGCTAGCTATCTG<br>    ACCGACAGGCTTAACAACGCAATATGGATACTCTGTTCACGGTCTTTTGGAGCCAATGCC<br>    TCCATTCCCCTGGACCATGTCAGAATACAAGACGTCAGAGACCGAAAAATTGGACCTTAA<br>    ATTATTCCATGGGTCCGCTCTGGGTTTGACGCGACATGTCCAGTTCGCGCGGCGTTCTCT<br>    TTCGCGTCGAGGAGCTCTGCTTAGAATGGACGTAGCCAGAAAAAATTGAGTGGGGCCAGT<br>    ATTAGATTGTTACCGTGGTTCAAGTGAATATCGCGAGTGT<br><br>Below is a description of a data file that is available for testing.<br><br>File name: DNA9.fasta<br><br>File contents:<br><br>    &gt; contig9<br>    GAGTGTTAAGAGTGCGCAGCATAAAAAAATCGCCGATGCACAAACCAGACTTTTGCGCGG<br>    AGAGCGGCAATGCCACGAGACACAACGGGGCGAGTAGCCTATCGAGCCCCGGCGCCAAGA<br>    AGCTATTCGTTTCGCATGTGATTAAGTTTCTACACTACATTGCGTAGAGGTCTAGTTATT<br>    TGTTAGCCAACGCTAGACTAGCATACCATGGTGTAATTATGGTAGACACCAAAGCGCGAA<br>    CGGAGAGCGTCCAATGTGATCACGAGCCCGGGAAGCAAAACCCGTAGATACAGATTCCGT<br>    ATTGCAAAGGGGGCACTGAATGCCCCCAATCAATCGTAAACGATTCAAGATAGTCCGAGT<br>    CTAATCGGTTGTCGTTGGTATCCCGAA<br><br>Below is a description of a data file that is available for testing.<br><br>File name: Placeholders.txt<br><br>File contents:<br><br>    Shortcut        Nucleotides<br>    N        A,C,G,T<br>    K        G,T<br>    M        A,C<br>    R        A,G<br>    Y        C,T<br><br>Below is a description of a data file that is available for testing.<br><br>File name: RestrictionEnzymes.txt<br><br>File contents:<br><br>    Name        Sequence<br>    ApoI        RAATKGCNTTY<br>    EcoRI        GAANNTTMC<br>    HgaI        GAMYCGC<br>    HinFI        GANNTCCC<br>    MspA1I        CMGCKYCCGG<br>    SbfI        CCTGCAGG</td><td class="s4" dir="ltr"></td><td class="s4" dir="ltr">Yes</td><td class="s4" dir="ltr">Yes</td><td class="s4">text</td><td class="s4" dir="ltr">SRP</td><td class="s5" dir="ltr">February 13, 2023</td><td class="s4" dir="ltr">Jan 30 Version Free Research Preview</td><td class="s4" dir="ltr">import sys<br>import re<br><br>def translate_sequence(seq):<br>    # Function to translate the sequence in the RestrictionEnzymes.txt file<br>    seq = seq.replace(&quot;R&quot;, &quot;[AG]&quot;).replace(&quot;Y&quot;, &quot;[CT]&quot;).replace(&quot;M&quot;, &quot;[AC]&quot;).replace(&quot;K&quot;, &quot;[GT]&quot;).replace(&quot;N&quot;, &quot;[ACGT]&quot;)<br>    return seq<br><br>def find_upstream_match(dna_seq, gene_seq, re_seq):<br>    # Function to find the matching upstream sequence and gene sequence<br>    upstream_match = re.search(re_seq, dna_seq)<br>    gene_match = re.search(re_seq, gene_seq)<br>    if upstream_match and not gene_match:<br>        return upstream_match.group()<br>    return None<br><br># Read the Placeholders.txt file<br>with open(&quot;Placeholders.txt&quot;, &quot;r&quot;) as f:<br>    placeholders = {}<br>    for line in f:<br>        if not line.startswith(&quot;Shortcut&quot;):<br>            shortcut, nucleotides = line.strip().split(&quot;\t&quot;)<br>            placeholders[shortcut] = nucleotides<br><br># Read the RestrictionEnzymes.txt file<br>with open(&quot;RestrictionEnzymes.txt&quot;, &quot;r&quot;) as f:<br>    restriction_enzymes = {}<br>    for line in f:<br>        if not line.startswith(&quot;Name&quot;):<br>            name, sequence = line.strip().split(&quot;\t&quot;)<br>            for k, v in placeholders.items():<br>                sequence = sequence.replace(k, v)<br>            restriction_enzymes[name] = translate_sequence(sequence)<br><br># Read the FASTA file<br>with open(sys.argv[1], &quot;r&quot;) as f:<br>    dna_seq = &quot;&quot;<br>    for line in f:<br>        if not line.startswith(&quot;&gt;&quot;):<br>            dna_seq += line.strip()<br>    for name, re_seq in restriction_enzymes.items():<br>        # Extract the gene sequence<br>        gene_start = dna_seq.index(&quot;AAATGGCGTTCCCACACCA&quot;)<br>        gene_end = gene_start + len(&quot;AAATGGCGTTCCCACACCA&quot;)<br>        gene_seq = dna_seq[gene_start:gene_end]<br>        # Find the matching upstream sequence<br>        upstream_seq = dna_seq[:gene_start]<br>        match = find_upstream_match(upstream_seq, gene_seq, re_seq)<br>        if match:<br>            print(&quot;{}: {}&quot;.format(name, match))<br>        else:<br>            print(&quot;{}: No match&quot;.format(name))</td><td class="s5">5</td><td class="s5" dir="ltr">0</td><td class="s5" dir="ltr">10</td><td class="s4" dir="ltr">ChatGPT never really figured this one out. Sometimes the code gave me errors. One time it was trying to parse the wrong file for one of the steps. Other times it was parsing files incorrectly. Occasionally, it passed 1-2 of the tests by accident. The final solution is somewhat close, but still far from an effective solution.</td><td class="s4"></td><td class="s4" dir="ltr">Yes</td><td class="s4" dir="ltr">Yes</td><td class="s4"></td><td class="s4" dir="ltr">Yes</td><td class="s4"></td><td class="s4"></td><td class="s4"></td><td class="s4"></td><td class="s4" dir="ltr">Yes</td><td class="s4"></td></tr></tbody></table></div>